<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-02-01" />

<title>DS108 ID-seq analysis: Comparisons (clean)</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapters on DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Paper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="y_paper_main.html">Main</a>
    </li>
    <li>
      <a href="z_paper_suppl.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteins ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="t_proteins_main.html.html">Main</a>
    </li>
    <li>
      <a href="u_proteins_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    States ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="v_states_main.html.html">Main</a>
    </li>
    <li>
      <a href="w_states_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
    <li>
      <a href="c_flow_vis.html">Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="DS108_p_seq_comparisons.html">DS108 clean</a>
    </li>
    <li>
      <a href="DS108_q_seq_thresholds.html">DS108 thresholds</a>
    </li>
    <li>
      <a href="DS108_r_seq_pca.html">DS108 PCA</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">DS108 ID-seq analysis: Comparisons
(clean)</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-02-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapters_DLBCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241022code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241022code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241022)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong94c226d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 94c226d
</a>
</p>
</div>
<div id="strongRepositoryversionstrong94c226d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 94c226d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/AL006_InhibCombiTest/
    Untracked:  data/DS102_StimTest/
    Untracked:  data/DS103_ViabilityTest/
    Untracked:  data/DS104_StimInhibTest/
    Untracked:  data/DS108_StimInhibIDseq/
    Untracked:  data/DS113_InhibCombiIDseq/
    Untracked:  output/DS102_StimTest/
    Untracked:  output/DS103_ViabilityTest/
    Untracked:  output/DS104_StimInhibTest/
    Untracked:  output/DS108_StimInhibIDseq/
    Untracked:  output/DS113_InhibCombiIDseq/
    Untracked:  output/figures/
    Untracked:  output/network_visual_DLBCL/
    Untracked:  output/network_visual_paper/

Unstaged changes:
    Deleted:    analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/DS108_p_seq_comparisons.Rmd</code>) and HTML
(<code>docs/DS108_p_seq_comparisons.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
6559698
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Paper figs + analysis files for ch figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)

textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
                  axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
                  axis.title = element_text(colour = &quot;black&quot;, size = 12), 
                  legend.title = element_text(colour = &quot;black&quot;, size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
                  strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
)

# textsize_small &lt;- theme(text = element_text(size = 7, family = &quot;sans&quot;, colour = &quot;black&quot;),
#                         plot.title = element_text(size = 8)
# )
textsize_small &lt;- theme(axis.text.x = element_text(colour = &quot;black&quot;, size = 9),
                        axis.text.y = element_text(colour = &quot;black&quot;, size = 9),
                        axis.title = element_text(colour = &quot;black&quot;, size = 10),
                        legend.text = element_text(colour = &quot;black&quot;, size = 9),
                        title = element_text(color = &quot;black&quot;, size = 10),
                        strip.text.x = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;),
                        strip.text.y = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;)
)

colors_dark9 &lt;- c(&quot;#4daf4a&quot;, &quot;#984ea3&quot;, &quot;#377eb8&quot;, &quot;#ff7f00&quot;, &quot;#f781bf&quot;, &quot;#ffff33&quot;, &quot;#e41a1c&quot;, &quot;#a65628&quot;, &quot;#999999&quot;)
colors_light12 &lt;- c(&quot;#8dd3c7&quot;,&quot;#ffffb3&quot;,&quot;#bebada&quot;,&quot;#fb8072&quot;,&quot;#80b1d3&quot;,&quot;#fdb462&quot;,&quot;#b3de69&quot;,&quot;#fccde5&quot;,&quot;#d9d9d9&quot;,&quot;#bc80bd&quot;,&quot;#ccebc5&quot;,&quot;#ffed6f&quot;)
colors_paired10 &lt;- c(&quot;#a6cee3&quot;, &quot;#1f78b4&quot;, &quot;#b2df8a&quot;, &quot;#33a02c&quot;, &quot;#fdbf6f&quot;, &quot;#ff7f00&quot;, &quot;#cab2d6&quot;, &quot;#6a3d9a&quot;, &quot;#fb9a99&quot;, &quot;#e31a1c&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)

colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

colors_proteins &lt;- c(&quot;pCD79a (Y182)&quot; = &quot;#892BE1&quot;, &quot;pSYK (Y525/Y526)&quot; = &quot;#1E8FFF&quot;, &quot;pPLCy2 (Y759)&quot; = &quot;#20B1A9&quot;)
colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#41ab5d&quot;, &quot;iBTK&quot; = &quot;#4292c6&quot;, &quot;iPI3Kd&quot; = &quot;#807dba&quot;, &quot;iNFkB&quot; = &quot;#f16913&quot;)
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#525252&quot;, &quot;aIg+H2O2&quot; = &quot;#fdae6b&quot;)
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#df65b0&quot;, &quot;OCI-Ly8&quot; = &quot;#41b6c4&quot;)</code></pre>
<pre class="r"><code>plates_DS108 &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;)
cells_DS108 &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
stim_DS108 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
inhib_DS108 &lt;- c(&quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)
inhib_conc_DS108 &lt;- c(&quot;0 uM (nostain)&quot;, &quot;0 uM (PBS)&quot;, &quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
inhib_conc_label &lt;- c(&quot;0 uM\n(nostain)&quot;, &quot;0 uM\n(PBS)&quot;, &quot;0 uM\n(DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
inhib_full_DS108 &lt;- c(&quot;0 uM iSYK (nostain)&quot;, &quot;0 uM iSYK (PBS)&quot;, &quot;0 uM iSYK (DMSO)&quot;, &quot;0.1 uM iSYK&quot;, &quot;1 uM iSYK&quot;, &quot;10 uM iSYK&quot;, &quot;100 uM iSYK&quot;, 
                      &quot;0 uM iBTK (nostain)&quot;, &quot;0 uM iBTK (PBS)&quot;, &quot;0 uM iBTK (DMSO)&quot;, &quot;0.1 uM iBTK&quot;, &quot;1 uM iBTK&quot;, &quot;10 uM iBTK&quot;, &quot;100 uM iBTK&quot;, 
                      &quot;0 uM iPI3Kd (nostain)&quot;, &quot;0 uM iPI3Kd (PBS)&quot;, &quot;0 uM iPI3Kd (DMSO)&quot;, &quot;0.1 uM iPI3Kd&quot;, &quot;1 uM iPI3Kd&quot;, &quot;10 uM iPI3Kd&quot;, &quot;100 uM iPI3Kd&quot;, 
                      &quot;0 uM iNFkB (nostain)&quot;, &quot;0 uM iNFkB (PBS)&quot;, &quot;0 uM iNFkB (DMSO)&quot;, &quot;0.1 uM iNFkB&quot;, &quot;1 uM iNFkB&quot;, &quot;10 uM iNFkB&quot;, &quot;100 uM iNFkB&quot;)</code></pre>
</div>
<div id="data-qc" class="section level2">
<h2>Data &amp; QC</h2>
<div id="raw-data" class="section level4">
<h4>Raw data</h4>
<pre class="r"><code># Load sparse matrix format all data
counts_PlateA &lt;- read_count_output(&quot;data/DS108_StimInhibIDseq/counts/PlateA/featurecounts&quot;, name = &quot;featurecounts&quot;)
counts_PlateB &lt;- read_count_output(&quot;data/DS108_StimInhibIDseq/counts/PlateB/featurecounts&quot;, name = &quot;featurecounts&quot;)
counts_PlateC &lt;- read_count_output(&quot;data/DS108_StimInhibIDseq/counts/PlateC/featurecounts&quot;, name = &quot;featurecounts&quot;)
counts_PlateD &lt;- read_count_output(&quot;data/DS108_StimInhibIDseq/counts/PlateD/featurecounts&quot;, name = &quot;featurecounts&quot;)

meta_wellID_plateA &lt;- read_csv(&quot;data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateA_HBL1pbs.csv&quot;)
meta_wellID_plateB &lt;- read_csv(&quot;data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateB_HBL1stim.csv&quot;)
meta_wellID_plateC &lt;- read_csv(&quot;data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateC_Ly8pbs.csv&quot;)
meta_wellID_plateD &lt;- read_csv(&quot;data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateD_Ly8stim.csv&quot;)
meta_wellID &lt;- list(meta_wellID_plateA, meta_wellID_plateB, meta_wellID_plateC, meta_wellID_plateD) %&gt;% purrr::reduce(full_join) %&gt;%
  mutate(plate_well = paste0(plate, &quot;_&quot;, well))

meta_Abs &lt;- read_excel(&quot;data/DS108_StimInhibIDseq/config/BulkIDseq_Ab_info_seq_analysis_202212.xlsx&quot;)

# Reshape counts in data table format
# Plate A
counts_plateA &lt;- data.frame(counts_PlateA) %&gt;% 
  mutate(barcode_name = rownames(counts_PlateA)) %&gt;% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %&gt;%
  gather(&quot;well_BC_seq&quot;, &quot;counts&quot;, 2:c(ncol(counts_PlateA)+1)) %&gt;%
  filter(counts &gt;= 1) %&gt;% # Remove undetected proteins counts
  left_join(meta_Abs) %&gt;%
  unique() %&gt;%
  # filter(!is.na(target_nospace)) %&gt;% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateA, plate == &quot;A&quot;)) %&gt;% # Add sample metadata
  filter(!is.na(experiment)) %&gt;%
  arrange(plate, well)

# Plate B
counts_plateB &lt;- data.frame(counts_PlateB) %&gt;% 
  mutate(barcode_name = rownames(counts_PlateB)) %&gt;% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %&gt;%
  gather(&quot;well_BC_seq&quot;, &quot;counts&quot;, 2:c(ncol(counts_PlateB)+1)) %&gt;%
  filter(counts &gt;= 1) %&gt;% # Remove undetected proteins counts
  left_join(meta_Abs) %&gt;%
  unique() %&gt;%
  # filter(!is.na(target_nospace)) %&gt;% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateB, plate == &quot;B&quot;)) %&gt;% # Add sample metadata
  filter(!is.na(experiment)) %&gt;%
  arrange(plate, well)

# Plate C
counts_plateC &lt;- data.frame(counts_PlateC) %&gt;% 
  mutate(barcode_name = rownames(counts_PlateC)) %&gt;% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %&gt;%
  gather(&quot;well_BC_seq&quot;, &quot;counts&quot;, 2:c(ncol(counts_PlateC)+1)) %&gt;%
  filter(counts &gt;= 1) %&gt;% # Remove undetected proteins counts
  left_join(meta_Abs) %&gt;%
  unique() %&gt;%
  # filter(!is.na(target_nospace)) %&gt;% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateC, plate == &quot;C&quot;)) %&gt;% # Add sample metadata
  filter(!is.na(experiment)) %&gt;%
  arrange(plate, well)

# Plate D
counts_plateD &lt;- data.frame(counts_PlateD) %&gt;% 
  mutate(barcode_name = rownames(counts_PlateD)) %&gt;% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %&gt;%
  gather(&quot;well_BC_seq&quot;, &quot;counts&quot;, 2:c(ncol(counts_PlateD)+1)) %&gt;%
  filter(counts &gt;= 1) %&gt;% # Remove undetected proteins counts
  left_join(meta_Abs) %&gt;%
  unique() %&gt;%
  # filter(!is.na(target_nospace)) %&gt;% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateD, plate == &quot;D&quot;)) %&gt;% # Add sample metadata
  filter(!is.na(experiment)) %&gt;%
  arrange(plate, well)

counts_dtbl &lt;- list(counts_plateA, counts_plateB, counts_plateC, counts_plateD) %&gt;% purrr::reduce(full_join)

counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(plate_well = paste0(plate, &quot;_&quot;, well), 
         stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhibitor = factor(inhibitor, inhib_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108),
         inhib_conc_text = factor(inhib_conc_text, levels = inhib_full_DS108))

# Calculate per-sample properties
data_properties &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount = sum(counts), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108))</code></pre>
<p><em><strong>Table</strong>: Unfiltered data-table properties loaded
into R</em></p>
<table>
<colgroup>
<col width="25%" />
<col width="37%" />
<col width="37%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>nCols (#samples)</th>
<th>nRows (# barcodes)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Plate A: HBL1 PBS</td>
<td>95</td>
<td>300</td>
</tr>
<tr class="even">
<td>Plate B: HBL1 aIg+H2O2</td>
<td>95</td>
<td>300</td>
</tr>
<tr class="odd">
<td>Plate C: OCI-Ly8 PBS</td>
<td>95</td>
<td>300</td>
</tr>
<tr class="even">
<td>Plate D: OCI-Ly8 aIg+H2O2</td>
<td>95</td>
<td>300</td>
</tr>
</tbody>
</table>
<p>The full panel of mapped barcodes and 96 wells are present in the
data table. First have a look at the total counts and number of detected
protein barcodes per sample: There are a lot more proteins detected than
put in the panel, due to barcode contamination and cross-ver between
wells. The nostain samples have relatively low counts and proteins
detected.</p>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(data_properties, aes(log10(nCount), nProt, color = factor(inhibitor, levels = inhib_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) +
  scale_color_manual(values = colors_inhib, name = &quot;Inhibitor&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation total counts and number of detected barcodes&quot;, 
       x = expression(&quot;Total counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_count_prot-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_prot-1">
Past versions of fig_count_prot-1.png
</button>
</p>
<div id="fig-fig_count_prot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(data_properties, aes(log10(nCount), nProt, color = factor(inhib_conc, levels = inhib_conc_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) + 
  scale_color_manual(values = colors_blue9[2:9], name = &quot;Inhibitor conc&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation total counts and number of detected barcodes&quot;, 
       x = expression(&quot;Total counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_count_prot-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_prot-2">
Past versions of fig_count_prot-2.png
</button>
</p>
<div id="fig-fig_count_prot-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Correlation between total counts and
number of detected protein barcodes. We observe many more detected
proteins than present in the staining.</em></p>
<p>As we have a 96-wells plate, we can also have a look at total counts
per plate detected:</p>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 6)

rng &lt;- range(log10(data_properties$nCount))

# for (this_plate in plates_DS108) {
#   plot &lt;- raw_map(data = log10(subset(data_properties, plate == this_plate)$nCount), well = subset(data_properties, plate == this_plate)$well, plate = 96, size = 7) +
#     scale_fill_viridis(expression(&quot;Total counts Log&quot;[10]), limits = c(rng[1], rng[2])) +
#     labs(title = paste(&quot;Plate&quot;, this_plate))
#   theme_bw()
#   
#   print(plot)
# }

plot &lt;- raw_grid(data = log10(data_properties$nCount), well = data_properties$well, plate_id = data_properties$plate, plate = 96, size = 6) +
  scale_fill_viridis(expression(&quot;Total counts Log&quot;[10]), limits = c(rng[1], rng[2])) +
  theme_bw()

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_plate_totalUMI-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_plate_totalUMI-1">
Past versions of fig_plate_totalUMI-1.png
</button>
</p>
<div id="fig-fig_plate_totalUMI-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Plate overviews of total counts. Row A
are no-stain samples.</em></p>
</div>
<div id="barcode-contamination-comp" class="section level4">
<h4>Barcode contamination comp</h4>
<p>Several barcodes are contaminated with &lt;2.5% of other barcodes
(all barcodes were sequenced individually to determine the
contamination). As this can skew the results, especially for lowly
present barcodes that are contaminated with highly present barcodes, we
will compensate for this.</p>
<pre class="r"><code>comp_matrix &lt;- read_csv(&quot;data/DS108_StimInhibIDseq/BC_compensation_matrix.csv&quot;)

data_comp &lt;- counts_dtbl %&gt;%
  # filter(well == &quot;A01&quot;) %&gt;%
  dplyr::select(plate, well, barcode_name, counts) %&gt;%
  left_join(comp_matrix, by = join_by(barcode_name == BC_to_correct)) #, multiple = &quot;all&quot;

# Get counts of OG BC
data_comp_OG &lt;- data_comp %&gt;%
  # filter(well == &quot;A01&quot;) %&gt;%
  filter(barcode_name == OG_BC) %&gt;%
  dplyr::select(plate, well, OG_BC, counts) %&gt;%
  dplyr::rename(counts_OG_BC = counts)

# Calculate false reads
data_false &lt;- left_join(data_comp, data_comp_OG) %&gt;% 
  mutate(false_counts = (counts_OG_BC / percent_OG_BC) * percent_BC_to_correct) %&gt;%
  filter(barcode_name != OG_BC) %&gt;%
  group_by(plate, well, barcode_name, counts) %&gt;%
  summarize(total_false_counts = sum(false_counts, na.rm = TRUE))

counts_dtbl &lt;- counts_dtbl %&gt;%
  # filter(well == &quot;A01&quot;) %&gt;%
  left_join(data_false) %&gt;% 
  mutate(total_false_counts = replace_na(total_false_counts, 0), 
         true_counts = as.integer(counts - total_false_counts), 
         percent_true = (true_counts / counts) * 100) %&gt;%
  filter(!is.na(target_nospace)) # Remove barcodes not in Ab metadata (not conjugated/added in panel)

compensation &lt;- counts_dtbl %&gt;%
  dplyr::select(plate, well, barcode_name, counts, total_false_counts, true_counts, percent_true, target_nospace)

# Calculate per-sample properties
data_properties_true &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount = sum(true_counts), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(counts_dtbl, aes(log10(counts), log10(true_counts), color = factor(inhibitor, levels = inhib_DS108))) +
  geom_point(alpha = 1, size = 0.8) +
  geom_abline(intercept = 0, slope = 1, linetype = &quot;dashed&quot;) + 
  facet_grid(cell_line~stimulus) +
  scale_color_manual(values = colors_inhib, name = &quot;Inhibitor&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation raw counts and compensated counts&quot;,
       x = expression(&quot;Raw counts Log&quot;[10]),
       y = expression(&quot;Compensated counts Log&quot;[10])) +
  textsize_small</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_count_raw_comp-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_raw_comp-1">
Past versions of fig_count_raw_comp-1.png
</button>
</p>
<div id="fig-fig_count_raw_comp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(data_properties_true, aes(log10(nCount), nProt, color = factor(inhib_conc, levels = inhib_conc_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) + # stimulus, staining_cells, staining_conc, staining_volume
  scale_color_manual(values = colors_blue9[2:9], name = &quot;Inhibitor conc&quot;) +
  # scale_color_binned(type = &quot;viridis&quot;, name = &quot;Staining conc (ug/mL)&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation total counts and number of detected barcodes&quot;, 
       x = expression(&quot;Total counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_count_raw_comp-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_raw_comp-2">
Past versions of fig_count_raw_comp-2.png
</button>
</p>
<div id="fig-fig_count_raw_comp-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Correlation between raw counts and
compensated counts, and between compensated counts and detected number
of barcodes. After compensation, all samples (except 2 nostain samples)
have 107 detected proteins (the number of Ab in the panel).</em></p>
</div>
<div id="count-normalisation" class="section level4">
<h4>Count normalisation</h4>
<p>To remove variation between replicates, the data is normalised per
cell line and per target protein with a geometric mean scaling factor:
First, the geometric mean of each target is calculated separately for
each cell line, and then the median of the protein scaling factors is
calculated separately for each sample.</p>
<pre class="r"><code>scaling_factors &lt;- counts_dtbl %&gt;%
  filter(staining == &quot;yes&quot;) %&gt;%
  # filter() %&gt;% #target_nospace == &quot;GAPDH&quot; # modification == &quot;no&quot;
  ungroup() %&gt;%
  group_by(cell_line, target_nospace) %&gt;%
  mutate(scaling_factor = counts / median(counts),
         scaling_factor_geo = counts / exp(mean(log((counts)))), 
         true_scaling_factor = true_counts / median(true_counts),
         true_scaling_factor_geo = true_counts / exp(mean(log((true_counts))))) %&gt;%
  ungroup() %&gt;%
  group_by(cell_line, descript_rep) %&gt;%
  summarize(scaling_factor = median(scaling_factor), .groups = &quot;keep&quot;,
            scaling_factor_geo = median(scaling_factor_geo), 
            true_scaling_factor = median(true_scaling_factor),
            true_scaling_factor_geo = median(true_scaling_factor_geo, na.rm = TRUE))

counts_dtbl &lt;- counts_dtbl %&gt;%
  filter(staining == &quot;yes&quot;) %&gt;%
  left_join(scaling_factors) %&gt;%
  ungroup() %&gt;%
  mutate(counts_norm = counts / scaling_factor,
         counts_norm_geo = counts / scaling_factor_geo, 
         true_counts_norm = true_counts / true_scaling_factor, 
         true_counts_norm_geo = true_counts / true_scaling_factor_geo)

counts_dtbl &lt;- filter(counts_dtbl, !is.na(target_nospace))

# Calculate per-sample properties
data_properties_norm &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount_norm = sum(counts_norm_geo), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108))

data_properties_norm_true &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount_norm = sum(true_counts_norm_geo), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(counts_dtbl, aes(log10(counts), log10(true_counts_norm_geo), color = factor(inhibitor, levels = inhib_DS108))) +
  geom_point(alpha = 1, size = 0.8) +
  geom_abline(intercept = 0, slope = 1, linetype = &quot;dashed&quot;) + 
  facet_grid(cell_line~stimulus) +
  scale_color_manual(values = colors_inhib, name = &quot;Inhibitor&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation raw counts and normalised counts&quot;,
       x = expression(&quot;Raw counts Log&quot;[10]),
       y = expression(&quot;Normalised counts Log&quot;[10])) +
  textsize_small</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_count_norm-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_norm-1">
Past versions of fig_count_norm-1.png
</button>
</p>
<div id="fig-fig_count_norm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(data_properties_norm_true, aes(log10(nCount_norm), nProt, color = factor(inhib_conc, levels = inhib_conc_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) + 
  scale_color_manual(values = colors_blue9[2:9], name = &quot;Inhibitor conc&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation normalised counts and number of detected barcodes&quot;, 
       x = expression(&quot;Normalised counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_count_norm-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_norm-2">
Past versions of fig_count_norm-2.png
</button>
</p>
<div id="fig-fig_count_norm-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Correlation between raw counts and
compensated counts.</em></p>
</div>
<div id="data-wrangling" class="section level4">
<h4>Data wrangling</h4>
<p>Calculate the following:</p>
<ul>
<li><p>Mean + sd for replicates (true_counts +
true_counts_norm_geo)</p></li>
<li><p>Manual fold changes of all samples compared to the mean of 0 uM
(DMSO) + PBS samples</p></li>
</ul>
<p><em>% phosphoprotein of total protein (only for proteins with both
total + phospho targets)</em></p>
<pre class="r"><code># Calculate mean + sd of true_counts and true_counts_norm_geo
mean_dtbl &lt;- counts_dtbl %&gt;%
  dplyr::group_by(description, target_nospace) %&gt;%
  summarise(
    counts_sd = sd(true_counts, na.rm = T),
    counts = mean(true_counts, na.rm = T), 
    counts_norm_sd = sd(true_counts_norm_geo, na.rm = T),
    counts_norm = mean(true_counts_norm_geo, na.rm = T)
    )

# Add metadata
mean_dtbl &lt;- mean_dtbl %&gt;%
  left_join(dplyr::select(counts_dtbl,
                   c(barcode_name, DS_Ab_nr, target, modification, epitope, marker_cell, marker_pathway, marker_pathway_detail, isotype, DS_barcode_nr, DS_barcode_seq, target_nospace, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, staining, description))) %&gt;%
  distinct()</code></pre>
<pre class="r"><code>prot &lt;- &quot;BTK_Y551&quot;

# Raw (compensated) counts
plot1 &lt;- ggplot(subset(mean_dtbl, staining == &quot;yes&quot; &amp; cell_line == &quot;HBL1&quot; &amp; inhibitor == &quot;iSYK&quot; &amp; target_nospace == prot)) +
  geom_col(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = counts, 
      group = stimulus,
      fill = stimulus),
      position = &quot;dodge2&quot;) + 
  geom_errorbar(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      ymin = counts - counts_sd,
      ymax = counts + counts_sd,
      group = stimulus),
      position = position_dodge2(width = 0.9),
      width = 0.9,
    ) +
  facet_wrap(vars(factor(cell_line, labels = &quot;Raw (compensated)&quot;)), scales = &quot;free_y&quot;) +
  theme_bw() +
  # scale_y_log10() +
  scale_fill_manual(values = colors_stim) +
  labs(x = &quot;&quot;, y = &quot;&quot;, title = prot) +
  textsize_small

# Normalised counts
plot2 &lt;- ggplot(subset(mean_dtbl, staining == &quot;yes&quot; &amp; cell_line == &quot;HBL1&quot; &amp; inhibitor == &quot;iSYK&quot; &amp; target_nospace == prot)) +
  geom_col(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = counts_norm, 
      group = stimulus,
      fill = stimulus),
      position = &quot;dodge2&quot;) + 
  geom_errorbar(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      ymin = counts_norm - counts_norm_sd,
      ymax = counts_norm + counts_norm_sd,
      group = stimulus),
      position = position_dodge2(width = 0.9),
      width = 0.9,
    ) +
  facet_wrap(vars(factor(cell_line, labels = &quot;Normalised&quot;)), scales = &quot;free_y&quot;) +
  theme_bw() +
  # scale_y_log10() +
  scale_fill_manual(values = colors_stim) +
  labs(x = &quot;&quot;, y = &quot;&quot;, title = prot) +
  textsize_small

plot_grid(plot1, plot2, ncol = 2)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_wrangle1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_wrangle1-1">
Past versions of fig_wrangle1-1.png
</button>
</p>
<div id="fig-fig_wrangle1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Mean signal (+ sd) of replicates for BTK
(Y551) (HBL1 cells, iSYK inhibitor). Raw counts on the left, normalised
counts on the right. Normalisation removes the variation between
replicates but not the effect of iSYK concentration.</em></p>
<pre class="r"><code># Manually calculate the fold change of true_counts and true_counts_norm_geo
# Use the mean of 0 uM (DMSO) + PBS as baseline
counts_FC0 &lt;- counts_dtbl %&gt;%
  filter(stimulus == &quot;PBS&quot; &amp; inhib_conc == &quot;0 uM (DMSO)&quot;) %&gt;%
  group_by(cell_line, target_nospace) %&gt;%
  summarise(counts_0 = mean(true_counts), 
            counts_norm_0 = mean(true_counts_norm_geo))

counts_dtbl &lt;- left_join(counts_dtbl, counts_FC0) %&gt;%
  mutate(man_FC_counts = (true_counts / counts_0), 
         man_log2FC_counts = log2(man_FC_counts), 
         man_FC_norm = (true_counts_norm_geo / counts_norm_0), 
         man_log2FC_norm = log2(man_FC_norm))

# Calculate mean + sd of manual fold changes of replicates
mean_FC &lt;- counts_dtbl %&gt;%
  dplyr::group_by(description, target_nospace) %&gt;%
  summarise(
    FC_counts_sd = sd(man_FC_counts, na.rm = T),
    FC_counts = mean(man_FC_counts, na.rm = T), 
    log2FC_counts_sd = sd(man_log2FC_counts, na.rm = T), 
    log2FC_counts = mean(man_log2FC_counts, na.rm = T), 
    FC_norm_sd = sd(man_FC_norm, na.rm = T),
    FC_norm = mean(man_FC_norm, na.rm = T), 
    log2FC_norm_sd = sd(man_log2FC_norm, na.rm = T), 
    log2FC_norm = mean(man_log2FC_norm, na.rm = T)
)

mean_dtbl &lt;- mean_dtbl %&gt;%
  left_join(mean_FC) %&gt;%
  distinct()</code></pre>
<pre class="r"><code>prot &lt;- &quot;BTK_Y551&quot;

# Normalised counts
plot &lt;- ggplot(subset(counts_dtbl, staining == &quot;yes&quot; &amp; cell_line == &quot;HBL1&quot; &amp; target_nospace == prot)) +
  geom_point(aes(
    x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = man_FC_norm,
    color = stimulus)) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_DS108))) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim) +
  labs(x = &quot;&quot;, y = &quot;Fold change over 0 uM DMSO + PBS \n(normalised counts)&quot;, title = prot) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_wrangle2-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_wrangle2-1">
Past versions of fig_wrangle2-1.png
</button>
</p>
<div id="fig-fig_wrangle2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Manually calculated fold change of
signal compared to 0 uM inhibitor (DMSO ctrl) + PBS stimulation for BTK
(Y551) (HBL1 cells, normalised counts). </em></p>
<p>Finally, we will save the datasets as .csv file</p>
<ul>
<li><p>IDseq_data_sample.csv: Dataset of counts per protein per
sample</p></li>
<li><p>IDseq_data_condition.csv: Dataset of counts per protein per
condition (mean of replicates)</p></li>
</ul>
<pre class="r"><code># Dataset of counts per protein per sample 
data_well_all &lt;- counts_dtbl %&gt;% 
  arrange(plate_well, target_nospace) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

write.csv(data_well_all, file = &quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;, row.names = F)

# Dataset of mean counts per protein per condition (replicates combined)
data_mean_all &lt;- mean_dtbl %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

write.csv(data_mean_all, file = &quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code>rm(list = ls(pattern = &quot;Plate&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9373542 500.7   16263484 868.6 16263484 868.6
Vcells 23591701 180.0   37544322 286.5 37540090 286.5</code></pre>
</div>
</div>
<div id="counts-analysis" class="section level2">
<h2>Counts analysis</h2>
<div id="counts-per-sample" class="section level4">
<h4>Counts per sample</h4>
<p>To gain insight into the effect of experimental conditions on total
signal, we first have a look at the total (raw) and normalised counts
per sample.</p>
<pre class="r"><code># Compensated counts
fig_sample_true &lt;- ggplot(data_properties_true, aes(inhib_conc, nCount, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10() +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) + 
  textsize_small

print(fig_sample_true)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_sample-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_sample-1">
Past versions of fig_counts_sample-1.png
</button>
</p>
<div id="fig-fig_counts_sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Normalised compensated counts
fig_sample_norm &lt;- ggplot(data_properties_norm_true, aes(inhib_conc, nCount_norm, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10() +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) + 
  textsize_small

print(fig_sample_norm)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_sample-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_sample-2">
Past versions of fig_counts_sample-2.png
</button>
</p>
<div id="fig-fig_counts_sample-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Compensated counts
fig_sample_true &lt;- ggplot(subset(data_properties_true, staining == &quot;yes&quot;), aes(inhib_conc, nCount, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(limits = c(5e4, 5e5)) +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) + 
  textsize_small

# Normalised compensated counts
fig_sample_norm &lt;- ggplot(data_properties_norm_true, aes(inhib_conc, nCount_norm, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(limits = c(5e4, 5e5)) +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) + 
  textsize_small

plot_grid(fig_sample_true, fig_sample_norm, ncol = 2)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_sample2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_sample2-1">
Past versions of fig_counts_sample2-1.png
</button>
</p>
<div id="fig-fig_counts_sample2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution counts per sample with
compensation (left) and normalisation (right).</em></p>
</div>
<div id="counts-per-ab" class="section level4">
<h4>Counts per Ab</h4>
<p>We can also look at the counts per Ab for selected samples to get an
indication of any differences between proteins and between aIg+H2O2 and
PBS stimulation.</p>
<pre class="r"><code># Compensated counts
fig_Ab_true &lt;- ggplot(unique(subset(counts_dtbl, inhib_conc == &quot;0 uM (DMSO)&quot;)), 
       aes(x = reorder(target_nospace, true_counts), 
           y = true_counts, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) +
  theme_bw() +
  facet_grid2(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

print(fig_Ab_true)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_Ab_counts-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts-1">
Past versions of fig_Ab_counts-1.png
</button>
</p>
<div id="fig-fig_Ab_counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution of compensated counts per
Ab-barcode for the 0 uM (DMSO) samples (non-normalised).</em></p>
<pre class="r"><code># Compensated counts
fig_Ab_norm &lt;- ggplot(unique(subset(counts_dtbl, inhib_conc == &quot;0 uM (DMSO)&quot;)), 
       aes(x = reorder(target_nospace, true_counts_norm_geo), 
           y = true_counts_norm_geo, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) +
  theme_bw() +
  facet_grid2(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

print(fig_Ab_norm)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_Ab_counts_norm-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts_norm-1">
Past versions of fig_Ab_counts_norm-1.png
</button>
</p>
<div id="fig-fig_Ab_counts_norm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution of normalised compensated
counts per Ab-barcode for the 0 uM (DMSO) samples. Here we see some
protein targets that clearly show a difference between the aIg+H2O2 and
PBS samples.</em></p>
<pre class="r"><code># Compensated
fig_Ab_true_HBL1 &lt;- ggplot(unique(subset(counts_dtbl, cell_line == &quot;HBL1&quot; &amp; inhib_conc_text == &quot;0 uM iSYK (DMSO)&quot;)), 
       aes(x = reorder(target_nospace, true_counts), 
           y = true_counts, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) +
  theme_bw() +
  facet_grid(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

# Normalised
fig_Ab_norm_HBL1 &lt;- ggplot(unique(subset(counts_dtbl, cell_line == &quot;HBL1&quot; &amp; inhib_conc_text == &quot;0 uM iSYK (DMSO)&quot;)), 
       aes(x = reorder(target_nospace, true_counts_norm_geo), 
           y = true_counts_norm_geo, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) +
  theme_bw() +
  facet_grid(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

plot_grid(fig_Ab_true_HBL1, fig_Ab_norm_HBL1, ncol = 1)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_Ab_counts_HBL1-1.png" width="1536" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts_HBL1-1">
Past versions of fig_Ab_counts_HBL1-1.png
</button>
</p>
<div id="fig-fig_Ab_counts_HBL1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution of counts per Ab-barcode
for the HBL1 0 uM iSYK (DMSO) samples. The normalisation removes the
variation between replicates, but preserves the variation between
proteins and highlights the differences between aIg+H2O2 and PBS
stimulated samples.</em></p>
</div>
</div>
<div id="basal-cellsignaling-state" class="section level2">
<h2>Basal cell/signaling state</h2>
<p>First, we look at the basal signaling levels in both HBL1 and OCI-Ly8
cells (without any inhibitor/DMSO and without any stimulus). We can
investigate differences in basal signaling between the cell lines.</p>
<div id="deseq2" class="section level3 tabset">
<h3 class="tabset">DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="cell" class="section level4">
<h4>Cell</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>0 uM inhibitor (PBS ctrls), PBS stimulus, both cell lines samples
(24 samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_line (ref: HBL1)</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- counts_dtbl  %&gt;%
  filter(stimulus == &quot;PBS&quot; &amp; inhib_conc == &quot;0 uM (PBS)&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(meta_wellID) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         cell_inhib = paste(cell_line, inhibitor)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ cell_line # cell_inhib

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_line &lt;- relevel(dds$cell_line, ref = &quot;HBL1&quot;)
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$cell_inhib &lt;- relevel(dds$cell_inhib, ref = &quot;HBL1 iSYK&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code>[1] &quot;Intercept&quot;                 &quot;cell_line_OCI.Ly8_vs_HBL1&quot;</code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_cell &lt;- results(dds, name = &quot;cell_line_OCI.Ly8_vs_HBL1&quot;, alpha = 0.05)

# results_iBTK &lt;- results(dds, name = &quot;inhibitor_iBTK_vs_iSYK&quot;, alpha = 0.05)
# results_iNFkB &lt;- results(dds, name = &quot;inhibitor_iNFkB_vs_iSYK&quot;, alpha = 0.05)
# results_iPI3Kd &lt;- results(dds, name = &quot;inhibitor_iPI3Kd_vs_iSYK&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_cell) 
                     # results_iBTK, results_iNFkB, results_iPI3Kd)

# Results summary
summary(results_cell, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 20, 19%
LFC &lt; 0 (down)     : 21, 20%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 555)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code># par(mfrow = c(1,3), mar = c(4,4,2,1))
par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_cell, alpha = signcutoff, main = &quot;OCI-Ly8 vs HBL1&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_basal-1.png" width="288" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_basal-1">
Past versions of fig_log2FC_basal-1.png
</button>
</p>
<div id="fig-fig_log2FC_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># plotMA(results_iBTK, alpha = signcutoff, main = &quot;iBTK vs iSYK&quot;)
# plotMA(results_iPI3Kd, alpha = signcutoff, main = &quot;iPI3Kd vs iSYK&quot;)</code></pre>
<div id="data-transformation" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_conc&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(cell_line, levels = cells_DS108), factor(cell_line, levels = cells_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhib_conc) +
    scale_color_manual(values = colors_cell, name = &quot;Cell line&quot;) +
    scale_fill_manual(values = colors_cell, name = &quot;Cell line&quot;) +
    scale_shape_manual(values = c(16, 17), name = &quot;Cell line&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_basal-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_basal-1">
Past versions of fig_PCA_basal-1.png
</button>
</p>
<div id="fig-fig_PCA_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_basal-2.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_basal-2">
Past versions of fig_PCA_basal-2.png
</button>
</p>
<div id="fig-fig_PCA_basal-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_basal-3.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_basal-3">
Past versions of fig_PCA_basal-3.png
</button>
</p>
<div id="fig-fig_PCA_basal-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 shows a clear separation on cell line, and PC2
contributes little. The different types of data transformation also give
similar results.</em></p>
</div>
<div id="variance" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_basal-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_basal-1">
Past versions of fig_sd_basal-1.png
</button>
</p>
<div id="fig-fig_sd_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform (log2(n+1))&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_basal-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_basal-2">
Past versions of fig_sd_basal-2.png
</button>
</p>
<div id="fig-fig_sd_basal-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising transform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_basal-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_basal-3">
Past versions of fig_sd_basal-3.png
</button>
</p>
<div id="fig-fig_sd_basal-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log transform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_basal-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_basal-4">
Past versions of fig_sd_basal-4.png
</button>
</p>
<div id="fig-fig_sd_basal-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$cell_inhib,
                                      list_datatransform[[data]]$replicate,
                                      sep = &quot; - &quot;)
  # rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_basal-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_basal-1">
Past versions of fig_distances_basal-1.png
</button>
</p>
<div id="fig-fig_distances_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_basal-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_basal-2">
Past versions of fig_distances_basal-2.png
</button>
</p>
<div id="fig-fig_distances_basal-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_basal-3.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_basal-3">
Past versions of fig_distances_basal-3.png
</button>
</p>
<div id="fig-fig_distances_basal-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_cell&quot;)
                        # &quot;volc_iBTK&quot;, &quot;volc_iNFkB&quot;, &quot;volc_iPI3Kd&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_cell)
                          # volc_iBTK, volc_iNFkB, volc_iPI3Kd)
titles_volcano &lt;- c(&quot;OCI-Ly8 vs HBL1 (basal signaling w/o inhibitor)&quot;)
                    # &quot;volc_iBTK&quot;, &quot;volc_iNFkB&quot;, &quot;volc_iPI3Kd&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_basal-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_basal-1">
Past versions of fig_volcano_basal-1.png
</button>
</p>
<div id="fig-fig_volcano_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model comparison:
OCI-Ly8 vs HBL1. Various (phospho)proteins are significantly lower in
the OCI-Ly8 cells (red), but there are also some (phospho)proteins with
higher presence in OCI-Ly8 (blue).</em></p>
</div>
<div id="heatmap" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_cell&quot;)
                        # &quot;heat_iBTK&quot;, &quot;heat_iNFkB&quot;, &quot;heat_iPI3Kd&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;cell_line&quot;, &quot;inhibitor&quot;)]) %&gt;%
  arrange(cell_line, factor(inhibitor, levels = inhib_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# OCI-Ly8 vs HBL1
heat_cell$marker_pathway[is.na(heat_cell$marker_pathway)] &lt;- &quot;Other&quot;

ann_colors_man &lt;- list(cell_line = colors_cell,
                       inhibitor = colors_inhib,
                       modification = c(phospho = &quot;#f768a1&quot;, no = &quot;grey&quot;, cleaved = &quot;#fff7bc&quot;),
                       marker_pathway = setNames(colors_light12[1:length(unique(heat_cell$marker_pathway))], unique(heat_cell$marker_pathway))
)

# Get the rld transformed data for the heatmap and scale per row
data_cell &lt;- t(scale(t(assay(rld)[unique(c(heat_cell$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_cell,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$cell_line,
  row_split = subset(heat_cell)$modification,
  cellwidth = 10,
  cellheight = 10,
  fontsize = 8,
  border_color = NA,
  # main = &quot;scaled per row, ComplexHeatmap&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_basal-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_basal-1">
Past versions of fig_heatmap_basal-1.png
</button>
</p>
<div id="fig-fig_heatmap_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
the OCI-Ly8 vs HBL1 comparison (no threshold for fold change).</em></p>
<p><em>The apoptosis markers cleaved PARP and active Caspase 3 are
higher in the OCI-Ly8 cells, which is consistent with the FACS data,
where we saw more debris and dead cells in the OCI-Ly8 samples.
Regarding the phospho-proteins, several upstream BCR signaling proteins
and MAPK-JNK signaling proteins show lower presence in OCI-Ly8, while
some JAK-STAT signaling proteins show higher presence. Regarding the
non-phospho-proteins, serveral surface markers are increased in OCI-Ly8,
including IgE, while IgM, CD79a and several intracellular proteins are
decreased in OCI-Ly8.</em></p>
</div>
</div>
</div>
<div id="protein-comparisons" class="section level3">
<h3>Protein comparisons</h3>
<p>Finally, we visualised the normalised true counts of all significant
proteins in the OCI-Ly8 vs HBL1 comparison (41 in total).</p>
<pre class="r"><code># Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins
sign_prot_cell &lt;- sort(unique(c(heat_cell$target_nospace)))
# print(c(&quot;Significant proteins (OCI-ly8 vs HBL1 basal signaling):&quot;, sign_prot_cell))

# Datasets generated during analysis
# heat_cell: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_cell: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
clean_data_well &lt;- data.frame(data_cell) %&gt;% 
  mutate(target_nospace = rownames(data_cell)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value&quot;, 2:c(ncol(data_cell)+1)) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_cell) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

prot_labels &lt;- sort(setNames(unique(pull(clean_data_well, target)), unique(pull(clean_data_well, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
clean_data_sign &lt;- heat_cell %&gt;%
  filter(target_nospace %in% sign_prot_cell) %&gt;%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean = baseMean, 
                log2FC = log2FoldChange, 
                log2FC_se = lfcSE)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_cell &amp; stimulus == &quot;PBS&quot; &amp; inhib_conc == &quot;0 uM (PBS)&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)</code></pre>
<pre class="r"><code>plot &lt;- ggplot(clean_data_well, 
               aes(factor(cell_line, levels = cells_DS108),
                   counts_norm, 
                   color = cell_line, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target_nospace, labels = prot_labels)), ncol = 5) +
  theme_bw() +
  scale_y_log10() +
  scale_color_manual(values = colors_cell, name = &quot;Cell line&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts (log10 scale)&quot;) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal_all-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal_all-1">
Past versions of fig_counts_basal_all-1.png
</button>
</p>
<div id="fig-fig_counts_basal_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Normalised counts for all significantly
changing proteins. For some proteins the effect size is quite
small.</em></p>
<pre class="r"><code>options(repr.plot.width = 3.5, repr.plot.height = 2.5)

# Re-redesigned fig
filt_clean_data_mean &lt;- clean_data_mean %&gt;%
  dplyr::group_by(cell_line, target_nospace) %&gt;%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_cell) {
  plot &lt;- ggplot() +
    geom_jitter(data = subset(clean_data_well, target_nospace == protein), 
                aes(factor(cell_line, levels = cells_DS108), counts_norm, color = cell_line), 
                alpha = 1, size = 1.5, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
                  aes(factor(cell_line, levels = cells_DS108),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = cell_line),
                  width = 0.5, size = 1, color = &quot;black&quot;) +
    # geom_col(data = subset(filt_clean_data_mean, target_nospace == protein), 
    #          aes(factor(cell_line, levels = cells_DS108), mean_counts_norm, group = cell_line, fill = cell_line), 
    #          position = &quot;dodge2&quot;, alpha = 0.5) +
    # geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
    #               aes(factor(cell_line, levels = cells_DS108),
    #                   ymin = mean_counts_norm - mean_counts_norm_sd,
    #                   ymax = mean_counts_norm + mean_counts_norm_sd,
    #                   group = cell_line),
    #               position = position_dodge2(width = 0.9), width = 0.5) +
    facet_wrap(vars(factor(target_nospace))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_cell, name = &quot;Cell line&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-1.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-1">
Past versions of fig_counts_basal-1.png
</button>
</p>
<div id="fig-fig_counts_basal-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-2.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-2">
Past versions of fig_counts_basal-2.png
</button>
</p>
<div id="fig-fig_counts_basal-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-3.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-3">
Past versions of fig_counts_basal-3.png
</button>
</p>
<div id="fig-fig_counts_basal-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-4.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-4">
Past versions of fig_counts_basal-4.png
</button>
</p>
<div id="fig-fig_counts_basal-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-5.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-5">
Past versions of fig_counts_basal-5.png
</button>
</p>
<div id="fig-fig_counts_basal-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-6.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-6">
Past versions of fig_counts_basal-6.png
</button>
</p>
<div id="fig-fig_counts_basal-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-7.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-7">
Past versions of fig_counts_basal-7.png
</button>
</p>
<div id="fig-fig_counts_basal-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-8.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-8">
Past versions of fig_counts_basal-8.png
</button>
</p>
<div id="fig-fig_counts_basal-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-9.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-9">
Past versions of fig_counts_basal-9.png
</button>
</p>
<div id="fig-fig_counts_basal-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-10.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-10">
Past versions of fig_counts_basal-10.png
</button>
</p>
<div id="fig-fig_counts_basal-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-11.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-11">
Past versions of fig_counts_basal-11.png
</button>
</p>
<div id="fig-fig_counts_basal-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-12.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-12">
Past versions of fig_counts_basal-12.png
</button>
</p>
<div id="fig-fig_counts_basal-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-13.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-13">
Past versions of fig_counts_basal-13.png
</button>
</p>
<div id="fig-fig_counts_basal-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-14.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-14">
Past versions of fig_counts_basal-14.png
</button>
</p>
<div id="fig-fig_counts_basal-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-15.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-15">
Past versions of fig_counts_basal-15.png
</button>
</p>
<div id="fig-fig_counts_basal-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-16.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-16">
Past versions of fig_counts_basal-16.png
</button>
</p>
<div id="fig-fig_counts_basal-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-17.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-17">
Past versions of fig_counts_basal-17.png
</button>
</p>
<div id="fig-fig_counts_basal-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-18.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-18">
Past versions of fig_counts_basal-18.png
</button>
</p>
<div id="fig-fig_counts_basal-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-19.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-19">
Past versions of fig_counts_basal-19.png
</button>
</p>
<div id="fig-fig_counts_basal-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-20.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-20">
Past versions of fig_counts_basal-20.png
</button>
</p>
<div id="fig-fig_counts_basal-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-21.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-21">
Past versions of fig_counts_basal-21.png
</button>
</p>
<div id="fig-fig_counts_basal-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-22.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-22">
Past versions of fig_counts_basal-22.png
</button>
</p>
<div id="fig-fig_counts_basal-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-23.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-23">
Past versions of fig_counts_basal-23.png
</button>
</p>
<div id="fig-fig_counts_basal-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-24.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-24">
Past versions of fig_counts_basal-24.png
</button>
</p>
<div id="fig-fig_counts_basal-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-25.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-25">
Past versions of fig_counts_basal-25.png
</button>
</p>
<div id="fig-fig_counts_basal-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-26.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-26">
Past versions of fig_counts_basal-26.png
</button>
</p>
<div id="fig-fig_counts_basal-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-27.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-27">
Past versions of fig_counts_basal-27.png
</button>
</p>
<div id="fig-fig_counts_basal-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-28.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-28">
Past versions of fig_counts_basal-28.png
</button>
</p>
<div id="fig-fig_counts_basal-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-29.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-29">
Past versions of fig_counts_basal-29.png
</button>
</p>
<div id="fig-fig_counts_basal-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-30.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-30">
Past versions of fig_counts_basal-30.png
</button>
</p>
<div id="fig-fig_counts_basal-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-31.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-31">
Past versions of fig_counts_basal-31.png
</button>
</p>
<div id="fig-fig_counts_basal-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-32.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-32">
Past versions of fig_counts_basal-32.png
</button>
</p>
<div id="fig-fig_counts_basal-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-33.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-33">
Past versions of fig_counts_basal-33.png
</button>
</p>
<div id="fig-fig_counts_basal-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-34.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-34">
Past versions of fig_counts_basal-34.png
</button>
</p>
<div id="fig-fig_counts_basal-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-35.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-35">
Past versions of fig_counts_basal-35.png
</button>
</p>
<div id="fig-fig_counts_basal-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-36.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-36">
Past versions of fig_counts_basal-36.png
</button>
</p>
<div id="fig-fig_counts_basal-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-37.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-37">
Past versions of fig_counts_basal-37.png
</button>
</p>
<div id="fig-fig_counts_basal-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-38.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-38">
Past versions of fig_counts_basal-38.png
</button>
</p>
<div id="fig-fig_counts_basal-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-39.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-39">
Past versions of fig_counts_basal-39.png
</button>
</p>
<div id="fig-fig_counts_basal-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-40.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-40">
Past versions of fig_counts_basal-40.png
</button>
</p>
<div id="fig-fig_counts_basal-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_basal-41.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_basal-41">
Past versions of fig_counts_basal-41.png
</button>
</p>
<div id="fig-fig_counts_basal-41" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-data" class="section level3">
<h3>Save data</h3>
<p>Save all information and datasets as .csv file</p>
<ul>
<li><p>sign_proteins.txt: List of significantly changing
proteins</p></li>
<li><p>sign_DESeq2data.csv: Dataset with significantly changing
proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>sign_sample.csv: Dataset of counts per sample per protein +
heatmap data for significantly changing proteins</p></li>
<li><p>sign_mean.csv: Dataset of counts per condition (mean of repl) per
protein for significantly changing proteins</p></li>
</ul>
<pre class="r"><code># List of significantly changing proteins: sign_prot_cell
write(sign_prot_cell, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_proteins.txt&quot;)

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_DESeq2data.csv&quot;, row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_sample.csv&quot;, row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9846053 525.9   16263484 868.6 16263484 868.6
Vcells 25183257 192.2   45133186 344.4 37540090 286.5</code></pre>
</div>
</div>
<div id="inhibitor-effects-on-basal-signaling" class="section level2">
<h2>Inhibitor effects on basal signaling</h2>
<p>Now, we look at the effect of the inhibitors and their different
concentrations on basal signaling levels in both HBL1 and OCI-Ly8 cells
(without any stimulus). We can investigate differences between
inhibitors, between inhibitor concentrations, and between cell
lines.</p>
<div id="deseq2-1" class="section level3 tabset">
<h3 class="tabset">DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="hbl1-conc" class="section level4">
<h4>HBL1 conc</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>4 inhibitors with all conc, PBS stimulus, HBL1 cells only (72
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(stimulus == &quot;PBS&quot; &amp; cell_line == &quot;HBL1&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhib_new &lt;- relevel(dds$inhib_new, ref = &quot;DMSO_ctrl&quot;)
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 uM (DMSO)&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                
 [2] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;  
 [3] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot; 
 [4] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;
 [5] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;  
 [6] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;    
 [7] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;   
 [8] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;  
 [9] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;    
[10] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;   
[11] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;  
[12] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot; 
[13] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;   
[14] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;  
[15] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot; 
[16] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;
[17] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;  
[18] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_PBS_iH &lt;- results(dds, name = &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK0.1_H &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_H &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_H &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_H &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_H &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_H &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_H &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_H &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_H &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_H &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_H &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_H &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_H &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_H &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_H &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_H &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_PBS_iH, 
                     results_iSYK0.1_H, results_iSYK1_H, results_iSYK10_H, results_iSYK100_H, 
                     results_iBTK0.1_H, results_iBTK1_H, results_iBTK10_H, results_iBTK100_H, 
                     results_iPI3Kd0.1_H, results_iPI3Kd1_H, results_iPI3Kd10_H, results_iPI3Kd100_H, 
                     results_iNFkB0.1_H, results_iNFkB1_H, results_iNFkB10_H, results_iNFkB100_H)

# Results summary
summary(results_iSYK100_H, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 6, 5.6%
LFC &lt; 0 (down)     : 7, 6.5%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 441)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_PBS_iH, alpha = signcutoff, main = &quot;PBS vs DMSO&quot;)
plotMA(results_iSYK100_H, alpha = signcutoff, main = &quot;100 uM iSYK vs DMSO&quot;)
plotMA(results_iNFkB100_H, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_bas_HBL1_conc-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_bas_HBL1_conc-1">
Past versions of fig_log2FC_bas_HBL1_conc-1.png
</button>
</p>
<div id="fig-fig_log2FC_bas_HBL1_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-1" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-1" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 8, repr.plot.height = 3)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_HBL1_conc-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_HBL1_conc-1">
Past versions of fig_PCA_bas_HBL1_conc-1.png
</button>
</p>
<div id="fig-fig_PCA_bas_HBL1_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_HBL1_conc-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_HBL1_conc-2">
Past versions of fig_PCA_bas_HBL1_conc-2.png
</button>
</p>
<div id="fig-fig_PCA_bas_HBL1_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_HBL1_conc-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_HBL1_conc-3">
Past versions of fig_PCA_bas_HBL1_conc-3.png
</button>
</p>
<div id="fig-fig_PCA_bas_HBL1_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is mainly the difference between the 100 uM iNFkB
samples and all other samples, and PC2 is mainly the difference between
1 repl of 100 uM iSYK and all other samples.</em></p>
</div>
<div id="variance-1" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_HBL1_conc-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_HBL1_conc-1">
Past versions of fig_sd_bas_HBL1_conc-1.png
</button>
</p>
<div id="fig-fig_sd_bas_HBL1_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_HBL1_conc-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_HBL1_conc-2">
Past versions of fig_sd_bas_HBL1_conc-2.png
</button>
</p>
<div id="fig-fig_sd_bas_HBL1_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_HBL1_conc-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_HBL1_conc-3">
Past versions of fig_sd_bas_HBL1_conc-3.png
</button>
</p>
<div id="fig-fig_sd_bas_HBL1_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_HBL1_conc-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_HBL1_conc-4">
Past versions of fig_sd_bas_HBL1_conc-4.png
</button>
</p>
<div id="fig-fig_sd_bas_HBL1_conc-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-1" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$inhib_conc_text,
                                      list_datatransform[[data]]$replicate,
                                      sep = &quot; - &quot;)
  # rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_HBL1_conc-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_HBL1_conc-1">
Past versions of fig_distances_bas_HBL1_conc-1.png
</button>
</p>
<div id="fig-fig_distances_bas_HBL1_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_HBL1_conc-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_HBL1_conc-2">
Past versions of fig_distances_bas_HBL1_conc-2.png
</button>
</p>
<div id="fig-fig_distances_bas_HBL1_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_HBL1_conc-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_HBL1_conc-3">
Past versions of fig_distances_bas_HBL1_conc-3.png
</button>
</p>
<div id="fig-fig_distances_bas_HBL1_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-1" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_PBS_H&quot;,
                        &quot;volc_iSYK0.1_H&quot;, &quot;volc_iSYK1_H&quot;, &quot;volc_iSYK10_H&quot;, &quot;volc_iSYK100_H&quot;,
                        &quot;volc_iBTK0.1_H&quot;, &quot;volc_iBTK1_H&quot;, &quot;volc_iBTK10_H&quot;, &quot;volc_iBTK100_H&quot;,
                        &quot;volc_iPI3Kd0.1_H&quot;, &quot;volc_iPI3Kd1_H&quot;, &quot;volc_iPI3Kd10_H&quot;, &quot;volc_iPI3Kd100_H&quot;,
                        &quot;volc_iNFkB0.1_H&quot;, &quot;volc_iNFkB1_H&quot;, &quot;volc_iNFkB10_H&quot;, &quot;volc_iNFkB100_H&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_PBS_H, 
                          volc_iSYK0.1_H, volc_iSYK1_H, volc_iSYK10_H, volc_iSYK100_H, 
                          volc_iBTK0.1_H, volc_iBTK1_H, volc_iBTK10_H, volc_iBTK100_H, 
                          volc_iPI3Kd0.1_H, volc_iPI3Kd1_H, volc_iPI3Kd10_H, volc_iPI3Kd100_H, 
                          volc_iNFkB0.1_H, volc_iNFkB1_H, volc_iNFkB10_H, volc_iNFkB100_H)
# titles_volcano &lt;- c(&quot;OCI-Ly8 vs HBL1 (... signaling w/o inhibitor)&quot;)
titles_volcano &lt;- c(&quot;PBS ctrl vs DMSO ctrl (HBL1 w/o stim)&quot;,
                    &quot;0.1 uM iSYK vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;1 uM iSYK vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;10 uM iSYK vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;100 uM iSYK vs DMSO ctrl (HBL1 w/o stim)&quot;,
                    &quot;0.1 uM iBTK vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;1 uM iBTK vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;10 uM iBTK vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;100 uM iBTK vs DMSO ctrl (HBL1 w/o stim)&quot;,
                    &quot;0.1 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)&quot;,
                    &quot;0.1 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-1">
Past versions of fig_volcano_bas_HBL1_conc-1.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-2">
Past versions of fig_volcano_bas_HBL1_conc-2.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-3">
Past versions of fig_volcano_bas_HBL1_conc-3.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-4">
Past versions of fig_volcano_bas_HBL1_conc-4.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-5">
Past versions of fig_volcano_bas_HBL1_conc-5.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-6">
Past versions of fig_volcano_bas_HBL1_conc-6.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-7">
Past versions of fig_volcano_bas_HBL1_conc-7.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-8">
Past versions of fig_volcano_bas_HBL1_conc-8.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-9">
Past versions of fig_volcano_bas_HBL1_conc-9.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-10">
Past versions of fig_volcano_bas_HBL1_conc-10.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-11">
Past versions of fig_volcano_bas_HBL1_conc-11.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-12">
Past versions of fig_volcano_bas_HBL1_conc-12.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-13">
Past versions of fig_volcano_bas_HBL1_conc-13.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-14">
Past versions of fig_volcano_bas_HBL1_conc-14.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-15">
Past versions of fig_volcano_bas_HBL1_conc-15.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-16">
Past versions of fig_volcano_bas_HBL1_conc-16.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_HBL1_conc-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_HBL1_conc-17">
Past versions of fig_volcano_bas_HBL1_conc-17.png
</button>
</p>
<div id="fig-fig_volcano_bas_HBL1_conc-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons: each inhibitor conc vs DMSO ctrl. iSYK shows 1 protein
oging down at 1 uM, more at 10 uM and even more at 100 uM. iBTK starts
showing significantly changing proteins at 10 uM. iPI3Kd shows 3
proteins significantly increasing at 0.1 uM, and only 1 protein
significantly decreasing at 100 uM. iNFkB shows several proteins
significantly changing at 10 uM, and even more proteins changing at 100
uM.</em></p>
</div>
<div id="heatmap-1" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_PBS_H&quot;,
                        &quot;heat_iSYK0.1_H&quot;, &quot;heat_iSYK1_H&quot;, &quot;heat_iSYK10_H&quot;, &quot;heat_iSYK100_H&quot;,
                        &quot;heat_iBTK0.1_H&quot;, &quot;heat_iBTK1_H&quot;, &quot;heat_iBTK10_H&quot;, &quot;heat_iBTK100_H&quot;,
                        &quot;heat_iPI3Kd0.1_H&quot;, &quot;heat_iPI3Kd1_H&quot;, &quot;heat_iPI3Kd10_H&quot;, &quot;heat_iPI3Kd100_H&quot;,
                        &quot;heat_iNFkB0.1_H&quot;, &quot;heat_iNFkB1_H&quot;, &quot;heat_iNFkB10_H&quot;, &quot;heat_iNFkB100_H&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_100uM_H &lt;- list(heat_iSYK100_H, heat_iBTK100_H, heat_iPI3Kd100_H, heat_iNFkB100_H) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# Get the rld transformed data for the heatmap and scale per row
data_100uM_H &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_H$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_100uM_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_HBL1_conc_100-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_HBL1_conc_100-1">
Past versions of fig_heatmap_bas_HBL1_conc_100-1.png
</button>
</p>
<div id="fig-fig_heatmap_bas_HBL1_conc_100-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM iNFkB vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iNFkB_H &lt;- t(scale(t(assay(rld)[unique(c(heat_iNFkB100_H$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_iNFkB_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iNFkB100_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins in 100 uM iNFkB (HBL1 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_HBL1_conc-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_HBL1_conc-1">
Past versions of fig_heatmap_bas_HBL1_conc-1.png
</button>
</p>
<div id="fig-fig_heatmap_bas_HBL1_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 100 uM iSYK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iSYK_H &lt;- t(scale(t(assay(rld)[unique(c(heat_iSYK100_H$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iSYK_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iSYK100_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins in 100 uM iSYK (HBL1 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_HBL1_conc-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_HBL1_conc-2">
Past versions of fig_heatmap_bas_HBL1_conc-2.png
</button>
</p>
<div id="fig-fig_heatmap_bas_HBL1_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 100 uM iBTK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iBTK_H &lt;- t(scale(t(assay(rld)[unique(c(heat_iBTK100_H$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iBTK_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iBTK100_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins in 100 uM iBTK (HBL1 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_HBL1_conc-3.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_HBL1_conc-3">
Past versions of fig_heatmap_bas_HBL1_conc-3.png
</button>
</p>
<div id="fig-fig_heatmap_bas_HBL1_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmaps of the significant results in
the 100 uM inhibitor vs DMSO ctrl comparison (no threshold for fold
change).</em></p>
</div>
</div>
<div id="oci-ly8-conc" class="section level4">
<h4>OCI-Ly8 conc</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>4 inhibitors with all conc, PBS stimulus, OCI-Ly8 cells only (72
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(stimulus == &quot;PBS&quot; &amp; cell_line == &quot;OCI-Ly8&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhib_new &lt;- relevel(dds$inhib_new, ref = &quot;DMSO_ctrl&quot;)
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 uM (DMSO)&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                
 [2] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;  
 [3] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot; 
 [4] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;
 [5] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;  
 [6] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;    
 [7] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;   
 [8] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;  
 [9] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;    
[10] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;   
[11] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;  
[12] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot; 
[13] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;   
[14] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;  
[15] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot; 
[16] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;
[17] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;  
[18] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_PBS_iO &lt;- results(dds, name = &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK0.1_O &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_O &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_O &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_O &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_O &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_O &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_O &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_O &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_O &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_O &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_O &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_O &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_O &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_O &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_O &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_O &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_PBS_iO, 
                     results_iSYK0.1_O, results_iSYK1_O, results_iSYK10_O, results_iSYK100_O, 
                     results_iBTK0.1_O, results_iBTK1_O, results_iBTK10_O, results_iBTK100_O, 
                     results_iPI3Kd0.1_O, results_iPI3Kd1_O, results_iPI3Kd10_O, results_iPI3Kd100_O, 
                     results_iNFkB0.1_O, results_iNFkB1_O, results_iNFkB10_O, results_iNFkB100_O)

# Results summary
summary(results_iSYK100_O, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 8, 7.5%
LFC &lt; 0 (down)     : 9, 8.4%
outliers [1]       : 0, 0%
low counts [2]     : 3, 2.8%
(mean count &lt; 708)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_PBS_iO, alpha = signcutoff, main = &quot;PBS vs DMSO&quot;)
plotMA(results_iSYK100_O, alpha = signcutoff, main = &quot;100 uM iSYK vs DMSO&quot;)
plotMA(results_iNFkB100_O, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_bas_Ly8_conc-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_bas_Ly8_conc-1">
Past versions of fig_log2FC_bas_Ly8_conc-1.png
</button>
</p>
<div id="fig-fig_log2FC_bas_Ly8_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-2" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-2" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 6, repr.plot.height = 4)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_Ly8_conc-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_Ly8_conc-1">
Past versions of fig_PCA_bas_Ly8_conc-1.png
</button>
</p>
<div id="fig-fig_PCA_bas_Ly8_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_Ly8_conc-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_Ly8_conc-2">
Past versions of fig_PCA_bas_Ly8_conc-2.png
</button>
</p>
<div id="fig-fig_PCA_bas_Ly8_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_Ly8_conc-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_Ly8_conc-3">
Past versions of fig_PCA_bas_Ly8_conc-3.png
</button>
</p>
<div id="fig-fig_PCA_bas_Ly8_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is mainly the difference between the 100 uM iNFkB
samples and all other samples, and PC2 is mainly the difference between
iSYK and iBTK inhibitor concentrations.</em></p>
</div>
<div id="variance-2" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_Ly8_conc-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_Ly8_conc-1">
Past versions of fig_sd_bas_Ly8_conc-1.png
</button>
</p>
<div id="fig-fig_sd_bas_Ly8_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_Ly8_conc-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_Ly8_conc-2">
Past versions of fig_sd_bas_Ly8_conc-2.png
</button>
</p>
<div id="fig-fig_sd_bas_Ly8_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_Ly8_conc-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_Ly8_conc-3">
Past versions of fig_sd_bas_Ly8_conc-3.png
</button>
</p>
<div id="fig-fig_sd_bas_Ly8_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_Ly8_conc-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_Ly8_conc-4">
Past versions of fig_sd_bas_Ly8_conc-4.png
</button>
</p>
<div id="fig-fig_sd_bas_Ly8_conc-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-2" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$inhib_conc_text,
                                      list_datatransform[[data]]$replicate,
                                      sep = &quot; - &quot;)
  # rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_Ly8_conc-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_Ly8_conc-1">
Past versions of fig_distances_bas_Ly8_conc-1.png
</button>
</p>
<div id="fig-fig_distances_bas_Ly8_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_Ly8_conc-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_Ly8_conc-2">
Past versions of fig_distances_bas_Ly8_conc-2.png
</button>
</p>
<div id="fig-fig_distances_bas_Ly8_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_Ly8_conc-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_Ly8_conc-3">
Past versions of fig_distances_bas_Ly8_conc-3.png
</button>
</p>
<div id="fig-fig_distances_bas_Ly8_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-2" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_PBS_O&quot;,
                        &quot;volc_iSYK0.1_O&quot;, &quot;volc_iSYK1_O&quot;, &quot;volc_iSYK10_O&quot;, &quot;volc_iSYK100_O&quot;,
                        &quot;volc_iBTK0.1_O&quot;, &quot;volc_iBTK1_O&quot;, &quot;volc_iBTK10_O&quot;, &quot;volc_iBTK100_O&quot;,
                        &quot;volc_iPI3Kd0.1_O&quot;, &quot;volc_iPI3Kd1_O&quot;, &quot;volc_iPI3Kd10_O&quot;, &quot;volc_iPI3Kd100_O&quot;,
                        &quot;volc_iNFkB0.1_O&quot;, &quot;volc_iNFkB1_O&quot;, &quot;volc_iNFkB10_O&quot;, &quot;volc_iNFkB100_O&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_PBS_O, 
                          volc_iSYK0.1_O, volc_iSYK1_O, volc_iSYK10_O, volc_iSYK100_O, 
                          volc_iBTK0.1_O, volc_iBTK1_O, volc_iBTK10_O, volc_iBTK100_O, 
                          volc_iPI3Kd0.1_O, volc_iPI3Kd1_O, volc_iPI3Kd10_O, volc_iPI3Kd100_O, 
                          volc_iNFkB0.1_O, volc_iNFkB1_O, volc_iNFkB10_O, volc_iNFkB100_O)
# titles_volcano &lt;- c(&quot;OCI-Ly8 vs HBL1 (... signaling w/o inhibitor)&quot;)
titles_volcano &lt;- c(&quot;PBS ctrl vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;,
                    &quot;0.1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;10 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;100 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;,
                    &quot;0.1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;10 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;100 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;,
                    &quot;0.1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;,
                    &quot;0.1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-1">
Past versions of fig_volcano_bas_Ly8_conc-1.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-2">
Past versions of fig_volcano_bas_Ly8_conc-2.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-3">
Past versions of fig_volcano_bas_Ly8_conc-3.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-4">
Past versions of fig_volcano_bas_Ly8_conc-4.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-5">
Past versions of fig_volcano_bas_Ly8_conc-5.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-6">
Past versions of fig_volcano_bas_Ly8_conc-6.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-7">
Past versions of fig_volcano_bas_Ly8_conc-7.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-8">
Past versions of fig_volcano_bas_Ly8_conc-8.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-9">
Past versions of fig_volcano_bas_Ly8_conc-9.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-10">
Past versions of fig_volcano_bas_Ly8_conc-10.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-11">
Past versions of fig_volcano_bas_Ly8_conc-11.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-12">
Past versions of fig_volcano_bas_Ly8_conc-12.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-13">
Past versions of fig_volcano_bas_Ly8_conc-13.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-14">
Past versions of fig_volcano_bas_Ly8_conc-14.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-15">
Past versions of fig_volcano_bas_Ly8_conc-15.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-16">
Past versions of fig_volcano_bas_Ly8_conc-16.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_Ly8_conc-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_Ly8_conc-17">
Past versions of fig_volcano_bas_Ly8_conc-17.png
</button>
</p>
<div id="fig-fig_volcano_bas_Ly8_conc-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons: each inhibitor conc vs DMSO ctrl. iSYK shows 1 protein
going down at 0.1 uM, and more at increasingly higher conc. iBTK starts
showing significantly changing proteins at 1 uM. iPI3Kd shows a few
proteins significantly changing at all conc. iNFkB shows several
proteins significantly changing at 10 uM, and even more proteins
changing at 100 uM.</em></p>
</div>
<div id="heatmap-2" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_PBS_O&quot;,
                        &quot;heat_iSYK0.1_O&quot;, &quot;heat_iSYK1_O&quot;, &quot;heat_iSYK10_O&quot;, &quot;heat_iSYK100_O&quot;,
                        &quot;heat_iBTK0.1_O&quot;, &quot;heat_iBTK1_O&quot;, &quot;heat_iBTK10_O&quot;, &quot;heat_iBTK100_O&quot;,
                        &quot;heat_iPI3Kd0.1_O&quot;, &quot;heat_iPI3Kd1_O&quot;, &quot;heat_iPI3Kd10_O&quot;, &quot;heat_iPI3Kd100_O&quot;,
                        &quot;heat_iNFkB0.1_O&quot;, &quot;heat_iNFkB1_O&quot;, &quot;heat_iNFkB10_O&quot;, &quot;heat_iNFkB100_O&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_100uM_O &lt;- list(heat_iSYK100_O, heat_iBTK100_O, heat_iPI3Kd100_O, heat_iNFkB100_O) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# Get the rld transformed data for the heatmap and scale per row
data_100uM_O &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_O$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_100uM_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_Ly8_conc_100-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_Ly8_conc_100-1">
Past versions of fig_heatmap_bas_Ly8_conc_100-1.png
</button>
</p>
<div id="fig-fig_heatmap_bas_Ly8_conc_100-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM iNFkB vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iNFkB_O &lt;- t(scale(t(assay(rld)[unique(c(heat_iNFkB100_O$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_iNFkB_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iNFkB100_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins in 100 uM iNFkB (OCI-Ly8 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_Ly8_conc-1.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_Ly8_conc-1">
Past versions of fig_heatmap_bas_Ly8_conc-1.png
</button>
</p>
<div id="fig-fig_heatmap_bas_Ly8_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 100 uM iSYK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iSYK_O &lt;- t(scale(t(assay(rld)[unique(c(heat_iSYK100_O$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iSYK_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iSYK100_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins in 100 uM iSYK (OCI-Ly8 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_Ly8_conc-2.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_Ly8_conc-2">
Past versions of fig_heatmap_bas_Ly8_conc-2.png
</button>
</p>
<div id="fig-fig_heatmap_bas_Ly8_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 100 uM iBTK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iBTK_O &lt;- t(scale(t(assay(rld)[unique(c(heat_iBTK100_O$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iBTK_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iBTK100_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins in 100 uM iBTK (OCI-Ly8 cells w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_Ly8_conc-3.png" width="1344" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_Ly8_conc-3">
Past versions of fig_heatmap_bas_Ly8_conc-3.png
</button>
</p>
<div id="fig-fig_heatmap_bas_Ly8_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
the 100 uM inhibitor vs DMSO ctrl comparisons (no threshold for fold
change).</em></p>
</div>
</div>
<div id="cellconc" class="section level4">
<h4>Cell:conc</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>4 inhibitors with all conc, PBS stimulus, both cell lines (144
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_line (ref: HBL1) + inhib_text_new (ref: DMSO
ctrl) + cell_line:inhib_text_new</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(stimulus == &quot;PBS&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ cell_line + inhib_text_new + cell_line:inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhib_new &lt;- relevel(dds$inhib_new, ref = &quot;DMSO_ctrl&quot;)
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
dds$cell_line &lt;- relevel(dds$cell_line, ref = &quot;HBL1&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                   
 [2] &quot;cell_line_OCI.Ly8_vs_HBL1&quot;                   
 [3] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;     
 [4] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;    
 [5] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;   
 [6] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;     
 [7] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;       
 [8] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;      
 [9] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;     
[10] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;       
[11] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;      
[12] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;     
[13] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;    
[14] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;      
[15] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;     
[16] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;    
[17] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;   
[18] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;     
[19] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;        
[20] &quot;cell_lineOCI.Ly8.inhib_text_new0.1.uM.iBTK&quot;  
[21] &quot;cell_lineOCI.Ly8.inhib_text_new0.1.uM.iNFkB&quot; 
[22] &quot;cell_lineOCI.Ly8.inhib_text_new0.1.uM.iPI3Kd&quot;
[23] &quot;cell_lineOCI.Ly8.inhib_text_new0.1.uM.iSYK&quot;  
[24] &quot;cell_lineOCI.Ly8.inhib_text_new1.uM.iBTK&quot;    
[25] &quot;cell_lineOCI.Ly8.inhib_text_new1.uM.iNFkB&quot;   
[26] &quot;cell_lineOCI.Ly8.inhib_text_new1.uM.iPI3Kd&quot;  
[27] &quot;cell_lineOCI.Ly8.inhib_text_new1.uM.iSYK&quot;    
[28] &quot;cell_lineOCI.Ly8.inhib_text_new10.uM.iBTK&quot;   
[29] &quot;cell_lineOCI.Ly8.inhib_text_new10.uM.iNFkB&quot;  
[30] &quot;cell_lineOCI.Ly8.inhib_text_new10.uM.iPI3Kd&quot; 
[31] &quot;cell_lineOCI.Ly8.inhib_text_new10.uM.iSYK&quot;   
[32] &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iBTK&quot;  
[33] &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iNFkB&quot; 
[34] &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iPI3Kd&quot;
[35] &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iSYK&quot;  
[36] &quot;cell_lineOCI.Ly8.inhib_text_newPBS.ctrl&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_cell_icc &lt;- results(dds, name = &quot;cell_line_OCI.Ly8_vs_HBL1&quot;, alpha = 0.05)

results_iSYK0.1_cc &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_cc &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_cc &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_cc &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_cc &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_cc &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_cc &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_cc &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_cc &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_cc &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_cc &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_cc &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_cc &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_cc &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_cc &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_cc &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_cell_iSYK_cc &lt;- results(dds, name = &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iSYK&quot;, alpha = 0.05)
results_cell_iBTK_cc &lt;- results(dds, name = &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iBTK&quot;, alpha = 0.05)
results_cell_iPI3Kd_cc &lt;- results(dds, name = &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iPI3Kd&quot;, alpha = 0.05)
results_cell_iNFkB_cc &lt;- results(dds, name = &quot;cell_lineOCI.Ly8.inhib_text_new100.uM.iNFkB&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_cell_icc, 
                     results_iSYK0.1_cc, results_iSYK1_cc, results_iSYK10_cc, results_iSYK100_cc,
                     results_iBTK0.1_cc, results_iBTK1_cc, results_iBTK10_cc, results_iBTK100_cc,
                     results_iPI3Kd0.1_cc, results_iPI3Kd1_cc, results_iPI3Kd10_cc, results_iPI3Kd100_cc,
                     results_iNFkB0.1_cc, results_iNFkB1_cc, results_iNFkB10_cc, results_iNFkB100_cc,
                     results_cell_iSYK_cc, results_cell_iBTK_cc, results_cell_iPI3Kd_cc, results_cell_iNFkB_cc)

# Results summary
summary(results_iNFkB100_cc, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 26, 24%
LFC &lt; 0 (down)     : 30, 28%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 446)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_cell_icc, alpha = signcutoff, main = &quot;OCI-Ly8 vs HBL1&quot;)
# plotMA(results_iSYK100_O, alpha = signcutoff, main = &quot;100 uM iSYK vs DMSO&quot;)
plotMA(results_iNFkB100_O, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)
plotMA(results_cell_iNFkB_cc, alpha = signcutoff, main = &quot;Effect of OCI-Ly8 on\n100 uM iNFkB vs DMSO&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_bas_cell_conc-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_bas_cell_conc-1">
Past versions of fig_log2FC_bas_cell_conc-1.png
</button>
</p>
<div id="fig-fig_log2FC_bas_cell_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-3" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-3" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 6, repr.plot.height = 4)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  # plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_color_manual(values = colors_blue9[4:9], name = &quot;Inhib conc&quot;) +
    scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    # scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    scale_shape_manual(values = c(16, 17), name = &quot;Cell line&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_cell_conc-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_cell_conc-1">
Past versions of fig_PCA_bas_cell_conc-1.png
</button>
</p>
<div id="fig-fig_PCA_bas_cell_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_cell_conc-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_cell_conc-2">
Past versions of fig_PCA_bas_cell_conc-2.png
</button>
</p>
<div id="fig-fig_PCA_bas_cell_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_bas_cell_conc-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_bas_cell_conc-3">
Past versions of fig_PCA_bas_cell_conc-3.png
</button>
</p>
<div id="fig-fig_PCA_bas_cell_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is mainly the difference between the cell lines,
and PC2 is mainly the difference between inhibitor concentrations
(especially the 100 uM iNFkB and the rest).</em></p>
</div>
<div id="variance-3" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_cell_conc-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_cell_conc-1">
Past versions of fig_sd_bas_cell_conc-1.png
</button>
</p>
<div id="fig-fig_sd_bas_cell_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_cell_conc-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_cell_conc-2">
Past versions of fig_sd_bas_cell_conc-2.png
</button>
</p>
<div id="fig-fig_sd_bas_cell_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_cell_conc-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_cell_conc-3">
Past versions of fig_sd_bas_cell_conc-3.png
</button>
</p>
<div id="fig-fig_sd_bas_cell_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_bas_cell_conc-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_bas_cell_conc-4">
Past versions of fig_sd_bas_cell_conc-4.png
</button>
</p>
<div id="fig-fig_sd_bas_cell_conc-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-3" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$cell_line, 
                                      list_datatransform[[data]]$inhib_conc_text,
                                      list_datatransform[[data]]$replicate,
                                      sep = &quot; - &quot;)
  # rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_cell_conc-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_cell_conc-1">
Past versions of fig_distances_bas_cell_conc-1.png
</button>
</p>
<div id="fig-fig_distances_bas_cell_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_cell_conc-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_cell_conc-2">
Past versions of fig_distances_bas_cell_conc-2.png
</button>
</p>
<div id="fig-fig_distances_bas_cell_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_bas_cell_conc-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_bas_cell_conc-3">
Past versions of fig_distances_bas_cell_conc-3.png
</button>
</p>
<div id="fig-fig_distances_bas_cell_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-3" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_cell_icc&quot;, 
                        &quot;volc_iSYK0.1_cc&quot;, &quot;volc_iSYK1_cc&quot;, &quot;volc_iSYK10_cc&quot;, &quot;volc_iSYK100_cc&quot;,
                        &quot;volc_iBTK0.1_cc&quot;, &quot;volc_iBTK1_cc&quot;, &quot;volc_iBTK10_cc&quot;, &quot;volc_iBTK100_cc&quot;,
                        &quot;volc_iPI3Kd0.1_cc&quot;, &quot;volc_iPI3Kd1_cc&quot;, &quot;volc_iPI3Kd10_cc&quot;, &quot;volc_iPI3Kd100_cc&quot;,
                        &quot;volc_iNFkB0.1_cc&quot;, &quot;volc_iNFkB1_cc&quot;, &quot;volc_iNFkB10_cc&quot;, &quot;volc_iNFkB100_cc&quot;,
                        &quot;volc_cell_iSYK_cc&quot;, &quot;volc_cell_iBTK_cc&quot;, &quot;volc_cell_iPI3Kd_cc&quot;, &quot;volc_cell_iNFkB_cc&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_cell_icc, 
                     volc_iSYK0.1_cc, volc_iSYK1_cc, volc_iSYK10_cc, volc_iSYK100_cc,
                     volc_iBTK0.1_cc, volc_iBTK1_cc, volc_iBTK10_cc, volc_iBTK100_cc,
                     volc_iPI3Kd0.1_cc, volc_iPI3Kd1_cc, volc_iPI3Kd10_cc, volc_iPI3Kd100_cc,
                     volc_iNFkB0.1_cc, volc_iNFkB1_cc, volc_iNFkB10_cc, volc_iNFkB100_cc,
                     volc_cell_iSYK_cc, volc_cell_iBTK_cc, volc_cell_iPI3Kd_cc, volc_cell_iNFkB_cc)
titles_volcano &lt;- c(&quot;OCI-Ly8 vs HBL1 (all inhibitor together? w/o stim)&quot;, 
                        &quot;0.1 uM iSYK vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;1 uM iSYK vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;10 uM iSYK vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;100 uM iSYK vs DMSO ctrl (both cell lines w/o stim)&quot;,
                        &quot;0.1 uM iBTK vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;1 uM iBTK vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;10 uM iBTK vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;100 uM iBTK vs DMSO ctrl (both cell lines w/o stim)&quot;,
                        &quot;0.1 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)&quot;,
                        &quot;0.1 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)&quot;,
                        &quot;Effect of OCI-Ly8 on 100 uM iSYK vs DMSO ctrl (w/o stim)&quot;, &quot;Effect of OCI-Ly8 on 100 uM iBTK vs DMSO ctrl (w/o stim)&quot;, &quot;Effect of OCI-Ly8 on 100 uM iPI3Kd vs DMSO ctrl (w/o stim)&quot;, &quot;Effect of OCI-Ly8 on 100 uM iNFkB vs DMSO ctrl (w/o stim)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-1">
Past versions of fig_volcano_bas_cell_conc-1.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-2">
Past versions of fig_volcano_bas_cell_conc-2.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-3">
Past versions of fig_volcano_bas_cell_conc-3.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-4">
Past versions of fig_volcano_bas_cell_conc-4.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-5">
Past versions of fig_volcano_bas_cell_conc-5.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-6">
Past versions of fig_volcano_bas_cell_conc-6.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-7">
Past versions of fig_volcano_bas_cell_conc-7.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-8">
Past versions of fig_volcano_bas_cell_conc-8.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-9">
Past versions of fig_volcano_bas_cell_conc-9.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-10">
Past versions of fig_volcano_bas_cell_conc-10.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-11">
Past versions of fig_volcano_bas_cell_conc-11.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-12">
Past versions of fig_volcano_bas_cell_conc-12.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-13">
Past versions of fig_volcano_bas_cell_conc-13.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-14">
Past versions of fig_volcano_bas_cell_conc-14.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-15">
Past versions of fig_volcano_bas_cell_conc-15.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-16">
Past versions of fig_volcano_bas_cell_conc-16.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-17">
Past versions of fig_volcano_bas_cell_conc-17.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-18.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-18">
Past versions of fig_volcano_bas_cell_conc-18.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-19.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-19">
Past versions of fig_volcano_bas_cell_conc-19.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-20.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-20">
Past versions of fig_volcano_bas_cell_conc-20.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_bas_cell_conc-21.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_bas_cell_conc-21">
Past versions of fig_volcano_bas_cell_conc-21.png
</button>
</p>
<div id="fig-fig_volcano_bas_cell_conc-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons: each inhibitor conc vs DMSO ctrl and the effect of OCI-Ly8
vs HBL1 on inhibitor conc vs DMSo ctrl. iSYK shows 2 protein going down
at 1 uM, and more at increasingly higher conc. iBTK starts showing
significantly changing proteins at 1 uM. iPI3Kd shows almost no
significantly changing proteins at all conc. iNFkB shows several
proteins significantly changing at 1 uM, and more at increasingly higher
conc. iSYK affects many proteins significantly less in OCI-Ly8 cells,
whereas the other inhibitors show only a few proteins that are
differently affected in OCI-Ly8 cells.</em></p>
</div>
<div id="heatmap-3" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_cell_icc&quot;, 
                        &quot;heat_iSYK0.1_cc&quot;, &quot;heat_iSYK1_cc&quot;, &quot;heat_iSYK10_cc&quot;, &quot;heat_iSYK100_cc&quot;,
                        &quot;heat_iBTK0.1_cc&quot;, &quot;heat_iBTK1_cc&quot;, &quot;heat_iBTK10_cc&quot;, &quot;heat_iBTK100_cc&quot;,
                        &quot;heat_iPI3Kd0.1_cc&quot;, &quot;heat_iPI3Kd1_cc&quot;, &quot;heat_iPI3Kd10_cc&quot;, &quot;heat_iPI3Kd100_cc&quot;,
                        &quot;heat_iNFkB0.1_cc&quot;, &quot;heat_iNFkB1_cc&quot;, &quot;heat_iNFkB10_cc&quot;, &quot;heat_iNFkB100_cc&quot;,
                        &quot;heat_cell_iSYK_cc&quot;, &quot;heat_cell_iBTK_cc&quot;, &quot;heat_cell_iPI3Kd_cc&quot;, &quot;heat_cell_iNFkB_cc&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_100uM_cc &lt;- list(heat_iSYK100_cc, heat_iBTK100_cc, heat_iPI3Kd100_cc, heat_iNFkB100_cc) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  distinct()

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_cell_inhib_cc &lt;- list(heat_cell_iSYK_cc, heat_cell_iBTK_cc, heat_cell_iPI3Kd_cc, heat_cell_iNFkB_cc) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;cell_line&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(cell_line, factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc)), 
                       cell_line = colors_cell
)

# 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_100uM_cc &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_cc$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_100uM_cc,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_cc)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significant proteins with 100 uM inhibitor (both cell lines w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_cell_conc-1.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_cell_conc-1">
Past versions of fig_heatmap_bas_cell_conc-1.png
</button>
</p>
<div id="fig-fig_heatmap_bas_cell_conc-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Effect of OCI-Ly8 on 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_cell_inhib_cc &lt;- t(scale(t(assay(rld)[unique(c(heat_cell_inhib_cc$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_cell_inhib_cc,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = paste(data_heatmap$inhib_new, data_heatmap$cell_line),
  row_split = subset(heat_cell_inhib_cc)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = &quot;Significantly different proteins in OCI-Ly8 with 100 uM inhibitor (w/o stim)&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_bas_cell_conc-2.png" width="2112" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_bas_cell_conc-2">
Past versions of fig_heatmap_bas_cell_conc-2.png
</button>
</p>
<div id="fig-fig_heatmap_bas_cell_conc-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
the 100 uM inhibitor vs DMSO comparison and the effect of OCI-Ly8 on 100
uM inhibitor vs DMSO ctrl comparison (no threshold for fold
change).</em></p>
</div>
</div>
</div>
<div id="protein-comparisons-1" class="section level3">
<h3>Protein comparisons</h3>
<p>Finally, we visualised the normalised true counts of all significant
proteins in the OCI-Ly8 vs HBL1 comparison (80 total).</p>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Datasets generated during analysis
# heat_xxx: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_xxx: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# List of all significantly changing proteins
sign_prot_100uM_H &lt;- sort(unique(c(heat_100uM_H$target_nospace)))
sign_prot_100uM_O &lt;- sort(unique(c(heat_100uM_O$target_nospace)))
sign_prot_100uM &lt;- sort(unique(c(sign_prot_100uM_H, sign_prot_100uM_O)))
sign_prot_100uM_cc &lt;- sort(unique(c(heat_100uM_cc$target_nospace)))
sign_prot_cell_cc &lt;- sort(unique(c(heat_cell_inhib_cc$target_nospace)))

sign_prot_all &lt;- sort(unique(c(sign_prot_100uM, sign_prot_cell_cc)))

# print(c(&quot;Significant proteins (OCI-ly8 vs HBL1 basal signaling):&quot;, sign_prot_cell))</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
# 100 uM inhibitor in HBL1
heat_list_H &lt;- list(heat_iSYK100_H, heat_iBTK100_H, heat_iPI3Kd100_H, heat_iNFkB100_H)
heat_names &lt;- c(&quot;sign_data_iSYK&quot;, &quot;sign_data_iBTK&quot;, &quot;sign_data_iPI3Kd&quot;, &quot;sign_data_iNFkB&quot;)
heat_suffix &lt;- c(&quot;_iSYK&quot;, &quot;_iBTK&quot;, &quot;_iPI3Kd&quot;, &quot;_iNFkB&quot;)
for (heat_data in c(1:length(heat_list_H))) {
  # print(heat_data)
  alt_data &lt;- heat_list_H[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_100uM_H) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;base_mean&quot;, &quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_100uM_H &lt;- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# 100 uM inhibitor in OCI-Ly8
heat_list_O &lt;- list(heat_iSYK100_O, heat_iBTK100_O, heat_iPI3Kd100_O, heat_iNFkB100_O)
heat_names &lt;- c(&quot;sign_data_iSYK&quot;, &quot;sign_data_iBTK&quot;, &quot;sign_data_iPI3Kd&quot;, &quot;sign_data_iNFkB&quot;)
heat_suffix &lt;- c(&quot;_iSYK&quot;, &quot;_iBTK&quot;, &quot;_iPI3Kd&quot;, &quot;_iNFkB&quot;)
for (heat_data in c(1:length(heat_list_O))) {
  # print(heat_data)
  alt_data &lt;- heat_list_O[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_100uM_H) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;base_mean&quot;, &quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_100uM_O &lt;- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of cell line on 100 uM inhibitor
heat_list_cc &lt;- list(heat_cell_iSYK_cc, heat_cell_iBTK_cc, heat_cell_iPI3Kd_cc, heat_cell_iNFkB_cc)
heat_names &lt;- c(&quot;sign_data_iSYK&quot;, &quot;sign_data_iBTK&quot;, &quot;sign_data_iPI3Kd&quot;, &quot;sign_data_iNFkB&quot;)
heat_suffix &lt;- c(&quot;_iSYK&quot;, &quot;_iBTK&quot;, &quot;_iPI3Kd&quot;, &quot;_iNFkB&quot;)
for (heat_data in c(1:length(heat_list_cc))) {
  # print(heat_data)
  alt_data &lt;- heat_list_cc[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_100uM_H) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;base_mean&quot;, &quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_cell_cc &lt;- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
# 100 uM inhibitor vs DMSO ctrl
alt_data_100uM_H &lt;- data.frame(data_100uM_H) %&gt;% 
  mutate(target_nospace = rownames(data_100uM_H)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_HBL1&quot;, 2:c(ncol(data_100uM_H)+1))
alt_data_100uM_O &lt;- data.frame(data_100uM_O) %&gt;% 
  mutate(target_nospace = rownames(data_100uM_O)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_Ly8&quot;, 2:c(ncol(data_100uM_O)+1))
clean_data_well_100uM &lt;- full_join(alt_data_100uM_H, alt_data_100uM_O) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_100uM) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_HBL1, heatmap_Ly8) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

# Effect of cell line on 100 uM inhibitor
clean_data_well_cell_cc &lt;- data.frame(data_cell_inhib_cc) %&gt;% 
  mutate(target_nospace = rownames(data_cell_inhib_cc)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value&quot;, 2:c(ncol(data_cell_inhib_cc)+1)) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_cell_cc) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

# All signifocant proteins for plotting
clean_data_well &lt;- counts_dtbl %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_all &amp; stimulus == &quot;PBS&quot;) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with mean values per condition (replicates combined)
# 100 uM inhibitor vs DMSO ctrl
clean_data_mean_100uM &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_100uM &amp; stimulus == &quot;PBS&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# Effect of cell line on 100 uM inhibitor
clean_data_mean_cell_cc &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_cell_cc &amp; stimulus == &quot;PBS&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# All signifocant proteins for plotting
clean_data_mean &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_all &amp; stimulus == &quot;PBS&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)</code></pre>
<pre class="r"><code>plot &lt;- ggplot(clean_data_well, 
               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                   counts_norm, 
                   color = cell_line, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_cell, name = &quot;Cell line&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib_all-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib_all-1">
Past versions of fig_counts_bas_inhib_all-1.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Normalised counts for all significantly
changing proteins. For some proteins the effect size is quite
small.</em></p>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

# Re-redesigned fig
filt_clean_data_mean &lt;- clean_data_mean %&gt;%
  dplyr::group_by(cell_line, inhibitor, inhib_conc, target_nospace) %&gt;%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_all) {
  plot &lt;- ggplot() +
    geom_jitter(data = subset(clean_data_well, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
                    counts_norm, 
                    color = inhibitor, 
                    # shape = cell_line
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = inhibitor,
                      # color = inhibitor
                      ),
                  width = 0.5, size = 0.5, color = &quot;black&quot;) + #
    # facet_wrap(vars(factor(target_nospace))) +
    # facet_wrap(vars(factor(inhibitor, levels = inhib_DS108)), ncol = 1) +
    facet_grid(vars(factor(cell_line, levels = cells_DS108)), vars(factor(inhibitor, levels = inhib_DS108))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_inhib, name = &quot;Inhibitors&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;, title = protein) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-1">
Past versions of fig_counts_bas_inhib-1.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-2">
Past versions of fig_counts_bas_inhib-2.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-3">
Past versions of fig_counts_bas_inhib-3.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-4">
Past versions of fig_counts_bas_inhib-4.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-5">
Past versions of fig_counts_bas_inhib-5.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-6.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-6">
Past versions of fig_counts_bas_inhib-6.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-7.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-7">
Past versions of fig_counts_bas_inhib-7.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-8.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-8">
Past versions of fig_counts_bas_inhib-8.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-9.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-9">
Past versions of fig_counts_bas_inhib-9.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-10.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-10">
Past versions of fig_counts_bas_inhib-10.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-11.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-11">
Past versions of fig_counts_bas_inhib-11.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-12.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-12">
Past versions of fig_counts_bas_inhib-12.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-13.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-13">
Past versions of fig_counts_bas_inhib-13.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-14.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-14">
Past versions of fig_counts_bas_inhib-14.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-15.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-15">
Past versions of fig_counts_bas_inhib-15.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-16.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-16">
Past versions of fig_counts_bas_inhib-16.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-17.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-17">
Past versions of fig_counts_bas_inhib-17.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-18.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-18">
Past versions of fig_counts_bas_inhib-18.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-19.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-19">
Past versions of fig_counts_bas_inhib-19.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-20.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-20">
Past versions of fig_counts_bas_inhib-20.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-21.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-21">
Past versions of fig_counts_bas_inhib-21.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-22.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-22">
Past versions of fig_counts_bas_inhib-22.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-23.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-23">
Past versions of fig_counts_bas_inhib-23.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-24.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-24">
Past versions of fig_counts_bas_inhib-24.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-25.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-25">
Past versions of fig_counts_bas_inhib-25.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-26.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-26">
Past versions of fig_counts_bas_inhib-26.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-27.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-27">
Past versions of fig_counts_bas_inhib-27.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-28.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-28">
Past versions of fig_counts_bas_inhib-28.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-29.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-29">
Past versions of fig_counts_bas_inhib-29.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-30.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-30">
Past versions of fig_counts_bas_inhib-30.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-31.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-31">
Past versions of fig_counts_bas_inhib-31.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-32.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-32">
Past versions of fig_counts_bas_inhib-32.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-33.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-33">
Past versions of fig_counts_bas_inhib-33.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-34.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-34">
Past versions of fig_counts_bas_inhib-34.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-35.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-35">
Past versions of fig_counts_bas_inhib-35.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-36.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-36">
Past versions of fig_counts_bas_inhib-36.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-37.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-37">
Past versions of fig_counts_bas_inhib-37.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-38.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-38">
Past versions of fig_counts_bas_inhib-38.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-39.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-39">
Past versions of fig_counts_bas_inhib-39.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-40.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-40">
Past versions of fig_counts_bas_inhib-40.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-41.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-41">
Past versions of fig_counts_bas_inhib-41.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-41" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-42.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-42">
Past versions of fig_counts_bas_inhib-42.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-42" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-43.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-43">
Past versions of fig_counts_bas_inhib-43.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-43" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-44.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-44">
Past versions of fig_counts_bas_inhib-44.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-44" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-45.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-45">
Past versions of fig_counts_bas_inhib-45.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-45" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-46.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-46">
Past versions of fig_counts_bas_inhib-46.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-46" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-47.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-47">
Past versions of fig_counts_bas_inhib-47.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-47" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-48.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-48">
Past versions of fig_counts_bas_inhib-48.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-48" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-49.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-49">
Past versions of fig_counts_bas_inhib-49.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-49" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-50.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-50">
Past versions of fig_counts_bas_inhib-50.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-50" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-51.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-51">
Past versions of fig_counts_bas_inhib-51.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-51" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-52.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-52">
Past versions of fig_counts_bas_inhib-52.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-52" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-53.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-53">
Past versions of fig_counts_bas_inhib-53.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-53" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-54.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-54">
Past versions of fig_counts_bas_inhib-54.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-54" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-55.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-55">
Past versions of fig_counts_bas_inhib-55.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-55" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-56.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-56">
Past versions of fig_counts_bas_inhib-56.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-56" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-57.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-57">
Past versions of fig_counts_bas_inhib-57.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-57" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-58.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-58">
Past versions of fig_counts_bas_inhib-58.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-58" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-59.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-59">
Past versions of fig_counts_bas_inhib-59.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-59" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-60.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-60">
Past versions of fig_counts_bas_inhib-60.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-60" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-61.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-61">
Past versions of fig_counts_bas_inhib-61.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-61" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-62.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-62">
Past versions of fig_counts_bas_inhib-62.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-62" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-63.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-63">
Past versions of fig_counts_bas_inhib-63.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-63" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-64.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-64">
Past versions of fig_counts_bas_inhib-64.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-64" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-65.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-65">
Past versions of fig_counts_bas_inhib-65.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-65" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-66.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-66">
Past versions of fig_counts_bas_inhib-66.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-66" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-67.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-67">
Past versions of fig_counts_bas_inhib-67.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-67" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-68.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-68">
Past versions of fig_counts_bas_inhib-68.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-68" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-69.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-69">
Past versions of fig_counts_bas_inhib-69.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-69" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-70.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-70">
Past versions of fig_counts_bas_inhib-70.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-70" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-71.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-71">
Past versions of fig_counts_bas_inhib-71.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-71" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-72.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-72">
Past versions of fig_counts_bas_inhib-72.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-72" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-73.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-73">
Past versions of fig_counts_bas_inhib-73.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-73" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-74.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-74">
Past versions of fig_counts_bas_inhib-74.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-74" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-75.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-75">
Past versions of fig_counts_bas_inhib-75.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-75" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-76.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-76">
Past versions of fig_counts_bas_inhib-76.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-76" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-77.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-77">
Past versions of fig_counts_bas_inhib-77.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-77" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-78.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-78">
Past versions of fig_counts_bas_inhib-78.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-78" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-79.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-79">
Past versions of fig_counts_bas_inhib-79.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-79" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_bas_inhib-80.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_bas_inhib-80">
Past versions of fig_counts_bas_inhib-80.png
</button>
</p>
<div id="fig-fig_counts_bas_inhib-80" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># for(protein in sign_prot_all[68]) {
#   plot &lt;- ggplot() +
#     geom_jitter(data = subset(clean_data_well, target_nospace == protein), 
#                 aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
#                     counts_norm, 
#                     color = cell_line, 
#                     shape = cell_line), 
#                 alpha = 1, size = 1.5, width = 0.1) +
#     # geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
#     #               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
#     #                   ymin = mean_counts_norm,
#     #                   ymax = mean_counts_norm,
#     #                   group = cell_line, 
#     #                   color = cell_line),
#     #               width = 0.5, size = 1) + #, color = &quot;black&quot;
#     geom_line(data = subset(filt_clean_data_mean, target_nospace == protein),
#               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
#                   mean_counts_norm, 
#                   group = cell_line, 
#                   color = cell_line)) +
#     # facet_wrap(vars(factor(target_nospace))) +
#     facet_grid(vars(factor(inhibitor, levels = inhib_DS108))) +
#     # facet_wrap(vars(factor(cell_line, levels = cells_DS108)), ncol = 1) +
#     # scale_y_log10() +
#     scale_color_manual(values = colors_cell, name = &quot;Cell line&quot;) + 
#     labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;, title = protein) +
#     theme_bw() +
#     theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
#     textsize_small
#   
#   print(plot)
# }</code></pre>
</div>
<div id="save-data-1" class="section level3">
<h3>Save data</h3>
<p>Save all information and datasets as .csv file</p>
<p>100 uM inhibitor vs DMSO ctrl comparisons (done separately for each
cell line):</p>
<ul>
<li><p>100uM_sign_proteins.txt: List of significantly changing proteins
in the 100 uM inhibitor vs DMSO ctrl comparisons (in at least one of the
cell lines)</p></li>
<li><p>100uM_HBL1_sign_DESeq2data.csv: Dataset with significantly
changing proteins, and log2FC + pvalues from DESeq2 analysis (HBL1
samples only)</p></li>
<li><p>100uM_Ly8_sign_DESeq2data.csv: Dataset with significantly
changing proteins, and log2FC + pvalues from DESeq2 analysis (OCI-Ly8
samples only)</p></li>
<li><p>100uM_sign_sample.csv: Dataset of counts per sample per protein +
heatmap data for significantly changing proteins (in at least one of the
cell lines)</p></li>
<li><p>100uM_sign_condition.csv: Dataset of counts per condition (mean
of repl) per protein for significantly changing proteins (in at least
one of the cell lines)</p></li>
</ul>
<p>Effect of cell line on 100 uM inhibitor vs DMSO ctrl:</p>
<ul>
<li><p>cell_100uM_sign_proteins.txt: List of significantly different
proteins in OCI-Ly8 vs HBL1 when looking at the 100 uM inhibitor vs DMSO
ctrl comparisons</p></li>
<li><p>cell_100uM_sign_DESeq2data.csv: Dataset with significantly
different proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>cell_100uM_sign_sample.csv: Dataset of counts per sample per
protein + heatmap data for significantly changing proteins</p></li>
<li><p>cell_100uM_sign_condition.csv: Dataset of counts per condition
(mean of repl) per protein for significantly changing proteins</p></li>
</ul>
<pre class="r"><code># List of significantly changing proteins: sign_prot_cell
write(sign_prot_100uM, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_sign_proteins.txt&quot;)
write(sign_prot_cell_cc, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_proteins.txt&quot;)

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_100uM_H, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_HBL1_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_100uM_O, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_Ly8_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_cell_cc, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_DESeq2data.csv&quot;, row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_100uM, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_sign_sample.csv&quot;, row.names = F)
write.csv(clean_data_well_cell_cc, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_sample.csv&quot;, row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_100uM, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_sign_condition.csv&quot;, row.names = F)
write.csv(clean_data_mean_cell_cc, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9859997 526.6   16263484 868.6 16263484 868.6
Vcells 25520317 194.8   54239823 413.9 39895894 304.4</code></pre>
</div>
</div>
<div id="activated-cellsignaling-state" class="section level2">
<h2>Activated cell/signaling state</h2>
<p>Now, we will look at the activated signaling levels in both HBL1 and
OCI-Ly8 cells (without any inhibitor/DMSO, but after 20 min incubation
with either PBS or aIg+H2O2). We can investigate the effect of
activation and differences between the cell lines.</p>
<div id="deseq2-2" class="section level3 tabset">
<h3 class="tabset">DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="interaction" class="section level4">
<h4>Interaction</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>0 uM inhibitor (PBS ctrls), PBS and aIg+H2O2 stimulus, both cell
lines samples (48 samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_line (ref: HBL1) + stimulus (ref: PBS) +
cell_line:stimulus</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- counts_dtbl  %&gt;%
  filter(inhib_conc == &quot;0 uM (PBS)&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(meta_wellID) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         stimulus = factor(stimulus, levels = stim_DS108), 
         cell_inhib = paste(cell_line, inhibitor)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ cell_line + stimulus + cell_line:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_line &lt;- relevel(dds$cell_line, ref = &quot;HBL1&quot;)
dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$cell_inhib &lt;- relevel(dds$cell_inhib, ref = &quot;HBL1 iSYK&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code>[1] &quot;Intercept&quot;                         &quot;cell_line_OCI.Ly8_vs_HBL1&quot;        
[3] &quot;stimulus_aIg.H2O2_vs_PBS&quot;          &quot;cell_lineOCI.Ly8.stimulusaIg.H2O2&quot;</code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_cell &lt;- results(dds, name = &quot;cell_line_OCI.Ly8_vs_HBL1&quot;, alpha = 0.05)
results_stim_i &lt;- results(dds, name = &quot;stimulus_aIg.H2O2_vs_PBS&quot;, alpha = 0.05)
results_cell_stim &lt;- results(dds, name = &quot;cell_lineOCI.Ly8.stimulusaIg.H2O2&quot;, alpha = 0.05)

list_results &lt;- list(results_cell, results_stim_i, results_cell_stim)

# Results summary
summary(results_cell, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 18, 17%
LFC &lt; 0 (down)     : 22, 21%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 533)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_cell, alpha = signcutoff, main = &quot;OCI-Ly8 vs HBL1&quot;)
plotMA(results_stim_i, alpha = signcutoff, main = &quot;aIg+H2O2 vs PBS&quot;)
plotMA(results_cell_stim, alpha = signcutoff, main = &quot;effect of OCI-Ly8 cell line\non aIg+H2O2 vs PBS&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_act_int-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_int-1">
Past versions of fig_log2FC_act_int-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-4" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-4" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_conc&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhib_conc) +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_fill_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17), name = &quot;Cell line&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_int-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_int-1">
Past versions of fig_PCA_act_int-1.png
</button>
</p>
<div id="fig-fig_PCA_act_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_int-2.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_int-2">
Past versions of fig_PCA_act_int-2.png
</button>
</p>
<div id="fig-fig_PCA_act_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_int-3.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_int-3">
Past versions of fig_PCA_act_int-3.png
</button>
</p>
<div id="fig-fig_PCA_act_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 shows a clear separation on stimulus, and PC2
separates the cell lines. The different types of data transformation
also give similar results.</em></p>
</div>
<div id="variance-4" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_int-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_int-1">
Past versions of fig_sd_act_int-1.png
</button>
</p>
<div id="fig-fig_sd_act_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Norm (log2(n+1))&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_int-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_int-2">
Past versions of fig_sd_act_int-2.png
</button>
</p>
<div id="fig-fig_sd_act_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_int-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_int-3">
Past versions of fig_sd_act_int-3.png
</button>
</p>
<div id="fig-fig_sd_act_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_int-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_int-4">
Past versions of fig_sd_act_int-4.png
</button>
</p>
<div id="fig-fig_sd_act_int-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-4" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$cell_line,
                                      list_datatransform[[data]]$stimulus,
                                      # list_datatransform[[data]]$inhibitor,
                                      list_datatransform[[data]]$replicate,
                                      sep = &quot; - &quot;)
  # rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_int-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_int-1">
Past versions of fig_distances_act_int-1.png
</button>
</p>
<div id="fig-fig_distances_act_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_int-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_int-2">
Past versions of fig_distances_act_int-2.png
</button>
</p>
<div id="fig-fig_distances_act_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_int-3.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_int-3">
Past versions of fig_distances_act_int-3.png
</button>
</p>
<div id="fig-fig_distances_act_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-4" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_cell&quot;, &quot;volc_stim_i&quot;, &quot;volc_cell_stim&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_cell, volc_stim_i, volc_cell_stim)
titles_volcano &lt;- c(&quot;OCI-Ly8 vs HBL1 (basal + activated signaling w/o inhibitor)&quot;, 
                    &quot;aIg+H2O2 vs PBS (both cell lines, w/o inhibitor)&quot;, 
                    &quot;Effect of OCI-Ly8 (vs HBL1) on aIg+H2O2 vs PBS (w/o inhibitor)&quot;)
# titles_volcano &lt;- names_results_volc

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_int-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_int-1">
Past versions of fig_volcano_act_int-1.png
</button>
</p>
<div id="fig-fig_volcano_act_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_int-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_int-2">
Past versions of fig_volcano_act_int-2.png
</button>
</p>
<div id="fig-fig_volcano_act_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_int-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_int-3">
Past versions of fig_volcano_act_int-3.png
</button>
</p>
<div id="fig-fig_volcano_act_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons.</em></p>
<p><em>As also seen in the basal signaling comparison, several
(phospho)proteins are significantly different in the OCI-Ly8 cells
compared to HBL1 cells. As expected, upon aIg+H2O2 stimulation, many
(phospho)proteins are increased (both cell line together).
Interestingly, many proteins are significantly less activated in OCI-Ly8
cells than in HBL1 cells, but a few proteins are activated more in
OCI0-Ly8 cells.</em></p>
</div>
<div id="heatmap-4" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_cell&quot;, &quot;heat_stim_i&quot;, &quot;heat_cell_stim&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;cell_line&quot;, &quot;stimulus&quot;)]) %&gt;%
  arrange(cell_line, factor(stimulus, levels = stim_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# aIg+H2O2 vs PBS
heat_stim_i$marker_pathway[is.na(heat_stim_i$marker_pathway)] &lt;- &quot;Other&quot;

ann_colors_man &lt;- list(cell_line = colors_cell,
                       stimulus = colors_stim,
                       # modification = c(phospho = &quot;#f768a1&quot;, no = &quot;grey&quot;, cleaved = &quot;#fff7bc&quot;),
                       marker_pathway = setNames(colors_light12[1:length(unique(heat_stim_i$marker_pathway))], unique(heat_stim_i$marker_pathway))
)

# Get the rld transformed data for the heatmap and scale per row
data_stim_i &lt;- t(scale(t(assay(rld)[unique(c(heat_stim_i$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_stim_i,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  annotation_row = heat_stim_i[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$cell_line,
  row_split = subset(heat_stim_i)$modification,
  cellwidth = 10,
  cellheight = 10,
  fontsize = 8,
  border_color = NA,
  # main = &quot;scaled per row, ComplexHeatmap&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_int-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_int-1">
Past versions of fig_heatmap_act_int-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
the aIg+H2O2 vs PBS comparison (no threshold for fold change).</em></p>
<p><em>Focussing on the phospho-proteins: Many upstream BCR proteins are
activated in the HBL1 cells and to a lesser extend in the OCI-Ly8 cells.
Some MAPK-JNK signaling proteins are activated more in the OCI-Ly8
cells. In addition, several JAK-STAT, ERK, and cell cylcle
phospho-proteins are decreased upon stimulation, in both cell
lines.</em></p>
</div>
</div>
<div id="separate-comp" class="section level4">
<h4>Separate comp</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>0 uM inhibitor (PBS ctrls), PBS and aIg+H2O2 stimulus, both cell
lines samples (48 samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_stim (ref: HBL1 PBS)</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(cell_stim = paste(cell_line, stimulus))

cts &lt;- counts_dtbl  %&gt;%
  filter(inhib_conc == &quot;0 uM (PBS)&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, cell_stim)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         stimulus = factor(stimulus, levels = stim_DS108), 
         cell_inhib = paste(cell_line, inhibitor)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ cell_stim

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_stim &lt;- relevel(dds$cell_stim, ref = &quot;HBL1 PBS&quot;)
# dds$cell_inhib &lt;- relevel(dds$cell_inhib, ref = &quot;HBL1 iSYK&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code>[1] &quot;Intercept&quot;                             
[2] &quot;cell_stim_HBL1.aIg.H2O2_vs_HBL1.PBS&quot;   
[3] &quot;cell_stim_OCI.Ly8.aIg.H2O2_vs_HBL1.PBS&quot;
[4] &quot;cell_stim_OCI.Ly8.PBS_vs_HBL1.PBS&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_HBL1 &lt;- results(dds, name = &quot;cell_stim_HBL1.aIg.H2O2_vs_HBL1.PBS&quot;, alpha = 0.05)
results_Ly8 &lt;- results(dds, contrast = c(&quot;cell_stim&quot;, &quot;OCI-Ly8 aIg+H2O2&quot;, &quot;OCI-Ly8 PBS&quot;), alpha = 0.05)

results_PBS &lt;- results(dds, name = &quot;cell_stim_OCI.Ly8.PBS_vs_HBL1.PBS&quot;, alpha = 0.05)
results_stim &lt;- results(dds, contrast = c(&quot;cell_stim&quot;, &quot;OCI-Ly8 aIg+H2O2&quot;, &quot;HBL1 aIg+H2O2&quot;), alpha = 0.05)

list_results &lt;- list(results_HBL1, results_Ly8, results_PBS, results_stim)

# Results summary
summary(results_HBL1, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 32, 30%
LFC &lt; 0 (down)     : 41, 38%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 533)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_HBL1, alpha = signcutoff, main = &quot;HBL1:\naIg+H2O2 vs PBS&quot;)
plotMA(results_Ly8, alpha = signcutoff, main = &quot;OCI-Ly8:\naIg+H2O2 vs PBS&quot;)
plotMA(results_stim, alpha = signcutoff, main = &quot;aIg+H2O2:\nOCI-Ly8 vs HBL1&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_act_sep-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_sep-1">
Past versions of fig_log2FC_act_sep-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-5" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-5" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_conc&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhib_conc) +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_fill_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17), name = &quot;Cell line&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_sep-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_sep-1">
Past versions of fig_PCA_act_sep-1.png
</button>
</p>
<div id="fig-fig_PCA_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_sep-2.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_sep-2">
Past versions of fig_PCA_act_sep-2.png
</button>
</p>
<div id="fig-fig_PCA_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_sep-3.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_sep-3">
Past versions of fig_PCA_act_sep-3.png
</button>
</p>
<div id="fig-fig_PCA_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 shows a clear separation on stimulus, and PC2
separates the cell lines. The different types of data transformation
also give similar results.</em></p>
</div>
<div id="variance-5" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_sep-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-1">
Past versions of fig_sd_act_sep-1.png
</button>
</p>
<div id="fig-fig_sd_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Norm (log2(n+1))&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_sep-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-2">
Past versions of fig_sd_act_sep-2.png
</button>
</p>
<div id="fig-fig_sd_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_sep-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-3">
Past versions of fig_sd_act_sep-3.png
</button>
</p>
<div id="fig-fig_sd_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_sep-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-4">
Past versions of fig_sd_act_sep-4.png
</button>
</p>
<div id="fig-fig_sd_act_sep-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-5" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$cell_line,
                                      list_datatransform[[data]]$stimulus,
                                      # list_datatransform[[data]]$inhibitor,
                                      list_datatransform[[data]]$replicate,
                                      sep = &quot; - &quot;)
  # rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_sep-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_sep-1">
Past versions of fig_distances_act_sep-1.png
</button>
</p>
<div id="fig-fig_distances_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_sep-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_sep-2">
Past versions of fig_distances_act_sep-2.png
</button>
</p>
<div id="fig-fig_distances_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_sep-3.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_sep-3">
Past versions of fig_distances_act_sep-3.png
</button>
</p>
<div id="fig-fig_distances_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-5" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_HBL1&quot;, &quot;volc_Ly8&quot;, &quot;volc_PBS&quot;, &quot;volc_stim&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_HBL1, volc_Ly8, volc_PBS, volc_stim)
titles_volcano &lt;- c(&quot;HBL1: aIg+H2O2 vs PBS (w/o inhibitor)&quot;,
                    &quot;OCI-Ly8: aIg+H2O2 vs PBS (w/o inhibitor)&quot;,
                    &quot;OCI-Ly8 vs HBL1 (basal signaling w/o inhibitor)&quot;, 
                    &quot;OCI-Ly8 vs HBL1 (activated signaling w/o inhibitor)&quot;)
# titles_volcano &lt;- names_results_volc

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_sep-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-1">
Past versions of fig_volcano_act_sep-1.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_sep-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-2">
Past versions of fig_volcano_act_sep-2.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_sep-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-3">
Past versions of fig_volcano_act_sep-3.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_sep-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-4">
Past versions of fig_volcano_act_sep-4.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons.</em></p>
<p><em>Stimulation in HBL1 cells results in the activation of many
phospho-proteins. Stimulation in OCI-Ly8 cells also results in the
activation of phospho-proteins, but to a much lesser extend (less
proteins and lower fold changes). Comparison of the activated signaling
states of OCI-Ly8 and HBL1 reveals many (phospho)proteins with
significantly lower levels in OCI-Ly8 cells, but also some with higher
levels in OCI-Ly8.</em></p>
</div>
<div id="heatmap-5" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_HBL1&quot;, &quot;heat_Ly8&quot;, &quot;heat_PBS&quot;, &quot;heat_stim&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;cell_line&quot;, &quot;stimulus&quot;)]) %&gt;%
  arrange(cell_line, factor(stimulus, levels = stim_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# aIg+H2O2 vs PBS in HBL1 cells
heat_HBL1$marker_pathway[is.na(heat_HBL1$marker_pathway)] &lt;- &quot;Other&quot;

ann_colors_man &lt;- list(cell_line = colors_cell,
                       stimulus = colors_stim,
                       # modification = c(phospho = &quot;#f768a1&quot;, no = &quot;grey&quot;, cleaved = &quot;#fff7bc&quot;),
                       marker_pathway = setNames(colors_light12[1:length(unique(heat_HBL1$marker_pathway))], unique(heat_HBL1$marker_pathway))
)

# Get the rld transformed data for the heatmap and scale per row
data_HBL1 &lt;- t(scale(t(assay(rld)[unique(c(heat_HBL1$target_nospace)), rownames(data_heatmap)])))
data_Ly8 &lt;- t(scale(t(assay(rld)[unique(c(heat_Ly8$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_HBL1,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  annotation_row = heat_HBL1[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$cell_line,
  row_split = subset(heat_HBL1)$modification,
  cellwidth = 10,
  cellheight = 10,
  fontsize = 8,
  border_color = NA,
  # main = &quot;scaled per row, ComplexHeatmap&quot;
)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_sep-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_sep-1">
Past versions of fig_heatmap_act_sep-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
the aIg+H2O2 vs PBS comparison for HBL1 cells (no threshold for fold
change).</em></p>
<p><em>Focussing on the phospho-proteins: Many upstream BCR proteins are
activated in the HBL1 cells and to a lesser extend in the OCI-Ly8 cells.
Some MAPK-JNK signaling proteins are activated more in the OCI-Ly8
cells. In addition, several JAK-STAT, ERK, and cell cylcle
phospho-proteins are decreased upon stimulation, in both cell
lines.</em></p>
</div>
</div>
</div>
<div id="protein-comparisons-2" class="section level3">
<h3>Protein comparisons</h3>
<p>Finally, we visualised the normalised true counts of all significant
proteins in the aIg+H2O2 vs PBS comparison (85 in total).</p>
<pre class="r"><code># Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins (for the HBL1 cells and the OCI-Ly8 cells)
sign_prot_cell_int &lt;- sort(unique(c(heat_stim_i$target_nospace)))
# sign_prot_cell_sep &lt;- sort(unique(c(heat_HBL1$target_nospace, heat_Ly8$target_nospace)))
# print(c(&quot;Significant proteins (aIg+H2O2 vs PBS in both HBL1 and OCI-Ly8):&quot;, sign_prot_cell_int))

# Datasets generated during analysis
# heat_stim_i: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_stim_i: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
clean_data_well_int &lt;- data.frame(data_stim_i) %&gt;% 
  mutate(target_nospace = rownames(data_stim_i)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value&quot;, 2:c(ncol(data_stim_i)+1)) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_cell_int) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, cell_stim, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

prot_labels_int &lt;- sort(setNames(unique(pull(clean_data_well_int, target)), unique(pull(clean_data_well_int, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
clean_data_sign_int &lt;- heat_stim_i %&gt;%
  filter(target_nospace %in% sign_prot_cell_int) %&gt;%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean = baseMean, 
                log2FC = log2FoldChange, 
                log2FC_se = lfcSE)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean_int &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_cell_int &amp; inhib_conc == &quot;0 uM (PBS)&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)</code></pre>
<pre class="r"><code># Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins (for the HBL1 cells and the OCI-Ly8 cells)
# sign_prot_cell_int &lt;- sort(unique(c(heat_stim_i$target_nospace)))
sign_prot_cell_sep &lt;- sort(unique(c(heat_HBL1$target_nospace, heat_Ly8$target_nospace)))
# print(c(&quot;Significant proteins (aIg+H2O2 vs PBS in HBL1 and OCI-Ly8 separately):&quot;, sign_prot_cell_sep))

# Datasets generated during analysis
# heat_HBL1: DESeq2 data of sign prot for HBL1 (repl combined), log2FC and pvalues per protein
# heat_Ly8: DESeq2 data of sign prot for OCI-Ly8 (repl combined), log2FC and pvalues per protein
# data_HBL1: matrix of data for heatmap for HBL1, well (col) x prot (row) (rld transformed, scaled per row)
# data_Ly8: matrix of data for heatmap for OCI-Ly8, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
heat_data_HBL1 &lt;- data.frame(data_HBL1) %&gt;% 
  mutate(target_nospace = rownames(data_HBL1)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value_HBL1&quot;, 2:c(ncol(data_HBL1)+1))
heat_data_Ly8 &lt;- data.frame(data_Ly8) %&gt;% 
  mutate(target_nospace = rownames(data_Ly8)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value_Ly8&quot;, 2:c(ncol(data_Ly8)+1))

clean_data_well_sep &lt;- full_join(heat_data_HBL1, heat_data_Ly8) %&gt;% 
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_cell_sep) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, cell_stim, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value_HBL1, heatmap_value_Ly8) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap_HBL1 = heatmap_value_HBL1, 
                heatmap_Ly8 = heatmap_value_Ly8)

prot_labels_sep &lt;- sort(setNames(unique(pull(clean_data_well_sep, target)), unique(pull(clean_data_well_sep, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
sign_data_HBL1 &lt;- heat_HBL1 %&gt;%
  filter(target_nospace %in% sign_prot_cell_sep) %&gt;%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean_HBL1 = baseMean, 
                log2FC_HBL1 = log2FoldChange, 
                log2FC_se_HBL1 = lfcSE, 
                stat_HBL1 = stat, 
                pvalue_HBL1 = pvalue,
                padj_HBL1 = padj, 
                diff_express_HBL1 = diff_express)
sign_data_Ly8 &lt;- heat_Ly8 %&gt;%
  filter(target_nospace %in% sign_prot_cell_sep) %&gt;%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean_Ly8 = baseMean, 
                log2FC_Ly8 = log2FoldChange, 
                log2FC_se_Ly8 = lfcSE, 
                stat_Ly8 = stat, 
                pvalue_Ly8 = pvalue,
                padj_Ly8 = padj,
                diff_express_Ly8 = diff_express)
clean_data_sign_sep &lt;- full_join(sign_data_HBL1, sign_data_Ly8)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean_sep &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_cell_sep &amp; inhib_conc == &quot;0 uM (PBS)&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)</code></pre>
<pre class="r"><code># I visualised all significant proteins from the separate analysis (aIg+H2O2 vs PBS for the HBL1 cells and OCI-Ly8 cells separately), as this were more proteins, and all proteins in the interaction analysis also popped up in the separate analysis

plot &lt;- ggplot(clean_data_well_sep, 
               aes(factor(cell_line, levels = cells_DS108),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  scale_y_log10() +
  scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts (log10 scale)&quot;) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_all-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_all-1">
Past versions of fig_counts_act_all-1.png
</button>
</p>
<div id="fig-fig_counts_act_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Normalised counts for all significantly
changing proteins. For some proteins the effect size is quite
small.</em></p>
<pre class="r"><code>options(repr.plot.width = 3.5, repr.plot.height = 2.5)

# Re-redesigned fig
filt_clean_data_mean &lt;- clean_data_mean_sep %&gt;%
  dplyr::group_by(cell_line, stimulus, target_nospace) %&gt;%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_cell_sep) {
  plot &lt;- ggplot() +
    geom_jitter(data = subset(clean_data_well_sep, target_nospace == protein), 
                aes(factor(cell_line, levels = cells_DS108), counts_norm, color = stimulus), 
                alpha = 1, size = 1.5, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
                  aes(factor(cell_line, levels = cells_DS108),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = cell_line, 
                      color = stimulus),
                  width = 0.5, size = 1) +
    facet_wrap(vars(factor(target))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-1.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-1">
Past versions of fig_counts_act-1.png
</button>
</p>
<div id="fig-fig_counts_act-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-2.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-2">
Past versions of fig_counts_act-2.png
</button>
</p>
<div id="fig-fig_counts_act-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-3.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-3">
Past versions of fig_counts_act-3.png
</button>
</p>
<div id="fig-fig_counts_act-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-4.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-4">
Past versions of fig_counts_act-4.png
</button>
</p>
<div id="fig-fig_counts_act-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-5.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-5">
Past versions of fig_counts_act-5.png
</button>
</p>
<div id="fig-fig_counts_act-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-6.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-6">
Past versions of fig_counts_act-6.png
</button>
</p>
<div id="fig-fig_counts_act-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-7.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-7">
Past versions of fig_counts_act-7.png
</button>
</p>
<div id="fig-fig_counts_act-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-8.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-8">
Past versions of fig_counts_act-8.png
</button>
</p>
<div id="fig-fig_counts_act-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-9.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-9">
Past versions of fig_counts_act-9.png
</button>
</p>
<div id="fig-fig_counts_act-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-10.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-10">
Past versions of fig_counts_act-10.png
</button>
</p>
<div id="fig-fig_counts_act-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-11.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-11">
Past versions of fig_counts_act-11.png
</button>
</p>
<div id="fig-fig_counts_act-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-12.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-12">
Past versions of fig_counts_act-12.png
</button>
</p>
<div id="fig-fig_counts_act-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-13.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-13">
Past versions of fig_counts_act-13.png
</button>
</p>
<div id="fig-fig_counts_act-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-14.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-14">
Past versions of fig_counts_act-14.png
</button>
</p>
<div id="fig-fig_counts_act-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-15.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-15">
Past versions of fig_counts_act-15.png
</button>
</p>
<div id="fig-fig_counts_act-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-16.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-16">
Past versions of fig_counts_act-16.png
</button>
</p>
<div id="fig-fig_counts_act-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-17.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-17">
Past versions of fig_counts_act-17.png
</button>
</p>
<div id="fig-fig_counts_act-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-18.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-18">
Past versions of fig_counts_act-18.png
</button>
</p>
<div id="fig-fig_counts_act-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-19.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-19">
Past versions of fig_counts_act-19.png
</button>
</p>
<div id="fig-fig_counts_act-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-20.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-20">
Past versions of fig_counts_act-20.png
</button>
</p>
<div id="fig-fig_counts_act-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-21.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-21">
Past versions of fig_counts_act-21.png
</button>
</p>
<div id="fig-fig_counts_act-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-22.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-22">
Past versions of fig_counts_act-22.png
</button>
</p>
<div id="fig-fig_counts_act-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-23.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-23">
Past versions of fig_counts_act-23.png
</button>
</p>
<div id="fig-fig_counts_act-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-24.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-24">
Past versions of fig_counts_act-24.png
</button>
</p>
<div id="fig-fig_counts_act-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-25.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-25">
Past versions of fig_counts_act-25.png
</button>
</p>
<div id="fig-fig_counts_act-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-26.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-26">
Past versions of fig_counts_act-26.png
</button>
</p>
<div id="fig-fig_counts_act-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-27.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-27">
Past versions of fig_counts_act-27.png
</button>
</p>
<div id="fig-fig_counts_act-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-28.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-28">
Past versions of fig_counts_act-28.png
</button>
</p>
<div id="fig-fig_counts_act-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-29.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-29">
Past versions of fig_counts_act-29.png
</button>
</p>
<div id="fig-fig_counts_act-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-30.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-30">
Past versions of fig_counts_act-30.png
</button>
</p>
<div id="fig-fig_counts_act-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-31.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-31">
Past versions of fig_counts_act-31.png
</button>
</p>
<div id="fig-fig_counts_act-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-32.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-32">
Past versions of fig_counts_act-32.png
</button>
</p>
<div id="fig-fig_counts_act-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-33.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-33">
Past versions of fig_counts_act-33.png
</button>
</p>
<div id="fig-fig_counts_act-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-34.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-34">
Past versions of fig_counts_act-34.png
</button>
</p>
<div id="fig-fig_counts_act-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-35.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-35">
Past versions of fig_counts_act-35.png
</button>
</p>
<div id="fig-fig_counts_act-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-36.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-36">
Past versions of fig_counts_act-36.png
</button>
</p>
<div id="fig-fig_counts_act-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-37.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-37">
Past versions of fig_counts_act-37.png
</button>
</p>
<div id="fig-fig_counts_act-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-38.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-38">
Past versions of fig_counts_act-38.png
</button>
</p>
<div id="fig-fig_counts_act-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-39.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-39">
Past versions of fig_counts_act-39.png
</button>
</p>
<div id="fig-fig_counts_act-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-40.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-40">
Past versions of fig_counts_act-40.png
</button>
</p>
<div id="fig-fig_counts_act-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-41.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-41">
Past versions of fig_counts_act-41.png
</button>
</p>
<div id="fig-fig_counts_act-41" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-42.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-42">
Past versions of fig_counts_act-42.png
</button>
</p>
<div id="fig-fig_counts_act-42" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-43.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-43">
Past versions of fig_counts_act-43.png
</button>
</p>
<div id="fig-fig_counts_act-43" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-44.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-44">
Past versions of fig_counts_act-44.png
</button>
</p>
<div id="fig-fig_counts_act-44" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-45.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-45">
Past versions of fig_counts_act-45.png
</button>
</p>
<div id="fig-fig_counts_act-45" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-46.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-46">
Past versions of fig_counts_act-46.png
</button>
</p>
<div id="fig-fig_counts_act-46" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-47.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-47">
Past versions of fig_counts_act-47.png
</button>
</p>
<div id="fig-fig_counts_act-47" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-48.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-48">
Past versions of fig_counts_act-48.png
</button>
</p>
<div id="fig-fig_counts_act-48" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-49.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-49">
Past versions of fig_counts_act-49.png
</button>
</p>
<div id="fig-fig_counts_act-49" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-50.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-50">
Past versions of fig_counts_act-50.png
</button>
</p>
<div id="fig-fig_counts_act-50" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-51.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-51">
Past versions of fig_counts_act-51.png
</button>
</p>
<div id="fig-fig_counts_act-51" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-52.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-52">
Past versions of fig_counts_act-52.png
</button>
</p>
<div id="fig-fig_counts_act-52" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-53.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-53">
Past versions of fig_counts_act-53.png
</button>
</p>
<div id="fig-fig_counts_act-53" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-54.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-54">
Past versions of fig_counts_act-54.png
</button>
</p>
<div id="fig-fig_counts_act-54" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-55.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-55">
Past versions of fig_counts_act-55.png
</button>
</p>
<div id="fig-fig_counts_act-55" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-56.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-56">
Past versions of fig_counts_act-56.png
</button>
</p>
<div id="fig-fig_counts_act-56" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-57.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-57">
Past versions of fig_counts_act-57.png
</button>
</p>
<div id="fig-fig_counts_act-57" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-58.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-58">
Past versions of fig_counts_act-58.png
</button>
</p>
<div id="fig-fig_counts_act-58" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-59.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-59">
Past versions of fig_counts_act-59.png
</button>
</p>
<div id="fig-fig_counts_act-59" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-60.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-60">
Past versions of fig_counts_act-60.png
</button>
</p>
<div id="fig-fig_counts_act-60" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-61.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-61">
Past versions of fig_counts_act-61.png
</button>
</p>
<div id="fig-fig_counts_act-61" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-62.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-62">
Past versions of fig_counts_act-62.png
</button>
</p>
<div id="fig-fig_counts_act-62" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-63.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-63">
Past versions of fig_counts_act-63.png
</button>
</p>
<div id="fig-fig_counts_act-63" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-64.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-64">
Past versions of fig_counts_act-64.png
</button>
</p>
<div id="fig-fig_counts_act-64" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-65.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-65">
Past versions of fig_counts_act-65.png
</button>
</p>
<div id="fig-fig_counts_act-65" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-66.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-66">
Past versions of fig_counts_act-66.png
</button>
</p>
<div id="fig-fig_counts_act-66" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-67.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-67">
Past versions of fig_counts_act-67.png
</button>
</p>
<div id="fig-fig_counts_act-67" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-68.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-68">
Past versions of fig_counts_act-68.png
</button>
</p>
<div id="fig-fig_counts_act-68" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-69.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-69">
Past versions of fig_counts_act-69.png
</button>
</p>
<div id="fig-fig_counts_act-69" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-70.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-70">
Past versions of fig_counts_act-70.png
</button>
</p>
<div id="fig-fig_counts_act-70" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-71.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-71">
Past versions of fig_counts_act-71.png
</button>
</p>
<div id="fig-fig_counts_act-71" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-72.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-72">
Past versions of fig_counts_act-72.png
</button>
</p>
<div id="fig-fig_counts_act-72" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-73.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-73">
Past versions of fig_counts_act-73.png
</button>
</p>
<div id="fig-fig_counts_act-73" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-74.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-74">
Past versions of fig_counts_act-74.png
</button>
</p>
<div id="fig-fig_counts_act-74" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-75.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-75">
Past versions of fig_counts_act-75.png
</button>
</p>
<div id="fig-fig_counts_act-75" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-76.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-76">
Past versions of fig_counts_act-76.png
</button>
</p>
<div id="fig-fig_counts_act-76" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-77.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-77">
Past versions of fig_counts_act-77.png
</button>
</p>
<div id="fig-fig_counts_act-77" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-78.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-78">
Past versions of fig_counts_act-78.png
</button>
</p>
<div id="fig-fig_counts_act-78" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-79.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-79">
Past versions of fig_counts_act-79.png
</button>
</p>
<div id="fig-fig_counts_act-79" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-80.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-80">
Past versions of fig_counts_act-80.png
</button>
</p>
<div id="fig-fig_counts_act-80" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-81.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-81">
Past versions of fig_counts_act-81.png
</button>
</p>
<div id="fig-fig_counts_act-81" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-82.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-82">
Past versions of fig_counts_act-82.png
</button>
</p>
<div id="fig-fig_counts_act-82" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-83.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-83">
Past versions of fig_counts_act-83.png
</button>
</p>
<div id="fig-fig_counts_act-83" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-84.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-84">
Past versions of fig_counts_act-84.png
</button>
</p>
<div id="fig-fig_counts_act-84" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act-85.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-85">
Past versions of fig_counts_act-85.png
</button>
</p>
<div id="fig-fig_counts_act-85" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-data-2" class="section level3">
<h3>Save data</h3>
<p>Save all information and datasets as .csv file</p>
<p>Interaction analysis for both cell lines together:</p>
<ul>
<li><p>int_sign_proteins.txt: List of significantly changing
proteins</p></li>
<li><p>int_sign_DESeq2data.csv: Dataset with significantly changing
proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>int_sign_sample.csv: Dataset of counts per sample per protein +
heatmap data for significantly changing proteins</p></li>
<li><p>int_sign_mean.csv: Dataset of counts per condition (mean of repl)
per protein for significantly changing proteins</p></li>
</ul>
<p>Separate analysis for each cell line:</p>
<ul>
<li><p>sep_sign_proteins.txt: List of significantly changing
proteins</p></li>
<li><p>sep_sign_DESeq2data.csv: Dataset with significantly changing
proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>sep_sign_sample.csv: Dataset of counts per sample per protein +
heatmap data for significantly changing proteins</p></li>
<li><p>sep_sign_mean.csv: Dataset of counts per condition (mean of repl)
per protein for significantly changing proteins</p></li>
</ul>
<pre class="r"><code># List of significantly changing proteins: sign_prot_cell
write(sign_prot_cell_int, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_proteins.txt&quot;)
write(sign_prot_cell_sep, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_proteins.txt&quot;)

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_int, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_sep, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_DESeq2data.csv&quot;, row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_int, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_sample.csv&quot;, row.names = F)
write.csv(clean_data_well_sep, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_sample.csv&quot;, row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_int, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_condition.csv&quot;, row.names = F)
write.csv(clean_data_mean_sep, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9866945 527.0   16263484 868.6 16263484 868.6
Vcells 25455920 194.3   54239823 413.9 39895894 304.4</code></pre>
</div>
</div>
<div id="inhibitor-effects-on-activated-signaling"
class="section level2">
<h2>Inhibitor effects on activated signaling</h2>
<p>Finally, we look at the effect of inhibitors on activated signaling
state. We can investigate the effect of different inhibitors and
concentrations on activation, in both cell lines separately.</p>
<div id="hbl1-deseq2" class="section level3 tabset">
<h3 class="tabset">HBL1: DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="inhib-concstim" class="section level4">
<h4>Inhib conc:stim</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>All HBL1 samples: 4 inhibitors, 6 conc, 2 stimuli (144
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(cell_line == &quot;HBL1&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 uM (DMSO)&quot;)
dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                   
 [2] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;     
 [3] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;    
 [4] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;   
 [5] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;     
 [6] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;       
 [7] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;      
 [8] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;     
 [9] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;       
[10] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;      
[11] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;     
[12] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;    
[13] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;      
[14] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;     
[15] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;    
[16] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;   
[17] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;     
[18] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;        
[19] &quot;stimulus_aIg.H2O2_vs_PBS&quot;                    
[20] &quot;inhib_text_new0.1.uM.iBTK.stimulusaIg.H2O2&quot;  
[21] &quot;inhib_text_new0.1.uM.iNFkB.stimulusaIg.H2O2&quot; 
[22] &quot;inhib_text_new0.1.uM.iPI3Kd.stimulusaIg.H2O2&quot;
[23] &quot;inhib_text_new0.1.uM.iSYK.stimulusaIg.H2O2&quot;  
[24] &quot;inhib_text_new1.uM.iBTK.stimulusaIg.H2O2&quot;    
[25] &quot;inhib_text_new1.uM.iNFkB.stimulusaIg.H2O2&quot;   
[26] &quot;inhib_text_new1.uM.iPI3Kd.stimulusaIg.H2O2&quot;  
[27] &quot;inhib_text_new1.uM.iSYK.stimulusaIg.H2O2&quot;    
[28] &quot;inhib_text_new10.uM.iBTK.stimulusaIg.H2O2&quot;   
[29] &quot;inhib_text_new10.uM.iNFkB.stimulusaIg.H2O2&quot;  
[30] &quot;inhib_text_new10.uM.iPI3Kd.stimulusaIg.H2O2&quot; 
[31] &quot;inhib_text_new10.uM.iSYK.stimulusaIg.H2O2&quot;   
[32] &quot;inhib_text_new100.uM.iBTK.stimulusaIg.H2O2&quot;  
[33] &quot;inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2&quot; 
[34] &quot;inhib_text_new100.uM.iPI3Kd.stimulusaIg.H2O2&quot;
[35] &quot;inhib_text_new100.uM.iSYK.stimulusaIg.H2O2&quot;  
[36] &quot;inhib_text_newPBS.ctrl.stimulusaIg.H2O2&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_stim_Hi &lt;- results(dds, name = &quot;stimulus_aIg.H2O2_vs_PBS&quot;, alpha = 0.05)

results_PBS_Hi &lt;- results(dds, name = &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK0.1_Hi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_Hi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_Hi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_Hi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_Hi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_Hi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_Hi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_Hi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_Hi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_Hi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_Hi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_Hi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_Hi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_Hi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_Hi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_Hi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK100_stim_Hi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iSYK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iBTK100_stim_Hi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iPI3Kd100_stim_Hi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iPI3Kd.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iNFkB100_stim_Hi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_stim_Hi, results_PBS_Hi, 
                     results_iSYK0.1_Hi, results_iSYK1_Hi, results_iSYK10_Hi, results_iSYK100_Hi, 
                     results_iBTK0.1_Hi, results_iBTK1_Hi, results_iBTK10_Hi, results_iBTK100_Hi, 
                     results_iPI3Kd0.1_Hi, results_iPI3Kd1_Hi, results_iPI3Kd10_Hi, results_iPI3Kd100_Hi, 
                     results_iNFkB0.1_Hi, results_iNFkB1_Hi, results_iNFkB10_Hi, results_iNFkB100_Hi, 
                     results_iSYK100_stim_Hi, results_iBTK100_stim_Hi, results_iPI3Kd100_stim_Hi, results_iNFkB100_stim_Hi)

# Results summary
summary(results_stim_Hi, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 33, 31%
LFC &lt; 0 (down)     : 32, 30%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 407)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_stim_Hi, alpha = signcutoff, main = &quot;aIg+H2O2 vs PBS&quot;)
plotMA(results_iNFkB100_Hi, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)
plotMA(results_iNFkB100_stim_Hi, alpha = signcutoff, main = &quot;Effect of 100 uM iNFkB on\n aIg+H2O2 vs PBS&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_act_HBL1_int-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_HBL1_int-1">
Past versions of fig_log2FC_act_HBL1_int-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_HBL1_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-6" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-6" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    # scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_fill_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_HBL1_int-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_HBL1_int-1">
Past versions of fig_PCA_act_HBL1_int-1.png
</button>
</p>
<div id="fig-fig_PCA_act_HBL1_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_HBL1_int-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_HBL1_int-2">
Past versions of fig_PCA_act_HBL1_int-2.png
</button>
</p>
<div id="fig-fig_PCA_act_HBL1_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_HBL1_int-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_HBL1_int-3">
Past versions of fig_PCA_act_HBL1_int-3.png
</button>
</p>
<div id="fig-fig_PCA_act_HBL1_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is mainly the difference between stimulus, but also
shows the effect of increasing inhibitor concentration (as the aIg+H2O2
samples with high inhibitor conc are closer to the PBS samples), and PC2
is mainly the difference between 100 uM iNFkB and all other
samples.</em></p>
</div>
<div id="variance-6" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_int-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_int-1">
Past versions of fig_sd_act_HBL1_int-1.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_int-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_int-2">
Past versions of fig_sd_act_HBL1_int-2.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_int-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_int-3">
Past versions of fig_sd_act_HBL1_int-3.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_int-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_int-4">
Past versions of fig_sd_act_HBL1_int-4.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_int-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-6" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = &quot; - &quot;)
  rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_HBL1_int-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_HBL1_int-1">
Past versions of fig_distances_act_HBL1_int-1.png
</button>
</p>
<div id="fig-fig_distances_act_HBL1_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_HBL1_int-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_HBL1_int-2">
Past versions of fig_distances_act_HBL1_int-2.png
</button>
</p>
<div id="fig-fig_distances_act_HBL1_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_HBL1_int-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_HBL1_int-3">
Past versions of fig_distances_act_HBL1_int-3.png
</button>
</p>
<div id="fig-fig_distances_act_HBL1_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-6" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_stim_Hi&quot;, &quot;volc_PBS_Hi&quot;, 
                        &quot;volc_iSYK0.1_Hi&quot;, &quot;volc_iSYK1_Hi&quot;, &quot;volc_iSYK10_Hi&quot;, &quot;volc_iSYK100_Hi&quot;, 
                        &quot;volc_iBTK0.1_Hi&quot;, &quot;volc_iBTK1_Hi&quot;, &quot;volc_iBTK10_Hi&quot;, &quot;volc_iBTK100_Hi&quot;, 
                        &quot;volc_iPI3Kd0.1_Hi&quot;, &quot;volc_iPI3Kd1_Hi&quot;, &quot;volc_iPI3Kd10_Hi&quot;, &quot;volc_iPI3Kd100_Hi&quot;, 
                        &quot;volc_iNFkB0.1_Hi&quot;, &quot;volc_iNFkB1_Hi&quot;, &quot;volc_iNFkB10_Hi&quot;, &quot;volc_iNFkB100_Hi&quot;, 
                        &quot;volc_iSYK100_stim_Hi&quot;, &quot;volc_iBTK100_stim_Hi&quot;, &quot;volc_iPI3Kd100_stim_Hi&quot;, &quot;volc_iNFkB100_stim_Hi&quot;)


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_stim_Hi, volc_PBS_Hi, 
                          volc_iSYK0.1_Hi, volc_iSYK1_Hi, volc_iSYK10_Hi, volc_iSYK100_Hi, 
                          volc_iBTK0.1_Hi, volc_iBTK1_Hi, volc_iBTK10_Hi, volc_iBTK100_Hi, 
                          volc_iPI3Kd0.1_Hi, volc_iPI3Kd1_Hi, volc_iPI3Kd10_Hi, volc_iPI3Kd100_Hi, 
                          volc_iNFkB0.1_Hi, volc_iNFkB1_Hi, volc_iNFkB10_Hi, volc_iNFkB100_Hi, 
                          volc_iSYK100_stim_Hi, volc_iBTK100_stim_Hi, volc_iPI3Kd100_stim_Hi, volc_iNFkB100_stim_Hi)
titles_volcano &lt;- c(&quot;aIg+H2O2 vs PBS (HBL1 w/ all inhibitors?)&quot;, &quot;PBS ctrl vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, 
                    &quot;0.1 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;1 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;10 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;100 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, 
                    &quot;0.1 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;1 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;10 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;100 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)&quot;,
                    &quot;0.1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)&quot;,
                    &quot;0.1 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)&quot;,
                    &quot;Effect of 100 uM iSYK on aIg+H2O2 vs PBS (HBL1)&quot;, &quot;Effect of 100 uM iBTK on aIg+H2O2 vs PBS (HBL1)&quot;, &quot;Effect of 100 uM iPI3Kd on aIg+H2O2 vs PBS (HBL1)&quot;, &quot;Effect of 100 uM iNFkB on aIg+H2O2 vs PBS (HBL1)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-1">
Past versions of fig_volcano_act_HBL1_int-1.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-2">
Past versions of fig_volcano_act_HBL1_int-2.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-3">
Past versions of fig_volcano_act_HBL1_int-3.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-4">
Past versions of fig_volcano_act_HBL1_int-4.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-5">
Past versions of fig_volcano_act_HBL1_int-5.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-6">
Past versions of fig_volcano_act_HBL1_int-6.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-7">
Past versions of fig_volcano_act_HBL1_int-7.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-8">
Past versions of fig_volcano_act_HBL1_int-8.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-9">
Past versions of fig_volcano_act_HBL1_int-9.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-10">
Past versions of fig_volcano_act_HBL1_int-10.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-11">
Past versions of fig_volcano_act_HBL1_int-11.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-12">
Past versions of fig_volcano_act_HBL1_int-12.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-13">
Past versions of fig_volcano_act_HBL1_int-13.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-14">
Past versions of fig_volcano_act_HBL1_int-14.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-15">
Past versions of fig_volcano_act_HBL1_int-15.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-16">
Past versions of fig_volcano_act_HBL1_int-16.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-17">
Past versions of fig_volcano_act_HBL1_int-17.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-18.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-18">
Past versions of fig_volcano_act_HBL1_int-18.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-19.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-19">
Past versions of fig_volcano_act_HBL1_int-19.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-20.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-20">
Past versions of fig_volcano_act_HBL1_int-20.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-21.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-21">
Past versions of fig_volcano_act_HBL1_int-21.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_int-22.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_int-22">
Past versions of fig_volcano_act_HBL1_int-22.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_int-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plots of DESeq2 model
comparisons: aIg+H2O2 vs PBS, each inhibitor conc vs DMSO ctrl, and the
effect of 100 uM inhibitor on aIg+H2O2 vs PBS (HBL1 cells only). Many
proteins are activated upon aIg+H2O2 stimulation. Higher conc of
inhibitor significantly affects more proteins, except iPI3Kd. And many
proteins are significantly less activated in the presence of 100 uM
inhibitor (again except iPI3Kd).</em></p>
</div>
<div id="heatmap-6" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_stim_Hi&quot;, &quot;heat_PBS_Hi&quot;, 
                        &quot;heat_iSYK0.1_Hi&quot;, &quot;heat_iSYK1_Hi&quot;, &quot;heat_iSYK10_Hi&quot;, &quot;heat_iSYK100_Hi&quot;, 
                        &quot;heat_iBTK0.1_Hi&quot;, &quot;heat_iBTK1_Hi&quot;, &quot;heat_iBTK10_Hi&quot;, &quot;heat_iBTK100_Hi&quot;, 
                        &quot;heat_iPI3Kd0.1_Hi&quot;, &quot;heat_iPI3Kd1_Hi&quot;, &quot;heat_iPI3Kd10_Hi&quot;, &quot;heat_iPI3Kd100_Hi&quot;, 
                        &quot;heat_iNFkB0.1_Hi&quot;, &quot;heat_iNFkB1_Hi&quot;, &quot;heat_iNFkB10_Hi&quot;, &quot;heat_iNFkB100_Hi&quot;, 
                        &quot;heat_iSYK100_stim_Hi&quot;, &quot;heat_iBTK100_stim_Hi&quot;, &quot;heat_iPI3Kd100_stim_Hi&quot;, &quot;heat_iNFkB100_stim_Hi&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Hi &lt;- list(heat_iSYK100_Hi, heat_iBTK100_Hi, heat_iPI3Kd100_Hi, heat_iNFkB100_Hi) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()

# Combine all heatmap datasets of 100 uM inhibitor effect on aIg+H2O2 vs PBS (without DESeq2 analysis values)
heat_100_stim_Hi &lt;- list(heat_iSYK100_stim_Hi, heat_iBTK100_stim_Hi, heat_iPI3Kd100_stim_Hi, heat_iNFkB100_stim_Hi) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;stimulus&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(stimulus, levels = stim_DS108), factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Hi &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_Hi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Hi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Hi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = &quot;Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)&quot;
)) +
  labs(title = &quot;100 uM inhibitor vs DMSO ctrl (HBL1 w/ both stimuli) (significant proteins)&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_HBL1_int_100-1.png" width="2016" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_HBL1_int_100-1">
Past versions of fig_heatmap_act_HBL1_int_100-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_HBL1_int_100-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 100 uM inhibitor effect on aIg+H2O2 vs PBS
# Get the rld transformed data for the heatmap and scale per row
data_100_stim_Hi &lt;- t(scale(t(assay(rld)[unique(c(heat_100_stim_Hi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

as.ggplot(ComplexHeatmap::pheatmap(
  data_100_stim_Hi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100_stim_Hi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = &quot;Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)&quot;
)) +
  labs(title = &quot;Effect of 100 uM inhibitor on aIg+H2O2 vs PBS (HBL1) (significant proteins)&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_HBL1_int_100-2.png" width="2016" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_HBL1_int_100-2">
Past versions of fig_heatmap_act_HBL1_int_100-2.png
</button>
</p>
<div id="fig-fig_heatmap_act_HBL1_int_100-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmaps of the significant results in
the 100 uM inhibitor vs DMSO ctrl comparisons and the effect of 100 uM
inhibitor on aIg+H2O2 vs PBS comparisons (no threshold for fold
change).</em></p>
</div>
</div>
<div id="conc-aig-only" class="section level4">
<h4>Conc (aIg only)</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>All HBL1 samples: 4 inhibitors, 6 conc, aIg+H2O2 stimulus (74
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(cell_line == &quot;HBL1&quot; &amp; stimulus == &quot;aIg+H2O2&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 uM (DMSO)&quot;)
# dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                
 [2] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;  
 [3] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot; 
 [4] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;
 [5] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;  
 [6] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;    
 [7] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;   
 [8] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;  
 [9] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;    
[10] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;   
[11] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;  
[12] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot; 
[13] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;   
[14] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;  
[15] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot; 
[16] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;
[17] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;  
[18] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_PBS_Hs &lt;- results(dds, name = &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK0.1_Hs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_Hs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_Hs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_Hs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_Hs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_Hs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_Hs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_Hs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_Hs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_Hs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_Hs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_Hs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_Hs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_Hs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_Hs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_Hs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_PBS_Hs, 
                     results_iSYK0.1_Hs, results_iSYK1_Hs, results_iSYK10_Hs, results_iSYK100_Hs, 
                     results_iBTK0.1_Hs, results_iBTK1_Hs, results_iBTK10_Hs, results_iBTK100_Hs, 
                     results_iPI3Kd0.1_Hs, results_iPI3Kd1_Hs, results_iPI3Kd10_Hs, results_iPI3Kd100_Hs, 
                     results_iNFkB0.1_Hs, results_iNFkB1_Hs, results_iNFkB10_Hs, results_iNFkB100_Hs)

# Results summary
summary(results_iNFkB100_Hs, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 35, 33%
LFC &lt; 0 (down)     : 36, 34%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 374)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_PBS_Hs, alpha = signcutoff, main = &quot;PBS vs DMSO&quot;)
plotMA(results_iNFkB100_Hs, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)
plotMA(results_iBTK100_Hs, alpha = signcutoff, main = &quot;100 uM iBTK vs DMSO&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_act_HBL1_stim-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_HBL1_stim-1">
Past versions of fig_log2FC_act_HBL1_stim-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_HBL1_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-7" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-7" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    # scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    # scale_fill_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_HBL1_stim-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_HBL1_stim-1">
Past versions of fig_PCA_act_HBL1_stim-1.png
</button>
</p>
<div id="fig-fig_PCA_act_HBL1_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_HBL1_stim-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_HBL1_stim-2">
Past versions of fig_PCA_act_HBL1_stim-2.png
</button>
</p>
<div id="fig-fig_PCA_act_HBL1_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_HBL1_stim-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_HBL1_stim-3">
Past versions of fig_PCA_act_HBL1_stim-3.png
</button>
</p>
<div id="fig-fig_PCA_act_HBL1_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is mainly the difference between inhibitor conc,
and PC2 is mainly the difference between 100 uM iNFkB and all other
samples.</em></p>
</div>
<div id="variance-7" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_stim-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_stim-1">
Past versions of fig_sd_act_HBL1_stim-1.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_stim-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_stim-2">
Past versions of fig_sd_act_HBL1_stim-2.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_stim-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_stim-3">
Past versions of fig_sd_act_HBL1_stim-3.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_HBL1_stim-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_HBL1_stim-4">
Past versions of fig_sd_act_HBL1_stim-4.png
</button>
</p>
<div id="fig-fig_sd_act_HBL1_stim-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-7" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = &quot; - &quot;)
  rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_HBL1_stim-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_HBL1_stim-1">
Past versions of fig_distances_act_HBL1_stim-1.png
</button>
</p>
<div id="fig-fig_distances_act_HBL1_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_HBL1_stim-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_HBL1_stim-2">
Past versions of fig_distances_act_HBL1_stim-2.png
</button>
</p>
<div id="fig-fig_distances_act_HBL1_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_HBL1_stim-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_HBL1_stim-3">
Past versions of fig_distances_act_HBL1_stim-3.png
</button>
</p>
<div id="fig-fig_distances_act_HBL1_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-7" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_PBS_Hs&quot;, 
                        &quot;volc_iSYK0.1_Hs&quot;, &quot;volc_iSYK1_Hs&quot;, &quot;volc_iSYK10_Hs&quot;, &quot;volc_iSYK100_Hs&quot;, 
                        &quot;volc_iBTK0.1_Hs&quot;, &quot;volc_iBTK1_Hs&quot;, &quot;volc_iBTK10_Hs&quot;, &quot;volc_iBTK100_Hs&quot;, 
                        &quot;volc_iPI3Kd0.1_Hs&quot;, &quot;volc_iPI3Kd1_Hs&quot;, &quot;volc_iPI3Kd10_Hs&quot;, &quot;volc_iPI3Kd100_Hs&quot;, 
                        &quot;volc_iNFkB0.1_Hs&quot;, &quot;volc_iNFkB1_Hs&quot;, &quot;volc_iNFkB10_Hs&quot;, &quot;volc_iNFkB100_Hs&quot;)


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_PBS_Hs, 
                          volc_iSYK0.1_Hs, volc_iSYK1_Hs, volc_iSYK10_Hs, volc_iSYK100_Hs, 
                          volc_iBTK0.1_Hs, volc_iBTK1_Hs, volc_iBTK10_Hs, volc_iBTK100_Hs, 
                          volc_iPI3Kd0.1_Hs, volc_iPI3Kd1_Hs, volc_iPI3Kd10_Hs, volc_iPI3Kd100_Hs, 
                          volc_iNFkB0.1_Hs, volc_iNFkB1_Hs, volc_iNFkB10_Hs, volc_iNFkB100_Hs)
titles_volcano &lt;- c(&quot;PBS ctrl vs DMSO ctrl (HBL1 w/ stim)&quot;,
                    &quot;0.1 uM iSYK vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;1 uM iSYK vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;10 uM iSYK vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;100 uM iSYK vs DMSO ctrl (HBL1 w/ stim)&quot;,
                    &quot;0.1 uM iBTK vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;1 uM iBTK vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;10 uM iBTK vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;100 uM iBTK vs DMSO ctrl (HBL1 w/ stim)&quot;,
                    &quot;0.1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)&quot;,
                    &quot;0.1 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0)) + 
    coord_cartesian(xlim = c(-1.5, 1.5), ylim = c(0, 175))
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-1">
Past versions of fig_volcano_act_HBL1_stim-1.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-2">
Past versions of fig_volcano_act_HBL1_stim-2.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-3">
Past versions of fig_volcano_act_HBL1_stim-3.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-4">
Past versions of fig_volcano_act_HBL1_stim-4.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-5">
Past versions of fig_volcano_act_HBL1_stim-5.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-6">
Past versions of fig_volcano_act_HBL1_stim-6.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-7">
Past versions of fig_volcano_act_HBL1_stim-7.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-8">
Past versions of fig_volcano_act_HBL1_stim-8.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-9">
Past versions of fig_volcano_act_HBL1_stim-9.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-10">
Past versions of fig_volcano_act_HBL1_stim-10.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-11">
Past versions of fig_volcano_act_HBL1_stim-11.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-12">
Past versions of fig_volcano_act_HBL1_stim-12.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-13">
Past versions of fig_volcano_act_HBL1_stim-13.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-14">
Past versions of fig_volcano_act_HBL1_stim-14.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-15">
Past versions of fig_volcano_act_HBL1_stim-15.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-16">
Past versions of fig_volcano_act_HBL1_stim-16.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_HBL1_stim-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_HBL1_stim-17">
Past versions of fig_volcano_act_HBL1_stim-17.png
</button>
</p>
<div id="fig-fig_volcano_act_HBL1_stim-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plots of DESeq2 model
comparisons: each inhibitor conc vs DMSO ctrl (HBL1 w/ aIg+H2O2 samples
only). Already at low conc inhibitor, many proteins are decreased
compared to the DMSO ctrl, and in general the effect size increases with
the inhibitor conc, except iPI3Kd.</em></p>
</div>
<div id="heatmap-7" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_PBS_Hs&quot;, 
                        &quot;heat_iSYK0.1_Hs&quot;, &quot;heat_iSYK1_Hs&quot;, &quot;heat_iSYK10_Hs&quot;, &quot;heat_iSYK100_Hs&quot;, 
                        &quot;heat_iBTK0.1_Hs&quot;, &quot;heat_iBTK1_Hs&quot;, &quot;heat_iBTK10_Hs&quot;, &quot;heat_iBTK100_Hs&quot;, 
                        &quot;heat_iPI3Kd0.1_Hs&quot;, &quot;heat_iPI3Kd1_Hs&quot;, &quot;heat_iPI3Kd10_Hs&quot;, &quot;heat_iPI3Kd100_Hs&quot;, 
                        &quot;heat_iNFkB0.1_Hs&quot;, &quot;heat_iNFkB1_Hs&quot;, &quot;heat_iNFkB10_Hs&quot;, &quot;heat_iNFkB100_Hs&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Hs &lt;- list(heat_iSYK100_Hs, heat_iBTK100_Hs, heat_iPI3Kd100_Hs, heat_iNFkB100_Hs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       # stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl (w/ stim)
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Hs &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_Hs$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Hs,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Hs)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = &quot;Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)&quot;
)) +
  labs(title = &quot;100 uM inhibitor vs DMSO ctrl (HBL1 w/ stim) (significant proteins)&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_HBL1_stim-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_HBL1_stim-1">
Past versions of fig_heatmap_act_HBL1_stim-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_HBL1_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of the significant results in
the 100 uM inhibitor vs DMSO ctrl comparisons (no threshold for fold
change).</em></p>
</div>
</div>
</div>
<div id="hbl1-protein-comparisons" class="section level3">
<h3>HBL1: Protein comparisons</h3>
<p>Finally, we visualised the normalised true counts of all proteins for
which the aIg+H2O2 vs PBS comparison was significantly affected by 100
uM inhibitor (31 in total).</p>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Datasets generated during analysis
# heat_xxx: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_xxx: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# List of all significantly changing proteins
sign_prot_Hi &lt;- sort(unique(c(heat_100_stim_Hi$target_nospace)))

heat_iSYK_Hs &lt;- list(heat_iSYK0.1_Hs, heat_iSYK1_Hs, heat_iSYK10_Hs, heat_iSYK100_Hs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iSYK_Hs &lt;- sort(unique(c(heat_iSYK_Hs$target_nospace)))

heat_iBTK_Hs &lt;- list(heat_iBTK0.1_Hs, heat_iBTK1_Hs, heat_iBTK10_Hs, heat_iBTK100_Hs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iBTK_Hs &lt;- sort(unique(c(heat_iBTK_Hs$target_nospace)))

heat_iPI3Kd_Hs &lt;- list(heat_iPI3Kd0.1_Hs, heat_iPI3Kd1_Hs, heat_iPI3Kd10_Hs, heat_iPI3Kd100_Hs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iPI3Kd_Hs &lt;- sort(unique(c(heat_iPI3Kd_Hs$target_nospace)))

heat_iNFkB_Hs &lt;- list(heat_iNFkB0.1_Hs, heat_iNFkB1_Hs, heat_iNFkB10_Hs, heat_iNFkB100_Hs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iNFkB_Hs &lt;- sort(unique(c(heat_iNFkB_Hs$target_nospace)))

sign_prot_inhib_Hs &lt;- sort(unique(c(sign_prot_iSYK_Hs, sign_prot_iBTK_Hs, sign_prot_iPI3Kd_Hs, sign_prot_iNFkB_Hs)))
sign_prot_all_H &lt;- sort(unique(c(sign_prot_Hi, sign_prot_inhib_Hs)))

# print(c(&quot;Significant proteins (OCI-ly8 vs HBL1 basal signaling):&quot;, sign_prot_cell))</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
# Effect of 100 uM inhibitor on stimulation
heat_list_Hi &lt;- list(heat_iSYK100_stim_Hi, heat_iBTK100_stim_Hi, heat_iPI3Kd100_stim_Hi, heat_iNFkB100_stim_Hi)
heat_names &lt;- c(&quot;sign_data_iSYK&quot;, &quot;sign_data_iBTK&quot;, &quot;sign_data_iPI3Kd&quot;, &quot;sign_data_iNFkB&quot;)
heat_suffix &lt;- c(&quot;_iSYK&quot;, &quot;_iBTK&quot;, &quot;_iPI3Kd&quot;, &quot;_iNFkB&quot;)
for (heat_data in c(1:length(heat_list_Hi))) {
  # print(heat_data)
  alt_data &lt;- heat_list_Hi[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_Hi) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_Hi &lt;- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iSYK conc on activated signaling
heat_list_iSYK_Hs &lt;- list(heat_iSYK0.1_Hs, heat_iSYK1_Hs, heat_iSYK10_Hs, heat_iSYK100_Hs)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iSYK_Hs))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iSYK_Hs[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iSYK_Hs) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iSYK_Hs &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iBTK conc on activated signaling
heat_list_iBTK_Hs &lt;- list(heat_iBTK0.1_Hs, heat_iBTK1_Hs, heat_iBTK10_Hs, heat_iBTK100_Hs)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iBTK_Hs))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iBTK_Hs[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iBTK_Hs) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iBTK_Hs &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iPI3Kd conc on activated signaling
heat_list_iPI3Kd_Hs &lt;- list(heat_iPI3Kd0.1_Hs, heat_iPI3Kd1_Hs, heat_iPI3Kd10_Hs, heat_iPI3Kd100_Hs)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iPI3Kd_Hs))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iPI3Kd_Hs[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iPI3Kd_Hs) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iPI3Kd_Hs &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iNFkB conc on activated signaling
heat_list_iNFkB_Hs &lt;- list(heat_iNFkB0.1_Hs, heat_iNFkB1_Hs, heat_iNFkB10_Hs, heat_iNFkB100_Hs)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iNFkB_Hs))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iNFkB_Hs[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iNFkB_Hs) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iNFkB_Hs &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
# Effect of 100 uM inhibitor on stimulation
clean_data_well_Hi &lt;- data.frame(data_100_stim_Hi) %&gt;% 
  mutate(target_nospace = rownames(data_100_stim_Hi)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value&quot;, 2:c(ncol(data_100_stim_Hi)+1)) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_Hi) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

# Effect of inhibitor conc on activated signaling
data_iSYK_Hs &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iSYK_Hs)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iSYK&quot;)))])))
alt_data_iSYK_Hs &lt;- data.frame(data_iSYK_Hs) %&gt;% 
  mutate(target_nospace = rownames(data_iSYK_Hs)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iSYK&quot;, 2:c(ncol(data_iSYK_Hs)+1))
data_iBTK_Hs &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iBTK_Hs)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iBTK&quot;)))])))
alt_data_iBTK_Hs &lt;- data.frame(data_iBTK_Hs) %&gt;% 
  mutate(target_nospace = rownames(data_iBTK_Hs)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iBTK&quot;, 2:c(ncol(data_iBTK_Hs)+1))
data_iPI3Kd_Hs &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iPI3Kd_Hs)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iPI3Kd&quot;)))])))
alt_data_iPI3Kd_Hs &lt;- data.frame(data_iPI3Kd_Hs) %&gt;% 
  mutate(target_nospace = rownames(data_iPI3Kd_Hs)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iPI3Kd&quot;, 2:c(ncol(data_iPI3Kd_Hs)+1))
data_iNFkB_Hs &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iNFkB_Hs)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iNFkB&quot;)))])))
alt_data_iNFkB_Hs &lt;- data.frame(data_iNFkB_Hs) %&gt;% 
  mutate(target_nospace = rownames(data_iNFkB_Hs)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iNFkB&quot;, 2:c(ncol(data_iNFkB_Hs)+1))

clean_data_well_inhib_Hs &lt;- list(alt_data_iSYK_Hs, alt_data_iBTK_Hs, alt_data_iPI3Kd_Hs, alt_data_iNFkB_Hs) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_inhib_Hs) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_iSYK, heatmap_iBTK, heatmap_iPI3Kd, heatmap_iNFkB) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

# All significant proteins for plotting
clean_data_well_H &lt;- counts_dtbl %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_Hi) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with mean values per condition (replicates combined)
# Effect of 100 uM inhibitor on stimulation
clean_data_mean_Hi &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_Hi &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# Effect of inhibitor conc on activated signaling
clean_data_mean_Hs &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_inhib_Hs &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# All significant proteins for plotting
clean_data_mean_H &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_Hi) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)</code></pre>
<pre class="r"><code>plot &lt;- ggplot(clean_data_well_Hi, 
               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1_all-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1_all-1">
Past versions of fig_counts_act_HBL1_all-1.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Normalised counts for all significantly
changing proteins. For some proteins the effect size is quite
small.</em></p>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

# Re-redesigned fig
filt_clean_data_mean_H &lt;- clean_data_mean_H %&gt;%
  dplyr::group_by(cell_line, inhibitor, inhib_conc, stimulus, target_nospace) %&gt;%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_Hi) {
  plot &lt;- ggplot() +
    geom_jitter(data = subset(clean_data_well_H, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
                    counts_norm, 
                    color = stimulus, 
                    # shape = cell_line
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean_H, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = stimulus,
                      color = stimulus
                      ),
                  width = 0.5, size = 0.75) + #, color = &quot;black&quot;
    # facet_wrap(vars(factor(target_nospace))) +
    # facet_wrap(vars(factor(inhibitor, levels = inhib_DS108)), ncol = 1) +
    facet_grid(vars(factor(cell_line, levels = cells_DS108)), vars(factor(inhibitor, levels = inhib_DS108))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;, title = protein) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-1">
Past versions of fig_counts_act_HBL1-1.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-2">
Past versions of fig_counts_act_HBL1-2.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-3">
Past versions of fig_counts_act_HBL1-3.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-4">
Past versions of fig_counts_act_HBL1-4.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-5">
Past versions of fig_counts_act_HBL1-5.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-6.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-6">
Past versions of fig_counts_act_HBL1-6.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-7.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-7">
Past versions of fig_counts_act_HBL1-7.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-8.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-8">
Past versions of fig_counts_act_HBL1-8.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-9.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-9">
Past versions of fig_counts_act_HBL1-9.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-10.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-10">
Past versions of fig_counts_act_HBL1-10.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-11.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-11">
Past versions of fig_counts_act_HBL1-11.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-12.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-12">
Past versions of fig_counts_act_HBL1-12.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-13.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-13">
Past versions of fig_counts_act_HBL1-13.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-14.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-14">
Past versions of fig_counts_act_HBL1-14.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-15.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-15">
Past versions of fig_counts_act_HBL1-15.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-16.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-16">
Past versions of fig_counts_act_HBL1-16.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-17.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-17">
Past versions of fig_counts_act_HBL1-17.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-18.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-18">
Past versions of fig_counts_act_HBL1-18.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-19.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-19">
Past versions of fig_counts_act_HBL1-19.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-20.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-20">
Past versions of fig_counts_act_HBL1-20.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-21.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-21">
Past versions of fig_counts_act_HBL1-21.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-22.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-22">
Past versions of fig_counts_act_HBL1-22.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-23.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-23">
Past versions of fig_counts_act_HBL1-23.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-24.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-24">
Past versions of fig_counts_act_HBL1-24.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-25.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-25">
Past versions of fig_counts_act_HBL1-25.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-26.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-26">
Past versions of fig_counts_act_HBL1-26.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-27.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-27">
Past versions of fig_counts_act_HBL1-27.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-28.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-28">
Past versions of fig_counts_act_HBL1-28.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-29.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-29">
Past versions of fig_counts_act_HBL1-29.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-30.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-30">
Past versions of fig_counts_act_HBL1-30.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_HBL1-31.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_HBL1-31">
Past versions of fig_counts_act_HBL1-31.png
</button>
</p>
<div id="fig-fig_counts_act_HBL1-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="hbl1-save-data" class="section level3">
<h3>HBL1: Save data</h3>
<p>Save all information and datasets as .csv file</p>
<p>Effect of 100 uM inhibitor on aIg+H2O2 vs PBS comparison (HBL1 cells
only):</p>
<ul>
<li><p>HBL1_100uM_stim_sign_proteins.txt: List of significantly affected
proteins by 100 uM inhibitor in the aIg+H2O2 vs PBS comparison (by at
least one inhibitor)</p></li>
<li><p>HBL1_100uM_stim_sign_DESeq2data.csv: Dataset with significantly
changing proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>HBL1_100uM_stim_sign_sample.csv: Dataset of counts per sample per
protein + heatmap data for significantly changing proteins</p></li>
<li><p>HBL1_100uM_stim_sign_condition.csv: Dataset of counts per
condition (mean of repl) per protein for significantly changing
proteins</p></li>
</ul>
<p>Effect of inhibitor conc vs DMSO ctrl (for each inhibitor, HBL1 w/
stim only): For each inhibitor (iSYK, iBTK, iPI3Kd, iNFkB):</p>
<ul>
<li><p>HBL1_inhib_sign_proteins.txt: List of significantly different
proteins in any of the inhibitor conc vs DMSO ctrl comparisons</p></li>
<li><p>HBL1_inhib_sign_DESeq2data.csv: Dataset with significantly
different proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>HBL1_inhib_sign_sample.csv: Dataset of counts per sample per
protein + heatmap data for significantly changing proteins</p></li>
<li><p>HBL1_inhib_sign_condition.csv: Dataset of counts per condition
(mean of repl) per protein for significantly changing proteins</p></li>
</ul>
<pre class="r"><code># List of significantly changing proteins: sign_prot_cell
write(sign_prot_Hi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_proteins.txt&quot;)
write(sign_prot_iSYK_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iSYK_sign_proteins.txt&quot;)
write(sign_prot_iBTK_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iBTK_sign_proteins.txt&quot;)
write(sign_prot_iPI3Kd_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iPI3Kd_sign_proteins.txt&quot;)
write(sign_prot_iNFkB_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iNFkB_sign_proteins.txt&quot;)

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_Hi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iSYK_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iSYK_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iBTK_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iBTK_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iPI3Kd_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iPI3Kd_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iNFkB_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iNFkB_sign_DESeq2data.csv&quot;, row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_Hi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_sample.csv&quot;, row.names = F)
write.csv(clean_data_well_inhib_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_inhib_sign_sample.csv&quot;, row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_Hi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_condition.csv&quot;, row.names = F)
write.csv(clean_data_mean_Hs, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_inhib_sign_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
</div>
<div id="oci-ly8-deseq2" class="section level3 tabset">
<h3 class="tabset">OCI-Ly8: DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="inhib-concstim-1" class="section level4">
<h4>Inhib conc:stim</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>All OCI-Ly8 samples: 4 inhibitors, 6 conc, 2 stimuli (144
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(cell_line == &quot;OCI-Ly8&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 uM (DMSO)&quot;)
dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                   
 [2] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;     
 [3] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;    
 [4] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;   
 [5] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;     
 [6] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;       
 [7] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;      
 [8] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;     
 [9] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;       
[10] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;      
[11] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;     
[12] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;    
[13] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;      
[14] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;     
[15] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;    
[16] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;   
[17] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;     
[18] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;        
[19] &quot;stimulus_aIg.H2O2_vs_PBS&quot;                    
[20] &quot;inhib_text_new0.1.uM.iBTK.stimulusaIg.H2O2&quot;  
[21] &quot;inhib_text_new0.1.uM.iNFkB.stimulusaIg.H2O2&quot; 
[22] &quot;inhib_text_new0.1.uM.iPI3Kd.stimulusaIg.H2O2&quot;
[23] &quot;inhib_text_new0.1.uM.iSYK.stimulusaIg.H2O2&quot;  
[24] &quot;inhib_text_new1.uM.iBTK.stimulusaIg.H2O2&quot;    
[25] &quot;inhib_text_new1.uM.iNFkB.stimulusaIg.H2O2&quot;   
[26] &quot;inhib_text_new1.uM.iPI3Kd.stimulusaIg.H2O2&quot;  
[27] &quot;inhib_text_new1.uM.iSYK.stimulusaIg.H2O2&quot;    
[28] &quot;inhib_text_new10.uM.iBTK.stimulusaIg.H2O2&quot;   
[29] &quot;inhib_text_new10.uM.iNFkB.stimulusaIg.H2O2&quot;  
[30] &quot;inhib_text_new10.uM.iPI3Kd.stimulusaIg.H2O2&quot; 
[31] &quot;inhib_text_new10.uM.iSYK.stimulusaIg.H2O2&quot;   
[32] &quot;inhib_text_new100.uM.iBTK.stimulusaIg.H2O2&quot;  
[33] &quot;inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2&quot; 
[34] &quot;inhib_text_new100.uM.iPI3Kd.stimulusaIg.H2O2&quot;
[35] &quot;inhib_text_new100.uM.iSYK.stimulusaIg.H2O2&quot;  
[36] &quot;inhib_text_newPBS.ctrl.stimulusaIg.H2O2&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_stim_Oi &lt;- results(dds, name = &quot;stimulus_aIg.H2O2_vs_PBS&quot;, alpha = 0.05)

results_PBS_Oi &lt;- results(dds, name = &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK0.1_Oi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_Oi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_Oi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_Oi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_Oi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_Oi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_Oi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_Oi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_Oi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_Oi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_Oi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_Oi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_Oi &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_Oi &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_Oi &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_Oi &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK100_stim_Oi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iSYK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iBTK100_stim_Oi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iPI3Kd100_stim_Oi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iPI3Kd.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iNFkB100_stim_Oi &lt;- results(dds, name = &quot;inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_stim_Oi, results_PBS_Oi, 
                     results_iSYK0.1_Oi, results_iSYK1_Oi, results_iSYK10_Oi, results_iSYK100_Oi, 
                     results_iBTK0.1_Oi, results_iBTK1_Oi, results_iBTK10_Oi, results_iBTK100_Oi, 
                     results_iPI3Kd0.1_Oi, results_iPI3Kd1_Oi, results_iPI3Kd10_Oi, results_iPI3Kd100_Oi, 
                     results_iNFkB0.1_Oi, results_iNFkB1_Oi, results_iNFkB10_Oi, results_iNFkB100_Oi, 
                     results_iSYK100_stim_Oi, results_iBTK100_stim_Oi, results_iPI3Kd100_stim_Oi, results_iNFkB100_stim_Oi)

# Results summary
summary(results_stim_Oi, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 24, 22%
LFC &lt; 0 (down)     : 25, 23%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 433)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_stim_Oi, alpha = signcutoff, main = &quot;aIg+H2O2 vs PBS&quot;)
plotMA(results_iNFkB100_Oi, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)
plotMA(results_iNFkB100_stim_Oi, alpha = signcutoff, main = &quot;Effect of 100 uM iNFkB on\n aIg+H2O2 vs PBS&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_act_Ly8_int-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_Ly8_int-1">
Past versions of fig_log2FC_act_Ly8_int-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_Ly8_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-8" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-8" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    # scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_fill_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_Ly8_int-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_Ly8_int-1">
Past versions of fig_PCA_act_Ly8_int-1.png
</button>
</p>
<div id="fig-fig_PCA_act_Ly8_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_Ly8_int-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_Ly8_int-2">
Past versions of fig_PCA_act_Ly8_int-2.png
</button>
</p>
<div id="fig-fig_PCA_act_Ly8_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_Ly8_int-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_Ly8_int-3">
Past versions of fig_PCA_act_Ly8_int-3.png
</button>
</p>
<div id="fig-fig_PCA_act_Ly8_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is the difference between stimuli and the effect of
increasing inhibitor concentration (as the aIg+H2O2 samples with high
inhibitor conc are closer to the PBS samples), and PC2 is mainly the
difference between 100 uM iNFkB and all other samples.</em></p>
</div>
<div id="variance-8" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_int-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_int-1">
Past versions of fig_sd_act_Ly8_int-1.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_int-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_int-2">
Past versions of fig_sd_act_Ly8_int-2.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_int-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_int-3">
Past versions of fig_sd_act_Ly8_int-3.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_int-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_int-4">
Past versions of fig_sd_act_Ly8_int-4.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_int-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-8" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = &quot; - &quot;)
  rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_Ly8_int-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_Ly8_int-1">
Past versions of fig_distances_act_Ly8_int-1.png
</button>
</p>
<div id="fig-fig_distances_act_Ly8_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_Ly8_int-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_Ly8_int-2">
Past versions of fig_distances_act_Ly8_int-2.png
</button>
</p>
<div id="fig-fig_distances_act_Ly8_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_Ly8_int-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_Ly8_int-3">
Past versions of fig_distances_act_Ly8_int-3.png
</button>
</p>
<div id="fig-fig_distances_act_Ly8_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-8" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_stim_Oi&quot;, &quot;volc_PBS_Oi&quot;, 
                        &quot;volc_iSYK0.1_Oi&quot;, &quot;volc_iSYK1_Oi&quot;, &quot;volc_iSYK10_Oi&quot;, &quot;volc_iSYK100_Oi&quot;, 
                        &quot;volc_iBTK0.1_Oi&quot;, &quot;volc_iBTK1_Oi&quot;, &quot;volc_iBTK10_Oi&quot;, &quot;volc_iBTK100_Oi&quot;, 
                        &quot;volc_iPI3Kd0.1_Oi&quot;, &quot;volc_iPI3Kd1_Oi&quot;, &quot;volc_iPI3Kd10_Oi&quot;, &quot;volc_iPI3Kd100_Oi&quot;, 
                        &quot;volc_iNFkB0.1_Oi&quot;, &quot;volc_iNFkB1_Oi&quot;, &quot;volc_iNFkB10_Oi&quot;, &quot;volc_iNFkB100_Oi&quot;, 
                        &quot;volc_iSYK100_stim_Oi&quot;, &quot;volc_iBTK100_stim_Oi&quot;, &quot;volc_iPI3Kd100_stim_Oi&quot;, &quot;volc_iNFkB100_stim_Oi&quot;)


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_stim_Oi, volc_PBS_Oi, 
                          volc_iSYK0.1_Oi, volc_iSYK1_Oi, volc_iSYK10_Oi, volc_iSYK100_Oi, 
                          volc_iBTK0.1_Oi, volc_iBTK1_Oi, volc_iBTK10_Oi, volc_iBTK100_Oi, 
                          volc_iPI3Kd0.1_Oi, volc_iPI3Kd1_Oi, volc_iPI3Kd10_Oi, volc_iPI3Kd100_Oi, 
                          volc_iNFkB0.1_Oi, volc_iNFkB1_Oi, volc_iNFkB10_Oi, volc_iNFkB100_Oi, 
                          volc_iSYK100_stim_Oi, volc_iBTK100_stim_Oi, volc_iPI3Kd100_stim_Oi, volc_iNFkB100_stim_Oi)
titles_volcano &lt;- c(&quot;aIg+H2O2 vs PBS (OCI-Ly8 w/ all inhibitors?)&quot;, &quot;PBS ctrl vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, 
                    &quot;0.1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;10 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;100 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, 
                    &quot;0.1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;10 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;100 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;,
                    &quot;0.1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;,
                    &quot;0.1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)&quot;,
                    &quot;Effect of 100 uM iSYK on aIg+H2O2 vs PBS (OCI-Ly8)&quot;, &quot;Effect of 100 uM iBTK on aIg+H2O2 vs PBS (OCI-Ly8)&quot;, &quot;Effect of 100 uM iPI3Kd on aIg+H2O2 vs PBS (OCI-Ly8)&quot;, &quot;Effect of 100 uM iNFkB on aIg+H2O2 vs PBS (OCI-Ly8)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-1">
Past versions of fig_volcano_act_Ly8_int-1.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-2">
Past versions of fig_volcano_act_Ly8_int-2.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-3">
Past versions of fig_volcano_act_Ly8_int-3.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-4">
Past versions of fig_volcano_act_Ly8_int-4.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-5">
Past versions of fig_volcano_act_Ly8_int-5.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-6">
Past versions of fig_volcano_act_Ly8_int-6.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-7">
Past versions of fig_volcano_act_Ly8_int-7.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-8">
Past versions of fig_volcano_act_Ly8_int-8.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-9">
Past versions of fig_volcano_act_Ly8_int-9.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-10">
Past versions of fig_volcano_act_Ly8_int-10.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-11">
Past versions of fig_volcano_act_Ly8_int-11.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-12">
Past versions of fig_volcano_act_Ly8_int-12.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-13">
Past versions of fig_volcano_act_Ly8_int-13.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-14">
Past versions of fig_volcano_act_Ly8_int-14.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-15">
Past versions of fig_volcano_act_Ly8_int-15.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-16">
Past versions of fig_volcano_act_Ly8_int-16.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-17">
Past versions of fig_volcano_act_Ly8_int-17.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-18.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-18">
Past versions of fig_volcano_act_Ly8_int-18.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-19.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-19">
Past versions of fig_volcano_act_Ly8_int-19.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-20.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-20">
Past versions of fig_volcano_act_Ly8_int-20.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-21.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-21">
Past versions of fig_volcano_act_Ly8_int-21.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_int-22.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_int-22">
Past versions of fig_volcano_act_Ly8_int-22.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_int-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plots of DESeq2 model
comparisons: aIg+H2O2 vs PBS, each inhibitor conc vs DMSO ctrl, and the
effect of 100 uM inhibitor on aIg+H2O2 vs PBS (OCI-Ly8 cells only). Many
proteins are activated upon aIg+H2O2 stimulation. Higher conc of
inhibitor significantly affects more proteins, except iPI3Kd. And many
proteins are significantly less activated in the presence of 100 uM
inhibitor (again except iPI3Kd).</em></p>
</div>
<div id="heatmap-8" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_stim_Oi&quot;, &quot;heat_PBS_Oi&quot;, 
                        &quot;heat_iSYK0.1_Oi&quot;, &quot;heat_iSYK1_Oi&quot;, &quot;heat_iSYK10_Oi&quot;, &quot;heat_iSYK100_Oi&quot;, 
                        &quot;heat_iBTK0.1_Oi&quot;, &quot;heat_iBTK1_Oi&quot;, &quot;heat_iBTK10_Oi&quot;, &quot;heat_iBTK100_Oi&quot;, 
                        &quot;heat_iPI3Kd0.1_Oi&quot;, &quot;heat_iPI3Kd1_Oi&quot;, &quot;heat_iPI3Kd10_Oi&quot;, &quot;heat_iPI3Kd100_Oi&quot;, 
                        &quot;heat_iNFkB0.1_Oi&quot;, &quot;heat_iNFkB1_Oi&quot;, &quot;heat_iNFkB10_Oi&quot;, &quot;heat_iNFkB100_Oi&quot;, 
                        &quot;heat_iSYK100_stim_Oi&quot;, &quot;heat_iBTK100_stim_Oi&quot;, &quot;heat_iPI3Kd100_stim_Oi&quot;, &quot;heat_iNFkB100_stim_Oi&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Oi &lt;- list(heat_iSYK100_Oi, heat_iBTK100_Oi, heat_iPI3Kd100_Oi, heat_iNFkB100_Oi) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()

# Combine all heatmap datasets of 100 uM inhibitor effect on aIg+H2O2 vs PBS (without DESeq2 analysis values)
heat_100_stim_Oi &lt;- list(heat_iSYK100_stim_Oi, heat_iBTK100_stim_Oi, heat_iPI3Kd100_stim_Oi, heat_iNFkB100_stim_Oi) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;stimulus&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(stimulus, levels = stim_DS108), factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Oi &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_Oi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Oi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Oi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = &quot;Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)&quot;
)) +
  labs(title = &quot;100 uM inhibitor vs DMSO ctrl (OCI-Ly8 w/ both stimuli) (significant proteins)&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_Ly8_int_100-1.png" width="2016" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_Ly8_int_100-1">
Past versions of fig_heatmap_act_Ly8_int_100-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_Ly8_int_100-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># 100 uM inhibitor effect on aIg+H2O2 vs PBS
# Get the rld transformed data for the heatmap and scale per row
data_100_stim_Oi &lt;- t(scale(t(assay(rld)[unique(c(heat_100_stim_Oi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

as.ggplot(ComplexHeatmap::pheatmap(
  data_100_stim_Oi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100_stim_Oi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = &quot;Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)&quot;
)) +
  labs(title = &quot;Effect of 100 uM inhibitor on aIg+H2O2 vs PBS (OCI-Ly8) (significant proteins)&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_Ly8_int_100-2.png" width="2016" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_Ly8_int_100-2">
Past versions of fig_heatmap_act_Ly8_int_100-2.png
</button>
</p>
<div id="fig-fig_heatmap_act_Ly8_int_100-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmaps of the significant results in
the 100 uM inhibitor vs DMSO ctrl comparisons and the effect of 100 uM
inhibitor on aIg+H2O2 vs PBS comparisons (no threshold for fold
change).</em></p>
</div>
</div>
<div id="conc-aig-only-1" class="section level4">
<h4>Conc (aIg only)</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>All OCI-Ly8 samples: 4 inhibitors, 6 conc, aIg+H2O2 stimulus (74
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p>
<p><em>Parameter options: cell_line + stimulus + inhibitor + inhib_conc
+ inhib_conc_text + description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))
inhib_new_DS108 &lt;- c(&quot;DMSO ctrl&quot;, &quot;PBS ctrl&quot;, &quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iPI3Kd&quot;, &quot;iNFkB&quot;)

cts &lt;- counts_dtbl  %&gt;%
  filter(cell_line == &quot;OCI-Ly8&quot; &amp; stimulus == &quot;aIg+H2O2&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %&gt;%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor &lt;- relevel(dds$inhibitor, ref = &quot;iSYK&quot;)
# dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 uM (DMSO)&quot;)
# dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                
 [2] &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;  
 [3] &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot; 
 [4] &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;
 [5] &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;  
 [6] &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;    
 [7] &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;   
 [8] &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;  
 [9] &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;    
[10] &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;   
[11] &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;  
[12] &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot; 
[13] &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;   
[14] &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;  
[15] &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot; 
[16] &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;
[17] &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;  
[18] &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_PBS_Os &lt;- results(dds, name = &quot;inhib_text_new_PBS.ctrl_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iSYK0.1_Os &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_Os &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_Os &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_Os &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_Os &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_Os &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_Os &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_Os &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iPI3Kd0.1_Os &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd1_Os &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd10_Os &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iPI3Kd100_Os &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_Os &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_Os &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_Os &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_Os &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

# results_CD53_aIg &lt;- results(dds, contrast = c(&quot;description_cell_stim&quot;, &quot;CD53_KO anti-Ig&quot;, &quot;WT_ctrl anti-Ig&quot;), alpha = 0.05)

list_results &lt;- list(results_PBS_Os, 
                     results_iSYK0.1_Os, results_iSYK1_Os, results_iSYK10_Os, results_iSYK100_Os, 
                     results_iBTK0.1_Os, results_iBTK1_Os, results_iBTK10_Os, results_iBTK100_Os, 
                     results_iPI3Kd0.1_Os, results_iPI3Kd1_Os, results_iPI3Kd10_Os, results_iPI3Kd100_Os, 
                     results_iNFkB0.1_Os, results_iNFkB1_Os, results_iNFkB10_Os, results_iNFkB100_Os)

# Results summary
summary(results_iNFkB100_Os, padj = 0.05)</code></pre>
<pre><code>
out of 107 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 36, 34%
LFC &lt; 0 (down)     : 37, 35%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 415)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_PBS_Os, alpha = signcutoff, main = &quot;PBS vs DMSO&quot;)
plotMA(results_iNFkB100_Os, alpha = signcutoff, main = &quot;100 uM iNFkB vs DMSO&quot;)
plotMA(results_iBTK100_Os, alpha = signcutoff, main = &quot;100 uM iBTK vs DMSO&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_log2FC_act_Ly8_stim-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_Ly8_stim-1">
Past versions of fig_log2FC_act_Ly8_stim-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_Ly8_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-9" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-9" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;cell_line&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_text_new&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhib conc&quot;) +
    # scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    # scale_fill_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = &quot;Inhib conc&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_Ly8_stim-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_Ly8_stim-1">
Past versions of fig_PCA_act_Ly8_stim-1.png
</button>
</p>
<div id="fig-fig_PCA_act_Ly8_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_Ly8_stim-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_Ly8_stim-2">
Past versions of fig_PCA_act_Ly8_stim-2.png
</button>
</p>
<div id="fig-fig_PCA_act_Ly8_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_PCA_act_Ly8_stim-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_Ly8_stim-3">
Past versions of fig_PCA_act_Ly8_stim-3.png
</button>
</p>
<div id="fig-fig_PCA_act_Ly8_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 is mainly the difference between inhibitor conc,
and PC2 is mainly the difference between 100 uM iNFkB and all other
samples.</em></p>
</div>
<div id="variance-9" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_stim-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_stim-1">
Past versions of fig_sd_act_Ly8_stim-1.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Normtransform&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_stim-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_stim-2">
Past versions of fig_sd_act_Ly8_stim-2.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_stim-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_stim-3">
Past versions of fig_sd_act_Ly8_stim-3.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_sd_act_Ly8_stim-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_Ly8_stim-4">
Past versions of fig_sd_act_Ly8_stim-4.png
</button>
</p>
<div id="fig-fig_sd_act_Ly8_stim-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-9" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = &quot; - &quot;)
  rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_Ly8_stim-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_Ly8_stim-1">
Past versions of fig_distances_act_Ly8_stim-1.png
</button>
</p>
<div id="fig-fig_distances_act_Ly8_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_Ly8_stim-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_Ly8_stim-2">
Past versions of fig_distances_act_Ly8_stim-2.png
</button>
</p>
<div id="fig-fig_distances_act_Ly8_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_distances_act_Ly8_stim-3.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_Ly8_stim-3">
Past versions of fig_distances_act_Ly8_stim-3.png
</button>
</p>
<div id="fig-fig_distances_act_Ly8_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-9" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins (no cutoff for fold
change):</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_PBS_Os&quot;, 
                        &quot;volc_iSYK0.1_Os&quot;, &quot;volc_iSYK1_Os&quot;, &quot;volc_iSYK10_Os&quot;, &quot;volc_iSYK100_Os&quot;, 
                        &quot;volc_iBTK0.1_Os&quot;, &quot;volc_iBTK1_Os&quot;, &quot;volc_iBTK10_Os&quot;, &quot;volc_iBTK100_Os&quot;, 
                        &quot;volc_iPI3Kd0.1_Os&quot;, &quot;volc_iPI3Kd1_Os&quot;, &quot;volc_iPI3Kd10_Os&quot;, &quot;volc_iPI3Kd100_Os&quot;, 
                        &quot;volc_iNFkB0.1_Os&quot;, &quot;volc_iNFkB1_Os&quot;, &quot;volc_iNFkB10_Os&quot;, &quot;volc_iNFkB100_Os&quot;)


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_PBS_Os, 
                          volc_iSYK0.1_Os, volc_iSYK1_Os, volc_iSYK10_Os, volc_iSYK100_Os, 
                          volc_iBTK0.1_Os, volc_iBTK1_Os, volc_iBTK10_Os, volc_iBTK100_Os, 
                          volc_iPI3Kd0.1_Os, volc_iPI3Kd1_Os, volc_iPI3Kd10_Os, volc_iPI3Kd100_Os, 
                          volc_iNFkB0.1_Os, volc_iNFkB1_Os, volc_iNFkB10_Os, volc_iNFkB100_Os)
titles_volcano &lt;- c(&quot;PBS ctrl vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;,
                    &quot;0.1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;10 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;100 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;,
                    &quot;0.1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;10 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;100 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;,
                    &quot;0.1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;10 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;100 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;,
                    &quot;0.1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;10 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;, &quot;100 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)&quot;)

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0)) + 
    coord_cartesian(xlim = c(-1.25, 1.25), ylim = c(0, 125))
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-1">
Past versions of fig_volcano_act_Ly8_stim-1.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-2">
Past versions of fig_volcano_act_Ly8_stim-2.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-3">
Past versions of fig_volcano_act_Ly8_stim-3.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-4">
Past versions of fig_volcano_act_Ly8_stim-4.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-5">
Past versions of fig_volcano_act_Ly8_stim-5.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-6">
Past versions of fig_volcano_act_Ly8_stim-6.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-7">
Past versions of fig_volcano_act_Ly8_stim-7.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-8">
Past versions of fig_volcano_act_Ly8_stim-8.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-9">
Past versions of fig_volcano_act_Ly8_stim-9.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-10">
Past versions of fig_volcano_act_Ly8_stim-10.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-11">
Past versions of fig_volcano_act_Ly8_stim-11.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-12">
Past versions of fig_volcano_act_Ly8_stim-12.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-13.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-13">
Past versions of fig_volcano_act_Ly8_stim-13.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-14.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-14">
Past versions of fig_volcano_act_Ly8_stim-14.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-15.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-15">
Past versions of fig_volcano_act_Ly8_stim-15.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-16.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-16">
Past versions of fig_volcano_act_Ly8_stim-16.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_volcano_act_Ly8_stim-17.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_Ly8_stim-17">
Past versions of fig_volcano_act_Ly8_stim-17.png
</button>
</p>
<div id="fig-fig_volcano_act_Ly8_stim-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plots of DESeq2 model
comparisons: each inhibitor conc vs DMSO ctrl (OCI-Ly8 w/ aIg+H2O2
samples only). Already at low conc inhibitor, several proteins are
decreased compared to the DMSO ctrl, and in general the number op
proteins, effect size and significance increase with the inhibitor conc,
except iPI3Kd.</em></p>
</div>
<div id="heatmap-9" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li>padj &lt; 0.05</li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_PBS_Os&quot;, 
                        &quot;heat_iSYK0.1_Os&quot;, &quot;heat_iSYK1_Os&quot;, &quot;heat_iSYK10_Os&quot;, &quot;heat_iSYK100_Os&quot;, 
                        &quot;heat_iBTK0.1_Os&quot;, &quot;heat_iBTK1_Os&quot;, &quot;heat_iBTK10_Os&quot;, &quot;heat_iBTK100_Os&quot;, 
                        &quot;heat_iPI3Kd0.1_Os&quot;, &quot;heat_iPI3Kd1_Os&quot;, &quot;heat_iPI3Kd10_Os&quot;, &quot;heat_iPI3Kd100_Os&quot;, 
                        &quot;heat_iNFkB0.1_Os&quot;, &quot;heat_iNFkB1_Os&quot;, &quot;heat_iNFkB10_Os&quot;, &quot;heat_iNFkB100_Os&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Os &lt;- list(heat_iSYK100_Os, heat_iBTK100_Os, heat_iPI3Kd100_Os, heat_iNFkB100_Os) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;inhib_new&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man &lt;- list(inhib_new = c(&quot;PBS ctrl&quot; = &quot;black&quot;, &quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib),
                       # stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl (w/ stim)
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Os &lt;- t(scale(t(assay(rld)[unique(c(heat_100uM_Os$target_nospace)), rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Os,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_cell[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Os)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = &quot;Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)&quot;
)) +
  labs(title = &quot;100 uM inhibitor vs DMSO ctrl (OCI-Ly8 w/ stim) (significant proteins)&quot;)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_heatmap_act_Ly8_stim-1.png" width="1248" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_Ly8_stim-1">
Past versions of fig_heatmap_act_Ly8_stim-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_Ly8_stim-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of the significant results in
the 100 uM inhibitor vs DMSO ctrl comparisons (no threshold for fold
change).</em></p>
</div>
</div>
</div>
<div id="oci-ly8-protein-comparisons" class="section level3">
<h3>OCI-Ly8: Protein comparisons</h3>
<p>Finally, we visualised the normalised true counts of all proteins for
which the aIg+H2O2 vs PBS comparison was significantly affected by 100
uM inhibitor (27 in total).</p>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Datasets generated during analysis
# heat_xxx: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_xxx: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# List of all significantly changing proteins
sign_prot_Oi &lt;- sort(unique(c(heat_100_stim_Oi$target_nospace)))

heat_iSYK_Os &lt;- list(heat_iSYK0.1_Os, heat_iSYK1_Os, heat_iSYK10_Os, heat_iSYK100_Os) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iSYK_Os &lt;- sort(unique(c(heat_iSYK_Os$target_nospace)))

heat_iBTK_Os &lt;- list(heat_iBTK0.1_Os, heat_iBTK1_Os, heat_iBTK10_Os, heat_iBTK100_Os) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iBTK_Os &lt;- sort(unique(c(heat_iBTK_Os$target_nospace)))

heat_iPI3Kd_Os &lt;- list(heat_iPI3Kd0.1_Os, heat_iPI3Kd1_Os, heat_iPI3Kd10_Os, heat_iPI3Kd100_Os) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iPI3Kd_Os &lt;- sort(unique(c(heat_iPI3Kd_Os$target_nospace)))

heat_iNFkB_Os &lt;- list(heat_iNFkB0.1_Os, heat_iNFkB1_Os, heat_iNFkB10_Os, heat_iNFkB100_Os) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  filter(target_nospace != &quot;BIM&quot;) %&gt;%
  distinct()
sign_prot_iNFkB_Os &lt;- sort(unique(c(heat_iNFkB_Os$target_nospace)))

sign_prot_inhib_Os &lt;- sort(unique(c(sign_prot_iSYK_Os, sign_prot_iBTK_Os, sign_prot_iPI3Kd_Os, sign_prot_iNFkB_Os)))
sign_prot_all_O &lt;- sort(unique(c(sign_prot_Oi, sign_prot_inhib_Os)))

# print(c(&quot;Significant proteins (OCI-ly8 vs OCI-Ly8 basal signaling):&quot;, sign_prot_cell))</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
# Effect of 100 uM inhibitor on stimulation
heat_list_Oi &lt;- list(heat_iSYK100_stim_Oi, heat_iBTK100_stim_Oi, heat_iPI3Kd100_stim_Oi, heat_iNFkB100_stim_Oi)
heat_names &lt;- c(&quot;sign_data_iSYK&quot;, &quot;sign_data_iBTK&quot;, &quot;sign_data_iPI3Kd&quot;, &quot;sign_data_iNFkB&quot;)
heat_suffix &lt;- c(&quot;_iSYK&quot;, &quot;_iBTK&quot;, &quot;_iPI3Kd&quot;, &quot;_iNFkB&quot;)
for (heat_data in c(1:length(heat_list_Oi))) {
  # print(heat_data)
  alt_data &lt;- heat_list_Oi[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_Oi) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_Oi &lt;- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iSYK conc on activated signaling
heat_list_iSYK_Os &lt;- list(heat_iSYK0.1_Os, heat_iSYK1_Os, heat_iSYK10_Os, heat_iSYK100_Os)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iSYK_Os))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iSYK_Os[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iSYK_Os) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iSYK_Os &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iBTK conc on activated signaling
heat_list_iBTK_Os &lt;- list(heat_iBTK0.1_Os, heat_iBTK1_Os, heat_iBTK10_Os, heat_iBTK100_Os)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iBTK_Os))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iBTK_Os[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iBTK_Os) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iBTK_Os &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iPI3Kd conc on activated signaling
heat_list_iPI3Kd_Os &lt;- list(heat_iPI3Kd0.1_Os, heat_iPI3Kd1_Os, heat_iPI3Kd10_Os, heat_iPI3Kd100_Os)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iPI3Kd_Os))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iPI3Kd_Os[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iPI3Kd_Os) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iPI3Kd_Os &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Effect of iNFkB conc on activated signaling
heat_list_iNFkB_Os &lt;- list(heat_iNFkB0.1_Os, heat_iNFkB1_Os, heat_iNFkB10_Os, heat_iNFkB100_Os)
heat_names &lt;- c(&quot;sign_data_0.1&quot;, &quot;sign_data_1&quot;, &quot;sign_data_10&quot;, &quot;sign_data_100&quot;)
heat_suffix &lt;- c(&quot;_0.1&quot;, &quot;_1&quot;, &quot;_10&quot;, &quot;_100&quot;)
for (heat_data in c(1:length(heat_list_iNFkB_Os))) {
  # print(heat_data)
  alt_data &lt;- heat_list_iNFkB_Os[[heat_data]] %&gt;%
    filter(target_nospace %in% sign_prot_iNFkB_Os) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iNFkB_Os &lt;- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
# Effect of 100 uM inhibitor on stimulation
clean_data_well_Oi &lt;- data.frame(data_100_stim_Oi) %&gt;% 
  mutate(target_nospace = rownames(data_100_stim_Oi)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value&quot;, 2:c(ncol(data_100_stim_Oi)+1)) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_Oi) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

# Effect of inhibitor conc on activated signaling
data_iSYK_Os &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iSYK_Os)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iSYK&quot;)))])))
alt_data_iSYK_Os &lt;- data.frame(data_iSYK_Os) %&gt;% 
  mutate(target_nospace = rownames(data_iSYK_Os)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iSYK&quot;, 2:c(ncol(data_iSYK_Os)+1))
data_iBTK_Os &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iBTK_Os)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iBTK&quot;)))])))
alt_data_iBTK_Os &lt;- data.frame(data_iBTK_Os) %&gt;% 
  mutate(target_nospace = rownames(data_iBTK_Os)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iBTK&quot;, 2:c(ncol(data_iBTK_Os)+1))
data_iPI3Kd_Os &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iPI3Kd_Os)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iPI3Kd&quot;)))])))
alt_data_iPI3Kd_Os &lt;- data.frame(data_iPI3Kd_Os) %&gt;% 
  mutate(target_nospace = rownames(data_iPI3Kd_Os)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iPI3Kd&quot;, 2:c(ncol(data_iPI3Kd_Os)+1))
data_iNFkB_Os &lt;- t(scale(t(assay(rld)[unique(c(sign_prot_iNFkB_Os)), rownames(filter(data_heatmap, inhib_new %in% c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;, &quot;iNFkB&quot;)))])))
alt_data_iNFkB_Os &lt;- data.frame(data_iNFkB_Os) %&gt;% 
  mutate(target_nospace = rownames(data_iNFkB_Os)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_iNFkB&quot;, 2:c(ncol(data_iNFkB_Os)+1))

clean_data_well_inhib_Os &lt;- list(alt_data_iSYK_Os, alt_data_iBTK_Os, alt_data_iPI3Kd_Os, alt_data_iNFkB_Os) %&gt;% 
  purrr::reduce(full_join) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_inhib_Os) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_iSYK, heatmap_iBTK, heatmap_iPI3Kd, heatmap_iNFkB) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

# All significant proteins for plotting
clean_data_well_O &lt;- counts_dtbl %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  filter(target_nospace %in% sign_prot_Oi) %&gt;%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)</code></pre>
<pre class="r"><code># Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with mean values per condition (replicates combined)
# Effect of 100 uM inhibitor on stimulation
clean_data_mean_Oi &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_Oi &amp; cell_line == &quot;OCI-Ly8&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# Effect of inhibitor conc on activated signaling
clean_data_mean_Os &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_inhib_Os &amp; cell_line == &quot;OCI-Ly8&quot;) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# All significant proteins for plotting
clean_data_mean_O &lt;- mean_dtbl %&gt;%
  filter(target_nospace %in% sign_prot_Oi) %&gt;%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)</code></pre>
<pre class="r"><code>plot &lt;- ggplot(clean_data_well_Oi, 
               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8_all-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8_all-1">
Past versions of fig_counts_act_Ly8_all-1.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Normalised counts for all significantly
changing proteins. For some proteins the effect size is quite
small.</em></p>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

# Re-redesigned fig
filt_clean_data_mean_O &lt;- clean_data_mean_O %&gt;%
  dplyr::group_by(cell_line, inhibitor, inhib_conc, stimulus, target_nospace) %&gt;%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_Oi) {
  plot &lt;- ggplot() +
    geom_jitter(data = subset(clean_data_well_O, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
                    counts_norm, 
                    color = stimulus, 
                    # shape = cell_line
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean_O, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = stimulus,
                      color = stimulus
                      ),
                  width = 0.5, size = 0.75) + #, color = &quot;black&quot;
    # facet_wrap(vars(factor(target_nospace))) +
    # facet_wrap(vars(factor(inhibitor, levels = inhib_DS108)), ncol = 1) +
    facet_grid(vars(factor(cell_line, levels = cells_DS108)), vars(factor(inhibitor, levels = inhib_DS108))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;, title = protein) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-1">
Past versions of fig_counts_act_Ly8-1.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-2">
Past versions of fig_counts_act_Ly8-2.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-3">
Past versions of fig_counts_act_Ly8-3.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-4">
Past versions of fig_counts_act_Ly8-4.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-5">
Past versions of fig_counts_act_Ly8-5.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-6.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-6">
Past versions of fig_counts_act_Ly8-6.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-7.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-7">
Past versions of fig_counts_act_Ly8-7.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-8.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-8">
Past versions of fig_counts_act_Ly8-8.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-9.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-9">
Past versions of fig_counts_act_Ly8-9.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-10.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-10">
Past versions of fig_counts_act_Ly8-10.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-11.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-11">
Past versions of fig_counts_act_Ly8-11.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-12.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-12">
Past versions of fig_counts_act_Ly8-12.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-13.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-13">
Past versions of fig_counts_act_Ly8-13.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-14.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-14">
Past versions of fig_counts_act_Ly8-14.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-15.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-15">
Past versions of fig_counts_act_Ly8-15.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-16.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-16">
Past versions of fig_counts_act_Ly8-16.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-17.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-17">
Past versions of fig_counts_act_Ly8-17.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-18.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-18">
Past versions of fig_counts_act_Ly8-18.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-19.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-19">
Past versions of fig_counts_act_Ly8-19.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-20.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-20">
Past versions of fig_counts_act_Ly8-20.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-21.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-21">
Past versions of fig_counts_act_Ly8-21.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-22.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-22">
Past versions of fig_counts_act_Ly8-22.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-23.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-23">
Past versions of fig_counts_act_Ly8-23.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-24.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-24">
Past versions of fig_counts_act_Ly8-24.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-25.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-25">
Past versions of fig_counts_act_Ly8-25.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-26.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-26">
Past versions of fig_counts_act_Ly8-26.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS108_p_seq_comparisons.Rmd/fig_counts_act_Ly8-27.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_Ly8-27">
Past versions of fig_counts_act_Ly8-27.png
</button>
</p>
<div id="fig-fig_counts_act_Ly8-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="oci-ly8-save-data" class="section level3">
<h3>OCI-Ly8: Save data</h3>
<p>Save all information and datasets as .csv file</p>
<p>Effect of 100 uM inhibitor on aIg+H2O2 vs PBS comparison (OCI-Ly8
cells only):</p>
<ul>
<li><p>Ly8_100uM_stim_sign_proteins.txt: List of significantly affected
proteins by 100 uM inhibitor in the aIg+H2O2 vs PBS comparison (by at
least one inhibitor)</p></li>
<li><p>Ly8_100uM_stim_sign_DESeq2data.csv: Dataset with significantly
changing proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>Ly8_100uM_stim_sign_sample.csv: Dataset of counts per sample per
protein + heatmap data for significantly changing proteins</p></li>
<li><p>Ly8_100uM_stim_sign_condition.csv: Dataset of counts per
condition (mean of repl) per protein for significantly changing
proteins</p></li>
</ul>
<p>Effect of inhibitor conc vs DMSO ctrl (for each inhibitor, OCI-Ly8 w/
stim only): For each inhibitor (iSYK, iBTK, iPI3Kd, iNFkB):</p>
<ul>
<li><p>Ly8_inhib_sign_proteins.txt: List of significantly different
proteins in any of the inhibitor conc vs DMSO ctrl comparisons</p></li>
<li><p>Ly8_inhib_sign_DESeq2data.csv: Dataset with significantly
different proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>Ly8_inhib_sign_sample.csv: Dataset of counts per sample per
protein + heatmap data for significantly changing proteins</p></li>
<li><p>Ly8_inhib_sign_condition.csv: Dataset of counts per condition
(mean of repl) per protein for significantly changing proteins</p></li>
</ul>
<pre class="r"><code># List of significantly changing proteins: sign_prot_cell
write(sign_prot_Oi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_proteins.txt&quot;)
write(sign_prot_iSYK_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iSYK_sign_proteins.txt&quot;)
write(sign_prot_iBTK_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iBTK_sign_proteins.txt&quot;)
write(sign_prot_iPI3Kd_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iPI3Kd_sign_proteins.txt&quot;)
write(sign_prot_iNFkB_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iNFkB_sign_proteins.txt&quot;)

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_Oi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iSYK_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iSYK_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iBTK_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iBTK_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iPI3Kd_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iPI3Kd_sign_DESeq2data.csv&quot;, row.names = F)
write.csv(clean_data_sign_iNFkB_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iNFkB_sign_DESeq2data.csv&quot;, row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_Oi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_sample.csv&quot;, row.names = F)
write.csv(clean_data_well_inhib_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_inhib_sign_sample.csv&quot;, row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_Oi, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_condition.csv&quot;, row.names = F)
write.csv(clean_data_mean_Os, file = &quot;output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_inhib_sign_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9879554 527.7   16263485 868.6 16263485 868.6
Vcells 25643060 195.7   54239823 413.9 40247824 307.1</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ashr_2.2-63                 apeglm_1.26.1              
 [3] MASS_7.3-60.2               ggbreak_0.1.2              
 [5] geomtextpath_0.1.4          ggh4x_0.2.8                
 [7] cowplot_1.1.3               patchwork_1.2.0            
 [9] greekLetters_1.0.2          ggpubr_0.6.0               
[11] ggrepel_0.9.5               scales_1.3.0               
[13] ggthemes_5.1.0              RColorBrewer_1.1-3         
[15] scico_1.5.0                 viridis_0.6.5              
[17] viridisLite_0.4.2           gridExtra_2.3              
[19] mmtable2_0.1.3              gt_0.10.1                  
[21] ggvenn_0.1.10               rgl_1.3.1                  
[23] plotly_4.10.4               ggcorrplot_0.1.4.1         
[25] corrplot_0.92               umap_0.2.10.0              
[27] ggplotify_0.1.2             dendsort_0.3.4             
[29] vsn_3.72.0                  ComplexHeatmap_2.20.0      
[31] platetools_0.1.7            kableExtra_1.4.0           
[33] DESeq2_1.44.0               SummarizedExperiment_1.34.0
[35] Biobase_2.64.0              MatrixGenerics_1.16.0      
[37] matrixStats_1.3.0           GenomicRanges_1.56.1       
[39] GenomeInfoDb_1.40.1         IRanges_2.38.0             
[41] S4Vectors_0.42.0            BiocGenerics_0.50.0        
[43] Seurat_5.1.0                SeuratObject_5.0.2         
[45] sp_2.1-4                    rstatix_0.7.2              
[47] factoextra_1.0.7            FactoMineR_2.11            
[49] readxl_1.4.3                Matrix_1.7-0               
[51] lubridate_1.9.3             forcats_1.0.0              
[53] stringr_1.5.1               dplyr_1.1.4                
[55] purrr_1.0.2                 readr_2.1.5                
[57] tidyr_1.3.1                 tibble_3.2.1               
[59] ggplot2_3.5.1               tidyverse_2.0.0            
[61] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             goftest_1.2-3           DT_0.33                
  [4] TH.data_1.1-2           vctrs_0.6.5             spatstat.random_3.2-3  
  [7] digest_0.6.36           png_0.1-8               shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       magick_2.8.3            SQUAREM_2021.1         
 [16] reshape2_1.4.4          httpuv_1.6.15           foreach_1.5.2          
 [19] withr_3.0.0             xfun_0.45               ggfun_0.1.5            
 [22] survival_3.6-4          memoise_2.0.1           hexbin_1.28.3          
 [25] emmeans_1.10.2          systemfonts_1.1.0       zoo_1.8-12             
 [28] GlobalOptions_0.1.2     pbapply_1.7-2           promises_1.3.0         
 [31] scatterplot3d_0.3-44    httr_1.4.7              globals_0.16.3         
 [34] fitdistrplus_1.1-11     ps_1.7.6                rstudioapi_0.16.0      
 [37] UCSC.utils_1.0.0        miniUI_0.1.1.1          generics_0.1.3         
 [40] base64enc_0.1-3         processx_3.8.4          zlibbioc_1.50.0        
 [43] polyclip_1.10-6         GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [46] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [49] S4Arrays_1.4.1          preprocessCore_1.66.0   hms_1.1.3              
 [52] irlba_2.3.5.1           colorspace_2.1-0        ROCR_1.0-11            
 [55] reticulate_1.38.0       spatstat.data_3.1-2     magrittr_2.0.3         
 [58] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [61] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [64] scattermore_1.2         RcppAnnoy_0.0.22        pillar_1.9.0           
 [67] nlme_3.1-164            iterators_1.0.14        compiler_4.4.1         
 [70] RSpectra_0.16-1         stringi_1.8.4           tensor_1.5             
 [73] plyr_1.8.9              crayon_1.5.3            abind_1.4-5            
 [76] truncnorm_1.0-9         gridGraphics_0.5-1      emdbook_1.3.13         
 [79] locfit_1.5-9.10         bit_4.0.5               sandwich_3.1-0         
 [82] whisker_0.4.1           codetools_0.2-20        multcomp_1.4-25        
 [85] textshaping_0.4.0       openssl_2.2.0           bslib_0.7.0            
 [88] GetoptLong_1.0.5        mime_0.12               splines_4.4.1          
 [91] circlize_0.4.16         Rcpp_1.0.12             fastDummies_1.7.3      
 [94] cellranger_1.1.0        Rttf2pt1_1.3.12         leaps_3.2              
 [97] knitr_1.47              utf8_1.2.4              clue_0.3-65            
[100] fs_1.6.4                listenv_0.9.1           ggsignif_0.6.4         
[103] estimability_1.5.1      callr_3.7.6             statmod_1.5.0          
[106] tzdb_0.4.0              svglite_2.1.3           pkgconfig_2.0.3        
[109] tools_4.4.1             cachem_1.1.0            numDeriv_2016.8-1.1    
[112] fastmap_1.2.0           rmarkdown_2.27          ica_1.0-3              
[115] broom_1.0.6             sass_0.4.9              coda_0.19-4.1          
[118] BiocManager_1.30.23     dotCall64_1.1-1         carData_3.0-5          
[121] RANN_2.6.1              farver_2.1.2            yaml_2.3.8             
[124] cli_3.6.3               leiden_0.4.3.1          lifecycle_1.0.4        
[127] askpass_1.2.0           uwot_0.2.2              mvtnorm_1.2-5          
[130] backports_1.5.0         BiocParallel_1.38.0     timechange_0.3.0       
[133] gtable_0.3.5            rjson_0.2.21            ggridges_0.5.6         
[136] progressr_0.14.0        parallel_4.4.1          limma_3.60.3           
[139] jsonlite_1.8.8          RcppHNSW_0.6.0          bit64_4.0.5            
[142] multcompView_0.1-10     assertthat_0.2.1        Rtsne_0.17             
[145] yulab.utils_0.1.4       spatstat.utils_3.0-5    bdsmatrix_1.3-7        
[148] highr_0.11              jquerylib_0.1.4         lazyeval_0.2.2         
[151] shiny_1.8.1.1           htmltools_0.5.8.1       affy_1.82.0            
[154] sctransform_0.4.1       glue_1.7.0              spam_2.10-0            
[157] XVector_0.44.0          rprojroot_2.0.4         flashClust_1.01-2      
[160] invgamma_1.1            igraph_2.0.3            extrafontdb_1.0        
[163] R6_2.5.1                labeling_0.4.3          bbmle_1.0.25.1         
[166] cluster_2.1.6           aplot_0.2.3             DelayedArray_0.30.1    
[169] tidyselect_1.2.1        xml2_1.3.6              car_3.1-2              
[172] future_1.33.2           munsell_0.5.1           KernSmooth_2.23-24     
[175] affyio_1.74.0           data.table_1.15.4       htmlwidgets_1.6.4      
[178] rlang_1.1.4             spatstat.sparse_3.1-0   extrafont_0.19         
[181] spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
