<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-05-29" />

<title>DS113 ID-seq analysis: Preliminary analysis</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="site_libs/plotly-binding-4.10.4/plotly.js"></script>
<script src="site_libs/typedarray-0.1/typedarray.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/plotly-htmlwidgets-css-2.11.1/plotly-htmlwidgets.css" rel="stylesheet" />
<script src="site_libs/plotly-main-2.11.1/plotly-latest.min.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapters on DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Paper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="y_paper_main.html">Main</a>
    </li>
    <li>
      <a href="z_paper_suppl.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteins ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="t_proteins_main.html.html">Main</a>
    </li>
    <li>
      <a href="u_proteins_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    States ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="v_states_main.html.html">Main</a>
    </li>
    <li>
      <a href="w_states_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
    <li>
      <a href="c_flow_vis.html">Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="DS108_p_seq_comparisons.html">DS108 clean</a>
    </li>
    <li>
      <a href="DS108_q_seq_thresholds.html">DS108 thresholds</a>
    </li>
    <li>
      <a href="DS108_r_seq_pca.html">DS108 PCA</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">DS113 ID-seq analysis: Preliminary
analysis</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-05-29</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapters_DLBCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241022code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241022code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241022)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong94c226d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 94c226d
</a>
</p>
</div>
<div id="strongRepositoryversionstrong94c226d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 94c226d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/AL006_InhibCombiTest/
    Untracked:  data/DS102_StimTest/
    Untracked:  data/DS103_ViabilityTest/
    Untracked:  data/DS104_StimInhibTest/
    Untracked:  data/DS108_StimInhibIDseq/
    Untracked:  data/DS113_InhibCombiIDseq/
    Untracked:  output/DS102_StimTest/
    Untracked:  output/DS103_ViabilityTest/
    Untracked:  output/DS104_StimInhibTest/
    Untracked:  output/DS108_StimInhibIDseq/
    Untracked:  output/DS113_InhibCombiIDseq/
    Untracked:  output/figures/
    Untracked:  output/network_visual_DLBCL/
    Untracked:  output/network_visual_paper/

Unstaged changes:
    Deleted:    analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/DS113_s_seq_prelim.Rmd</code>)
and HTML (<code>docs/DS113_s_seq_prelim.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
6559698
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Paper figs + analysis files for ch figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)

textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
                  axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
                  axis.title = element_text(colour = &quot;black&quot;, size = 12), 
                  legend.title = element_text(colour = &quot;black&quot;, size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
                  strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
)

# textsize_small &lt;- theme(text = element_text(size = 7, family = &quot;sans&quot;, colour = &quot;black&quot;),
#                         plot.title = element_text(size = 8)
# )
textsize_small &lt;- theme(axis.text.x = element_text(colour = &quot;black&quot;, size = 9),
                        axis.text.y = element_text(colour = &quot;black&quot;, size = 9),
                        axis.title = element_text(colour = &quot;black&quot;, size = 10),
                        legend.text = element_text(colour = &quot;black&quot;, size = 9),
                        title = element_text(color = &quot;black&quot;, size = 10),
                        strip.text.x = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;),
                        strip.text.y = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;)
)

colors_dark9 &lt;- c(&quot;#4daf4a&quot;, &quot;#984ea3&quot;, &quot;#377eb8&quot;, &quot;#ff7f00&quot;, &quot;#f781bf&quot;, &quot;#ffff33&quot;, &quot;#e41a1c&quot;, &quot;#a65628&quot;, &quot;#999999&quot;)
colors_light12 &lt;- c(&quot;#8dd3c7&quot;,&quot;#ffffb3&quot;,&quot;#bebada&quot;,&quot;#fb8072&quot;,&quot;#80b1d3&quot;,&quot;#fdb462&quot;,&quot;#b3de69&quot;,&quot;#fccde5&quot;,&quot;#d9d9d9&quot;,&quot;#bc80bd&quot;,&quot;#ccebc5&quot;,&quot;#ffed6f&quot;)
colors_paired10 &lt;- c(&quot;#a6cee3&quot;, &quot;#1f78b4&quot;, &quot;#b2df8a&quot;, &quot;#33a02c&quot;, &quot;#fdbf6f&quot;, &quot;#ff7f00&quot;, &quot;#cab2d6&quot;, &quot;#6a3d9a&quot;, &quot;#fb9a99&quot;, &quot;#e31a1c&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)

colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

colors_proteins &lt;- c(&quot;pCD79a (Y182)&quot; = &quot;#892BE1&quot;, &quot;pSYK (Y525/Y526)&quot; = &quot;#1E8FFF&quot;, &quot;pPLCy2 (Y759)&quot; = &quot;#20B1A9&quot;)
colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#41ab5d&quot;, &quot;iBTK&quot; = &quot;#4292c6&quot;, &quot;iPI3Kd&quot; = &quot;#807dba&quot;, &quot;iNFkB&quot; = &quot;#f16913&quot;)
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#525252&quot;, &quot;aIg+H2O2&quot; = &quot;#fdae6b&quot;)
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#df65b0&quot;, &quot;OCI-Ly8&quot; = &quot;#41b6c4&quot;)
colors_conc &lt;- c(&quot;0 + 0 + 0 uM&quot; = &quot;#dadaeb&quot;, &quot;1 + 1 + 1 uM&quot; = &quot;#bcbddc&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot; = &quot;#6a51a3&quot;, &quot;10 + 10 + 10 uM&quot; = &quot;#3f007d&quot;, 
                 &quot;10 + 0 + 0 uM&quot; = &quot;#54278f&quot;, &quot;0 + 10 + 0 uM&quot; = &quot;#54278f&quot;, &quot;0 + 0 + 10 uM&quot; = &quot;#54278f&quot;, 
                 &quot;2.5 + 2.5 + 0 uM&quot; = &quot;#9e9ac8&quot;, &quot;2.5 + 0 + 2.5 uM&quot; = &quot;#9e9ac8&quot;, &quot;0 + 2.5 + 2.5 uM&quot; = &quot;#9e9ac8&quot;, 
                 &quot;5 + 1 + 1 uM&quot; = &quot;#807dba&quot;, &quot;1 + 5 + 1 uM&quot; = &quot;#807dba&quot;, &quot;1 + 1 + 5 uM&quot; = &quot;#807dba&quot;, 
                 &quot;10 + 10 + 10 uM (nostain)&quot; = &quot;#969696&quot;)
colors_conc_uniq &lt;- c(
  &quot;0 + 0 + 0 uM&quot; = &quot;#bcbddc&quot;, &quot;1 + 1 + 1 uM&quot; = &quot;#9e9ac8&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot; = &quot;#6a51a3&quot;, &quot;10 + 10 + 10 uM&quot; = &quot;#3f007d&quot;, 
  &quot;10 + 0 + 0 uM&quot; = &quot;#006d2c&quot;, &quot;0 + 10 + 0 uM&quot; = &quot;#41ab5d&quot;, &quot;0 + 0 + 10 uM&quot; = &quot;#a1d99b&quot;, 
  &quot;2.5 + 2.5 + 0 uM&quot; = &quot;#a63603&quot;, &quot;2.5 + 0 + 2.5 uM&quot; = &quot;#f16913&quot;, &quot;0 + 2.5 + 2.5 uM&quot; = &quot;#fdae6b&quot;, 
  &quot;5 + 1 + 1 uM&quot; = &quot;#08519c&quot;, &quot;1 + 5 + 1 uM&quot; = &quot;#4292c6&quot;, &quot;1 + 1 + 5 uM&quot; = &quot;#9ecae1&quot;, 
  &quot;10 + 10 + 10 uM (nostain)&quot; = &quot;#969696&quot;)</code></pre>
<pre class="r"><code>stim_DS113 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label &lt;- c(&quot;basal&quot;, &quot;activated&quot;)
inhib_DS113 &lt;- c(&quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iNFkB&quot;)
legend_label &lt;- &quot;iSYK + iBTK + iNFkB&quot;
inhib_conc_DS113 &lt;- c(&quot;0 + 0 + 0 uM&quot;, &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 10 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, &quot;2.5 + 2.5 + 0 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 5 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, &quot;10 + 10 + 10 uM (nostain)&quot;)
inhib_single_conc_DS113 &lt;- c(0, 1, 2.5, 10, 5)
inhib_group_DS113 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)</code></pre>
</div>
<div id="data-qc" class="section level2">
<h2>Data &amp; QC</h2>
<div id="raw-data" class="section level4">
<h4>Raw data</h4>
<pre class="r"><code># Load sparse matrix format all data
counts_PlateA &lt;- read_count_output(&quot;data/DS113_InhibCombiIDseq/counts/PlateA/featurecounts&quot;, name = &quot;featurecounts&quot;)
meta_wellID &lt;- read_csv(&quot;data/DS113_InhibCombiIDseq/config/DynSign.113_wellBC_metadata_seq.csv&quot;)

meta_Abs &lt;- read_excel(&quot;data/DS113_InhibCombiIDseq/config/BulkIDseq_Ab_info_seq_analysis_202405.xlsx&quot;)

# Reshape counts in data table format
counts_dtbl &lt;- data.frame(counts_PlateA) %&gt;% 
  mutate(barcode_name = rownames(counts_PlateA)) %&gt;% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %&gt;%
  gather(&quot;well_BC_seq&quot;, &quot;counts&quot;, 2:c(ncol(counts_PlateA)+1)) %&gt;%
  dplyr::filter(counts &gt;= 1) %&gt;% # Remove undetected proteins counts
  left_join(meta_Abs) %&gt;%
  unique() %&gt;%
  # dplyr::filter(!is.na(target_nospace)) %&gt;% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID, plate == &quot;A&quot;)) %&gt;% # Add sample metadata
  dplyr::filter(!is.na(experiment)) %&gt;%
  arrange(plate, well)

counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(plate_well = paste0(plate, &quot;_&quot;, well), 
         stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113))

# Calculate per-sample properties
data_properties &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount = sum(counts), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113))</code></pre>
<p><em><strong>Table</strong>: Unfiltered data-table properties loaded
into R</em></p>
<table>
<colgroup>
<col width="25%" />
<col width="37%" />
<col width="37%" />
</colgroup>
<thead>
<tr class="header">
<th></th>
<th>nCols (#samples)</th>
<th>nRows (# barcodes)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Plate A</td>
<td>94</td>
<td>300</td>
</tr>
</tbody>
</table>
<p>The full panel of mapped barcodes and 96 wells are present in the
data table. First have a look at the total counts and number of detected
protein barcodes per sample: There are a lot more proteins detected than
put in the panel, due to barcode contamination and cross-ver between
wells. The nostain samples have relatively low counts and proteins
detected.</p>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(data_properties, aes(log10(nCount), nProt, color = factor(inhib_conc, levels = inhib_conc_DS113))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(~stimulus) +
  scale_color_manual(values = colors_conc, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation total counts and number of detected barcodes&quot;, 
       x = expression(&quot;Total counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_count_prot-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_prot-1">
Past versions of fig_count_prot-1.png
</button>
</p>
<div id="fig-fig_count_prot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Correlation between total counts and
number of detected protein barcodes. We observe many more detected
proteins than present in the staining.</em></p>
<p>As we have a 96-wells plate, we can also have a look at total counts
per plate detected:</p>
<pre class="r"><code>options(repr.plot.width = 10, repr.plot.height = 6)

rng &lt;- range(log10(data_properties$nCount))

plot &lt;- raw_grid(data = log10(data_properties$nCount), well = data_properties$well, plate_id = data_properties$plate, plate = 96, size = 6) +
  scale_fill_viridis(expression(&quot;Total counts Log&quot;[10]), limits = c(rng[1], rng[2])) +
  theme_bw()

print(plot)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_plate_totalUMI-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_plate_totalUMI-1">
Past versions of fig_plate_totalUMI-1.png
</button>
</p>
<div id="fig-fig_plate_totalUMI-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Plate overviews of total
counts.</em></p>
</div>
<div id="barcode-contamination-comp" class="section level4">
<h4>Barcode contamination comp</h4>
<p>Several barcodes are contaminated with &lt;2.5% of other barcodes
(all barcodes were sequenced individually to determine the
contamination). As this can skew the results, especially for lowly
present barcodes that are contaminated with highly present barcodes, we
will compensate for this.</p>
<pre class="r"><code>comp_matrix &lt;- read_csv(&quot;data/DS113_InhibCombiIDseq/BC_compensation_matrix.csv&quot;)

data_comp &lt;- counts_dtbl %&gt;%
  # dplyr::filter(well == &quot;A01&quot;) %&gt;%
  dplyr::select(plate, well, barcode_name, counts) %&gt;%
  left_join(comp_matrix, by = join_by(barcode_name == BC_to_correct)) #, multiple = &quot;all&quot;

# Get counts of OG BC
data_comp_OG &lt;- data_comp %&gt;%
  # dplyr::filter(well == &quot;A01&quot;) %&gt;%
  dplyr::filter(barcode_name == OG_BC) %&gt;%
  dplyr::select(plate, well, OG_BC, counts) %&gt;%
  dplyr::rename(counts_OG_BC = counts)

# Calculate false reads
data_false &lt;- left_join(data_comp, data_comp_OG) %&gt;% 
  mutate(false_counts = (counts_OG_BC / percent_OG_BC) * percent_BC_to_correct) %&gt;%
  dplyr::filter(barcode_name != OG_BC) %&gt;%
  group_by(plate, well, barcode_name, counts) %&gt;%
  summarize(total_false_counts = sum(false_counts, na.rm = TRUE))

counts_dtbl &lt;- counts_dtbl %&gt;%
  # dplyr::filter(well == &quot;A01&quot;) %&gt;%
  left_join(data_false) %&gt;% 
  mutate(total_false_counts = replace_na(total_false_counts, 0), 
         true_counts = as.integer(counts - total_false_counts), 
         percent_true = (true_counts / counts) * 100) %&gt;%
  dplyr::filter(!is.na(target_nospace)) # Remove barcodes not in Ab metadata (not conjugated/added in panel)

compensation &lt;- counts_dtbl %&gt;%
  dplyr::select(plate, well, barcode_name, counts, total_false_counts, true_counts, percent_true, target_nospace)

# Calculate per-sample properties
data_properties_true &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount = sum(true_counts), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113))</code></pre>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(counts_dtbl, aes(log10(counts), log10(true_counts), color = factor(inhib_conc, levels = inhib_conc_DS113))) +
  geom_point(alpha = 1, size = 0.8) +
  geom_abline(intercept = 0, slope = 1, linetype = &quot;dashed&quot;) + 
  facet_grid(~stimulus) +
  scale_color_manual(values = colors_conc, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation raw counts and compensated counts&quot;,
       x = expression(&quot;Raw counts Log&quot;[10]),
       y = expression(&quot;Compensated counts Log&quot;[10])) +
  textsize_small</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_count_raw_comp-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_raw_comp-1">
Past versions of fig_count_raw_comp-1.png
</button>
</p>
<div id="fig-fig_count_raw_comp-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(data_properties_true, aes(log10(nCount), nProt, color = factor(inhib_conc, levels = inhib_conc_DS113))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(~stimulus) + 
  scale_color_manual(values = colors_conc, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation total counts and number of detected barcodes&quot;, 
       x = expression(&quot;Total counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_count_raw_comp-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_raw_comp-2">
Past versions of fig_count_raw_comp-2.png
</button>
</p>
<div id="fig-fig_count_raw_comp-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Correlation between raw counts and
compensated counts, and between compensated counts and detected number
of barcodes. After compensation, all samples (except 2 nostain samples)
have 111 detected proteins (the number of Ab in the panel).</em></p>
</div>
<div id="count-normalisation" class="section level4">
<h4>Count normalisation</h4>
<p>To remove variation between replicates, the data is normalised per
target protein with a geometric mean scaling factor: First, the
geometric mean of each target is calculated separately for each cell
line, and then the median of the protein scaling factors is calculated
separately for each sample.</p>
<pre class="r"><code>scaling_factors &lt;- counts_dtbl %&gt;%
  dplyr::filter(staining == &quot;yes&quot;) %&gt;%
  # dplyr::filter() %&gt;% #target_nospace == &quot;GAPDH&quot; # modification == &quot;no&quot;
  ungroup() %&gt;%
  group_by(target_nospace) %&gt;%
  mutate(scaling_factor = counts / median(counts),
         scaling_factor_geo = counts / exp(mean(log((counts)))), 
         true_scaling_factor = true_counts / median(true_counts),
         true_scaling_factor_geo = true_counts / exp(mean(log((true_counts))))) %&gt;%
  ungroup() %&gt;%
  group_by(descript_rep) %&gt;%
  summarize(scaling_factor = median(scaling_factor), .groups = &quot;keep&quot;,
            scaling_factor_geo = median(scaling_factor_geo), 
            true_scaling_factor = median(true_scaling_factor),
            true_scaling_factor_geo = median(true_scaling_factor_geo, na.rm = TRUE))

counts_dtbl &lt;- counts_dtbl %&gt;%
  dplyr::filter(staining == &quot;yes&quot;) %&gt;%
  left_join(scaling_factors) %&gt;%
  ungroup() %&gt;%
  mutate(counts_norm = counts / scaling_factor,
         counts_norm_geo = counts / scaling_factor_geo, 
         true_counts_norm = true_counts / true_scaling_factor, 
         true_counts_norm_geo = true_counts / true_scaling_factor_geo)

counts_dtbl &lt;- dplyr::filter(counts_dtbl, !is.na(target_nospace))

# Calculate per-sample properties
data_properties_norm &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount_norm = sum(counts_norm_geo), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113))

data_properties_norm_true &lt;- counts_dtbl %&gt;%
  unique()%&gt;%
  group_by(plate, well_BC_seq) %&gt;%
  summarize(nCount_norm = sum(true_counts_norm_geo), 
            nProt = n()) %&gt;%
  left_join(meta_wellID) %&gt;% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113))</code></pre>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(counts_dtbl, aes(log10(counts), log10(true_counts_norm_geo), color = factor(inhib_conc, levels = inhib_conc_DS113))) +
  geom_point(alpha = 1, size = 0.8) +
  geom_abline(intercept = 0, slope = 1, linetype = &quot;dashed&quot;) + 
  facet_grid(~stimulus) +
  scale_color_manual(values = colors_conc, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation raw counts and normalised counts&quot;,
       x = expression(&quot;Raw counts Log&quot;[10]),
       y = expression(&quot;Normalised counts Log&quot;[10])) +
  textsize_small</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_count_norm-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_norm-1">
Past versions of fig_count_norm-1.png
</button>
</p>
<div id="fig-fig_count_norm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(data_properties_norm_true, aes(log10(nCount_norm), nProt, color = factor(inhib_conc, levels = inhib_conc_DS113))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(~stimulus) + 
  scale_color_manual(values = colors_conc, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  theme_bw() +
  labs(title = &quot;Correlation normalised counts and number of detected barcodes&quot;, 
       x = expression(&quot;Normalised counts Log&quot;[10]),
       y = &quot;Detected number of protein barcodes&quot;) +
  textsize_small</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_count_norm-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_count_norm-2">
Past versions of fig_count_norm-2.png
</button>
</p>
<div id="fig-fig_count_norm-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Correlation between raw counts and
compensated counts.</em></p>
</div>
<div id="data-wrangling" class="section level4">
<h4>Data wrangling</h4>
<p>Calculate the following:</p>
<ul>
<li>Mean + sd for replicates (true_counts + true_counts_norm_geo)</li>
</ul>
<p><em>Manual fold changes of all samples compared to the mean of 0 uM
(DMSO) + PBS samples</em></p>
<p><em>% phosphoprotein of total protein (only for proteins with both
total + phospho targets)</em></p>
<pre class="r"><code># Calculate mean + sd of true_counts and true_counts_norm_geo
mean_dtbl &lt;- counts_dtbl %&gt;%
  dplyr::group_by(description, target_nospace) %&gt;%
  summarise(
    counts_sd = sd(true_counts, na.rm = T),
    counts = mean(true_counts, na.rm = T), 
    counts_norm_sd = sd(true_counts_norm_geo, na.rm = T),
    counts_norm = mean(true_counts_norm_geo, na.rm = T)
    )

# Add metadata
mean_dtbl &lt;- mean_dtbl %&gt;%
  left_join(dplyr::select(counts_dtbl,
                   c(barcode_name, DS_Ab_nr, target, modification, epitope, marker_cell, marker_pathway, marker_pathway_detail, isotype, DS_barcode_nr, DS_barcode_seq, target_nospace, cell_line, stimulus, stim_clean, inhibitor, inhib_new, inhib_conc, inhib_conc_new, inhib_group, iSYK_conc_uM, iBTK_conc_uM, iNFkB_conc_uM, inhib_conc_text, inhib_text_new, staining, description))) %&gt;%
  distinct()</code></pre>
<pre class="r"><code>prot &lt;- &quot;BTK_Y551&quot;
# prot &lt;- &quot;VAV1_Y174&quot;

# Raw (compensated) counts
plot1 &lt;- ggplot(subset(mean_dtbl, staining == &quot;yes&quot; &amp; target_nospace == prot)) +
  geom_col(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS113), 
      y = counts, 
      group = stimulus,
      fill = stimulus),
      position = &quot;dodge2&quot;) + 
  geom_errorbar(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS113), 
      ymin = counts - counts_sd,
      ymax = counts + counts_sd,
      group = stimulus),
      position = position_dodge2(width = 0.9),
      width = 0.9,
    ) +
  facet_wrap(vars(factor(cell_line, labels = &quot;Raw (compensated)&quot;)), scales = &quot;free_y&quot;) +
  theme_bw() +
  # scale_y_log10() +
  RotatedAxis() + 
  scale_fill_manual(values = colors_stim) +
  labs(x = &quot;&quot;, y = &quot;&quot;, title = prot) +
  textsize_small

# Normalised counts
plot2 &lt;- ggplot(subset(mean_dtbl, staining == &quot;yes&quot; &amp; target_nospace == prot)) +
  geom_col(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS113), 
      y = counts_norm, 
      group = stimulus,
      fill = stimulus),
      position = &quot;dodge2&quot;) + 
  geom_errorbar(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS113), 
      ymin = counts_norm - counts_norm_sd,
      ymax = counts_norm + counts_norm_sd,
      group = stimulus),
      position = position_dodge2(width = 0.9),
      width = 0.9,
    ) +
  facet_wrap(vars(factor(cell_line, labels = &quot;Normalised&quot;)), scales = &quot;free_y&quot;) +
  theme_bw() +
  # scale_y_log10() +
  RotatedAxis() +
  scale_fill_manual(values = colors_stim) +
  labs(x = &quot;&quot;, y = &quot;&quot;, title = prot) +
  textsize_small

plot_grid(plot1, plot2, ncol = 2)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_wrangle1-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_wrangle1-1">
Past versions of fig_wrangle1-1.png
</button>
</p>
<div id="fig-fig_wrangle1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Mean signal (+ sd) of replicates for BTK
(Y551). Raw counts on the left, normalised counts on the right.
Normalisation removes the variation between replicates but not the
inhibitor effects.</em></p>
<pre class="r"><code># Manually calculate the fold change of true_counts and true_counts_norm_geo
# Use the mean of same inhibitor conc + PBS as baseline
counts_FC0 &lt;- counts_dtbl %&gt;%
  dplyr::filter(stimulus == &quot;PBS&quot;) %&gt;% # &amp; inhib_conc == &quot;0 uM (DMSO)&quot;
  group_by(cell_line, inhib_conc_text, target_nospace) %&gt;%
  summarise(counts_0 = mean(true_counts), 
            counts_norm_0 = mean(true_counts_norm_geo))

counts_dtbl &lt;- left_join(counts_dtbl, counts_FC0) %&gt;%
  mutate(man_FC_counts = (true_counts / counts_0), 
         man_log2FC_counts = log2(man_FC_counts), 
         man_FC_norm = (true_counts_norm_geo / counts_norm_0), 
         man_log2FC_norm = log2(man_FC_norm))

# Calculate mean + sd of manual fold changes of replicates
mean_FC &lt;- counts_dtbl %&gt;%
  dplyr::group_by(description, target_nospace) %&gt;%
  summarise(
    FC_counts_sd = sd(man_FC_counts, na.rm = T),
    FC_counts = mean(man_FC_counts, na.rm = T), 
    log2FC_counts_sd = sd(man_log2FC_counts, na.rm = T), 
    log2FC_counts = mean(man_log2FC_counts, na.rm = T), 
    FC_norm_sd = sd(man_FC_norm, na.rm = T),
    FC_norm = mean(man_FC_norm, na.rm = T), 
    log2FC_norm_sd = sd(man_log2FC_norm, na.rm = T), 
    log2FC_norm = mean(man_log2FC_norm, na.rm = T)
)

mean_dtbl &lt;- mean_dtbl %&gt;%
  left_join(mean_FC) %&gt;%
  distinct()</code></pre>
<pre class="r"><code>prot &lt;- &quot;BTK_Y551&quot;

# Normalised counts
plot &lt;- ggplot(subset(counts_dtbl, staining == &quot;yes&quot; &amp; cell_line == &quot;HBL1&quot; &amp; target_nospace == prot)) +
  geom_point(aes(
    x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = man_FC_norm,
    color = stimulus)) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_DS108))) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim) +
  labs(x = &quot;&quot;, y = &quot;Fold change over PBS (same inhib conc) \n(normalised counts)&quot;, title = prot) +
  textsize_small

print(plot)</code></pre>
<p><em><strong>Figure:</strong> Manually calculated fold change of
signal compared to 0 uM inhibitor (DMSO ctrl) + PBS stimulation for BTK
(Y551) (HBL1 cells, normalised counts). </em></p>
<p>Finally, we will save the datasets as .csv file</p>
<ul>
<li><p>IDseq_data_sample.csv: Dataset of counts per protein per
sample</p></li>
<li><p>IDseq_data_condition.csv: Dataset of counts per protein per
condition (mean of replicates)</p></li>
</ul>
<pre class="r"><code># Dataset of counts per protein per sample 
meta_cols_well &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;,
                    &quot;target_nospace&quot;, &quot;target&quot;, &quot;modification&quot;, &quot;marker_cell&quot;, &quot;marker_pathway&quot;, &quot;marker_pathway_detail&quot;, &quot;barcode_name&quot;, &quot;DS_Ab_nr&quot;, 
                    &quot;true_counts&quot;, &quot;true_scaling_factor_geo&quot;, &quot;true_counts_norm_geo&quot;)
data_well_all &lt;- counts_dtbl %&gt;% 
  arrange(plate_well, target_nospace) %&gt;%
  dplyr::select(meta_cols_well) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

write.csv(data_well_all, file = &quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;, row.names = F)

# Dataset of mean counts per protein per condition (replicates combined)
meta_cols_mean &lt;- c(&quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;description&quot;, 
                    &quot;target_nospace&quot;, &quot;target&quot;, &quot;modification&quot;, &quot;marker_cell&quot;, &quot;marker_pathway&quot;, &quot;marker_pathway_detail&quot;, &quot;barcode_name&quot;, &quot;DS_Ab_nr&quot;, 
                    &quot;counts&quot;, &quot;counts_sd&quot;, &quot;counts_norm&quot;, &quot;counts_norm_sd&quot;)
data_mean_all &lt;- mean_dtbl %&gt;%
  dplyr::select(meta_cols_mean) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name)

write.csv(data_mean_all, file = &quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code>rm(list = ls(pattern = &quot;Plate&quot;))
rm(list = ls(pattern = &quot;comp&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9345699 499.2   14474182 773.1 14474182 773.1
Vcells 16798120 128.2   26573924 202.8 22078266 168.5</code></pre>
</div>
</div>
<div id="counts-analysis" class="section level2">
<h2>Counts analysis</h2>
<div id="counts-per-sample" class="section level4">
<h4>Counts per sample</h4>
<p>To gain insight into the effect of experimental conditions on total
signal, we first have a look at the total (raw) and normalised counts
per sample.</p>
<pre class="r"><code># Compensated counts
fig_sample_true &lt;- ggplot(data_properties_true, aes(inhib_conc, nCount, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_wrap(vars(factor(inhib_group, levels = inhib_group_DS113)), nrow = 1, scales = &quot;free_x&quot;) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10() +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Compensated counts (log10 scale)&quot;, title = &quot;Compensated counts&quot;) + 
  textsize_small

print(fig_sample_true)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_sample-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_sample-1">
Past versions of fig_counts_sample-1.png
</button>
</p>
<div id="fig-fig_counts_sample-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Normalised compensated counts
fig_sample_norm &lt;- ggplot(data_properties_norm_true, aes(inhib_conc, nCount_norm, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_wrap(vars(factor(inhib_group, levels = inhib_group_DS113)), nrow = 1, scales = &quot;free_x&quot;) +
  theme_bw() +
  RotatedAxis() +
  # scale_y_log10() +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) + 
  textsize_small

print(fig_sample_norm)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_sample-2.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_sample-2">
Past versions of fig_counts_sample-2.png
</button>
</p>
<div id="fig-fig_counts_sample-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Compensated counts
fig_sample_true &lt;- ggplot(subset(data_properties_true, staining == &quot;yes&quot;), aes(inhib_conc, nCount, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_wrap(vars(factor(inhib_group, levels = inhib_group_DS113)), nrow = 1, scales = &quot;free_x&quot;) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(limits = c(5e4, 5e5)) +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) + 
  textsize_small

# Normalised compensated counts
fig_sample_norm &lt;- ggplot(data_properties_norm_true, aes(inhib_conc, nCount_norm, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_wrap(vars(factor(inhib_group, levels = inhib_group_DS113)), nrow = 1, scales = &quot;free_x&quot;) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(limits = c(5e4, 5e5)) +
  scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) + 
  textsize_small

plot_grid(fig_sample_true, fig_sample_norm, ncol = 2)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_sample2-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_sample2-1">
Past versions of fig_counts_sample2-1.png
</button>
</p>
<div id="fig-fig_counts_sample2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution counts per sample with
compensation (left) and normalisation (right).</em></p>
</div>
<div id="counts-per-ab" class="section level4">
<h4>Counts per Ab</h4>
<p>We can also look at the counts per Ab for selected samples to get an
indication of any differences between proteins and between aIg+H2O2 and
PBS stimulation.</p>
<pre class="r"><code># Compensated counts
fig_Ab_true &lt;- ggplot(unique(subset(counts_dtbl, inhib_conc == &quot;0 + 0 + 0 uM&quot;)), 
       aes(x = reorder(target_nospace, true_counts), 
           y = true_counts, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  RotatedAxis() + 
  # theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) +
  theme_bw() +
  facet_wrap(~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

print(fig_Ab_true)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_Ab_counts-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts-1">
Past versions of fig_Ab_counts-1.png
</button>
</p>
<div id="fig-fig_Ab_counts-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution of compensated counts per
Ab-barcode for the 0 uM (DMSO) samples (non-normalised).</em></p>
<pre class="r"><code># Compensated counts
fig_Ab_norm &lt;- ggplot(unique(subset(counts_dtbl, inhib_conc == &quot;0 + 0 + 0 uM&quot;)), 
       aes(x = reorder(target_nospace, true_counts_norm_geo), 
           y = true_counts_norm_geo, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  RotatedAxis() + 
  # theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) +
  theme_bw() +
  facet_wrap(~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

print(fig_Ab_norm)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_Ab_counts_norm-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts_norm-1">
Past versions of fig_Ab_counts_norm-1.png
</button>
</p>
<div id="fig-fig_Ab_counts_norm-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution of normalised compensated
counts per Ab-barcode for the 0 uM (DMSO) samples. Here we see some
protein targets that clearly show a difference between the aIg+H2O2 and
PBS samples.</em></p>
<pre class="r"><code>plot_grid(fig_Ab_true, fig_Ab_norm, ncol = 1)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_Ab_counts_HBL1-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts_HBL1-1">
Past versions of fig_Ab_counts_HBL1-1.png
</button>
</p>
<div id="fig-fig_Ab_counts_HBL1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Compensated
fig_Ab_true_HBL1 &lt;- ggplot(unique(subset(counts_dtbl, inhib_conc == &quot;10 + 10 + 10 uM&quot;)), 
       aes(x = reorder(target_nospace, true_counts), 
           y = true_counts, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Compensated counts&quot;, title = &quot;Compensated counts&quot;) +
  theme_bw() +
  facet_wrap(~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

# Normalised
fig_Ab_norm_HBL1 &lt;- ggplot(unique(subset(counts_dtbl, inhib_conc == &quot;10 + 10 + 10 uM&quot;)), 
       aes(x = reorder(target_nospace, true_counts_norm_geo), 
           y = true_counts_norm_geo, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = &quot;black&quot;, size = 6)) +
  scale_y_log10(
    breaks = trans_breaks(&quot;log10&quot;, function(x) 10^x),
    labels = trans_format(&quot;log10&quot;, math_format(10^.x))) +
  annotation_logticks(sides = &quot;l&quot;) +
  labs(x = &quot;Protein&quot;, y = &quot;Normalised compensated counts&quot;, title = &quot;Normalised compensated counts&quot;) +
  theme_bw() +
  facet_wrap(~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

plot_grid(fig_Ab_true_HBL1, fig_Ab_norm_HBL1, ncol = 1)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_Ab_counts_HBL1-2.png" width="1920" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_Ab_counts_HBL1-2">
Past versions of fig_Ab_counts_HBL1-2.png
</button>
</p>
<div id="fig-fig_Ab_counts_HBL1-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Distribution of counts per Ab-barcode
for the HBL1 0 uM iSYK (DMSO) samples. The normalisation removes the
variation between replicates, but preserves the variation between
proteins and highlights the differences between aIg+H2O2 and PBS
stimulated samples.</em></p>
</div>
</div>
<div id="pca" class="section level2">
<h2>PCA</h2>
<p>Principal component analysis on all samples (package: FactoMineR)</p>
<div id="data" class="section level3">
<h3>Data</h3>
<pre class="r"><code># Cleaned data
# data_well_all

# Prepare data for FactoMineR package
# cts: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- data_well_all  %&gt;%
  dplyr::filter(counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0)

cts &lt;- column_to_rownames(cts, &quot;plate_well&quot;)

metadata_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;stimulus&quot;, &quot;inhibitor&quot;, &quot;inhib_conc&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)
coldata &lt;- data.frame(plate_well = rownames(cts)) %&gt;%
  left_join(distinct(dplyr::select(data_well_all, all_of(metadata_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113), 
         inhib_group = factor(inhib_group, levels = inhib_group_DS113)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well</code></pre>
<pre class="r"><code># Results
pca_results &lt;- PCA(cts, scale.unit = FALSE, ncp = 5, graph = FALSE)

# Percentage variance explained by each PC
pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;%
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

# Results focused on variables (proteins)
pca_variab &lt;- get_pca_var(pca_results) 
# head(variab$contrib)

# Results focused on individuals (wells)
pca_indiv &lt;- get_pca_ind(pca_results) 
# head(indiv$contrib)
pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
</div>
<div id="visualisation" class="section level3">
<h3>Visualisation</h3>
<div id="basic" class="section level4">
<h4>Basic</h4>
<pre class="r"><code># Variance explained by each PC (eigenvalues)
fviz_eig(pca_results, addlabels = TRUE)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/pca_basic-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_basic-1">
Past versions of pca_basic-1.png
</button>
</p>
<div id="fig-pca_basic-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># fviz_pca_var(results_HBL1, col.var = &quot;black&quot;)

# Quality of representation for variables
# corrplot(variab_HBL1$cos2, is.corr = FALSE)
# corrplot(variab_HBL1$contrib, is.corr = FALSE)

contrib_2.5p &lt;- as.data.frame(pca_variab$contrib) %&gt;%
  dplyr::filter(Dim.1 &gt;= 2.5 | Dim.2 &gt;= 2.5 | Dim.3 &gt;= 2.5) %&gt;% # | Dim.4 &gt;= 1 | Dim.5 &gt;= 1
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  as.matrix()
corrplot(contrib_2.5p, is.corr = FALSE, method = &quot;circle&quot;, outline = TRUE, 
         tl.col = &quot;black&quot;, cl.ratio = 0.75, cl.offset = 1, 
         title = &quot; &quot;)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/pca_basic-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_basic-2">
Past versions of pca_basic-2.png
</button>
</p>
<div id="fig-pca_basic-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Top10 contributing proteins to PC1-3 
fig_PC1 &lt;- fviz_contrib(pca_results, choice = &quot;var&quot;, axes = 1, top = 10)
fig_PC2 &lt;- fviz_contrib(pca_results, choice = &quot;var&quot;, axes = 2, top = 10)
fig_PC3 &lt;- fviz_contrib(pca_results, choice = &quot;var&quot;, axes = 3, top = 10)
plot_grid(fig_PC1, fig_PC2, fig_PC3, ncol = 3)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/pca_basic-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-pca_basic-3">
Past versions of pca_basic-3.png
</button>
</p>
<div id="fig-pca_basic-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># fviz_pca_ind(pca_results,
#              axes = c(1, 2),
#              geom.ind = &quot;point&quot;,
#              mean.point = FALSE,
#              col.ind = paste(coldata$inhib_conc),
#              addEllipses = TRUE)
# fviz_pca_ind(pca_results,
#              axes = c(1, 3),
#              geom.ind = &quot;point&quot;,
#              mean.point = FALSE,
#              col.ind = paste(coldata$inhib_text_new),
#              addEllipses = TRUE)
# fviz_pca_ind(pca_results,
#              axes = c(2, 3),
#              geom.ind = &quot;point&quot;,
#              mean.point = FALSE,
#              # col.ind = paste(coldata$cell_line, coldata$stimulus),
#              col.ind = paste(coldata$inhib_text_new),
#              addEllipses = TRUE)</code></pre>
</div>
<div id="d" class="section level4">
<h4>2D</h4>
<pre class="r"><code># pca_data
fig_PC12 &lt;- ggplot(pca_data) + 
  geom_point(aes(x = PC1, y = PC2, shape = stimulus, color = inhib_conc), size = 3) +
  # geom_point(aes(x = PC1, y = PC2, shape = stimulus, color = inhib_text_new), size = 3, alpha = 0.8) + 
  facet_wrap(~inhib_group, ncol = 1) +
  scale_color_manual(values = colors_conc_uniq, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) + 
  ggtitle(&quot;Normalised counts\nPC1 vs PC2&quot;) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  textsize_small
# fig_PC12
fig_PC13 &lt;- ggplot(pca_data) + 
  geom_point(aes(x = PC1, y = PC3, shape = stimulus, color = inhib_conc), size = 3) +
  facet_wrap(~inhib_group, ncol = 1) +
  # scale_color_manual(values = c(colors_orange9[3:7], colors_grey9[3:7]), name = &quot;Stim - Inhib conc&quot;) +
  scale_color_manual(values = colors_conc_uniq, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  ggtitle(&quot;\nPC1 vs PC3&quot;) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  textsize_small
# fig_PC13
fig_PC23 &lt;- ggplot(pca_data) + 
  geom_point(aes(x = PC2, y = PC3, shape = stimulus, color = inhib_conc), size = 3) +
  facet_wrap(~inhib_group, ncol = 1) +
  scale_color_manual(values = colors_conc_uniq, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  xlab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  ggtitle(&quot;\nPC2 vs PC3&quot;) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;) +
  textsize_small
# fig_PC23

fig &lt;- plot_grid(fig_PC12, fig_PC13, fig_PC23, ncol = 3, align = &quot;h&quot;, rel_widths = c(1, 1, 1.5))
print(fig)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/Facto_2D_HBL1-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Facto_2D_HBL1-1">
Past versions of Facto_2D_HBL1-1.png
</button>
</p>
<div id="fig-Facto_2D_HBL1-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="d-1" class="section level4">
<h4>3D</h4>
<pre class="r"><code># 3D scatter plot of PC1-3
# stimulus
fig &lt;- plot_ly(pca_data, x = ~PC1, y = ~PC2, z = ~PC3, color = ~stimulus, colors = colors_stim) %&gt;%
  add_markers(size = 12) %&gt;%
  layout(title = paste(&quot;Variance explained:&quot;, sum(pca_expl_var[1:3])))
fig</code></pre>
<div class="plotly html-widget html-fill-item" id="htmlwidget-3b205314ad47e0460929" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-3b205314ad47e0460929">{"x":{"visdat":{"23984a3471ed":["function () ","plotlyVisDat"]},"cur_data":"23984a3471ed","attrs":{"23984a3471ed":{"x":{},"y":{},"z":{},"color":{},"colors":["#525252","#fdae6b"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d","mode":"markers","size":12,"inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"title":"Variance explained: 90.2","scene":{"xaxis":{"title":"PC1"},"yaxis":{"title":"PC2"},"zaxis":{"title":"PC3"}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[-1515.7983039303695,-1748.8019028919869,-1392.5791455809519,-1774.8444653152555,-1163.5136720637161,-1665.7890718475719],"y":[-2863.1740838511487,-766.9862012826178,-2367.021600311823,-1112.8009316292478,-3208.6690806265378,-1463.2896918948229],"z":[-307.5782657343637,376.90011758283708,382.0449725904607,334.1087247961018,298.99529997634943,428.28673479703735],"type":"scatter3d","mode":"markers","name":"PBS","marker":{"color":"rgba(82,82,82,1)","size":[55,55,55,55,55,55],"sizemode":"area","line":{"color":"rgba(82,82,82,1)"}},"textfont":{"color":"rgba(82,82,82,1)","size":55},"error_y":{"color":"rgba(82,82,82,1)","width":55},"error_x":{"color":"rgba(82,82,82,1)","width":55},"line":{"color":"rgba(82,82,82,1)","width":55},"frame":null},{"x":[5174.2601125188303,315.75089380052793,6156.1077300275592,464.3728875922788,5432.2619140649331,330.29627862299475,-552.23772407131798,-830.46927528498372,-455.0745237515311,-834.00415720219542,-585.44918820093221,-968.61841938126202,-1051.1712769994958,761.75524091745399,-1566.8305639613677,356.56349521628948,-1100.1953645113285,275.53613062972897,-1832.4399700604104,-1524.6637158733724,-2034.0425975975677,-1566.3029386439298,-1808.7903088480684,-1556.4875192298784,792.20273637621199,-693.25776768626577,979.42560750604264,-652.13291154980573,779.67513187786244,-612.2926992100879,2576.8488399399334,-861.2393238422402,2785.6568811905095,-845.25190106771174,1907.8432345951746,-1101.6344739621766,1182.0380443890854,1147.8360145758115,875.48200872454504],"y":[-653.36921046293935,321.17757510784821,98.318222854255197,-499.19920494103565,-440.81781555428171,-845.16282817610602,357.0099428354838,767.84075100056793,184.81989400064421,645.00307816445502,-78.291128837122727,-218.50748736868783,-552.80032609961108,1641.888550899318,976.08774643616846,-572.59883344605475,648.73315527315617,-429.42026963330443,1331.930354224012,927.35681636201787,1565.0351969536712,1246.8176933783309,928.51956282453898,736.7087412312676,833.83369352497402,274.38481575882622,1321.9619862428171,718.26169892571352,261.99857587694521,232.25322607035815,495.22543142101523,631.71692093001627,745.39283800811836,99.342523886308257,-800.42232502739455,-227.16231259234436,402.49153890267587,210.61980395140139,-1505.0370033098629],"z":[911.70087948984963,-340.31855481979898,831.82855274682743,-424.67581580282939,751.43616817940654,-479.01163400843899,-113.46857453832516,303.05901437518423,-78.799211037082586,364.55558369930009,-97.427515812700619,293.60221914071326,16.683942627507701,-712.46301442947288,513.56224870731171,-716.53165264438462,287.00619526184425,-473.48882113905387,146.71346892511582,377.6644660272475,152.94399340327021,307.85738537465647,303.41581448419993,262.3334640358575,64.025957000247658,-27.739785860678015,115.9524270178083,-104.01335793352901,72.120197346918289,6.0468216416934082,-646.77749093875218,5.0776681148862206,-810.00963489595574,-227.699951131072,-1013.9866116787367,-479.86079992646012,-234.54543942369486,-276.12095111971576,-343.40523446760312],"type":"scatter3d","mode":"markers","name":"aIg+H2O2","marker":{"color":"rgba(253,174,107,1)","size":[55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55,55],"sizemode":"area","line":{"color":"rgba(253,174,107,1)"}},"textfont":{"color":"rgba(253,174,107,1)","size":55},"error_y":{"color":"rgba(253,174,107,1)","width":55},"error_x":{"color":"rgba(253,174,107,1)","width":55},"line":{"color":"rgba(253,174,107,1)","width":55},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code># inhib_conc
fig &lt;- plot_ly(pca_data, x = ~PC1, y = ~PC2, z = ~PC3, color = ~inhib_conc, colors = colors_conc_uniq) %&gt;%
  add_markers(size = 12) %&gt;%
  layout(title = paste(&quot;Variance explained:&quot;, sum(pca_expl_var[1:3])))
fig</code></pre>
<div class="plotly html-widget html-fill-item" id="htmlwidget-156f262d82fdef13d33e" style="width:672px;height:480px;"></div>
<script type="application/json" data-for="htmlwidget-156f262d82fdef13d33e">{"x":{"visdat":{"23985d45476a":["function () ","plotlyVisDat"]},"cur_data":"23985d45476a","attrs":{"23985d45476a":{"x":{},"y":{},"z":{},"color":{},"colors":["#bcbddc","#9e9ac8","#6a51a3","#3f007d","#006d2c","#41ab5d","#a1d99b","#a63603","#f16913","#fdae6b","#08519c","#4292c6","#9ecae1","#969696"],"alpha_stroke":1,"sizes":[10,100],"spans":[1,20],"type":"scatter3d","mode":"markers","size":12,"inherit":true}},"layout":{"margin":{"b":40,"l":60,"t":25,"r":10},"title":"Variance explained: 90.2","scene":{"xaxis":{"title":"PC1"},"yaxis":{"title":"PC2"},"zaxis":{"title":"PC3"}},"hovermode":"closest","showlegend":true},"source":"A","config":{"modeBarButtonsToAdd":["hoverclosest","hovercompare"],"showSendToCloud":false},"data":[{"x":[5174.2601125188303,6156.1077300275592,5432.2619140649331,-1515.7983039303695,-1392.5791455809519,-1163.5136720637161],"y":[-653.36921046293935,98.318222854255197,-440.81781555428171,-2863.1740838511487,-2367.021600311823,-3208.6690806265378],"z":[911.70087948984963,831.82855274682743,751.43616817940654,-307.5782657343637,382.0449725904607,298.99529997634943],"type":"scatter3d","mode":"markers","name":"0 + 0 + 0 uM","marker":{"color":"rgba(188,189,220,1)","size":[55,55,55,55,55,55],"sizemode":"area","line":{"color":"rgba(188,189,220,1)"}},"textfont":{"color":"rgba(188,189,220,1)","size":55},"error_y":{"color":"rgba(188,189,220,1)","width":55},"error_x":{"color":"rgba(188,189,220,1)","width":55},"line":{"color":"rgba(188,189,220,1)","width":55},"frame":null},{"x":[-552.23772407131798,-455.0745237515311,-585.44918820093221],"y":[357.0099428354838,184.81989400064421,-78.291128837122727],"z":[-113.46857453832516,-78.799211037082586,-97.427515812700619],"type":"scatter3d","mode":"markers","name":"1 + 1 + 1 uM","marker":{"color":"rgba(158,154,200,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(158,154,200,1)"}},"textfont":{"color":"rgba(158,154,200,1)","size":55},"error_y":{"color":"rgba(158,154,200,1)","width":55},"error_x":{"color":"rgba(158,154,200,1)","width":55},"line":{"color":"rgba(158,154,200,1)","width":55},"frame":null},{"x":[-1051.1712769994958,-1566.8305639613677,-1100.1953645113285],"y":[-552.80032609961108,976.08774643616846,648.73315527315617],"z":[16.683942627507701,513.56224870731171,287.00619526184425],"type":"scatter3d","mode":"markers","name":"2.5 + 2.5 + 2.5 uM","marker":{"color":"rgba(106,81,163,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(106,81,163,1)"}},"textfont":{"color":"rgba(106,81,163,1)","size":55},"error_y":{"color":"rgba(106,81,163,1)","width":55},"error_x":{"color":"rgba(106,81,163,1)","width":55},"line":{"color":"rgba(106,81,163,1)","width":55},"frame":null},{"x":[-1832.4399700604104,-2034.0425975975677,-1808.7903088480684,-1748.8019028919869,-1774.8444653152555,-1665.7890718475719],"y":[1331.930354224012,1565.0351969536712,928.51956282453898,-766.9862012826178,-1112.8009316292478,-1463.2896918948229],"z":[146.71346892511582,152.94399340327021,303.41581448419993,376.90011758283708,334.1087247961018,428.28673479703735],"type":"scatter3d","mode":"markers","name":"10 + 10 + 10 uM","marker":{"color":"rgba(63,0,125,1)","size":[55,55,55,55,55,55],"sizemode":"area","line":{"color":"rgba(63,0,125,1)"}},"textfont":{"color":"rgba(63,0,125,1)","size":55},"error_y":{"color":"rgba(63,0,125,1)","width":55},"error_x":{"color":"rgba(63,0,125,1)","width":55},"line":{"color":"rgba(63,0,125,1)","width":55},"frame":null},{"x":[792.20273637621199,979.42560750604264,779.67513187786244],"y":[833.83369352497402,1321.9619862428171,261.99857587694521],"z":[64.025957000247658,115.9524270178083,72.120197346918289],"type":"scatter3d","mode":"markers","name":"10 + 0 + 0 uM","marker":{"color":"rgba(0,109,44,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(0,109,44,1)"}},"textfont":{"color":"rgba(0,109,44,1)","size":55},"error_y":{"color":"rgba(0,109,44,1)","width":55},"error_x":{"color":"rgba(0,109,44,1)","width":55},"line":{"color":"rgba(0,109,44,1)","width":55},"frame":null},{"x":[2576.8488399399334,2785.6568811905095,1907.8432345951746],"y":[495.22543142101523,745.39283800811836,-800.42232502739455],"z":[-646.77749093875218,-810.00963489595574,-1013.9866116787367],"type":"scatter3d","mode":"markers","name":"0 + 10 + 0 uM","marker":{"color":"rgba(65,171,93,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(65,171,93,1)"}},"textfont":{"color":"rgba(65,171,93,1)","size":55},"error_y":{"color":"rgba(65,171,93,1)","width":55},"error_x":{"color":"rgba(65,171,93,1)","width":55},"line":{"color":"rgba(65,171,93,1)","width":55},"frame":null},{"x":[1182.0380443890854,1147.8360145758115,875.48200872454504],"y":[402.49153890267587,210.61980395140139,-1505.0370033098629],"z":[-234.54543942369486,-276.12095111971576,-343.40523446760312],"type":"scatter3d","mode":"markers","name":"0 + 0 + 10 uM","marker":{"color":"rgba(161,217,155,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(161,217,155,1)"}},"textfont":{"color":"rgba(161,217,155,1)","size":55},"error_y":{"color":"rgba(161,217,155,1)","width":55},"error_x":{"color":"rgba(161,217,155,1)","width":55},"line":{"color":"rgba(161,217,155,1)","width":55},"frame":null},{"x":[315.75089380052793,464.3728875922788,330.29627862299475],"y":[321.17757510784821,-499.19920494103565,-845.16282817610602],"z":[-340.31855481979898,-424.67581580282939,-479.01163400843899],"type":"scatter3d","mode":"markers","name":"2.5 + 2.5 + 0 uM","marker":{"color":"rgba(166,54,3,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(166,54,3,1)"}},"textfont":{"color":"rgba(166,54,3,1)","size":55},"error_y":{"color":"rgba(166,54,3,1)","width":55},"error_x":{"color":"rgba(166,54,3,1)","width":55},"line":{"color":"rgba(166,54,3,1)","width":55},"frame":null},{"x":[-830.46927528498372,-834.00415720219542,-968.61841938126202],"y":[767.84075100056793,645.00307816445502,-218.50748736868783],"z":[303.05901437518423,364.55558369930009,293.60221914071326],"type":"scatter3d","mode":"markers","name":"2.5 + 0 + 2.5 uM","marker":{"color":"rgba(241,105,19,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(241,105,19,1)"}},"textfont":{"color":"rgba(241,105,19,1)","size":55},"error_y":{"color":"rgba(241,105,19,1)","width":55},"error_x":{"color":"rgba(241,105,19,1)","width":55},"line":{"color":"rgba(241,105,19,1)","width":55},"frame":null},{"x":[761.75524091745399,356.56349521628948,275.53613062972897],"y":[1641.888550899318,-572.59883344605475,-429.42026963330443],"z":[-712.46301442947288,-716.53165264438462,-473.48882113905387],"type":"scatter3d","mode":"markers","name":"0 + 2.5 + 2.5 uM","marker":{"color":"rgba(253,174,107,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(253,174,107,1)"}},"textfont":{"color":"rgba(253,174,107,1)","size":55},"error_y":{"color":"rgba(253,174,107,1)","width":55},"error_x":{"color":"rgba(253,174,107,1)","width":55},"line":{"color":"rgba(253,174,107,1)","width":55},"frame":null},{"x":[-1524.6637158733724,-1566.3029386439298,-1556.4875192298784],"y":[927.35681636201787,1246.8176933783309,736.7087412312676],"z":[377.6644660272475,307.85738537465647,262.3334640358575],"type":"scatter3d","mode":"markers","name":"5 + 1 + 1 uM","marker":{"color":"rgba(8,81,156,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(8,81,156,1)"}},"textfont":{"color":"rgba(8,81,156,1)","size":55},"error_y":{"color":"rgba(8,81,156,1)","width":55},"error_x":{"color":"rgba(8,81,156,1)","width":55},"line":{"color":"rgba(8,81,156,1)","width":55},"frame":null},{"x":[-693.25776768626577,-652.13291154980573,-612.2926992100879],"y":[274.38481575882622,718.26169892571352,232.25322607035815],"z":[-27.739785860678015,-104.01335793352901,6.0468216416934082],"type":"scatter3d","mode":"markers","name":"1 + 5 + 1 uM","marker":{"color":"rgba(66,146,198,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(66,146,198,1)"}},"textfont":{"color":"rgba(66,146,198,1)","size":55},"error_y":{"color":"rgba(66,146,198,1)","width":55},"error_x":{"color":"rgba(66,146,198,1)","width":55},"line":{"color":"rgba(66,146,198,1)","width":55},"frame":null},{"x":[-861.2393238422402,-845.25190106771174,-1101.6344739621766],"y":[631.71692093001627,99.342523886308257,-227.16231259234436],"z":[5.0776681148862206,-227.699951131072,-479.86079992646012],"type":"scatter3d","mode":"markers","name":"1 + 1 + 5 uM","marker":{"color":"rgba(158,202,225,1)","size":[55,55,55],"sizemode":"area","line":{"color":"rgba(158,202,225,1)"}},"textfont":{"color":"rgba(158,202,225,1)","size":55},"error_y":{"color":"rgba(158,202,225,1)","width":55},"error_x":{"color":"rgba(158,202,225,1)","width":55},"line":{"color":"rgba(158,202,225,1)","width":55},"frame":null}],"highlight":{"on":"plotly_click","persistent":false,"dynamic":false,"selectize":false,"opacityDim":0.20000000000000001,"selected":{"opacity":1},"debounce":0},"shinyEvents":["plotly_hover","plotly_click","plotly_selected","plotly_relayout","plotly_brushed","plotly_brushing","plotly_clickannotation","plotly_doubleclick","plotly_deselect","plotly_afterplot","plotly_sunburstclick"],"base_url":"https://plot.ly"},"evals":[],"jsHooks":[]}</script>
</div>
<div id="umap" class="section level4">
<h4>UMAP</h4>
<pre class="r"><code># Data
pca_data_5PC &lt;- as.data.frame(pca_indiv$coord[, 1:5]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC53 = Dim.5) %&gt;% 
  left_join(coldata)

# UMAP data reduction
umap_pca &lt;- umap::umap(as.data.frame(pca_indiv$coord[, 1:5]))
# umap.defaults
# head(umap_pca$layout)
umap_pca_ann &lt;- as.data.frame(umap_pca$layout) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(UMAP1 = V1, UMAP2 = V2) %&gt;% 
  left_join(coldata)

# Basic fig of everything
fig &lt;- ggplot(umap_pca_ann, aes(x = UMAP1, y = UMAP2)) + 
  geom_point(aes(shape = stimulus, color = inhib_conc), size = 2) + 
  scale_color_manual(values = colors_conc_uniq, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  ggtitle(&quot;UMAP with PC1-5&quot;) +
  theme_bw() +
  textsize_small
fig</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/Facto_UMAP_PCA-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Facto_UMAP_PCA-1">
Past versions of Facto_UMAP_PCA-1.png
</button>
</p>
<div id="fig-Facto_UMAP_PCA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fig &lt;- ggplot(umap_pca_ann, aes(x = UMAP1, y = UMAP2)) + 
  geom_point(aes(shape = stimulus, color = inhib_conc), size = 2) + 
  facet_wrap(~inhib_group, ncol = 1) +
  scale_color_manual(values = colors_conc_uniq, name = &quot;iSYK + iBTK + iNFkB&quot;) +
  ggtitle(&quot;UMAP with PC1-5&quot;) +
  theme_bw() +
  textsize_small
fig</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/Facto_UMAP_PCA-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-Facto_UMAP_PCA-2">
Past versions of Facto_UMAP_PCA-2.png
</button>
</p>
<div id="fig-Facto_UMAP_PCA-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;PC&quot;))
gc()</code></pre>
<pre><code>           used (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9549472  510   14474182 773.1 14474182 773.1
Vcells 17422374  133   31968708 244.0 25106744 191.6</code></pre>
</div>
</div>
</div>
<div id="basal-vs-activated" class="section level2">
<h2>Basal vs activated</h2>
<p>Now, we will look at the basal vs activated signaling levels with and
without inhibitors.</p>
<div id="deseq2" class="section level3 tabset">
<h3 class="tabset">DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="separate-comp" class="section level4">
<h4>Separate comp</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>PBS and aIg+H2O2 stimulus, 0 + 0 + 0 uM and 10 + 10 + 10 uM (12
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_stim (ref: 0 + 0 + 0 uM - PBS)</p>
<p><em>Parameter options: stimulus + inhib_conc + inhib_conc_text +
description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
counts_dtbl &lt;- counts_dtbl %&gt;%
  mutate(inhib_stim = paste(inhib_conc, &quot;-&quot;, stimulus))

cts &lt;- counts_dtbl  %&gt;%
  dplyr::filter(inhib_conc %in% c(&quot;0 + 0 + 0 uM&quot;, &quot;10 + 10 + 10 uM&quot;) &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  dplyr::filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

metadata_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;stimulus&quot;, &quot;inhibitor&quot;, &quot;inhib_conc&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;, &quot;inhib_stim&quot;)
coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, all_of(metadata_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_group = factor(inhib_group, levels = inhib_group_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113), 
         inhib_stim = paste(inhib_conc, &quot;-&quot;, stimulus)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_stim

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_stim &lt;- relevel(dds$inhib_stim, ref = &quot;0 + 0 + 0 uM - PBS&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code>[1] &quot;Intercept&quot;                                                  
[2] &quot;inhib_stim_0...0...0.uM...aIg.H2O2_vs_0...0...0.uM...PBS&quot;   
[3] &quot;inhib_stim_10...10...10.uM...aIg.H2O2_vs_0...0...0.uM...PBS&quot;
[4] &quot;inhib_stim_10...10...10.uM...PBS_vs_0...0...0.uM...PBS&quot;     </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_active_wo &lt;- results(dds, name = &quot;inhib_stim_0...0...0.uM...aIg.H2O2_vs_0...0...0.uM...PBS&quot;, alpha = 0.05)
results_active_w &lt;- results(dds, contrast = c(&quot;inhib_stim&quot;, &quot;10 + 10 + 10 uM - aIg+H2O2&quot;, &quot;10 + 10 + 10 uM - PBS&quot;), alpha = 0.05)

results_PBS &lt;- results(dds, name = &quot;inhib_stim_10...10...10.uM...PBS_vs_0...0...0.uM...PBS&quot;, alpha = 0.05)
results_stim &lt;- results(dds, contrast = c(&quot;inhib_stim&quot;, &quot;10 + 10 + 10 uM - aIg+H2O2&quot;, &quot;0 + 0 + 0 uM - aIg+H2O2&quot;), alpha = 0.05)

list_results &lt;- list(results_active_wo, results_active_w, results_PBS, results_stim)

# Results summary
summary(results_active_wo, padj = 0.05)</code></pre>
<pre><code>
out of 111 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 24, 22%
LFC &lt; 0 (down)     : 14, 13%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 518)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_active_wo, alpha = signcutoff, main = &quot;aIg+H2O2 vs PBS\nw/o inhib&quot;)
plotMA(results_active_w, alpha = signcutoff, main = &quot;aIg+H2O2 vs PBS\nw/ inhib&quot;)
plotMA(results_stim, alpha = signcutoff, main = &quot;0 vs 10 uM equal\nin activated cells&quot;)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_log2FC_act_sep-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_act_sep-1">
Past versions of fig_log2FC_act_sep-1.png
</button>
</p>
<div id="fig-fig_log2FC_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-1" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;inhib_group&quot;, &quot;inhib_conc&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS113), shape = factor(inhib_conc, levels = inhib_conc_DS113))) + #, shape = factor(cell_line, levels = cells_DS113)
    geom_point(size = 3) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    facet_wrap(~inhib_group) +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
    scale_shape_manual(values = c(16, 17), name = &quot;iSYK + iBTK + iNFkB&quot;) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_PCA_act_sep-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_sep-1">
Past versions of fig_PCA_act_sep-1.png
</button>
</p>
<div id="fig-fig_PCA_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_PCA_act_sep-2.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_sep-2">
Past versions of fig_PCA_act_sep-2.png
</button>
</p>
<div id="fig-fig_PCA_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_PCA_act_sep-3.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_act_sep-3">
Past versions of fig_PCA_act_sep-3.png
</button>
</p>
<div id="fig-fig_PCA_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis. PC1 shows a clear separation of the activated samples
without inhibitors and the other conditions, and PC2 separates the
samples with and without inhibitors. The different types of data
transformation also give similar results.</em></p>
</div>
<div id="variance" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_act_sep-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-1">
Past versions of fig_sd_act_sep-1.png
</button>
</p>
<div id="fig-fig_sd_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Norm (log2(n+1))&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_act_sep-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-2">
Past versions of fig_sd_act_sep-2.png
</button>
</p>
<div id="fig-fig_sd_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_act_sep-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-3">
Past versions of fig_sd_act_sep-3.png
</button>
</p>
<div id="fig-fig_sd_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_act_sep-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_act_sep-4">
Past versions of fig_sd_act_sep-4.png
</button>
</p>
<div id="fig-fig_sd_act_sep-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$cell_line,
  #                                     list_datatransform[[data]]$stimulus,
  #                                     # list_datatransform[[data]]$inhibitor,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = &quot; - &quot;)
  rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$descript_rep
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_distances_act_sep-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_sep-1">
Past versions of fig_distances_act_sep-1.png
</button>
</p>
<div id="fig-fig_distances_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_distances_act_sep-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_sep-2">
Past versions of fig_distances_act_sep-2.png
</button>
</p>
<div id="fig-fig_distances_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_distances_act_sep-3.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_act_sep-3">
Past versions of fig_distances_act_sep-3.png
</button>
</p>
<div id="fig-fig_distances_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins:</p>
<ul>
<li><p>padj &lt; 0.05</p></li>
<li><p>log2FC &gt; log2(1.1)</p></li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_active_wo&quot;, &quot;volc_active_w&quot;, &quot;volc_PBS&quot;, &quot;volc_stim&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_active_wo, volc_active_w, volc_PBS, volc_stim)
titles_volcano &lt;- c(&quot;active vs basal (0 + 0 + 0 uM inhib)&quot;,
                    &quot;active vs basal (10 + 10 + 10 uM inhib)&quot;,
                    &quot;10 + 10 + 10 uM vs 0 + 0 + 0 uM inhib (basal)&quot;,
                    &quot;10 + 10 + 10 uM vs 0 + 0 + 0 uM inhib (active)&quot;)
# titles_volcano &lt;- names_results_volc

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(-log2(1.1), log2(1.1)))
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_act_sep-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-1">
Past versions of fig_volcano_act_sep-1.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_act_sep-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-2">
Past versions of fig_volcano_act_sep-2.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_act_sep-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-3">
Past versions of fig_volcano_act_sep-3.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_act_sep-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_act_sep-4">
Past versions of fig_volcano_act_sep-4.png
</button>
</p>
<div id="fig-fig_volcano_act_sep-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons.</em></p>
<p><em>Activation results in the increase of many phospho-proteins. In
the presence of 10 + 10 + 10 uM inhibitors, activation is drastically
reduced (only a few total protein levels). In the basal state, only a
few proteins are significantly decreased by high inhibitor
concentrations, while many phospho-proteins are decreased by inhibitors
in the activated state.</em></p>
</div>
<div id="heatmap" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li><p>padj &lt; 0.05</p></li>
<li><p>log2FC &gt; log2(1.1)</p></li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_active_wo&quot;, &quot;heat_active_w&quot;, &quot;heat_PBS&quot;, &quot;heat_stim&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;stimulus&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(stimulus, levels = stim_DS113), factor(inhib_conc, levels = inhib_conc_DS113))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# active vs basal (0 + 0 + 0 uM inhib)
heat_active_wo$marker_pathway[is.na(heat_active_wo$marker_pathway)] &lt;- &quot;Other&quot;

ann_colors_man &lt;- list(stimulus = colors_stim,
                       inhib_conc = colors_conc
                       # modification = c(phospho = &quot;#f768a1&quot;, no = &quot;grey&quot;, cleaved = &quot;#fff7bc&quot;),
                       # marker_pathway = setNames(colors_light12[1:length(unique(heat_active_wo$marker_pathway))], unique(heat_active_wo$marker_pathway))
)

# All significant proteins in all comparisons
sign_prot_sep &lt;- sort(unique(c(heat_active_wo$target_nospace, 
                               heat_active_w$target_nospace, 
                               heat_PBS$target_nospace, 
                               heat_stim$target_nospace)))

# Get the rld transformed data for the heatmap and scale per row
# data_active_wo &lt;- t(scale(t(assay(rld)[unique(c(heat_active_wo$target_nospace)), rownames(data_heatmap)])))
data_act_sep &lt;- t(scale(t(assay(rld)[sign_prot_sep, rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_act_sep,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_HBL1[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_conc,
  # column_split = data_heatmap$stimulus,
  # row_split = subset(heat_active_wo)$modification,
  cellwidth = 15,
  cellheight = 15,
  fontsize = 12,
  # border_color = NA,
  # main = &quot;scaled per row, ComplexHeatmap&quot;
)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_heatmap_act_sep-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_act_sep-1">
Past versions of fig_heatmap_act_sep-1.png
</button>
</p>
<div id="fig-fig_heatmap_act_sep-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
all comparisons.</em></p>
<p><em>There are many phospho-proteins that are activated upon
stimulation, and are inhibited by 10 + 10 + 10 uM inhibitors. pCREB and
pS6 are inhibited in both basal and activated conditions. Various
non-phospho-proteins are increased in both basal and activated
conditions (more so in the activated conditions). And lastly, some
proteins are decreased upon stimulation, and less decreased upon
inhibitor treatment.</em></p>
</div>
</div>
<div id="interaction" class="section level4">
<h4>Interaction</h4>
<p>Add later.</p>
</div>
</div>
<div id="protein-comparisons" class="section level3">
<h3>Protein comparisons</h3>
<p>Finally, we visualised the normalised true counts of all significant
proteins in the comparisons (40 in total).</p>
<pre class="r"><code># Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins 
sign_prot_sep</code></pre>
<pre><code> [1] &quot;AKT_T308&quot;           &quot;AMPKb1_S182&quot;        &quot;ATF2_T71&quot;          
 [4] &quot;BAD_S112&quot;           &quot;BIM&quot;                &quot;BLNK_Y84&quot;          
 [7] &quot;BTK&quot;                &quot;BTK_Y223&quot;           &quot;BTK_Y551&quot;          
[10] &quot;CD20&quot;               &quot;CD43&quot;               &quot;CD79a&quot;             
[13] &quot;CD79a_Y182&quot;         &quot;CDK1_T14_Y15&quot;       &quot;cFOS_S32&quot;          
[16] &quot;cMYC_T58&quot;           &quot;CREB_S133&quot;          &quot;Cyclin_D1&quot;         
[19] &quot;GAPDH&quot;              &quot;GSK3b_S9&quot;           &quot;IgM&quot;               
[22] &quot;IKKa_b_S176_S180&quot;   &quot;JAK1_Y1022_Y1023&quot;   &quot;Ki67&quot;              
[25] &quot;LYN_Y397&quot;           &quot;MAPK_p38_T180_Y182&quot; &quot;MKK4_S257_T261&quot;    
[28] &quot;PARP_D214&quot;          &quot;PKCb1&quot;              &quot;PLCy2&quot;             
[31] &quot;PYK2_Y402&quot;          &quot;S6_S235_S236&quot;       &quot;S6_S240_S244&quot;      
[34] &quot;SHP2&quot;               &quot;SHP2_Y580&quot;          &quot;SRC_Y416&quot;          
[37] &quot;STAT6&quot;              &quot;SYK_Y525_Y526&quot;      &quot;VAV1_Y174&quot;         
[40] &quot;ZAP70_Y319&quot;        </code></pre>
<pre class="r"><code># Datasets generated during analysis
# heat_active_wo + heat_active_w + heat_PBS + heat_stim: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_act_sep: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
clean_data_well_sep &lt;- data.frame(data_act_sep) %&gt;% 
  mutate(target_nospace = rownames(data_act_sep)) %&gt;%
  dplyr::select(target_nospace, everything()) %&gt;%
  gather(&quot;plate_well&quot;, &quot;heatmap_value&quot;, 2:c(ncol(data_act_sep)+1)) %&gt;%
  left_join(counts_dtbl) %&gt;%
  arrange(plate_well, target_nospace) %&gt;%
  dplyr::filter(target_nospace %in% sign_prot_sep) %&gt;%
  dplyr::select(c(meta_cols_well, heatmap_value)) %&gt;% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

prot_labels_sep &lt;- sort(setNames(unique(pull(clean_data_well_sep, target)), unique(pull(clean_data_well_sep, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
heat_list_act_sep &lt;- list(heat_active_wo, heat_active_w, heat_PBS, heat_stim)
heat_names &lt;- c(&quot;sign_data_active_wo&quot;, &quot;sign_data_active_w&quot;, &quot;sign_data_PBS&quot;, &quot;sign_data_stim&quot;)
heat_suffix &lt;- c(&quot;_actWO&quot;, &quot;_actW&quot;, &quot;_PBS&quot;, &quot;_stim&quot;)
for (heat_data in c(1:length(heat_list_act_sep))) {
  # print(heat_data)
  alt_data &lt;- heat_list_act_sep[[heat_data]] %&gt;%
    dplyr::filter(target_nospace %in% sign_prot_sep) %&gt;%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
                  baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %&gt;% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %&gt;%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c(&quot;log2FC&quot;, &quot;log2FC_se&quot;, &quot;stat&quot;, &quot;pvalue&quot;, &quot;padj&quot;, &quot;diff_express&quot;))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_sep &lt;- list(sign_data_active_wo, sign_data_active_w, sign_data_PBS, sign_data_stim) %&gt;%
  purrr::reduce(full_join) %&gt;%
  arrange(target_nospace)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean_sep &lt;- mean_dtbl %&gt;%
  dplyr::filter(target_nospace %in% sign_prot_sep &amp; inhib_conc %in% c(&quot;0 + 0 + 0 uM&quot;, &quot;10 + 10 + 10 uM&quot;)) %&gt;%
  dplyr::select(all_of(meta_cols_mean)) %&gt;% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name)</code></pre>
<pre class="r"><code>plot &lt;- ggplot(clean_data_well_sep, 
               aes(factor(inhib_conc, levels = inhib_conc_DS113),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhib_conc)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  scale_y_log10() +
  scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts (log10 scale)&quot;) +
  textsize_small

print(plot)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act_all-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act_all-1">
Past versions of fig_counts_act_all-1.png
</button>
</p>
<div id="fig-fig_counts_act_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Normalised counts for all significantly
changing proteins. For some proteins the effect size is quite
small.</em></p>
<pre class="r"><code>options(repr.plot.width = 3.5, repr.plot.height = 2.5)

# Re-redesigned fig
# filt_clean_data_mean &lt;- clean_data_mean_sep %&gt;%
#   dplyr::group_by(cell_line, stimulus, target_nospace) %&gt;%
#   summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
#             mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_sep) {
  plot &lt;- ggplot() +
    geom_jitter(data = subset(clean_data_well_sep, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS113), counts_norm, color = stimulus), 
                alpha = 1, size = 1.5, width = 0.1) +
    geom_errorbar(data = subset(clean_data_mean_sep, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS113),
                      ymin = counts_norm,
                      ymax = counts_norm,
                      group = inhib_conc, 
                      color = stimulus),
                  width = 0.5, size = 1) +
    facet_wrap(vars(factor(target))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-1.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-1">
Past versions of fig_counts_act-1.png
</button>
</p>
<div id="fig-fig_counts_act-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-2.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-2">
Past versions of fig_counts_act-2.png
</button>
</p>
<div id="fig-fig_counts_act-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-3.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-3">
Past versions of fig_counts_act-3.png
</button>
</p>
<div id="fig-fig_counts_act-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-4.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-4">
Past versions of fig_counts_act-4.png
</button>
</p>
<div id="fig-fig_counts_act-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-5.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-5">
Past versions of fig_counts_act-5.png
</button>
</p>
<div id="fig-fig_counts_act-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-6.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-6">
Past versions of fig_counts_act-6.png
</button>
</p>
<div id="fig-fig_counts_act-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-7.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-7">
Past versions of fig_counts_act-7.png
</button>
</p>
<div id="fig-fig_counts_act-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-8.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-8">
Past versions of fig_counts_act-8.png
</button>
</p>
<div id="fig-fig_counts_act-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-9.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-9">
Past versions of fig_counts_act-9.png
</button>
</p>
<div id="fig-fig_counts_act-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-10.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-10">
Past versions of fig_counts_act-10.png
</button>
</p>
<div id="fig-fig_counts_act-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-11.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-11">
Past versions of fig_counts_act-11.png
</button>
</p>
<div id="fig-fig_counts_act-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-12.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-12">
Past versions of fig_counts_act-12.png
</button>
</p>
<div id="fig-fig_counts_act-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-13.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-13">
Past versions of fig_counts_act-13.png
</button>
</p>
<div id="fig-fig_counts_act-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-14.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-14">
Past versions of fig_counts_act-14.png
</button>
</p>
<div id="fig-fig_counts_act-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-15.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-15">
Past versions of fig_counts_act-15.png
</button>
</p>
<div id="fig-fig_counts_act-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-16.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-16">
Past versions of fig_counts_act-16.png
</button>
</p>
<div id="fig-fig_counts_act-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-17.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-17">
Past versions of fig_counts_act-17.png
</button>
</p>
<div id="fig-fig_counts_act-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-18.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-18">
Past versions of fig_counts_act-18.png
</button>
</p>
<div id="fig-fig_counts_act-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-19.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-19">
Past versions of fig_counts_act-19.png
</button>
</p>
<div id="fig-fig_counts_act-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-20.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-20">
Past versions of fig_counts_act-20.png
</button>
</p>
<div id="fig-fig_counts_act-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-21.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-21">
Past versions of fig_counts_act-21.png
</button>
</p>
<div id="fig-fig_counts_act-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-22.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-22">
Past versions of fig_counts_act-22.png
</button>
</p>
<div id="fig-fig_counts_act-22" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-23.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-23">
Past versions of fig_counts_act-23.png
</button>
</p>
<div id="fig-fig_counts_act-23" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-24.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-24">
Past versions of fig_counts_act-24.png
</button>
</p>
<div id="fig-fig_counts_act-24" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-25.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-25">
Past versions of fig_counts_act-25.png
</button>
</p>
<div id="fig-fig_counts_act-25" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-26.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-26">
Past versions of fig_counts_act-26.png
</button>
</p>
<div id="fig-fig_counts_act-26" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-27.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-27">
Past versions of fig_counts_act-27.png
</button>
</p>
<div id="fig-fig_counts_act-27" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-28.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-28">
Past versions of fig_counts_act-28.png
</button>
</p>
<div id="fig-fig_counts_act-28" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-29.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-29">
Past versions of fig_counts_act-29.png
</button>
</p>
<div id="fig-fig_counts_act-29" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-30.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-30">
Past versions of fig_counts_act-30.png
</button>
</p>
<div id="fig-fig_counts_act-30" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-31.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-31">
Past versions of fig_counts_act-31.png
</button>
</p>
<div id="fig-fig_counts_act-31" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-32.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-32">
Past versions of fig_counts_act-32.png
</button>
</p>
<div id="fig-fig_counts_act-32" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-33.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-33">
Past versions of fig_counts_act-33.png
</button>
</p>
<div id="fig-fig_counts_act-33" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-34.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-34">
Past versions of fig_counts_act-34.png
</button>
</p>
<div id="fig-fig_counts_act-34" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-35.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-35">
Past versions of fig_counts_act-35.png
</button>
</p>
<div id="fig-fig_counts_act-35" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-36.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-36">
Past versions of fig_counts_act-36.png
</button>
</p>
<div id="fig-fig_counts_act-36" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-37.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-37">
Past versions of fig_counts_act-37.png
</button>
</p>
<div id="fig-fig_counts_act-37" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-38.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-38">
Past versions of fig_counts_act-38.png
</button>
</p>
<div id="fig-fig_counts_act-38" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-39.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-39">
Past versions of fig_counts_act-39.png
</button>
</p>
<div id="fig-fig_counts_act-39" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_counts_act-40.png" width="240" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_counts_act-40">
Past versions of fig_counts_act-40.png
</button>
</p>
<div id="fig-fig_counts_act-40" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-data" class="section level3">
<h3>Save data</h3>
<p>Save all information and datasets as .csv file</p>
<p>Separate analysis comparing conditions:</p>
<ul>
<li><p>sep_sign_proteins.txt: List of significantly changing
proteins</p></li>
<li><p>sep_sign_DESeq2data.csv: Dataset with significantly changing
proteins, and log2FC + pvalues from DESeq2 analysis</p></li>
<li><p>sep_sign_sample.csv: Dataset of counts per sample per protein +
heatmap data for significantly changing proteins</p></li>
<li><p>sep_sign_condition.csv: Dataset of counts per condition (mean of
repl) per protein for significantly changing proteins</p></li>
</ul>
<pre class="r"><code># List of significantly changing proteins: sign_prot_cell
write(sign_prot_sep, file = &quot;output/DS113_InhibCombiIDseq/IDseq_sign_threshold/basal_active/sep_sign_proteins.txt&quot;)

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_sep, file = &quot;output/DS113_InhibCombiIDseq/IDseq_sign_threshold/basal_active/sep_sign_DESeq2data.csv&quot;, row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_sep, file = &quot;output/DS113_InhibCombiIDseq/IDseq_sign_threshold/basal_active/sep_sign_sample.csv&quot;, row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_sep, file = &quot;output/DS113_InhibCombiIDseq/IDseq_sign_threshold/basal_active/sep_sign_condition.csv&quot;, row.names = F)</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9943460 531.1   14474182 773.1 14474182 773.1
Vcells 18314535 139.8   31968708 244.0 25106744 191.6</code></pre>
</div>
</div>
<div id="active-w-inhib" class="section level2">
<h2>Active w/ inhib</h2>
<div id="deseq2-1" class="section level3 tabset">
<h3 class="tabset">DEseq2</h3>
<p><a
href="https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html"
class="uri">https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html</a></p>
<div id="comp-to-000-um" class="section level4">
<h4>Comp to 0+0+0 uM</h4>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>aIg+H2O2 stimulus only, all inhibitor conditions (39
samples)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_conc (ref: 0 + 0 + 0 uM)</p>
<p><em>Parameter options: stimulus + inhib_conc + inhib_conc_text +
description</em></p></li>
</ul>
<pre class="r"><code># Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- counts_dtbl  %&gt;%
  dplyr::filter(stimulus == &quot;aIg+H2O2&quot; &amp; true_counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, true_counts) %&gt;%
  dplyr::filter(!is.na(true_counts)) %&gt;%
  spread(plate_well, true_counts) %&gt;%
  replace(is.na(.), 0)

rownames(cts) &lt;- cts$target_nospace
cts &lt;- as.matrix(as.data.frame(cts[2:ncol(cts)]))

metadata_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;stimulus&quot;, &quot;inhibitor&quot;, &quot;inhib_conc&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)
coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(counts_dtbl, all_of(metadata_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_DS113), 
         inhib_group = factor(inhib_group, levels = inhib_group_DS113), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS113)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_conc

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_conc &lt;- relevel(dds$inhib_conc, ref = &quot;0 + 0 + 0 uM&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
DESeq_comparisons</code></pre>
<pre><code> [1] &quot;Intercept&quot;                                    
 [2] &quot;inhib_conc_1...1...1.uM_vs_0...0...0.uM&quot;      
 [3] &quot;inhib_conc_2.5...2.5...2.5.uM_vs_0...0...0.uM&quot;
 [4] &quot;inhib_conc_10...10...10.uM_vs_0...0...0.uM&quot;   
 [5] &quot;inhib_conc_10...0...0.uM_vs_0...0...0.uM&quot;     
 [6] &quot;inhib_conc_0...10...0.uM_vs_0...0...0.uM&quot;     
 [7] &quot;inhib_conc_0...0...10.uM_vs_0...0...0.uM&quot;     
 [8] &quot;inhib_conc_2.5...2.5...0.uM_vs_0...0...0.uM&quot;  
 [9] &quot;inhib_conc_2.5...0...2.5.uM_vs_0...0...0.uM&quot;  
[10] &quot;inhib_conc_0...2.5...2.5.uM_vs_0...0...0.uM&quot;  
[11] &quot;inhib_conc_5...1...1.uM_vs_0...0...0.uM&quot;      
[12] &quot;inhib_conc_1...5...1.uM_vs_0...0...0.uM&quot;      
[13] &quot;inhib_conc_1...1...5.uM_vs_0...0...0.uM&quot;      </code></pre>
<p><strong>Log2FC results:</strong> Significance threshold: padj &lt;
0.05</p>
<pre class="r"><code># Create results for all DESeq_comparisons
results_comp0_1_1_1 &lt;- results(dds, name = &quot;inhib_conc_1...1...1.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_2.5_2.5_2.5 &lt;- results(dds, name = &quot;inhib_conc_2.5...2.5...2.5.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_10_10_10 &lt;- results(dds, name = &quot;inhib_conc_10...10...10.uM_vs_0...0...0.uM&quot;, alpha = 0.05)

results_comp0_10_0_0 &lt;- results(dds, name = &quot;inhib_conc_10...0...0.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_0_10_0 &lt;- results(dds, name = &quot;inhib_conc_0...10...0.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_0_0_10 &lt;- results(dds, name = &quot;inhib_conc_0...0...10.uM_vs_0...0...0.uM&quot;, alpha = 0.05)

results_comp0_2.5_2.5_0 &lt;- results(dds, name = &quot;inhib_conc_2.5...2.5...0.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_2.5_0_2.5 &lt;- results(dds, name = &quot;inhib_conc_2.5...0...2.5.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_0_2.5_2.5 &lt;- results(dds, name = &quot;inhib_conc_0...2.5...2.5.uM_vs_0...0...0.uM&quot;, alpha = 0.05)

results_comp0_5_1_1 &lt;- results(dds, name = &quot;inhib_conc_5...1...1.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_1_5_1 &lt;- results(dds, name = &quot;inhib_conc_1...5...1.uM_vs_0...0...0.uM&quot;, alpha = 0.05)
results_comp0_1_1_5 &lt;- results(dds, name = &quot;inhib_conc_1...1...5.uM_vs_0...0...0.uM&quot;, alpha = 0.05)

list_results &lt;- list(results_comp0_1_1_1, results_comp0_2.5_2.5_2.5, results_comp0_10_10_10, 
                     results_comp0_10_0_0, results_comp0_0_10_0, results_comp0_0_0_10, 
                     results_comp0_2.5_2.5_0, results_comp0_2.5_0_2.5, results_comp0_0_2.5_2.5,
                     results_comp0_5_1_1, results_comp0_1_5_1, results_comp0_1_1_5)

# Results summary
summary(results_comp0_2.5_2.5_2.5, padj = 0.05)</code></pre>
<pre><code>
out of 111 with nonzero total read count
adjusted p-value &lt; 0.05
LFC &gt; 0 (up)       : 10, 9%
LFC &lt; 0 (down)     : 22, 20%
outliers [1]       : 0, 0%
low counts [2]     : 0, 0%
(mean count &lt; 514)
[1] see &#39;cooksCutoff&#39; argument of ?results
[2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># sum(results_aIg$padj &lt; 0.05, na.rm = TRUE)</code></pre>
<pre class="r"><code>par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff &lt;- 0.05
plotMA(results_comp0_2.5_2.5_2.5, alpha = signcutoff, main = &quot;2.5+2.5+2.5 uM vs\n0+0+0 uM&quot;)
plotMA(results_comp0_0_0_10, alpha = signcutoff, main = &quot;0+0+10 uM vs\n0+0+0 uM&quot;)
plotMA(results_comp0_5_1_1, alpha = signcutoff, main = &quot;5+1+1 uM vs\n0+0+0 uM&quot;)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_log2FC_comp0-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_log2FC_comp0-1">
Past versions of fig_log2FC_comp0-1.png
</button>
</p>
<div id="fig-fig_log2FC_comp0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="data-transformation-1" class="section level5 tabset">
<h5 class="tabset">Data transformation</h5>
<pre class="r"><code># Normtransform (log2(n+1))
ntd &lt;- normTransform(dds)

# Variance stabilising transform
vsd &lt;- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)</code></pre>
<div id="pca-2" class="section level6">
<h6>PCA</h6>
<pre class="r"><code>list_datatransform &lt;- list(ntd, vsd, rld)
titles_datatransform &lt;- c(&quot;ntd&quot;, &quot;vsd&quot;, &quot;rld&quot;)

for (data in 1:3){
  options(repr.plot.width = 6, repr.plot.height = 6)
  pcaData &lt;- plotPCA(unlist(list_datatransform[[data]]), intgroup = c(&quot;stimulus&quot;, &quot;inhib_group&quot;, &quot;inhib_conc&quot;), returnData = TRUE)
  percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))
  
  plot &lt;- ggplot(pcaData, aes(PC1, PC2, shape = inhib_group, color = factor(inhib_conc, levels = inhib_conc_DS113))) + 
    geom_point(size = 3) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;)) + 
    coord_fixed() +
    # facet_wrap(~inhib_group) +
    scale_color_manual(values = colors_conc_uniq, name = legend_label) +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_PCA_comp0-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_comp0-1">
Past versions of fig_PCA_comp0-1.png
</button>
</p>
<div id="fig-fig_PCA_comp0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_PCA_comp0-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_comp0-2">
Past versions of fig_PCA_comp0-2.png
</button>
</p>
<div id="fig-fig_PCA_comp0-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_PCA_comp0-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_PCA_comp0-3">
Past versions of fig_PCA_comp0-3.png
</button>
</p>
<div id="fig-fig_PCA_comp0-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> PCA of all samples included in the
DESeq2 analysis.</em></p>
</div>
<div id="variance-1" class="section level6">
<h6>Variance</h6>
<pre class="r"><code>options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd &lt;- meanSdPlot(assay(ntd))</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_comp0-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_comp0-1">
Past versions of fig_sd_comp0-1.png
</button>
</p>
<div id="fig-fig_sd_comp0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_ntd &lt;- msd_ntd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Norm (log2(n+1))&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_vsd &lt;- meanSdPlot(assay(vsd))</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_comp0-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_comp0-2">
Past versions of fig_sd_comp0-2.png
</button>
</p>
<div id="fig-fig_sd_comp0-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_vsd &lt;- msd_vsd$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Variance stabilising&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

msd_rld &lt;- meanSdPlot(assay(rld))</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_comp0-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_comp0-3">
Past versions of fig_sd_comp0-3.png
</button>
</p>
<div id="fig-fig_sd_comp0-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plot_rld &lt;- msd_rld$gg + 
  labs(x = &quot;Ranked mean&quot;, y = &quot;Standard deviation&quot;, title = &quot;Regularized log&quot;) + 
  theme_bw() + theme(legend.position = &quot;none&quot;) + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_sd_comp0-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_sd_comp0-4">
Past versions of fig_sd_comp0-4.png
</button>
</p>
<div id="fig-fig_sd_comp0-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample-to-sample-distances-1" class="section level6">
<h6>Sample-to-sample distances</h6>
<pre class="r"><code>options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists &lt;- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix &lt;- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) &lt;- paste(list_datatransform[[data]]$cell_line,
  #                                     list_datatransform[[data]]$stimulus,
  #                                     # list_datatransform[[data]]$inhibitor,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = &quot; - &quot;)
  rownames(sampleDistMatrix) &lt;- list_datatransform[[data]]$descript_rep
  colnames(sampleDistMatrix) &lt;- NULL
  colors &lt;- colorRampPalette(rev(brewer.pal(9, &quot;Blues&quot;)))(255)
  
  plot &lt;- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_distances_comp0-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_comp0-1">
Past versions of fig_distances_comp0-1.png
</button>
</p>
<div id="fig-fig_distances_comp0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_distances_comp0-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_comp0-2">
Past versions of fig_distances_comp0-2.png
</button>
</p>
<div id="fig-fig_distances_comp0-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_distances_comp0-3.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_distances_comp0-3">
Past versions of fig_distances_comp0-3.png
</button>
</p>
<div id="fig-fig_distances_comp0-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="volcano-1" class="section level5">
<h5>Volcano</h5>
<p>Highlight the significantly changing proteins:</p>
<ul>
<li><p>padj &lt; 0.05</p></li>
<li><p>log2FC &gt; log2(1.1)</p></li>
</ul>
<pre class="r"><code># Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc &lt;- c(&quot;volc_comp0_1_1_1&quot;, &quot;volc_comp0_2.5_2.5_2.5&quot;, &quot;volc_comp0_10_10_10&quot;, 
                        &quot;volc_comp0_10_0_0&quot;, &quot;volc_comp0_0_10_0&quot;, &quot;volc_comp0_0_0_10&quot;, 
                        &quot;volc_comp0_2.5_2.5_0&quot;, &quot;volc_comp0_2.5_0_2.5&quot;, &quot;volc_comp0_0_2.5_2.5&quot;,
                        &quot;volc_comp0_5_1_1&quot;, &quot;volc_comp0_1_5_1&quot;, &quot;volc_comp0_1_1_5&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}</code></pre>
<pre class="r"><code># Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results &lt;- function(data = data, 
                                title_plot = &quot;&quot;, 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c(&quot;red&quot;), 
                                point_colors = c(UP = &quot;navy&quot;, NO = &quot;black&quot;, DOWN = &quot;firebrick&quot;)){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = &quot;y&quot;,
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange &lt; 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = &#39;black&#39;, size = 9),
          axis.text.y = element_text(colour = &#39;black&#39;, size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_comp0_1_1_1, volc_comp0_2.5_2.5_2.5, volc_comp0_10_10_10, 
                          volc_comp0_10_0_0, volc_comp0_0_10_0, volc_comp0_0_0_10, 
                          volc_comp0_2.5_2.5_0, volc_comp0_2.5_0_2.5, volc_comp0_0_2.5_2.5,
                          volc_comp0_5_1_1, volc_comp0_1_5_1, volc_comp0_1_1_5)

titles_volcano &lt;- c(&quot;1 + 1 + 1 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;2.5 + 2.5 + 2.5 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;10 + 10 + 10 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, 
                     &quot;10 + 0 + 0 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;0 + 10 + 0 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;0 + 0 + 10 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, 
                     &quot;2.5 + 2.5 + 0 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;2.5 + 0 + 2.5 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;0 + 2.5 + 2.5 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;,
                     &quot;5 + 1 + 1 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;1 + 5 + 1 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;, &quot;1 + 1 + 5 uM vs 0 uM (iSYK + iBTK + iNFkB)&quot;)
# titles_volcano &lt;- names_results_volc

for(dataset in c(1:length(list_results_volc))){
  plot &lt;- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(-log2(1.1), log2(1.1)))
  print(plot)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-1">
Past versions of fig_volcano_comp0-1.png
</button>
</p>
<div id="fig-fig_volcano_comp0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-2.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-2">
Past versions of fig_volcano_comp0-2.png
</button>
</p>
<div id="fig-fig_volcano_comp0-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-3.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-3">
Past versions of fig_volcano_comp0-3.png
</button>
</p>
<div id="fig-fig_volcano_comp0-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-4.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-4">
Past versions of fig_volcano_comp0-4.png
</button>
</p>
<div id="fig-fig_volcano_comp0-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-5.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-5">
Past versions of fig_volcano_comp0-5.png
</button>
</p>
<div id="fig-fig_volcano_comp0-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-6.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-6">
Past versions of fig_volcano_comp0-6.png
</button>
</p>
<div id="fig-fig_volcano_comp0-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-7.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-7">
Past versions of fig_volcano_comp0-7.png
</button>
</p>
<div id="fig-fig_volcano_comp0-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-8.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-8">
Past versions of fig_volcano_comp0-8.png
</button>
</p>
<div id="fig-fig_volcano_comp0-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-9.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-9">
Past versions of fig_volcano_comp0-9.png
</button>
</p>
<div id="fig-fig_volcano_comp0-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-10.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-10">
Past versions of fig_volcano_comp0-10.png
</button>
</p>
<div id="fig-fig_volcano_comp0-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-11.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-11">
Past versions of fig_volcano_comp0-11.png
</button>
</p>
<div id="fig-fig_volcano_comp0-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_volcano_comp0-12.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_volcano_comp0-12">
Past versions of fig_volcano_comp0-12.png
</button>
</p>
<div id="fig-fig_volcano_comp0-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Volcano plot of DESeq2 model
comparisons.</em></p>
<p><em>Add summary of results</em></p>
</div>
<div id="heatmap-1" class="section level5">
<h5>Heatmap</h5>
<p>We continue with the regularised log transformed results (see Data
transform section). We create heatmaps of significant changes:</p>
<ul>
<li><p>padj &lt; 0.05</p></li>
<li><p>log2FC &gt; log2(1.1)</p></li>
</ul>
<pre class="r"><code># Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat &lt;- c(&quot;heat_comp0_1_1_1&quot;, &quot;heat_comp0_2.5_2.5_2.5&quot;, &quot;heat_comp0_10_10_10&quot;, 
                        &quot;heat_comp0_10_0_0&quot;, &quot;heat_comp0_0_10_0&quot;, &quot;heat_comp0_0_0_10&quot;, 
                        &quot;heat_comp0_2.5_2.5_0&quot;, &quot;heat_comp0_2.5_0_2.5&quot;, &quot;heat_comp0_0_2.5_2.5&quot;,
                        &quot;heat_comp0_5_1_1&quot;, &quot;heat_comp0_1_5_1&quot;, &quot;heat_comp0_1_1_5&quot;)

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}

# heat_results_all &lt;- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap &lt;- as.data.frame(colData(dds)[, c(&quot;inhib_group&quot;, &quot;inhib_conc&quot;)]) %&gt;%
  arrange(factor(inhib_group, levels = inhib_group_DS113), factor(inhib_conc, levels = inhib_conc_DS113))</code></pre>
<pre class="r"><code># , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# 10 uM equal vs 0 uM equal
heat_comp0_10_10_10$marker_pathway[is.na(heat_comp0_10_10_10$marker_pathway)] &lt;- &quot;Other&quot;

ann_colors_man &lt;- list(stimulus = colors_stim,
                       inhib_conc = colors_conc_uniq, 
                       inhib_group = c(&quot;DMSO ctrl&quot; = &quot;#737373&quot;, three_equal = &quot;#6a51a3&quot;, single = &quot;#41ab5d&quot;, two = &quot;#f16913&quot;, three_mix = &quot;#4292c6&quot;)
                       # modification = c(phospho = &quot;#f768a1&quot;, no = &quot;grey&quot;, cleaved = &quot;#fff7bc&quot;),
                       # marker_pathway = setNames(colors_light12[1:length(unique(heat_active_wo$marker_pathway))], unique(heat_active_wo$marker_pathway))
)

# All significant proteins in all comparisons
sign_prot_comp0 &lt;- sort(unique(c(heat_comp0_1_1_1$target_nospace, 
                               heat_comp0_2.5_2.5_2.5$target_nospace, 
                               heat_comp0_10_10_10$target_nospace, 
                               heat_comp0_10_0_0$target_nospace, 
                               heat_comp0_0_10_0$target_nospace, 
                               heat_comp0_0_0_10$target_nospace, 
                               heat_comp0_2.5_2.5_0$target_nospace, 
                               heat_comp0_2.5_0_2.5$target_nospace, 
                               heat_comp0_0_2.5_2.5$target_nospace, 
                               heat_comp0_5_1_1$target_nospace, 
                               heat_comp0_1_5_1$target_nospace, 
                               heat_comp0_1_1_5$target_nospace)))

# Get the rld transformed data for the heatmap and scale per row
# data_comp0_10_10_10 &lt;- t(scale(t(assay(rld)[unique(c(heat_comp0_10_10_10$target_nospace)), rownames(data_heatmap)])))
data_comp0 &lt;- t(scale(t(assay(rld)[sign_prot_comp0, rownames(data_heatmap)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

ComplexHeatmap::pheatmap(
  data_comp0,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c(&quot;modification&quot;, &quot;marker_pathway&quot;)],
  # annotation_row = heat_HBL1[, &quot;marker_pathway&quot;, drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = &quot;row&quot;,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = data_heatmap$inhib_group,
  # column_split = data_heatmap$stimulus,
  # row_split = subset(heat_comp0_10_10_10)$modification,
  # cellwidth = 15,
  # cellheight = 15,
  # fontsize = 12,
  # border_color = NA,
  # main = &quot;scaled per row, ComplexHeatmap&quot;
)</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_heatmap_comp0-1.png" width="1440" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heatmap_comp0-1">
Past versions of fig_heatmap_comp0-1.png
</button>
</p>
<div id="fig-fig_heatmap_comp0-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><em><strong>Figure:</strong> Heatmap of all significant results in
all comparisons.</em></p>
<p><em>Higher concentrations of equal amounts of inhibitors have
stronger inhibitory effects. The single inhibitors conditions show some
variation in the inhibition of core BCR phospho-proteins. The two
inhibitors conditions show that iSYK + iNFKB inhibits core BCR proteins
quite well, while the other two combinations better inhibit several
total protein levels. The three inhibitor combinations inhibit the BCR
phospho-proteins very well, especially the combination with higher iSYK
levels.</em></p>
</div>
</div>
<div id="comp-to-101010-um" class="section level4">
<h4>Comp to 10+10+10 uM</h4>
</div>
<div id="comp-to-10-um-isyk" class="section level4">
<h4>Comp to 10 uM iSYK</h4>
</div>
<div id="comp-to-10-um-ibtk" class="section level4">
<h4>Comp to 10 uM iBTK</h4>
</div>
<div id="comp-to-10-um-infkb" class="section level4">
<h4>Comp to 10 uM iNFkB</h4>
</div>
</div>
<div id="individual-proteins" class="section level3">
<h3>Individual proteins</h3>
<pre class="r"><code># options(repr.plot.width = 5, repr.plot.height = 5)

# all_prot &lt;- sort(unique(c(data_well_all$target_nospace)))
# sign_prot_comp0
select_prot &lt;- c(&quot;BLNK_Y84&quot;, &quot;IgM&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;VAV1_Y174&quot;)

for(protein in select_prot) {
  plot1 &lt;- ggplot() +
    geom_jitter(data = subset(data_well_all, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS113), 
                    counts_norm, 
                    color = stimulus, 
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(data_mean_all, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS113),
                      ymin = counts_norm,
                      ymax = counts_norm,
                      group = stimulus,
                      color = stimulus
                      ),
                  width = 0.5, size = 0.75) + #, color = &quot;black&quot;
    facet_wrap(vars(factor(inhib_group, levels = inhib_group_DS113)), scales = &quot;free_x&quot;, nrow = 1) +
    # scale_y_log10() +
    # RotatedAxis() +
    scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
    labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;, title = protein) +
    theme_bw() +
    theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank(), axis.text.x = element_text(angle = 45, hjust = 1, vjust = 1)) +
    textsize_small
  print(plot1)
}</code></pre>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_dot_all-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_dot_all-1">
Past versions of fig_dot_all-1.png
</button>
</p>
<div id="fig-fig_dot_all-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_dot_all-2.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_dot_all-2">
Past versions of fig_dot_all-2.png
</button>
</p>
<div id="fig-fig_dot_all-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_dot_all-3.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_dot_all-3">
Past versions of fig_dot_all-3.png
</button>
</p>
<div id="fig-fig_dot_all-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_dot_all-4.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_dot_all-4">
Past versions of fig_dot_all-4.png
</button>
</p>
<div id="fig-fig_dot_all-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/DS113_s_seq_prelim.Rmd/fig_dot_all-5.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_dot_all-5">
Past versions of fig_dot_all-5.png
</button>
</p>
<div id="fig-fig_dot_all-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-data-1" class="section level3">
<h3>Save data</h3>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;clean_&quot;))
rm(list = ls(pattern = &quot;sign_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9946133 531.2   14474182 773.1 14474182 773.1
Vcells 18362097 140.1   31968708 244.0 25106744 191.6</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ashr_2.2-63                 apeglm_1.26.1              
 [3] MASS_7.3-60.2               ggbreak_0.1.2              
 [5] geomtextpath_0.1.4          ggh4x_0.2.8                
 [7] cowplot_1.1.3               patchwork_1.2.0            
 [9] greekLetters_1.0.2          ggpubr_0.6.0               
[11] ggrepel_0.9.5               scales_1.3.0               
[13] ggthemes_5.1.0              RColorBrewer_1.1-3         
[15] scico_1.5.0                 viridis_0.6.5              
[17] viridisLite_0.4.2           gridExtra_2.3              
[19] mmtable2_0.1.3              gt_0.10.1                  
[21] ggvenn_0.1.10               rgl_1.3.1                  
[23] plotly_4.10.4               ggcorrplot_0.1.4.1         
[25] corrplot_0.92               umap_0.2.10.0              
[27] ggplotify_0.1.2             dendsort_0.3.4             
[29] vsn_3.72.0                  ComplexHeatmap_2.20.0      
[31] platetools_0.1.7            kableExtra_1.4.0           
[33] DESeq2_1.44.0               SummarizedExperiment_1.34.0
[35] Biobase_2.64.0              MatrixGenerics_1.16.0      
[37] matrixStats_1.3.0           GenomicRanges_1.56.1       
[39] GenomeInfoDb_1.40.1         IRanges_2.38.0             
[41] S4Vectors_0.42.0            BiocGenerics_0.50.0        
[43] Seurat_5.1.0                SeuratObject_5.0.2         
[45] sp_2.1-4                    rstatix_0.7.2              
[47] factoextra_1.0.7            FactoMineR_2.11            
[49] readxl_1.4.3                Matrix_1.7-0               
[51] lubridate_1.9.3             forcats_1.0.0              
[53] stringr_1.5.1               dplyr_1.1.4                
[55] purrr_1.0.2                 readr_2.1.5                
[57] tidyr_1.3.1                 tibble_3.2.1               
[59] ggplot2_3.5.1               tidyverse_2.0.0            
[61] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             goftest_1.2-3           DT_0.33                
  [4] TH.data_1.1-2           vctrs_0.6.5             spatstat.random_3.2-3  
  [7] digest_0.6.36           png_0.1-8               shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       magick_2.8.3            SQUAREM_2021.1         
 [16] reshape2_1.4.4          httpuv_1.6.15           foreach_1.5.2          
 [19] withr_3.0.0             xfun_0.45               ggfun_0.1.5            
 [22] survival_3.6-4          memoise_2.0.1           hexbin_1.28.3          
 [25] emmeans_1.10.2          systemfonts_1.1.0       zoo_1.8-12             
 [28] GlobalOptions_0.1.2     pbapply_1.7-2           promises_1.3.0         
 [31] scatterplot3d_0.3-44    httr_1.4.7              globals_0.16.3         
 [34] fitdistrplus_1.1-11     ps_1.7.6                rstudioapi_0.16.0      
 [37] UCSC.utils_1.0.0        miniUI_0.1.1.1          generics_0.1.3         
 [40] base64enc_0.1-3         processx_3.8.4          zlibbioc_1.50.0        
 [43] polyclip_1.10-6         GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [46] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [49] S4Arrays_1.4.1          preprocessCore_1.66.0   hms_1.1.3              
 [52] irlba_2.3.5.1           colorspace_2.1-0        ROCR_1.0-11            
 [55] reticulate_1.38.0       spatstat.data_3.1-2     magrittr_2.0.3         
 [58] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [61] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [64] scattermore_1.2         RcppAnnoy_0.0.22        pillar_1.9.0           
 [67] nlme_3.1-164            iterators_1.0.14        compiler_4.4.1         
 [70] RSpectra_0.16-1         stringi_1.8.4           tensor_1.5             
 [73] plyr_1.8.9              crayon_1.5.3            abind_1.4-5            
 [76] truncnorm_1.0-9         gridGraphics_0.5-1      emdbook_1.3.13         
 [79] locfit_1.5-9.10         bit_4.0.5               sandwich_3.1-0         
 [82] whisker_0.4.1           codetools_0.2-20        multcomp_1.4-25        
 [85] textshaping_0.4.0       openssl_2.2.0           crosstalk_1.2.1        
 [88] bslib_0.7.0             GetoptLong_1.0.5        mime_0.12              
 [91] splines_4.4.1           circlize_0.4.16         Rcpp_1.0.12            
 [94] fastDummies_1.7.3       cellranger_1.1.0        Rttf2pt1_1.3.12        
 [97] leaps_3.2               knitr_1.47              utf8_1.2.4             
[100] clue_0.3-65             fs_1.6.4                listenv_0.9.1          
[103] ggsignif_0.6.4          estimability_1.5.1      callr_3.7.6            
[106] statmod_1.5.0           tzdb_0.4.0              svglite_2.1.3          
[109] pkgconfig_2.0.3         tools_4.4.1             cachem_1.1.0           
[112] numDeriv_2016.8-1.1     fastmap_1.2.0           rmarkdown_2.27         
[115] ica_1.0-3               broom_1.0.6             sass_0.4.9             
[118] coda_0.19-4.1           BiocManager_1.30.23     dotCall64_1.1-1        
[121] carData_3.0-5           RANN_2.6.1              farver_2.1.2           
[124] yaml_2.3.8              cli_3.6.3               leiden_0.4.3.1         
[127] lifecycle_1.0.4         askpass_1.2.0           uwot_0.2.2             
[130] mvtnorm_1.2-5           backports_1.5.0         BiocParallel_1.38.0    
[133] timechange_0.3.0        gtable_0.3.5            rjson_0.2.21           
[136] ggridges_0.5.6          progressr_0.14.0        parallel_4.4.1         
[139] limma_3.60.3            jsonlite_1.8.8          RcppHNSW_0.6.0         
[142] bit64_4.0.5             multcompView_0.1-10     assertthat_0.2.1       
[145] Rtsne_0.17              yulab.utils_0.1.4       spatstat.utils_3.0-5   
[148] bdsmatrix_1.3-7         highr_0.11              jquerylib_0.1.4        
[151] lazyeval_0.2.2          shiny_1.8.1.1           htmltools_0.5.8.1      
[154] affy_1.82.0             sctransform_0.4.1       glue_1.7.0             
[157] spam_2.10-0             XVector_0.44.0          rprojroot_2.0.4        
[160] flashClust_1.01-2       invgamma_1.1            igraph_2.0.3           
[163] extrafontdb_1.0         R6_2.5.1                labeling_0.4.3         
[166] bbmle_1.0.25.1          cluster_2.1.6           aplot_0.2.3            
[169] DelayedArray_0.30.1     tidyselect_1.2.1        xml2_1.3.6             
[172] car_3.1-2               future_1.33.2           munsell_0.5.1          
[175] KernSmooth_2.23-24      affyio_1.74.0           data.table_1.15.4      
[178] htmlwidgets_1.6.4       rlang_1.1.4             spatstat.sparse_3.1-0  
[181] extrafont_0.19          spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
