<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-10-22" />

<title>DLBCL proteins chapter main figures</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapters on DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Paper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="y_paper_main.html">Main</a>
    </li>
    <li>
      <a href="z_paper_suppl.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteins ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="t_proteins_main.html.html">Main</a>
    </li>
    <li>
      <a href="u_proteins_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    States ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="v_states_main.html.html">Main</a>
    </li>
    <li>
      <a href="w_states_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
    <li>
      <a href="c_flow_vis.html">Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="DS108_p_seq_comparisons.html">DS108 clean</a>
    </li>
    <li>
      <a href="DS108_q_seq_thresholds.html">DS108 thresholds</a>
    </li>
    <li>
      <a href="DS108_r_seq_pca.html">DS108 PCA</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">DLBCL proteins chapter main figures</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-10-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapters_DLBCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241022code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241022code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241022)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong94c226d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 94c226d
</a>
</p>
</div>
<div id="strongRepositoryversionstrong94c226d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 94c226d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/AL006_InhibCombiTest/
    Untracked:  data/DS102_StimTest/
    Untracked:  data/DS103_ViabilityTest/
    Untracked:  data/DS104_StimInhibTest/
    Untracked:  data/DS108_StimInhibIDseq/
    Untracked:  data/DS113_InhibCombiIDseq/
    Untracked:  output/DS102_StimTest/
    Untracked:  output/DS103_ViabilityTest/
    Untracked:  output/DS104_StimInhibTest/
    Untracked:  output/DS108_StimInhibIDseq/
    Untracked:  output/DS113_InhibCombiIDseq/
    Untracked:  output/figures/
    Untracked:  output/network_visual_DLBCL/
    Untracked:  output/network_visual_paper/

Unstaged changes:
    Deleted:    analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/t_proteins_main.Rmd</code>) and
HTML (<code>docs/t_proteins_main.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
1ffbd94
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Revised chapter figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
6559698
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Paper figs + analysis files for ch figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)
PANEL_labels &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;,&quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;)

textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
                  axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
                  axis.title = element_text(colour = &quot;black&quot;, size = 12), 
                  legend.title = element_text(colour = &quot;black&quot;, size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
                  strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
)

textsize_small &lt;- theme(text = element_text(size = 9, family = &quot;sans&quot;, colour = &quot;black&quot;),
                        plot.title = element_text(size = 10)
)</code></pre>
<pre class="r"><code>colors_nord &lt;- c(&quot;#2e3440&quot;, &quot;#3b4252&quot;, &quot;#434c5e&quot;, &quot;#4c566a&quot;, 
                 &quot;#d8dee9&quot;, &quot;#e5e9f0&quot;, &quot;#eceff4&quot;, 
                 &quot;#8fbcbb&quot;, &quot;#88c0d0&quot;, &quot;#81a1c1&quot;, &quot;#5e81ac&quot;, 
                 &quot;#bf616a&quot;, &quot;#d08770&quot;, &quot;#ebcb8b&quot;, &quot;#a3be8c&quot;, &quot;#b48ead&quot;)

colors_toll_bright &lt;- c(&quot;#4477AA&quot;, &quot;#66CCEE&quot;, &quot;#228833&quot;, &quot;#CCBB44&quot;, &quot;#EE6677&quot;, &quot;#AA3377&quot;, &quot;#BBBBBB&quot;)
colors_toll_mute &lt;- c(&quot;#332288&quot;, &quot;#88CCEE&quot;, &quot;#44AA99&quot;, &quot;#117733&quot;, &quot;#999933&quot;, &quot;#DDCC77&quot;, &quot;#CC6677&quot;, &quot;#882255&quot;, &quot;#AA4499&quot;, &quot;#DDDDDD&quot;)
colors_toll_br &lt;- c(&quot;#364B9A&quot;, &quot;#4A7BB7&quot;, &quot;#6EA6CD&quot;, &quot;#98CAE1&quot;, &quot;#C2E4EF&quot;, &quot;#EAECCC&quot;, &quot;#FEDA8B&quot;, &quot;#FDB366&quot;, &quot;#F67A4B&quot;, &quot;#DD3D2D&quot;, &quot;#A50026&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)
colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

# Variable-specific colors
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8&quot; = &quot;#332288&quot;, 
                 &quot;ABC subtype&quot; = &quot;#AA4499&quot;, &quot;GCB subtype&quot; = &quot;#332288&quot;)
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#737373&quot;, &quot;aIg+H2O2&quot; = &quot;#FDB366&quot;, 
                 &quot;Basal&quot; = &quot;#737373&quot;, &quot;Activated&quot; = &quot;#FDB366&quot;)
colors_cell_stim &lt;- c(&quot;HBL1 - PBS&quot; = &quot;#c994c7&quot;, &quot;HBL1 - aIg+H2O2&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8 - PBS&quot; = &quot;#88CCEE&quot;, &quot;OCI-Ly8 - aIg+H2O2&quot; = &quot;#332288&quot;)
# colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#228833&quot;, &quot;iBTK&quot; = &quot;#4477AA&quot;, &quot;iNFkB&quot; = &quot;#AA3377&quot;)
colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#9dce5c&quot;, &quot;iBTK&quot; = &quot;#5c9dce&quot;, &quot;iNFkB&quot; = &quot;#ce5c9d&quot;)
colors_conc &lt;- c(&quot;#737373&quot;, &quot;#dadaeb&quot;, &quot;#9e9ac8&quot;, &quot;#6a51a3&quot;, &quot;#3f007d&quot;)
colors_iSYK &lt;- colors_green9[c(3, 5, 6, 8, 9)]
colors_iBTK &lt;- colors_blue9[c(3, 5, 6, 8, 9)]
colors_iNFkB &lt;- colors_pr9[c(3, 5, 6, 8, 9)]
colors_conc_stim &lt;- colors_orange9[c(2, 3, 5, 6, 8, 9)]
colors_conc_PBS &lt;- colors_grey9[c(2, 3, 5, 6, 8, 9)]</code></pre>
</div>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<p><em>Sequencing-based characterisation of BCR signaling in two DLBCL
subtype cell lines.</em></p>
<p>Input:</p>
<ul>
<li>BioRender schematic figures</li>
</ul>
<div id="panels" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># # Placeholder
# fig1_panelA &lt;- ggplot() +
#   geom_blank() +
#   scale_x_continuous(limits = c(0, 10)) +
#   scale_y_continuous(limits = c(0, 10)) +
#   # labs(title = &quot;BCR signaling network&quot;) +
#   theme_bw() +
#   theme(axis.text.x = element_blank(),
#         axis.text.y = element_blank(),
#         axis.ticks = element_blank(),
#         panel.grid = element_blank()) +
#   textsize_small
# # fig1_panelA

# PNG
fig1_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/ch4_BCR_network_simple_inhib_DLBCL.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
fig1_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig1_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig1_panelA-1">
Past versions of fig1_panelA-1.png
</button>
</p>
<div id="fig-fig1_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># # Placeholder
# fig1_panelB &lt;- ggplot() +
#   geom_blank() +
#   scale_x_continuous(limits = c(0, 10)) +
#   scale_y_continuous(limits = c(0, 10)) +
#   # labs(title = &quot;BCR signaling network&quot;) +
#   theme_bw() +
#   theme(axis.text.x = element_blank(),
#         axis.text.y = element_blank(),
#         axis.ticks = element_blank(),
#         panel.grid = element_blank()) +
#   textsize_small
# # fig1_panelB

# PNG
fig1_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/ch4_DLBCL_inhibition_workflow.png&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
fig1_panelB</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig1_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig1_panelB-1">
Past versions of fig1_panelB-1.png
</button>
</p>
<div id="fig-fig1_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig1 &lt;- plot_grid(fig1_panelA, fig1_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 2.1), label_size = 10)
fig1</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig1_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig1_combi-1">
Past versions of fig1_combi-1.png
</button>
</p>
<div id="fig-fig1_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig1,
  filename = &quot;output/figures/DLBCL_proteins/main_fig1.pdf&quot;,
  width = 6,
  height = 2.4,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig1,
  filename = &quot;output/figures/DLBCL_proteins/main_fig1.jpg&quot;,
  width = 6,
  height = 2.4,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Sequencing-based characterisation of BCR signaling in two DLBCL
subtype cell lines.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig1&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9156973 489.1   16807764 897.7  11029544 589.1
Vcells 37750365 288.1  130872887 998.5 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<p><em>ID-seq confirms basal and activated BCR signaling in ABC- and
GCB-type DLBCL.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li><p>DESeq2 analysis</p></li>
<li><p>Count data</p></li>
</ul></li>
</ul>
<div id="data" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig2_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) %&gt;%
  subset(inhib_new == &quot;PBS ctrl&quot;)
fig2_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) %&gt;%
  subset(inhib_new == &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig2_cells &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
fig2_cells_clean &lt;- c(&quot;ABC subtype&quot;, &quot;GCB subtype&quot;)
fig2_cells_short &lt;- c(&quot;ABC&quot;, &quot;GCB&quot;)
fig2_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig2_stim_clean &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig2_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;cell_clean&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;cell_stim&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-1" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-1" class="section level4">
<h4>Panel A</h4>
<p>Differential expression analysis comparing basal signaling in HBL1
and OCI-Ly8 cells</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>2 cell lines, PBS stimulation only</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_clean</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig2_data_id  %&gt;%
  dplyr::filter(stim_clean == &quot;Basal&quot; &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig2_data_id, all_of(fig2_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ cell_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_clean &lt;- relevel(dds$cell_clean, ref = &quot;ABC subtype&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_cell &lt;- results(dds, name = &quot;cell_clean_GCB.subtype_vs_ABC.subtype&quot;, alpha = 0.05)

list_results &lt;- list(results_cell)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_results_volc &lt;- c(&quot;volc_cell&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.05)))
}</code></pre>
<pre class="r"><code># Select data
fig2A_dataset &lt;- volc_cell
fig2A_dataset$wrapper = &quot;Basal signaling: ABC vs GCB subtype&quot;

# Figure
fig2_panelA &lt;- ggplot(data = fig2A_dataset,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.85,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(fig2A_dataset$log2FoldChange &lt; 0, -0.5 - fig2A_dataset$log2FoldChange, 0.5 - fig2A_dataset$log2FoldChange), 
                  max.overlaps = 9
  ) +
  # facet_wrap(~wrapper, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#332288&quot;, NO = &quot;black&quot;, DOWN = &quot;#AA4499&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;GCB subtype&quot;, &quot;No difference&quot;, &quot;ABC subtype&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.05), log2(1.05)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;Basal signaling: ABC vs GCB subtype&quot;) +
  theme_bw() +
  theme(panel.grid = element_blank(), legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  textsize_small
fig2_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig2_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig2_panelA-1">
Past versions of fig2_panelA-1.png
</button>
</p>
<div id="fig-fig2_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-1" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
fig2B_proteins &lt;- c(&quot;CREB_S133&quot;, &quot;GSK3b_S9&quot;)#&quot;IgM&quot;, &quot;CD43&quot;, 
fig2B_proteins_name &lt;- c(&quot;CREB (S133)&quot;, &quot;GSK3b (S9)&quot;) #&quot;IgM&quot;, &quot;CD43&quot;, 

fig2B_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2B_proteins &amp; stim_clean == &quot;Basal&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig2B_proteins))
fig2B_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2B_proteins &amp; stim_clean == &quot;Basal&quot;) %&gt;%
  group_by(target_nospace, cell_clean) %&gt;%
  mutate(counts_norm_avg = mean(counts_norm), 
         target_nospace = factor(target_nospace, levels = fig2B_proteins))

# Statistics
# Kruskal-Wallis test
fig2B_kruskal &lt;- fig2B_data %&gt;%
  group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ cell_clean)
fig2B_kruskal</code></pre>
<pre><code># A tibble: 2 × 7
  target_nospace .y.             n statistic    df        p method        
* &lt;fct&gt;          &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;         
1 CREB_S133      counts_norm    18      11.0     1 0.000929 Kruskal-Wallis
2 GSK3b_S9       counts_norm    18      12.8     1 0.000349 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
fig2B_dunn &lt;- fig2B_data %&gt;%
  group_by(target_nospace) %&gt;%
  dunn_test(counts_norm ~ cell_clean, p.adjust.method = &quot;BH&quot;) #%&gt;%
  # dplyr::filter(
  #   (group1 == &quot;Activated\nDMSO ctrl&quot; &amp; group2 == &quot;Basal\nDMSO ctrl&quot;) |
  #     (group1 == &quot;Activated\n100 uM iBTK&quot; &amp; group2 == &quot;Basal\n100 uM iBTK&quot;) |
  #     (group1 == &quot;Basal\n100 uM iBTK&quot; &amp; group2 == &quot;Basal\nDMSO ctrl&quot;) |
  #     (group1 == &quot;Activated\n100 uM iBTK&quot; &amp; group2 == &quot;Activated\nDMSO ctrl&quot;)
  #     ) %&gt;%
  # mutate(group1 = case_when(group1 == &quot;Basal\n100 uM iBTK&quot; ~ &quot;Basal\n100 uM\niBTK&quot;, 
  #                           group1 == &quot;Activated\n100 uM iBTK&quot; ~ &quot;Activated\n100 uM\niBTK&quot;, 
  #                           .default = group1), 
  #        group2 = case_when(group2 == &quot;Basal\n100 uM iBTK&quot; ~ &quot;Basal\n100 uM\niBTK&quot;, 
  #                           group2 == &quot;Activated\n100 uM iBTK&quot; ~ &quot;Activated\n100 uM\niBTK&quot;, 
  #                           .default = group2))
# group_by(target_nospace) %&gt;%
# adjust_pvalue(method = &quot;BH&quot;)
fig2B_dunn</code></pre>
<pre><code># A tibble: 2 × 10
  target_nospace .y.         group1 group2    n1    n2 statistic       p   p.adj
* &lt;fct&gt;          &lt;chr&gt;       &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
1 CREB_S133      counts_norm ABC s… GCB s…     9     9     -3.31 9.29e-4 9.29e-4
2 GSK3b_S9       counts_norm ABC s… GCB s…     9     9      3.58 3.49e-4 3.49e-4
# ℹ 1 more variable: p.adj.signif &lt;chr&gt;</code></pre>
<pre class="r"><code># Manual stat table (from DEA; Fig. 1)
fig2B_manual_stat &lt;- tibble(target_nospace = fig2B_dunn$target_nospace,
                            .y. = &quot;counts_norm&quot;,
                            group1 = fig2B_dunn$group1,
                            group2 = fig2B_dunn$group2,
                            p.adj.signif = fig2B_dunn$p.adj.signif,
                            y.position = c(2125, 1050)) #5100, 6100, 
fig2B_manual_stat</code></pre>
<pre><code># A tibble: 2 × 6
  target_nospace .y.         group1      group2      p.adj.signif y.position
  &lt;fct&gt;          &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;       &lt;chr&gt;             &lt;dbl&gt;
1 CREB_S133      counts_norm ABC subtype GCB subtype ***                2125
2 GSK3b_S9       counts_norm ABC subtype GCB subtype ***                1050</code></pre>
<pre class="r"><code># Figure
fig2_panelB &lt;- ggplot() +
  geom_jitter(data = fig2B_data,
              aes(factor(cell_clean, levels = fig2_cells_clean), counts_norm, color = factor(cell_clean, levels = fig2_cells_clean)), 
              alpha = 1, size = 1, width = 0.1) +
  geom_errorbar(data = fig2B_mean,
                aes(factor(cell_clean, levels = fig2_cells_clean),
                    ymin = counts_norm_avg, ymax = counts_norm_avg,
                    group = factor(cell_clean, levels = fig2_cells_clean),
                    color = factor(cell_clean, levels = fig2_cells_clean)),
                width = 0.75, size = 1) +
  facet_wrap(vars(factor(target_nospace, levels = fig2B_proteins, labels = fig2B_proteins_name)), ncol = 1, scales = &quot;free_y&quot;) +
  scale_color_manual(values = colors_cell, breaks = fig2_cells_clean, labels = fig2_cells_short, name = &quot;DLBCL subtype&quot;) +
  labs(x = element_blank(), y = &quot;Normalised counts&quot;) +
  # coord_cartesian(ylim = c(1500, 10200)) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) +
  stat_pvalue_manual(fig2B_manual_stat, label = &quot;p.adj.signif&quot;, tip.length = 0.05,
                     step.increase = 0.22, group = &quot;target_nospace&quot;, step.group.by = &quot;target_nospace&quot;, size = 3, vjust = 1.5) + 
  scale_x_discrete(labels = fig2_cells_short)
fig2_panelB</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig2_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig2_panelB-1">
Past versions of fig2_panelB-1.png
</button>
</p>
<div id="fig-fig2_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c" class="section level4">
<h4>Panel C</h4>
<p>Differential expression analysis of activated vs basal signaling</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>2 cell lines, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_stim</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig2_data_id  %&gt;%
  dplyr::filter(counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig2_data_id, all_of(fig2_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ cell_stim

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_stim &lt;- relevel(dds$cell_stim, ref = &quot;Basal\nABC subtype&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_HBL1_stim &lt;- results(dds, name = &quot;cell_stim_Activated.ABC.subtype_vs_Basal.ABC.subtype&quot;, alpha = 0.05)
results_OCI_stim &lt;- results(dds, contrast = c(&quot;cell_stim&quot;, &quot;Activated\nGCB subtype&quot;, &quot;Basal\nGCB subtype&quot;), alpha = 0.05)

list_results &lt;- list(results_HBL1_stim, results_OCI_stim)

# Data transformation
# Regularized log transform
rld_stim &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_volc &lt;- c(&quot;volc_HBL1_stim&quot;, &quot;volc_OCI_stim&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_HBL1_stim, volc_OCI_stim)
names_heat &lt;- c(&quot;heat_HBL1_stim&quot;, &quot;heat_OCI_stim&quot;)
for(dataset in c(1:length(list_results_volc))){
  assign(names_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}</code></pre>
<pre class="r"><code># Select data
# Combine heatmap datasets (without DESeq2 analysis values)
heat_inhib &lt;- list(heat_HBL1_stim, heat_OCI_stim) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  left_join(fig2_data_id[, 14:21]) %&gt;%
  # dplyr::filter(modification == &quot;phospho&quot;) %&gt;%
  distinct()

fig2C_prot_HBL1 &lt;- sort(unique(c(heat_HBL1_stim$target_nospace)))
fig2C_prot_OCI &lt;- sort(unique(c(heat_OCI_stim$target_nospace)))

# Subset the full dataframe to get metadata for heatmap plotting
fig2C_info &lt;- fig2_data_id %&gt;%
  dplyr::select(plate_well, stim_clean, cell_clean) %&gt;%
  distinct() %&gt;%
  # dplyr::filter(cell_line == &quot;HBL1&quot; &amp;inhibitor == &quot;iBTK&quot; &amp; inhib_conc %in% fig3_conc_all &amp; !(stim_clean == &quot;Basal&quot; &amp; inhib_conc %in% fig3_conc_all[2:5])) %&gt;%
  # mutate(inhibitor = factor(inhibitor, levels = fig3_inhib)) %&gt;%
  as.data.frame()
rownames(fig2C_info) &lt;- fig2C_info$plate_well
fig2C_info &lt;- dplyr::select(fig2C_info, -c(plate_well))

# Get the rld transformed data for the heatmap and scale per row
fig2C_data_heat &lt;- t(scale(t(assay(rld_stim)[c(heat_inhib$target_nospace), rownames(fig2C_info)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

# Figure
fig2C_ann_colors &lt;- list(&quot;Stimulation&quot; = colors_stim, 
                         &quot;DLBCL subtype&quot; = colors_cell)
fig2C_heat_legend &lt;- list(title = &quot;Scaled\nexpression&quot;, title_gp = gpar(fontsize = 7), labels_gp = gpar(fontsize = 6), 
                          grid_width = unit(2, &quot;mm&quot;), grid_height = unit(20, &quot;mm&quot;))

fig2C_col_ann &lt;- HeatmapAnnotation(&quot;DLBCL subtype&quot; = fig2C_info$cell_clean, 
                                   &quot;Stimulation&quot; = fig2C_info$stim_clean, 
                                   simple_anno_size = unit(2.5, &quot;mm&quot;), 
                                   annotation_name_gp = gpar(fontsize = 7),
                                   col = fig2C_ann_colors, 
                                   annotation_legend_param = list(&quot;DLBCL subtype&quot; = list(title = &quot;DLBCL subtype&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 6), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  &quot;Stimulation&quot; = list(title = &quot;Stimulation&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 6), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;))), 
                                   show_legend = c(TRUE, TRUE))

fig2_panelC &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  fig2C_data_heat,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  # treeheight_row = 20,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  top_annotation = fig2C_col_ann,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = fig2C_info$cell_clean,
  row_split = subset(heat_inhib)$modification,
  row_gap = unit(1, &quot;mm&quot;), 
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 5.5,
  row_title_gp = gpar(fontsize = 7),
  column_title_gp = gpar(fontsize = 7), 
  annotation_names_col = gpar(fontsize = 6),
  border_color = NA,
  heatmap_legend_param = fig2C_heat_legend, 
  # annotation_legend = FALSE,
)) +
  ggtitle(&quot;          Activated vs basal signaling&quot;) +
  textsize_small
fig2_panelC</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig2_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig2_panelC-1">
Past versions of fig2_panelC-1.png
</button>
</p>
<div id="fig-fig2_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Select data
fig2D_proteins &lt;- c(&quot;BTK&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;) #, &quot;SHP2_Y580&quot;, &quot;ATF2_T71&quot;
# fig2D_proteins_name &lt;- fig2D_proteins
fig2D_proteins_name &lt;- c(&quot;BTK&quot;, &quot;BTK (Y223)&quot;, &quot;BTK (Y551)&quot;)
fig2D_cell_stim &lt;- c(&quot;Basal\nABC subtype&quot;, &quot;Activated\nABC subtype&quot;, &quot;Basal\nGCB subtype&quot;, &quot;Activated\nGCB subtype&quot;)
fig2D_cell_stim_label &lt;- c(&quot;Basal\n             ABC subtype&quot;, &quot;Active&quot;, &quot;Basal\n             GCB subtype&quot;, &quot;Active&quot;)

fig2D_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2D_proteins) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig2D_proteins), 
         cell_stim = factor(cell_stim, levels = fig2D_cell_stim))
fig2D_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2D_proteins) %&gt;%
  group_by(target_nospace, cell_stim) %&gt;%
  mutate(counts_norm_avg = mean(counts_norm),
         target_nospace = factor(target_nospace, levels = fig2D_proteins), 
         cell_stim = factor(cell_stim, levels = fig2D_cell_stim))

# Statistics
# Kruskal-Wallis test
fig2D_kruskal &lt;- fig2D_data %&gt;%
  group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ cell_stim)
fig2D_kruskal</code></pre>
<pre><code># A tibble: 3 × 7
  target_nospace .y.             n statistic    df           p method        
* &lt;fct&gt;          &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;       &lt;dbl&gt; &lt;chr&gt;         
1 BTK            counts_norm    36      28.9     3 0.0000023   Kruskal-Wallis
2 BTK_Y223       counts_norm    36      29.6     3 0.00000171  Kruskal-Wallis
3 BTK_Y551       counts_norm    36      32.8     3 0.000000348 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
fig2D_dunn &lt;- fig2D_data %&gt;%
  group_by(target_nospace) %&gt;%
  dunn_test(counts_norm ~ cell_stim, p.adjust.method = &quot;none&quot;) %&gt;%
  slice(-c(3, 4, 9, 10, 15, 16)) %&gt;% #, 21, 22, 27, 28
  # mutate(group1 = case_when(group1 == &quot;Basal\n100 uM iBTK&quot; ~ &quot;Basal\n100 uM\niBTK&quot;, 
  #                           group1 == &quot;Activated\n100 uM iBTK&quot; ~ &quot;Activated\n100 uM\niBTK&quot;, 
  #                           .default = group1), 
  #        group2 = case_when(group2 == &quot;Basal\n100 uM iBTK&quot; ~ &quot;Basal\n100 uM\niBTK&quot;, 
  #                           group2 == &quot;Activated\n100 uM iBTK&quot; ~ &quot;Activated\n100 uM\niBTK&quot;, 
  #                           .default = group2))
  group_by(target_nospace) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  mutate(y.position = rep(c(1900, 2250, 3100), each = 4))
fig2D_dunn</code></pre>
<pre><code># A tibble: 12 × 11
   target_nospace .y.        group1 group2    n1    n2 statistic       p   p.adj
   &lt;fct&gt;          &lt;chr&gt;      &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;
 1 BTK            counts_no… &quot;Basa… &quot;Acti…     9     9    1.41   1.59e-1 2.12e-1
 2 BTK            counts_no… &quot;Basa… &quot;Basa…     9     9   -3.33   8.58e-4 1.72e-3
 3 BTK            counts_no… &quot;Acti… &quot;Acti…     9     9   -3.92   9.04e-5 3.62e-4
 4 BTK            counts_no… &quot;Basa… &quot;Acti…     9     9    0.828  4.08e-1 4.08e-1
 5 BTK_Y223       counts_no… &quot;Basa… &quot;Acti…     9     9    5.44   5.44e-8 2.18e-7
 6 BTK_Y223       counts_no… &quot;Basa… &quot;Basa…     9     9    2.68   7.26e-3 9.68e-3
 7 BTK_Y223       counts_no… &quot;Acti… &quot;Acti…     9     9   -2.68   7.26e-3 9.68e-3
 8 BTK_Y223       counts_no… &quot;Basa… &quot;Acti…     9     9    0.0671 9.46e-1 9.46e-1
 9 BTK_Y551       counts_no… &quot;Basa… &quot;Acti…     9     9    5.44   5.44e-8 2.18e-7
10 BTK_Y551       counts_no… &quot;Basa… &quot;Basa…     9     9    1.81   7.00e-2 7.00e-2
11 BTK_Y551       counts_no… &quot;Acti… &quot;Acti…     9     9   -1.81   7.00e-2 7.00e-2
12 BTK_Y551       counts_no… &quot;Basa… &quot;Acti…     9     9    1.81   7.00e-2 7.00e-2
# ℹ 2 more variables: p.adj.signif &lt;chr&gt;, y.position &lt;dbl&gt;</code></pre>
<pre class="r"><code># Figure
fig2_panelD &lt;- ggplot() +
  geom_jitter(data = fig2D_data,
              aes(factor(cell_stim, fig2D_cell_stim), counts_norm, color = factor(stim_clean, levels = fig2_stim_clean)), 
              alpha = 1, size = 1, width = 0.1) +
  geom_errorbar(data = fig2D_mean,
                aes(factor(cell_stim, levels = fig2D_cell_stim),
                    ymin = counts_norm_avg, ymax = counts_norm_avg,
                    group = factor(cell_clean, levels = fig2_cells_clean),
                    color = factor(stim_clean, levels = fig2_stim_clean)),
                width = 0.75, size = 1) +
  stat_pvalue_manual(fig2D_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.05,
                     step.increase = 0.22, group = &quot;target_nospace&quot;, step.group.by = &quot;target_nospace&quot;, size = 3, vjust = 1.2) + 
  facet_wrap(vars(factor(target_nospace, levels = fig2D_proteins, labels = fig2D_proteins_name)), nrow = 1, scales = &quot;free_y&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig2_stim_clean, name = &quot;Stimulation&quot;) +
  scale_x_discrete(breaks = fig2D_cell_stim, labels = fig2D_cell_stim_label) + 
  labs(x = element_blank(), y = &quot;Normalised counts&quot;) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.position = &quot;bottom&quot;, legend.justification = &quot;right&quot;, legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.margin = margin(c(0, 0, 0, 0)))
fig2_panelD</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig2_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig2_panelD-1">
Past versions of fig2_panelD-1.png
</button>
</p>
<div id="fig-fig2_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-1" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig2_AB &lt;- plot_grid(fig2_panelA, fig2_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1.25, 1), label_size = 10)
fig2_CD &lt;- plot_grid(fig2_panelC, fig2_panelD, labels = PANEL_labels[c(3, 4)], ncol = 1, rel_heights = c(1.4, 1), label_size = 10)
fig2 &lt;- plot_grid(fig2_AB, fig2_CD, ncol = 1, rel_heights = c(1, 2), label_size = 10)
fig2</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig2_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig2_combi-1">
Past versions of fig2_combi-1.png
</button>
</p>
<div id="fig-fig2_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig2,
  filename = &quot;output/figures/DLBCL_proteins/main_fig2.pdf&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig2,
  filename = &quot;output/figures/DLBCL_proteins/main_fig2.jpg&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>ID-seq confirms basal and activated BCR signaling in ABC- and
GCB-DLBCL.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig2&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9691631 517.6   16807764 897.7  16807764 897.7
Vcells 17114126 130.6  104698310 798.8 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-3" class="section level2">
<h2>Figure 3</h2>
<p><em>High doses of inhibitors drastically decrease many
phospho-protein levels.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li><p>Count data</p></li>
<li><p>DESeq2 analysis</p></li>
</ul></li>
</ul>
<div id="data-1" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig3_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)
fig3_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig3_cells &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
fig3_cells_clean &lt;- c(&quot;ABC subtype&quot;, &quot;GCB subtype&quot;)
fig3_cells_short &lt;- c(&quot;ABC&quot;, &quot;GCB&quot;)
fig3_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig3_stim_clean &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig3_inhib &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
fig3_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
fig3_conc_label &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
fig3_conc_all &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
# label_conc &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

fig3_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;cell_clean&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;cell_stim&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-2" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-2" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
fig3A_protein &lt;- c(&quot;BTK_Y551&quot;) # &quot;PYK2_Y402&quot; &quot;ATF2_T71&quot;
fig3A_protein_name &lt;- c(&quot;BTK (Y551)&quot;)
fig3A_cell_stim &lt;- c(&quot;Basal\nABC subtype&quot;, &quot;Activated\nABC subtype&quot;, &quot;Basal\nGCB subtype&quot;, &quot;Activated\nGCB subtype&quot;)
# fig3A_cell_stim_label &lt;- c(&quot;Basal\n             ABC subtype&quot;, &quot;Active&quot;, &quot;Basal\n             GCB subtype&quot;, &quot;Active&quot;)
fig3A_cell_stim_conc &lt;- c(&quot;0 uM\nBasal\nABC subtype&quot;, &quot;0 uM\nActivated\nABC subtype&quot;, 
                          &quot;100 uM\nBasal\nABC subtype&quot;, &quot;100 uM\nActivated\nABC subtype&quot;, 
                          &quot;0 uM\nBasal\nGCB subtype&quot;, &quot;0 uM\nActivated\nGCB subtype&quot;, 
                          &quot;100 uM\nBasal\nGCB subtype&quot;, &quot;100 uM\nActivated\nGCB subtype&quot;)
fig3A_cell_stim_conc_label &lt;- c(&quot;Bas.\n      0 uM&quot;, &quot;Act.&quot;, 
                                &quot;Bas.\n     100 uM&quot;, &quot;Act.&quot;, 
                                &quot;Bas.\n      0 uM&quot;, &quot;Act.&quot;, 
                                &quot;Bas.\n     100 uM&quot;, &quot;Act.&quot;)
fig3A_colors_stim &lt;- c(&quot;0 uM\nBasal\nABC subtype&quot; = &quot;#737373&quot;, &quot;0 uM\nActivated\nABC subtype&quot; = &quot;#FDB366&quot;, 
                          &quot;100 uM\nBasal\nABC subtype&quot; = &quot;#bdbdbd&quot;, &quot;100 uM\nActivated\nABC subtype&quot; = &quot;#FEDA8B&quot;, 
                          &quot;0 uM\nBasal\nGCB subtype&quot; = &quot;#737373&quot;, &quot;0 uM\nActivated\nGCB subtype&quot; = &quot;#FDB366&quot;, 
                          &quot;100 uM\nBasal\nGCB subtype&quot; = &quot;#bdbdbd&quot;, &quot;100 uM\nActivated\nGCB subtype&quot; = &quot;#FEDA8B&quot;)

fig3A_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3A_protein &amp; inhib_conc %in% fig3_conc) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig3A_protein), 
         cell_stim = factor(cell_stim, levels = fig3A_cell_stim), 
         cell_stim_conc = paste0(inhib_conc_uM, &quot; uM\n&quot;, cell_stim),
         inhibitor = factor(inhibitor, levels = fig3_inhib))
fig3A_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3A_protein &amp; inhib_conc %in% fig3_conc) %&gt;%
  # group_by(target_nospace, cell_stim) %&gt;%
  mutate(#counts_norm_avg = mean(counts_norm),
         target_nospace = factor(target_nospace, levels = fig3A_protein), 
         cell_stim = factor(cell_stim, levels = fig3A_cell_stim), 
         cell_stim_conc = paste0(inhib_conc_uM, &quot; uM\n&quot;, cell_stim),
         inhibitor = factor(inhibitor, levels = fig3_inhib))

# Statistics
# Kruskal-Wallis test
fig3A_kruskal &lt;- fig3A_data %&gt;%
  group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ cell_stim_conc)
fig3A_kruskal</code></pre>
<pre><code># A tibble: 1 × 7
  target_nospace .y.             n statistic    df        p method        
* &lt;fct&gt;          &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;         
1 BTK_Y551       counts_norm    72      63.5     7 3.03e-11 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
fig3A_dunn &lt;- fig3A_data %&gt;%
  group_by(inhibitor, cell_clean) %&gt;%
  dunn_test(counts_norm ~ cell_stim_conc, p.adjust.method = &quot;none&quot;) %&gt;%
  slice(-c(3, 4, 9, 10, 16, 17, 21, 22, 27, 28, 33, 34)) %&gt;%
  group_by(inhibitor, cell_clean) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  mutate(y.position = 3000, 
         inhib_cell = paste(inhibitor, cell_clean))
fig3A_dunn</code></pre>
<pre><code># A tibble: 24 × 13
   inhibitor cell_clean  .y.   group1 group2    n1    n2 statistic      p  p.adj
   &lt;fct&gt;     &lt;chr&gt;       &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 iBTK      ABC subtype coun… &quot;0 uM… &quot;0 uM…     3     3    -2.15  0.0315 0.108 
 2 iBTK      ABC subtype coun… &quot;0 uM… &quot;100 …     3     3    -1.02  0.308  0.411 
 3 iBTK      ABC subtype coun… &quot;0 uM… &quot;100 …     3     3    -0.793 0.428  0.428 
 4 iBTK      ABC subtype coun… &quot;100 … &quot;100 …     3     3    -1.92  0.0542 0.108 
 5 iBTK      GCB subtype coun… &quot;0 uM… &quot;0 uM…     3     3    -1.81  0.0700 0.140 
 6 iBTK      GCB subtype coun… &quot;0 uM… &quot;100 …     3     3    -1.25  0.213  0.213 
 7 iBTK      GCB subtype coun… &quot;0 uM… &quot;100 …     3     3    -1.25  0.213  0.213 
 8 iBTK      GCB subtype coun… &quot;100 … &quot;100 …     3     3    -1.81  0.0700 0.140 
 9 iSYK      ABC subtype coun… &quot;0 uM… &quot;0 uM…     3     3    -2.04  0.0415 0.0554
10 iSYK      ABC subtype coun… &quot;0 uM… &quot;100 …     3     3    -1.02  0.308  0.308 
# ℹ 14 more rows
# ℹ 3 more variables: p.adj.signif &lt;chr&gt;, y.position &lt;dbl&gt;, inhib_cell &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
fig3_panelA &lt;- ggplot() +
  geom_jitter(data = fig3A_data,
              aes(factor(cell_stim_conc, levels = fig3A_cell_stim_conc), counts_norm, color = factor(cell_stim_conc, levels = fig3A_cell_stim_conc)), 
              alpha = 1, size = 1, width = 0.15) +
  geom_errorbar(data = fig3A_mean,
                aes(factor(cell_stim_conc, levels = fig3A_cell_stim_conc),
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(cell_stim_conc, levels = fig3A_cell_stim_conc),
                    color = factor(cell_stim_conc, levels = fig3A_cell_stim_conc)),
                width = 0.95, size = 1) +
  # stat_pvalue_manual(fig3A_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.05,
  #                    step.increase = 0.22, group = &quot;inhib_cell&quot;, step.group.by = &quot;inhib_cell&quot;, size = 3, vjust = 1.2) +
  # facet_wrap(vars(factor(inhibitor, levels = fig3_inhib)), nrow = 1) +
  # facet_nested(~ inhibitor + cell_clean, scales = &quot;free_x&quot;) +
  facet_nested(, vars(factor(inhibitor), factor(cell_clean, labels = fig3_cells_short)), scales = &quot;free_x&quot;) +
  # facet_grid(target_nospace~inhibitor, scales = &quot;free_y&quot;) +
  # scale_color_manual(values = colors_stim, breaks = fig3_stim_clean, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = fig3A_colors_stim) +
  scale_x_discrete(breaks = fig3A_cell_stim_conc, labels = fig3A_cell_stim_conc_label) +
  labs(x = element_blank(), y = &quot;Normalised counts&quot;, title = fig3A_protein_name) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.position = &quot;none&quot;, legend.justification = &quot;right&quot;, legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.margin = margin(c(0, 0, 0, 0)), 
        axis.text.x = element_text(size = 6), plot.title = element_text(size = 8.5), plot.title.position = &quot;panel&quot;)
fig3_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_panelA-1">
Past versions of fig3_panelA-1.png
</button>
</p>
<div id="fig-fig3_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-2" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
fig3B_protein &lt;- c(&quot;PYK2_Y402&quot;) # &quot;PYK2_Y402&quot; &quot;ATF2_T71&quot;
fig3B_protein_name &lt;- c(&quot;PYK2 (Y402)&quot;)
fig3B_cell_stim &lt;- c(&quot;Basal\nABC subtype&quot;, &quot;Activated\nABC subtype&quot;, &quot;Basal\nGCB subtype&quot;, &quot;Activated\nGCB subtype&quot;)
# fig3B_cell_stim_label &lt;- c(&quot;Basal\n             ABC subtype&quot;, &quot;Active&quot;, &quot;Basal\n             GCB subtype&quot;, &quot;Active&quot;)
fig3B_cell_stim_conc &lt;- c(&quot;0 uM\nBasal\nABC subtype&quot;, &quot;0 uM\nActivated\nABC subtype&quot;, 
                          &quot;100 uM\nBasal\nABC subtype&quot;, &quot;100 uM\nActivated\nABC subtype&quot;, 
                          &quot;0 uM\nBasal\nGCB subtype&quot;, &quot;0 uM\nActivated\nGCB subtype&quot;, 
                          &quot;100 uM\nBasal\nGCB subtype&quot;, &quot;100 uM\nActivated\nGCB subtype&quot;)
fig3B_cell_stim_conc_label &lt;- c(&quot;Bas.\n      0 uM&quot;, &quot;Act.&quot;, 
                                &quot;Bas.\n     100 uM&quot;, &quot;Act.&quot;, 
                                &quot;Bas.\n      0 uM&quot;, &quot;Act.&quot;, 
                                &quot;Bas.\n     100 uM&quot;, &quot;Act.&quot;)

fig3B_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3B_protein &amp; inhib_conc %in% fig3_conc) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig3B_protein), 
         cell_stim = factor(cell_stim, levels = fig3B_cell_stim), 
         cell_stim_conc = paste0(inhib_conc_uM, &quot; uM\n&quot;, cell_stim),
         inhibitor = factor(inhibitor, levels = fig3_inhib))
fig3B_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3B_protein &amp; inhib_conc %in% fig3_conc) %&gt;%
  # group_by(target_nospace, cell_stim) %&gt;%
  mutate(#counts_norm_avg = mean(counts_norm),
         target_nospace = factor(target_nospace, levels = fig3B_protein), 
         cell_stim = factor(cell_stim, levels = fig3B_cell_stim), 
         cell_stim_conc = paste0(inhib_conc_uM, &quot; uM\n&quot;, cell_stim),
         inhibitor = factor(inhibitor, levels = fig3_inhib))

# Statistics
# Kruskal-Wallis test
fig3B_kruskal &lt;- fig3B_data %&gt;%
  group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ cell_stim_conc)
fig3B_kruskal</code></pre>
<pre><code># A tibble: 1 × 7
  target_nospace .y.             n statistic    df             p method        
* &lt;fct&gt;          &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;         &lt;dbl&gt; &lt;chr&gt;         
1 PYK2_Y402      counts_norm    72      54.5     7 0.00000000189 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
fig3B_dunn &lt;- fig3B_data %&gt;%
  group_by(inhibitor, cell_clean) %&gt;%
  dunn_test(counts_norm ~ cell_stim_conc, p.adjust.method = &quot;none&quot;) %&gt;%
  slice(-c(3, 4, 9, 10, 16, 17, 21, 22, 27, 28, 33, 34)) %&gt;%
  group_by(inhibitor, cell_clean) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  mutate(y.position = 3000, 
         inhib_cell = paste(inhibitor, cell_clean))
fig3B_dunn</code></pre>
<pre><code># A tibble: 24 × 13
   inhibitor cell_clean .y.   group1 group2    n1    n2 statistic       p  p.adj
   &lt;fct&gt;     &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;
 1 iBTK      ABC subty… coun… &quot;0 uM… &quot;0 uM…     3     3    -2.60  0.00921 0.0368
 2 iBTK      ABC subty… coun… &quot;0 uM… &quot;100 …     3     3    -1.02  0.308   0.411 
 3 iBTK      ABC subty… coun… &quot;0 uM… &quot;100 …     3     3     0.113 0.910   0.910 
 4 iBTK      ABC subty… coun… &quot;100 … &quot;100 …     3     3    -1.47  0.141   0.282 
 5 iBTK      GCB subty… coun… &quot;0 uM… &quot;0 uM…     3     3    -2.49  0.0127  0.0510
 6 iBTK      GCB subty… coun… &quot;0 uM… &quot;100 …     3     3    -2.04  0.0415  0.0831
 7 iBTK      GCB subty… coun… &quot;0 uM… &quot;100 …     3     3     0.906 0.365   0.487 
 8 iBTK      GCB subty… coun… &quot;100 … &quot;100 …     3     3     0.453 0.651   0.651 
 9 iSYK      ABC subty… coun… &quot;0 uM… &quot;0 uM…     3     3    -2.49  0.0127  0.0255
10 iSYK      ABC subty… coun… &quot;0 uM… &quot;100 …     3     3    -1.02  0.308   0.308 
# ℹ 14 more rows
# ℹ 3 more variables: p.adj.signif &lt;chr&gt;, y.position &lt;dbl&gt;, inhib_cell &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
fig3_panelB &lt;- ggplot() +
  geom_jitter(data = fig3B_data,
              aes(factor(cell_stim_conc, levels = fig3B_cell_stim_conc), counts_norm, color = factor(cell_stim_conc, levels = fig3A_cell_stim_conc)), 
              alpha = 1, size = 1, width = 0.15) +
  geom_errorbar(data = fig3B_mean,
                aes(factor(cell_stim_conc, levels = fig3B_cell_stim_conc),
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(cell_stim_conc, levels = fig3B_cell_stim_conc),
                    color = factor(cell_stim_conc, levels = fig3A_cell_stim_conc)),
                width = 0.95, size = 1) +
  # stat_pvalue_manual(fig3B_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.05,
  #                    step.increase = 0.22, group = &quot;inhib_cell&quot;, step.group.by = &quot;inhib_cell&quot;, size = 3, vjust = 1.2) +
  facet_nested(, vars(factor(inhibitor), factor(cell_clean, labels = fig3_cells_short)), scales = &quot;free_x&quot;) +
  scale_color_manual(values = fig3A_colors_stim) +
  scale_x_discrete(breaks = fig3B_cell_stim_conc, labels = fig3B_cell_stim_conc_label) +
  labs(x = element_blank(), y = &quot;Normalised counts&quot;, title = fig3B_protein_name) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.position = &quot;none&quot;, legend.justification = &quot;right&quot;, legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.margin = margin(c(0, 0, 0, 0)), 
        axis.text.x = element_text(size = 6), plot.title = element_text(size = 8.5), plot.title.position = &quot;panel&quot;)
fig3_panelB</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_panelB-1">
Past versions of fig3_panelB-1.png
</button>
</p>
<div id="fig-fig3_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-1" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data
fig3C_protein &lt;- c(&quot;ATF2_T71&quot;) # &quot;PYK2_Y402&quot; &quot;ATF2_T71&quot;
fig3C_protein_name &lt;- c(&quot;ATF2 (T71)&quot;)
fig3C_cell_stim &lt;- c(&quot;Basal\nABC subtype&quot;, &quot;Activated\nABC subtype&quot;, &quot;Basal\nGCB subtype&quot;, &quot;Activated\nGCB subtype&quot;)
# fig3C_cell_stim_label &lt;- c(&quot;Basal\n             ABC subtype&quot;, &quot;Active&quot;, &quot;Basal\n             GCB subtype&quot;, &quot;Active&quot;)
fig3C_cell_stim_conc &lt;- c(&quot;0 uM\nBasal\nABC subtype&quot;, &quot;0 uM\nActivated\nABC subtype&quot;, 
                          &quot;100 uM\nBasal\nABC subtype&quot;, &quot;100 uM\nActivated\nABC subtype&quot;, 
                          &quot;0 uM\nBasal\nGCB subtype&quot;, &quot;0 uM\nActivated\nGCB subtype&quot;, 
                          &quot;100 uM\nBasal\nGCB subtype&quot;, &quot;100 uM\nActivated\nGCB subtype&quot;)
fig3C_cell_stim_conc_label &lt;- c(&quot;Bas.\n      0 uM&quot;, &quot;Act.&quot;,
                                &quot;Bas.\n     100 uM&quot;, &quot;Act.&quot;,
                                &quot;Bas.\n      0 uM&quot;, &quot;Act.&quot;,
                                &quot;Bas.\n     100 uM&quot;, &quot;Act.&quot;)

fig3C_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3C_protein &amp; inhib_conc %in% fig3_conc) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig3C_protein), 
         cell_stim = factor(cell_stim, levels = fig3C_cell_stim), 
         cell_stim_conc = paste0(inhib_conc_uM, &quot; uM\n&quot;, cell_stim),
         inhibitor = factor(inhibitor, levels = fig3_inhib))
fig3C_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3C_protein &amp; inhib_conc %in% fig3_conc) %&gt;%
  # group_by(target_nospace, cell_stim) %&gt;%
  mutate(#counts_norm_avg = mean(counts_norm),
         target_nospace = factor(target_nospace, levels = fig3C_protein), 
         cell_stim = factor(cell_stim, levels = fig3C_cell_stim), 
         cell_stim_conc = paste0(inhib_conc_uM, &quot; uM\n&quot;, cell_stim),
         inhibitor = factor(inhibitor, levels = fig3_inhib))

# Statistics
# Kruskal-Wallis test
fig3C_kruskal &lt;- fig3C_data %&gt;%
  group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ cell_stim_conc)
fig3C_kruskal</code></pre>
<pre><code># A tibble: 1 × 7
  target_nospace .y.             n statistic    df            p method        
* &lt;fct&gt;          &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;        &lt;dbl&gt; &lt;chr&gt;         
1 ATF2_T71       counts_norm    72      46.0     7 0.0000000867 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
fig3C_dunn &lt;- fig3C_data %&gt;%
  group_by(inhibitor, cell_clean) %&gt;%
  dunn_test(counts_norm ~ cell_stim_conc, p.adjust.method = &quot;none&quot;) %&gt;%
  slice(-c(3, 4, 9, 10, 16, 17, 21, 22, 27, 28, 33, 34)) %&gt;%
  group_by(inhibitor, cell_clean) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  mutate(y.position = 3000, 
         inhib_cell = paste(inhibitor, cell_clean))
fig3C_dunn</code></pre>
<pre><code># A tibble: 24 × 13
   inhibitor cell_clean .y.   group1 group2    n1    n2 statistic       p  p.adj
   &lt;fct&gt;     &lt;chr&gt;      &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;
 1 iBTK      ABC subty… coun… &quot;0 uM… &quot;0 uM…     3     3    -0.566 0.571   0.571 
 2 iBTK      ABC subty… coun… &quot;0 uM… &quot;100 …     3     3    -2.60  0.00921 0.0368
 3 iBTK      ABC subty… coun… &quot;0 uM… &quot;100 …     3     3    -1.47  0.141   0.282 
 4 iBTK      ABC subty… coun… &quot;100 … &quot;100 …     3     3     0.566 0.571   0.571 
 5 iBTK      GCB subty… coun… &quot;0 uM… &quot;0 uM…     3     3    -1.02  0.308   0.411 
 6 iBTK      GCB subty… coun… &quot;0 uM… &quot;100 …     3     3    -2.72  0.00658 0.0263
 7 iBTK      GCB subty… coun… &quot;0 uM… &quot;100 …     3     3    -1.36  0.174   0.348 
 8 iBTK      GCB subty… coun… &quot;100 … &quot;100 …     3     3     0.340 0.734   0.734 
 9 iSYK      ABC subty… coun… &quot;0 uM… &quot;0 uM…     3     3    -2.38  0.0174  0.0697
10 iSYK      ABC subty… coun… &quot;0 uM… &quot;100 …     3     3    -0.113 0.910   0.910 
# ℹ 14 more rows
# ℹ 3 more variables: p.adj.signif &lt;chr&gt;, y.position &lt;dbl&gt;, inhib_cell &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
fig3_panelC &lt;- ggplot() +
  geom_jitter(data = fig3C_data,
              aes(factor(cell_stim_conc, levels = fig3C_cell_stim_conc), counts_norm, color = factor(cell_stim_conc, levels = fig3A_cell_stim_conc)), 
              alpha = 1, size = 1, width = 0.15) +
  geom_errorbar(data = fig3C_mean,
                aes(factor(cell_stim_conc, levels = fig3C_cell_stim_conc),
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(cell_stim_conc, levels = fig3C_cell_stim_conc),
                    color = factor(cell_stim_conc, levels = fig3A_cell_stim_conc)),
                width = 0.95, size = 1) +
  # stat_pvalue_manual(fig3C_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.05,
  #                    step.increase = 0.22, group = &quot;inhib_cell&quot;, step.group.by = &quot;inhib_cell&quot;, size = 3, vjust = 1.2) +
  facet_nested(, vars(factor(inhibitor), factor(cell_clean, labels = fig3_cells_short)), scales = &quot;free_x&quot;) +
  scale_color_manual(values = fig3A_colors_stim, breaks = c(&quot;0 uM\nBasal\nABC subtype&quot;, &quot;0 uM\nActivated\nABC subtype&quot;), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  # scale_color_manual(values = colors_stim, breaks = fig3_stim_clean, name = &quot;Stimulus&quot;) +
  scale_x_discrete(breaks = fig3C_cell_stim_conc, labels = fig3C_cell_stim_conc_label) +
  labs(x = element_blank(), y = &quot;Normalised counts&quot;, title = fig3C_protein_name) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.position = &quot;bottom&quot;, legend.justification = &quot;right&quot;, legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.margin = margin(c(0, 0, 0, 0)), 
        axis.text.x = element_text(size = 6), plot.title = element_text(size = 8.5), plot.title.position = &quot;panel&quot;)
fig3_panelC</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_panelC-1">
Past versions of fig3_panelC-1.png
</button>
</p>
<div id="fig-fig3_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d-1" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Select data
# Protein lists
# HBL1
fig3D_HBL1_iBTK &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/HBL1_iBTK_stim_sign_proteins.txt&quot;)
fig3D_HBL1_iSYK &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/HBL1_iSYK_stim_sign_proteins.txt&quot;)
fig3D_HBL1_iNFkB &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/HBL1_iNFkB_stim_sign_proteins.txt&quot;)
# any dose inhib effect on act vs basal: HBL1_iBTK_stim_sign_proteins.txt

# OCI-Ly8
fig3D_OCI_iBTK &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/Ly8_iBTK_stim_sign_proteins.txt&quot;)
fig3D_OCI_iSYK &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/Ly8_iSYK_stim_sign_proteins.txt&quot;)
fig3D_OCI_iNFkB &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/Ly8_iNFkB_stim_sign_proteins.txt&quot;)
# any dose inhib effect on act vs basal: Ly8_iBTK_stim_sign_proteins.txt

# Figure
# HBL1
fig3D_HBL1_data &lt;- list(fig3D_HBL1_iBTK, fig3D_HBL1_iSYK, fig3D_HBL1_iNFkB)
fig3D_HBL1_names &lt;- c(paste0(&quot;iBTK (&quot;, length(fig3D_HBL1_iBTK), &quot; prot.)&quot;), 
                     paste0(&quot;iSYK (&quot;, length(fig3D_HBL1_iSYK), &quot; prot.)&quot;), 
                     paste0(&quot;iNFkB (&quot;, length(fig3D_HBL1_iNFkB), &quot; prot.)&quot;))
names(fig3D_HBL1_data) &lt;- fig3D_HBL1_names
fig3_panelD_HBL1 &lt;- ggvenn(data = fig3D_HBL1_data, 
               fill_color = c(&quot;#5c9dce&quot;, &quot;#9dce5c&quot;, &quot;#ce5c9d&quot;),
               # auto_scale = TRUE,
               show_percentage = FALSE,
               # show_elements = FALSE,
               stroke_size = 0.35, 
               text_size = 2.5, 
               set_name_size = 2.5) + 
  ggtitle(&quot;   Effect of inhibitors on\n   activated vs basal signaling\n\n   ABC subtype&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8.5))
fig3_panelD_HBL1</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_panelD-1">
Past versions of fig3_panelD-1.png
</button>
</p>
<div id="fig-fig3_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># OCI-Ly8
fig3D_OCI_data &lt;- list(fig3D_OCI_iBTK, fig3D_OCI_iSYK, fig3D_OCI_iNFkB)
fig3D_OCI_names &lt;- c(paste0(&quot;iBTK (&quot;, length(fig3D_OCI_iBTK), &quot; prot.)&quot;), 
                     paste0(&quot;iSYK (&quot;, length(fig3D_OCI_iSYK), &quot; prot.)&quot;), 
                     paste0(&quot;iNFkB (&quot;, length(fig3D_OCI_iNFkB), &quot; prot.)&quot;))
names(fig3D_OCI_data) &lt;- fig3D_OCI_names
fig3_panelD_OCI &lt;- ggvenn(data = fig3D_OCI_data, 
               fill_color = c(&quot;#5c9dce&quot;, &quot;#9dce5c&quot;, &quot;#ce5c9d&quot;),
               # auto_scale = TRUE,
               show_percentage = FALSE,
               # show_elements = TRUE,
               stroke_size = 0.5, 
               text_size = 2.5, 
               set_name_size = 2.5) + 
  ggtitle(&quot;\n   GCB subtype&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8.5))
fig3_panelD_OCI</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_panelD-2.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_panelD-2">
Past versions of fig3_panelD-2.png
</button>
</p>
<div id="fig-fig3_panelD-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fig3_panelD &lt;- plot_grid(fig3_panelD_HBL1, fig3_panelD_OCI, ncol = 1, rel_heights = c(1, 1))
fig3_panelD</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_panelD-3.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_panelD-3">
Past versions of fig3_panelD-3.png
</button>
</p>
<div id="fig-fig3_panelD-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fig3_panelD_alt &lt;- plot_grid(fig3_panelD_HBL1, fig3_panelD_OCI, nrow = 1, rel_widths = c(1, 1))
# fig3_panelD_alt</code></pre>
</div>
</div>
<div id="combined-fig-2" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
# fig3 &lt;- plot_grid(fig3_panelA, fig3_panelB, fig3_panelC, fig3_panelD_alt, labels = PANEL_labels[c(1:4)], ncol = 1, rel_heights = c(1, 1, 1.1, 1.2), label_size = 10)
fig3_ABC &lt;- plot_grid(fig3_panelA, fig3_panelB, fig3_panelC, labels = PANEL_labels[c(1:3)], ncol = 1, rel_heights = c(1, 1, 1.1), label_size = 10)
fig3 &lt;- plot_grid(fig3_ABC, fig3_panelD, labels = c(NA, PANEL_labels[c(4)]), nrow = 1, rel_widths = c(2.9, 1), label_size = 10)
fig3</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig3_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig3_combi-1">
Past versions of fig3_combi-1.png
</button>
</p>
<div id="fig-fig3_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig3,
  filename = &quot;output/figures/DLBCL_proteins/main_fig3.pdf&quot;,
  width = 6,
  height = 5.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig3,
  filename = &quot;output/figures/DLBCL_proteins/main_fig3.jpg&quot;,
  width = 6,
  height = 5.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>High doses of inhibitors drastically decrease many
phospho-protein levels.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig3&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9752038 520.9   16807764 897.7  16807764 897.7
Vcells 17228477 131.5   83758648 639.1 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<p><em>High doses of inhibitors cause network-wide disruption in the
activated signaling response.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li>Network projection</li>
</ul></li>
</ul>
<div id="data-2" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig4_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)
fig4_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)

fig4_node_info &lt;- read_excel(&quot;output/network_visual_DLBCL/node_info2.xlsx&quot;)</code></pre>
<pre class="r"><code>fig4_cells &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
fig4_cells_clean &lt;- c(&quot;ABC subtype&quot;, &quot;GCB subtype&quot;)
fig4_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig4_stim_clean &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig4_inhib &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
fig4_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
fig4_conc_label &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
fig4_conc_all &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
# label_conc &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

fig4_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;cell_clean&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;cell_stim&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="diff-express-analysis" class="section level3 tabset">
<h3 class="tabset">Diff express analysis</h3>
<div id="activated-vs-basal-pbs-ctrl" class="section level4">
<h4>Activated vs basal (PBS ctrl)</h4>
<p>Differential expression analysis of activated vs basal signaling</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>2 cell lines, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: cell_stim</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig4_data_id  %&gt;%
  dplyr::filter(inhib_new == &quot;PBS ctrl&quot; &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig4_data_id, all_of(fig4_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ cell_stim

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_stim &lt;- relevel(dds$cell_stim, ref = &quot;Basal\nABC subtype&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_HBL1_stim &lt;- results(dds, name = &quot;cell_stim_Activated.ABC.subtype_vs_Basal.ABC.subtype&quot;, alpha = 0.05)
results_OCI_stim &lt;- results(dds, contrast = c(&quot;cell_stim&quot;, &quot;Activated\nGCB subtype&quot;, &quot;Basal\nGCB subtype&quot;), alpha = 0.05)

list_results &lt;- list(results_HBL1_stim, results_OCI_stim)

# Data transformation
# Regularized log transform
rld_stim &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
volc_HBL1_stim &lt;- prep_forvulcano(dataset = results_HBL1_stim, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_HBL1_StimvsPBS_log2FC = log2FoldChange, 
                prot_HBL1_StimvsPBS_padj = padj,
                prot_HBL1_StimvsPBS_log10padj = log10padj, 
                prot_HBL1_StimvsPBS_DE = diff_express)

volc_OCI_stim &lt;- prep_forvulcano(dataset = results_OCI_stim, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_OCI_StimvsPBS_log2FC = log2FoldChange, 
                prot_OCI_StimvsPBS_padj = padj,
                prot_OCI_StimvsPBS_log10padj = log10padj, 
                prot_OCI_StimvsPBS_DE = diff_express)</code></pre>
</div>
<div id="hbl1-100-um-inhib-vs-dmso-ctrl-activated"
class="section level4">
<h4>HBL1: 100 uM inhib vs DMSO ctrl (activated)</h4>
<p>Differential expression analysis of stimulation in HBL1</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>3 inhibitors, 0 uM (DMSO) + 100 uM inhib, aIg+H2O2, HBL1</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig4_data_id  %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; stimulus == &quot;aIg+H2O2&quot; &amp; inhib_conc %in% c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;) &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig4_data_id, all_of(fig4_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_HBL1_iSYK100 &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_HBL1_iBTK100 &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_HBL1_iNFkB100 &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
# list_results &lt;- list(results_iSYK100_stim, results_iBTK100_stim, results_iNFkB100_stim)
# summary(results_HBL1, padj = 0.05)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Volcano preparation/data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
volc_HBL1_iSYK100 &lt;- prep_forvulcano(dataset = results_HBL1_iSYK100, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                               .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                               .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_HBL1_iSYK100vsDMSO_act_log2FC = log2FoldChange, 
                prot_HBL1_iSYK100vsDMSO_act_padj = padj,
                prot_HBL1_iSYK100vsDMSO_act_log10padj = log10padj, 
                prot_HBL1_iSYK100vsDMSO_act_DE = diff_express)
# write.csv(volc_HBL1_iSYK100, file = &quot;output/network_visual_paper/node_info_HBL1_iSYK100uMvsDMSOctrl_StimOnly.csv&quot;, row.names = F, na = &quot;&quot;)

volc_HBL1_iBTK100 &lt;- prep_forvulcano(dataset = results_HBL1_iBTK100, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                               .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                               .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_HBL1_iBTK100vsDMSO_act_log2FC = log2FoldChange, 
                prot_HBL1_iBTK100vsDMSO_act_padj = padj,
                prot_HBL1_iBTK100vsDMSO_act_log10padj = log10padj, 
                prot_HBL1_iBTK100vsDMSO_act_DE = diff_express)
# write.csv(volc_HBL1_iBTK100, file = &quot;output/network_visual_paper/node_info_HBL1_iBTK100uMvsDMSOctrl_StimOnly.csv&quot;, row.names = F, na = &quot;&quot;)

volc_HBL1_iNFkB100 &lt;- prep_forvulcano(dataset = results_HBL1_iNFkB100, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                               .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                               .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_HBL1_iNFkB100vsDMSO_act_log2FC = log2FoldChange, 
                prot_HBL1_iNFkB100vsDMSO_act_padj = padj,
                prot_HBL1_iNFkB100vsDMSO_act_log10padj = log10padj, 
                prot_HBL1_iNFkB100vsDMSO_act_DE = diff_express)
# write.csv(volc_HBL1_iNFkB100, file = &quot;output/network_visual_paper/node_info_HBL1_iNFkB100uMvsDMSOctrl_StimOnly.csv&quot;, row.names = F, na = &quot;&quot;)</code></pre>
</div>
<div id="oci-100-um-inhib-vs-dmso-ctrl-activated"
class="section level4">
<h4>OCI: 100 uM inhib vs DMSO ctrl (activated)</h4>
<p>Differential expression analysis of stimulation in OCI</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>3 inhibitors, 0 uM (DMSO) + 100 uM inhib, aIg+H2O2, OCI</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig4_data_id  %&gt;%
  dplyr::filter(cell_line == &quot;OCI-Ly8&quot; &amp; stimulus == &quot;aIg+H2O2&quot; &amp; inhib_conc %in% c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;) &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig4_data_id, all_of(fig4_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_OCI_iSYK100 &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_OCI_iBTK100 &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_OCI_iNFkB100 &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
# list_results &lt;- list(results_iSYK100_stim, results_iBTK100_stim, results_iNFkB100_stim)
# summary(results_OCI, padj = 0.05)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Volcano preparation/data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
volc_OCI_iSYK100 &lt;- prep_forvulcano(dataset = results_OCI_iSYK100, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                               .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                               .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_OCI_iSYK100vsDMSO_act_log2FC = log2FoldChange, 
                prot_OCI_iSYK100vsDMSO_act_padj = padj,
                prot_OCI_iSYK100vsDMSO_act_log10padj = log10padj, 
                prot_OCI_iSYK100vsDMSO_act_DE = diff_express)
# write.csv(volc_OCI_iSYK100, file = &quot;output/network_visual_paper/node_info_OCI_iSYK100uMvsDMSOctrl_StimOnly.csv&quot;, row.names = F, na = &quot;&quot;)

volc_OCI_iBTK100 &lt;- prep_forvulcano(dataset = results_OCI_iBTK100, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                               .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                               .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_OCI_iBTK100vsDMSO_act_log2FC = log2FoldChange, 
                prot_OCI_iBTK100vsDMSO_act_padj = padj,
                prot_OCI_iBTK100vsDMSO_act_log10padj = log10padj, 
                prot_OCI_iBTK100vsDMSO_act_DE = diff_express)
# write.csv(volc_OCI_iBTK100, file = &quot;output/network_visual_paper/node_info_OCI_iBTK100uMvsDMSOctrl_StimOnly.csv&quot;, row.names = F, na = &quot;&quot;)

volc_OCI_iNFkB100 &lt;- prep_forvulcano(dataset = results_OCI_iNFkB100, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  mutate(log10padj = -log10(padj)) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, log10padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                               .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                               .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(prot_OCI_iNFkB100vsDMSO_act_log2FC = log2FoldChange, 
                prot_OCI_iNFkB100vsDMSO_act_padj = padj,
                prot_OCI_iNFkB100vsDMSO_act_log10padj = log10padj, 
                prot_OCI_iNFkB100vsDMSO_act_DE = diff_express)
# write.csv(volc_OCI_iNFkB100, file = &quot;output/network_visual_paper/node_info_OCI_iNFkB100uMvsDMSOctrl_StimOnly.csv&quot;, row.names = F, na = &quot;&quot;)</code></pre>
</div>
<div id="combined-dataset" class="section level4">
<h4>Combined dataset</h4>
<pre class="r"><code>fig4_DESeq2_node_info &lt;- list(volc_HBL1_stim, volc_HBL1_iBTK100, volc_HBL1_iSYK100, volc_HBL1_iNFkB100, 
                              volc_OCI_stim, volc_OCI_iBTK100, volc_OCI_iSYK100, volc_OCI_iNFkB100) %&gt;%
  purrr::reduce(full_join)
write.csv(fig4_DESeq2_node_info, file = &quot;output/network_visual_DLBCL/data_node_info_proteins_all.csv&quot;, row.names = F, na = &quot;&quot;)</code></pre>
</div>
</div>
<div id="panels-3" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-3" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
fig4_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig4_panelA

# HBL1: Active
fig4_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_prot_HBL1_ActvsBas.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;    ABC subtype: Activated vs basal&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
fig4_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig4_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig4_panelA-1">
Past versions of fig4_panelA-1.png
</button>
</p>
<div id="fig-fig4_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># legend
fig4_network_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections.png&quot;) %&gt;%
# fig4_network_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections_vert.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  # ggtitle(&quot;\n&quot;) +
  textsize_small
# fig4_network_legend</code></pre>
</div>
<div id="panel-b-3" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
fig4_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig4_panelB

# HBL1: 100 uM iBTK
fig4_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_prot_HBL1_iBTK100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;    ABC subtype: 100 uM iBTK in activated cells&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
fig4_panelB</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig4_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig4_panelB-1">
Past versions of fig4_panelB-1.png
</button>
</p>
<div id="fig-fig4_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-2" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Placeholder
fig4_panelC &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig4_panelC

# OCI: Active
fig4_panelC &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_prot_OCI_ActvsBas.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;    GCB subtype: Activated vs basal&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
fig4_panelC</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig4_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig4_panelC-1">
Past versions of fig4_panelC-1.png
</button>
</p>
<div id="fig-fig4_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d-2" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Placeholder
fig4_panelD &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig4_panelD

# OCI: 100 uM iBTK
fig4_panelD &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_prot_OCI_iBTK100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;    GCB subtype: 100 uM iBTK in activated cells&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
fig4_panelD</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig4_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig4_panelD-1">
Past versions of fig4_panelD-1.png
</button>
</p>
<div id="fig-fig4_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-3" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig4_networks &lt;- plot_grid(fig4_panelA, fig4_panelB, fig4_panelC, fig4_panelD, labels = PANEL_labels[c(1:4)], ncol = 2, rel_widths = c(1, 1), rel_heights = c(1, 1), label_size = 10)
fig4 &lt;- plot_grid(fig4_networks, fig4_network_legend, ncol = 1, rel_heights = c(10, 1))
fig4</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig4_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig4_combi-1">
Past versions of fig4_combi-1.png
</button>
</p>
<div id="fig-fig4_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig4,
  filename = &quot;output/figures/DLBCL_proteins/main_fig4.pdf&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig4,
  filename = &quot;output/figures/DLBCL_proteins/main_fig4.jpg&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>High doses of inhibitors cause network-wide disruption in the
activated signaling response.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig4&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9745036 520.5   16807764 897.7  16807764 897.7
Vcells 21078526 160.9   67006919 511.3 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-5" class="section level2">
<h2>Figure 5</h2>
<p><em>Inhibition of core BCR signaling and MAPK signaling in ABC-DLBCL
subtype model.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li><p>Count data</p></li>
<li><p>DESeq2 analysis</p></li>
</ul></li>
</ul>
<div id="data-3" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig5_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)
fig5_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig5_cells &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
fig5_cells_clean &lt;- c(&quot;ABC subtype&quot;, &quot;GCB subtype&quot;)
fig5_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig5_stim_clean &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig5_inhib &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
# fig5_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
# fig5_conc_label &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
fig5_conc_all &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig5_conc_all_label &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig5_conc_all_label_alt &lt;- c(&quot;0&quot;, &quot;0.1&quot;, &quot;1&quot;, &quot;10&quot;, &quot;100&quot;)

fig5_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;cell_clean&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;cell_stim&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;inhib_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-4" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-4" class="section level4">
<h4>Panel A</h4>
<p>Differential expression analysis of the effect of inhibitor
concentration on the stimulation in HBL1</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1: 3 inhibitors, 5 conc, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig5_data_id  %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; inhib_conc %in% fig5_conc_all &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig5_data_id, all_of(fig5_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new + stim_clean + inhib_text_new:stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_HBL1_stim &lt;- results(dds, name = &quot;stim_clean_Activated_vs_Basal&quot;, alpha = 0.05)
results_HBL1_iBTK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)
results_HBL1_iSYK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iSYK.stim_cleanActivated&quot;, alpha = 0.05)
results_HBL1_iNFkB100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iNFkB.stim_cleanActivated&quot;, alpha = 0.05)

list_results &lt;- list(results_HBL1_stim, results_HBL1_iBTK100, results_HBL1_iSYK100, results_HBL1_iNFkB100)

# Data transformation
# Regularized log transform
rld_HBL1 &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_volc &lt;- c(&quot;volc_HBL1_stim&quot;, &quot;volc_HBL1_iBTK100&quot;, &quot;volc_HBL1_iSYK100&quot;, &quot;volc_HBL1_iNFkB100&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_HBL1_stim, volc_HBL1_iBTK100, volc_HBL1_iSYK100, volc_HBL1_iNFkB100)
names_heat &lt;- c(&quot;heat_HBL1_stim&quot;, &quot;heat_HBL1_iBTK100&quot;, &quot;heat_HBL1_iSYK100&quot;, &quot;heat_HBL1_iNFkB100&quot;)
for(dataset in c(1:length(list_results_volc))){
  assign(names_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}</code></pre>
<pre class="r"><code># Data
# Combine heatmap datasets (without DESeq2 analysis values)
heat_inhib_HBL1 &lt;- list(heat_HBL1_stim, heat_HBL1_iBTK100, heat_HBL1_iSYK100, heat_HBL1_iNFkB100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  left_join(fig5_data_id[, 14:21]) %&gt;%
  dplyr::filter(modification == &quot;phospho&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
fig5A_info_HBL1 &lt;- fig5_data_id %&gt;%
  dplyr::select(plate_well, stim_clean, inhib_conc, inhibitor, cell_line) %&gt;%
  distinct() %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; inhib_conc %in% fig5_conc_all &amp; !(stim_clean == &quot;Basal&quot; &amp; inhib_conc %in% fig5_conc_all[2:5])) %&gt;%
  mutate(inhibitor = factor(inhibitor, levels = fig5_inhib)) %&gt;%
  as.data.frame()
rownames(fig5A_info_HBL1) &lt;- fig5A_info_HBL1$plate_well
fig5A_info_HBL1 &lt;- dplyr::select(fig5A_info_HBL1, -c(plate_well, cell_line))

# Get the rld transformed data for the heatmap and scale per row
fig5A_data_heat_HBL1 &lt;- t(scale(t(assay(rld_HBL1)[c(heat_inhib_HBL1$target_nospace), rownames(fig5A_info_HBL1)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

# Figure
fig5A_ann_colors &lt;- list(Stimulation = colors_stim, 
                         Inhibitor = colors_inhib,
                         # inhib_new = c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib), 
                         &quot;Inhibitor conc&quot; = setNames(colors_conc, unique(fig5A_info_HBL1$inhib_conc)))
fig5A_heat_legend &lt;- list(title = &quot;Scaled\nexpression&quot;, title_gp = gpar(fontsize = 7), labels_gp = gpar(fontsize = 7), 
                          grid_width = unit(2, &quot;mm&quot;), grid_height = unit(20, &quot;mm&quot;))

fig5A_col_ann &lt;- HeatmapAnnotation(Inhibitor = fig5A_info_HBL1$inhibitor, 
                                   &quot;Inhibitor conc&quot; = fig5A_info_HBL1$inhib_conc, 
                                   Stimulation = fig5A_info_HBL1$stim_clean, 
                                   simple_anno_size = unit(2.5, &quot;mm&quot;), 
                                   annotation_name_gp = gpar(fontsize = 7),
                                   col = fig5A_ann_colors, 
                                   annotation_legend_param = list(Inhibitor = list(title = &quot;Inhibitor&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  &quot;Inhibitor conc&quot; = list(title = &quot;Inhibitor\nconcentration&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  Stimulation = list(title = &quot;Stimulation&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;))), 
                                   show_legend = c(FALSE, TRUE, TRUE))

fig5_panelA &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  fig5A_data_heat_HBL1,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  # treeheight_row = 20,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_col = fig5A_info_Ly8,
  top_annotation = fig5A_col_ann,
  # annotation_colors = fig5A_ann_colors,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = fig5A_info_HBL1$inhibitor,
  row_split = subset(heat_inhib_HBL1)$modification,
  row_gap = unit(1, &quot;mm&quot;), 
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 5.75,
  row_title_gp = gpar(fontsize = 7),
  column_title_gp = gpar(fontsize = 7), 
  annotation_names_col = gpar(fontsize = 7),
  border_color = NA,
  heatmap_legend_param = fig5A_heat_legend, 
  # annotation_legend = FALSE,
)) +
  ggtitle(&quot;          ABC subtype: Effect of inhibitor treatment on activated vs basal signaling&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8.5))
fig5_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_panelA-1">
Past versions of fig5_panelA-1.png
</button>
</p>
<div id="fig-fig5_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-4" class="section level4">
<h4>Panel B</h4>
<p>Differential expression analysis of the effect of inhibitor
concentration on activated signaling</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1 samples: 3 inhibitors, 5 conc, 1 stimulus</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

fig5B_inhib_new &lt;- c(&quot;DMSO ctrl&quot;, &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

cts &lt;- fig5_data_id  %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; stimulus == &quot;aIg+H2O2&quot; &amp; inhib_conc != &quot;0 uM (PBS)&quot; &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig5_data_id, all_of(fig5_meta_cols)))) %&gt;%
  distinct() %&gt;%
  mutate(inhib_new = factor(inhib_new, levels = fig5B_inhib_new), 
         inhib_conc = factor(inhib_conc, levels = fig5_conc_all), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = fig5_stim)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new
# modeldesign &lt;- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_iSYK0.1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_Hcs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_Hcs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_Hcs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_Hcs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_Hcs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_Hcs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

# list_results &lt;- list(results_iBTK0.1_Hsc, results_iBTK1_Hsc, results_iBTK10_Hsc, results_iBTK100_Hsc)
list_results &lt;- list(results_iSYK0.1_Hcs, results_iSYK1_Hcs, results_iSYK10_Hcs, results_iSYK100_Hcs,
                     results_iBTK0.1_Hcs, results_iBTK1_Hcs, results_iBTK10_Hcs, results_iBTK100_Hcs,
                     results_iNFkB0.1_Hcs, results_iNFkB1_Hcs, results_iNFkB10_Hcs, results_iNFkB100_Hcs)

# summary(results_iSYK100_Hcs, padj = 0.05)

# Data transformation
# Regularized log transform
rld_HBL1 &lt;- rlog(dds, blind = FALSE)</code></pre>
<pre class="r"><code># Data
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
# names_results_volc &lt;- c(&quot;volc_iBTK0.1_Hcs&quot;, &quot;volc_iBTK1_Hcs&quot;, &quot;volc_iBTK10_Hcs&quot;, &quot;volc_iBTK100_Hcs&quot;)
names_results_volc &lt;- c(&quot;volc_iSYK0.1_Hcs&quot;, &quot;volc_iSYK1_Hcs&quot;, &quot;volc_iSYK10_Hcs&quot;, &quot;volc_iSYK100_Hcs&quot;,
                        &quot;volc_iBTK0.1_Hcs&quot;, &quot;volc_iBTK1_Hcs&quot;, &quot;volc_iBTK10_Hcs&quot;, &quot;volc_iBTK100_Hcs&quot;,
                        &quot;volc_iNFkB0.1_Hcs&quot;, &quot;volc_iNFkB1_Hcs&quot;, &quot;volc_iNFkB10_Hcs&quot;, &quot;volc_iNFkB100_Hcs&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Add column with comparison for faceting
fig5B_comp &lt;- c(&quot;0.1 uM iSYK&quot;, &quot;1 uM iSYK&quot;, &quot;10 uM iSYK&quot;, &quot;100 uM iSYK&quot;,
                &quot;0.1 uM iBTK&quot;, &quot;1 uM iBTK&quot;, &quot;10 uM iBTK&quot;, &quot;100 uM iBTK&quot;, 
                &quot;0.1 uM iNFkB&quot;, &quot;1 uM iNFkB&quot;, &quot;10 uM iNFkB&quot;, &quot;100 uM iNFkB&quot;)

volc_iSYK0.1_Hcs$comparison &lt;- fig5B_comp[1]
volc_iSYK1_Hcs$comparison &lt;- fig5B_comp[2]
volc_iSYK10_Hcs$comparison &lt;- fig5B_comp[3]
volc_iSYK100_Hcs$comparison &lt;- fig5B_comp[4]
volc_iBTK0.1_Hcs$comparison &lt;- fig5B_comp[5]
volc_iBTK1_Hcs$comparison &lt;- fig5B_comp[6]
volc_iBTK10_Hcs$comparison &lt;- fig5B_comp[7]
volc_iBTK100_Hcs$comparison &lt;- fig5B_comp[8]
volc_iNFkB0.1_Hcs$comparison &lt;- fig5B_comp[9]
volc_iNFkB1_Hcs$comparison &lt;- fig5B_comp[10]
volc_iNFkB10_Hcs$comparison &lt;- fig5B_comp[11]
volc_iNFkB100_Hcs$comparison &lt;- fig5B_comp[12]
fig5B_dataset &lt;- list(volc_iSYK0.1_Hcs, volc_iSYK1_Hcs, volc_iSYK10_Hcs, volc_iSYK100_Hcs,
                      volc_iBTK0.1_Hcs, volc_iBTK1_Hcs, volc_iBTK10_Hcs, volc_iBTK100_Hcs,
                      volc_iNFkB0.1_Hcs, volc_iNFkB1_Hcs, volc_iNFkB10_Hcs, volc_iNFkB100_Hcs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  mutate(comparison = factor(comparison, levels = fig5B_comp)) %&gt;%
  arrange(comparison) 

fig5B_dataset_subset &lt;- fig5B_dataset %&gt;%
  subset(comparison %in% c(&quot;0.1 uM iNFkB&quot;, &quot;1 uM iNFkB&quot;, &quot;10 uM iNFkB&quot;)) #, &quot;100 uM iNFkB&quot;

fig5A_prot_iNFkB100 &lt;- sort(unique(c(subset(as.data.frame(volc_HBL1_iNFkB100), diff_express != &quot;NO&quot;)$target_nospace))) 
fig5B_prot_all &lt;- sort(unique(c(subset(fig5B_dataset, comparison %in% c(&quot;0.1 uM iNFkB&quot;, &quot;1 uM iNFkB&quot;, &quot;10 uM iNFkB&quot;, &quot;100 uM iNFkB&quot;) &amp; diff_express != &quot;NO&quot;)$target_nospace)))
fig5B_prot_new &lt;- setdiff(fig5B_prot_all, fig5A_prot_iNFkB100)

fig5B_new &lt;- tibble(target_nospace = fig5B_prot_new, fig5B_new = fig5B_prot_new, font_face = &quot;bold.italic&quot;)
fig5B_dataset_subset &lt;- left_join(fig5B_dataset_subset, fig5B_new) %&gt;%
  mutate(font_face = replace_na(font_face, &quot;plain&quot;))
fig5_dataset_label &lt;- subset(fig5B_dataset_subset, font_face == &quot;bold.italic&quot;)

# Figure
fig5_panelB &lt;- ggplot(data = fig5B_dataset_subset,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.25) +
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  geom_text_repel(data = fig5_dataset_label, aes(fontface = font_face), 
                  size = 1.75,
                  segment.size = 0.2,
                  hjust = 1,
                  direction = &quot;y&quot;,
                  max.overlaps = 15,
                  nudge_y = 20,
                  nudge_x = ifelse(fig5_dataset_label$log2FoldChange &lt; 0, -0.4 - fig5_dataset_label$log2FoldChange, 0.5 - fig5_dataset_label$log2FoldChange)
  ) +
  facet_wrap(~comparison, ncol = 4) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;iNFkB versus DMSO control in activated cells&quot;) +
  # coord_cartesian(xlim = c(-0.8, 0.3), ylim = c(0, 50)) + 
  theme_bw() +
  textsize_small +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, plot.title = element_text(size = 8.5)) + #legend.position = &quot;none&quot;
  guides(shape = guide_legend(nrow = 3, bycol = TRUE), color = guide_legend(nrow = 3, bycol = TRUE))
fig5_panelB</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_panelB-1">
Past versions of fig5_panelB-1.png
</button>
</p>
<div id="fig-fig5_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Figure
fig5_panelB_test &lt;- ggplot(data = fig5B_dataset,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.25) +
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  geom_text_repel(#aes(fontface = font_face), 
                  size = 1.5,
                  segment.size = 0.2,
                  hjust = 1,
                  direction = &quot;y&quot;,
                  max.overlaps = 8,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(fig5B_dataset$log2FoldChange &lt; 0, -0.5 - fig5B_dataset$log2FoldChange, 0.35 - fig5B_dataset$log2FoldChange)
  ) +
  facet_wrap(~comparison, ncol = 4) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;iNFkB versus DMSO ctrl in activated cells&quot;) +
  # coord_cartesian(xlim = c(-0.8, 0.3), ylim = c(0, 50)) + 
  theme_bw() +
  textsize_small +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, plot.title = element_text(size = 8.5)) + #legend.position = &quot;none&quot;
  guides(shape = guide_legend(nrow = 3, bycol = TRUE), color = guide_legend(nrow = 3, bycol = TRUE))
fig5_panelB_test</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_panelB_test-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_panelB_test-1">
Past versions of fig5_panelB_test-1.png
</button>
</p>
<div id="fig-fig5_panelB_test-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-3" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Data
fig5C_prot &lt;- c(
  &quot;BTK_Y551&quot;
  # &quot;ERK1_2_T202_Y204&quot;, 
  # &quot;MAPK_p38_T180_Y182&quot;
) 
fig5C_prot_label &lt;- c(&quot;BTK (Y551)&quot;)
fig5C_data &lt;- fig5_data_id %&gt;%
  subset(target_nospace %in% fig5C_prot &amp; inhib_conc %in% fig5_conc_all &amp; cell_line == &quot;HBL1&quot;)
fig5C_mean &lt;- fig5_mean_id %&gt;%
  subset(target_nospace %in% fig5C_prot &amp; inhib_conc %in% fig5_conc_all &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig5_panelC &lt;- ggplot() +
  geom_point(data = subset(fig5C_data, target_nospace == fig5C_prot), 
             aes(factor(inhib_conc, levels = fig5_conc_all, labels = fig5_conc_all_label_alt), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 1) +
  geom_line(data = subset(fig5C_mean, target_nospace == fig5C_prot), 
            aes(factor(inhib_conc, levels = fig5_conc_all, labels = fig5_conc_all_label_alt), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig5_inhib)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = fig5_stim, labels = fig5_stim_clean, name = &quot;Stimulation&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig5_stim, labels = fig5_stim_clean, name = &quot;Stimulation&quot;) +
  scale_y_continuous(breaks = c(1000, 2000, 3000)) +
  labs(x = &quot;Inhibitor conc. (uM)&quot;, y = &quot;Normalised counts&quot;, title = fig5C_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), plot.title = element_text(size = 8), axis.text.x = element_text(size = 7)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig5_panelC</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_panelC-1">
Past versions of fig5_panelC-1.png
</button>
</p>
<div id="fig-fig5_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d-3" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Data
fig5D_prot &lt;- c(
  # &quot;BTK_Y551&quot;
  &quot;ERK1_2_T202_Y204&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
) 
fig5D_prot_label &lt;- c(&quot;ERK1/2 (T202/Y204)&quot;)
fig5D_data &lt;- fig5_data_id %&gt;%
  subset(target_nospace %in% fig5D_prot &amp; inhib_conc %in% fig5_conc_all &amp; cell_line == &quot;HBL1&quot;)
fig5D_mean &lt;- fig5_mean_id %&gt;%
  subset(target_nospace %in% fig5D_prot &amp; inhib_conc %in% fig5_conc_all &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig5_panelD &lt;- ggplot() +
  geom_point(data = subset(fig5D_data, target_nospace == fig5D_prot), 
             aes(factor(inhib_conc, levels = fig5_conc_all, labels = fig5_conc_all_label_alt), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 1) +
  geom_line(data = subset(fig5D_mean, target_nospace == fig5D_prot), 
            aes(factor(inhib_conc, levels = fig5_conc_all, labels = fig5_conc_all_label_alt), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig5_inhib)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = fig5_stim, labels = fig5_stim_clean, name = &quot;Stimulation&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig5_stim, labels = fig5_stim_clean, name = &quot;Stimulation&quot;) +
  scale_y_continuous(breaks = c(1700, 1900, 2100)) +
  labs(x = &quot;Inhibitor conc. (uM)&quot;, y = &quot;Normalised counts&quot;, title = fig5D_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), plot.title = element_text(size = 8), axis.text.x = element_text(size = 7)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig5_panelD</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_panelD-1">
Past versions of fig5_panelD-1.png
</button>
</p>
<div id="fig-fig5_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-e" class="section level4">
<h4>Panel E</h4>
<pre class="r"><code># Data
fig5E_prot &lt;- c(
  # &quot;BTK_Y551&quot;
  # &quot;ERK1_2_T202_Y204&quot;, 
  &quot;MAPK_p38_T180_Y182&quot;
) 
fig5E_prot_label &lt;- c(&quot;MAPK p38 (T180/Y182)&quot;)
fig5E_data &lt;- fig5_data_id %&gt;%
  subset(target_nospace %in% fig5E_prot &amp; inhib_conc %in% fig5_conc_all &amp; cell_line == &quot;HBL1&quot;)
fig5E_mean &lt;- fig5_mean_id %&gt;%
  subset(target_nospace %in% fig5E_prot &amp; inhib_conc %in% fig5_conc_all &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig5_panelE &lt;- ggplot() +
  geom_point(data = subset(fig5E_data, target_nospace == fig5E_prot), 
             aes(factor(inhib_conc, levels = fig5_conc_all, labels = fig5_conc_all_label_alt), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 1) +
  geom_line(data = subset(fig5E_mean, target_nospace == fig5E_prot), 
            aes(factor(inhib_conc, levels = fig5_conc_all, labels = fig5_conc_all_label_alt), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig5_inhib)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = fig5_stim, labels = fig5_stim_clean, name = &quot;Stimulation&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig5_stim, labels = fig5_stim_clean, name = &quot;Stimulation&quot;) +
  scale_y_continuous(breaks = c(1300, 1500, 1700)) +
  labs(x = &quot;Inhibitor conc. (uM)&quot;, y = &quot;Normalised counts&quot;, title = fig5E_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), plot.title = element_text(size = 8), axis.text.x = element_text(size = 7)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig5_panelE</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_panelE-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_panelE-1">
Past versions of fig5_panelE-1.png
</button>
</p>
<div id="fig-fig5_panelE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-4" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig5_AB &lt;- plot_grid(fig5_panelA, fig5_panelB, labels = PANEL_labels[c(1, 2)], ncol = 1, rel_heights = c(1.2, 1), label_size = 10)
fig5_CDE &lt;- plot_grid(fig5_panelC, fig5_panelD, fig5_panelE, labels = PANEL_labels[c(3, 4, 5)], nrow = 1, rel_widths = c(1, 1, 1), label_size = 10, align = &quot;h&quot;)
fig5 &lt;- plot_grid(fig5_AB, fig5_CDE, ncol = 1, rel_heights = c(1.65, 1), label_size = 10)
fig5</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig5_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig5_combi-1">
Past versions of fig5_combi-1.png
</button>
</p>
<div id="fig-fig5_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig5,
  filename = &quot;output/figures/DLBCL_proteins/main_fig5.pdf&quot;,
  width = 6,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig5,
  filename = &quot;output/figures/DLBCL_proteins/main_fig5.jpg&quot;,
  width = 6,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Inhibition of core BCR signaling and MAPK signaling in ABC-DLBCL
subtype model.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig5&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9775483 522.1   16807764 897.7  16807764 897.7
Vcells 17358027 132.5   53605536 409.0 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-6" class="section level2">
<h2>Figure 6</h2>
<p><em>Inhibition of PI3K-AKT signaling and transcription factors in
GCB-DLBCL subtype model.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li><p>Count data</p></li>
<li><p>DESeq2 analysis</p></li>
</ul></li>
</ul>
<div id="data-4" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig6_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;OCI-Ly8&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)
fig6_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;OCI-Ly8&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new), 
         cell_clean = case_when(cell_line == &quot;HBL1&quot; ~ &quot;ABC subtype&quot;, 
                                cell_line == &quot;OCI-Ly8&quot; ~ &quot;GCB subtype&quot;), 
         cell_stim = paste0(stim_clean, &quot;\n&quot;, cell_clean)) #%&gt;%
  # subset(inhib_new == &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig6_cells &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
fig6_cells_clean &lt;- c(&quot;ABC subtype&quot;, &quot;GCB subtype&quot;)
fig6_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig6_stim_clean &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig6_inhib &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
# fig6_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
# fig6_conc_label &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
fig6_conc_all &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig6_conc_all_label &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig6_conc_all_label_alt &lt;- c(&quot;0&quot;, &quot;0.1&quot;, &quot;1&quot;, &quot;10&quot;, &quot;100&quot;)

fig6_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;cell_clean&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;cell_stim&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;inhib_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-5" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-5" class="section level4">
<h4>Panel A</h4>
<p>Differential expression analysis of the effect of inhibitor
concentration on the stimulation in OCI</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>OCI: 3 inhibitors, 5 conc, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig6_data_id  %&gt;%
  dplyr::filter(cell_line == &quot;OCI-Ly8&quot; &amp; inhib_conc %in% fig6_conc_all &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig6_data_id, all_of(fig6_meta_cols)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new + stim_clean + inhib_text_new:stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_OCI_stim &lt;- results(dds, name = &quot;stim_clean_Activated_vs_Basal&quot;, alpha = 0.05)
results_OCI_iBTK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)
results_OCI_iSYK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iSYK.stim_cleanActivated&quot;, alpha = 0.05)
results_OCI_iNFkB100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iNFkB.stim_cleanActivated&quot;, alpha = 0.05)

list_results &lt;- list(results_OCI_stim, results_OCI_iBTK100, results_OCI_iSYK100, results_OCI_iNFkB100)

# Data transformation
# Regularized log transform
rld_OCI &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_volc &lt;- c(&quot;volc_OCI_stim&quot;, &quot;volc_OCI_iBTK100&quot;, &quot;volc_OCI_iSYK100&quot;, &quot;volc_OCI_iNFkB100&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_OCI_stim, volc_OCI_iBTK100, volc_OCI_iSYK100, volc_OCI_iNFkB100)
names_heat &lt;- c(&quot;heat_OCI_stim&quot;, &quot;heat_OCI_iBTK100&quot;, &quot;heat_OCI_iSYK100&quot;, &quot;heat_OCI_iNFkB100&quot;)
for(dataset in c(1:length(list_results_volc))){
  assign(names_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}</code></pre>
<pre class="r"><code># Data
# Combine heatmap datasets (without DESeq2 analysis values)
heat_inhib_OCI &lt;- list(heat_OCI_stim, heat_OCI_iBTK100, heat_OCI_iSYK100, heat_OCI_iNFkB100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  left_join(fig6_data_id[, 14:21]) %&gt;%
  dplyr::filter(modification == &quot;phospho&quot;) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
fig6A_info_OCI &lt;- fig6_data_id %&gt;%
  dplyr::select(plate_well, stim_clean, inhib_conc, inhibitor, cell_line) %&gt;%
  distinct() %&gt;%
  dplyr::filter(cell_line == &quot;OCI-Ly8&quot; &amp; inhib_conc %in% fig6_conc_all &amp; !(stim_clean == &quot;Basal&quot; &amp; inhib_conc %in% fig6_conc_all[2:5])) %&gt;%
  mutate(inhibitor = factor(inhibitor, levels = fig6_inhib)) %&gt;%
  as.data.frame()
rownames(fig6A_info_OCI) &lt;- fig6A_info_OCI$plate_well
fig6A_info_OCI &lt;- dplyr::select(fig6A_info_OCI, -c(plate_well, cell_line))

# Get the rld transformed data for the heatmap and scale per row
fig6A_data_heat_OCI &lt;- t(scale(t(assay(rld_OCI)[c(heat_inhib_OCI$target_nospace), rownames(fig6A_info_OCI)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

# Figure
fig6A_ann_colors &lt;- list(Stimulation = colors_stim, 
                         Inhibitor = colors_inhib,
                         # inhib_new = c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib), 
                         &quot;Inhibitor conc&quot; = setNames(colors_conc, unique(fig6A_info_OCI$inhib_conc)))
fig6A_heat_legend &lt;- list(title = &quot;Scaled\nexpression&quot;, title_gp = gpar(fontsize = 7), labels_gp = gpar(fontsize = 7), 
                          grid_width = unit(2, &quot;mm&quot;), grid_height = unit(20, &quot;mm&quot;))

fig6A_col_ann &lt;- HeatmapAnnotation(Inhibitor = fig6A_info_OCI$inhibitor, 
                                   &quot;Inhibitor conc&quot; = fig6A_info_OCI$inhib_conc, 
                                   Stimulation = fig6A_info_OCI$stim_clean, 
                                   simple_anno_size = unit(2.5, &quot;mm&quot;), 
                                   annotation_name_gp = gpar(fontsize = 7),
                                   col = fig6A_ann_colors, 
                                   annotation_legend_param = list(Inhibitor = list(title = &quot;Inhibitor&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  &quot;Inhibitor conc&quot; = list(title = &quot;Inhibitor\nconcentration&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  Stimulation = list(title = &quot;Stimulation&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;))), 
                                   show_legend = c(FALSE, TRUE, TRUE))

fig6_panelA &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  fig6A_data_heat_OCI,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  # treeheight_row = 20,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_col = fig6A_info_Ly8,
  top_annotation = fig6A_col_ann,
  # annotation_colors = fig6A_ann_colors,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = fig6A_info_OCI$inhibitor,
  row_split = subset(heat_inhib_OCI)$modification,
  row_gap = unit(1, &quot;mm&quot;), 
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 5.75,
  row_title_gp = gpar(fontsize = 7),
  column_title_gp = gpar(fontsize = 7), 
  annotation_names_col = gpar(fontsize = 7),
  border_color = NA,
  heatmap_legend_param = fig6A_heat_legend, 
  # annotation_legend = FALSE,
)) +
  ggtitle(&quot;          GCB subtype: Effect of inhibitor treatment on activated vs basal signaling&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8.5))
fig6_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_panelA-1">
Past versions of fig6_panelA-1.png
</button>
</p>
<div id="fig-fig6_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-5" class="section level4">
<h4>Panel B</h4>
<p>Differential expression analysis of the effect of inhibitor
concentration on activated signaling</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>OCI samples: 3 inhibitors, 5 conc, 1 stimulus</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

fig6B_inhib_new &lt;- c(&quot;DMSO ctrl&quot;, &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

cts &lt;- fig6_data_id  %&gt;%
  dplyr::filter(cell_line == &quot;OCI-Ly8&quot; &amp; stimulus == &quot;aIg+H2O2&quot; &amp; inhib_conc != &quot;0 uM (PBS)&quot; &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(fig6_data_id, all_of(fig6_meta_cols)))) %&gt;%
  distinct() %&gt;%
  mutate(inhib_new = factor(inhib_new, levels = fig6B_inhib_new), 
         inhib_conc = factor(inhib_conc, levels = fig6_conc_all), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = fig6_stim)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign &lt;- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign &lt;- ~ inhib_text_new
# modeldesign &lt;- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds &lt;- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_iSYK0.1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK10_Hcs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK100_Hcs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iBTK0.1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK10_Hcs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTK100_Hcs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)

results_iNFkB0.1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB1_Hcs &lt;- results(dds, name = &quot;inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB10_Hcs &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB100_Hcs &lt;- results(dds, name = &quot;inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)

# list_results &lt;- list(results_iBTK0.1_Hsc, results_iBTK1_Hsc, results_iBTK10_Hsc, results_iBTK100_Hsc)
list_results &lt;- list(results_iSYK0.1_Hcs, results_iSYK1_Hcs, results_iSYK10_Hcs, results_iSYK100_Hcs,
                     results_iBTK0.1_Hcs, results_iBTK1_Hcs, results_iBTK10_Hcs, results_iBTK100_Hcs,
                     results_iNFkB0.1_Hcs, results_iNFkB1_Hcs, results_iNFkB10_Hcs, results_iNFkB100_Hcs)

# summary(results_iSYK100_Hcs, padj = 0.05)

# Data transformation
# Regularized log transform
rld_OCI &lt;- rlog(dds, blind = FALSE)</code></pre>
<pre class="r"><code># Data
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
# names_results_volc &lt;- c(&quot;volc_iBTK0.1_Hcs&quot;, &quot;volc_iBTK1_Hcs&quot;, &quot;volc_iBTK10_Hcs&quot;, &quot;volc_iBTK100_Hcs&quot;)
names_results_volc &lt;- c(&quot;volc_iSYK0.1_Hcs&quot;, &quot;volc_iSYK1_Hcs&quot;, &quot;volc_iSYK10_Hcs&quot;, &quot;volc_iSYK100_Hcs&quot;,
                        &quot;volc_iBTK0.1_Hcs&quot;, &quot;volc_iBTK1_Hcs&quot;, &quot;volc_iBTK10_Hcs&quot;, &quot;volc_iBTK100_Hcs&quot;,
                        &quot;volc_iNFkB0.1_Hcs&quot;, &quot;volc_iNFkB1_Hcs&quot;, &quot;volc_iNFkB10_Hcs&quot;, &quot;volc_iNFkB100_Hcs&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Add column with comparison for faceting
fig6B_comp &lt;- c(&quot;0.1 uM iSYK&quot;, &quot;1 uM iSYK&quot;, &quot;10 uM iSYK&quot;, &quot;100 uM iSYK&quot;,
                &quot;0.1 uM iBTK&quot;, &quot;1 uM iBTK&quot;, &quot;10 uM iBTK&quot;, &quot;100 uM iBTK&quot;, 
                &quot;0.1 uM iNFkB&quot;, &quot;1 uM iNFkB&quot;, &quot;10 uM iNFkB&quot;, &quot;100 uM iNFkB&quot;)

volc_iSYK0.1_Hcs$comparison &lt;- fig6B_comp[1]
volc_iSYK1_Hcs$comparison &lt;- fig6B_comp[2]
volc_iSYK10_Hcs$comparison &lt;- fig6B_comp[3]
volc_iSYK100_Hcs$comparison &lt;- fig6B_comp[4]
volc_iBTK0.1_Hcs$comparison &lt;- fig6B_comp[5]
volc_iBTK1_Hcs$comparison &lt;- fig6B_comp[6]
volc_iBTK10_Hcs$comparison &lt;- fig6B_comp[7]
volc_iBTK100_Hcs$comparison &lt;- fig6B_comp[8]
volc_iNFkB0.1_Hcs$comparison &lt;- fig6B_comp[9]
volc_iNFkB1_Hcs$comparison &lt;- fig6B_comp[10]
volc_iNFkB10_Hcs$comparison &lt;- fig6B_comp[11]
volc_iNFkB100_Hcs$comparison &lt;- fig6B_comp[12]
fig6B_dataset &lt;- list(volc_iSYK0.1_Hcs, volc_iSYK1_Hcs, volc_iSYK10_Hcs, volc_iSYK100_Hcs,
                      volc_iBTK0.1_Hcs, volc_iBTK1_Hcs, volc_iBTK10_Hcs, volc_iBTK100_Hcs,
                      volc_iNFkB0.1_Hcs, volc_iNFkB1_Hcs, volc_iNFkB10_Hcs, volc_iNFkB100_Hcs) %&gt;%
  purrr::reduce(full_join) %&gt;%
  mutate(comparison = factor(comparison, levels = fig6B_comp)) %&gt;%
  arrange(comparison)

fig6B_dataset_subset &lt;- fig6B_dataset %&gt;%
  subset(comparison %in% c(&quot;0.1 uM iSYK&quot;, &quot;1 uM iSYK&quot;, &quot;10 uM iSYK&quot;)) #, &quot;100 uM iSYK&quot;

fig6A_prot_iSYK100 &lt;- sort(unique(c(subset(as.data.frame(volc_OCI_iSYK100), diff_express != &quot;NO&quot;)$target_nospace))) 
fig6B_prot_all &lt;- sort(unique(c(subset(fig6B_dataset, comparison %in% c(&quot;0.1 uM iSYK&quot;, &quot;1 uM iSYK&quot;, &quot;10 uM iSYK&quot;, &quot;100 uM iSYK&quot;) &amp; diff_express != &quot;NO&quot;)$target_nospace)))
fig6B_prot_new &lt;- setdiff(fig6B_prot_all, fig6A_prot_iSYK100)

fig6B_new &lt;- tibble(target_nospace = fig6B_prot_new, fig6B_new = fig6B_prot_new, font_face = &quot;bold.italic&quot;)
fig6B_dataset_subset &lt;- left_join(fig6B_dataset_subset, fig6B_new) %&gt;%
  mutate(font_face = replace_na(font_face, &quot;plain&quot;))
fig6B_dataset_label &lt;- subset(fig6B_dataset_subset, font_face == &quot;bold.italic&quot;)

# Figure
fig6_panelB &lt;- ggplot(data = fig6B_dataset_subset,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.25) +
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  geom_text_repel(data = fig6B_dataset_label, aes(fontface = font_face), 
                  size = 1.75,
                  segment.size = 0.2,
                  hjust = 1,
                  direction = &quot;y&quot;,
                  max.overlaps = 15,
                  nudge_y = 20,
                  nudge_x = ifelse(fig6B_dataset_label$log2FoldChange &lt; 0, -0.4 - fig6B_dataset_label$log2FoldChange, 0.25 - fig6B_dataset_label$log2FoldChange)
  ) +
  facet_wrap(~comparison, ncol = 4) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;iSYK versus DMSO control in activated cells&quot;) +
  # coord_cartesian(xlim = c(-0.8, 0.3), ylim = c(0, 50)) + 
  theme_bw() +
  textsize_small +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, plot.title = element_text(size = 8.5)) + #legend.position = &quot;none&quot;
  guides(shape = guide_legend(nrow = 3, bycol = TRUE), color = guide_legend(nrow = 3, bycol = TRUE))
fig6_panelB</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_panelB-1">
Past versions of fig6_panelB-1.png
</button>
</p>
<div id="fig-fig6_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Figure
fig6_panelB_test &lt;- ggplot(data = fig6B_dataset,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.25) +
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  geom_text_repel(#aes(fontface = font_face), 
                  size = 1.5,
                  segment.size = 0.2,
                  hjust = 1,
                  direction = &quot;y&quot;,
                  max.overlaps = 8,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(fig6B_dataset$log2FoldChange &lt; 0, -0.5 - fig6B_dataset$log2FoldChange, 0.35 - fig6B_dataset$log2FoldChange)
  ) +
  facet_wrap(~comparison, ncol = 4) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;iSYK versus DMSO ctrl in activated cells&quot;) +
  # coord_cartesian(xlim = c(-0.8, 0.3), ylim = c(0, 50)) + 
  theme_bw() +
  textsize_small +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, plot.title = element_text(size = 8.5)) + #legend.position = &quot;none&quot;
  guides(shape = guide_legend(nrow = 3, bycol = TRUE), color = guide_legend(nrow = 3, bycol = TRUE))
fig6_panelB_test</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_panelB_test-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_panelB_test-1">
Past versions of fig6_panelB_test-1.png
</button>
</p>
<div id="fig-fig6_panelB_test-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-4" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Data
fig6C_prot &lt;- c(
  &quot;S6_S240_S244&quot;
  # &quot;&quot;CREB_S133&quot;
  # &quot;VAV1_Y174&quot;
) 
fig6C_prot_label &lt;- c(&quot;S6 (S240/S244)&quot;)
fig6C_data &lt;- fig6_data_id %&gt;%
  subset(target_nospace %in% fig6C_prot &amp; inhib_conc %in% fig6_conc_all &amp; cell_line == &quot;OCI-Ly8&quot;)
fig6C_mean &lt;- fig6_mean_id %&gt;%
  subset(target_nospace %in% fig6C_prot &amp; inhib_conc %in% fig6_conc_all &amp; cell_line == &quot;OCI-Ly8&quot;) 

# Figure
fig6_panelC &lt;- ggplot() +
  geom_point(data = subset(fig6C_data, target_nospace == fig6C_prot), 
             aes(factor(inhib_conc, levels = fig6_conc_all, labels = fig6_conc_all_label_alt), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig6C_mean, target_nospace == fig6C_prot), 
            aes(factor(inhib_conc, levels = fig6_conc_all, labels = fig6_conc_all_label_alt), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig6_inhib)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = fig6_stim, labels = fig6_stim_clean, name = &quot;Stimulation&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig6_stim, labels = fig6_stim_clean, name = &quot;Stimulation&quot;) +
  scale_y_continuous(breaks = c(1000, 1300, 1600)) +
  labs(x = &quot;Inhibitor conc. (uM)&quot;, y = &quot;Normalised counts&quot;, title = fig6C_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), plot.title = element_text(size = 8), axis.text.x = element_text(size = 7)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig6_panelC</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_panelC-1">
Past versions of fig6_panelC-1.png
</button>
</p>
<div id="fig-fig6_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d-4" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Data
fig6D_prot &lt;- c(
  # &quot;S6_S240_S244&quot;
  &quot;CREB_S133&quot;
  # &quot;VAV1_Y174&quot;
) 
fig6D_prot_label &lt;- c(&quot;CREB (S133)&quot;)
fig6D_data &lt;- fig6_data_id %&gt;%
  subset(target_nospace %in% fig6D_prot &amp; inhib_conc %in% fig6_conc_all &amp; cell_line == &quot;OCI-Ly8&quot;)
fig6D_mean &lt;- fig6_mean_id %&gt;%
  subset(target_nospace %in% fig6D_prot &amp; inhib_conc %in% fig6_conc_all &amp; cell_line == &quot;OCI-Ly8&quot;) 

# Figure
fig6_panelD &lt;- ggplot() +
  geom_point(data = subset(fig6D_data, target_nospace == fig6D_prot), 
             aes(factor(inhib_conc, levels = fig6_conc_all, labels = fig6_conc_all_label_alt), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig6D_mean, target_nospace == fig6D_prot), 
            aes(factor(inhib_conc, levels = fig6_conc_all, labels = fig6_conc_all_label_alt), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig6_inhib)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = fig6_stim, labels = fig6_stim_clean, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig6_stim, labels = fig6_stim_clean, name = &quot;Stimulus&quot;) +
  scale_y_continuous(breaks = c(1800, 2100, 2400)) +
  labs(x = &quot;Inhibitor conc. (uM)&quot;, y = &quot;Normalised counts&quot;, title = fig6D_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), plot.title = element_text(size = 8), axis.text.x = element_text(size = 7)) #+
  # guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig6_panelD</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_panelD-1">
Past versions of fig6_panelD-1.png
</button>
</p>
<div id="fig-fig6_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-e-1" class="section level4">
<h4>Panel E</h4>
<pre class="r"><code># Data
fig6E_prot &lt;- c(
  # &quot;S6_S240_S244&quot;
  # &quot;&quot;CREB_S133&quot;
  &quot;VAV1_Y174&quot;
) 
fig6E_prot_label &lt;- c(&quot;VAV1 (Y174)&quot;)
fig6E_data &lt;- fig6_data_id %&gt;%
  subset(target_nospace %in% fig6E_prot &amp; inhib_conc %in% fig6_conc_all &amp; cell_line == &quot;OCI-Ly8&quot;)
fig6E_mean &lt;- fig6_mean_id %&gt;%
  subset(target_nospace %in% fig6E_prot &amp; inhib_conc %in% fig6_conc_all &amp; cell_line == &quot;OCI-Ly8&quot;) 

# Figure
fig6_panelE &lt;- ggplot() +
  geom_point(data = subset(fig6E_data, target_nospace == fig6E_prot), 
             aes(factor(inhib_conc, levels = fig6_conc_all, labels = fig6_conc_all_label_alt), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig6E_mean, target_nospace == fig6E_prot), 
            aes(factor(inhib_conc, levels = fig6_conc_all, labels = fig6_conc_all_label_alt), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig6_inhib)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = fig6_stim, labels = fig6_stim_clean, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig6_stim, labels = fig6_stim_clean, name = &quot;Stimulus&quot;) +
  scale_y_continuous(breaks = c(1500, 2000, 2500)) +
  labs(x = &quot;Inhibitor conc. (uM)&quot;, y = &quot;Normalised counts&quot;, title = fig6E_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), plot.title = element_text(size = 8), axis.text.x = element_text(size = 7)) #+
  # guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig6_panelE</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_panelE-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_panelE-1">
Past versions of fig6_panelE-1.png
</button>
</p>
<div id="fig-fig6_panelE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-5" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig6_AB &lt;- plot_grid(fig6_panelA, fig6_panelB, labels = PANEL_labels[c(1, 2)], ncol = 1, rel_heights = c(1.2, 1), label_size = 10)
fig6_CDE &lt;- plot_grid(fig6_panelC, fig6_panelD, fig6_panelE, labels = PANEL_labels[c(3, 4, 5)], nrow = 1, rel_widths = c(1, 1, 1), label_size = 10, align = &quot;h&quot;)
fig6 &lt;- plot_grid(fig6_AB, fig6_CDE, ncol = 1, rel_heights = c(1.65, 1), label_size = 10)
fig6</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig6_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig6_combi-1">
Past versions of fig6_combi-1.png
</button>
</p>
<div id="fig-fig6_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig6,
  filename = &quot;output/figures/DLBCL_proteins/main_fig6.pdf&quot;,
  width = 6,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig6,
  filename = &quot;output/figures/DLBCL_proteins/main_fig6.jpg&quot;,
  width = 6,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Inhibition of PI3K-AKT signaling and transcription factors in
GCB-DLBCL subtype model.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig6&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9774280 522.1   16807764 897.7  16807764 897.7
Vcells 17357214 132.5   53605536 409.0 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-7" class="section level2">
<h2>Figure 7</h2>
<p><em>DLBCL subtype-dependent small molecule inhibitor
effects.</em></p>
<p>Input:</p>
<ul>
<li>BioRender schematic figure</li>
</ul>
<div id="panels-6" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-6" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
fig7_panelA_place &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig7_panelA_place

# PNG
fig7_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/ch4_DLBCL_inhibition_summary.png&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
fig7_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig7_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig7_panelA-1">
Past versions of fig7_panelA-1.png
</button>
</p>
<div id="fig-fig7_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-6" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig7_panelA</code></pre>
<p><img src="figure/t_proteins_main.Rmd/fig7_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig7_combi-1">
Past versions of fig7_combi-1.png
</button>
</p>
<div id="fig-fig7_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig7_panelA_place,
  filename = &quot;output/figures/DLBCL_proteins/main_fig7_place.pdf&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig7_panelA_place,
  filename = &quot;output/figures/DLBCL_proteins/main_fig7_place.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>DLBCL subtype-dependent small molecule inhibitor
effects.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig7&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb)  max used  (Mb)
Ncells  9734397 519.9   16807764 897.7  16807764 897.7
Vcells 17254570 131.7   53605536 409.0 107717937 821.9</code></pre>
</div>
</div>
<div id="figure-0" class="section level2">
<h2>Figure 0</h2>
<p><em>Figure title.</em></p>
<p>Input: - BioRender schematic figures</p>
<ul>
<li><p>… phosphoflow data</p></li>
<li><p>… ID-seq data</p></li>
</ul>
<div id="data-5" class="section level3">
<h3>Data</h3>
</div>
<div id="panels-7" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-7" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
fig0_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig0_panelA

# PNG
png_fig0 &lt;- image_read(&quot;output/figures/non_R_figs/xxx.png&quot;) %&gt;%
  image_ggplot()
# png_fig0</code></pre>
</div>
<div id="panel-b-6" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
<div id="panel-c-5" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
</div>
<div id="combined-fig-7" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig0 &lt;- plot_grid(panelA, panelB, labels = PANEL_labels[c(1, 2)], ncol = 1, rel_heights = c(1, 1), label_size = 10)
fig0

# Save figure as pdf, jpg
# ggsave(
#   fig0,
#   filename = &quot;output/figures/DLBCL_proteins/main_fig0.pdf&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  fig0,
  filename = &quot;output/figures/DLBCL_proteins/main_fig0.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure title.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig0&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ashr_2.2-63                 apeglm_1.26.1              
 [3] MASS_7.3-60.2               ggbreak_0.1.2              
 [5] geomtextpath_0.1.4          ggh4x_0.2.8                
 [7] cowplot_1.1.3               patchwork_1.2.0            
 [9] greekLetters_1.0.2          ggpubr_0.6.0               
[11] ggrepel_0.9.5               scales_1.3.0               
[13] ggthemes_5.1.0              RColorBrewer_1.1-3         
[15] scico_1.5.0                 viridis_0.6.5              
[17] viridisLite_0.4.2           gridExtra_2.3              
[19] mmtable2_0.1.3              gt_0.10.1                  
[21] ggvenn_0.1.10               rgl_1.3.1                  
[23] plotly_4.10.4               ggcorrplot_0.1.4.1         
[25] corrplot_0.92               umap_0.2.10.0              
[27] ggplotify_0.1.2             dendsort_0.3.4             
[29] vsn_3.72.0                  ComplexHeatmap_2.20.0      
[31] platetools_0.1.7            kableExtra_1.4.0           
[33] DESeq2_1.44.0               SummarizedExperiment_1.34.0
[35] Biobase_2.64.0              MatrixGenerics_1.16.0      
[37] matrixStats_1.3.0           GenomicRanges_1.56.1       
[39] GenomeInfoDb_1.40.1         IRanges_2.38.0             
[41] S4Vectors_0.42.0            BiocGenerics_0.50.0        
[43] Seurat_5.1.0                SeuratObject_5.0.2         
[45] sp_2.1-4                    rstatix_0.7.2              
[47] factoextra_1.0.7            FactoMineR_2.11            
[49] readxl_1.4.3                Matrix_1.7-0               
[51] lubridate_1.9.3             forcats_1.0.0              
[53] stringr_1.5.1               dplyr_1.1.4                
[55] purrr_1.0.2                 readr_2.1.5                
[57] tidyr_1.3.1                 tibble_3.2.1               
[59] ggplot2_3.5.1               tidyverse_2.0.0            
[61] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             goftest_1.2-3           DT_0.33                
  [4] TH.data_1.1-2           vctrs_0.6.5             spatstat.random_3.2-3  
  [7] digest_0.6.36           png_0.1-8               shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       magick_2.8.3            SQUAREM_2021.1         
 [16] reshape2_1.4.4          httpuv_1.6.15           foreach_1.5.2          
 [19] withr_3.0.0             xfun_0.45               ggfun_0.1.5            
 [22] survival_3.6-4          memoise_2.0.1           emmeans_1.10.2         
 [25] systemfonts_1.1.0       ragg_1.3.2              zoo_1.8-12             
 [28] GlobalOptions_0.1.2     pbapply_1.7-2           promises_1.3.0         
 [31] scatterplot3d_0.3-44    httr_1.4.7              globals_0.16.3         
 [34] fitdistrplus_1.1-11     ps_1.7.6                rstudioapi_0.16.0      
 [37] UCSC.utils_1.0.0        miniUI_0.1.1.1          generics_0.1.3         
 [40] base64enc_0.1-3         processx_3.8.4          zlibbioc_1.50.0        
 [43] polyclip_1.10-6         GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [46] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [49] S4Arrays_1.4.1          preprocessCore_1.66.0   hms_1.1.3              
 [52] irlba_2.3.5.1           colorspace_2.1-0        ROCR_1.0-11            
 [55] reticulate_1.38.0       spatstat.data_3.1-2     magrittr_2.0.3         
 [58] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [61] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [64] scattermore_1.2         RcppAnnoy_0.0.22        pillar_1.9.0           
 [67] nlme_3.1-164            iterators_1.0.14        compiler_4.4.1         
 [70] RSpectra_0.16-1         stringi_1.8.4           tensor_1.5             
 [73] plyr_1.8.9              crayon_1.5.3            abind_1.4-5            
 [76] truncnorm_1.0-9         gridGraphics_0.5-1      emdbook_1.3.13         
 [79] locfit_1.5-9.10         bit_4.0.5               sandwich_3.1-0         
 [82] whisker_0.4.1           codetools_0.2-20        multcomp_1.4-25        
 [85] textshaping_0.4.0       openssl_2.2.0           bslib_0.7.0            
 [88] GetoptLong_1.0.5        mime_0.12               splines_4.4.1          
 [91] circlize_0.4.16         Rcpp_1.0.12             fastDummies_1.7.3      
 [94] cellranger_1.1.0        Rttf2pt1_1.3.12         leaps_3.2              
 [97] knitr_1.47              utf8_1.2.4              clue_0.3-65            
[100] fs_1.6.4                listenv_0.9.1           ggsignif_0.6.4         
[103] estimability_1.5.1      callr_3.7.6             statmod_1.5.0          
[106] tzdb_0.4.0              svglite_2.1.3           pkgconfig_2.0.3        
[109] tools_4.4.1             cachem_1.1.0            numDeriv_2016.8-1.1    
[112] fastmap_1.2.0           rmarkdown_2.27          ica_1.0-3              
[115] broom_1.0.6             sass_0.4.9              coda_0.19-4.1          
[118] BiocManager_1.30.23     dotCall64_1.1-1         carData_3.0-5          
[121] RANN_2.6.1              farver_2.1.2            yaml_2.3.8             
[124] cli_3.6.3               leiden_0.4.3.1          lifecycle_1.0.4        
[127] askpass_1.2.0           uwot_0.2.2              mvtnorm_1.2-5          
[130] backports_1.5.0         BiocParallel_1.38.0     timechange_0.3.0       
[133] gtable_0.3.5            rjson_0.2.21            ggridges_0.5.6         
[136] progressr_0.14.0        parallel_4.4.1          limma_3.60.3           
[139] jsonlite_1.8.8          RcppHNSW_0.6.0          bit64_4.0.5            
[142] multcompView_0.1-10     assertthat_0.2.1        Rtsne_0.17             
[145] yulab.utils_0.1.4       spatstat.utils_3.0-5    bdsmatrix_1.3-7        
[148] highr_0.11              jquerylib_0.1.4         lazyeval_0.2.2         
[151] shiny_1.8.1.1           htmltools_0.5.8.1       affy_1.82.0            
[154] sctransform_0.4.1       glue_1.7.0              spam_2.10-0            
[157] XVector_0.44.0          rprojroot_2.0.4         flashClust_1.01-2      
[160] invgamma_1.1            igraph_2.0.3            extrafontdb_1.0        
[163] R6_2.5.1                labeling_0.4.3          bbmle_1.0.25.1         
[166] cluster_2.1.6           aplot_0.2.3             DelayedArray_0.30.1    
[169] tidyselect_1.2.1        xml2_1.3.6              car_3.1-2              
[172] future_1.33.2           munsell_0.5.1           KernSmooth_2.23-24     
[175] affyio_1.74.0           data.table_1.15.4       htmlwidgets_1.6.4      
[178] rlang_1.1.4             spatstat.sparse_3.1-0   extrafont_0.19         
[181] spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
