<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-10-22" />

<title>DLBCL states chapter main figures</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapters on DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Paper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="y_paper_main.html">Main</a>
    </li>
    <li>
      <a href="z_paper_suppl.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteins ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="t_proteins_main.html.html">Main</a>
    </li>
    <li>
      <a href="u_proteins_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    States ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="v_states_main.html.html">Main</a>
    </li>
    <li>
      <a href="w_states_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
    <li>
      <a href="c_flow_vis.html">Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="DS108_p_seq_comparisons.html">DS108 clean</a>
    </li>
    <li>
      <a href="DS108_q_seq_thresholds.html">DS108 thresholds</a>
    </li>
    <li>
      <a href="DS108_r_seq_pca.html">DS108 PCA</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">DLBCL states chapter main figures</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-10-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapters_DLBCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241022code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241022code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241022)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong1ffbd94">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 1ffbd94
</a>
</p>
</div>
<div id="strongRepositoryversionstrong1ffbd94"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 1ffbd94.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/AL006_InhibCombiTest/
    Untracked:  data/DS102_StimTest/
    Untracked:  data/DS103_ViabilityTest/
    Untracked:  data/DS104_StimInhibTest/
    Untracked:  data/DS108_StimInhibIDseq/
    Untracked:  data/DS113_InhibCombiIDseq/
    Untracked:  output/DS102_StimTest/
    Untracked:  output/DS103_ViabilityTest/
    Untracked:  output/DS104_StimInhibTest/
    Untracked:  output/DS108_StimInhibIDseq/
    Untracked:  output/DS113_InhibCombiIDseq/
    Untracked:  output/figures/
    Untracked:  output/network_visual_DLBCL/
    Untracked:  output/network_visual_paper/

Unstaged changes:
    Deleted:    analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/v_states_main.Rmd</code>) and
HTML (<code>docs/v_states_main.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
1ffbd94
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Revised chapter figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
6559698
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Paper figs + analysis files for ch figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)
PANEL_labels &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;,&quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;)

textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
                  axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
                  axis.title = element_text(colour = &quot;black&quot;, size = 12), 
                  legend.title = element_text(colour = &quot;black&quot;, size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
                  strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
)

textsize_small &lt;- theme(text = element_text(size = 9, family = &quot;sans&quot;, colour = &quot;black&quot;),
                        plot.title = element_text(size = 10)
)</code></pre>
<pre class="r"><code>colors_nord &lt;- c(&quot;#2e3440&quot;, &quot;#3b4252&quot;, &quot;#434c5e&quot;, &quot;#4c566a&quot;, 
                 &quot;#d8dee9&quot;, &quot;#e5e9f0&quot;, &quot;#eceff4&quot;, 
                 &quot;#8fbcbb&quot;, &quot;#88c0d0&quot;, &quot;#81a1c1&quot;, &quot;#5e81ac&quot;, 
                 &quot;#bf616a&quot;, &quot;#d08770&quot;, &quot;#ebcb8b&quot;, &quot;#a3be8c&quot;, &quot;#b48ead&quot;)

colors_toll_bright &lt;- c(&quot;#4477AA&quot;, &quot;#66CCEE&quot;, &quot;#228833&quot;, &quot;#CCBB44&quot;, &quot;#EE6677&quot;, &quot;#AA3377&quot;, &quot;#BBBBBB&quot;)
colors_toll_mute &lt;- c(&quot;#332288&quot;, &quot;#88CCEE&quot;, &quot;#44AA99&quot;, &quot;#117733&quot;, &quot;#999933&quot;, &quot;#DDCC77&quot;, &quot;#CC6677&quot;, &quot;#882255&quot;, &quot;#AA4499&quot;, &quot;#DDDDDD&quot;)
colors_toll_br &lt;- c(&quot;#364B9A&quot;, &quot;#4A7BB7&quot;, &quot;#6EA6CD&quot;, &quot;#98CAE1&quot;, &quot;#C2E4EF&quot;, &quot;#EAECCC&quot;, &quot;#FEDA8B&quot;, &quot;#FDB366&quot;, &quot;#F67A4B&quot;, &quot;#DD3D2D&quot;, &quot;#A50026&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)
colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

# Variable-specific colors
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8&quot; = &quot;#332288&quot;)
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#737373&quot;, &quot;aIg+H2O2&quot; = &quot;#FDB366&quot;, 
                 &quot;Basal&quot; = &quot;#737373&quot;, &quot;Activated&quot; = &quot;#FDB366&quot;)
colors_cell_stim &lt;- c(&quot;HBL1 - PBS&quot; = &quot;#c994c7&quot;, &quot;HBL1 - aIg+H2O2&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8 - PBS&quot; = &quot;#88CCEE&quot;, &quot;OCI-Ly8 - aIg+H2O2&quot; = &quot;#332288&quot;)
# colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#228833&quot;, &quot;iBTK&quot; = &quot;#4477AA&quot;, &quot;iNFkB&quot; = &quot;#AA3377&quot;)
colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#9dce5c&quot;, &quot;iBTK&quot; = &quot;#5c9dce&quot;, &quot;iNFkB&quot; = &quot;#ce5c9d&quot;)
colors_conc &lt;- c(&quot;#737373&quot;, &quot;#dadaeb&quot;, &quot;#9e9ac8&quot;, &quot;#6a51a3&quot;, &quot;#3f007d&quot;)
colors_iSYK &lt;- colors_green9[c(3, 5, 6, 8, 9)]
colors_iBTK &lt;- colors_blue9[c(3, 5, 6, 8, 9)]
colors_iNFkB &lt;- colors_pr9[c(3, 5, 6, 8, 9)]
colors_conc_stim &lt;- colors_orange9[c(2, 3, 5, 6, 8, 9)]
colors_conc_PBS &lt;- colors_grey9[c(2, 3, 5, 6, 8, 9)]</code></pre>
</div>
<div id="figure-1" class="section level2">
<h2>Figure 1</h2>
<p><em>B-cell activation and inhibition by iBTK results in distinct
signaling states.</em></p>
<p>Input:</p>
<ul>
<li><p>BioRender schematic figures</p></li>
<li><p>DS108 ID-seq data</p>
<ul>
<li>PCA on all samples</li>
</ul></li>
</ul>
<div id="data" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig1_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
fig1_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig1_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig1_label_stim &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
fig1_label_stim_short &lt;- c(&quot;Bas.&quot;, &quot;Act.&quot;)
fig1_inhib &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
fig1_stim_inhib &lt;- c(&quot;Basal\nDMSO ctrl&quot;, &quot;Activated\nDMSO ctrl&quot;, &quot;Basal\n100 uM iBTK&quot;, &quot;Activated\n100 uM iBTK&quot;)
fig1_stim_inhib_label &lt;- c(&quot;Basal\nDMSO\nctrl&quot;, &quot;Activated\nDMSO\nctrl&quot;, &quot;Basal\n100 uM\niBTK&quot;, &quot;Activated\n100 uM\niBTK&quot;)

fig1_conc_select &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
fig1_label_conc_select &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
fig1_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig1_label_conc &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

fig1_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- fig1_data_id  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(fig1_data_id, all_of(fig1_meta_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = fig1_stim), 
         inhibitor = factor(inhibitor, levels = fig1_inhib), 
         inhib_conc = factor(inhib_conc, levels = fig1_conc)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
</div>
<div id="panels" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># # Placeholder
# fig1_panelA &lt;- ggplot() +
#   geom_blank() +
#   scale_x_continuous(limits = c(0, 10)) +
#   scale_y_continuous(limits = c(0, 10)) +
#   # labs(title = &quot;BCR signaling network&quot;) +
#   theme_bw() +
#   theme(axis.text.x = element_blank(),
#         axis.text.y = element_blank(),
#         axis.ticks = element_blank(),
#         panel.grid = element_blank()) +
#   textsize_small
# # fig1_panelA

# PNG
fig1_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/BCR_network_simple_inhib_v1.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
fig1_panelA</code></pre>
<p><img src="figure/v_states_main.Rmd/fig1_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># # Placeholder
# fig1_panelB &lt;- ggplot() +
#   geom_blank() +
#   scale_x_continuous(limits = c(0, 10)) +
#   scale_y_continuous(limits = c(0, 10)) +
#   # labs(title = &quot;BCR signaling network&quot;) +
#   theme_bw() +
#   theme(axis.text.x = element_blank(),
#         axis.text.y = element_blank(),
#         axis.ticks = element_blank(),
#         panel.grid = element_blank()) +
#   textsize_small
# # fig1_panelB

# PNG
fig1_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/signaling_states_IDseq.png&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
fig1_panelB</code></pre>
<p><img src="figure/v_states_main.Rmd/fig1_panelB-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># PNG
fig1_panelB_alt &lt;- magick::image_read(&quot;output/figures/non_R_figs/signaling_states_IDseq_square.png&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
# fig1_panelB_alt</code></pre>
</div>
<div id="panel-c" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data
fig1C_data &lt;- pca_data %&gt;%
  subset(inhibitor == &quot;iBTK&quot; &amp; inhib_text_new != &quot;PBS ctrl&quot;)

fig1C_stim_inhib &lt;- c(&quot;Basal\nDMSO ctrl&quot;, 
                      &quot;Basal\n0.1 uM iBTK&quot;, &quot;Basal\n1 uM iBTK&quot;, &quot;Basal\n10 uM iBTK&quot;, &quot;Basal\n100 uM iBTK&quot;, 
                      &quot;Activated\nDMSO ctrl&quot;, 
                      &quot;Activated\n0.1 uM iBTK&quot;, &quot;Activated\n1 uM iBTK&quot;, &quot;Activated\n10 uM iBTK&quot;, &quot;Activated\n100 uM iBTK&quot;)

fig1C_labels &lt;- c(&quot;Basal\nDMSO ctrl&quot;, &quot;0.1 uM iBTK&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;, 
                  &quot;Activated\nDMSO ctrl&quot;, &quot;0.1 uM iBTK&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig1C_select_colors &lt;- c(&quot;#bdbdbd&quot;, 
                         &quot;#cfd9e0&quot;, &quot;#bcc9d4&quot;, &quot;#a0b3c2&quot;, &quot;#90a1ae&quot;, 
                         &quot;#737373&quot;, 
                         &quot;#bdd7eb&quot;, &quot;#9dc4e1&quot;, &quot;#7cb0d7&quot;, &quot;#5c9dce&quot;)
fig1C_select_shape &lt;- c(16, 
                        16, 16, 16, 16, 
                        17, 
                        17, 17, 17, 17)

names(fig1C_labels) &lt;- fig1C_stim_inhib
names(fig1C_select_colors) &lt;- fig1C_stim_inhib
names(fig1C_select_shape) &lt;- fig1C_stim_inhib

# Figure
fig1_panelC &lt;- ggplot(fig1C_data) + 
  geom_point(aes(x = PC1, y = PC3, shape = stim_inhib, color = stim_inhib), size = 2.5) +
  scale_color_manual(values = fig1C_select_colors, breaks = names(fig1C_select_colors), labels = fig1C_labels, name = &quot;Treatment&quot;) +
  scale_shape_manual(values = fig1C_select_shape, breaks = names(fig1C_select_shape), labels = fig1C_labels, name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) +
  theme_bw() +
  textsize_small + 
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.5, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, axis.title = element_text(size = 8)) + 
  guides(shape = guide_legend(ncol = 1, byrow = TRUE), color = guide_legend(ncol = 1, byrow = TRUE))
fig1_panelC</code></pre>
<p><img src="figure/v_states_main.Rmd/fig1_panelC-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-d" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Idea:
# Visualise proteins contributing to PCs
# Quality of representation: cos2 - size of dot
# Percentage contribution to one PC: contrib - colour of dot

# Select data
data_contrib_2.5p &lt;- as.data.frame(pca_variab$contrib) %&gt;%
  dplyr::filter(Dim.1 &gt;= 2.5 | Dim.2 &gt;= 2.5 | Dim.3 &gt;= 2.5) %&gt;% # | Dim.4 &gt;= 1 | Dim.5 &gt;= 1
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  mutate_all(function(x){x/100}) %&gt;%
  as.matrix()
data_contrib_prot &lt;- factor(rownames(data_contrib_2.5p))

data_cos2_2.5p &lt;- as.data.frame(pca_variab$cos2) %&gt;%
  rownames_to_column(&quot;target_nospace&quot;) %&gt;%
  dplyr::filter(target_nospace %in% data_contrib_prot) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  as.matrix()

# Tidy data format
data_cos2_tidy &lt;- as.data.frame(data_cos2_2.5p) %&gt;%
  rownames_to_column(&quot;target_nospace&quot;) %&gt;%
  pivot_longer(2:4, names_to = &quot;PC&quot;, values_to = &quot;cos2&quot;)
data_contrib_tidy &lt;- as.data.frame(data_contrib_2.5p) %&gt;%
  rownames_to_column(&quot;target_nospace&quot;) %&gt;%
  pivot_longer(2:4, names_to = &quot;PC&quot;, values_to = &quot;contrib&quot;) %&gt;%
  mutate(contrib = contrib * 100)
data_contrib_all &lt;- full_join(data_cos2_tidy, data_contrib_tidy) %&gt;%
  left_join(distinct(dplyr::select(fig1_data_id, c(target_nospace, target))))
fig1D_prot &lt;- sort(unique(data_contrib_all$target))
fig1D_prot_label &lt;- c(&quot;AMPKa1/2\n(T183/T172)&quot;, &quot;BLNK (Y84)&quot;, &quot;BTK (Y551)&quot;, &quot;CD38&quot;, &quot;CD79a&quot;, &quot;GAPDH&quot;, &quot;IgM&quot;, &quot;LYN (Y397)&quot;, &quot;PKC-b1&quot;, &quot;S6 (S235/S236)&quot;, &quot;S6 (S240/S244)&quot;, &quot;SRC (Y416)&quot;, &quot;VAV1 (Y174)&quot;, &quot;ZAP70 (Y319)&quot;)

# Figure
fig1_panelD &lt;- ggplot() +
  geom_point(data = data_contrib_all, aes(x = PC, y = factor(target, levels = rev(fig1D_prot), labels = rev(fig1D_prot_label)), size = cos2, fill = contrib), color = &quot;black&quot;, pch = 21) +
  scale_size_area(max_size = 4, breaks = c(0, 0.25, 0.5, 0.75, 1), name = &quot;Quality of\nrepresentation\nin PCs&quot;) +
  scale_fill_gradient(low = &quot;#FEDA8B&quot;, high = &quot;#73001a&quot;, name = &quot;Percentage\ncontribution\nwithin one PC&quot;) + #7f2704 #67000d
  labs(x = &quot;&quot;, y = &quot;&quot;) + 
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), axis.text.y = element_text(size = 6), legend.title = element_text(size = 7), legend.position = &quot;right&quot;) + 
  # guides(size = guide_legend(ncol = 1, byrow = TRUE)) +
  textsize_small #+ 
  # theme(legend.title = element_text(size = 8), axis.text.y = element_text(size = 8))
fig1_panelD</code></pre>
<p><img src="figure/v_states_main.Rmd/fig1_panelD-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-e" class="section level4">
<h4>Panel E</h4>
<pre class="r"><code># Select data
fig1E_conc &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM iBTK&quot;)
fig1E_label_conc &lt;- c(&quot;DMSO\nctrl&quot;, &quot;100 uM\niBTK&quot;)
fig1E_proteins &lt;- c(&quot;VAV1_Y174&quot;, &quot;IgM&quot;, &quot;S6_S235_S236&quot;) #, &quot;BLNK_Y84&quot;, &quot;AMPKa1_2_T183_T172&quot;, &quot;CD79a&quot;
fig1E_proteins_name &lt;- c(&quot;VAV1 (Y174)&quot;, &quot;IgM&quot;, &quot;S6 (S235/S236)&quot;) #, &quot;BLNK (SLP-65) (Y84)&quot;, &quot;AMPKa1/2 (T183/T172)&quot;, &quot;CD79a (Iga)&quot;
# fig1E_proteins_label &lt;- c(&quot;VAV1 (Y174)&quot;, &quot;IgM&quot;, &quot;S6 (S235/S236)&quot;, &quot;BLNK (Y84)&quot;, &quot;AMPKa1/2 (T183/T172)&quot;, &quot;CD79a&quot;)
fig1E_facet_labels &lt;- c(&quot;PC1: VAV1 (Y174)&quot;, &quot;PC2: IgM&quot;, &quot;PC3: S6 (S235/S236)&quot;)
fig1E_data &lt;- fig1_data_id %&gt;%
  subset(inhibitor == &quot;iBTK&quot; &amp; target_nospace %in% fig1E_proteins &amp; inhib_text_new %in% fig1E_conc) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig1E_proteins))
fig1E_mean &lt;- fig1_mean_id %&gt;%
  subset(inhibitor == &quot;iBTK&quot; &amp; target_nospace %in% fig1E_proteins &amp; inhib_text_new %in% fig1E_conc)

# Statistics
# Kruskal-Wallis test
fig1E_kruskal &lt;- fig1E_data %&gt;%
  group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ stim_inhib)
fig1E_kruskal</code></pre>
<pre><code># A tibble: 3 × 7
  target_nospace .y.             n statistic    df      p method        
* &lt;fct&gt;          &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;         
1 VAV1_Y174      counts_norm    12      9.97     3 0.0188 Kruskal-Wallis
2 IgM            counts_norm    12      8.13     3 0.0434 Kruskal-Wallis
3 S6_S235_S236   counts_norm    12      9.67     3 0.0216 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
fig1E_dunn &lt;- fig1E_data %&gt;%
  group_by(target_nospace) %&gt;%
  dunn_test(counts_norm ~ stim_inhib, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group1 == &quot;Activated\nDMSO ctrl&quot; &amp; group2 == &quot;Basal\nDMSO ctrl&quot;) |
      (group1 == &quot;Activated\n100 uM iBTK&quot; &amp; group2 == &quot;Basal\n100 uM iBTK&quot;) |
      (group1 == &quot;Basal\n100 uM iBTK&quot; &amp; group2 == &quot;Basal\nDMSO ctrl&quot;) |
      (group1 == &quot;Activated\n100 uM iBTK&quot; &amp; group2 == &quot;Activated\nDMSO ctrl&quot;)
      ) %&gt;%
  mutate(group1 = case_when(group1 == &quot;Basal\n100 uM iBTK&quot; ~ &quot;Basal\n100 uM\niBTK&quot;, 
                            group1 == &quot;Activated\n100 uM iBTK&quot; ~ &quot;Activated\n100 uM\niBTK&quot;, 
                            group1 == &quot;Basal\nDMSO ctrl&quot; ~ &quot;Basal\nDMSO\nctrl&quot;, 
                            group1 == &quot;Activated\nDMSO ctrl&quot; ~ &quot;Activated\nDMSO\nctrl&quot;, 
                            .default = group1), 
         group2 = case_when(group2 == &quot;Basal\n100 uM iBTK&quot; ~ &quot;Basal\n100 uM\niBTK&quot;, 
                            group2 == &quot;Activated\n100 uM iBTK&quot; ~ &quot;Activated\n100 uM\niBTK&quot;, 
                            group2 == &quot;Basal\nDMSO ctrl&quot; ~ &quot;Basal\nDMSO\nctrl&quot;, 
                            group2 == &quot;Activated\nDMSO ctrl&quot; ~ &quot;Activated\nDMSO\nctrl&quot;, 
                            .default = group2))
fig1E_dunn</code></pre>
<pre><code># A tibble: 12 × 10
   target_nospace .y.         group1  group2    n1    n2 statistic      p  p.adj
   &lt;fct&gt;          &lt;chr&gt;       &lt;chr&gt;   &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;
 1 VAV1_Y174      counts_norm &quot;Activ… &quot;Acti…     3     3     1.02  0.308  0.308 
 2 VAV1_Y174      counts_norm &quot;Activ… &quot;Basa…     3     3    -1.92  0.0542 0.0542
 3 VAV1_Y174      counts_norm &quot;Activ… &quot;Basa…     3     3    -2.15  0.0315 0.0315
 4 VAV1_Y174      counts_norm &quot;Basal… &quot;Basa…     3     3     0.793 0.428  0.428 
 5 IgM            counts_norm &quot;Activ… &quot;Acti…     3     3    -1.59  0.113  0.113 
 6 IgM            counts_norm &quot;Activ… &quot;Basa…     3     3    -1.70  0.0894 0.0894
 7 IgM            counts_norm &quot;Activ… &quot;Basa…     3     3    -1.25  0.213  0.213 
 8 IgM            counts_norm &quot;Basal… &quot;Basa…     3     3    -1.13  0.258  0.258 
 9 S6_S235_S236   counts_norm &quot;Activ… &quot;Acti…     3     3     1.81  0.0700 0.0700
10 S6_S235_S236   counts_norm &quot;Activ… &quot;Basa…     3     3    -1.02  0.308  0.308 
11 S6_S235_S236   counts_norm &quot;Activ… &quot;Basa…     3     3    -0.566 0.571  0.571 
12 S6_S235_S236   counts_norm &quot;Basal… &quot;Basa…     3     3     2.26  0.0235 0.0235
# ℹ 1 more variable: p.adj.signif &lt;chr&gt;</code></pre>
<pre class="r"><code># Manual stat table (from DEA)
fig1E_manual_stat &lt;- tibble(target_nospace = fig1E_dunn$target_nospace, 
                            .y. = &quot;counts_norm&quot;, 
                            group1 = fig1E_dunn$group1, 
                            group2 = fig1E_dunn$group2, 
                            p.adj.signif = c(&quot;*&quot;, &quot;*&quot;, &quot;*&quot;, &quot;ns&quot;, 
                                             &quot;*&quot;, &quot;*&quot;, &quot;*&quot;, &quot;*&quot;, 
                                             &quot;*&quot;, &quot;ns&quot;, &quot;ns&quot;, &quot;*&quot;), 
                            y.position = rep(c(6500, 6500, 5000), each = 4))
fig1E_manual_stat</code></pre>
<pre><code># A tibble: 12 × 6
   target_nospace .y.         group1              group2 p.adj.signif y.position
   &lt;fct&gt;          &lt;chr&gt;       &lt;chr&gt;               &lt;chr&gt;  &lt;chr&gt;             &lt;dbl&gt;
 1 VAV1_Y174      counts_norm &quot;Activated\n100 uM… &quot;Acti… *                  6500
 2 VAV1_Y174      counts_norm &quot;Activated\n100 uM… &quot;Basa… *                  6500
 3 VAV1_Y174      counts_norm &quot;Activated\nDMSO\n… &quot;Basa… *                  6500
 4 VAV1_Y174      counts_norm &quot;Basal\n100 uM\niB… &quot;Basa… ns                 6500
 5 IgM            counts_norm &quot;Activated\n100 uM… &quot;Acti… *                  6500
 6 IgM            counts_norm &quot;Activated\n100 uM… &quot;Basa… *                  6500
 7 IgM            counts_norm &quot;Activated\nDMSO\n… &quot;Basa… *                  6500
 8 IgM            counts_norm &quot;Basal\n100 uM\niB… &quot;Basa… *                  6500
 9 S6_S235_S236   counts_norm &quot;Activated\n100 uM… &quot;Acti… *                  5000
10 S6_S235_S236   counts_norm &quot;Activated\n100 uM… &quot;Basa… ns                 5000
11 S6_S235_S236   counts_norm &quot;Activated\nDMSO\n… &quot;Basa… ns                 5000
12 S6_S235_S236   counts_norm &quot;Basal\n100 uM\niB… &quot;Basa… *                  5000</code></pre>
<pre class="r"><code># Figure
fig1_panelE &lt;- ggplot() +
  geom_jitter(data = fig1E_data,
              aes(factor(stim_inhib, levels = fig1_stim_inhib, labels = fig1_stim_inhib_label), counts_norm, color = factor(stimulus, levels = fig1_stim)), 
              alpha = 1, size = 1.5, width = 0.1) +
  geom_errorbar(data = fig1E_mean,
                aes(factor(stim_inhib, levels = fig1_stim_inhib, labels = fig1_stim_inhib_label),
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stimulus, levels = fig1_stim),
                    color = factor(stimulus, levels = fig1_stim)),
                width = 0.5, size = 1) +
  facet_wrap(vars(factor(target_nospace, levels = fig1E_proteins, labels = fig1E_facet_labels)), nrow = 1, scales = &quot;free_y&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig1_stim, labels = fig1_label_stim, name = &quot;Stimulus&quot;) +
  labs(x = element_blank(), y = &quot;Normalised counts&quot;) +
  scale_y_continuous(expand = c(0.15, 0.15)) +
  # coord_cartesian(ylim = c(1500, 10200)) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;right&quot;, axis.text.x = element_text(size = 6.5)) +
  stat_pvalue_manual(fig1E_manual_stat, label = &quot;p.adj.signif&quot;, tip.length = 0.05,
                     step.increase = 0.25, group = &quot;target_nospace&quot;, step.group.by = &quot;target_nospace&quot;, size = 3, vjust = 0.1) 
fig1_panelE</code></pre>
<p><img src="figure/v_states_main.Rmd/fig1_panelE-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
# fig1_AB &lt;- plot_grid(fig1_panelA, fig1_panelB, labels = PANEL_labels[c(1,2)], ncol = 2, rel_widths = c(1, 3.9), scale = 0.98, label_size = 10)
fig1_AB &lt;- plot_grid(fig1_panelA, fig1_panelB_alt, labels = PANEL_labels[c(1,2)], nrow = 1, rel_widths = c(1, 1.35), label_size = 10)
fig1_CD &lt;- plot_grid(fig1_panelC, fig1_panelD, labels = PANEL_labels[c(3,4)], ncol = 2, rel_widths = c(1.4, 1), label_size = 10)
fig1 &lt;- plot_grid(fig1_AB, fig1_CD, fig1_panelE, labels = c(NA, NA, PANEL_labels[c(5)]), ncol = 1, rel_heights = c(1.5, 1.4, 1), label_size = 10)
fig1</code></pre>
<p><img src="figure/v_states_main.Rmd/fig1_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig1,
  filename = &quot;output/figures/DLBCL_states/main_fig1.pdf&quot;,
  width = 6,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig1,
  filename = &quot;output/figures/DLBCL_states/main_fig1.jpg&quot;,
  width = 6,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>B-cell activation and inhibition by iBTK results in distinct
signaling states.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig1&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9436998 504.0   16905207 902.9 16905207 902.9
Vcells 32070216 244.7   95666217 729.9 93646971 714.5</code></pre>
</div>
</div>
<div id="figure-2" class="section level2">
<h2>Figure 2</h2>
<p><em>Network-wide signaling inhibition along different repression axes
by iSYK and iNFκB.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li><p>PCA on all samples</p></li>
<li><p>Count data</p></li>
</ul></li>
</ul>
<div id="data-1" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
fig2_data_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
fig2_mean_id &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig2_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig2_label_stim &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
fig2_inhib &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

# fig2_conc_select &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
# fig2_label_conc_select &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
fig2_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig2_label_conc &lt;- c(&quot;DMSO\nctrl&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
fig2_label_conc_alt &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

fig2_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- fig2_data_id  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(fig2_data_id, all_of(fig2_meta_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = fig2_stim), 
         inhibitor = factor(inhibitor, levels = fig2_inhib), 
         inhib_conc = factor(inhib_conc, levels = fig2_conc)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
</div>
<div id="panels-1" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-1" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
# fig2A_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;#bdbdbd&quot;, &quot;100 uM iSYK&quot; = &quot;#9dce5c&quot;, &quot;100 uM iBTK&quot; = &quot;#5c9dce&quot;, &quot;100 uM iNFkB&quot; = &quot;#ce5c9d&quot;)
pca_data_select &lt;- pca_data %&gt;%
  dplyr::filter(!(stimulus == &quot;PBS&quot; &amp; inhib_conc %in% fig2_conc[2:5])) %&gt;%
  drop_na() %&gt;%
  dplyr::arrange(factor(stimulus, levels = fig2_stim), factor(inhibitor, levels =  fig2_inhib))

pca_data_iSYK &lt;- pca_data_select %&gt;%
  dplyr::filter(inhib_new == &quot;DMSO ctrl&quot; | inhib_new == &quot;iBTK&quot; | inhib_new == &quot;iSYK&quot;) %&gt;%
  mutate(inhib_plot = &quot;iSYK vs iBTK&quot;)
pca_data_iNFkB &lt;- pca_data_select %&gt;%
  dplyr::filter(inhib_new == &quot;DMSO ctrl&quot; | inhib_new == &quot;iBTK&quot; | inhib_new == &quot;iNFkB&quot;) %&gt;%
  mutate(inhib_plot = &quot;iNFkB vs iBTK&quot;)
pca_data_select_all &lt;- full_join(pca_data_iSYK, pca_data_iNFkB)

pca_legend_shapes &lt;- c(16, 16, 16, 
                       17, 17, 17, 17, 17, 
                       17, 17, 17, 17, 17, 
                       17, 17, 17, 17, 17)
pca_legend_values &lt;- c(&quot;#bdbdbd&quot;, &quot;#bdbdbd&quot;, &quot;#bdbdbd&quot;, 
                       &quot;#737373&quot;, &quot;#d7ebbd&quot;, &quot;#c4e19d&quot;,  &quot;#b0d77c&quot;, &quot;#9dce5c&quot;, 
                       &quot;#737373&quot;, &quot;#bdd7eb&quot;, &quot;#9dc4e1&quot;,  &quot;#7cb0d7&quot;, &quot;#5c9dce&quot;, 
                       &quot;#737373&quot;, &quot;#ebbdd7&quot;, &quot;#e19dc4&quot;,  &quot;#d77cb0&quot;, &quot;#ce5c9d&quot;)
pca_legend_levels &lt;- c(&quot;0 uM iSYK (DMSO) PBS&quot;, &quot;0 uM iBTK (DMSO) PBS&quot;, &quot;0 uM iNFkB (DMSO) PBS&quot;, 
                       &quot;0 uM iSYK (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iSYK aIg+H2O2&quot;, &quot;1 uM iSYK aIg+H2O2&quot;, &quot;10 uM iSYK aIg+H2O2&quot;, &quot;100 uM iSYK aIg+H2O2&quot;, 
                       &quot;0 uM iBTK (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iBTK aIg+H2O2&quot;, &quot;1 uM iBTK aIg+H2O2&quot;, &quot;10 uM iBTK aIg+H2O2&quot;, &quot;100 uM iBTK aIg+H2O2&quot;, 
                       &quot;0 uM iNFkB (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iNFkB aIg+H2O2&quot;, &quot;1 uM iNFkB aIg+H2O2&quot;, &quot;10 uM iNFkB aIg+H2O2&quot;, &quot;100 uM iNFkB aIg+H2O2&quot;)
names(pca_legend_shapes) &lt;- pca_legend_levels
names(pca_legend_values) &lt;- pca_legend_levels
pca_legend_breaks &lt;- c(&quot;0 uM iBTK (DMSO) PBS&quot;, &quot;0 uM iBTK (DMSO) aIg+H2O2&quot;, 
                       &quot;0.1 uM iBTK aIg+H2O2&quot;, &quot;1 uM iBTK aIg+H2O2&quot;, &quot;10 uM iBTK aIg+H2O2&quot;, &quot;100 uM iBTK aIg+H2O2&quot;, 
                       &quot;100 uM iSYK aIg+H2O2&quot;, &quot;100 uM iNFkB aIg+H2O2&quot;)
pca_legend_labels &lt;- c(&quot;0 uM - Basal&quot;, &quot;0 uM - Activated&quot;, &quot;0.1 uM - Activated&quot;, &quot;1 uM - Activated&quot;, &quot;10 uM - Activated&quot;, &quot;100 uM - Activated&quot;)
pca_legend_labels_alt &lt;- c(&quot;0 uM (basal)&quot;, &quot;0 uM (activated)&quot;, 
                           &quot;0.1 uM iBTK&quot;, &quot;1 uM iBTK&quot;, &quot;10 uM iBTK&quot;, &quot;100 uM iBTK&quot;, 
                           &quot;iSYK&quot;, &quot;iNFkB&quot;)

# Figure
fig2_panelA &lt;- ggplot(pca_data_select_all) + 
  geom_point(aes(x = PC1, y = PC3, 
                 shape = paste(inhib_conc_text, stimulus), 
                 color = paste(inhib_conc_text, stimulus), 
                 alpha = inhib_new), 
             size = 3) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels_alt,
                     name = &quot;Inhibitor\nconc.&quot;) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels_alt,
                     name = &quot;Inhibitor\nconc.&quot;) +
  scale_alpha_manual(values = c(0.8, 0.4, 1, 1)) +
  facet_wrap(vars(factor(inhib_plot, levels = c(&quot;iSYK vs iBTK&quot;, &quot;iNFkB vs iBTK&quot;))), ncol = 2) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;) +
  guides(alpha = &quot;none&quot;)
fig2_panelA</code></pre>
<p><img src="figure/v_states_main.Rmd/fig2_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-1" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
fig2B_prot &lt;- c(
  &quot;JAK1_Y1022_Y1023&quot;
  # &quot;IKKa_b_S176_S180&quot;
  # &quot;CD79a_Y182&quot;
  # &quot;S6_S235_S236&quot;
  
  # &quot;S6_S240_S244&quot; 
  # &quot;BLNK_Y84&quot;
  # &quot;LYN_Y397&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
  # &quot;CDK1_T14_Y15&quot;
  # &quot;PYK2_Y402&quot;
) 
fig2B_prot_label &lt;- c(
  &quot;JAK1 (Y1022/Y1023)&quot;
  # &quot;IKKa/b (S176/S180)&quot;
  # &quot;CD79a (Y182)&quot;
  # &quot;S6 (S235/S236)&quot;
  )
fig2B_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2B_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;)
fig2B_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2B_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig2_panelB &lt;- ggplot() +
  geom_point(data = fig2B_data, 
             aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = fig2B_mean, 
            aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig2_inhib)), ncol = 1) +
  # facet_grid(target_nospace~inhibitor, scales = &quot;free_y&quot;) +
  scale_shape_manual(values = c(17, 16), breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalised counts&quot;, title = fig2B_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), 
        axis.text.x = element_text(size = 6), axis.title = element_text(size = 8), legend.title = element_text(size = 8), plot.title = element_text(size = 8)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig2_panelB</code></pre>
<p><img src="figure/v_states_main.Rmd/fig2_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-c-1" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data
fig2C_prot &lt;- c(
  # &quot;JAK1_Y1022_Y1023&quot;
  # &quot;IKKa_b_S176_S180&quot;
  &quot;CD79a_Y182&quot;
  # &quot;S6_S235_S236&quot;
  
  # &quot;S6_S240_S244&quot; 
  # &quot;BLNK_Y84&quot;
  # &quot;LYN_Y397&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
  # &quot;CDK1_T14_Y15&quot;
  # &quot;PYK2_Y402&quot;
) 
fig2C_prot_label &lt;- c(
  # &quot;JAK1 (Y1022/Y1023)&quot;
  # &quot;IKKa/b (S176/S180)&quot;
  &quot;CD79a (Y182)&quot;
  # &quot;S6 (S235/S236)&quot;
  )
fig2C_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2C_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;)
fig2C_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2C_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig2_panelC &lt;- ggplot() +
  geom_point(data = fig2C_data, 
             aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = fig2C_mean, 
            aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig2_inhib)), ncol = 1) +
  # facet_grid(target_nospace~inhibitor, scales = &quot;free_y&quot;) +
  scale_shape_manual(values = c(17, 16), breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalised counts&quot;, title = fig2C_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), 
        axis.text.x = element_text(size = 6), axis.title = element_text(size = 8), legend.title = element_text(size = 8), plot.title = element_text(size = 8)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig2_panelC</code></pre>
<p><img src="figure/v_states_main.Rmd/fig2_panelC-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-d-1" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Select data
fig2D_prot &lt;- c(
  # &quot;JAK1_Y1022_Y1023&quot;
  # &quot;IKKa_b_S176_S180&quot;
  # &quot;CD79a_Y182&quot;
  &quot;S6_S235_S236&quot;
  # &quot;CREB_S133&quot;
  
  # &quot;S6_S240_S244&quot; 
  # &quot;BLNK_Y84&quot;
  # &quot;LYN_Y397&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
  # &quot;CDK1_T14_Y15&quot;
  # &quot;PYK2_Y402&quot;
) 
fig2D_prot_label &lt;- c(
  # &quot;JAK1 (Y1022/Y1023)&quot;
  # &quot;IKKa/b (S176/S180)&quot;
  # &quot;CD79a (Y182)&quot;
  &quot;S6 (S235/S236)&quot;
  # &quot;CREB (S133)&quot;
  )
fig2D_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2D_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;)
fig2D_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2D_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig2_panelD &lt;- ggplot() +
  geom_point(data = fig2D_data, 
             aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = fig2D_mean, 
            aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig2_inhib)), ncol = 1) +
  # facet_grid(target_nospace~inhibitor, scales = &quot;free_y&quot;) +
  scale_shape_manual(values = c(17, 16), breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalised counts&quot;, title = fig2D_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), 
        axis.text.x = element_text(size = 6), axis.title = element_text(size = 8), legend.title = element_text(size = 8), plot.title = element_text(size = 8)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig2_panelD</code></pre>
<p><img src="figure/v_states_main.Rmd/fig2_panelD-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-e-1" class="section level4">
<h4>Panel E</h4>
<pre class="r"><code># Select data
fig2E_prot &lt;- c(
  # &quot;JAK1_Y1022_Y1023&quot;
  &quot;IKKa_b_S176_S180&quot;
  # &quot;CD79a_Y182&quot;
  # &quot;S6_S235_S236&quot;
  
  # &quot;S6_S240_S244&quot; 
  # &quot;BLNK_Y84&quot;
  # &quot;LYN_Y397&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
  # &quot;CDK1_T14_Y15&quot;
  # &quot;PYK2_Y402&quot;
) 
fig2E_prot_label &lt;- c(
  # &quot;JAK1 (Y1022/Y1023)&quot;
  &quot;IKKa/b (S176/S180)&quot;
  # &quot;CD79a (Y182)&quot;
  # &quot;S6 (S235/S236)&quot;
  )
fig2E_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2E_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;)
fig2E_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2E_prot &amp; inhib_conc %in% fig2_conc &amp; cell_line == &quot;HBL1&quot;) 

# Figure
fig2_panelE &lt;- ggplot() +
  geom_point(data = fig2E_data, 
             aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = fig2E_mean, 
            aes(factor(inhib_conc, levels = fig2_conc, labels = fig2_label_conc), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig2_inhib)), ncol = 1) +
  # facet_grid(target_nospace~inhibitor, scales = &quot;free_y&quot;) +
  scale_shape_manual(values = c(17, 16), breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = fig2_stim, labels = fig2_label_stim, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalised counts&quot;, title = fig2E_prot_label) +
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), 
        axis.text.x = element_text(size = 6), axis.title = element_text(size = 8), legend.title = element_text(size = 8), plot.title = element_text(size = 8)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE))
fig2_panelE</code></pre>
<p><img src="figure/v_states_main.Rmd/fig2_panelE-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-1" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig2_BCDE &lt;- plot_grid(fig2_panelB, fig2_panelC, fig2_panelD, labels = PANEL_labels[c(2:4)], nrow = 1, rel_widths = c(1, 1, 1), label_size = 10, align = &quot;h&quot;)
fig2 &lt;- plot_grid(fig2_panelA, fig2_BCDE, labels = PANEL_labels[c(1)], ncol = 1, rel_heights = c(1, 1.4), label_size = 10)
fig2</code></pre>
<p><img src="figure/v_states_main.Rmd/fig2_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig2,
  filename = &quot;output/figures/DLBCL_states/main_fig2.pdf&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig2,
  filename = &quot;output/figures/DLBCL_states/main_fig2.jpg&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Network-wide signaling inhibition along different repression axes
by iSYK and iNFκB.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig2&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9415254 502.9   16905207 902.9 16905207 902.9
Vcells 16534097 126.2   76532974 584.0 93646971 714.5</code></pre>
</div>
</div>
<div id="figure-3" class="section level2">
<h2>Figure 3</h2>
<p><em>Inhibitor combination treatments effectively induce repressed
signaling states.</em></p>
<p>Input:</p>
<ul>
<li><p>DS113 ID-seq data</p>
<ul>
<li>PCA on all samples</li>
</ul></li>
</ul>
<div id="data-2" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS113 (counts per well/sample and mean per condition)
fig3_data_id &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
fig3_mean_id &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>fig3_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig3_label_stim &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig3_legend_label &lt;- &quot;iSYK + iBTK + iNFkB&quot;
fig3_inhib_group &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)
fig3_inhib &lt;- c(&quot;DMSO ctrl&quot;, 
                &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                &quot;iSYK + iBTK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                &quot;iSYK + iBTK + iNFkB&quot;)
fig3_label_inhib &lt;- c(&quot;DMSO ctrl&quot;, 
                      &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                      &quot;iBTK+ iSYK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                      &quot;iBTK + iSYK + iNFkB&quot;)
fig3_conc_single &lt;- c(0, 1, 2.5, 5, 10)
fig3_conc_all1 &lt;- c(&quot;0 + 0 + 0 uM&quot;, 
                    &quot;0 + 10 + 0 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, 
                    &quot;2.5 + 2.5 + 0 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, 
                    &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, 
                    &quot;1 + 5 + 1 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, 
                    &quot;10 + 10 + 10 uM (nostain)&quot;)
fig3_conc_all2 &lt;- c(&quot;DMSO ctrl&quot;, 
                    &quot;10 uM iBTK&quot;, &quot;10 uM iSYK&quot;, &quot;10 uM iNFkB&quot;, 
                    &quot;2.5 uM of iSYK + iBTK&quot;, &quot;2.5 uM of iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iNFkB&quot;, 
                    &quot;1 uM of iSYK + iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;10 uM of iSYK + iBTK + iNFkB&quot;, 
                    &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)

fig3_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- fig3_data_id  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(fig3_data_id, all_of(fig3_meta_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = fig3_stim), 
         inhibitor = factor(inhibitor, levels = fig3_inhib), 
         inhib_conc = factor(inhib_conc, levels = fig3_conc_all1)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  # gather(key = component, value = PC_value, -plate_well) %&gt;%
  left_join(coldata)</code></pre>
</div>
<div id="panels-2" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-2" class="section level4">
<h4>Panel A</h4>
<p>PCA visualisation of the DMSO ctrl and single inhibitor
treatments</p>
<pre class="r"><code># Data
fig3A_group_select &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;)
fig3A_data &lt;- pca_data %&gt;%
   dplyr::filter(inhib_group %in% fig3A_group_select)

# fig3_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;#bdbdbd&quot;, &quot;100 uM iSYK&quot; = &quot;#9dce5c&quot;, &quot;100 uM iBTK&quot; = &quot;#5c9dce&quot;, &quot;100 uM iNFkB&quot; = &quot;#ce5c9d&quot;)

pca_conditions &lt;- c(&quot;Basal\nDMSO ctrl&quot;, &quot;Activated\nDMSO ctrl&quot;, 
                    &quot;Activated\n10 uM iBTK&quot;, &quot;Activated\n10 uM iSYK&quot;, &quot;Activated\n10 uM iNFkB&quot;, 
                    &quot;Activated\n2.5 uM of iSYK + iBTK&quot;, &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;, &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;, 
                    &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;, &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;, &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;, 
                    &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)

pca_legend_alpha &lt;- c(&quot;DMSO ctrl&quot; = 0.8, &quot;single&quot; = 0.4, 
                      &quot;two&quot; = 1, &quot;three_equal&quot; = 1, &quot;three_mix&quot; = 1)
pca_legend_shapes &lt;- c(16, 17, 
                       17, 17, 17, 
                       17, 17, 17, 
                       17, 17, 17, 16, 
                       17, 17, 17)
pca_legend_values &lt;- c(&quot;#bdbdbd&quot;, &quot;#737373&quot;, 
                       &quot;#5c9dce&quot;, &quot;#9dce5c&quot;, &quot;#ce5c9d&quot;, 
                       &quot;#5cce8d&quot;, &quot;#8d5cce&quot;, &quot;#ce8d5c&quot;, 
                       &quot;#d3bceb&quot;, &quot;#ae82da&quot;, &quot;#8351b6&quot;,  &quot;#857892&quot;, 
                       &quot;#729cb9&quot;, &quot;#9bb677&quot;, &quot;#b6779b&quot;)
names(pca_legend_shapes) &lt;- pca_conditions
names(pca_legend_values) &lt;- pca_conditions
pca_legend_breaks &lt;- c(
  &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Activated\nDMSO ctrl&quot;, 
  &quot;Activated\n10 uM iBTK&quot;,
  &quot;Activated\n10 uM iSYK&quot;,
  &quot;Activated\n10 uM iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot;,
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;,
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;
)
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO basal)&quot;, 
  &quot;Activated\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO activated)&quot;, 
  &quot;Activated\n10 uM iBTK&quot; = &quot;10       -           -&quot;,
  &quot;Activated\n10 uM iSYK&quot; = &quot;-          10        -&quot;,
  &quot;Activated\n10 uM iNFkB&quot; = &quot;-          -          10&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5      2.5       -&quot;,
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5      -           2.5&quot;,
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;-          2.5       2.5&quot;,
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1         1          1&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5      2.5       2.5&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10&quot;,
  &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10\n(basal)&quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5          1         1&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;1          5         1&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;1          1         5&quot;
)
pca_legend_title &lt;- &quot;     Treatment\n      iBTK + iSYK + iNFkB\n      (uM)&quot;

# Figure
fig3_panelA &lt;- ggplot(fig3A_data) + 
  geom_point(aes(x = PC1, y = PC2, 
                 shape = stim_inhib, 
                 color = stim_inhib), #alpha = inhib_group
             size = 2) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  facet_wrap(vars(factor(experiment, labels = &quot;Single inhibitor&quot;)), ncol = 1) +
  coord_cartesian(xlim = c(-2300, 6400), ylim = c(-3400, 1900), expand = FALSE) +
  xlab(paste0(&quot;PC1 (&quot;, pca_expl_var[1], &quot;%)&quot;)) +
  ylab(paste0(&quot;PC2 (&quot;, pca_expl_var[2], &quot;%)&quot;)) + 
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;, 
        axis.title = element_text(size = 7), legend.title = element_text(size = 7), legend.text = element_text(size = 7), strip.text.x = element_text(margin = margin(0.1, 0, 0.1, 0, &quot;cm&quot;))) +
  # guides(alpha = &quot;none&quot;, shape = guide_legend(ncol = 1, byrow = TRUE), color = guide_legend(ncol = 1, byrow = TRUE))
  guides(alpha = &quot;none&quot;)
fig3_panelA</code></pre>
<p><img src="figure/v_states_main.Rmd/fig3_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-b-2" class="section level4">
<h4>Panel B</h4>
<p>PCA visualisation of the DMSO ctrl and two inhibitor treatments</p>
<pre class="r"><code># Data
fig3B_group_select &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;)
fig3B_data &lt;- pca_data %&gt;%
   dplyr::filter(inhib_group %in% fig3B_group_select)

pca_legend_breaks &lt;- c(
  &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Activated\nDMSO ctrl&quot;, 
  # &quot;Activated\n10 uM iBTK&quot;,
  # &quot;Activated\n10 uM iSYK&quot;,
  # &quot;Activated\n10 uM iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot;,
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;
  # &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;,
  # &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;,
  # &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;
)
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO basal)&quot;, 
  &quot;Activated\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO activated)&quot;, 
  # &quot;Activated\n10 uM iBTK&quot; = &quot;10       -           -&quot;,
  # &quot;Activated\n10 uM iSYK&quot; = &quot;-          10        -&quot;,
  # &quot;Activated\n10 uM iNFkB&quot; = &quot;-          -          10&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5      2.5       -&quot;,
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5      -           2.5&quot;,
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;-          2.5       2.5&quot;
  # &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1         1          1&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5      2.5       2.5&quot;,
  # &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10&quot;,
  # &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10\n(basal)&quot;,
  # &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5          1         1&quot;,
  # &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;1          5         1&quot;,
  # &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;1          1         5&quot;
)

# Figure
fig3_panelB &lt;- ggplot(fig3B_data) + 
  geom_point(aes(x = PC1, y = PC2, 
                 shape = stim_inhib, 
                 color = stim_inhib, 
                 alpha = inhib_group), #alpha = inhib_group
             size = 2) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_alpha_manual(values = pca_legend_alpha) +
  facet_wrap(vars(factor(experiment, labels = &quot;Two inhibitors&quot;)), ncol = 1) +
  coord_cartesian(xlim = c(-2300, 6400), ylim = c(-3400, 1900), expand = FALSE) +
  xlab(paste0(&quot;PC1 (&quot;, pca_expl_var[1], &quot;%)&quot;)) +
  ylab(paste0(&quot;PC2 (&quot;, pca_expl_var[2], &quot;%)&quot;)) + 
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;, 
        axis.title = element_text(size = 7), legend.title = element_text(size = 7), legend.text = element_text(size = 7), strip.text.x = element_text(margin = margin(0.1, 0, 0.1, 0, &quot;cm&quot;))) +
  # guides(alpha = &quot;none&quot;, shape = guide_legend(ncol = 1, byrow = TRUE), color = guide_legend(ncol = 1, byrow = TRUE))
  guides(alpha = &quot;none&quot;)
fig3_panelB</code></pre>
<p><img src="figure/v_states_main.Rmd/fig3_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-c-2" class="section level4">
<h4>Panel C</h4>
<p>PCA visualisation of the DMSO ctrl and three equal inhibitor
treatments</p>
<pre class="r"><code># Data
fig3C_group_select &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;three_equal&quot;)
fig3C_data &lt;- pca_data %&gt;%
   dplyr::filter(inhib_group %in% fig3C_group_select &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;)

pca_legend_breaks &lt;- c(
  &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Activated\nDMSO ctrl&quot;, 
  # &quot;Activated\n10 uM iBTK&quot;,
  # &quot;Activated\n10 uM iSYK&quot;,
  # &quot;Activated\n10 uM iNFkB&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK&quot;,
  # &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;,
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;
  # &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;,
  # &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;,
  # &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;
)
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO basal)&quot;, 
  &quot;Activated\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO activated)&quot;, 
  # &quot;Activated\n10 uM iBTK&quot; = &quot;10       -           -&quot;,
  # &quot;Activated\n10 uM iSYK&quot; = &quot;-          10        -&quot;,
  # &quot;Activated\n10 uM iNFkB&quot; = &quot;-          -          10&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5      2.5       -&quot;,
  # &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5      -           2.5&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;-          2.5       2.5&quot;,
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1         1          1&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5      2.5       2.5&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10&quot;
  # &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10\n(basal)&quot;,
  # &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5          1         1&quot;,
  # &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;1          5         1&quot;,
  # &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;1          1         5&quot;
)

# Figure
fig3_panelC &lt;- ggplot(fig3C_data) + 
  geom_point(aes(x = PC1, y = PC2, 
                 shape = stim_inhib, 
                 color = stim_inhib, 
                 alpha = inhib_group), #alpha = inhib_group
             size = 2) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_alpha_manual(values = pca_legend_alpha) +
  facet_wrap(vars(factor(experiment, labels = &quot;Three inhibitors equal&quot;)), ncol = 1) +
  coord_cartesian(xlim = c(-2300, 6400), ylim = c(-3400, 1900), expand = FALSE) +
  xlab(paste0(&quot;PC1 (&quot;, pca_expl_var[1], &quot;%)&quot;)) +
  ylab(paste0(&quot;PC2 (&quot;, pca_expl_var[2], &quot;%)&quot;)) + 
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;, 
        axis.title = element_text(size = 7), legend.title = element_text(size = 7), legend.text = element_text(size = 7), strip.text.x = element_text(margin = margin(0.1, 0, 0.1, 0, &quot;cm&quot;))) +
  # guides(alpha = &quot;none&quot;, shape = guide_legend(ncol = 1, byrow = TRUE), color = guide_legend(ncol = 1, byrow = TRUE))
  guides(alpha = &quot;none&quot;)
fig3_panelC</code></pre>
<p><img src="figure/v_states_main.Rmd/fig3_panelC-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-d-2" class="section level4">
<h4>Panel D</h4>
<p>PCA visualisation of the DMSO ctrl and three mixed inhibitor
treatments</p>
<pre class="r"><code># Data
fig3D_group_select &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;three_mix&quot;)
fig3D_data &lt;- pca_data %&gt;%
   dplyr::filter(inhib_group %in% fig3D_group_select)

pca_legend_breaks &lt;- c(
  &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Activated\nDMSO ctrl&quot;, 
  # &quot;Activated\n10 uM iBTK&quot;,
  # &quot;Activated\n10 uM iSYK&quot;,
  # &quot;Activated\n10 uM iNFkB&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK&quot;,
  # &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;,
  # &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;,
  # &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;
)
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO basal)&quot;, 
  &quot;Activated\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO activated)&quot;, 
  # &quot;Activated\n10 uM iBTK&quot; = &quot;10       -           -&quot;,
  # &quot;Activated\n10 uM iSYK&quot; = &quot;-          10        -&quot;,
  # &quot;Activated\n10 uM iNFkB&quot; = &quot;-          -          10&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5      2.5       -&quot;,
  # &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5      -           2.5&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;-          2.5       2.5&quot;,
  # &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1         1          1&quot;,
  # &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5      2.5       2.5&quot;,
  # &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10&quot;,
  # &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10\n(basal)&quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5          1         1&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;1          5         1&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;1          1         5&quot;
)

# Figure
fig3_panelD &lt;- ggplot(fig3D_data) + 
  geom_point(aes(x = PC1, y = PC2, 
                 shape = stim_inhib, 
                 color = stim_inhib, 
                 alpha = inhib_group), #alpha = inhib_group
             size = 2) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_alpha_manual(values = pca_legend_alpha) +
  facet_wrap(vars(factor(experiment, labels = &quot;Three inhibitors mixed&quot;)), ncol = 1) +
  coord_cartesian(xlim = c(-2300, 6400), ylim = c(-3400, 1900), expand = FALSE) +
  xlab(paste0(&quot;PC1 (&quot;, pca_expl_var[1], &quot;%)&quot;)) +
  ylab(paste0(&quot;PC2 (&quot;, pca_expl_var[2], &quot;%)&quot;)) + 
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;, 
        axis.title = element_text(size = 7), legend.title = element_text(size = 7), legend.text = element_text(size = 7), strip.text.x = element_text(margin = margin(0.1, 0, 0.1, 0, &quot;cm&quot;))) +
  # guides(alpha = &quot;none&quot;, shape = guide_legend(ncol = 1, byrow = TRUE), color = guide_legend(ncol = 1, byrow = TRUE))
  guides(alpha = &quot;none&quot;)
fig3_panelD</code></pre>
<p><img src="figure/v_states_main.Rmd/fig3_panelD-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="full-legend" class="section level4">
<h4>Full legend</h4>
<pre class="r"><code># Data
fig3leg_data &lt;- pca_data %&gt;%
  dplyr::filter(stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  add_row(stim_inhib = &quot;Empty1&quot;, PC1 = 0, PC2 = 0, inhib_group = &quot;Empty&quot;) %&gt;%
  add_row(stim_inhib = &quot;Empty2&quot;, PC1 = 100, PC2 = 100, inhib_group = &quot;Empty&quot;) %&gt;%
  add_row(stim_inhib = &quot;Empty3&quot;, PC1 = 200, PC2 = 200, inhib_group = &quot;Empty&quot;) %&gt;%
  add_row(stim_inhib = &quot;Empty4&quot;, PC1 = 300, PC2 = 300, inhib_group = &quot;Empty&quot;)

pca_legend_alpha &lt;- c(&quot;Empty&quot; = 0, &quot;DMSO ctrl&quot; = 0.8, &quot;single&quot; = 0.4, 
                      &quot;two&quot; = 1, &quot;three_equal&quot; = 1, &quot;three_mix&quot; = 1)
pca_legend_shapes &lt;- c(16, 17,
                       15, 
                       17, 17, 17,
                       15, 
                       17, 17, 17,
                       15, 
                       17, 17, 17,
                       15,
                       17, 17, 17)
pca_legend_values &lt;- c(&quot;#bdbdbd&quot;, &quot;#737373&quot;,
                       &quot;white&quot;, 
                       &quot;#5c9dce&quot;, &quot;#9dce5c&quot;, &quot;#ce5c9d&quot;,
                       &quot;white&quot;, 
                       &quot;#5cce8d&quot;, &quot;#8d5cce&quot;, &quot;#ce8d5c&quot;,
                       &quot;white&quot;, 
                       &quot;#d3bceb&quot;, &quot;#ae82da&quot;, &quot;#8351b6&quot;,
                       &quot;white&quot;, 
                       &quot;#729cb9&quot;, &quot;#9bb677&quot;, &quot;#b6779b&quot;)
pca_legend_breaks &lt;- c(
  &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Activated\nDMSO ctrl&quot;, 
  &quot;Empty1&quot;, 
  &quot;Activated\n10 uM iBTK&quot;,
  &quot;Activated\n10 uM iSYK&quot;,
  &quot;Activated\n10 uM iNFkB&quot;,
  &quot;Empty2&quot;, 
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot;,
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;,
  &quot;Empty3&quot;, 
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;,
  &quot;Empty4&quot;, 
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;
)
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO basal)&quot;, 
  &quot;Activated\nDMSO ctrl&quot; = &quot;-          -           -\n(DMSO activated)&quot;, 
  &quot;Empty1&quot; = &quot; &quot;,
  &quot;Activated\n10 uM iBTK&quot; = &quot;10       -           -&quot;,
  &quot;Activated\n10 uM iSYK&quot; = &quot;-          10        -&quot;,
  &quot;Activated\n10 uM iNFkB&quot; = &quot;-          -          10&quot;,
  &quot;Empty2&quot; = &quot; &quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5      2.5       -&quot;,
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5      -           2.5&quot;,
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;-          2.5       2.5&quot;,
  &quot;Empty3&quot; = &quot; &quot;,
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1         1          1&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5      2.5       2.5&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10       10        10&quot;,
  &quot;Empty4&quot; = &quot; &quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5          1         1&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;1          5         1&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;1          1         5&quot;
)

# Figure
fig3_plot_legend &lt;- ggplot(fig3leg_data) + 
  geom_point(aes(x = PC1, y = PC2, 
                 shape = stim_inhib, 
                 color = stim_inhib, 
                 alpha = inhib_group), #alpha = inhib_group
             size = 2) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels,
                     name = pca_legend_title) +
  scale_alpha_manual(values = pca_legend_alpha) +
  # facet_wrap(vars(factor(experiment, labels = &quot;Three inhibitors mixed&quot;)), ncol = 1) +
  coord_cartesian(xlim = c(-2300, 6400), ylim = c(-3400, 1900), expand = FALSE) +
  xlab(paste0(&quot;PC1 (&quot;, pca_expl_var[1], &quot;%)&quot;)) +
  ylab(paste0(&quot;PC2 (&quot;, pca_expl_var[2], &quot;%)&quot;)) + 
  theme_bw() +
  textsize_small +
  theme(panel.grid.minor = element_blank(), , strip.text.x = element_text(margin = margin(0.1, 0, 0.1, 0, &quot;cm&quot;)), 
        legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.01, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, 
        axis.title = element_text(size = 7), legend.title = element_text(size = 7.25), legend.text = element_text(size  = 7.25)) +
  # guides(alpha = &quot;none&quot;, shape = guide_legend(ncol = 1, byrow = TRUE), color = guide_legend(ncol = 1, byrow = TRUE))
  guides(alpha = &quot;none&quot;)
# fig3_plot_legend
fig3_legend &lt;- as_ggplot(ggpubr::get_legend(fig3_plot_legend))
fig3_legend</code></pre>
<p><img src="figure/v_states_main.Rmd/fig3_legend-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-2" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig3_ABCD &lt;- plot_grid(fig3_panelA, fig3_panelB, fig3_panelC, fig3_panelD, labels = PANEL_labels[c(1:4)], ncol = 2, rel_heights = c(1, 1), rel_widths = c(1, 1), label_size = 10)
fig3 &lt;- plot_grid(fig3_ABCD, fig3_legend, nrow = 1, rel_widths = c(4, 1), label_size = 10)
fig3</code></pre>
<p><img src="figure/v_states_main.Rmd/fig3_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg
ggsave(
  fig3,
  filename = &quot;output/figures/DLBCL_states/main_fig3.pdf&quot;,
  width = 6,
  height = 4.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig3,
  filename = &quot;output/figures/DLBCL_states/main_fig3.jpg&quot;,
  width = 6,
  height = 4.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Inhibitor combination treatments effectively induce repressed
signaling states.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig3&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9418182 503.0   16905207 902.9 16905207 902.9
Vcells 16525260 126.1   61226380 467.2 93646971 714.5</code></pre>
</div>
</div>
<div id="figure-4" class="section level2">
<h2>Figure 4</h2>
<p><em>Strong network-wide disruptions of activated B-cell signaling by
combination treatments.</em></p>
<p>Input:</p>
<ul>
<li><p>DS113 ID-seq data</p>
<ul>
<li><p>DESeq2 analysis</p></li>
<li><p>Cytoscape networks</p></li>
</ul></li>
</ul>
<div id="data-3" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS113 (counts per well/sample and mean per condition)
fig4_data_id &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
fig4_mean_id &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)

fig4_node_info &lt;- read_excel(&quot;output/network_visual_DLBCL/node_info2.xlsx&quot;)</code></pre>
<pre class="r"><code>fig4_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
fig4_label_stim &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

fig4_legend_label &lt;- &quot;iSYK + iBTK + iNFkB&quot;
fig4_inhib_group &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)
fig4_inhib &lt;- c(&quot;DMSO ctrl&quot;, 
                &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                &quot;iSYK + iBTK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                &quot;iSYK + iBTK + iNFkB&quot;)
fig4_label_inhib &lt;- c(&quot;DMSO ctrl&quot;, 
                      &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                      &quot;iBTK+ iSYK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                      &quot;iBTK + iSYK + iNFkB&quot;)
fig4_conc_single &lt;- c(0, 1, 2.5, 5, 10)
fig4_conc_all1 &lt;- c(&quot;0 + 0 + 0 uM&quot;, 
                    &quot;0 + 10 + 0 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, 
                    &quot;2.5 + 2.5 + 0 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, 
                    &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, 
                    &quot;1 + 5 + 1 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, 
                    &quot;10 + 10 + 10 uM (nostain)&quot;)
fig4_conc_all2 &lt;- c(&quot;DMSO ctrl&quot;, 
                    &quot;10 uM iBTK&quot;, &quot;10 uM iSYK&quot;, &quot;10 uM iNFkB&quot;, 
                    &quot;2.5 uM of iSYK + iBTK&quot;, &quot;2.5 uM of iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iNFkB&quot;, 
                    &quot;1 uM of iSYK + iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;10 uM of iSYK + iBTK + iNFkB&quot;, 
                    &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)

fig4_meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="diff-express-analysis" class="section level3 tabset">
<h3 class="tabset">Diff express analysis</h3>
<div id="activated-vs-basal-dmso-ctrl" class="section level4">
<h4>Activated vs basal (DMSO ctrl)</h4>
<p>Differential expression analysis of activation</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>DMSO ctrl samples (basal and activated)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: stim_clean (ref: Basal)</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig4_data_id  %&gt;%
  dplyr::filter(inhib_new == &quot;DMSO ctrl&quot; &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(fig4_data_id[1:20]) %&gt;% 
  distinct() %&gt;%
  mutate(stim_clean = factor(stim_clean, levels = fig4_label_stim)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_stim &lt;- results(dds, name = &quot;stim_clean_Activated_vs_Basal&quot;, alpha = 0.05)
# summary(results_HBL1, padj = 0.05)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Volcano preparation/data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
volc_stim &lt;- prep_forvulcano(dataset = results_stim, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_ActvsBas_log2FC = log2FoldChange, 
                states_ActvsBas_padj = padj,
                states_ActvsBas_DE = diff_express)

# write.csv(volc_stim, file = &quot;output/network_visual_paper/node_info_combi_ActvsBas.csv&quot;, row.names = F, na = &quot;&quot;)</code></pre>
</div>
<div id="inhibitors-vs-dmso-ctrl" class="section level4">
<h4>Inhibitors vs DMSO ctrl</h4>
<p>Differential expression analysis of 9inhibition in activated HBL1
cells</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>all inhibitor treatments + DMSO ctrl samples (activated)</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl)</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- fig4_data_id  %&gt;%
  dplyr::filter(stim_clean == &quot;Activated&quot; &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(fig4_data_id[c(1:20, 32)]) %&gt;% 
  distinct() %&gt;%
  mutate(stim_clean = factor(stim_clean, levels = fig4_label_stim), 
         inhib_text_new = factor(inhib_text_new, levels = fig4_conc_all2)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_iBTK &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iSYK &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iNFkB &lt;- results(dds, name = &quot;inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_iBTKiSYK &lt;- results(dds, name = &quot;inhib_text_new_2.5.uM.of.iSYK...iBTK_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_equal1 &lt;- results(dds, name = &quot;inhib_text_new_1.uM.of.iSYK...iBTK...iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_equal10 &lt;- results(dds, name = &quot;inhib_text_new_10.uM.of.iSYK...iBTK...iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
results_threeiSYK &lt;- results(dds, name = &quot;inhib_text_new_5.uM.iSYK...1.uM.of.iBTK...iNFkB_vs_DMSO.ctrl&quot;, alpha = 0.05)
# list_results &lt;- list(results_iSYK100_stim, results_iBTK100_stim, results_iNFkB100_stim)
# summary(results_HBL1, padj = 0.05)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Volcano preparation/data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
volc_iBTK &lt;- prep_forvulcano(dataset = results_iBTK, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_iBTKvsDMSO_combi_log2FC = log2FoldChange, 
                states_iBTKvsDMSO_combi_padj = padj,
                states_iBTKvsDMSO_combi_DE = diff_express)
# write.csv(volc_iBTK, file = &quot;output/network_visual_paper/node_info_combi_iBTKvsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)

volc_iSYK &lt;- prep_forvulcano(dataset = results_iSYK, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_iSYKvsDMSO_combi_log2FC = log2FoldChange, 
                states_iSYKvsDMSO_combi_padj = padj,
                states_iSYKvsDMSO_combi_DE = diff_express)
# write.csv(volc_iSYK, file = &quot;output/network_visual_paper/node_info_combi_iSYKvsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)

volc_iNFkB &lt;- prep_forvulcano(dataset = results_iNFkB, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_iNFkBvsDMSO_combi_log2FC = log2FoldChange, 
                states_iNFkBvsDMSO_combi_padj = padj,
                states_iNFkBvsDMSO_combi_DE = diff_express)
# write.csv(volc_iNFkB, file = &quot;output/network_visual_paper/node_info_combi_iNFkBvsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)

volc_iBTKiSYK &lt;- prep_forvulcano(dataset = results_iBTKiSYK, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_iBTKiSYKvsDMSO_combi_log2FC = log2FoldChange, 
                states_iBTKiSYKvsDMSO_combi_padj = padj,
                states_iBTKiSYKvsDMSO_combi_DE = diff_express)
# write.csv(volc_iBTKiSYK, file = &quot;output/network_visual_paper/node_info_combi_iBTKiSYKvsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)

volc_equal1 &lt;- prep_forvulcano(dataset = results_equal1, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_equal1vsDMSO_combi_log2FC = log2FoldChange, 
                states_equal1vsDMSO_combi_padj = padj,
                states_equal1vsDMSO_combi_DE = diff_express)
# write.csv(volc_equal1, file = &quot;output/network_visual_paper/node_info_combi_equal1vsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)

volc_equal10 &lt;- prep_forvulcano(dataset = results_equal10, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_equal10vsDMSO_combi_log2FC = log2FoldChange, 
                states_equal10vsDMSO_combi_padj = padj,
                states_equal10vsDMSO_combi_DE = diff_express)
# write.csv(volc_equal10, file = &quot;output/network_visual_paper/node_info_combi_equal10vsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)

volc_threeiSYK &lt;- prep_forvulcano(dataset = results_threeiSYK, foldchange_filter = 0, padj_filter = 0.05) %&gt;%
  subset(diff_express == &quot;UP&quot; | diff_express == &quot;DOWN&quot;) %&gt;%
  dplyr::select(c(target_nospace, log2FoldChange, padj, diff_express)) %&gt;%
  full_join(fig4_node_info) %&gt;%
  mutate(log2FoldChange = case_when(in_panel == &quot;yes&quot; &amp; is.na(log2FoldChange) ~ 0, 
                                    .default = log2FoldChange), 
         padj = case_when(in_panel == &quot;yes&quot; &amp; is.na(padj) ~ 0.051, 
                          .default = padj)) %&gt;%
  subset(in_network == &quot;yes&quot; &amp; in_panel == &quot;yes&quot;) %&gt;%
  dplyr::rename(states_threeiSYKvsDMSO_combi_log2FC = log2FoldChange, 
                states_threeiSYKvsDMSO_combi_padj = padj,
                states_threeiSYKvsDMSO_combi_DE = diff_express)
# write.csv(volc_threeiSYK, file = &quot;output/network_visual_paper/node_info_combi_threeiSYKvsDMSOctrl.csv&quot;, row.names = F, na = &quot;&quot;)</code></pre>
</div>
<div id="combined-dataset" class="section level4">
<h4>Combined dataset</h4>
<pre class="r"><code>fig4_DESeq2_node_info &lt;- list(volc_stim, volc_iBTK, volc_iSYK, volc_iNFkB, volc_iBTKiSYK, volc_equal1, volc_equal10, volc_threeiSYK) %&gt;%
  purrr::reduce(full_join)
write.csv(fig4_DESeq2_node_info, file = &quot;output/network_visual_DLBCL/data_node_info_combi_all.csv&quot;, row.names = F, na = &quot;&quot;)</code></pre>
</div>
</div>
<div id="panels-3" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-3" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># # Placeholder
# fig4_panelA &lt;- ggplot() +
#   geom_blank() +
#   scale_x_continuous(limits = c(0, 10)) +
#   scale_y_continuous(limits = c(0, 10)) +
#   labs(title = &quot;Network projections of combination treatment experiment&quot;) +
#   theme_bw() +
#   theme(axis.text = element_text(color = &quot;white&quot;),
#         axis.ticks = element_blank(),
#         panel.grid = element_blank()) +
#   textsize_small
# fig4_panelA

# Active
fig4_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_states_ActvsBas.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;    Activated vs basal&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8))
fig4_panelA</code></pre>
<p><img src="figure/v_states_main.Rmd/fig4_panelA-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># legend
fig4_network_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  # ggtitle(&quot;\n&quot;) +
  textsize_small
# fig4_network_legend</code></pre>
</div>
<div id="panel-b-3" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># iBTK
fig4_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_states_SingleiBTKvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;    10 uM iBTK vs DMSO ctrl (act. cells)&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8))
fig4_panelB</code></pre>
<p><img src="figure/v_states_main.Rmd/fig4_panelB-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-c-3" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># equal 1
fig4_panelC &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_states_Equal1vsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;    1 uM of each vs DMSO ctrl (act. cells)&quot;) +
  textsize_small + 
  theme(plot.title = element_text(size = 8))
fig4_panelC</code></pre>
<p><img src="figure/v_states_main.Rmd/fig4_panelC-1.png" style="display: block; margin: auto;" /></p>
</div>
<div id="panel-d-3" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># # equal 10
# fig4_panelD &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_states_Equal10vsDMSO.png&quot;) %&gt;%
#   magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
#   magick::image_ggplot() +
#   ggtitle(&quot;    10 uM of each vs DMSO ctrl&quot;) +
#   textsize_small + 
#   theme(plot.title = element_text(size = 8))
# fig4_panelD

# Three iSYK
fig4_panelD &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_states_ThreeiSYKvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;    5 uM iSYK + 1 uM iBTK + iNFkB vs DMSO ctrl (act. cells)&quot;) +
  textsize_small +
  theme(plot.title = element_text(size = 8))
fig4_panelD</code></pre>
<p><img src="figure/v_states_main.Rmd/fig4_panelD-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># 
# # 2.5 iBTK + iSYK
# fig4_panelD_alt2 &lt;- magick::image_read(&quot;output/figures/non_R_figs/Cytoscape_states_iBTKiSYKvsDMSO.png&quot;) %&gt;%
#   magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
#   magick::image_ggplot() +
#   ggtitle(&quot; 2.5 uM iBTK + iSYK vs DMSO ctrl&quot;) +
#   textsize_small + 
#   theme(plot.title = element_text(size = 8))
# fig4_panelD_alt2</code></pre>
</div>
</div>
<div id="combined-fig-3" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig4_networks &lt;- plot_grid(fig4_panelA, fig4_panelB, fig4_panelC, fig4_panelD, labels = PANEL_labels[c(1:4)], ncol = 2, rel_widths = c(1, 1), rel_heights = c(1, 1), label_size = 10)
fig4 &lt;- plot_grid(fig4_networks, fig4_network_legend, ncol = 1, rel_heights = c(10, 1))
fig4</code></pre>
<p><img src="figure/v_states_main.Rmd/fig4_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># fig4_alt &lt;- plot_grid(fig4_panelD_alt1, fig4_panelD_alt2, ncol = 2, rel_widths = c(1, 1))
# fig4_test &lt;- plot_grid(fig4, fig4_alt, ncol = 1, rel_heights = c(3, 1))

# Save figure as pdf, jpg
ggsave(
  fig4,
  filename = &quot;output/figures/DLBCL_states/main_fig4.pdf&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig4,
  filename = &quot;output/figures/DLBCL_states/main_fig4.jpg&quot;,
  width = 6,
  height = 6.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Strong network-wide disruptions of activated B-cell signaling by
combination treatments.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig4&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9490182 506.9   16905207 902.9 16905207 902.9
Vcells 20625132 157.4   61226380 467.2 93646971 714.5</code></pre>
</div>
</div>
<div id="figure-5" class="section level2">
<h2>Figure 5</h2>
<p><em>Inhibition of the B-cell network results in a transition from an
activated signaling state to a repressed state.</em></p>
<p>Input:</p>
<ul>
<li>BioRender schematic figure</li>
</ul>
<div id="panels-4" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-4" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
fig5_panelA_place &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  labs(title = &quot;Summary figure&quot;) +
  theme_bw() +
  theme(axis.text.x = element_blank(), 
        axis.text.y = element_blank(),
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig5_panelA_place

# PNG
fig5_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/signaling_states_summary.png&quot;) %&gt;%
  magick::image_ggplot()
fig5_panelA</code></pre>
<p><img src="figure/v_states_main.Rmd/fig5_panelA-1.png" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="combined-fig-4" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig5_panelA</code></pre>
<p><img src="figure/v_states_main.Rmd/fig5_combi-1.png" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Save figure as pdf, jpg
# ggsave(
#   fig5,
#   filename = &quot;output/figures/DLBCL_states/main_fig5.pdf&quot;,
#   width = 6,
#   height = 2.5,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  fig5_panelA_place,
  filename = &quot;output/figures/DLBCL_states/main_fig5_place.jpg&quot;,
  width = 6,
  height = 2.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Inhibition of the B-cell network results in a transition from an
activated signaling state to a repressed state.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig5&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger  (Mb) max used  (Mb)
Ncells  9467486 505.7   16905207 902.9 16905207 902.9
Vcells 16691034 127.4   61226380 467.2 93646971 714.5</code></pre>
</div>
</div>
<div id="figure-0" class="section level2">
<h2>Figure 0</h2>
<p><em>Figure title.</em></p>
<p>Input: - BioRender schematic figures</p>
<ul>
<li><p>… phosphoflow data</p></li>
<li><p>… ID-seq data</p></li>
</ul>
<div id="data-4" class="section level3">
<h3>Data</h3>
</div>
<div id="panels-5" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-5" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
fig0_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# fig0_panelA

# PNG
png_fig0 &lt;- image_read(&quot;output/figures/non_R_figs/xxx.png&quot;) %&gt;%
  image_ggplot()
# png_fig0</code></pre>
</div>
<div id="panel-b-4" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
<div id="panel-c-4" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
</div>
<div id="combined-fig-5" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
fig0 &lt;- plot_grid(panelA, panelB, labels = panel_labels[c(1, 2)], ncol = 1, rel_heights = c(1, 1), label_size = 10)
fig0

# Save figure as pdf, jpg
# ggsave(
#   fig0,
#   filename = &quot;output/figures/DLBCL_states/main_fig0.pdf&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  fig0,
  filename = &quot;output/figures/DLBCL_states/main_fig0.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure title.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig0&quot;))
rm(list = ls(pattern = &quot;png&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;rld&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
gc()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ashr_2.2-63                 apeglm_1.26.1              
 [3] MASS_7.3-60.2               ggbreak_0.1.2              
 [5] geomtextpath_0.1.4          ggh4x_0.2.8                
 [7] cowplot_1.1.3               patchwork_1.2.0            
 [9] greekLetters_1.0.2          ggpubr_0.6.0               
[11] ggrepel_0.9.5               scales_1.3.0               
[13] ggthemes_5.1.0              RColorBrewer_1.1-3         
[15] scico_1.5.0                 viridis_0.6.5              
[17] viridisLite_0.4.2           gridExtra_2.3              
[19] mmtable2_0.1.3              gt_0.10.1                  
[21] ggvenn_0.1.10               rgl_1.3.1                  
[23] plotly_4.10.4               ggcorrplot_0.1.4.1         
[25] corrplot_0.92               umap_0.2.10.0              
[27] ggplotify_0.1.2             dendsort_0.3.4             
[29] vsn_3.72.0                  ComplexHeatmap_2.20.0      
[31] platetools_0.1.7            kableExtra_1.4.0           
[33] DESeq2_1.44.0               SummarizedExperiment_1.34.0
[35] Biobase_2.64.0              MatrixGenerics_1.16.0      
[37] matrixStats_1.3.0           GenomicRanges_1.56.1       
[39] GenomeInfoDb_1.40.1         IRanges_2.38.0             
[41] S4Vectors_0.42.0            BiocGenerics_0.50.0        
[43] Seurat_5.1.0                SeuratObject_5.0.2         
[45] sp_2.1-4                    rstatix_0.7.2              
[47] factoextra_1.0.7            FactoMineR_2.11            
[49] readxl_1.4.3                Matrix_1.7-0               
[51] lubridate_1.9.3             forcats_1.0.0              
[53] stringr_1.5.1               dplyr_1.1.4                
[55] purrr_1.0.2                 readr_2.1.5                
[57] tidyr_1.3.1                 tibble_3.2.1               
[59] ggplot2_3.5.1               tidyverse_2.0.0            
[61] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             goftest_1.2-3           DT_0.33                
  [4] TH.data_1.1-2           vctrs_0.6.5             spatstat.random_3.2-3  
  [7] digest_0.6.36           png_0.1-8               shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       magick_2.8.3            SQUAREM_2021.1         
 [16] reshape2_1.4.4          httpuv_1.6.15           foreach_1.5.2          
 [19] withr_3.0.0             xfun_0.45               ggfun_0.1.5            
 [22] survival_3.6-4          memoise_2.0.1           emmeans_1.10.2         
 [25] systemfonts_1.1.0       ragg_1.3.2              zoo_1.8-12             
 [28] GlobalOptions_0.1.2     pbapply_1.7-2           promises_1.3.0         
 [31] scatterplot3d_0.3-44    httr_1.4.7              globals_0.16.3         
 [34] fitdistrplus_1.1-11     ps_1.7.6                rstudioapi_0.16.0      
 [37] UCSC.utils_1.0.0        miniUI_0.1.1.1          generics_0.1.3         
 [40] base64enc_0.1-3         processx_3.8.4          zlibbioc_1.50.0        
 [43] polyclip_1.10-6         GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [46] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [49] S4Arrays_1.4.1          preprocessCore_1.66.0   hms_1.1.3              
 [52] irlba_2.3.5.1           colorspace_2.1-0        ROCR_1.0-11            
 [55] reticulate_1.38.0       spatstat.data_3.1-2     magrittr_2.0.3         
 [58] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [61] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [64] scattermore_1.2         RcppAnnoy_0.0.22        pillar_1.9.0           
 [67] nlme_3.1-164            iterators_1.0.14        compiler_4.4.1         
 [70] RSpectra_0.16-1         stringi_1.8.4           tensor_1.5             
 [73] plyr_1.8.9              crayon_1.5.3            abind_1.4-5            
 [76] truncnorm_1.0-9         gridGraphics_0.5-1      emdbook_1.3.13         
 [79] locfit_1.5-9.10         bit_4.0.5               sandwich_3.1-0         
 [82] whisker_0.4.1           codetools_0.2-20        multcomp_1.4-25        
 [85] textshaping_0.4.0       openssl_2.2.0           bslib_0.7.0            
 [88] GetoptLong_1.0.5        mime_0.12               splines_4.4.1          
 [91] circlize_0.4.16         Rcpp_1.0.12             fastDummies_1.7.3      
 [94] cellranger_1.1.0        Rttf2pt1_1.3.12         leaps_3.2              
 [97] knitr_1.47              utf8_1.2.4              clue_0.3-65            
[100] fs_1.6.4                listenv_0.9.1           ggsignif_0.6.4         
[103] estimability_1.5.1      callr_3.7.6             statmod_1.5.0          
[106] tzdb_0.4.0              svglite_2.1.3           pkgconfig_2.0.3        
[109] tools_4.4.1             cachem_1.1.0            numDeriv_2016.8-1.1    
[112] fastmap_1.2.0           rmarkdown_2.27          ica_1.0-3              
[115] broom_1.0.6             sass_0.4.9              coda_0.19-4.1          
[118] BiocManager_1.30.23     dotCall64_1.1-1         carData_3.0-5          
[121] RANN_2.6.1              farver_2.1.2            yaml_2.3.8             
[124] cli_3.6.3               leiden_0.4.3.1          lifecycle_1.0.4        
[127] askpass_1.2.0           uwot_0.2.2              mvtnorm_1.2-5          
[130] backports_1.5.0         BiocParallel_1.38.0     timechange_0.3.0       
[133] gtable_0.3.5            rjson_0.2.21            ggridges_0.5.6         
[136] progressr_0.14.0        parallel_4.4.1          limma_3.60.3           
[139] jsonlite_1.8.8          RcppHNSW_0.6.0          bit64_4.0.5            
[142] multcompView_0.1-10     assertthat_0.2.1        Rtsne_0.17             
[145] yulab.utils_0.1.4       spatstat.utils_3.0-5    bdsmatrix_1.3-7        
[148] highr_0.11              jquerylib_0.1.4         lazyeval_0.2.2         
[151] shiny_1.8.1.1           htmltools_0.5.8.1       affy_1.82.0            
[154] sctransform_0.4.1       glue_1.7.0              spam_2.10-0            
[157] XVector_0.44.0          rprojroot_2.0.4         flashClust_1.01-2      
[160] invgamma_1.1            igraph_2.0.3            extrafontdb_1.0        
[163] R6_2.5.1                labeling_0.4.3          bbmle_1.0.25.1         
[166] cluster_2.1.6           aplot_0.2.3             DelayedArray_0.30.1    
[169] tidyselect_1.2.1        xml2_1.3.6              car_3.1-2              
[172] future_1.33.2           munsell_0.5.1           KernSmooth_2.23-24     
[175] affyio_1.74.0           data.table_1.15.4       htmlwidgets_1.6.4      
[178] rlang_1.1.4             spatstat.sparse_3.1-0   extrafont_0.19         
[181] spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
