<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-06-04" />

<title>FASEB conference poster figures</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapters on DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Paper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="y_paper_main.html">Main</a>
    </li>
    <li>
      <a href="z_paper_suppl.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteins ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="t_proteins_main.html.html">Main</a>
    </li>
    <li>
      <a href="u_proteins_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    States ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="v_states_main.html.html">Main</a>
    </li>
    <li>
      <a href="w_states_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
    <li>
      <a href="c_flow_vis.html">Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="DS108_p_seq_comparisons.html">DS108 clean</a>
    </li>
    <li>
      <a href="DS108_q_seq_thresholds.html">DS108 thresholds</a>
    </li>
    <li>
      <a href="DS108_r_seq_pca.html">DS108 PCA</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">FASEB conference poster figures</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-06-04</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapters_DLBCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241022code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241022code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241022)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong94c226d">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 94c226d
</a>
</p>
</div>
<div id="strongRepositoryversionstrong94c226d"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 94c226d.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/AL006_InhibCombiTest/
    Untracked:  data/DS102_StimTest/
    Untracked:  data/DS103_ViabilityTest/
    Untracked:  data/DS104_StimInhibTest/
    Untracked:  data/DS108_StimInhibIDseq/
    Untracked:  data/DS113_InhibCombiIDseq/
    Untracked:  output/DS102_StimTest/
    Untracked:  output/DS103_ViabilityTest/
    Untracked:  output/DS104_StimInhibTest/
    Untracked:  output/DS108_StimInhibIDseq/
    Untracked:  output/DS113_InhibCombiIDseq/
    Untracked:  output/figures/
    Untracked:  output/network_visual_DLBCL/
    Untracked:  output/network_visual_paper/

Unstaged changes:
    Deleted:    analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/x_poster_FASEB.Rmd</code>) and
HTML (<code>docs/x_poster_FASEB.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
ffc633f
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
6559698
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Paper figs + analysis files for ch figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)
PANEL_labels &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;,&quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;)

# textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
#                   axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
#                   axis.title = element_text(colour = &quot;black&quot;, size = 12), 
#                   legend.title = element_text(colour = &quot;black&quot;, size = 12),
#                   # legend.title = element_blank(), 
#                   legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
#                   strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
# )
textsize &lt;- theme(text = element_text(size = 10, family = &quot;sans&quot;, colour = &quot;black&quot;),
                        plot.title = element_text(size = 11)
)

textsize_small &lt;- theme(text = element_text(size = 7, family = &quot;sans&quot;, colour = &quot;black&quot;),
                        plot.title = element_text(size = 8)
)
# textsize_small &lt;- theme(axis.text.x = element_text(colour = &quot;black&quot;, size = 9),
#                         axis.text.y = element_text(colour = &quot;black&quot;, size = 9),
#                         axis.title = element_text(colour = &quot;black&quot;, size = 10),
#                         legend.text = element_text(colour = &quot;black&quot;, size = 9),
#                         title = element_text(color = &quot;black&quot;, size = 10),
#                         strip.text.x = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;),
#                         strip.text.y = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;)
# )</code></pre>
<pre class="r"><code>colors_nord &lt;- c(&quot;#2e3440&quot;, &quot;#3b4252&quot;, &quot;#434c5e&quot;, &quot;#4c566a&quot;, 
                 &quot;#d8dee9&quot;, &quot;#e5e9f0&quot;, &quot;#eceff4&quot;, 
                 &quot;#8fbcbb&quot;, &quot;#88c0d0&quot;, &quot;#81a1c1&quot;, &quot;#5e81ac&quot;, 
                 &quot;#bf616a&quot;, &quot;#d08770&quot;, &quot;#ebcb8b&quot;, &quot;#a3be8c&quot;, &quot;#b48ead&quot;)

colors_toll_bright &lt;- c(&quot;#4477AA&quot;, &quot;#66CCEE&quot;, &quot;#228833&quot;, &quot;#CCBB44&quot;, &quot;#EE6677&quot;, &quot;#AA3377&quot;, &quot;#BBBBBB&quot;)
colors_toll_mute &lt;- c(&quot;#332288&quot;, &quot;#88CCEE&quot;, &quot;#44AA99&quot;, &quot;#117733&quot;, &quot;#999933&quot;, &quot;#DDCC77&quot;, &quot;#CC6677&quot;, &quot;#882255&quot;, &quot;#AA4499&quot;, &quot;#DDDDDD&quot;)
colors_toll_br &lt;- c(&quot;#364B9A&quot;, &quot;#4A7BB7&quot;, &quot;#6EA6CD&quot;, &quot;#98CAE1&quot;, &quot;#C2E4EF&quot;, &quot;#EAECCC&quot;, &quot;#FEDA8B&quot;, &quot;#FDB366&quot;, &quot;#F67A4B&quot;, &quot;#DD3D2D&quot;, &quot;#A50026&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)
colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

# Variable-specific colors
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8&quot; = &quot;#332288&quot;)
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#525252&quot;, &quot;aIg+H2O2&quot; = &quot;#FDB366&quot;)
colors_cell_stim &lt;- c(&quot;HBL1 - PBS&quot; = &quot;#CC6677&quot;, &quot;HBL1 - aIg+H2O2&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8 - PBS&quot; = &quot;#88CCEE&quot;, &quot;OCI-Ly8 - aIg+H2O2&quot; = &quot;#332288&quot;)
# colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#228833&quot;, &quot;iBTK&quot; = &quot;#4477AA&quot;, &quot;iNFkB&quot; = &quot;#AA3377&quot;)
colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#9dce5c&quot;, &quot;iBTK&quot; = &quot;#5c9dce&quot;, &quot;iNFkB&quot; = &quot;#ce5c9d&quot;)
colors_conc &lt;- colors_purple9[c(3, 5, 6, 8, 9)]
colors_iSYK &lt;- colors_green9[c(3, 5, 6, 8, 9)]
colors_iBTK &lt;- colors_blue9[c(3, 5, 6, 8, 9)]
colors_iNFkB &lt;- colors_pr9[c(3, 5, 6, 8, 9)]
colors_conc_stim &lt;- colors_orange9[c(2, 3, 5, 6, 8, 9)]
colors_conc_PBS &lt;- colors_grey9[c(2, 3, 5, 6, 8, 9)]</code></pre>
</div>
<div id="data" class="section level2">
<h2>Data</h2>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
data_sample &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;% 
  mutate(cell_stim = paste(cell_line, stimulus), 
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))

data_condition &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;% 
  mutate(cell_stim = paste(cell_line, stimulus), 
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    .default = inhib_conc_text))

meta_cols &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<pre class="r"><code>order_stim &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
label_stim &lt;- c(&quot;Control&quot;, &quot;Stimulated&quot;)
order_inhib &lt;- c(&quot;iSYK&quot;, &quot;iBTK&quot;, &quot;iNFkB&quot;)
order_conc &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
label_conc &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
order_conc_select &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;1 uM&quot;, &quot;100 uM&quot;)</code></pre>
</div>
<div id="network-wide-disruptions" class="section level2">
<h2>Network-wide disruptions</h2>
<div id="deseq2-analysis" class="section level3">
<h3>DESeq2 analysis</h3>
<p>DESeq2: Effect of 1 uM or 100 uM inhib on stim vs PBS - interaction,
HBL1</p>
<p>Analysis parameters:</p>
<ul>
<li><p>All HBL1 samples: 4 inhibitors, 5 conc, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- data_sample  %&gt;%
  dplyr::filter(inhib_conc %in% order_conc_select &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(data_sample[c(1:13, 28:30)]) %&gt;% 
  distinct() %&gt;%
  mutate(stimulus = factor(stimulus, levels = order_stim), 
         inhib_conc = factor(inhib_conc, levels = order_conc)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)
dds$stimulus &lt;- relevel(dds$stimulus, ref = &quot;PBS&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_iSYK1_stim &lt;- results(dds, name = &quot;inhib_text_new1.uM.iSYK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iBTK1_stim &lt;- results(dds, name = &quot;inhib_text_new1.uM.iBTK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iNFkB1_stim &lt;- results(dds, name = &quot;inhib_text_new1.uM.iNFkB.stimulusaIg.H2O2&quot;, alpha = 0.05)

results_iSYK100_stim &lt;- results(dds, name = &quot;inhib_text_new100.uM.iSYK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iBTK100_stim &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stimulusaIg.H2O2&quot;, alpha = 0.05)
results_iNFkB100_stim &lt;- results(dds, name = &quot;inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2&quot;, alpha = 0.05)
list_results &lt;- list(results_iSYK1_stim, results_iBTK1_stim, results_iNFkB1_stim, 
                     results_iSYK100_stim, results_iBTK100_stim, results_iNFkB100_stim)
# summary(results_HBL1, padj = 0.05)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Volcano preparation/data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), filter = TRUE){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.1)
names_results_volc &lt;- c(&quot;volc_iSYK1_stim&quot;, &quot;volc_iBTK1_stim&quot;, &quot;volc_iNFkB1_stim&quot;, 
                     &quot;volc_iSYK100_stim&quot;, &quot;volc_iBTK100_stim&quot;, &quot;volc_iNFkB100_stim&quot;)

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Function to filter the data
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.1)
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_iSYK1_stim, volc_iBTK1_stim, volc_iNFkB1_stim, 
                     volc_iSYK100_stim, volc_iBTK100_stim, volc_iNFkB100_stim)
names_results_heat &lt;- c(&quot;heat_iSYK1_stim&quot;, &quot;heat_iBTK1_stim&quot;, &quot;heat_iNFkB1_stim&quot;, 
                     &quot;heat_iSYK100_stim&quot;, &quot;heat_iBTK100_stim&quot;, &quot;heat_iNFkB100_stim&quot;)
for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}</code></pre>
</div>
<div id="heatmap" class="section level3">
<h3>Heatmap</h3>
<pre class="r"><code># Data
# Combine heatmap datasets (without DESeq2 analysis values)
heat_1_100 &lt;- list(heat_iSYK1_stim, heat_iBTK1_stim, heat_iNFkB1_stim, 
                   heat_iSYK100_stim, heat_iBTK100_stim, heat_iNFkB100_stim) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express, proteins, delabel)) %&gt;%
  left_join(data_sample[, 14:21]) %&gt;%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
heat_info &lt;- data_sample %&gt;%
  dplyr::select(plate_well, inhib_conc, stimulus, inhibitor) %&gt;%
  distinct() %&gt;%
  dplyr::filter(inhib_conc %in% order_conc_select &amp; !(stimulus == &quot;PBS&quot; &amp; inhib_conc %in% order_conc_select[2:3])) %&gt;% 
  mutate(inhibitor = factor(inhibitor, levels = order_inhib)) %&gt;%
  as.data.frame()
rownames(heat_info) &lt;- heat_info$plate_well
heat_info &lt;- dplyr::select(heat_info, -c(plate_well))

# Get the rld transformed data for the heatmap and scale per row
heat_data &lt;- t(scale(t(assay(rld)[c(heat_1_100$target_nospace), rownames(heat_info)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

# Figure
fig_heat_ann_colors &lt;- list(stimulus = colors_stim, 
                         inhibitor = colors_inhib,
                         # inhib_new = c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib), 
                         inhib_conc = setNames(colors_conc[c(1:3)], unique(heat_info$inhib_conc)))

fig_heat_auto &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  heat_data,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  treeheight_row = 0,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  annotation_col = heat_info,
  annotation_colors = fig_heat_ann_colors,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = heat_info$inhibitor,
  row_split = subset(heat_1_100)$modification,
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 5,
  border_color = NA,
  # legend = FALSE,
  # annotation_legend = FALSE,
  # main = &quot;aIg+H2O2 vs PBS (without inhibitors)&quot;
)) +
  ggtitle(&quot;auto&quot;)
# fig_heat_auto

#################

fig_heat_colors &lt;- list(Stimulation = colors_stim, 
                        Inhibitor = colors_inhib,
                        &quot;Inhibitor conc&quot; = setNames(colors_conc[c(1, 3, 5)], unique(heat_info$inhib_conc)), 
                        labels_gp = gpar(fontsize = 5), 
                        simple_anno_size = unit(8, &quot;mm&quot;))
fig_heat_legend &lt;- list(title = &quot;Scaled\nexpression&quot;, title_gp = gpar(fontsize = 6), labels_gp = gpar(fontsize = 6), 
                        grid_width = unit(2, &quot;mm&quot;), grid_height = unit(20, &quot;mm&quot;))

fig_heat_col_ann &lt;- HeatmapAnnotation(Inhibitor = heat_info$inhibitor, 
                                      Stimulation = factor(heat_info$stimulus, levels = order_stim), 
                                      &quot;Inhibitor conc&quot; = heat_info$inhib_conc, 
                                      simple_anno_size = unit(1.5, &quot;mm&quot;), 
                                      annotation_name_gp = gpar(fontsize = 6),
                                      col = fig_heat_colors, 
                                      annotation_legend_param = list(Inhibitor = list(title = &quot;Inhibitor&quot;, 
                                                                                      title_gp = gpar(fontsize = 6), 
                                                                                      labels_gp = gpar(fontsize = 6), 
                                                                                      grid_width = unit(2, &quot;mm&quot;), 
                                                                                      grid_height = unit(1, &quot;mm&quot;)), 
                                                                     Stimulation = list(title = &quot;Stimulation&quot;, 
                                                                                        labels = c(&quot;Basal&quot;, &quot;Activated&quot;),
                                                                                     title_gp = gpar(fontsize = 6), 
                                                                                     labels_gp = gpar(fontsize = 6), 
                                                                                     grid_width = unit(2, &quot;mm&quot;), 
                                                                                     grid_height = unit(1, &quot;mm&quot;)), 
                                                                     &quot;Inhibitor conc&quot; = list(title = &quot;Inhibitor\nconcentration&quot;, 
                                                                                             title_gp = gpar(fontsize = 6), 
                                                                                             labels_gp = gpar(fontsize = 6), 
                                                                                             grid_width = unit(2, &quot;mm&quot;), 
                                                                                             grid_height = unit(1, &quot;mm&quot;))), 
                                      show_legend = c(FALSE, TRUE, TRUE))

fig_heat_man &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  heat_data,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  treeheight_row = 0,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_col = heat_info,
  # annotation_colors = fig_heat_colors,
  top_annotation = fig_heat_col_ann,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = heat_info$inhibitor,
  row_split = subset(heat_1_100)$modification,
  row_gap = unit(1, &quot;mm&quot;),
  cellwidth = 10,
  cellheight = 10,
  fontsize = 5,
  row_title_gp = gpar(fontsize = 6),
  column_title_gp = gpar(fontsize = 6), 
  annotation_names_col = gpar(fontsize = 6),
  border_color = NA,
  heatmap_legend_param = fig_heat_legend
  # annotation_legend = FALSE,
)) +
  theme(legend.position = &quot;bottom&quot;) +
  textsize_small
fig_heat_man</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_heat-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_heat-1">
Past versions of fig_heat-1.png
</button>
</p>
<div id="fig-fig_heat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_heat_man,
  filename = &quot;output/figures/poster/fig_heat_1_100.jpg&quot;,
  width = 7.5,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
</div>
<div id="venn-diagram" class="section level3">
<h3>Venn diagram</h3>
<p>Venn diagrams of proteins where the aIg+H2O2 vs PBS comparison is
significantly affected by 1 uM or 100 uM inhibitor</p>
<pre class="r"><code># Data
# Protein lists
prot_iSYK1 &lt;- sort(unique(c(heat_iSYK1_stim$target_nospace)))
prot_iBTK1 &lt;- sort(unique(c(heat_iBTK1_stim$target_nospace)))
prot_iNFkB1 &lt;- sort(unique(c(heat_iNFkB1_stim$target_nospace)))

prot_iSYK100 &lt;- sort(unique(c(heat_iSYK100_stim$target_nospace)))
prot_iBTK100 &lt;- sort(unique(c(heat_iBTK100_stim$target_nospace)))
prot_iNFkB100 &lt;- sort(unique(c(heat_iNFkB100_stim$target_nospace)))

# Figure
# 1 uM inhibitor
prot_data_1uM &lt;- list(prot_iSYK1, prot_iBTK1, prot_iNFkB1)
prot_names_1uM &lt;- c(paste0(&quot;iSYK (&quot;, length(prot_iSYK1), &quot; total)&quot;), 
                  paste0(&quot;iBTK (&quot;, length(prot_iBTK1), &quot; total)&quot;), 
                  paste0(&quot;iNFkB (&quot;, length(prot_iNFkB1), &quot; total)&quot;))
names(prot_data_1uM) &lt;- prot_names_1uM
fig_venn_1uM &lt;- ggvenn(data = prot_data_1uM, 
               fill_color = c(&quot;#228833&quot;, &quot;#4477AA&quot;, &quot;#AA3377&quot;),
               # auto_scale = TRUE,
               # show_percentage = FALSE,
               # show_elements = TRUE,
               stroke_size = 0.5, 
               text_size = 2.5, 
               set_name_size = 3.5) + 
  ggtitle(&quot;Proteins significantly affected by 1uM inhibitor&quot;) +
  textsize_small
# fig_venn_1uM

# 100 uM inhibitor
prot_data_100uM &lt;- list(prot_iSYK100, prot_iBTK100, prot_iNFkB100)
prot_names_100uM &lt;- c(paste0(&quot;iSYK (&quot;, length(prot_iSYK100), &quot; total)&quot;), 
                  paste0(&quot;iBTK (&quot;, length(prot_iBTK100), &quot; total)&quot;), 
                  paste0(&quot;iNFkB (&quot;, length(prot_iNFkB100), &quot; total)&quot;))
names(prot_data_100uM) &lt;- prot_names_100uM
fig_venn_100uM &lt;- ggvenn(data = prot_data_100uM, 
               fill_color = c(&quot;#228833&quot;, &quot;#4477AA&quot;, &quot;#AA3377&quot;),
               # auto_scale = TRUE,
               # show_percentage = FALSE,
               # show_elements = TRUE,
               stroke_size = 0.5, 
               text_size = 2.5, 
               set_name_size = 3.5) + 
  ggtitle(&quot;Proteins significantly affected by 100 uM inhibitor&quot;) +
  textsize_small
# fig_venn_100uM

fig_venn1_100 &lt;- plot_grid(fig_venn_1uM, fig_venn_100uM, nrow = 1)
fig_venn1_100</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_venn1_100-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_venn1_100-1">
Past versions of fig_venn1_100-1.png
</button>
</p>
<div id="fig-fig_venn1_100-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Data
# Protein lists
prot_iSYK &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/HBL1_iSYK_stim_sign_proteins.txt&quot;)
prot_iBTK &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/HBL1_iBTK_stim_sign_proteins.txt&quot;)
prot_iNFkB &lt;- read_lines(&quot;output/DS108_StimInhibIDseq/IDseq_sign_threshold/active_inhibitor/HBL1_iNFkB_stim_sign_proteins.txt&quot;)

# Figure
# HBL1
prot_data &lt;- list(prot_iSYK, prot_iBTK, prot_iNFkB)
prot_names &lt;- c(paste0(&quot;iSYK (&quot;, length(prot_iSYK), &quot; total)&quot;), 
                  paste0(&quot;iBTK (&quot;, length(prot_iBTK), &quot; total)&quot;), 
                  paste0(&quot;iNFkB (&quot;, length(prot_iNFkB), &quot; total)&quot;))
names(prot_data) &lt;- prot_names
fig_venn_total &lt;- ggvenn(data = prot_data, 
               fill_color = c(&quot;#9dce5c&quot;, &quot;#5c9dce&quot;, &quot;#ce5c9d&quot;),
               # auto_scale = TRUE,
               # show_percentage = FALSE,
               # show_elements = TRUE,
               stroke_size = 0.75, 
               text_size = 3, 
               set_name_size = 4) + 
  # ggtitle(&quot;Proteins significantly affected by inhibitors&quot;) +
  textsize_small
fig_venn_total</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_venn_total-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_venn_total-1">
Past versions of fig_venn_total-1.png
</button>
</p>
<div id="fig-fig_venn_total-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_venn1_100,
  filename = &quot;output/figures/poster/fig_venn_1_100.jpg&quot;,
  width = 10,
  height = 5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig_venn_total,
  filename = &quot;output/figures/poster/fig_venn_total.jpg&quot;,
  width = 5,
  height = 5,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
</div>
<div id="network-projections" class="section level3">
<h3>Network projections</h3>
<pre class="r"><code># Active
fig_network_active &lt;- magick::image_read(&quot;output/figures/non_R_figs/HBL1_StimvsPBS_PBSctrl.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;\nActivated vs basal&quot;) +
  textsize
# fig_network_active

# iSYK
fig_network_iSYK &lt;- magick::image_read(&quot;output/figures/non_R_figs/HBL1_StimOnly_iSYK100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;100 uM iSYK vs DMSO ctrl\nin activated cells&quot;) +
  textsize
# fig_network_iSYK

# iBTK
fig_network_iBTK &lt;- magick::image_read(&quot;output/figures/non_R_figs/HBL1_StimOnly_iBTK100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;100 uM iBTK vs DMSO ctrl\nin activated cells&quot;) +
  textsize
# fig_network_iBTK

# iNFkB
fig_network_iNFkB &lt;- magick::image_read(&quot;output/figures/non_R_figs/HBL1_StimOnly_iNFkB100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;100 uM iNFkB vs DMSO ctrl\nin activated cells&quot;) +
  textsize
# fig_network_iNFkB

# legend
fig_network_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections_vert.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot() +
  ggtitle(&quot;\n&quot;) +
  textsize
# fig_network_legend

# Combined
# fig_network_data &lt;- plot_grid(fig_network_active, fig_network_iSYK, fig_network_iBTK, fig_network_iNFkB, nrow = 1, rel_widths = c(1, 1, 1, 1), align = &quot;h&quot;, label_size = 10)
# fig_network &lt;- plot_grid(fig_network_data, fig_network_legend, ncol = 1, rel_heights = c(5, 1))
# fig_network

# fig_network &lt;- plot_grid(fig_network_active, fig_network_iSYK, fig_network_iBTK, fig_network_iNFkB, fig_network_legend, nrow = 1, rel_widths = c(2, 2, 2, 2, 1))
# fig_network

fig_network_data &lt;- plot_grid(fig_network_active, fig_network_iSYK, fig_network_iBTK, fig_network_iNFkB, ncol = 2, rel_widths = c(1, 1), rel_heights = c(1, 1), align = &quot;h&quot;, scale = 0.95)
# fig_network_data
fig_network &lt;- plot_grid(fig_network_data, fig_network_legend, nrow = 1, rel_widths = c(5, 0.75))
fig_network</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_network-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_network-1">
Past versions of fig_network-1.png
</button>
</p>
<div id="fig-fig_network-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_network,
  filename = &quot;output/figures/poster/fig_network.jpg&quot;,
  width = 8,
  height = 7.5,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
</div>
</div>
<div id="repressed-signaling-state" class="section level2">
<h2>Repressed signaling state</h2>
<div id="pca" class="section level3">
<h3>PCA</h3>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- data_sample  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(data_sample, all_of(meta_cols)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = order_stim), 
         inhibitor = factor(inhibitor, levels = order_inhib), 
         inhib_conc = factor(inhib_conc, levels = order_conc)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
<p>Basic: only basal + activated + 100 uM inhibited</p>
<pre class="r"><code># Figure
# Selected samples only
conditions &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
# select_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, &quot;100 uM iSYK&quot; = &quot;green&quot;, &quot;100 uM iBTK&quot; = &quot;blue&quot;, &quot;100 uM iNFkB&quot; = &quot;red&quot;)
# select_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, &quot;100 uM iSYK&quot; = &quot;#228833&quot;, &quot;100 uM iBTK&quot; = &quot;#4477AA&quot;, &quot;100 uM iNFkB&quot; = &quot;#AA3377&quot;)
select_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, &quot;100 uM iSYK&quot; = &quot;#9dce5c&quot;, &quot;100 uM iBTK&quot; = &quot;#5c9dce&quot;, &quot;100 uM iNFkB&quot; = &quot;#ce5c9d&quot;)
fig_pca_basic &lt;- ggplot(subset(pca_data, inhib_conc %in% conditions)) + 
  geom_point(aes(x = PC1, y = PC3, shape = stimulus, color = inhib_text_new), size = 3) +
  # scale_color_manual(values = c(colors_orange9[3:7], colors_grey9[3:7]), name = &quot;Stim - Inhib conc&quot;) +
  scale_color_manual(values = select_colors, breaks = names(select_colors), name = &quot;Inhibitor&quot;) +
  scale_shape_manual(values = c(16, 17), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) +
  # ggtitle(&quot;PC1 vs PC3&quot;) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;) +
  textsize
fig_pca_basic</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_pca_basic-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_pca_basic-1">
Past versions of fig_pca_basic-1.png
</button>
</p>
<div id="fig-fig_pca_basic-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># PC1-3
fig_PC12 &lt;- ggplot(subset(pca_data, inhib_conc %in% conditions)) + 
  geom_point(aes(x = PC1, y = PC2, shape = stimulus, color = inhib_text_new), size = 3) +
  scale_color_manual(values = select_colors, breaks = names(select_colors), name = &quot;Inhibitor&quot;) +
  scale_shape_manual(values = c(16, 17), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) + 
  ggtitle(&quot;PC1 vs PC2&quot;) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  textsize
# fig_PC12
fig_PC13 &lt;- ggplot(subset(pca_data, inhib_conc %in% conditions)) + 
  geom_point(aes(x = PC1, y = PC3, shape = stimulus, color = inhib_text_new), size = 3) +
  scale_color_manual(values = select_colors, breaks = names(select_colors), name = &quot;Inhibitor&quot;) +
  scale_shape_manual(values = c(16, 17), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  ggtitle(&quot;PC1 vs PC3&quot;) +
  theme_bw() +
  theme(legend.position = &quot;none&quot;) +
  textsize
# fig_PC13
fig_PC23 &lt;- ggplot(subset(pca_data, inhib_conc %in% conditions)) + 
  geom_point(aes(x = PC2, y = PC3, shape = stimulus, color = inhib_text_new), size = 3) +
  scale_color_manual(values = select_colors, breaks = names(select_colors), name = &quot;Inhibitor&quot;) +
  scale_shape_manual(values = c(16, 17), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  xlab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  ggtitle(&quot;PC2 vs PC3&quot;) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;) +
  textsize
# fig_PC23
fig_pca_PC13 &lt;- plot_grid(fig_PC12, fig_PC13, fig_PC23, ncol = 3, align = &quot;h&quot;, rel_widths = c(1, 1, 1.4))
print(fig_pca_PC13)</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_pca_basic-2.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_pca_basic-2">
Past versions of fig_pca_basic-2.png
</button>
</p>
<div id="fig-fig_pca_basic-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_pca_basic,
  filename = &quot;output/figures/poster/fig_pca_basic.jpg&quot;,
  width = 4.5,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig_pca_PC13,
  filename = &quot;output/figures/poster/fig_pca_PC13.jpg&quot;,
  width = 10,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
<p>Detailed: all concentrations</p>
<pre class="r"><code># Figure
select_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, &quot;100 uM iSYK&quot; = &quot;#9dce5c&quot;, &quot;100 uM iBTK&quot; = &quot;#5c9dce&quot;, &quot;100 uM iNFkB&quot; = &quot;#ce5c9d&quot;)

fig_pca_detail1 &lt;- ggplot(drop_na(dplyr::filter(pca_data, !(stimulus == &quot;PBS&quot; &amp; inhib_conc %in% order_conc[2:5])))) + 
  geom_point(aes(x = PC1, y = PC3, shape = paste(stimulus, &quot;-&quot;, inhib_conc), color = paste(stimulus, &quot;-&quot;, inhib_conc)), size = 3) +
  scale_color_manual(values = c(&quot;grey&quot;, colors_orange9[3:7]),
                     breaks = c(&quot;PBS - 0 uM (DMSO)&quot;, &quot;aIg+H2O2 - 0 uM (DMSO)&quot;, &quot;aIg+H2O2 - 0.1 uM&quot;, &quot;aIg+H2O2 - 1 uM&quot;, &quot;aIg+H2O2 - 10 uM&quot;, &quot;aIg+H2O2 - 100 uM&quot;),
                     # labels = c(),
                     name = &quot;Treatment&quot;) +
  # scale_color_manual(values = c(colors_orange9[3:7], colors_grey9[3:7]), name = &quot;Stim - Inhib conc&quot;) +
  # scale_color_manual(values = select_colors, breaks = names(select_colors), name = &quot;Inhibitor&quot;) +
  scale_shape_manual(values = c(16, 17, 17, 17, 17, 17),
                     breaks = c(&quot;PBS - 0 uM (DMSO)&quot;, &quot;aIg+H2O2 - 0 uM (DMSO)&quot;, &quot;aIg+H2O2 - 0.1 uM&quot;, &quot;aIg+H2O2 - 1 uM&quot;, &quot;aIg+H2O2 - 10 uM&quot;, &quot;aIg+H2O2 - 100 uM&quot;),
                     name = &quot;Treatment&quot;) +
  facet_wrap(~inhibitor, ncol = 1) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  theme_bw() +
  # theme(legend.position = &quot;none&quot;) +
  textsize
fig_pca_detail1</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_pca_detail-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_pca_detail-1">
Past versions of fig_pca_detail-1.png
</button>
</p>
<div id="fig-fig_pca_detail-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>pca_data_select &lt;- pca_data %&gt;%
  dplyr::filter(!(stimulus == &quot;PBS&quot; &amp; inhib_conc %in% order_conc[2:5])) %&gt;%
  drop_na() %&gt;%
  dplyr::arrange(factor(stimulus, levels = order_stim), factor(inhibitor, levels =  order_inhib))

pca_legend_shapes &lt;- c(16, 16, 16, 
                       17, 17, 17, 17, 17, 
                       17, 17, 17, 17, 17, 
                       17, 17, 17, 17, 17)
pca_legend_values &lt;- c(&quot;grey&quot;, &quot;grey&quot;, &quot;grey&quot;, 
                       &quot;#737373&quot;, &quot;#d7ebbd&quot;, &quot;#c4e19d&quot;,  &quot;#b0d77c&quot;, &quot;#9dce5c&quot;, 
                       &quot;#737373&quot;, &quot;#bdd7eb&quot;, &quot;#9dc4e1&quot;,  &quot;#7cb0d7&quot;, &quot;#5c9dce&quot;, 
                       &quot;#737373&quot;, &quot;#ebbdd7&quot;, &quot;#e19dc4&quot;,  &quot;#d77cb0&quot;, &quot;#ce5c9d&quot;)
pca_legend_levels &lt;- c(&quot;0 uM iSYK (DMSO) PBS&quot;, &quot;0 uM iBTK (DMSO) PBS&quot;, &quot;0 uM iNFkB (DMSO) PBS&quot;, 
                       &quot;0 uM iSYK (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iSYK aIg+H2O2&quot;, &quot;1 uM iSYK aIg+H2O2&quot;, &quot;10 uM iSYK aIg+H2O2&quot;, &quot;100 uM iSYK aIg+H2O2&quot;, 
                       &quot;0 uM iBTK (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iBTK aIg+H2O2&quot;, &quot;1 uM iBTK aIg+H2O2&quot;, &quot;10 uM iBTK aIg+H2O2&quot;, &quot;100 uM iBTK aIg+H2O2&quot;, 
                       &quot;0 uM iNFkB (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iNFkB aIg+H2O2&quot;, &quot;1 uM iNFkB aIg+H2O2&quot;, &quot;10 uM iNFkB aIg+H2O2&quot;, &quot;100 uM iNFkB aIg+H2O2&quot;)
names(pca_legend_shapes) &lt;- pca_legend_levels
names(pca_legend_values) &lt;- pca_legend_levels
pca_legend_breaks &lt;- c(&quot;0 uM iNFkB (DMSO) PBS&quot;, &quot;0 uM iSYK (DMSO) aIg+H2O2&quot;, &quot;0.1 uM iSYK aIg+H2O2&quot;, &quot;1 uM iSYK aIg+H2O2&quot;, &quot;10 uM iSYK aIg+H2O2&quot;, &quot;100 uM iSYK aIg+H2O2&quot;)
pca_legend_labels &lt;- c(&quot;0 uM - Basal&quot;, &quot;0 uM - Activated&quot;, &quot;0.1 uM - Activated&quot;, &quot;1 uM - Activated&quot;, &quot;10 uM - Activated&quot;, &quot;100 uM - Activated&quot;)
pca_legend_labels_alt &lt;- c(&quot;0 uM (basal)&quot;, &quot;0 uM (activated)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

fig_pca_detail2 &lt;- ggplot(pca_data_select) + 
  geom_point(aes(x = PC1, y = PC3, shape = stimulus, color = paste(inhib_conc_text, stimulus)), size = 3) +
  # scale_color_manual(values = colors_inhib_conc) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_breaks,
                     labels = pca_legend_labels_alt,
                     name = &quot;Inhibitor\nconcentration&quot;) +
  # scale_shape_manual(values = pca_legend_shapes,
  #                    breaks = pca_legend_breaks,
  #                    labels = pca_legend_labels,
  #                    name = &quot;Treatment&quot;) +
  scale_shape_manual(values = c(16, 17), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  facet_wrap(~inhibitor, ncol = 1) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  theme_bw() +
  textsize
fig_pca_detail2</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_pca_detail-2.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_pca_detail-2">
Past versions of fig_pca_detail-2.png
</button>
</p>
<div id="fig-fig_pca_detail-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_pca_detail1,
  filename = &quot;output/figures/poster/fig_pca_detail1.jpg&quot;,
  width = 4.5,
  height = 6,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  fig_pca_detail2,
  filename = &quot;output/figures/poster/fig_pca_detail2.jpg&quot;,
  width = 4.5,
  height = 6,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
</div>
<div id="contributing-proteins" class="section level3">
<h3>Contributing proteins</h3>
<p>Protein contributions to each PC (&gt;2.5%) and quality of
representation in PCS</p>
<pre class="r"><code># Idea:
# Visualise proteins contributing to PCs
# Quality of representation: cos2 - size of dot
# Percentage contribution to one PC: contrib - colour of dot

# Select data
data_contrib_2.5p &lt;- as.data.frame(pca_variab$contrib) %&gt;%
  dplyr::filter(Dim.1 &gt;= 2.5 | Dim.2 &gt;= 2.5 | Dim.3 &gt;= 2.5) %&gt;% # | Dim.4 &gt;= 1 | Dim.5 &gt;= 1
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  mutate_all(function(x){x/100}) %&gt;%
  as.matrix()
data_contrib_prot &lt;- factor(rownames(data_contrib_2.5p))

data_cos2_2.5p &lt;- as.data.frame(pca_variab$cos2) %&gt;%
  rownames_to_column(&quot;proteins&quot;) %&gt;%
  dplyr::filter(proteins %in% data_contrib_prot) %&gt;%
  column_to_rownames(&quot;proteins&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  as.matrix()

# Tidy data format
data_cos2_tidy &lt;- as.data.frame(data_cos2_2.5p) %&gt;%
  rownames_to_column(&quot;proteins&quot;) %&gt;%
  pivot_longer(2:4, names_to = &quot;PC&quot;, values_to = &quot;cos2&quot;)
data_contrib_tidy &lt;- as.data.frame(data_contrib_2.5p) %&gt;%
  rownames_to_column(&quot;proteins&quot;) %&gt;%
  pivot_longer(2:4, names_to = &quot;PC&quot;, values_to = &quot;contrib&quot;) %&gt;%
  mutate(contrib = contrib * 100)
data_contrib_all &lt;- full_join(data_cos2_tidy, data_contrib_tidy)

# Figure
# ggplot geom_count
fig_contrib &lt;- ggplot() +
  geom_point(data = data_contrib_all, aes(x = PC, y = factor(proteins, levels = rev(data_contrib_prot)), size = cos2, fill = contrib), color = &quot;black&quot;, pch = 21) +
  scale_size_area(max_size = 5, breaks = c(0, 0.25, 0.5, 0.75, 1), name = &quot;Quality of\nrepresentation\nin PCs&quot;) +
  scale_fill_gradient(low = &quot;#FEDA8B&quot;, high = &quot;#73001a&quot;, name = &quot;Percentage\ncontribution\nwithin one PC&quot;) + #7f2704 #67000d
  labs(x = &quot;&quot;, y = &quot;&quot;) + 
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), axis.text.y = element_text(size = 9), legend.title = element_text(size = 8)) + #legend.position = &quot;none&quot;
  textsize #+ 
  # theme(legend.title = element_text(size = 8), axis.text.y = element_text(size = 8))
fig_contrib</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_contrib-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_contrib-1">
Past versions of fig_contrib-1.png
</button>
</p>
<div id="fig-fig_contrib-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_contrib,
  filename = &quot;output/figures/poster/fig_contributions.jpg&quot;,
  width = 3.5,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
</div>
<div id="protein-examples" class="section level3">
<h3>Protein examples</h3>
<p>Effect of inhibitor concentration on phospho-protein levels:
Increased or decreased levels upon inhibition depending on the inhibitor
in question</p>
<pre class="r"><code># Select data
fig_proteins &lt;- c(&quot;VAV1_Y174&quot;, &quot;IgM&quot;, &quot;S6_S235_S236&quot;) # &quot;PARP_D214&quot;
data_prot_sample &lt;- data_sample %&gt;%
  subset(target_nospace %in% fig_proteins &amp; inhib_conc %in% order_conc) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = fig_proteins))
data_prot_condition &lt;- data_condition %&gt;%
  subset(target_nospace %in% fig_proteins &amp; inhib_conc %in% order_conc) 
fig_proteins_name &lt;- unique(data_prot_sample$target)

# Figure
# VAV1
fig_prot_VAV1 &lt;- ggplot() +
  geom_jitter(data = subset(data_prot_sample, target_nospace == fig_proteins[1]), 
              aes(factor(inhib_conc, levels = order_conc, labels = label_conc), counts_norm, color = stimulus), 
              alpha = 1, size = 1.5, width = 0.1) +
  # geom_errorbar(data = subset(data_prot_condition, target_nospace == fig_proteins[1]), 
  #               aes(factor(inhib_conc, levels = order_conc, labels = label_conc),
  #                   ymin = counts_norm, ymax = counts_norm, group = stimulus, color = stimulus),
  #               width = 0.5, size = 1) +
  geom_line(data = subset(data_prot_condition, target_nospace == fig_proteins[1]), 
                aes(factor(inhib_conc, levels = order_conc, labels = label_conc),
                    counts_norm, group = stimulus, color = stimulus),
                size = 1) +
  facet_wrap(vars(factor(inhibitor, levels = order_inhib))) +
  scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor concentration&quot;, y = &quot;Normalized counts&quot;, title = fig_proteins[1]) +
  theme_bw() +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) + 
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;, axis.title.x = element_blank()) +
  textsize
# fig_prot_VAV1

# IgM
fig_prot_IgM &lt;- ggplot() +
  geom_jitter(data = subset(data_prot_sample, target_nospace == fig_proteins[2]), 
              aes(factor(inhib_conc, levels = order_conc, labels = label_conc), counts_norm, color = stimulus), 
              alpha = 1, size = 1.5, width = 0.1) +
  geom_line(data = subset(data_prot_condition, target_nospace == fig_proteins[2]), 
                aes(factor(inhib_conc, levels = order_conc, labels = label_conc),
                    counts_norm, group = stimulus, color = stimulus),
                size = 1) +
  facet_wrap(vars(factor(inhibitor, levels = order_inhib))) +
  scale_color_manual(values = colors_stim, breaks = names(colors_stim), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  labs(x = &quot;Inhibitor concentration&quot;, y = &quot;Normalized counts&quot;, title = fig_proteins[2]) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;right&quot;, axis.title.x = element_blank()) +
  textsize
# fig_prot_IgM

# S6
fig_prot_S6 &lt;- ggplot() +
  geom_jitter(data = subset(data_prot_sample, target_nospace == fig_proteins[3]), 
              aes(factor(inhib_conc, levels = order_conc, labels = label_conc), counts_norm, color = stimulus), 
              alpha = 1, size = 1.5, width = 0.1) +
  geom_line(data = subset(data_prot_condition, target_nospace == fig_proteins[3]),
                aes(factor(inhib_conc, levels = order_conc, labels = label_conc),
                    counts_norm, group = stimulus, color = stimulus),
                size = 1) +
  facet_wrap(vars(factor(inhibitor, levels = order_inhib))) +
  scale_color_manual(values = colors_stim, breaks = names(colors_stim), labels = c(&quot;Basal&quot;, &quot;Activated&quot;), name = &quot;Stimulation&quot;) +
  labs(x = &quot;Inhibitor concentration&quot;, y = &quot;Normalized counts&quot;, title = fig_proteins[3]) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;bottom&quot;, legend.justification = &quot;right&quot;, legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) +
  textsize
# fig_prot_S6

# Combined figure
# fig_prot &lt;- plot_grid(fig_prot_VAV1, fig_prot_IgM, fig_prot_S6, labels = PANEL_labels[1:3], ncol = 1, rel_heights = c(1, 1, 1, 1.3), label_size = 10, align = &quot;v&quot;)
# fig_prot
fig_prot &lt;- plot_grid(fig_prot_VAV1, fig_prot_S6, ncol = 1, rel_heights = c(1, 1.2), label_size = 10)
fig_prot</code></pre>
<p><img src="figure/x_poster_FASEB.Rmd/fig_prot-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-fig_prot-1">
Past versions of fig_prot-1.png
</button>
</p>
<div id="fig-fig_prot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as jpg and png
ggsave(
  fig_prot,
  filename = &quot;output/figures/poster/fig_proteins.jpg&quot;,
  width = 7,
  height = 6,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   fig,
#   filename = &quot;output/figures/fig.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;fig&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
gc()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ashr_2.2-63                 apeglm_1.26.1              
 [3] MASS_7.3-60.2               ggbreak_0.1.2              
 [5] geomtextpath_0.1.4          ggh4x_0.2.8                
 [7] cowplot_1.1.3               patchwork_1.2.0            
 [9] greekLetters_1.0.2          ggpubr_0.6.0               
[11] ggrepel_0.9.5               scales_1.3.0               
[13] ggthemes_5.1.0              RColorBrewer_1.1-3         
[15] scico_1.5.0                 viridis_0.6.5              
[17] viridisLite_0.4.2           gridExtra_2.3              
[19] mmtable2_0.1.3              gt_0.10.1                  
[21] ggvenn_0.1.10               rgl_1.3.1                  
[23] plotly_4.10.4               ggcorrplot_0.1.4.1         
[25] corrplot_0.92               umap_0.2.10.0              
[27] ggplotify_0.1.2             dendsort_0.3.4             
[29] vsn_3.72.0                  ComplexHeatmap_2.20.0      
[31] platetools_0.1.7            kableExtra_1.4.0           
[33] DESeq2_1.44.0               SummarizedExperiment_1.34.0
[35] Biobase_2.64.0              MatrixGenerics_1.16.0      
[37] matrixStats_1.3.0           GenomicRanges_1.56.1       
[39] GenomeInfoDb_1.40.1         IRanges_2.38.0             
[41] S4Vectors_0.42.0            BiocGenerics_0.50.0        
[43] Seurat_5.1.0                SeuratObject_5.0.2         
[45] sp_2.1-4                    rstatix_0.7.2              
[47] factoextra_1.0.7            FactoMineR_2.11            
[49] readxl_1.4.3                Matrix_1.7-0               
[51] lubridate_1.9.3             forcats_1.0.0              
[53] stringr_1.5.1               dplyr_1.1.4                
[55] purrr_1.0.2                 readr_2.1.5                
[57] tidyr_1.3.1                 tibble_3.2.1               
[59] ggplot2_3.5.1               tidyverse_2.0.0            
[61] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             goftest_1.2-3           DT_0.33                
  [4] TH.data_1.1-2           vctrs_0.6.5             spatstat.random_3.2-3  
  [7] digest_0.6.36           png_0.1-8               shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       magick_2.8.3            SQUAREM_2021.1         
 [16] reshape2_1.4.4          httpuv_1.6.15           foreach_1.5.2          
 [19] withr_3.0.0             xfun_0.45               ggfun_0.1.5            
 [22] survival_3.6-4          memoise_2.0.1           emmeans_1.10.2         
 [25] systemfonts_1.1.0       ragg_1.3.2              zoo_1.8-12             
 [28] GlobalOptions_0.1.2     pbapply_1.7-2           promises_1.3.0         
 [31] scatterplot3d_0.3-44    httr_1.4.7              globals_0.16.3         
 [34] fitdistrplus_1.1-11     ps_1.7.6                rstudioapi_0.16.0      
 [37] UCSC.utils_1.0.0        miniUI_0.1.1.1          generics_0.1.3         
 [40] base64enc_0.1-3         processx_3.8.4          zlibbioc_1.50.0        
 [43] polyclip_1.10-6         GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [46] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [49] S4Arrays_1.4.1          preprocessCore_1.66.0   hms_1.1.3              
 [52] irlba_2.3.5.1           colorspace_2.1-0        ROCR_1.0-11            
 [55] reticulate_1.38.0       spatstat.data_3.1-2     magrittr_2.0.3         
 [58] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [61] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [64] scattermore_1.2         RcppAnnoy_0.0.22        pillar_1.9.0           
 [67] nlme_3.1-164            iterators_1.0.14        compiler_4.4.1         
 [70] RSpectra_0.16-1         stringi_1.8.4           tensor_1.5             
 [73] plyr_1.8.9              crayon_1.5.3            abind_1.4-5            
 [76] truncnorm_1.0-9         gridGraphics_0.5-1      emdbook_1.3.13         
 [79] locfit_1.5-9.10         bit_4.0.5               sandwich_3.1-0         
 [82] whisker_0.4.1           codetools_0.2-20        multcomp_1.4-25        
 [85] textshaping_0.4.0       openssl_2.2.0           bslib_0.7.0            
 [88] GetoptLong_1.0.5        mime_0.12               splines_4.4.1          
 [91] circlize_0.4.16         Rcpp_1.0.12             fastDummies_1.7.3      
 [94] cellranger_1.1.0        Rttf2pt1_1.3.12         leaps_3.2              
 [97] knitr_1.47              utf8_1.2.4              clue_0.3-65            
[100] fs_1.6.4                listenv_0.9.1           ggsignif_0.6.4         
[103] estimability_1.5.1      callr_3.7.6             statmod_1.5.0          
[106] tzdb_0.4.0              svglite_2.1.3           pkgconfig_2.0.3        
[109] tools_4.4.1             cachem_1.1.0            numDeriv_2016.8-1.1    
[112] fastmap_1.2.0           rmarkdown_2.27          ica_1.0-3              
[115] broom_1.0.6             sass_0.4.9              coda_0.19-4.1          
[118] BiocManager_1.30.23     dotCall64_1.1-1         carData_3.0-5          
[121] RANN_2.6.1              farver_2.1.2            yaml_2.3.8             
[124] cli_3.6.3               leiden_0.4.3.1          lifecycle_1.0.4        
[127] askpass_1.2.0           uwot_0.2.2              mvtnorm_1.2-5          
[130] backports_1.5.0         BiocParallel_1.38.0     timechange_0.3.0       
[133] gtable_0.3.5            rjson_0.2.21            ggridges_0.5.6         
[136] progressr_0.14.0        parallel_4.4.1          limma_3.60.3           
[139] jsonlite_1.8.8          RcppHNSW_0.6.0          bit64_4.0.5            
[142] multcompView_0.1-10     assertthat_0.2.1        Rtsne_0.17             
[145] yulab.utils_0.1.4       spatstat.utils_3.0-5    bdsmatrix_1.3-7        
[148] highr_0.11              jquerylib_0.1.4         lazyeval_0.2.2         
[151] shiny_1.8.1.1           htmltools_0.5.8.1       affy_1.82.0            
[154] sctransform_0.4.1       glue_1.7.0              spam_2.10-0            
[157] XVector_0.44.0          rprojroot_2.0.4         flashClust_1.01-2      
[160] invgamma_1.1            igraph_2.0.3            extrafontdb_1.0        
[163] R6_2.5.1                labeling_0.4.3          bbmle_1.0.25.1         
[166] cluster_2.1.6           aplot_0.2.3             DelayedArray_0.30.1    
[169] tidyselect_1.2.1        xml2_1.3.6              car_3.1-2              
[172] future_1.33.2           munsell_0.5.1           KernSmooth_2.23-24     
[175] affyio_1.74.0           data.table_1.15.4       htmlwidgets_1.6.4      
[178] rlang_1.1.4             spatstat.sparse_3.1-0   extrafont_0.19         
[181] spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
