<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="mwitmond" />

<meta name="date" content="2024-03-11" />

<title>Paper supplementary figures</title>

<script src="site_libs/header-attrs-2.27/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/readable.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}

.tocify-subheader {
  display: inline;
}
.tocify-subheader .tocify-item {
  font-size: 0.95em;
}

</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Chapters on DLBCL</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Paper
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="y_paper_main.html">Main</a>
    </li>
    <li>
      <a href="z_paper_suppl.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Proteins ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="t_proteins_main.html.html">Main</a>
    </li>
    <li>
      <a href="u_proteins_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    States ch
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="v_states_main.html.html">Main</a>
    </li>
    <li>
      <a href="w_states_suppl.html.html">Supplementary</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Flow
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="a_flow_gating.html">Gating</a>
    </li>
    <li>
      <a href="b_flow_ann.html">Annotation</a>
    </li>
    <li>
      <a href="c_flow_vis.html">Visualisation</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    IDseq
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="d_IDseq_count.html">Count tables</a>
    </li>
    <li>
      <a href="DS108_p_seq_comparisons.html">DS108 clean</a>
    </li>
    <li>
      <a href="DS108_q_seq_thresholds.html">DS108 thresholds</a>
    </li>
    <li>
      <a href="DS108_r_seq_pca.html">DS108 PCA</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Paper supplementary figures</h1>
<h4 class="author">mwitmond</h4>
<h4 class="date">2024-03-11</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-02-12
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>chapters_DLBCL/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20241022code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20241022)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20241022code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20241022)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong1ffbd94">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 1ffbd94
</a>
</p>
</div>
<div id="strongRepositoryversionstrong1ffbd94"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 1ffbd94.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/figure/

Untracked files:
    Untracked:  code/packages_FACS.R
    Untracked:  code/packages_seq.R
    Untracked:  data/AL006_InhibCombiTest/
    Untracked:  data/DS102_StimTest/
    Untracked:  data/DS103_ViabilityTest/
    Untracked:  data/DS104_StimInhibTest/
    Untracked:  data/DS108_StimInhibIDseq/
    Untracked:  data/DS113_InhibCombiIDseq/
    Untracked:  output/DS102_StimTest/
    Untracked:  output/DS103_ViabilityTest/
    Untracked:  output/DS104_StimInhibTest/
    Untracked:  output/DS108_StimInhibIDseq/
    Untracked:  output/DS113_InhibCombiIDseq/
    Untracked:  output/figures/
    Untracked:  output/network_visual_DLBCL/
    Untracked:  output/network_visual_paper/

Unstaged changes:
    Deleted:    analysis/about.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/z_paper_suppl.Rmd</code>) and
HTML (<code>docs/z_paper_suppl.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
1ffbd94
</td>
<td>
mwitmond
</td>
<td>
2025-02-12
</td>
<td>
Revised chapter figs
</td>
</tr>
<tr>
<td>
html
</td>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
6559698
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
<td>
Paper figs + analysis files for ch figs
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="set-up" class="section level2">
<h2>Set-up</h2>
<pre class="r"><code>row_order = c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, &quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;)
col_order = c(&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;10&quot;, &quot;11&quot;, &quot;12&quot;)
panel_labels &lt;- c(&quot;a&quot;, &quot;b&quot;, &quot;c&quot;,&quot;d&quot;, &quot;e&quot;, &quot;f&quot;, &quot;g&quot;, &quot;h&quot;, &quot;i&quot;, &quot;j&quot;, &quot;k&quot;, &quot;l&quot;, &quot;m&quot;)
PANEL_labels &lt;- c(&quot;A&quot;, &quot;B&quot;, &quot;C&quot;,&quot;D&quot;, &quot;E&quot;, &quot;F&quot;, &quot;G&quot;, &quot;H&quot;, &quot;I&quot;, &quot;J&quot;, &quot;K&quot;, &quot;L&quot;, &quot;M&quot;)

textsize &lt;- theme(axis.text.x = element_text(colour = &quot;grey&quot;, size = 11), #, face = &quot;bold&quot;
                  axis.text.y = element_text(colour = &quot;grey&quot;, size = 11),
                  axis.title = element_text(colour = &quot;black&quot;, size = 12), 
                  legend.title = element_text(colour = &quot;black&quot;, size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = &quot;grey&quot;, size = 11), 
                  strip.text.x = element_text(colour = &quot;black&quot;, size = 12)
)

textsize_small &lt;- theme(text = element_text(size = 9, family = &quot;sans&quot;, colour = &quot;black&quot;),
                        plot.title = element_text(size = 10)
)
# textsize_small &lt;- theme(axis.text.x = element_text(colour = &quot;black&quot;, size = 9),
#                         axis.text.y = element_text(colour = &quot;black&quot;, size = 9),
#                         axis.title = element_text(colour = &quot;black&quot;, size = 10),
#                         legend.text = element_text(colour = &quot;black&quot;, size = 9),
#                         title = element_text(color = &quot;black&quot;, size = 10),
#                         strip.text.x = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;),
#                         strip.text.y = element_text(colour = &quot;black&quot;, size = 10, face = &quot;bold&quot;)
# )</code></pre>
<pre class="r"><code>colors_nord &lt;- c(&quot;#2e3440&quot;, &quot;#3b4252&quot;, &quot;#434c5e&quot;, &quot;#4c566a&quot;, 
                 &quot;#d8dee9&quot;, &quot;#e5e9f0&quot;, &quot;#eceff4&quot;, 
                 &quot;#8fbcbb&quot;, &quot;#88c0d0&quot;, &quot;#81a1c1&quot;, &quot;#5e81ac&quot;, 
                 &quot;#bf616a&quot;, &quot;#d08770&quot;, &quot;#ebcb8b&quot;, &quot;#a3be8c&quot;, &quot;#b48ead&quot;)

colors_toll_bright &lt;- c(&quot;#4477AA&quot;, &quot;#66CCEE&quot;, &quot;#228833&quot;, &quot;#CCBB44&quot;, &quot;#EE6677&quot;, &quot;#AA3377&quot;, &quot;#BBBBBB&quot;)
colors_toll_mute &lt;- c(&quot;#332288&quot;, &quot;#88CCEE&quot;, &quot;#44AA99&quot;, &quot;#117733&quot;, &quot;#999933&quot;, &quot;#DDCC77&quot;, &quot;#CC6677&quot;, &quot;#882255&quot;, &quot;#AA4499&quot;, &quot;#DDDDDD&quot;)
colors_toll_br &lt;- c(&quot;#364B9A&quot;, &quot;#4A7BB7&quot;, &quot;#6EA6CD&quot;, &quot;#98CAE1&quot;, &quot;#C2E4EF&quot;, &quot;#EAECCC&quot;, &quot;#FEDA8B&quot;, &quot;#FDB366&quot;, &quot;#F67A4B&quot;, &quot;#DD3D2D&quot;, &quot;#A50026&quot;)

colors_blue9 &lt;- c(&quot;#f7fbff&quot;, &quot;#deebf7&quot;, &quot;#c6dbef&quot;, &quot;#9ecae1&quot;, &quot;#6baed6&quot;, &quot;#4292c6&quot;, &quot;#2171b5&quot;, &quot;#08519c&quot;, &quot;#08306b&quot;)
colors_green9 &lt;- c(&quot;#f7fcf5&quot;, &quot;#e5f5e0&quot;, &quot;#c7e9c0&quot;, &quot;#a1d99b&quot;, &quot;#74c476&quot;, &quot;#41ab5d&quot;, &quot;#238b45&quot;, &quot;#006d2c&quot;, &quot;#00441b&quot;)
colors_purple9 &lt;- c(&quot;#fcfbfd&quot;, &quot;#efedf5&quot;, &quot;#dadaeb&quot;, &quot;#bcbddc&quot;, &quot;#9e9ac8&quot;, &quot;#807dba&quot;, &quot;#6a51a3&quot;, &quot;#54278f&quot;, &quot;#3f007d&quot;)
colors_red9 &lt;- c(&quot;#fff5f0&quot;, &quot;#fee0d2&quot;, &quot;#fcbba1&quot;, &quot;#fc9272&quot;, &quot;#fb6a4a&quot;, &quot;#ef3b2c&quot;, &quot;#cb181d&quot;, &quot;#a50f15&quot;, &quot;#67000d&quot;)
colors_orange9 &lt;- c(&quot;#fff5eb&quot;, &quot;#fee6ce&quot;, &quot;#fdd0a2&quot;, &quot;#fdae6b&quot;, &quot;#fd8d3c&quot;, &quot;#f16913&quot;, &quot;#d94801&quot;, &quot;#a63603&quot;, &quot;#7f2704&quot;)
colors_grey9 &lt;- c(&quot;#ffffff&quot;, &quot;#f0f0f0&quot;, &quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, &quot;#969696&quot;, &quot;#737373&quot;, &quot;#525252&quot;, &quot;#252525&quot;, &quot;#000000&quot;)
colors_yb9 &lt;- c(&quot;#ffffd9&quot;, &quot;#edf8b1&quot;, &quot;#c7e9b4&quot;, &quot;#7fcdbb&quot;, &quot;#41b6c4&quot;, &quot;#1d91c0&quot;, &quot;#225ea8&quot;, &quot;#253494&quot;, &quot;#081d58&quot;)
colors_pr9 &lt;- c(&quot;#f7f4f9&quot;, &quot;#e7e1ef&quot;, &quot;#d4b9da&quot;, &quot;#c994c7&quot;, &quot;#df65b0&quot;, &quot;#e7298a&quot;, &quot;#ce1256&quot;, &quot;#980043&quot;, &quot;#67001f&quot;)

# Variable-specific colors
colors_cell &lt;- c(&quot;HBL1&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8&quot; = &quot;#332288&quot;)
colors_stim &lt;- c(&quot;PBS&quot; = &quot;#737373&quot;, &quot;aIg+H2O2&quot; = &quot;#FDB366&quot;, 
                 &quot;Basal&quot; = &quot;#737373&quot;, &quot;Activated&quot; = &quot;#FDB366&quot;)
colors_cell_stim &lt;- c(&quot;HBL1 - PBS&quot; = &quot;#c994c7&quot;, &quot;HBL1 - aIg+H2O2&quot; = &quot;#AA4499&quot;, &quot;OCI-Ly8 - PBS&quot; = &quot;#88CCEE&quot;, &quot;OCI-Ly8 - aIg+H2O2&quot; = &quot;#332288&quot;)
# colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#228833&quot;, &quot;iBTK&quot; = &quot;#4477AA&quot;, &quot;iNFkB&quot; = &quot;#AA3377&quot;)
colors_inhib &lt;- c(&quot;iSYK&quot; = &quot;#9dce5c&quot;, &quot;iBTK&quot; = &quot;#5c9dce&quot;, &quot;iNFkB&quot; = &quot;#ce5c9d&quot;)
colors_conc &lt;- c(&quot;#737373&quot;, &quot;#dadaeb&quot;, &quot;#9e9ac8&quot;, &quot;#6a51a3&quot;, &quot;#3f007d&quot;)
colors_iSYK &lt;- colors_green9[c(3, 5, 6, 8, 9)]
colors_iBTK &lt;- colors_blue9[c(3, 5, 6, 8, 9)]
colors_iNFkB &lt;- colors_pr9[c(3, 5, 6, 8, 9)]
colors_conc_stim &lt;- colors_orange9[c(2, 3, 5, 6, 8, 9)]
colors_conc_PBS &lt;- colors_grey9[c(2, 3, 5, 6, 8, 9)]</code></pre>
</div>
<div id="suppl.-figure-s1" class="section level2">
<h2>Suppl. figure S1</h2>
<p><em>Detailed schematic of the B-cell signaling network, with proteins
measured in ID-seq panel indicated.</em></p>
<p>Input:</p>
<ul>
<li>Cytoscape blank network</li>
</ul>
<div id="panels" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS1_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS1_panelA

# PNG
png_suppl_figS1 &lt;- magick::image_read(&quot;output/figures/non_R_figs/BCR_network_organised.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
png_suppl_figS1</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS1_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS1_panelA-1">
Past versions of suppl_figS1_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS1_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Legend
png_suppl_figS1_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections_blank.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE)
# png_suppl_figS1_legend</code></pre>
</div>
</div>
<div id="combined-fig" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS1 &lt;- plot_grid(png_suppl_figS1, png_suppl_figS1_legend, ncol = 1, rel_heights = c(8, 1), label_size = 10)
suppl_figS1</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS1_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS1_combi-1">
Past versions of suppl_figS1_combi-1.png
</button>
</p>
<div id="fig-suppl_figS1_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS1,
  filename = &quot;output/figures/paper/suppl_figS1.pdf&quot;,
  width = 7.3,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS1,
  filename = &quot;output/figures/paper/suppl_figS1.jpg&quot;,
  width = 7.3,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S1: Detailed schematic of the B-cell signaling network,
with proteins measured in ID-seq panel indicated.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS1&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb) max used  (Mb)
Ncells 10602175 566.3   19604257 1047.0 12335743 658.8
Vcells 19458613 148.5   31982714  244.1 26548076 202.6</code></pre>
</div>
</div>
<div id="suppl.-figure-s2" class="section level2">
<h2>Suppl. figure S2</h2>
<p><em>Increased activation upon treatment with both anti-Ig and H2O2
combined compared to treatment with anti-Ig or H2O2 alone.</em></p>
<p>Input:</p>
<ul>
<li>DS102 phosphoflow data</li>
</ul>
<div id="data" class="section level3">
<h3>Data</h3>
<pre class="r"><code>data_suppl_figS2 &lt;- read_csv(&quot;output/DS102_StimTest/flow_ann/flow_data_DS102.csv&quot;)
medians_suppl_figS2 &lt;- read_csv(&quot;output/DS102_StimTest/flow_ann/flow_medians_DS102.csv&quot;)</code></pre>
<pre class="r"><code>proteins_suppl_figS2 &lt;- c(&quot;pCD79a (Y182)&quot;, &quot;pSYK (Y525/Y526)&quot;) # &quot;pPLCy2 (Y759)&quot;
stim_suppl_figS2 &lt;- c(&quot;aIg&quot;, &quot;H2O2&quot;, &quot;H2O2 + aIg&quot;)
stim_labels_suppl_figS2 &lt;- c(&quot;Anti-Ig&quot;, &quot;H2O2&quot;, &quot;Combination\nAnti-Ig + H2O2&quot;)
conc_suppl_figS2 &lt;- c(&quot;PBS&quot;, 
                      &quot;10 ug/mL aIg&quot;, &quot;20 ug/mL aIg&quot;,
                      &quot;10 mM H2O2&quot;, &quot;20 mM H2O2&quot;, 
                      &quot;15 H2O2 + 15 aIg&quot;)
conc_labels_suppl_figS2 &lt;- c(&quot;PBS&quot;, 
                             &quot;10 ug/mL\nanti-Ig&quot;, &quot;20 ug/mL\nanti-Ig&quot;,
                             &quot;10 mM\nH2O2&quot;, &quot;20 mM\nH2O2&quot;,
                             &quot;15 anti-Ig +\n15 H2O2&quot;)
colors_conc_suppl_figS2 &lt;- c(&quot;PBS&quot; = &quot;#737373&quot;, 
                             &quot;10 ug/mL aIg&quot; = &quot;#a1d99b&quot;, &quot;20 ug/mL aIg&quot; = &quot;#74c476&quot;,
                             &quot;10 mM H2O2&quot; = &quot;#98CAE1&quot;, &quot;20 mM H2O2&quot; = &quot;#6EA6CD&quot;, 
                             &quot;15 H2O2 + 15 aIg&quot; = &quot;#FDB366&quot;)</code></pre>
</div>
<div id="panels-1" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-1" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Ridge test
# Data
data_suppl_figS2A &lt;- data_suppl_figS2 %&gt;%
  subset(cell_line == &quot;HBL1&quot; &amp; protein %in% proteins_suppl_figS2 &amp; conc_text %in% conc_suppl_figS2)

# Figure
suppl_figS2_panelA_alt &lt;- ggplot(data_suppl_figS2A, aes(x = fluorescence)) +
  geom_density_ridges(
    aes(y = factor(conc_text, levels = rev(conc_suppl_figS2)), fill = factor(conc_text, levels = conc_suppl_figS2)),
    scale = 1.5,
    # fill = &quot;#006699&quot;,
    alpha = 0.6
  ) +
  geom_vline(aes(xintercept = zero_quant97.5), linewidth = 0.5) +
  facet_wrap(vars(factor(protein, levels = proteins_suppl_figS2))) +
  scale_x_logicle() + # logicle scale instead of log10 scale
  scale_fill_manual(values = colors_conc_suppl_figS2, labels = conc_labels_suppl_figS2, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Fluorescent intensity&quot;, y = &quot;&quot;) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.minor = element_blank()) +
  textsize_small
suppl_figS2_panelA_alt</code></pre>
<pre class="r"><code># Data
data_suppl_figS2A &lt;- medians_suppl_figS2 %&gt;%
  subset(cell_line == &quot;HBL1&quot; &amp; protein == proteins_suppl_figS2[1] &amp; conc_text %in% conc_suppl_figS2)
data_suppl_figS2A_stat &lt;- data_suppl_figS2 %&gt;%
  subset(cell_line == &quot;HBL1&quot; &amp; protein %in% proteins_suppl_figS2[1] &amp; conc_text %in% conc_suppl_figS2)

# Statistics
# Kruskal-Wallis test
figS2A_kruskal &lt;- data_suppl_figS2A_stat %&gt;%
  # group_by(target_nospace) %&gt;%
  kruskal_test(fluorescence ~ conc_text)
figS2A_kruskal</code></pre>
<pre><code># A tibble: 1 × 6
  .y.               n statistic    df     p method        
* &lt;chr&gt;         &lt;int&gt;     &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;         
1 fluorescence 234536   151977.     5     0 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS2A_dunn &lt;- data_suppl_figS2A_stat %&gt;%
  # group_by(target_nospace) %&gt;%
  dunn_test(fluorescence ~ conc_text, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group1 == &quot;15 H2O2 + 15 aIg&quot;) | (group2 == &quot;15 H2O2 + 15 aIg&quot;)
      ) %&gt;%
# group_by(target_nospace) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;conc_text&quot;, ref.group = &quot;15 H2O2 + 15 aIg&quot;)
  mutate(xmin = c(4, 2, 6, 6, 6), 
         xmax = c(6, 6, 5, 3, 1), 
         plot_order = c(4, 2, 5, 3, 1)) %&gt;%
  arrange(plot_order)
figS2A_dunn</code></pre>
<pre><code># A tibble: 5 × 12
  .y.   group1 group2    n1    n2 statistic     p p.adj p.adj.signif  xmin  xmax
  &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt; &lt;dbl&gt;
1 fluo… 15 H2… PBS    37262 40876    -346.      0     0 ****             6     1
2 fluo… 10 ug… 15 H2… 37243 37262     230.      0     0 ****             2     6
3 fluo… 15 H2… 20 ug… 37262 38679     -90.2     0     0 ****             6     3
4 fluo… 10 mM… 15 H2… 40257 37262     149.      0     0 ****             4     6
5 fluo… 15 H2… 20 mM… 37262 40219     -87.5     0     0 ****             6     5
# ℹ 1 more variable: plot_order &lt;dbl&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS2_panelA &lt;- ggplot(data_suppl_figS2A) +
  geom_col(aes(
    x = factor(conc_text, levels = conc_suppl_figS2, labels = conc_labels_suppl_figS2),
    y = fluorescence_median, 
    group = factor(conc_text, levels = conc_suppl_figS2),
    fill = factor(conc_text, levels = conc_suppl_figS2)),
    position = &quot;dodge2&quot;) + 
  stat_pvalue_manual(figS2A_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.01, 
                     y.position = 15150, step.increase = 0.05, size = 2.25) +
  facet_wrap(vars(factor(protein, levels = proteins_suppl_figS2)), scale = &quot;free_y&quot;) +
  scale_fill_manual(values = colors_conc_suppl_figS2, labels = conc_labels_suppl_figS2, name = &quot;Stimulus&quot;) +
  labs(x = &quot;&quot;, y = &quot;MFI&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;none&quot;, panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), axis.text.x = element_text(size = 5)) #+
# suppl_figS2_panelA

# Figure alt
suppl_figS2_panelA_alt &lt;- ggplot() +
  geom_jitter(data = data_suppl_figS2A_stat,
              aes(
                x = factor(conc_text, levels = conc_suppl_figS2, labels = conc_labels_suppl_figS2),
                y = fluorescence,
                group = factor(conc_text, levels = conc_suppl_figS2),
                color = factor(conc_text, levels = conc_suppl_figS2)),
              # position = &quot;dodge2&quot;,
              width = 0.2, size = 0.2, alpha = 0.1
              ) +
  geom_errorbar(data = data_suppl_figS2A,
                aes(x = factor(conc_text, levels = conc_suppl_figS2, labels = conc_labels_suppl_figS2),
                    ymin = fluorescence_median, ymax = fluorescence_median,
                    group = factor(conc_text, levels = conc_suppl_figS2)),
                color = &quot;black&quot;, width = 0.5, size = 0.5) +
  stat_pvalue_manual(figS2A_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.01,
                     y.position = 7e4, step.increase = 0.05, size = 2.25) +
  coord_cartesian(ylim = c(0, 1e5)) +
  facet_wrap(vars(factor(protein, levels = proteins_suppl_figS2)), scale = &quot;free_y&quot;) +
  # scale_y_log10() +
  scale_color_manual(values = colors_conc_suppl_figS2, labels = conc_labels_suppl_figS2, name = &quot;Stimulus&quot;) +
  labs(x = &quot;&quot;, y = &quot;Fluorescent intensity (A.U.)&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;none&quot;, panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), axis.text.x = element_text(size = 6))
suppl_figS2_panelA_alt</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS2_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS2_panelA-1">
Past versions of suppl_figS2_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS2_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Data
data_suppl_figS2B &lt;- medians_suppl_figS2 %&gt;%
  subset(cell_line == &quot;HBL1&quot; &amp; protein == proteins_suppl_figS2[2] &amp; conc_text %in% conc_suppl_figS2)
data_suppl_figS2B_stat &lt;- data_suppl_figS2 %&gt;%
  subset(cell_line == &quot;HBL1&quot; &amp; protein %in% proteins_suppl_figS2[2] &amp; conc_text %in% conc_suppl_figS2)

# Statistics
# Kruskal-Wallis test
figS2B_kruskal &lt;- data_suppl_figS2B_stat %&gt;%
  # group_by(target_nospace) %&gt;%
  kruskal_test(fluorescence ~ conc_text)
figS2B_kruskal</code></pre>
<pre><code># A tibble: 1 × 6
  .y.               n statistic    df     p method        
* &lt;chr&gt;         &lt;int&gt;     &lt;dbl&gt; &lt;int&gt; &lt;dbl&gt; &lt;chr&gt;         
1 fluorescence 234536   155207.     5     0 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS2B_dunn &lt;- data_suppl_figS2B_stat %&gt;%
  # group_by(target_nospace) %&gt;%
  dunn_test(fluorescence ~ conc_text, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group1 == &quot;15 H2O2 + 15 aIg&quot;) | (group2 == &quot;15 H2O2 + 15 aIg&quot;)
      ) %&gt;%
# group_by(target_nospace) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;conc_text&quot;, ref.group = &quot;15 H2O2 + 15 aIg&quot;)
  mutate(xmin = c(4, 2, 6, 6, 6), 
         xmax = c(6, 6, 5, 3, 1), 
         plot_order = c(4, 2, 5, 3, 1)) %&gt;%
  arrange(plot_order)
figS2B_dunn</code></pre>
<pre><code># A tibble: 5 × 12
  .y.       group1 group2    n1    n2 statistic         p     p.adj p.adj.signif
  &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;     &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;       
1 fluoresc… 15 H2… PBS    37262 40876    -328.  0         0         ****        
2 fluoresc… 10 ug… 15 H2… 37243 37262     220.  0         0         ****        
3 fluoresc… 15 H2… 20 ug… 37262 38679    -135.  0         0         ****        
4 fluoresc… 10 mM… 15 H2… 40257 37262      89.1 0         0         ****        
5 fluoresc… 15 H2… 20 mM… 37262 40219     -29.3 2.40e-189 2.40e-189 ****        
# ℹ 3 more variables: xmin &lt;dbl&gt;, xmax &lt;dbl&gt;, plot_order &lt;dbl&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS2_panelB &lt;- ggplot(data_suppl_figS2B) +
  geom_col(aes(
    x = factor(conc_text, levels = conc_suppl_figS2, labels = conc_labels_suppl_figS2),
    y = fluorescence_median, 
    group = factor(conc_text, levels = conc_suppl_figS2),
    fill = factor(conc_text, levels = conc_suppl_figS2)),
    position = &quot;dodge2&quot;) + 
  stat_pvalue_manual(figS2B_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.01,
                     y.position = 42000, step.increase = 0.05, size = 2.25) +
  facet_wrap(vars(factor(protein, levels = proteins_suppl_figS2))) +
  scale_fill_manual(values = colors_conc_suppl_figS2, labels = conc_labels_suppl_figS2, name = &quot;Stimulus&quot;) +
  labs(x = &quot;&quot;, y = &quot;MFI&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;right&quot;, panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), axis.text.x = element_text(size = 5)) #+
# suppl_figS2_panelB

# Figure alt
suppl_figS2_panelB_alt &lt;- ggplot() +
  geom_jitter(data = data_suppl_figS2B_stat,
              aes(
                x = factor(conc_text, levels = conc_suppl_figS2, labels = conc_labels_suppl_figS2),
                y = fluorescence,
                group = factor(conc_text, levels = conc_suppl_figS2),
                color = factor(conc_text, levels = conc_suppl_figS2)),
              # position = &quot;dodge2&quot;,
              width = 0.2, size = 0.2, alpha = 0.1
              ) +
  geom_errorbar(data = data_suppl_figS2B,
                aes(x = factor(conc_text, levels = conc_suppl_figS2, labels = conc_labels_suppl_figS2),
                    ymin = fluorescence_median, ymax = fluorescence_median,
                    group = factor(conc_text, levels = conc_suppl_figS2)),
                color = &quot;black&quot;, width = 0.5, size = 0.5) +
  stat_pvalue_manual(figS2B_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.01,
                     y.position = 2.5e5, step.increase = 0.05, size = 2.25) +
  coord_cartesian(ylim = c(0, 3.1e5)) +
  facet_wrap(vars(factor(protein, levels = proteins_suppl_figS2)), scale = &quot;free_y&quot;) +
  # scale_y_log10() +
  scale_color_manual(values = colors_conc_suppl_figS2, labels = conc_labels_suppl_figS2, name = &quot;Stimulus&quot;) +
  labs(x = &quot;&quot;, y = &quot;Fluorescent intensity (A.U.)&quot;) +
  theme_bw() +
  textsize_small +
  theme(legend.position = &quot;none&quot;, panel.grid.minor = element_blank(), panel.grid.major.x = element_blank(), axis.text.x = element_text(size = 6))
suppl_figS2_panelB_alt</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS2_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS2_panelB-1">
Past versions of suppl_figS2_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS2_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-1" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS2 &lt;- plot_grid(suppl_figS2_panelA_alt, suppl_figS2_panelB_alt, labels = PANEL_labels[c(1, 2)], ncol = 2, rel_widths = c(1, 1), label_size = 10)
suppl_figS2</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS2_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS2_combi-1">
Past versions of suppl_figS2_combi-1.png
</button>
</p>
<div id="fig-suppl_figS2_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS2,
  filename = &quot;output/figures/paper/suppl_figS2.pdf&quot;,
  width = 6.125,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS2,
  filename = &quot;output/figures/paper/suppl_figS2.jpg&quot;,
  width = 6.125,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S2: Increased activation upon treatment with both anti-Ig
and H2O2 combined compared to treatment with anti-Ig or H2O2
alone.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS2&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used  (Mb)
Ncells 10840177 579.0   19604257 1047.0  13514554 721.8
Vcells 30894088 235.8  123837680  944.9 128930891 983.7</code></pre>
</div>
</div>
<div id="suppl.-figure-s3" class="section level2">
<h2>Suppl. figure S3</h2>
<p><em>Schematic overview of the experimental workflow of
ID-seq.</em></p>
<p>Input:</p>
<ul>
<li>BioRender schematic figure</li>
</ul>
<div id="panels-2" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-2" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS3_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  labs(title = &quot;ID-seq workflow&quot;) +
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS3_panelA

# PNG
png_suppl_figS3 &lt;- magick::image_read(&quot;output/figures/non_R_figs/bulkIDseq_workflow_detailed.png&quot;) %&gt;%
  magick::image_ggplot()
png_suppl_figS3</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS3_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS3_panelA-1">
Past versions of suppl_figS3_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS3_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-2" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
png_suppl_figS3</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS3_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS3_combi-1">
Past versions of suppl_figS3_combi-1.png
</button>
</p>
<div id="fig-suppl_figS3_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
# ggsave(
#   suppl_figS3_panelA,
#   filename = &quot;output/figures/paper/suppl_figS3_place.pdf&quot;,
#   width = 7.3,
#   height = 3.5,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  suppl_figS3_panelA,
  filename = &quot;output/figures/paper/suppl_figS3_place.jpg&quot;,
  width = 7.3,
  height = 3.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S3: Schematic overview of the experimental workflow of
ID-seq.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS3&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 10832165 578.6   19604257 1047.0  13514554  721.8
Vcells 18916420 144.4  142801808 1089.5 139788691 1066.6</code></pre>
</div>
</div>
<div id="suppl.-figure-s4" class="section level2">
<h2>Suppl. figure S4</h2>
<p><em>Phospho-specific flow cytometry measurements of CD79a (Y182)
shows high concordance with ID-seq results.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 phosphoflow data</p></li>
<li><p>DS108 ID-seq data</p>
<ul>
<li>Count data</li>
</ul></li>
</ul>
<div id="data-1" class="section level3">
<h3>Data</h3>
<pre class="r"><code># Phospshoflow data 
data_flow_suppl_figS4 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/flow_ann/flow_data_DS108.csv&quot;) %&gt;%
  subset(cell_line == &quot;HBL1&quot;)

# ID-seq data (counts per well/sample and mean per condition)
data_IDsample_suppl_figS4 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
data_IDmean_suppl_figS4 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS4 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS4 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
inhib_suppl_figS4 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

conc_flow_suppl_figS4 &lt;- c(&quot;0 uM&quot;, &quot;100 uM&quot;)
conc_ID_suppl_figS4 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
conc_label_suppl_figS4 &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)

protein_flow_suppl_figS4 &lt;- &quot;pCD79a (Y182)&quot;
protein_ID_suppl_figS4 &lt;- &quot;CD79a_Y182&quot;</code></pre>
</div>
<div id="panels-3" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-3" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
data_flow_suppl_figS4A &lt;- data_flow_suppl_figS4 %&gt;%
  subset(protein == protein_flow_suppl_figS4 &amp; inhib_conc %in% conc_flow_suppl_figS4 &amp; inhibitor %in% inhib_suppl_figS4)

# Figure
suppl_figS4_panelA &lt;- ggplot(data_flow_suppl_figS4A, aes(x = fluorescence)) +
  geom_density_ridges(
    aes(y = factor(inhib_conc, levels = rev(conc_flow_suppl_figS4), labels = rev(conc_label_suppl_figS4)), fill = factor(stimulus, levels = stim_suppl_figS4)),
    scale = 1, 
    alpha = 0.5
  ) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_suppl_figS4)), nrow = 1) +
  scale_x_logicle() + # logicle scale instead of log10 scale
  scale_fill_manual(values = colors_stim, labels = stim_label_suppl_figS4, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Fluorescent intensity (A.U.)&quot;, y = &quot;&quot;) +
  theme_bw() +
  # theme(legend.position = &quot;right&quot;, panel.grid.minor = element_blank()) +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;) +
  textsize_small
suppl_figS4_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS4_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS4_panelA-1">
Past versions of suppl_figS4_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS4_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-1" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
data_IDsample_suppl_figS4B &lt;- data_IDsample_suppl_figS4 %&gt;%
  subset(target_nospace == protein_ID_suppl_figS4 &amp; inhib_conc %in% conc_ID_suppl_figS4)
data_IDmean_suppl_figS4B &lt;- data_IDmean_suppl_figS4 %&gt;%
  subset(target_nospace == protein_ID_suppl_figS4 &amp; inhib_conc %in% conc_ID_suppl_figS4) # &amp; inhibitor %in% inhib_suppl_figS4

# Statistics
# Kruskal-Wallis test
figS4B_kruskal &lt;- data_IDsample_suppl_figS4B %&gt;%
  group_by(inhibitor) %&gt;%
  kruskal_test(counts_norm ~ stim_inhib)
figS4B_kruskal</code></pre>
<pre><code># A tibble: 3 × 7
  inhibitor .y.             n statistic    df      p method        
* &lt;chr&gt;     &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;  &lt;dbl&gt; &lt;chr&gt;         
1 iBTK      counts_norm    12      9.36     3 0.0249 Kruskal-Wallis
2 iNFkB     counts_norm    12      7.82     3 0.0499 Kruskal-Wallis
3 iSYK      counts_norm    12      9.36     3 0.0249 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS4B_dunn &lt;- data_IDsample_suppl_figS4B %&gt;%
  group_by(inhibitor) %&gt;%
  dunn_test(counts_norm ~ stim_inhib, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
      (group1 == &quot;Activated\n100 uM iBTK&quot; &amp; group2 == &quot;Activated\nDMSO ctrl&quot;) | 
      (group1 == &quot;Activated\n100 uM iSYK&quot; &amp; group2 == &quot;Activated\nDMSO ctrl&quot;) |  
      (group1 == &quot;Activated\n100 uM iNFkB&quot; &amp; group2 == &quot;Activated\nDMSO ctrl&quot;)
      ) %&gt;%
  group_by(inhibitor) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;conc_text&quot;, ref.group = &quot;15 H2O2 + 15 aIg&quot;)
  mutate(xmin = c(2, 2, 2), 
         xmax = c(1, 1, 1), 
         plot_order = c(1, 3, 2)) %&gt;%
  arrange(plot_order)
figS4B_dunn</code></pre>
<pre><code># A tibble: 3 × 13
  inhibitor .y.     group1 group2    n1    n2 statistic     p p.adj p.adj.signif
  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       
1 iBTK      counts… &quot;Acti… &quot;Acti…     3     3     1.02  0.308 0.308 ns          
2 iSYK      counts… &quot;Acti… &quot;Acti…     3     3     0.113 0.910 0.910 ns          
3 iNFkB     counts… &quot;Acti… &quot;Acti…     3     3     1.36  0.174 0.174 ns          
# ℹ 3 more variables: xmin &lt;dbl&gt;, xmax &lt;dbl&gt;, plot_order &lt;dbl&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS4_panelB &lt;- ggplot() +
  geom_jitter(data = data_IDsample_suppl_figS4B, 
              aes(x = factor(inhib_conc, levels = conc_ID_suppl_figS4, labels = conc_label_suppl_figS4), 
                  y = counts_norm, 
                  color = factor(stimulus, levels = stim_suppl_figS4)), 
              alpha = 1, size = 1, width = 0.1) +
  geom_errorbar(data = data_IDmean_suppl_figS4B,
                aes(x = factor(inhib_conc, levels = conc_ID_suppl_figS4, labels = conc_label_suppl_figS4), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stimulus, levels = stim_suppl_figS4),
                    color = factor(stimulus, levels = stim_suppl_figS4)),
                width = 0.35, size = 0.75) +
  # stat_pvalue_manual(figS4B_dunn, label = &quot;p.adj&quot;, tip.length = 0.01,
  #                    y.position = 800, step.increase = 0, size = 2.25) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_suppl_figS4)), nrow = 1) +
  scale_color_manual(values = colors_stim, labels = stim_label_suppl_figS4, name = &quot;Stimulus&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;) +
  textsize_small
suppl_figS4_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS4_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS4_panelB-1">
Past versions of suppl_figS4_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS4_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-3" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS4 &lt;- plot_grid(suppl_figS4_panelA, suppl_figS4_panelB, labels = PANEL_labels[c(1, 2)], ncol = 1, rel_heights = c(1, 1), label_size = 10, align = &quot;v&quot;)
suppl_figS4</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS4_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS4_combi-1">
Past versions of suppl_figS4_combi-1.png
</button>
</p>
<div id="fig-suppl_figS4_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS4,
  filename = &quot;output/figures/paper/suppl_figS4.pdf&quot;,
  width = 5,
  height = 3.25,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS4,
  filename = &quot;output/figures/paper/suppl_figS4.jpg&quot;,
  width = 5,
  height = 3.25,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S4: Phospho-specific flow cytometry measurements of CD79a
(Y182) shows high concordance with ID-seq results.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS4&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 10880663 581.1   19604257 1047.0  19604257 1047.0
Vcells 27125058 207.0  140318495 1070.6 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s5" class="section level2">
<h2>Suppl. figure S5</h2>
<p><em>DMSO has no effect on signaling activation.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li>DESeq2 analysis of PBS and DMSO controls</li>
</ul></li>
</ul>
<div id="data-2" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
data_ID_suppl_figS5 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS5 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS5 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
# inhib_suppl_figS5 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
conc_suppl_figS5 &lt;- c(&quot;0 uM (PBS)&quot;, &quot;0 uM (DMSO)&quot;)
conc_label_suppl_figS5 &lt;- c(&quot;PBS ctrl&quot;, &quot;DMSO ctrl&quot;)

meta_cols_suppl_figS5 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<p>Differential expression analysis of the effect of DMSO vs PBS on the
stimulation in HBL1</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1: PBS + DMSO ctrls 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: PBS ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- data_ID_suppl_figS5  %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; inhib_conc %in% conc_suppl_figS5 &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS5, all_of(meta_cols_suppl_figS5)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new + stim_clean + inhib_text_new:stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;PBS ctrl&quot;)
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_stim &lt;- results(dds, name = &quot;stim_clean_Activated_vs_Basal&quot;, alpha = 0.05)
results_DMSO &lt;- results(dds, name = &quot;inhib_text_new_DMSO.ctrl_vs_PBS.ctrl&quot;, alpha = 0.05)
results_DMSO_stim &lt;- results(dds, name = &quot;inhib_text_newDMSO.ctrl.stim_cleanActivated&quot;, alpha = 0.05)

list_results &lt;- list(results_stim, results_DMSO, results_DMSO_stim)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_results_volc &lt;- c(&quot;volc_stim&quot;, &quot;volc_DMSO&quot;, &quot;volc_DMSO_stim&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Add column with comparison for faceting
comp_suppl_figS5 &lt;- c(&quot;Activated vs basal&quot;, &quot;DMSO vs PBS&quot;, &quot;Effect DMSO on activation&quot;)
volc_stim$comparison &lt;- comp_suppl_figS5[1]
volc_DMSO$comparison &lt;- comp_suppl_figS5[2]
volc_DMSO_stim$comparison &lt;- comp_suppl_figS5[3]
dataset_suppl_figS5 &lt;- list(volc_stim, volc_DMSO, volc_DMSO_stim) %&gt;%
  purrr::reduce(full_join)</code></pre>
</div>
<div id="panels-4" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-4" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
dataset_suppl_figS5A &lt;- dataset_suppl_figS5 %&gt;%
  subset(comparison == &quot;Activated vs basal&quot;)

# Figure
suppl_figS5_panelA &lt;- ggplot(data = dataset_suppl_figS5A,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.5,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS5A$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS5A$log2FoldChange, 0.4 - dataset_suppl_figS5A$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-0.25, 1), ylim = c(0, 275)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  guides(color = guide_legend(ncol = 1, byrow = TRUE)) +
  textsize_small
suppl_figS5_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS5_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS5_panelA-1">
Past versions of suppl_figS5_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS5_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-2" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
dataset_suppl_figS5B &lt;- dataset_suppl_figS5 %&gt;%
  subset(comparison == &quot;DMSO vs PBS&quot;)

# Figure
suppl_figS5_panelB &lt;- ggplot(data = dataset_suppl_figS5B,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.5,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS5B$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS5B$log2FoldChange, 0.4 - dataset_suppl_figS5B$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-0.25, 1), ylim = c(0, 275)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  # guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS5_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS5_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS5_panelB-1">
Past versions of suppl_figS5_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS5_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data
dataset_suppl_figS5C &lt;- dataset_suppl_figS5 %&gt;%
  subset(comparison == &quot;Effect DMSO on activation&quot;)

# Figure
suppl_figS5_panelC &lt;- ggplot(data = dataset_suppl_figS5C,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.5,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS5C$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS5C$log2FoldChange, 0.4 - dataset_suppl_figS5C$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-0.25, 1), ylim = c(0, 275)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  # guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS5_panelC</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS5_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS5_panelC-1">
Past versions of suppl_figS5_panelC-1.png
</button>
</p>
<div id="fig-suppl_figS5_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-4" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS5 &lt;- plot_grid(suppl_figS5_panelA, suppl_figS5_panelB, suppl_figS5_panelC, labels = PANEL_labels[c(1, 2, 3)], nrow = 1, rel_widths = c(1, 1, 1), label_size = 10, align = &quot;h&quot;)
suppl_figS5</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS5_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS5_combi-1">
Past versions of suppl_figS5_combi-1.png
</button>
</p>
<div id="fig-suppl_figS5_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS5,
  filename = &quot;output/figures/paper/suppl_figS5.pdf&quot;,
  width = 7.3,
  height = 3.25,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS5,
  filename = &quot;output/figures/paper/suppl_figS5.jpg&quot;,
  width = 7.3,
  height = 3.25,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS5&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 10988352 586.9   19604257 1047.0  19604257 1047.0
Vcells 19295994 147.3  112254796  856.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s6" class="section level2">
<h2>Suppl. figure S6</h2>
<p><em>Complete PCA signaling state space of the inhibitor concentration
experiment.</em></p>
<p>Input: - DS108 ID-seq data</p>
<pre><code>-   PCA on all samples</code></pre>
<div id="data-3" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data (counts per well/sample and mean per condition)
data_ID_suppl_figS6 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS6 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS6 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
inhib_suppl_figS6 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

conc_suppl_figS6 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
conc_label__suppl_figS6 &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

meta_cols_suppl_figS6 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-5" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-5" class="section level4">
<h4>Panel A</h4>
<p>PCA of all samples (PCA performed on all samples, all samples
plotted).</p>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- data_ID_suppl_figS6  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS6, all_of(meta_cols_suppl_figS6)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_suppl_figS6), 
         inhibitor = factor(inhibitor, levels = inhib_suppl_figS6), 
         inhib_conc = factor(inhib_conc, levels = conc_suppl_figS6)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
<pre class="r"><code># Data
pca_legend_levels &lt;- c(&quot;Basal\nPBS ctrl&quot;, &quot;Basal\nDMSO ctrl&quot;, 
                       &quot;Basal\n0.1 uM iBTK&quot;, &quot;Basal\n1 uM iBTK&quot;, &quot;Basal\n10 uM iBTK&quot;, &quot;Basal\n100 uM iBTK&quot;, 
                       &quot;Basal\n0.1 uM iSYK&quot;, &quot;Basal\n1 uM iSYK&quot;, &quot;Basal\n10 uM iSYK&quot;, &quot;Basal\n100 uM iSYK&quot;, 
                       &quot;Basal\n0.1 uM iNFkB&quot;, &quot;Basal\n1 uM iNFkB&quot;, &quot;Basal\n10 uM iNFkB&quot;, &quot;Basal\n100 uM iNFkB&quot;, 
                       &quot;Activated\nPBS ctrl&quot;, &quot;Activated\nDMSO ctrl&quot;, 
                       &quot;Activated\n0.1 uM iBTK&quot;, &quot;Activated\n1 uM iBTK&quot;, &quot;Activated\n10 uM iBTK&quot;, &quot;Activated\n100 uM iBTK&quot;, 
                       &quot;Activated\n0.1 uM iSYK&quot;, &quot;Activated\n1 uM iSYK&quot;, &quot;Activated\n10 uM iSYK&quot;, &quot;Activated\n100 uM iSYK&quot;, 
                       &quot;Activated\n0.1 uM iNFkB&quot;, &quot;Activated\n1 uM iNFkB&quot;, &quot;Activated\n10 uM iNFkB&quot;, &quot;Activated\n100 uM iNFkB&quot;)
pca_legend_shapes &lt;- c(16, 16, 
                       16, 16, 16, 16, 
                       16, 16, 16, 16, 
                       16, 16, 16, 16, 
                       17, 17,
                       17, 17, 17, 17, 
                       17, 17, 17, 17,  
                       17, 17, 17, 17)
pca_legend_values &lt;- c(&quot;#d9d9d9&quot;, &quot;#bdbdbd&quot;, 
                       &quot;#cfd9e0&quot;, &quot;#bcc9d4&quot;,  &quot;#a0b3c2&quot;, &quot;#90a1ae&quot;, 
                       &quot;#d9e0cf&quot;, &quot;#c9d4bc&quot;,  &quot;#b3c2a0&quot;, &quot;#a1ae90&quot;, 
                       &quot;#e0cfd9&quot;, &quot;#d4bcc9&quot;,  &quot;#c2a0b3&quot;, &quot;#ae90a1&quot;, 
                       &quot;#969696&quot;, &quot;#737373&quot;, 
                       &quot;#bdd7eb&quot;, &quot;#9dc4e1&quot;,  &quot;#7cb0d7&quot;, &quot;#5c9dce&quot;, 
                       &quot;#d7ebbd&quot;, &quot;#c4e19d&quot;,  &quot;#b0d77c&quot;, &quot;#9dce5c&quot;, 
                       &quot;#ebbdd7&quot;, &quot;#e19dc4&quot;,  &quot;#d77cb0&quot;, &quot;#ce5c9d&quot;)
names(pca_legend_shapes) &lt;- pca_legend_levels
names(pca_legend_values) &lt;- pca_legend_levels

# Figure
suppl_figS6_panelA &lt;- ggplot(pca_data) + 
  geom_point(aes(x = PC1, y = PC3, 
                 shape = stim_inhib, color = stim_inhib), 
             size = 2.5) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_levels,
                     name = &quot;Treatment&quot;) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_levels,
                     name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.65, &quot;cm&quot;), legend.spacing.y = unit(0.5, &quot;cm&quot;), legend.box.spacing = unit(0.2, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;, legend.text = element_text(size = 6.5)) +
  textsize_small
suppl_figS6_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS6_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS6_panelA-1">
Past versions of suppl_figS6_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS6_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-5" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS6_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS6_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS6_combi-1">
Past versions of suppl_figS6_combi-1.png
</button>
</p>
<div id="fig-suppl_figS6_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS6_panelA,
  filename = &quot;output/figures/paper/suppl_figS6.pdf&quot;,
 width = 6.125,
  height = 4.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS6_panelA,
  filename = &quot;output/figures/paper/suppl_figS6.jpg&quot;,
  width = 6.125,
  height = 4.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S6: Complete PCA signaling state space of the inhibitor
concentration experiment.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;suppl_figS6&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 10984433 586.7   19604257 1047.0  19604257 1047.0
Vcells 19288800 147.2   89803837  685.2 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s7" class="section level2">
<h2>Suppl. figure S7</h2>
<p><em>Principal components PC1-3 visualized for DMSO controls (basal
and activated) and 100 uM iBTK treatment.</em></p>
<p>Input: - DS108 ID-seq data</p>
<pre><code>-   PCA on all samples</code></pre>
<div id="data-4" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data (counts per well/sample and mean per condition)
data_ID_suppl_figS7 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS7 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS7 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
inhib_suppl_figS7 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

conc_suppl_figS7 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
conc_label__suppl_figS7 &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
conc_select_suppl_figS7 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)

meta_cols_suppl_figS7 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)

conditions_suppl_figS7 &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM iBTK&quot;)
stim_inhib_suppl_figS7 &lt;- c(&quot;Basal\nDMSO ctrl&quot;, &quot;Basal\n100 uM iBTK&quot;, &quot;Activated\nDMSO ctrl&quot;, &quot;Activated\n100 uM iBTK&quot;)
# select_colors &lt;- c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, &quot;100 uM iSYK&quot; = &quot;#9dce5c&quot;, &quot;100 uM iBTK&quot; = &quot;#5c9dce&quot;, &quot;100 uM iNFkB&quot; = &quot;#ce5c9d&quot;)
colors_suppl_figS7 &lt;- c(&quot;Basal\nDMSO ctrl&quot; = &quot;#bdbdbd&quot;, &quot;Basal\n100 uM iBTK&quot; = &quot;#9dc4e1&quot;, 
                               &quot;Activated\nDMSO ctrl&quot; = &quot;#737373&quot;, &quot;Activated\n100 uM iBTK&quot; = &quot;#5c9dce&quot;)
shape_suppl_figS7 &lt;- c(&quot;Basal\nDMSO ctrl&quot; = 16, &quot;Basal\n100 uM iBTK&quot; = 16, 
                              &quot;Activated\nDMSO ctrl&quot; = 17, &quot;Activated\n100 uM iBTK&quot; = 17)</code></pre>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- data_ID_suppl_figS7  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS7, all_of(meta_cols_suppl_figS7)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_suppl_figS7), 
         inhibitor = factor(inhibitor, levels = inhib_suppl_figS7), 
         inhib_conc = factor(inhib_conc, levels = conc_suppl_figS7)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
</div>
<div id="panels-6" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-6" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Figure
# Selected samples only
suppl_figS7_panelA &lt;- ggplot(subset(pca_data, inhib_text_new %in% conditions_suppl_figS7)) + 
  geom_point(aes(x = PC1, y = PC2, shape = stim_inhib, color = stim_inhib), size = 2.5) +
  scale_color_manual(values = colors_suppl_figS7, breaks = names(colors_suppl_figS7), name = &quot;Treatment&quot;) +
  scale_shape_manual(values = shape_suppl_figS7, breaks = names(shape_suppl_figS7), name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) +
  facet_wrap(vars(factor(experiment, labels = &quot;PC1 vs PC2&quot;))) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.5, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) +
  textsize_small
suppl_figS7_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS7_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS7_panelA-1">
Past versions of suppl_figS7_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS7_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-3" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Figure
# Selected samples only
suppl_figS7_panelB &lt;- ggplot(subset(pca_data, inhib_text_new %in% conditions_suppl_figS7)) + 
  geom_point(aes(x = PC3, y = PC2, shape = stim_inhib, color = stim_inhib), size = 2.5) +
  scale_color_manual(values = colors_suppl_figS7, breaks = names(colors_suppl_figS7), name = &quot;Treatment&quot;) +
  scale_shape_manual(values = shape_suppl_figS7, breaks = names(shape_suppl_figS7), name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) +
  facet_wrap(vars(factor(experiment, labels = &quot;PC3 vs PC2&quot;))) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.65, &quot;cm&quot;), legend.spacing.y = unit(0.75, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) +
  guides(color = guide_legend(by_row = TRUE), shape = guide_legend(by_row = TRUE)) +
  textsize_small
suppl_figS7_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS7_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS7_panelB-1">
Past versions of suppl_figS7_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS7_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-6" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS7 &lt;- plot_grid(suppl_figS7_panelA, suppl_figS7_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1.55), label_size = 10)
suppl_figS7</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS7_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS7_combi-1">
Past versions of suppl_figS7_combi-1.png
</button>
</p>
<div id="fig-suppl_figS7_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS7,
  filename = &quot;output/figures/paper/suppl_figS7.pdf&quot;,
  width = 5,
  height = 2,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS7,
  filename = &quot;output/figures/paper/suppl_figS7.jpg&quot;,
  width = 5,
  height = 2,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S7: Principal components PC1-3 visualized for DMSO
controls (basal and activated) and 100 uM iBTK treatment.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS7&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 10993401 587.2   19604257 1047.0  19604257 1047.0
Vcells 19325477 147.5   71843070  548.2 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s8" class="section level2">
<h2>Suppl. figure S8</h2>
<p><em>Effect of iBTK concentration on the comparison between activated
and basal signaling. Volcano plot visualization of the data represented
in the heatmap of Figure 2A.</em></p>
<p>Input: - DS108 ID-seq data</p>
<pre><code>-   DESeq2 analysis of basal and activated signaling with inhibitors</code></pre>
<div id="data-5" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
data_ID_suppl_figS8 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS8 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS8 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
inhib_suppl_figS8 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
conc_suppl_figS8 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
conc_label_suppl_figS8 &lt;- c(&quot;DMSO ctrl&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

meta_cols_suppl_figS8 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<p>Differential expression analysis of the effect of inhibitor
concentration on the stimulation in HBL1</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1: 1 inhibitor, 5 conc, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- data_ID_suppl_figS8  %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; inhib_conc %in% conc_suppl_figS8 &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS8, all_of(meta_cols_suppl_figS8)))) %&gt;%
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new + stim_clean + inhib_text_new:stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_stim &lt;- results(dds, name = &quot;stim_clean_Activated_vs_Basal&quot;, alpha = 0.05)
results_iBTK0.1 &lt;- results(dds, name = &quot;inhib_text_new0.1.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)
results_iBTK1 &lt;- results(dds, name = &quot;inhib_text_new1.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)
results_iBTK10 &lt;- results(dds, name = &quot;inhib_text_new10.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)
results_iBTK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)

list_results &lt;- list(results_stim, results_iBTK0.1, results_iBTK1, results_iBTK10, results_iBTK100)

# Data transformation
# Regularized log transform
rld &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(as.data.frame(dataset))
}

# Apply function to each dataset
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_results_volc &lt;- c(&quot;volc_stim&quot;, &quot;volc_iBTK0.1&quot;, &quot;volc_iBTK1&quot;, &quot;volc_iBTK10&quot;, &quot;volc_iBTK100&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Add column with comparison for faceting
comp_suppl_figS8 &lt;- c(&quot;Activated vs basal&quot;, 
                      &quot;Effect of 0.1 uM iBTK&quot;, &quot;Effect of 1 uM iBTK&quot;, &quot;Effect of 10 uM iBTK&quot;, &quot;Effect of 100 uM iBTK&quot;)
volc_stim$comparison &lt;- comp_suppl_figS8[1]
volc_iBTK0.1$comparison &lt;- comp_suppl_figS8[2]
volc_iBTK1$comparison &lt;- comp_suppl_figS8[3]
volc_iBTK10$comparison &lt;- comp_suppl_figS8[4]
volc_iBTK100$comparison &lt;- comp_suppl_figS8[5]
dataset_suppl_figS8 &lt;- list(volc_stim, volc_iBTK0.1, volc_iBTK1, volc_iBTK10, volc_iBTK100) %&gt;%
  purrr::reduce(full_join)</code></pre>
</div>
<div id="panels-7" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-7" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
dataset_suppl_figS8A &lt;- dataset_suppl_figS8 %&gt;%
  subset(comparison == &quot;Activated vs basal&quot;)
# comp_suppl_figS8 &lt;- c(&quot;Activated vs basal&quot;, &quot;Effect of 0.1 uM iBTK&quot;, &quot;Effect of 1 uM iBTK&quot;, &quot;Effect of 10 uM iBTK&quot;, &quot;Effect of 100 uM iBTK&quot;)

# Figure
suppl_figS8_panelA &lt;- ggplot(data = dataset_suppl_figS8A,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.75,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS8A$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS8A$log2FoldChange, 0.4 - dataset_suppl_figS8A$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially\nexpressed\nproteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-0.5, 1.5), ylim = c(0, 325)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  # guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS8_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS8_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS8_panelA-1">
Past versions of suppl_figS8_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS8_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-4" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data
dataset_suppl_figS8B &lt;- dataset_suppl_figS8 %&gt;%
  subset(comparison == &quot;Effect of 0.1 uM iBTK&quot;)
# comp_suppl_figS8 &lt;- c(&quot;Activated vs basal&quot;, &quot;Effect of 0.1 uM iBTK&quot;, &quot;Effect of 1 uM iBTK&quot;, &quot;Effect of 10 uM iBTK&quot;, &quot;Effect of 100 uM iBTK&quot;)

# Figure
suppl_figS8_panelB &lt;- ggplot(data = dataset_suppl_figS8B,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.75,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS8B$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS8B$log2FoldChange, 0.4 - dataset_suppl_figS8B$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially expressed proteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-1.5, 0.5), ylim = c(0, 75)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS8_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS8_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS8_panelB-1">
Past versions of suppl_figS8_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS8_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-1" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Select data
dataset_suppl_figS8C &lt;- dataset_suppl_figS8 %&gt;%
  subset(comparison == &quot;Effect of 1 uM iBTK&quot;)
# comp_suppl_figS8 &lt;- c(&quot;Activated vs basal&quot;, &quot;Effect of 0.1 uM iBTK&quot;, &quot;Effect of 1 uM iBTK&quot;, &quot;Effect of 10 uM iBTK&quot;, &quot;Effect of 100 uM iBTK&quot;)

# Figure
suppl_figS8_panelC &lt;- ggplot(data = dataset_suppl_figS8C,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.75,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS8C$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS8C$log2FoldChange, 0.4 - dataset_suppl_figS8C$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially expressed proteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-1.5, 0.5), ylim = c(0, 75)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS8_panelC</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS8_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS8_panelC-1">
Past versions of suppl_figS8_panelC-1.png
</button>
</p>
<div id="fig-suppl_figS8_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code># Select data
dataset_suppl_figS8D &lt;- dataset_suppl_figS8 %&gt;%
  subset(comparison == &quot;Effect of 10 uM iBTK&quot;)
# comp_suppl_figS8 &lt;- c(&quot;Activated vs basal&quot;, &quot;Effect of 0.1 uM iBTK&quot;, &quot;Effect of 1 uM iBTK&quot;, &quot;Effect of 10 uM iBTK&quot;, &quot;Effect of 100 uM iBTK&quot;)

# Figure
suppl_figS8_panelD &lt;- ggplot(data = dataset_suppl_figS8D,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.75,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS8D$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS8D$log2FoldChange, 0.4 - dataset_suppl_figS8D$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially expressed proteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-1.5, 0.5), ylim = c(0, 75)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS8_panelD</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS8_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS8_panelD-1">
Past versions of suppl_figS8_panelD-1.png
</button>
</p>
<div id="fig-suppl_figS8_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-e" class="section level4">
<h4>Panel E</h4>
<pre class="r"><code># Select data
dataset_suppl_figS8E &lt;- dataset_suppl_figS8 %&gt;%
  subset(comparison == &quot;Effect of 100 uM iBTK&quot;)
# comp_suppl_figS8 &lt;- c(&quot;Activated vs basal&quot;, &quot;Effect of 0.1 uM iBTK&quot;, &quot;Effect of 1 uM iBTK&quot;, &quot;Effect of 10 uM iBTK&quot;, &quot;Effect of 100 uM iBTK&quot;)

# Figure
suppl_figS8_panelE &lt;- ggplot(data = dataset_suppl_figS8E,
                      aes(x = log2FoldChange, y = -log10(padj), color = diff_express, label = delabel)) +
  geom_point(size = 0.5) +
  geom_text_repel(size = 1.75,
                  segment.size = 0.2,
                  hjust = 0,
                  direction = &quot;y&quot;,
                  nudge_y = -log10(0.005),
                  nudge_x = ifelse(dataset_suppl_figS8E$log2FoldChange &lt; 0, -0.4 - dataset_suppl_figS8E$log2FoldChange, 0.4 - dataset_suppl_figS8E$log2FoldChange)
  ) +
  facet_wrap(~comparison, nrow = 1) +
  scale_color_manual(values = c(UP = &quot;#A50026&quot;, NO = &quot;black&quot;, DOWN = &quot;#364B9A&quot;),
                     breaks = c(&quot;UP&quot;, &quot;NO&quot;, &quot;DOWN&quot;),
                     labels = c(&quot;Increased&quot;, &quot;No difference&quot;, &quot;Decreased&quot;),
                     name = &quot;Differentially expressed proteins&quot;) + 
  geom_vline(xintercept = c(-log2(1.1), log2(1.1)), color = &quot;#CC6677&quot;) +
  geom_hline(yintercept = -log10(0.05), color = &quot;#CC6677&quot;) +
  labs(x = &quot;log2(fold change)&quot;, y = &quot;-log10(adjusted p-value)&quot;, title = &quot;&quot;) +
  coord_cartesian(xlim = c(-1.5, 0.5), ylim = c(0, 75)) +
  theme_bw() +
  theme(legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) + #legend.position = &quot;none&quot;
  guides(color = guide_legend(nrow = 3, byrow = TRUE)) +
  textsize_small
suppl_figS8_panelE</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS8_panelE-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS8_panelE-1">
Past versions of suppl_figS8_panelE-1.png
</button>
</p>
<div id="fig-suppl_figS8_panelE-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-7" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS8_A &lt;- plot_grid(suppl_figS8_panelA, NULL, labels = PANEL_labels[c(1)], nrow = 1, rel_widths = c(1.75, 1), label_size = 10)
suppl_figS8_BCDE &lt;- plot_grid(suppl_figS8_panelB, suppl_figS8_panelC, suppl_figS8_panelD, suppl_figS8_panelE, labels = PANEL_labels[c(2:5)], ncol = 2, rel_widths = c(1, 1), rel_heights = c(1, 1), label_size = 10)
suppl_figS8 &lt;- plot_grid(suppl_figS8_A, suppl_figS8_BCDE, ncol = 1, rel_heights = c(1, 2))
suppl_figS8</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS8_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS8_combi-1">
Past versions of suppl_figS8_combi-1.png
</button>
</p>
<div id="fig-suppl_figS8_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS8,
  filename = &quot;output/figures/paper/suppl_figS8.pdf&quot;,
  width = 6.125,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS8,
  filename = &quot;output/figures/paper/suppl_figS8.jpg&quot;,
  width = 6.125,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S8: Effect of iBTK concentration on the comparison between
activated and basal signaling. Volcano plot visualization of the data
represented in the heatmap of Figure 2A.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS8&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11012824 588.2   19604257 1047.0  19604257 1047.0
Vcells 19404149 148.1   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s9" class="section level2">
<h2>Suppl. figure S9</h2>
<p><em>Inhibition effect of 100 uM iSYK or 100 uM iNFkB versus DMSO
control in activated cells.</em></p>
<p>Input: - Cytoscape network figures</p>
<div id="panels-8" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-8" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS9_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS9_panelA

# PNG
# 100 uM iSYK (DS108)
png_suppl_figS9_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/HBL1_StimOnly_iSYK100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;100 uM iSYK vs DMSO ctrl in activated cells&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
png_suppl_figS9_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS9_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS9_panelA-1">
Past versions of suppl_figS9_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS9_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># legend
png_suppl_figS9_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  # ggtitle(&quot;\n&quot;) +
  textsize_small
# png_suppl_figS9_legend</code></pre>
</div>
<div id="panel-b-5" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
suppl_figS9_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS9_panelB

# PNG
# 100 uM iNFkB (DS108)
png_suppl_figS9_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/HBL1_StimOnly_iNFkB100uMvsDMSO.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;100 uM iNFkB vs DMSO ctrl in activated cells&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
png_suppl_figS9_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS9_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS9_panelB-1">
Past versions of suppl_figS9_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS9_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-8" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS9_AB &lt;- plot_grid(png_suppl_figS9_panelA, png_suppl_figS9_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1), label_size = 10)
suppl_figS9 &lt;- plot_grid(suppl_figS9_AB, png_suppl_figS9_legend, ncol = 1, rel_heights = c(7, 1), label_size = 10)
suppl_figS9</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS9_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS9_combi-1">
Past versions of suppl_figS9_combi-1.png
</button>
</p>
<div id="fig-suppl_figS9_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS9,
  filename = &quot;output/figures/paper/suppl_figS9.pdf&quot;,
  width = 7.3,
  height = 4.6,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS9,
  filename = &quot;output/figures/paper/suppl_figS9.jpg&quot;,
  width = 7.3,
  height = 4.6,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S9: Inhibition effect of 100 uM iSYK or 100 uM iNFkB
versus DMSO control in activated cells.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS9&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11002062 587.6   19604257 1047.0  19604257 1047.0
Vcells 21506527 164.1   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s10" class="section level2">
<h2>Suppl. figure S10</h2>
<p><em>Effect of iNFkB concentration on the comparison between activated
and basal signaling.</em></p>
<p>Input: - DS108 ID-seq data</p>
<pre><code>-   DESeq2 analysis of basal and activated signaling with inhibitors</code></pre>
<div id="data-6" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
data_ID_suppl_figS10 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS10 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS10 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
inhib_suppl_figS10 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)

conc_suppl_figS10 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)
conc_label_suppl_figS10 &lt;- c(&quot;DMSO ctrl&quot;, &quot;100 uM&quot;)
conc_all_suppl_figS10 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
conc_all_label_suppl_figS10 &lt;- c(&quot;DMSO\nctrl&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

meta_cols_suppl_figS10 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-9" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-9" class="section level4">
<h4>Panel A</h4>
<p>Differential expression analysis of the effect of inhibitor
concentration on the stimulation in HBL1</p>
<p>DESeq2 analysis parameters:</p>
<ul>
<li><p>HBL1: 3 inhibitors, 5 conc, 2 stimuli</p></li>
<li><p>Compensated counts (&gt;= 1)</p></li>
<li><p>Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref:
PBS) + inhib_text_new:stimulus</p></li>
</ul>
<pre class="r"><code># Differential expression analysis
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts &lt;- data_ID_suppl_figS10  %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; inhib_conc %in% conc_all_suppl_figS10 &amp; counts &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts) %&gt;%
  dplyr::filter(!is.na(counts)) %&gt;%
  spread(plate_well, counts) %&gt;%
  replace(is.na(.), 0) %&gt;%
  column_to_rownames(&quot;target_nospace&quot;)

cts &lt;- as.matrix(as.data.frame(cts))

coldata &lt;- data.frame(plate_well = colnames(cts)) %&gt;%
  left_join(data_ID_suppl_figS10[c(1:13, 28:31)]) %&gt;% 
  distinct()

rownames(coldata) &lt;- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign &lt;- ~ inhib_text_new + stim_clean + inhib_text_new:stim_clean

# Then create the DESeq dataset
dds &lt;- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$inhib_text_new &lt;- relevel(dds$inhib_text_new, ref = &quot;DMSO ctrl&quot;)
dds$stim_clean &lt;- relevel(dds$stim_clean, ref = &quot;Basal&quot;)

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

dds &lt;- DESeq(dds, test = &quot;Wald&quot;, fitType = &quot;local&quot;)
DESeq_comparisons &lt;- resultsNames(dds) # lists the coefficients
# DESeq_comparisons

# Results
results_HBL1_stim &lt;- results(dds, name = &quot;stim_clean_Activated_vs_Basal&quot;, alpha = 0.05)
# results_HBL1_iSYK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iSYK.stim_cleanActivated&quot;, alpha = 0.05)
# results_HBL1_iBTK100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iBTK.stim_cleanActivated&quot;, alpha = 0.05)
results_HBL1_iNFkB100 &lt;- results(dds, name = &quot;inhib_text_new100.uM.iNFkB.stim_cleanActivated&quot;, alpha = 0.05)

list_results &lt;- list(results_HBL1_stim, results_HBL1_iNFkB100)

# Data transformation
# Regularized log transform
rld_HBL1 &lt;- rlog(dds, blind = FALSE)

# Data wrangling
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins &lt;- rownames(dataset)
  dataset$target_nospace &lt;- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express &lt;- &quot;NO&quot;
  dataset$diff_express[dataset$log2FoldChange &gt; foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;UP&quot;
  dataset$diff_express[dataset$log2FoldChange &lt; -foldchange_filter &amp; dataset$padj &lt; padj_filter] &lt;- &quot;DOWN&quot;
  
  # Add a column for labeling
  dataset$delabel &lt;- NA
  dataset$delabel[dataset$diff_express != &quot;NO&quot;] &lt;- dataset$proteins[dataset$diff_express != &quot;NO&quot;]
  
  # Add Ab metadata
  # dataset &lt;- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
# Applied thresholds:
# padj &lt; 0.05
# log2FC &gt; log2(1.05)
names_volc &lt;- c(&quot;volc_HBL1_stim&quot;, &quot;volc_HBL1_iNFkB100&quot;)
for(dataset in c(1:length(list_results))){
  assign(names_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = log2(1.1)))
}

# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap &lt;- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data &lt;- as.data.frame(dataset) %&gt;%
    dplyr::filter(padj &lt;= padj_filter &amp; abs(log2FoldChange) &gt;= foldchange_filter) %&gt;%
    arrange(-log2FoldChange) 
  rownames(data) &lt;- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top &lt;- data %&gt;%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
list_results_volc &lt;- list(volc_HBL1_stim, volc_HBL1_iNFkB100)
names_heat &lt;- c(&quot;heat_HBL1_stim&quot;, &quot;heat_HBL1_iNFkB100&quot;)
for(dataset in c(1:length(list_results_volc))){
  assign(names_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = log2(1.1)))
  dataset
}</code></pre>
<pre class="r"><code># Select data
# Combine heatmap datasets (without DESeq2 analysis values)
heat_inhib_HBL1 &lt;- list(heat_HBL1_stim, heat_HBL1_iNFkB100) %&gt;%
  purrr::reduce(full_join) %&gt;%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %&gt;%
  left_join(data_ID_suppl_figS10[, 14:21]) %&gt;%
  dplyr::filter(modification == &quot;phospho&quot;) %&gt;%
  distinct()

figS10A_prot_stim &lt;- sort(unique(c(heat_HBL1_stim$target_nospace)))
figS10A_prot_iBTK &lt;- sort(unique(c(heat_HBL1_iNFkB100$target_nospace)))

# Subset the full dataframe to get metadata for heatmap plotting
info_suppl_figS10 &lt;- data_ID_suppl_figS10 %&gt;%
  dplyr::select(plate_well, stim_clean, inhib_conc, inhibitor, cell_line) %&gt;%
  distinct() %&gt;%
  dplyr::filter(cell_line == &quot;HBL1&quot; &amp; inhibitor == &quot;iNFkB&quot; &amp; inhib_conc %in% conc_all_suppl_figS10 &amp; !(stim_clean == &quot;Basal&quot; &amp; inhib_conc %in% conc_all_suppl_figS10[2:5])) %&gt;%
  mutate(inhibitor = factor(inhibitor, levels = inhib_suppl_figS10)) %&gt;%
  as.data.frame()
rownames(info_suppl_figS10) &lt;- info_suppl_figS10$plate_well
info_suppl_figS10 &lt;- dplyr::select(info_suppl_figS10, -c(plate_well, cell_line))

# Get the rld transformed data for the heatmap and scale per row
data_heat_suppl_figS10 &lt;- t(scale(t(assay(rld_HBL1)[c(heat_inhib_HBL1$target_nospace), rownames(info_suppl_figS10)])))
# sort_hclust &lt;- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows &lt;- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = &quot;Unsorted Dendrogram&quot;, xlab = &quot;&quot;, sub = &quot;&quot;)

# Figure
ann_colors_suppl_figS10 &lt;- list(stim_clean = colors_stim, 
                         inhibitor = colors_inhib,
                         # inhib_new = c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib), 
                         inhib_conc = setNames(colors_conc, unique(info_suppl_figS10$inhib_conc)))

suppl_figS10_panelA &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  data_heat_suppl_figS10,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  treeheight_row = 0,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  annotation_col = info_suppl_figS10,
  annotation_colors = ann_colors_suppl_figS10,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = info_suppl_figS10$inhibitor,
  row_split = subset(heat_inhib_HBL1)$modification,
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 5,
  border_color = NA,
  # legend = FALSE,
  # annotation_legend = FALSE,
  # main = &quot;aIg+H2O2 vs PBS (without inhibitors)&quot;
))
# suppl_figS10_panelA

#################
ann_colors_suppl_figS10 &lt;- list(Stimulus = colors_stim, 
                         Inhibitor = colors_inhib,
                         # inhib_new = c(&quot;DMSO ctrl&quot; = &quot;grey&quot;, colors_inhib), 
                         &quot;Inhibitor conc&quot; = setNames(colors_conc, unique(info_suppl_figS10$inhib_conc)))
heat_legend_suppl_figS10 &lt;- list(title = &quot;Scaled\nexpression&quot;, title_gp = gpar(fontsize = 7), labels_gp = gpar(fontsize = 7), 
                          grid_width = unit(2, &quot;mm&quot;), grid_height = unit(20, &quot;mm&quot;))

col_ann_suppl_figS10 &lt;- HeatmapAnnotation(Inhibitor = info_suppl_figS10$inhibitor, 
                                   &quot;Inhibitor conc&quot; = info_suppl_figS10$inhib_conc, 
                                   Stimulus = info_suppl_figS10$stim_clean, 
                                   simple_anno_size = unit(2.5, &quot;mm&quot;), 
                                   annotation_name_gp = gpar(fontsize = 7),
                                   col = ann_colors_suppl_figS10, 
                                   annotation_legend_param = list(Inhibitor = list(title = &quot;Inhibitor&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  &quot;Inhibitor conc&quot; = list(title = &quot;Inhibitor\nconcentration&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;)), 
                                                                  Stimulus = list(title = &quot;Stimulus&quot;, 
                                                                                   title_gp = gpar(fontsize = 7), 
                                                                                   labels_gp = gpar(fontsize = 7), 
                                                                                   grid_width = unit(2, &quot;mm&quot;), 
                                                                                   grid_height = unit(1, &quot;mm&quot;))), 
                                   show_legend = c(FALSE, TRUE, TRUE))

suppl_figS10_panelA &lt;- ggplotify::as.ggplot(ComplexHeatmap::pheatmap(
  data_heat_suppl_figS10,
  scale = &quot;none&quot;,
  cluster_rows = TRUE,
  # treeheight_row = 10,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_col = fig3A_info_Ly8,
  top_annotation = col_ann_suppl_figS10,
  # annotation_colors = fig3A_ann_colors,
  colorRampPalette(c(&quot;navy&quot;, &quot;white&quot;, &quot;firebrick3&quot;))(50),
  column_split = info_suppl_figS10$inhibitor,
  # row_split = subset(heat_inhib_HBL1)$modification,
  row_gap = unit(1, &quot;mm&quot;), 
  # cellwidth = 10,
  # cellheight = 10,
  fontsize = 6,
  row_title_gp = gpar(fontsize = 7),
  column_title_gp = gpar(fontsize = 7), 
  annotation_names_col = gpar(fontsize = 7),
  border_color = NA,
  heatmap_legend_param = heat_legend_suppl_figS10, 
  # annotation_legend = FALSE,
)) +
  ggtitle(&quot;          Effect of 100 uM iNFkB on activated vs basal signaling&quot;) +
  textsize_small
suppl_figS10_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS10_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS10_panelA-1">
Past versions of suppl_figS10_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS10_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-9" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS10_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS10_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS10_combi-1">
Past versions of suppl_figS10_combi-1.png
</button>
</p>
<div id="fig-suppl_figS10_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS10_panelA,
  filename = &quot;output/figures/paper/suppl_figS10.pdf&quot;,
  width = 5,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS10_panelA,
  filename = &quot;output/figures/paper/suppl_figS10.jpg&quot;,
  width = 5,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure: S10: Effect of iNFkB concentration on the comparison
between activated and basal signaling.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS10&quot;))
rm(list = ls(pattern = &quot;results_&quot;))
rm(list = ls(pattern = &quot;volc_&quot;))
rm(list = ls(pattern = &quot;heat_&quot;))
rm(list = ls(pattern = &quot;list_&quot;))
rm(list = ls(pattern = &quot;names_&quot;))
rm(list = ls(pattern = &quot;titles_&quot;))
rm(list = ls(pattern = &quot;data_&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11194270 597.9   19604257 1047.0  19604257 1047.0
Vcells 19815867 151.2   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s11" class="section level2">
<h2>Suppl. figure S11</h2>
<p><em>Phosphorylation levels of CD79a, IKKa/b, and MAPK p38 upon
activation and inhibition.</em></p>
<p>Input:</p>
<ul>
<li><p>DS108 ID-seq data</p>
<ul>
<li>Counts</li>
</ul></li>
</ul>
<div id="data-7" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data DS108 (counts per well/sample and mean per condition)
data_ID_suppl_figS11 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
mean_ID_suppl_figS11 &lt;- read_csv(&quot;output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_clean = case_when(stimulus == &quot;PBS&quot; ~ &quot;Basal&quot;, 
                                stimulus == &quot;aIg+H2O2&quot; ~ &quot;Activated&quot;),
         inhib_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                               inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == &quot;0 uM (DMSO)&quot; ~ &quot;DMSO ctrl&quot;, 
                                    inhib_conc == &quot;0 uM (PBS)&quot; ~ &quot;PBS ctrl&quot;, 
                                    .default = inhib_conc_text), 
         stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code># fig3_cells &lt;- c(&quot;HBL1&quot;, &quot;OCI-Ly8&quot;)
stim_suppl_figS11 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS11 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)
inhib_suppl_figS11 &lt;- c(&quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;)
conc_all_suppl_figS11 &lt;- c(&quot;0 uM (DMSO)&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
conc_label_suppl_figS11 &lt;- c(&quot;DMSO\nctrl&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)
# label_conc &lt;- c(&quot;0 uM&quot;, &quot;0.1 uM&quot;, &quot;1 uM&quot;, &quot;10 uM&quot;, &quot;100 uM&quot;)

meta_cols_suppl_figS11 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc_uM&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-10" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-10" class="section level4">
<h4>Panel A</h4>
<p>Example protein CD79a</p>
<pre class="r"><code># Data
figS11A_prot &lt;- c(
  &quot;CD79a_Y182&quot; 
  # &quot;IKKa_b_S176_S180&quot;,
  # &quot;JAK1_Y1022_Y1023&quot;
  # &quot;LYN_Y397&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
  # &quot;CDK1_T14_Y15&quot;
  # &quot;PYK2_Y402&quot;
) 
figS11A_prot_label &lt;- c(&quot;CD79a (Y182)&quot;)
figS11A_data &lt;- data_ID_suppl_figS11 %&gt;%
  subset(target_nospace %in% figS11A_prot &amp; inhib_conc %in% conc_all_suppl_figS11 &amp; cell_line == &quot;HBL1&quot;)
figS11A_mean &lt;- mean_ID_suppl_figS11 %&gt;%
  subset(target_nospace %in% figS11A_prot &amp; inhib_conc %in% conc_all_suppl_figS11 &amp; cell_line == &quot;HBL1&quot;) 

# Figure
suppl_figS11_panelA &lt;- ggplot() +
  geom_point(data = figS11A_data, 
             aes(factor(inhib_conc, levels = conc_all_suppl_figS11, labels = conc_label_suppl_figS11), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = figS11A_mean, 
            aes(factor(inhib_conc, levels = conc_all_suppl_figS11, labels = conc_label_suppl_figS11), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_suppl_figS11)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = stim_suppl_figS11, labels = stim_label_suppl_figS11, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS11, labels = stim_label_suppl_figS11, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = figS11A_prot_label) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;bottom&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), axis.text.x = element_text(size = 6)) +
  guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE)) +
  textsize_small
suppl_figS11_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS11_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS11_panelA-1">
Past versions of suppl_figS11_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS11_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-6" class="section level4">
<h4>Panel B</h4>
<p>Example protein IKKa/b</p>
<pre class="r"><code># Data
figS11B_prot &lt;- c(
  # &quot;CD79a_Y182&quot; 
  &quot;IKKa_b_S176_S180&quot;
  # &quot;MAPK_p38_T180_Y182&quot;
) 
figS11B_prot_label &lt;- c(&quot;IKKa/b (S176/S180)&quot;)
figS11B_data &lt;- data_ID_suppl_figS11 %&gt;%
  subset(target_nospace %in% figS11B_prot &amp; inhib_conc %in% conc_all_suppl_figS11 &amp; cell_line == &quot;HBL1&quot;)
figS11B_mean &lt;- mean_ID_suppl_figS11 %&gt;%
  subset(target_nospace %in% figS11B_prot &amp; inhib_conc %in% conc_all_suppl_figS11 &amp; cell_line == &quot;HBL1&quot;) 

# Figure
suppl_figS11_panelB &lt;- ggplot() +
  geom_point(data = figS11B_data, 
             aes(factor(inhib_conc, levels = conc_all_suppl_figS11, labels = conc_label_suppl_figS11), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = figS11B_mean, 
            aes(factor(inhib_conc, levels = conc_all_suppl_figS11, labels = conc_label_suppl_figS11), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_suppl_figS11)), ncol = 1) +
  scale_shape_manual(values = c(17, 16), breaks = stim_suppl_figS11, labels = stim_label_suppl_figS11, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS11, labels = stim_label_suppl_figS11, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = figS11B_prot_label) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), axis.text.x = element_text(size = 6)) +
  # guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE)) +
  textsize_small
suppl_figS11_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS11_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS11_panelB-1">
Past versions of suppl_figS11_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS11_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-2" class="section level4">
<h4>Panel C</h4>
<p>Example protein MAPK p38</p>
<pre class="r"><code># Data
figS11C_prot &lt;- c(
  # &quot;CD79a_Y182&quot; 
  # &quot;IKKa_b_S176_S180&quot;,
  &quot;MAPK_p38_T180_Y182&quot;
) 
figS11C_prot_label &lt;- c(&quot;MAPK p38 (T180/Y182)&quot;)
figS11C_data &lt;- data_ID_suppl_figS11 %&gt;%
  subset(target_nospace %in% figS11C_prot &amp; inhib_conc %in% conc_all_suppl_figS11 &amp; cell_line == &quot;HBL1&quot;)
figS11C_mean &lt;- mean_ID_suppl_figS11 %&gt;%
  subset(target_nospace %in% figS11C_prot &amp; inhib_conc %in% conc_all_suppl_figS11 &amp; cell_line == &quot;HBL1&quot;) 

# Figure
suppl_figS11_panelC &lt;- ggplot() +
  geom_point(data = figS11C_data, 
             aes(factor(inhib_conc, levels = conc_all_suppl_figS11, labels = conc_label_suppl_figS11), 
                 counts_norm, 
                 color = stimulus, 
                 shape = stimulus
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = figS11C_mean, 
            aes(factor(inhib_conc, levels = conc_all_suppl_figS11, labels = conc_label_suppl_figS11), 
                counts_norm,
                group = paste(inhibitor, stimulus),
                color = stimulus
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_suppl_figS11)), ncol = 1) +
  # facet_grid(inhibitor~target_nospace, scales = &quot;free_y&quot;) +
  scale_shape_manual(values = c(17, 16), breaks = stim_suppl_figS11, labels = stim_label_suppl_figS11, name = &quot;Stimulus&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS11, labels = stim_label_suppl_figS11, name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = figS11C_prot_label) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), legend.position = &quot;none&quot;, strip.text.x = element_text(margin = margin(0.05, 0, 0.05, 0, &quot;cm&quot;)), axis.text.x = element_text(size = 6)) +
  # guides(shape = guide_legend(nrow = 2, byrow = TRUE), color = guide_legend(nrow = 2, byrow = TRUE)) +
  textsize_small
suppl_figS11_panelC</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS11_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS11_panelC-1">
Past versions of suppl_figS11_panelC-1.png
</button>
</p>
<div id="fig-suppl_figS11_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-10" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS11 &lt;- plot_grid(suppl_figS11_panelA, suppl_figS11_panelB, suppl_figS11_panelC, labels = PANEL_labels[c(1:3)], nrow = 1, rel_widths = c(1, 1, 1), label_size = 10, align = &quot;h&quot;)
suppl_figS11</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS11_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS11_combi-1">
Past versions of suppl_figS11_combi-1.png
</button>
</p>
<div id="fig-suppl_figS11_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS11,
  filename = &quot;output/figures/paper/suppl_figS11.pdf&quot;,
  width = 6.125,
  height = 3.8,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS11,
  filename = &quot;output/figures/paper/suppl_figS11.jpg&quot;,
  width = 6.125,
  height = 3.8,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S11: Phosphorylation levels of CD79a, IKKa/b, and MAPK p38
upon activation and inhibition.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS11&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11201188 598.3   19604257 1047.0  19604257 1047.0
Vcells 19837941 151.4   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s12" class="section level2">
<h2>Suppl. figure S12</h2>
<p><em>Complete PCA signaling state space of the inhibitor combination
experiment.</em></p>
<p>Input: - DS113 ID-seq data</p>
<pre><code>-   PCA on all samples</code></pre>
<div id="data-8" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data (counts per well/sample and mean per condition)
data_ID_suppl_figS12 &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS12 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS12 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

legend_label_suppl_figS12 &lt;- &quot;iSYK + iBTK + iNFkB&quot;
inhib_group_suppl_figS12 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)
inhib_suppl_figS12 &lt;- c(&quot;DMSO ctrl&quot;, 
                        &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                        &quot;iSYK + iBTK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                        &quot;iSYK + iBTK + iNFkB&quot;)
inhib_label_suppl_figS12 &lt;- c(&quot;DMSO ctrl&quot;, 
                              &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                              &quot;iBTK+ iSYK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                              &quot;iBTK + iSYK + iNFkB&quot;)
conc_single_suppl_figS12 &lt;- c(0, 1, 2.5, 5, 10)
conc_all1_suppl_figS12 &lt;- c(&quot;0 + 0 + 0 uM&quot;, 
                            &quot;0 + 10 + 0 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, 
                            &quot;2.5 + 2.5 + 0 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, 
                            &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, 
                            &quot;1 + 5 + 1 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, 
                            &quot;10 + 10 + 10 uM (nostain)&quot;)
conc_all2_suppl_figS12 &lt;- c(&quot;DMSO ctrl&quot;, 
                            &quot;10 uM iBTK&quot;, &quot;10 uM iSYK&quot;, &quot;10 uM iNFkB&quot;, 
                            &quot;2.5 uM of iSYK + iBTK&quot;, &quot;2.5 uM of iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iNFkB&quot;, 
                            &quot;1 uM of iSYK + iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;10 uM of iSYK + iBTK + iNFkB&quot;, 
                            &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)

meta_cols_suppl_figS12 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-11" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-11" class="section level4">
<h4>Panel A</h4>
<p>PCA of all samples (PCA performed on all samples, all samples
plotted).</p>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- data_ID_suppl_figS12  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS12, all_of(meta_cols_suppl_figS12)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_suppl_figS12), 
         inhibitor = factor(inhibitor, levels = inhib_suppl_figS12), 
         inhib_conc = factor(inhib_conc, levels = conc_all1_suppl_figS12)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
<pre class="r"><code># Data
pca_legend_levels &lt;- c(&quot;Basal\nDMSO ctrl&quot;, &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;, 
                       &quot;Activated\nDMSO ctrl&quot;, 
                       &quot;Activated\n10 uM iBTK&quot;, &quot;Activated\n10 uM iSYK&quot;, &quot;Activated\n10 uM iNFkB&quot;, 
                       &quot;Activated\n2.5 uM of iSYK + iBTK&quot;, &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;, &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;, 
                       &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;, &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;, 
                       &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)
pca_legend_shapes &lt;- c(16, 16, 
                       17, 
                       17, 17, 17, 
                       17, 17, 17, 
                       17, 17, 17, 
                       17, 17, 17)
pca_legend_values &lt;- c(&quot;#bdbdbd&quot;, &quot;#857892&quot;, 
                       &quot;#737373&quot;, 
                       &quot;#5c9dce&quot;, &quot;#9dce5c&quot;, &quot;#ce5c9d&quot;, 
                       &quot;#5cce8d&quot;, &quot;#8d5cce&quot;, &quot;#ce8d5c&quot;, 
                       &quot;#d3bceb&quot;, &quot;#ae82da&quot;, &quot;#8351b6&quot;,  
                       &quot;#729cb9&quot;, &quot;#9bb677&quot;, &quot;#b6779b&quot;)
names(pca_legend_shapes) &lt;- pca_legend_levels
names(pca_legend_values) &lt;- pca_legend_levels
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;Basal\n10 uM of each&quot;,
  &quot;Activated\nDMSO ctrl&quot; = &quot;Activated\nDMSO ctrl&quot;, 
  &quot;Activated\n10 uM iBTK&quot; = &quot;10 uM iBTK&quot;,
  &quot;Activated\n10 uM iSYK&quot; = &quot;10 uM iSYK&quot;,
  &quot;Activated\n10 uM iNFkB&quot; = &quot;10 uM iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5 uM iBTK + iSYK&quot;, 
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5 uM iBTK + iNFkB&quot;, 
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;2.5 uM iSYK + iNFkB&quot;,  
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1 uM of each&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5 uM of each&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10 uM of each&quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5 uM iBTK + \n1 uM iSYK + iNFkB&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;5 uM iSYK + \n1 uM iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;5 uM iNFkB + \n1 uM iSYK + iBTK&quot;)

# Figure
suppl_figS12_panelA &lt;- ggplot(pca_data) + 
  geom_point(aes(x = PC1, y = PC2, 
                 shape = stim_inhib, color = stim_inhib), 
             size = 3) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_levels,
                     labels = pca_legend_labels,
                     name = &quot;Treatment&quot;) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_levels,
                     labels = pca_legend_labels,
                     name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) + 
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.7, &quot;cm&quot;), legend.spacing.y = unit(0.5, &quot;cm&quot;), legend.box.spacing = unit(0.2, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) +
  textsize_small
suppl_figS12_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS12_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS12_panelA-1">
Past versions of suppl_figS12_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS12_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-11" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS12_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS12_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS12_combi-1">
Past versions of suppl_figS12_combi-1.png
</button>
</p>
<div id="fig-suppl_figS12_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS12_panelA,
  filename = &quot;output/figures/paper/suppl_figS12.pdf&quot;,
  width = 6.125,
  height = 4.7,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS12_panelA,
  filename = &quot;output/figures/paper/suppl_figS12.jpg&quot;,
  width = 6.125,
  height = 4.7,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S12: Complete PCA signaling state space of the inhibitor
combination experiment.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS12&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11187180 597.5   19604257 1047.0  19604257 1047.0
Vcells 19800827 151.1   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s13" class="section level2">
<h2>Suppl. figure S13</h2>
<p><em>Proteins contributing &gt;2.5% to a PC in the inhibitor
combination experiment. </em></p>
<p>Input: - DS113 ID-seq data</p>
<pre><code>-   PCA on all samples</code></pre>
<div id="data-9" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data (counts per well/sample and mean per condition)
data_ID_suppl_figS13 &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS13 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS13 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

legend_label_suppl_figS13 &lt;- &quot;iSYK + iBTK + iNFkB&quot;
inhib_group_suppl_figS13 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)
group_select_suppl_figS13 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;)
inhib_suppl_figS13 &lt;- c(&quot;DMSO ctrl&quot;, 
                        &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                        &quot;iSYK + iBTK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                        &quot;iSYK + iBTK + iNFkB&quot;)
inhib_label_suppl_figS13 &lt;- c(&quot;DMSO ctrl&quot;, 
                              &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                              &quot;iBTK+ iSYK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                              &quot;iBTK + iSYK + iNFkB&quot;)
conc_single_suppl_figS13 &lt;- c(0, 1, 2.5, 5, 10)
conc_all1_suppl_figS13 &lt;- c(&quot;0 + 0 + 0 uM&quot;, 
                            &quot;0 + 10 + 0 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, 
                            &quot;2.5 + 2.5 + 0 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, 
                            &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, 
                            &quot;1 + 5 + 1 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, 
                            &quot;10 + 10 + 10 uM (nostain)&quot;)
conc_all2_suppl_figS13 &lt;- c(&quot;DMSO ctrl&quot;, 
                            &quot;10 uM iBTK&quot;, &quot;10 uM iSYK&quot;, &quot;10 uM iNFkB&quot;, 
                            &quot;2.5 uM of iSYK + iBTK&quot;, &quot;2.5 uM of iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iNFkB&quot;, 
                            &quot;1 uM of iSYK + iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;10 uM of iSYK + iBTK + iNFkB&quot;, 
                            &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)

meta_cols_suppl_figS13 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<p>PCA of all samples (PCA performed on all samples, selected samples
plotted).</p>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- data_ID_suppl_figS13  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS13, all_of(meta_cols_suppl_figS13)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_suppl_figS13), 
         inhibitor = factor(inhibitor, levels = inhib_suppl_figS13), 
         inhib_conc = factor(inhib_conc, levels = conc_all1_suppl_figS13)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
</div>
<div id="panels-12" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-12" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Idea:
# Visualise proteins contributing to PCs
# Quality of representation: cos2 - size of dot
# Percentage contribution to one PC: contrib - colour of dot

# Select data
data_contrib_2.5p &lt;- as.data.frame(pca_variab$contrib) %&gt;%
  dplyr::filter(Dim.1 &gt;= 2.5 | Dim.2 &gt;= 2.5 | Dim.3 &gt;= 2.5) %&gt;% # | Dim.4 &gt;= 1 | Dim.5 &gt;= 1
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  mutate_all(function(x){x/100}) %&gt;%
  as.matrix()
data_contrib_prot &lt;- factor(rownames(data_contrib_2.5p))

data_cos2_2.5p &lt;- as.data.frame(pca_variab$cos2) %&gt;%
  rownames_to_column(&quot;proteins&quot;) %&gt;%
  dplyr::filter(proteins %in% data_contrib_prot) %&gt;%
  column_to_rownames(&quot;proteins&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 =  Dim.2, PC3 = Dim.3, PC4 = Dim.4, PC5 = Dim.5) %&gt;%
  dplyr::select(c(PC1, PC2, PC3)) %&gt;%
  as.matrix()

# Tidy data format
data_cos2_tidy &lt;- as.data.frame(data_cos2_2.5p) %&gt;%
  rownames_to_column(&quot;proteins&quot;) %&gt;%
  pivot_longer(2:4, names_to = &quot;PC&quot;, values_to = &quot;cos2&quot;)
data_contrib_tidy &lt;- as.data.frame(data_contrib_2.5p) %&gt;%
  rownames_to_column(&quot;proteins&quot;) %&gt;%
  pivot_longer(2:4, names_to = &quot;PC&quot;, values_to = &quot;contrib&quot;) %&gt;%
  mutate(contrib = contrib * 100)
data_contrib_all &lt;- full_join(data_cos2_tidy, data_contrib_tidy)

# Figure
# ggplot geom_count
suppl_figS13_panelA &lt;- ggplot() +
  geom_point(data = data_contrib_all, aes(x = PC, y = factor(proteins, levels = rev(data_contrib_prot)), size = cos2, fill = contrib), color = &quot;black&quot;, pch = 21) +
  scale_size_area(max_size = 10, breaks = c(0, 0.25, 0.5, 0.75, 1), name = &quot;Quality of\nrepresentation\nin PCs&quot;) +
  scale_fill_gradient(low = &quot;#FEDA8B&quot;, high = &quot;#73001a&quot;, name = &quot;Percentage\ncontribution\nwithin one PC&quot;) + #7f2704 #67000d
  labs(x = &quot;&quot;, y = &quot;&quot;) + 
  theme_bw() +
  # theme(legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.25, &quot;cm&quot;), legend.box.spacing = unit(0.25, &quot;cm&quot;), axis.text = element_text(size = 8), legend.title = element_text(size = 9), legend.text = element_text(size = 8), legend.position = &quot;right&quot;, panel.grid = element_line(color = &quot;#f0f0f0&quot;)) + 
  theme(legend.key.size = unit(0.4, &quot;cm&quot;), legend.spacing.y = unit(0.25, &quot;cm&quot;), legend.box.spacing = unit(0.25, &quot;cm&quot;), legend.position = &quot;right&quot;, panel.grid = element_line(color = &quot;#f0f0f0&quot;)) + 
  textsize_small
suppl_figS13_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS13_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS13_panelA-1">
Past versions of suppl_figS13_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS13_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-12" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS13_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS13_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS13_combi-1">
Past versions of suppl_figS13_combi-1.png
</button>
</p>
<div id="fig-suppl_figS13_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS13_panelA,
  filename = &quot;output/figures/paper/suppl_figS13.pdf&quot;,
  width = 3.5,
  height = 5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS13_panelA,
  filename = &quot;output/figures/paper/suppl_figS13.jpg&quot;,
  width = 3.5,
  height = 5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S13: Proteins contributing &gt;2.5% to a PC in the
inhibitor combination experiment. </em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS13&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11198660 598.1   19604257 1047.0  19604257 1047.0
Vcells 19833025 151.4   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s14" class="section level2">
<h2>Suppl. figure S14</h2>
<p><em>Principal components PC1-3 visualized for DMSO controls (basal
and activated) and 10 uM single inhibitor treatment.</em></p>
<p>Input: - DS113 ID-seq data</p>
<pre><code>-   PCA on all samples</code></pre>
<div id="data-10" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data (counts per well/sample and mean per condition)
data_ID_suppl_figS14 &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS14 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS14 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

legend_label_suppl_figS14 &lt;- &quot;iSYK + iBTK + iNFkB&quot;
inhib_group_suppl_figS14 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)
group_select_suppl_figS14 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;)
inhib_suppl_figS14 &lt;- c(&quot;DMSO ctrl&quot;, 
                        &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                        &quot;iSYK + iBTK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                        &quot;iSYK + iBTK + iNFkB&quot;)
inhib_label_suppl_figS14 &lt;- c(&quot;DMSO ctrl&quot;, 
                              &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                              &quot;iBTK+ iSYK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                              &quot;iBTK + iSYK + iNFkB&quot;)
conc_single_suppl_figS14 &lt;- c(0, 1, 2.5, 5, 10)
conc_all1_suppl_figS14 &lt;- c(&quot;0 + 0 + 0 uM&quot;, 
                            &quot;0 + 10 + 0 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, 
                            &quot;2.5 + 2.5 + 0 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, 
                            &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, 
                            &quot;1 + 5 + 1 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, 
                            &quot;10 + 10 + 10 uM (nostain)&quot;)
conc_all2_suppl_figS14 &lt;- c(&quot;DMSO ctrl&quot;, 
                            &quot;10 uM iBTK&quot;, &quot;10 uM iSYK&quot;, &quot;10 uM iNFkB&quot;, 
                            &quot;2.5 uM of iSYK + iBTK&quot;, &quot;2.5 uM of iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iNFkB&quot;, 
                            &quot;1 uM of iSYK + iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;10 uM of iSYK + iBTK + iNFkB&quot;, 
                            &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)

meta_cols_suppl_figS14 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
<p>PCA of all samples (PCA performed on all samples, selected samples
plotted).</p>
<pre class="r"><code># Prepare data for FactoMineR package
# cts_pca: dataframe with wells as row names, proteins as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts_pca &lt;- data_ID_suppl_figS14  %&gt;%
  dplyr::filter(counts_norm &gt;= 1) %&gt;%
  dplyr::select(target_nospace, plate_well, counts_norm) %&gt;%
  dplyr::filter(!is.na(counts_norm)) %&gt;%
  spread(target_nospace, counts_norm) %&gt;%
  replace(is.na(.), 0) %&gt;% 
  column_to_rownames(&quot;plate_well&quot;)

# cts_pca &lt;- scale(cts_pca)

coldata &lt;- data.frame(plate_well = rownames(cts_pca)) %&gt;%
  left_join(distinct(dplyr::select(data_ID_suppl_figS14, all_of(meta_cols_suppl_figS14)))) %&gt;%
  mutate(stimulus = factor(stimulus, levels = stim_suppl_figS14), 
         inhibitor = factor(inhibitor, levels = inhib_suppl_figS14), 
         inhib_conc = factor(inhib_conc, levels = conc_all1_suppl_figS14)) #If necessary, make factors of some metadata columns

rownames(coldata) &lt;- coldata$plate_well

# Results
pca_results &lt;- PCA(cts_pca, scale.unit = FALSE, ncp = 5, graph = FALSE)

pca_expl_var &lt;- as.data.frame(get_eigenvalue(pca_results)) %&gt;% # Percentage variance explained by each PC
  rownames_to_column(&quot;PC&quot;) %&gt;%
  pull(var = variance.percent, name = PC) %&gt;%
  round(digits = 1)

pca_variab &lt;- get_pca_var(pca_results) # Results focused on variables (proteins)
pca_indiv &lt;- get_pca_ind(pca_results) # Results focused on individuals (wells)

pca_data &lt;- as.data.frame(pca_indiv$coord[, 1:3]) %&gt;%
  rownames_to_column(&quot;plate_well&quot;) %&gt;%
  dplyr::rename(PC1 = Dim.1, PC2 = Dim.2, PC3 = Dim.3) %&gt;% 
  left_join(coldata)</code></pre>
</div>
<div id="panels-13" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-13" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select Data
data_suppl_figS14 &lt;- pca_data %&gt;%
   dplyr::filter(inhib_group %in% group_select_suppl_figS14)

pca_legend_levels &lt;- c(&quot;Basal\nDMSO ctrl&quot;, 
                       &quot;Activated\nDMSO ctrl&quot;, 
                       &quot;Activated\n10 uM iBTK&quot;, &quot;Activated\n10 uM iSYK&quot;, &quot;Activated\n10 uM iNFkB&quot;, 
                       &quot;Activated\n2.5 uM of iSYK + iBTK&quot;, &quot;Activated\n2.5 uM of iBTK + iNFkB&quot;, &quot;Activated\n2.5 uM of iSYK + iNFkB&quot;, 
                       &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot;, &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot;, &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;, 
                       &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)
pca_legend_shapes &lt;- c(16, 
                       17, 
                       17, 17, 17, 
                       17, 17, 17, 
                       17, 17, 17, 16, 
                       17, 17, 17)
pca_legend_values &lt;- c(&quot;#bdbdbd&quot;, 
                       &quot;#737373&quot;, 
                       &quot;#5c9dce&quot;, &quot;#9dce5c&quot;, &quot;#ce5c9d&quot;, 
                       &quot;#5cce8d&quot;, &quot;#8d5cce&quot;, &quot;#ce8d5c&quot;, 
                       &quot;#d3bceb&quot;, &quot;#ae82da&quot;, &quot;#8351b6&quot;, &quot;#857892&quot;, 
                       &quot;#729cb9&quot;, &quot;#9bb677&quot;, &quot;#b6779b&quot;)
names(pca_legend_shapes) &lt;- pca_legend_levels
names(pca_legend_values) &lt;- pca_legend_levels
pca_legend_labels &lt;- c(
  &quot;Basal\nDMSO ctrl&quot; = &quot;Basal\nDMSO ctrl&quot;, 
  &quot;Activated\nDMSO ctrl&quot; = &quot;Activated\nDMSO ctrl&quot;, 
  &quot;Activated\n10 uM iBTK&quot; = &quot;10 uM iBTK&quot;,
  &quot;Activated\n10 uM iSYK&quot; = &quot;10 uM iSYK&quot;,
  &quot;Activated\n10 uM iNFkB&quot; = &quot;10 uM iNFkB&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK&quot; = &quot;2.5 uM iBTK + iSYK&quot;, 
  &quot;Activated\n2.5 uM of iBTK + iNFkB&quot; = &quot;2.5 uM iBTK + iNFkB&quot;, 
  &quot;Activated\n2.5 uM of iSYK + iNFkB&quot; = &quot;2.5 uM iSYK + iNFkB&quot;,  
  &quot;Activated\n1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1 uM of each&quot;,
  &quot;Activated\n2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5 uM of each&quot;,
  &quot;Activated\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10 uM of each&quot;,
  &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot; = &quot;Basal\n10 uM of each&quot;,
  &quot;Activated\n5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5 uM iBTK + \n1 uM iSYK + iNFkB&quot;,
  &quot;Activated\n5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;5 uM iSYK + \n1 uM iBTK + iNFkB&quot;,
  &quot;Activated\n5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;5 uM iNFkB + \n1 uM iSYK + iBTK&quot;)

# Figure
suppl_figS14_panelA &lt;- ggplot(data_suppl_figS14) + 
  geom_point(aes(x = PC1, y = PC3, 
                 shape = stim_inhib, color = stim_inhib), 
             size = 2.5) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_levels,
                     labels = pca_legend_labels,
                     name = &quot;Treatment&quot;) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_levels,
                     labels = pca_legend_labels,
                     name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC1: &quot;, pca_expl_var[1], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  facet_wrap(vars(factor(experiment, labels = &quot;PC1 vs PC3&quot;))) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.6, &quot;cm&quot;), legend.spacing.y = unit(0.5, &quot;cm&quot;), legend.box.spacing = unit(0.2, &quot;cm&quot;), legend.position = &quot;none&quot;, legend.justification = &quot;left&quot;) +
  textsize_small
suppl_figS14_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS14_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS14_panelA-1">
Past versions of suppl_figS14_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS14_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-7" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Figure
suppl_figS14_panelB &lt;- ggplot(data_suppl_figS14) + 
  geom_point(aes(x = PC2, y = PC3, 
                 shape = stim_inhib, color = stim_inhib), 
             size = 2.5) +
  scale_color_manual(values = pca_legend_values,
                     breaks = pca_legend_levels,
                     labels = pca_legend_labels,
                     name = &quot;Treatment&quot;) +
  scale_shape_manual(values = pca_legend_shapes,
                     breaks = pca_legend_levels,
                     labels = pca_legend_labels,
                     name = &quot;Treatment&quot;) +
  xlab(paste0(&quot;PC2: &quot;, pca_expl_var[2], &quot;% variance&quot;)) +
  ylab(paste0(&quot;PC3: &quot;, pca_expl_var[3], &quot;% variance&quot;)) + 
  facet_wrap(vars(factor(experiment, labels = &quot;PC1 vs PC3&quot;))) +
  theme_bw() +
  theme(panel.grid.minor = element_blank(), legend.key.size = unit(0.6, &quot;cm&quot;), legend.spacing.y = unit(0.5, &quot;cm&quot;), legend.box.spacing = unit(0.2, &quot;cm&quot;), legend.position = &quot;right&quot;, legend.justification = &quot;left&quot;) +
  textsize_small
suppl_figS14_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS14_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS14_panelB-1">
Past versions of suppl_figS14_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS14_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-13" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS14 &lt;- plot_grid(suppl_figS14_panelA, suppl_figS14_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1.35), label_size = 10)
suppl_figS14</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS14_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS14_combi-1">
Past versions of suppl_figS14_combi-1.png
</button>
</p>
<div id="fig-suppl_figS14_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS14,
  filename = &quot;output/figures/paper/suppl_figS14.pdf&quot;,
  width = 6.125,
  height = 2.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS14,
  filename = &quot;output/figures/paper/suppl_figS14.jpg&quot;,
  width = 6.125,
  height = 2.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S14: Principal components PC1-3 visualized for DMSO
controls (basal and activated) and 10 uM single inhibitor
treatment.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS14&quot;))
rm(list = ls(pattern = &quot;pca&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11206641 598.5   19604257 1047.0  19604257 1047.0
Vcells 19839089 151.4   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s15" class="section level2">
<h2>Suppl. figure S15</h2>
<p><em>Inhibition effect of 10 uM iSYK or 100 uM iNFkB versus DMSO
control in activated cells (inhibitor combination experiment).</em></p>
<p>Input: - Cytoscape network figures</p>
<div id="panels-14" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-14" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS15_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS15_panelA

# PNG
# 100 uM iSYK (DS108)
png_suppl_figS15_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/combi_iSYKvsDMSOctrl.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;10 uM iSYK vs DMSO ctrl in activated cells&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
png_suppl_figS15_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS15_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS15_panelA-1">
Past versions of suppl_figS15_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS15_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># legend
png_suppl_figS15_legend &lt;- magick::image_read(&quot;output/figures/non_R_figs/legend_network_projections.png&quot;) %&gt;%
  # magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  # ggtitle(&quot;\n&quot;) +
  textsize_small
# png_suppl_figS15_legend</code></pre>
</div>
<div id="panel-b-8" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
suppl_figS15_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS15_panelB

# PNG
# 100 uM iNFkB (DS108)
png_suppl_figS15_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/combi_iNFkBvsDMSOctrl.png&quot;) %&gt;%
  magick::image_border(color = &quot;black&quot;, geometry = &quot;1x1&quot;) %&gt;%
  magick::image_ggplot(interpolate = TRUE) +
  ggtitle(&quot;10 uM iNFkB vs DMSO ctrl in activated cells&quot;) +
  # theme(axis.title = element_text(colour = &quot;white&quot;), axis.text = element_blank()) +
  textsize_small
png_suppl_figS15_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS15_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS15_panelB-1">
Past versions of suppl_figS15_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS15_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-14" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS15_AB &lt;- plot_grid(png_suppl_figS15_panelA, png_suppl_figS15_panelB, labels = PANEL_labels[c(1, 2)], nrow = 1, rel_widths = c(1, 1), label_size = 10)
suppl_figS15 &lt;- plot_grid(suppl_figS15_AB, png_suppl_figS15_legend, ncol = 1, rel_heights = c(7, 1), label_size = 10)
suppl_figS15</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS15_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS15_combi-1">
Past versions of suppl_figS15_combi-1.png
</button>
</p>
<div id="fig-suppl_figS15_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS15,
  filename = &quot;output/figures/paper/suppl_figS15.pdf&quot;,
  width = 7.3,
  height = 4.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS15,
  filename = &quot;output/figures/paper/suppl_figS15.jpg&quot;,
  width = 7.3,
  height = 4.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S15: Inhibition effect of 10 uM iSYK or 100 uM iNFkB
versus DMSO control in activated cells (inhibitor combination
experiment).</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS15&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11212543 598.9   19604257 1047.0  19604257 1047.0
Vcells 21984363 167.8   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s16" class="section level2">
<h2>Suppl. figure S16</h2>
<p><em>Individual protein data exemplifying either strong repression
across all inhibitor combinations that were investigated, or
inhibitor-dependent differences in the strength of repression.</em></p>
<p>Input: - DS113 ID-seq data</p>
<pre><code>-   Count data</code></pre>
<div id="data-11" class="section level3">
<h3>Data</h3>
<pre class="r"><code># ID-seq data (counts per well/sample and mean per condition)
data_ID_suppl_figS16 &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_sample.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)
mean_ID_suppl_figS16 &lt;- read_csv(&quot;output/DS113_InhibCombiIDseq/IDseq_ann/IDseq_data_condition.csv&quot;) %&gt;%
  subset(inhibitor != &quot;iPI3Kd&quot; &amp; target_nospace != &quot;BIM&quot; &amp; target_nospace != &quot;Histone_H2AX_S139&quot; &amp; cell_line == &quot;HBL1&quot;) %&gt;%
  mutate(stim_inhib = paste0(stim_clean, &quot;\n&quot;, inhib_text_new)) #%&gt;%
  # subset(inhib_new != &quot;PBS ctrl&quot;)</code></pre>
<pre class="r"><code>stim_suppl_figS16 &lt;- c(&quot;PBS&quot;, &quot;aIg+H2O2&quot;)
stim_label_suppl_figS16 &lt;- c(&quot;Basal&quot;, &quot;Activated&quot;)

legend_label_suppl_figS16 &lt;- &quot;iSYK + iBTK + iNFkB&quot;
inhib_group_suppl_figS16 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;, &quot;two&quot;, &quot;three_equal&quot;, &quot;three_mix&quot;)
inhib_group_label_suppl_figS16 &lt;- c(&quot;DMSO ctrl&quot;, &quot;Single inhibitor&quot;, &quot;Two inhibitors&quot;, &quot;Three inhibitors equal&quot;, &quot;Three inhibitors mixed&quot;)
group_select_suppl_figS16 &lt;- c(&quot;DMSO ctrl&quot;, &quot;single&quot;)
inhib_suppl_figS16 &lt;- c(&quot;DMSO ctrl&quot;, 
                        &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                        &quot;iSYK + iBTK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                        &quot;iSYK + iBTK + iNFkB&quot;)
inhib_label_suppl_figS16 &lt;- c(&quot;DMSO ctrl&quot;, 
                              &quot;iBTK&quot;, &quot;iSYK&quot;, &quot;iNFkB&quot;, 
                              &quot;iBTK+ iSYK&quot;, &quot;iBTK + iNFkB&quot;, &quot;iSYK +iNFkB&quot;, 
                              &quot;iBTK + iSYK + iNFkB&quot;)
conc_single_suppl_figS16 &lt;- c(0, 1, 2.5, 5, 10)
conc_all1_suppl_figS16 &lt;- c(&quot;0 + 0 + 0 uM&quot;, 
                            &quot;0 + 10 + 0 uM&quot;, &quot;10 + 0 + 0 uM&quot;, &quot;0 + 0 + 10 uM&quot;, 
                            &quot;2.5 + 2.5 + 0 uM&quot;, &quot;0 + 2.5 + 2.5 uM&quot;, &quot;2.5 + 0 + 2.5 uM&quot;, 
                            &quot;1 + 1 + 1 uM&quot;, &quot;2.5 + 2.5 + 2.5 uM&quot;, &quot;10 + 10 + 10 uM&quot;, 
                            &quot;1 + 5 + 1 uM&quot;, &quot;5 + 1 + 1 uM&quot;, &quot;1 + 1 + 5 uM&quot;, 
                            &quot;10 + 10 + 10 uM (nostain)&quot;)
conc_all2_suppl_figS16 &lt;- c(&quot;DMSO ctrl&quot;, 
                            &quot;10 uM iBTK&quot;, &quot;10 uM iSYK&quot;, &quot;10 uM iNFkB&quot;, 
                            &quot;2.5 uM of iSYK + iBTK&quot;, &quot;2.5 uM of iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iNFkB&quot;, 
                            &quot;1 uM of iSYK + iBTK + iNFkB&quot;, &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;, &quot;10 uM of iSYK + iBTK + iNFkB&quot;, 
                            &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot;, &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;, &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot;)
conc_label_suppl_figS16 &lt;- c(
  &quot;DMSO ctrl&quot; = &quot;iBTK             -\niSYK             -\niNFkB            -&quot;, #&quot;iBTK\niSYK\niNFkB&quot;
  &quot;10 uM iBTK&quot; = &quot;10\n-\n-&quot;, 
  &quot;10 uM iSYK&quot; = &quot;-\n10\n-&quot;, 
  &quot;10 uM iNFkB&quot; = &quot;-\n-\n10&quot;,
  &quot;2.5 uM of iSYK + iBTK&quot; = &quot;2.5\n2.5\n-&quot;, 
  &quot;2.5 uM of iBTK + iNFkB&quot; = &quot;2.5\n-\n2.5&quot;, 
  &quot;2.5 uM of iSYK + iNFkB&quot; = &quot;-\n2.5\n2.5&quot;, 
  &quot;1 uM of iSYK + iBTK + iNFkB&quot; = &quot;1\n1\n1&quot;, 
  &quot;2.5 uM of iSYK + iBTK + iNFkB&quot; = &quot;2.5\n2.5\n2.5&quot;, 
  &quot;10 uM of iSYK + iBTK + iNFkB&quot; = &quot;10\n10\n10&quot;, 
  &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot; = &quot;5\n1\n1&quot;, 
  &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot; = &quot;1\n5\n1&quot;, 
  &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; = &quot;1\n1\n5&quot;)

meta_cols_suppl_figS16 &lt;- c(&quot;plate_well&quot;, &quot;plate&quot;, &quot;well&quot;, &quot;experiment&quot;, &quot;cell_line&quot;, &quot;stimulus&quot;, &quot;stim_clean&quot;, &quot;inhibitor&quot;, &quot;inhib_new&quot;, &quot;inhib_conc&quot;, &quot;inhib_conc_new&quot;, &quot;inhib_group&quot;, &quot;iSYK_conc_uM&quot;, &quot;iBTK_conc_uM&quot;, &quot;iNFkB_conc_uM&quot;, &quot;inhib_conc_text&quot;, &quot;inhib_text_new&quot;, &quot;stim_inhib&quot;, &quot;replicate&quot;, &quot;description&quot;, &quot;descript_rep&quot;)</code></pre>
</div>
<div id="panels-15" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-15" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Select data
proteins_suppl_figS16 &lt;- c(&quot;LYN_Y397&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;LYN (Y397)&quot;)

data_select_suppl_figS16A &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16A &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Figure
suppl_figS16_panelA &lt;- ggplot() +
  # geom_jitter(data = subset(fig1F_data, target_nospace == fig1F_proteins[1]), 
  geom_jitter(data = data_select_suppl_figS16A, 
              aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  counts_norm, 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16A,
                aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stimulus, levels = stim_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.75) +
  facet_grid(cols = vars(factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)), 
             # rows = vars(factor(target_nospace, levels = proteins_suppl_figS16)), 
             scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 6)) +
  textsize_small
suppl_figS16_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelA-1">
Past versions of suppl_figS16_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Select data
proteins_suppl_figS16 &lt;- c(&quot;LYN_Y397&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;LYN (Y397)&quot;)

data_select_suppl_figS16A &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16A &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Statistics
# Kruskal-Wallis test
figS16A_kruskal &lt;- subset(data_select_suppl_figS16A, stim_clean == &quot;Activated&quot;) %&gt;%
  # group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ inhib_text_new)
figS16A_kruskal</code></pre>
<pre><code># A tibble: 1 × 6
  .y.             n statistic    df       p method        
* &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;   &lt;dbl&gt; &lt;chr&gt;         
1 counts_norm    39      27.0    12 0.00774 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS16A_dunn &lt;- subset(data_select_suppl_figS16A, stim_clean == &quot;Activated&quot;) %&gt;%
  dunn_test(counts_norm ~ inhib_text_new, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group2 == &quot;DMSO ctrl&quot;) | 
      (group1 == &quot;10 uM iBTK&quot; &amp; (group2 == &quot;10 uM iSYK&quot; | group2 == &quot;10 uM iNFkB&quot;)) | 
      (group1 == &quot;10 uM iNFkB&quot; &amp; group2 == &quot;10 uM iSYK&quot;) |
      (group1 == &quot;2.5 uM of iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK&quot; | group2 == &quot;2.5 uM of iSYK + iNFkB&quot;)) | 
      (group1 == &quot;2.5 uM of iSYK + iBTK&quot; &amp; group2 == &quot;2.5 uM of iSYK + iNFkB&quot;) | 
      (group1 == &quot;1 uM of iSYK + iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot; | group2 == &quot;10 uM of iSYK + iBTK + iNFkB&quot;)) |
      (group1 == &quot;10 uM of iSYK + iBTK + iNFkB&quot; &amp; group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;) | 
      
      (group1 == &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot; &amp; (group2 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; | group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)) | 
      (group1 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; &amp; group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)
      ) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;inhib_text_new&quot;)
  mutate(xmin = c(8,  8, 8, 2, 2, 2, 4, 4, 3, 10, 10, 6, 6, 6, 5, 5, 9, 7, 11, 11, 11, 13, 13, 12),
         xmax = c(10, 9, 1, 4, 3, 1, 3, 1, 1, 9,  1,  5, 7, 1, 7, 1, 1, 1, 13, 12, 1,  12, 1,  1),
         plot_order = c(20, 19, 7, 14, 13, 1, 15, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11), 
         plot_order_alt = c(20, 19, 7, -1, -2, 1, 0, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11)) %&gt;%
  arrange(plot_order) %&gt;%
    mutate(grouping = c(rep(&quot;DMSO ctrl&quot;, times = 12), 
                        rep(&quot;Single inhibitor&quot;, times = 3), 
                        rep(&quot;Two inhibitors&quot;, times = 3), 
                        rep(&quot;Three inhibitors equal&quot;, times = 3), 
                        rep(&quot;Three inhibitors mixed&quot;, times = 3)), 
           grouping_alt = c(rep(&quot;lvl1&quot;, times = 12), 
                         rep(&quot;lvl1&quot;, times = 3), 
                         rep(&quot;lvl2&quot;, times = 3), 
                         rep(&quot;lvl3&quot;, times = 3), 
                         rep(&quot;lvl4&quot;, times = 3))) %&gt;%
  arrange(plot_order_alt)
figS16A_dunn</code></pre>
<pre><code># A tibble: 24 × 15
   .y.     group1 group2    n1    n2 statistic       p  p.adj p.adj.signif  xmin
   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;
 1 counts… 10 uM… 10 uM…     3     3     0.967 0.334   0.572  ns               2
 2 counts… 10 uM… 10 uM…     3     3     0.501 0.616   0.845  ns               2
 3 counts… 10 uM… 10 uM…     3     3     0.465 0.642   0.845  ns               4
 4 counts… 10 uM… DMSO …     3     3     1.29  0.197   0.431  ns               2
 5 counts… 10 uM… DMSO …     3     3     0.322 0.747   0.845  ns               3
 6 counts… 10 uM… DMSO …     3     3     0.788 0.431   0.646  ns               4
 7 counts… 2.5 u… DMSO …     3     3     2.86  0.00418 0.0334 **               5
 8 counts… 2.5 u… DMSO …     3     3     2.65  0.00806 0.0477 **               6
 9 counts… 2.5 u… DMSO …     3     3     1.54  0.124   0.330  ns               7
10 counts… 1 uM … DMSO …     3     3     2.29  0.0219  0.0752 *                8
# ℹ 14 more rows
# ℹ 5 more variables: xmax &lt;dbl&gt;, plot_order &lt;dbl&gt;, plot_order_alt &lt;dbl&gt;,
#   grouping &lt;chr&gt;, grouping_alt &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS16_panelA_stat &lt;- ggplot() +
  geom_jitter(data = data_select_suppl_figS16A, 
              aes(x = interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                  factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                  , sep = &quot;!&quot;), 
                  y = counts_norm, 
                  group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16A,
                aes(interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                , sep = &quot;!&quot;), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.6) +
  stat_pvalue_manual(figS16A_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.025,
                     y.position = 4300, step.increase = 0.1, step.group.by = &quot;grouping_alt&quot;, size = 2.5, vjust = 0.3) + 
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  scale_x_discrete(guide = guide_axis_nested(delim = &quot;!&quot;), name = &quot;&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 7)) +
  textsize_small
suppl_figS16_panelA_stat</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelA_stat-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelA_stat-1">
Past versions of suppl_figS16_panelA_stat-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelA_stat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-9" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code>## Select data
proteins_suppl_figS16 &lt;- c(&quot;BTK_Y223&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;BTK (Y223)&quot;)

data_select_suppl_figS16B &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16B &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Figure
suppl_figS16_panelB &lt;- ggplot() +
  # geom_jitter(data = subset(fig1F_data, target_nospace == fig1F_proteins[1]), 
  geom_jitter(data = data_select_suppl_figS16B, 
              aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  counts_norm, 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16B,
                aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stimulus, levels = stim_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.75) +
  facet_grid(cols = vars(factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)), 
             # rows = vars(factor(target_nospace, levels = proteins_suppl_figS16)), 
             scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 6)) +
  textsize_small
suppl_figS16_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelB-1">
Past versions of suppl_figS16_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Select data
proteins_suppl_figS16 &lt;- c(&quot;BTK_Y223&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;BTK (Y223)&quot;)

data_select_suppl_figS16B &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16B &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Statistics
# Kruskal-Wallis test
figS16B_kruskal &lt;- subset(data_select_suppl_figS16B, stim_clean == &quot;Activated&quot;) %&gt;%
  # group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ inhib_text_new)
figS16B_kruskal</code></pre>
<pre><code># A tibble: 1 × 6
  .y.             n statistic    df       p method        
* &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;   &lt;dbl&gt; &lt;chr&gt;         
1 counts_norm    39      28.5    12 0.00463 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS16B_dunn &lt;- subset(data_select_suppl_figS16B, stim_clean == &quot;Activated&quot;) %&gt;%
  dunn_test(counts_norm ~ inhib_text_new, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group2 == &quot;DMSO ctrl&quot;) | 
      (group1 == &quot;10 uM iBTK&quot; &amp; (group2 == &quot;10 uM iSYK&quot; | group2 == &quot;10 uM iNFkB&quot;)) | 
      (group1 == &quot;10 uM iNFkB&quot; &amp; group2 == &quot;10 uM iSYK&quot;) |
      (group1 == &quot;2.5 uM of iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK&quot; | group2 == &quot;2.5 uM of iSYK + iNFkB&quot;)) | 
      (group1 == &quot;2.5 uM of iSYK + iBTK&quot; &amp; group2 == &quot;2.5 uM of iSYK + iNFkB&quot;) | 
      (group1 == &quot;1 uM of iSYK + iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot; | group2 == &quot;10 uM of iSYK + iBTK + iNFkB&quot;)) |
      (group1 == &quot;10 uM of iSYK + iBTK + iNFkB&quot; &amp; group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;) | 
      
      (group1 == &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot; &amp; (group2 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; | group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)) | 
      (group1 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; &amp; group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)
      ) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;inhib_text_new&quot;)
  mutate(xmin = c(8,  8, 8, 2, 2, 2, 4, 4, 3, 10, 10, 6, 6, 6, 5, 5, 9, 7, 11, 11, 11, 13, 13, 12),
         xmax = c(10, 9, 1, 4, 3, 1, 3, 1, 1, 9,  1,  5, 7, 1, 7, 1, 1, 1, 13, 12, 1,  12, 1,  1),
         plot_order = c(20, 19, 7, 14, 13, 1, 15, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11), 
         plot_order_alt = c(20, 19, 7, -1, -2, 1, 0, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11)) %&gt;%
  arrange(plot_order) %&gt;%
    mutate(grouping = c(rep(&quot;DMSO ctrl&quot;, times = 12), 
                        rep(&quot;Single inhibitor&quot;, times = 3), 
                        rep(&quot;Two inhibitors&quot;, times = 3), 
                        rep(&quot;Three inhibitors equal&quot;, times = 3), 
                        rep(&quot;Three inhibitors mixed&quot;, times = 3)), 
           grouping_alt = c(rep(&quot;lvl1&quot;, times = 12), 
                         rep(&quot;lvl1&quot;, times = 3), 
                         rep(&quot;lvl2&quot;, times = 3), 
                         rep(&quot;lvl3&quot;, times = 3), 
                         rep(&quot;lvl4&quot;, times = 3))) %&gt;%
  arrange(plot_order_alt)
figS16B_dunn</code></pre>
<pre><code># A tibble: 24 × 15
   .y.     group1 group2    n1    n2 statistic       p  p.adj p.adj.signif  xmin
   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;
 1 counts… 10 uM… 10 uM…     3     3     0.394 0.694   0.757  ns               2
 2 counts… 10 uM… 10 uM…     3     3     1.47  0.142   0.260  ns               2
 3 counts… 10 uM… 10 uM…     3     3    -1.07  0.283   0.357  ns               4
 4 counts… 10 uM… DMSO …     3     3     2.36  0.0181  0.0725 *                2
 5 counts… 10 uM… DMSO …     3     3     1.97  0.0489  0.142  *                3
 6 counts… 10 uM… DMSO …     3     3     0.895 0.371   0.445  ns               4
 7 counts… 2.5 u… DMSO …     3     3     1.43  0.152   0.260  ns               5
 8 counts… 2.5 u… DMSO …     3     3     2.65  0.00806 0.0477 **               6
 9 counts… 2.5 u… DMSO …     3     3     1.25  0.210   0.298  ns               7
10 counts… 1 uM … DMSO …     3     3     1.47  0.142   0.260  ns               8
# ℹ 14 more rows
# ℹ 5 more variables: xmax &lt;dbl&gt;, plot_order &lt;dbl&gt;, plot_order_alt &lt;dbl&gt;,
#   grouping &lt;chr&gt;, grouping_alt &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS16_panelB_stat &lt;- ggplot() +
  geom_jitter(data = data_select_suppl_figS16B, 
              aes(x = interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                  factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                  , sep = &quot;!&quot;), 
                  y = counts_norm, 
                  group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16B,
                aes(interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                , sep = &quot;!&quot;), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.6) +
  stat_pvalue_manual(figS16B_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.01,
                     y.position = 2000, step.increase = 0.1, step.group.by = &quot;grouping_alt&quot;, size = 1.25, vjust = 0.4) + 
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  scale_x_discrete(guide = guide_axis_nested(delim = &quot;!&quot;), name = &quot;&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 6)) +
  textsize_small
suppl_figS16_panelB_stat</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelB_stat-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelB_stat-1">
Past versions of suppl_figS16_panelB_stat-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelB_stat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-3" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code>## Select data
proteins_suppl_figS16 &lt;- c(&quot;S6_S240_S244&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;S6 (S240/S244)&quot;)

data_select_suppl_figS16C &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16C &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Figure
suppl_figS16_panelC &lt;- ggplot() +
  # geom_jitter(data = subset(fig1F_data, target_nospace == fig1F_proteins[1]), 
  geom_jitter(data = data_select_suppl_figS16C, 
              aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  counts_norm, 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16C,
                aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stimulus, levels = stim_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.75) +
  facet_grid(cols = vars(factor(inhib_group, levels = inhib_group_suppl_figS16, label = inhib_group_label_suppl_figS16)), 
             # rows = vars(factor(target_nospace, levels = proteins_suppl_figS16)), 
             scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 6)) +
  textsize_small
suppl_figS16_panelC</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelC-1">
Past versions of suppl_figS16_panelC-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## Select data
proteins_suppl_figS16 &lt;- c(&quot;S6_S240_S244&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;S6 (S240/S244)&quot;)

data_select_suppl_figS16C &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16C &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Statistics
# Kruskal-Wallis test
figS16C_kruskal &lt;- subset(data_select_suppl_figS16C, stim_clean == &quot;Activated&quot;) %&gt;%
  # group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ inhib_text_new)
figS16C_kruskal</code></pre>
<pre><code># A tibble: 1 × 6
  .y.             n statistic    df        p method        
* &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;         
1 counts_norm    39      34.5    12 0.000557 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS16C_dunn &lt;- subset(data_select_suppl_figS16C, stim_clean == &quot;Activated&quot;) %&gt;%
  dunn_test(counts_norm ~ inhib_text_new, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group2 == &quot;DMSO ctrl&quot;) | 
      (group1 == &quot;10 uM iBTK&quot; &amp; (group2 == &quot;10 uM iSYK&quot; | group2 == &quot;10 uM iNFkB&quot;)) | 
      (group1 == &quot;10 uM iNFkB&quot; &amp; group2 == &quot;10 uM iSYK&quot;) |
      (group1 == &quot;2.5 uM of iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK&quot; | group2 == &quot;2.5 uM of iSYK + iNFkB&quot;)) | 
      (group1 == &quot;2.5 uM of iSYK + iBTK&quot; &amp; group2 == &quot;2.5 uM of iSYK + iNFkB&quot;) | 
      (group1 == &quot;1 uM of iSYK + iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot; | group2 == &quot;10 uM of iSYK + iBTK + iNFkB&quot;)) |
      (group1 == &quot;10 uM of iSYK + iBTK + iNFkB&quot; &amp; group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;) | 
      
      (group1 == &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot; &amp; (group2 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; | group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)) | 
      (group1 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; &amp; group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)
      ) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;inhib_text_new&quot;)
  mutate(xmin = c(8,  8, 8, 2, 2, 2, 4, 4, 3, 10, 10, 6, 6, 6, 5, 5, 9, 7, 11, 11, 11, 13, 13, 12),
         xmax = c(10, 9, 1, 4, 3, 1, 3, 1, 1, 9,  1,  5, 7, 1, 7, 1, 1, 1, 13, 12, 1,  12, 1,  1),
         plot_order = c(20, 19, 7, 14, 13, 1, 15, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11), 
         plot_order_alt = c(20, 19, 7, -1, -2, 1, 0, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11)) %&gt;%
  arrange(plot_order) %&gt;%
    mutate(grouping = c(rep(&quot;DMSO ctrl&quot;, times = 12), 
                        rep(&quot;Single inhibitor&quot;, times = 3), 
                        rep(&quot;Two inhibitors&quot;, times = 3), 
                        rep(&quot;Three inhibitors equal&quot;, times = 3), 
                        rep(&quot;Three inhibitors mixed&quot;, times = 3)), 
           grouping_alt = c(rep(&quot;lvl1&quot;, times = 12), 
                         rep(&quot;lvl1&quot;, times = 3), 
                         rep(&quot;lvl2&quot;, times = 3), 
                         rep(&quot;lvl3&quot;, times = 3), 
                         rep(&quot;lvl4&quot;, times = 3))) %&gt;%
  arrange(plot_order_alt)
figS16C_dunn</code></pre>
<pre><code># A tibble: 24 × 15
   .y.     group1 group2    n1    n2 statistic       p  p.adj p.adj.signif  xmin
   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;
 1 counts… 10 uM… 10 uM…     3     3    -0.967 0.334   0.445  ns               2
 2 counts… 10 uM… 10 uM…     3     3     1.65  0.0995  0.199  ns               2
 3 counts… 10 uM… 10 uM…     3     3    -2.61  0.00895 0.0430 **               4
 4 counts… 10 uM… DMSO …     3     3     1.43  0.152   0.260  ns               2
 5 counts… 10 uM… DMSO …     3     3     2.40  0.0164  0.0658 *                3
 6 counts… 10 uM… DMSO …     3     3    -0.215 0.830   0.830  ns               4
 7 counts… 2.5 u… DMSO …     3     3     1.54  0.124   0.228  ns               5
 8 counts… 2.5 u… DMSO …     3     3     0.358 0.720   0.752  ns               6
 9 counts… 2.5 u… DMSO …     3     3     2.04  0.0413  0.124  *                7
10 counts… 1 uM … DMSO …     3     3     0.788 0.431   0.544  ns               8
# ℹ 14 more rows
# ℹ 5 more variables: xmax &lt;dbl&gt;, plot_order &lt;dbl&gt;, plot_order_alt &lt;dbl&gt;,
#   grouping &lt;chr&gt;, grouping_alt &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS16_panelC_stat &lt;- ggplot() +
  geom_jitter(data = data_select_suppl_figS16C, 
              aes(x = interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                  factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                  , sep = &quot;!&quot;), 
                  y = counts_norm, 
                  group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16C,
                aes(interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                , sep = &quot;!&quot;), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.6) +
  stat_pvalue_manual(figS16C_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.025,
                     y.position = 1750, step.increase = 0.1, step.group.by = &quot;grouping_alt&quot;, size = 2.5, vjust = 0.3) + 
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  scale_x_discrete(guide = guide_axis_nested(delim = &quot;!&quot;), name = &quot;&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 7)) +
  textsize_small
suppl_figS16_panelC_stat</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelC_stat-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelC_stat-1">
Past versions of suppl_figS16_panelC_stat-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelC_stat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-d-1" class="section level4">
<h4>Panel D</h4>
<pre class="r"><code>## Select data
proteins_suppl_figS16 &lt;- c(&quot;BTK_Y551&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;BTK (Y551)&quot;)

data_select_suppl_figS16D &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16D &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Figure
suppl_figS16_panelD &lt;- ggplot() +
  # geom_jitter(data = subset(fig1F_data, target_nospace == fig1F_proteins[1]), 
  geom_jitter(data = data_select_suppl_figS16D, 
              aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  counts_norm, 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16D,
                aes(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(stimulus, levels = stim_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.75) +
  facet_grid(cols = vars(factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)), 
             # rows = vars(factor(target_nospace, levels = proteins_suppl_figS16)), 
             scales = &quot;free_x&quot;, space = &quot;free_x&quot;) +
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 6)) +
  textsize_small
suppl_figS16_panelD</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelD-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelD-1">
Past versions of suppl_figS16_panelD-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelD-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## Select data
proteins_suppl_figS16 &lt;- c(&quot;BTK_Y551&quot;)
# c(&quot;LYN_Y397&quot;, &quot;BTK_Y223&quot;, &quot;S6_S240_S244&quot;, &quot;BTK_Y551&quot;)
# c(&quot;VAV1_Y174&quot;, &quot;BTK_Y223&quot;, &quot;BTK_Y551&quot;, &quot;CD79a_Y182&quot;, &quot;JAK1_Y1022_Y1023&quot;, &quot;LYN_Y397&quot;, &quot;S6_S235_S236&quot;, &quot;S6_S240_S244&quot;)
proteins_name_suppl_figS16 &lt;- c(&quot;BTK (Y551)&quot;)

data_select_suppl_figS16D &lt;- data_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

mean_select_suppl_figS16D &lt;- mean_ID_suppl_figS16 %&gt;%
  subset(target_nospace %in% proteins_suppl_figS16 &amp; stim_inhib != &quot;Basal\n10 uM of iSYK + iBTK + iNFkB&quot;) %&gt;%
  mutate(target_nospace = factor(target_nospace, levels = proteins_suppl_figS16), 
         inhib_conc_new = factor(inhib_text_new, levels = conc_all2_suppl_figS16))

# Statistics
# Kruskal-Wallis test
figS16D_kruskal &lt;- subset(data_select_suppl_figS16D, stim_clean == &quot;Activated&quot;) %&gt;%
  # group_by(target_nospace) %&gt;%
  kruskal_test(counts_norm ~ inhib_text_new)
figS16D_kruskal</code></pre>
<pre><code># A tibble: 1 × 6
  .y.             n statistic    df        p method        
* &lt;chr&gt;       &lt;int&gt;     &lt;dbl&gt; &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;         
1 counts_norm    39      37.1    12 0.000217 Kruskal-Wallis</code></pre>
<pre class="r"><code># Post-hoc Dunn&#39;s test (without correction)
figS16D_dunn &lt;- subset(data_select_suppl_figS16D, stim_clean == &quot;Activated&quot;) %&gt;%
  dunn_test(counts_norm ~ inhib_text_new, p.adjust.method = &quot;none&quot;) %&gt;%
  dplyr::filter(
    (group2 == &quot;DMSO ctrl&quot;) | 
      (group1 == &quot;10 uM iBTK&quot; &amp; (group2 == &quot;10 uM iSYK&quot; | group2 == &quot;10 uM iNFkB&quot;)) | 
      (group1 == &quot;10 uM iNFkB&quot; &amp; group2 == &quot;10 uM iSYK&quot;) |
      (group1 == &quot;2.5 uM of iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK&quot; | group2 == &quot;2.5 uM of iSYK + iNFkB&quot;)) | 
      (group1 == &quot;2.5 uM of iSYK + iBTK&quot; &amp; group2 == &quot;2.5 uM of iSYK + iNFkB&quot;) | 
      (group1 == &quot;1 uM of iSYK + iBTK + iNFkB&quot; &amp; (group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot; | group2 == &quot;10 uM of iSYK + iBTK + iNFkB&quot;)) |
      (group1 == &quot;10 uM of iSYK + iBTK + iNFkB&quot; &amp; group2 == &quot;2.5 uM of iSYK + iBTK + iNFkB&quot;) | 
      
      (group1 == &quot;5 uM iBTK + 1 uM of iSYK + iNFkB&quot; &amp; (group2 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; | group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)) | 
      (group1 == &quot;5 uM iNFkB + 1 uM of iSYK + iBTK&quot; &amp; group2 == &quot;5 uM iSYK + 1 uM of iBTK + iNFkB&quot;)
      ) %&gt;%
  adjust_pvalue(method = &quot;BH&quot;) %&gt;%
  # add_xy_position(x = &quot;inhib_text_new&quot;)
  mutate(xmin = c(8,  8, 8, 2, 2, 2, 4, 4, 3, 10, 10, 6, 6, 6, 5, 5, 9, 7, 11, 11, 11, 13, 13, 12),
         xmax = c(10, 9, 1, 4, 3, 1, 3, 1, 1, 9,  1,  5, 7, 1, 7, 1, 1, 1, 13, 12, 1,  12, 1,  1),
         plot_order = c(20, 19, 7, 14, 13, 1, 15, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11), 
         plot_order_alt = c(20, 19, 7, -1, -2, 1, 0, 3, 2, 21, 9, 16, 18, 5, 17, 4, 8, 6, 23, 22, 10, 24, 12, 11)) %&gt;%
  arrange(plot_order) %&gt;%
    mutate(grouping = c(rep(&quot;DMSO ctrl&quot;, times = 12), 
                        rep(&quot;Single inhibitor&quot;, times = 3), 
                        rep(&quot;Two inhibitors&quot;, times = 3), 
                        rep(&quot;Three inhibitors equal&quot;, times = 3), 
                        rep(&quot;Three inhibitors mixed&quot;, times = 3)), 
           grouping_alt = c(rep(&quot;lvl1&quot;, times = 12), 
                         rep(&quot;lvl1&quot;, times = 3), 
                         rep(&quot;lvl2&quot;, times = 3), 
                         rep(&quot;lvl3&quot;, times = 3), 
                         rep(&quot;lvl4&quot;, times = 3))) %&gt;%
  arrange(plot_order_alt)
figS16D_dunn</code></pre>
<pre><code># A tibble: 24 × 15
   .y.     group1 group2    n1    n2 statistic       p  p.adj p.adj.signif  xmin
   &lt;chr&gt;   &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        &lt;dbl&gt;
 1 counts… 10 uM… 10 uM…     3     3    -0.788 0.431   0.638  ns               2
 2 counts… 10 uM… 10 uM…     3     3    -0.430 0.667   0.728  ns               2
 3 counts… 10 uM… 10 uM…     3     3    -0.358 0.720   0.747  ns               4
 4 counts… 10 uM… DMSO …     3     3     0.322 0.747   0.747  ns               2
 5 counts… 10 uM… DMSO …     3     3     1.11  0.267   0.452  ns               3
 6 counts… 10 uM… DMSO …     3     3     0.752 0.452   0.638  ns               4
 7 counts… 2.5 u… DMSO …     3     3     1.58  0.115   0.276  ns               5
 8 counts… 2.5 u… DMSO …     3     3     1.07  0.283   0.452  ns               6
 9 counts… 2.5 u… DMSO …     3     3     2.72  0.00650 0.0312 **               7
10 counts… 1 uM … DMSO …     3     3     2.08  0.0378  0.130  *                8
# ℹ 14 more rows
# ℹ 5 more variables: xmax &lt;dbl&gt;, plot_order &lt;dbl&gt;, plot_order_alt &lt;dbl&gt;,
#   grouping &lt;chr&gt;, grouping_alt &lt;chr&gt;</code></pre>
<pre class="r"><code># Figure
suppl_figS16_panelD_stat &lt;- ggplot() +
  geom_jitter(data = data_select_suppl_figS16D, 
              aes(x = interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                  factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                  , sep = &quot;!&quot;), 
                  y = counts_norm, 
                  group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16), 
                  color = factor(stimulus, levels = stim_suppl_figS16)), 
              alpha = 1, size = 1, width = 0.16) +
  geom_errorbar(data = mean_select_suppl_figS16D,
                aes(interaction(factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                                factor(inhib_group, levels = inhib_group_suppl_figS16, labels = inhib_group_label_suppl_figS16)
                                , sep = &quot;!&quot;), 
                    ymin = counts_norm, ymax = counts_norm,
                    group = factor(inhib_text_new, levels = conc_all2_suppl_figS16, labels = conc_label_suppl_figS16),
                    color = factor(stimulus, levels = stim_suppl_figS16)),
                width = 0.5, size = 0.6) +
  stat_pvalue_manual(figS16D_dunn, label = &quot;p.adj.signif&quot;, tip.length = 0.01,
                     y.position = 3500, step.increase = 0.1, step.group.by = &quot;grouping_alt&quot;, size = 1.25, vjust = 0.4) + 
  scale_color_manual(values = colors_stim, breaks = stim_suppl_figS16, labels = stim_label_suppl_figS16, name = &quot;Stimulus&quot;) +
  scale_x_discrete(guide = guide_axis_nested(delim = &quot;!&quot;), name = &quot;&quot;) +
  labs(x = element_blank(), y = &quot;Normalized counts&quot;, title = proteins_name_suppl_figS16) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), 
        legend.key.size = unit(0.5, &quot;cm&quot;), legend.spacing.y = unit(0.2, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;), 
        legend.position = &quot;right&quot;, legend.justification = &quot;right&quot;, 
        axis.text.x = element_text(hjust = 0.9, size = 6)) +
  textsize_small
suppl_figS16_panelD_stat</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_panelD_stat-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_panelD_stat-1">
Past versions of suppl_figS16_panelD_stat-1.png
</button>
</p>
<div id="fig-suppl_figS16_panelD_stat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-15" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS16 &lt;- plot_grid(suppl_figS16_panelA, suppl_figS16_panelB, suppl_figS16_panelC, suppl_figS16_panelD, labels = PANEL_labels[c(1:4)], ncol = 1, rel_heights = c(1, 1, 1, 1), label_size = 10)
suppl_figS16

# Save figure as pdf, jpg, and png
ggsave(
  suppl_figS16,
  filename = &quot;output/figures/paper/suppl_figS16.pdf&quot;,
  width = 7.3,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS16,
  filename = &quot;output/figures/paper/suppl_figS16.jpg&quot;,
  width = 7.3,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<pre class="r"><code># Combine all panels of the figure
# suppl_figS16 &lt;- plot_grid(suppl_figS16_panelA_stat, suppl_figS16_panelB_stat, suppl_figS16_panelC_stat, suppl_figS16_panelD_stat, labels = PANEL_labels[c(1:4)], ncol = 1, rel_heights = c(1, 1, 1, 1), label_size = 10)
suppl_figS16 &lt;- plot_grid(suppl_figS16_panelA_stat, suppl_figS16_panelC_stat, labels = PANEL_labels[c(1:2)], ncol = 1, rel_heights = c(1, 1, 1, 1), label_size = 10)
suppl_figS16</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS16_combi_stat-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS16_combi_stat-1">
Past versions of suppl_figS16_combi_stat-1.png
</button>
</p>
<div id="fig-suppl_figS16_combi_stat-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS16,
  filename = &quot;output/figures//papersuppl_figS16.pdf&quot;,
  width = 7.3,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS16,
  filename = &quot;output/figures//papersuppl_figS16.jpg&quot;,
  width = 7.3,
  height = 8,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S16: Individual protein data exemplifying either strong
repression across all inhibitor combinations that were investigated, or
inhibitor-dependent differences in the strength of repression.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS16&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11231255 599.9   19604257 1047.0  19604257 1047.0
Vcells 19898114 151.9   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s17" class="section level2">
<h2>Suppl. figure S17</h2>
<p><em>Flow cytometry gating strategy.</em></p>
<p>Input: - DS108 phosphoflow data</p>
<pre><code>-   Gating plots</code></pre>
<div id="panels-16" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-16" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS17_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS17_panelA

# PNG
png_suppl_figS17_panelA &lt;- magick::image_read(&quot;output/figures/gating_figs/fig_debris_DS108.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_suppl_figS17_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS17_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS17_panelA-1">
Past versions of suppl_figS17_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS17_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-10" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
suppl_figS17_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS17_panelB

# PNG
png_suppl_figS17_panelB &lt;- magick::image_read(&quot;output/figures/gating_figs/fig_singlets_DS108.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_suppl_figS17_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS17_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS17_panelB-1">
Past versions of suppl_figS17_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS17_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-4" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Placeholder
suppl_figS17_panelC &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS17_panelC

# PNG
png_suppl_figS17_panelC &lt;- magick::image_read(&quot;output/figures/gating_figs/fig_live_DS108.png&quot;) %&gt;%
  image_ggplot(interpolate = TRUE)
png_suppl_figS17_panelC</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS17_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS17_panelC-1">
Past versions of suppl_figS17_panelC-1.png
</button>
</p>
<div id="fig-suppl_figS17_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-16" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS17 &lt;- plot_grid(png_suppl_figS17_panelA, png_suppl_figS17_panelB, png_suppl_figS17_panelC, labels = PANEL_labels[c(1, 2, 3)], ncol = 3, rel_widths = c(1, 1, 1), label_size = 10)
suppl_figS17</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS17_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS17_combi-1">
Past versions of suppl_figS17_combi-1.png
</button>
</p>
<div id="fig-suppl_figS17_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS17,
  filename = &quot;output/figures/paper/suppl_figS17.pdf&quot;,
  width = 7.3,
  height = 2.5,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS17,
  filename = &quot;output/figures/paper/suppl_figS17.jpg&quot;,
  width = 7.3,
  height = 2.5,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S17: Flow cytometry gating strategy.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS17&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11259674 601.4   19604257 1047.0  19604257 1047.0
Vcells 22374275 170.8   57474456  438.5 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s18" class="section level2">
<h2>Suppl. figure S18</h2>
<p><em>Antibody barcode design for the ID-seq protocol.</em></p>
<p>Input: - BioRender schematic figure</p>
<div id="panels-17" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-17" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS18_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  labs(title = &quot;Ab barcode design&quot;) +
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS18_panelA

# PNG
png_suppl_figS18 &lt;- magick::image_read(&quot;output/figures/non_R_figs/Ab_barcode_primers_design.png&quot;) %&gt;%
  magick::image_ggplot()
png_suppl_figS18</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS18_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS18_panelA-1">
Past versions of suppl_figS18_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS18_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-17" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
png_suppl_figS18</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS18_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS18_combi-1">
Past versions of suppl_figS18_combi-1.png
</button>
</p>
<div id="fig-suppl_figS18_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
# ggsave(
#   suppl_figS18,
#   filename = &quot;output/figures/paper/suppl_figS18.pdf&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  suppl_figS18_panelA,
  filename = &quot;output/figures/paper/suppl_figS18_place.jpg&quot;,
  width = 7.3,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S18: Antibody barcode design for the ID-seq
protocol.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS18&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11218316 599.2   19604257 1047.0  19604257 1047.0
Vcells 19862545 151.6   69049347  526.9 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s19" class="section level2">
<h2>Suppl. figure S19</h2>
<p><em>Examples of successful antibody-DNA tag conjugation for ID-seq.
Samples were run on a non-reducing SDS-PAGE gel
electrophoresis.</em></p>
<p>Input: - SDS gel images</p>
<div id="panels-18" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-18" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS19_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;Gel example A&quot;) +
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS19_panelA

# PNG
png_suppl_figS19_panelA &lt;- magick::image_read(&quot;output/figures/non_R_figs/conjugation_example_2.png&quot;) %&gt;%
  magick::image_ggplot()
png_suppl_figS19_panelA</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS19_panelA-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS19_panelA-1">
Past versions of suppl_figS19_panelA-1.png
</button>
</p>
<div id="fig-suppl_figS19_panelA-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-b-11" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Placeholder
suppl_figS19_panelB &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;Gel example A&quot;) +
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS19_panelB

# PNG
png_suppl_figS19_panelB &lt;- magick::image_read(&quot;output/figures/non_R_figs/conjugation_example_1.png&quot;) %&gt;%
  magick::image_ggplot()
png_suppl_figS19_panelB</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS19_panelB-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS19_panelB-1">
Past versions of suppl_figS19_panelB-1.png
</button>
</p>
<div id="fig-suppl_figS19_panelB-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="panel-c-5" class="section level4">
<h4>Panel C</h4>
<pre class="r"><code># Placeholder
suppl_figS19_panelC &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;Gel example A&quot;) +
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS19_panelC

# PNG
png_suppl_figS19_panelC &lt;- magick::image_read(&quot;output/figures/non_R_figs/conjugation_example_3.png&quot;) %&gt;%
  magick::image_ggplot()
png_suppl_figS19_panelC</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS19_panelC-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS19_panelC-1">
Past versions of suppl_figS19_panelC-1.png
</button>
</p>
<div id="fig-suppl_figS19_panelC-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="combined-fig-18" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS19 &lt;- plot_grid(png_suppl_figS19_panelA, png_suppl_figS19_panelB, png_suppl_figS19_panelC, labels = PANEL_labels[c(1, 2, 3)], nrow = 1, rel_widths =  c(1.4, 1.05, 1.01), label_size = 10, label_colour = &quot;white&quot;)
suppl_figS19</code></pre>
<p><img src="figure/z_paper_suppl.Rmd/suppl_figS19_combi-1.png" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-suppl_figS19_combi-1">
Past versions of suppl_figS19_combi-1.png
</button>
</p>
<div id="fig-suppl_figS19_combi-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
4492de7
</td>
<td>
mwitmond
</td>
<td>
2024-10-22
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Save figure as pdf, jpg, and png
ggsave(
  suppl_figS19,
  filename = &quot;output/figures/paper/suppl_figS19.pdf&quot;,
  width = 7.3,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )
ggsave(
  suppl_figS19,
  filename = &quot;output/figures/paper/suppl_figS19.jpg&quot;,
  width = 7.3,
  height = 4,
  units = &quot;in&quot;,
  dpi = 300
  )</code></pre>
<p><em>Figure S19: Examples of successful antibody-DNA tag conjugation
for ID-seq. Samples were run on a non-reducing SDS-PAGE gel
electrophoresis.</em></p>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;figS19&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
<pre><code>           used  (Mb) gc trigger   (Mb)  max used   (Mb)
Ncells 11259766 601.4   19604257 1047.0  19604257 1047.0
Vcells 25651480 195.8   69049347  526.9 175260797 1337.2</code></pre>
</div>
</div>
<div id="suppl.-figure-s0" class="section level2">
<h2>Suppl. figure S0</h2>
<p><em>Figure title.</em></p>
<p>Input: - BioRender schematic figures</p>
<ul>
<li><p>… phosphoflow data</p></li>
<li><p>… ID-seq data</p></li>
</ul>
<div id="data-12" class="section level3">
<h3>Data</h3>
</div>
<div id="panels-19" class="section level3 tabset">
<h3 class="tabset">Panels</h3>
<div id="panel-a-19" class="section level4">
<h4>Panel A</h4>
<pre class="r"><code># Placeholder
suppl_figS0_panelA &lt;- ggplot() +
  geom_blank() +
  scale_x_continuous(limits = c(0, 10)) + 
  scale_y_continuous(limits = c(0, 10)) + 
  # labs(title = &quot;BCR signaling network&quot;) + 
  theme_bw() +
  theme(axis.text = element_text(color = &quot;white&quot;), 
        axis.ticks = element_blank(), 
        panel.grid = element_blank()) +
  textsize_small
# suppl_figS0_panelA

# PNG
png_suppl_figS0 &lt;- image_read(&quot;output/figures/non_R_figs/xxx.png&quot;) %&gt;%
  image_ggplot()
# png_suppl_figS0</code></pre>
</div>
<div id="panel-b-12" class="section level4">
<h4>Panel B</h4>
<pre class="r"><code># Select data

# Figure</code></pre>
</div>
<div id="panel-c-6" class="section level4">
<h4>Panel C</h4>
</div>
</div>
<div id="combined-fig-19" class="section level3">
<h3>Combined fig</h3>
<pre class="r"><code># Combine all panels of the figure
suppl_figS0 &lt;- plot_grid(panelA, panelB, labels = panel_labels[c(1, 2)], ncol = 1, rel_heights = c(1, 1), label_size = 10)
suppl_figS0

# Save figure as pdf, jpg, and png
# ggsave(
#   suppl_figS1,
#   filename = &quot;output/figures/paper/suppl_figS1.pdf&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )
ggsave(
  suppl_figS0,
  filename = &quot;output/figures/paper/suppl_figS0.jpg&quot;,
  width = 6,
  height = 3,
  units = &quot;in&quot;,
  dpi = 300
  )
# ggsave(
#   suppl_figS1,
#   filename = &quot;output/figures/paper/suppl_figS1.png&quot;,
#   width = 6,
#   height = 3,
#   units = &quot;in&quot;,
#   dpi = 300
#   )</code></pre>
<pre class="r"><code># Remove unnecessary files to clear up memory
rm(list = ls(pattern = &quot;suppl_figS1&quot;))
# rm(list = ls(pattern = &quot;_fig1&quot;))
rm(list = ls(pattern = &quot;png_&quot;))
gc()</code></pre>
</div>
</div>
<div id="moved-from-main" class="section level2">
<h2>Moved from main</h2>
<div id="main-figure-2" class="section level3">
<h3>Main figure 2</h3>
<div id="panel-e-1" class="section level4">
<h4>Panel E</h4>
<p>Effect of ID-seq data normalisation on counts per sample</p>
<pre class="r"><code># Select data
fig2E1_data &lt;- fig2_data_id %&gt;%
  unique()%&gt;%
  group_by(plate_well) %&gt;%
  summarize(counts_total = sum(counts), 
            step = &quot;Raw counts&quot;) %&gt;%
  left_join(fig2_data_id[, 1:13]) %&gt;%
  distinct()

fig2E2_data &lt;- fig2_data_id %&gt;%
  unique()%&gt;%
  group_by(plate_well) %&gt;%
  summarize(counts_total = sum(counts_norm), 
            step = &quot;Normalised counts&quot;) %&gt;%
  left_join(fig2_data_id[, 1:13]) %&gt;%
  distinct()

fig2E_data &lt;- full_join(fig2E1_data, fig2E2_data)

# Figure
fig2_panelE &lt;- ggplot(fig2E_data, aes(cell_line, counts_total, color = factor(inhib_conc, levels = fig2_conc), shape = stimulus)) +
  geom_jitter(alpha = 1, size = 0.75, width = 0.1) +
  facet_wrap(vars(factor(step, levels =  c(&quot;Raw counts&quot;, &quot;Normalised counts&quot;)))) +
  scale_y_log10() +
  # scale_color_manual(values = c(&quot;PBS&quot; = &quot;black&quot;, &quot;aIg+H2O2&quot; = &quot;orange&quot;), name = &quot;&quot;) +
  scale_color_manual(values = colors_purple9[c(2, 3, 5, 6, 8, 9)], name = &quot;Inhibitor conc.&quot;) +
  scale_shape_manual(values = c(16, 17), name = &quot;Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc&quot;, y = &quot;Counts (log10 scale)&quot;) + 
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.2, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  guides(fill = guide_legend(byrow = TRUE)) +
  textsize_small
fig2_panelE</code></pre>
</div>
<div id="panel-f" class="section level4">
<h4>Panel F</h4>
<p>Signal of isotype controls</p>
<pre class="r"><code># Select data
fig2F_prot &lt;- c(&quot;Isotype_Mouse_IgG1&quot;, &quot;Isotype_Rabbit_IgG&quot;)
fig2F_data &lt;- fig2_data_id %&gt;%
  subset(target_nospace %in% fig2F_prot &amp; inhib_conc %in% c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;))
fig2F_mean &lt;- fig2_mean_id %&gt;%
  subset(target_nospace %in% fig2F_prot &amp; inhib_conc %in% c(&quot;0 uM (DMSO)&quot;, &quot;100 uM&quot;)) %&gt;%
  dplyr::group_by(cell_line, stimulus, inhib_conc, target) %&gt;%
  summarise(counts_norm = mean(counts_norm, na.rm = T))

# Figure
fig2_panelF &lt;- ggplot() + 
  geom_jitter(data = fig2F_data,
              aes(inhib_conc, 
                  counts_norm, 
                  color = stimulus, 
                  # shape = cell_line
              ), 
              alpha = 1, size = 0.75, width = 0.1) +
  geom_errorbar(data = fig2F_mean,
                aes(inhib_conc,
                    ymin = counts_norm,
                    ymax = counts_norm,
                    group = stimulus,
                    color = stimulus
                ),
                width = 0.5, size = 0.5) + #, color = &quot;black&quot;
  # facet_wrap(vars(factor(target_nospace, levels = prot_test4)), ncol = 4) +
  facet_wrap(vars(factor(target), factor(cell_line, levels = fig2_cells)), nrow = 1) +
  # scale_y_log10() +
  scale_y_continuous(limits = c(900, 2500)) +
  scale_color_manual(values = colors_stim, name = &quot;Stimulus&quot;) + 
  labs(x = &quot;&quot;, y = &quot;Normalized counts&quot;) +
  theme_bw() +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  textsize_small
fig2_panelF</code></pre>
</div>
</div>
<div id="main-figure-3" class="section level3">
<h3>Main figure 3</h3>
<div id="panel-b-13" class="section level4">
<h4>Panel B</h4>
<p>Effect of inhibitor concentration on phospho-protein levels: Clear
activation and dose-response inhibition in both cell lines</p>
<pre class="r"><code># Select data
fig3B_prot &lt;- c(&quot;BTK_Y551&quot;, &quot;JAK1_Y1022_Y1023&quot;)
fig3B_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3B_prot &amp; inhib_conc %in% fig3_conc)
fig3B_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3B_prot &amp; inhib_conc %in% fig3_conc) 
fig3B_prot_name &lt;- unique(fig3B_data$target)

# Figure
# BTK
fig3_panelB1 &lt;- ggplot() +
  geom_point(data = subset(fig3B_data, target_nospace == fig3B_prot[1]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3B_mean, target_nospace == fig3B_prot[1]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = paste0(&quot;Dose-dependent inhibition in both cell lines\n&quot;, fig3B_prot_name[1])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
# fig3_panelB1

# JAK1
fig3_panelB2 &lt;- ggplot() +
  geom_point(data = subset(fig3B_data, target_nospace == fig3B_prot[2]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3B_mean, target_nospace == fig3B_prot[2]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = paste0(&quot;\n&quot;, fig3B_prot_name[2])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  textsize_small
# fig3_panelB2

fig3_panelB &lt;- plot_grid(fig3_panelB1, fig3_panelB2, ncol = 2, rel_widths = c(1, 1.5), align = &quot;h&quot;)
fig3_panelB</code></pre>
</div>
<div id="panel-c-7" class="section level4">
<h4>Panel C</h4>
<p>Effect of inhibitor concentration on phospho-protein levels:
Activation and dose-response inhibition in one cell line</p>
<pre class="r"><code># Select data
fig3C_prot &lt;- c(&quot;CREB_S133&quot;, &quot;MAPK_p38_T180_Y182&quot;)
fig3C_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3C_prot &amp; inhib_conc %in% fig3_conc)
fig3C_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3C_prot &amp; inhib_conc %in% fig3_conc) 
fig3C_prot_name &lt;- unique(fig3C_data$target)

# Figure
# CREB
fig3_panelC1 &lt;- ggplot() +
  geom_point(data = subset(fig3C_data, target_nospace == fig3C_prot[1]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3C_mean, target_nospace == fig3C_prot[1]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  # facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = paste0(&quot;Dose-dependent inhibition in one cell line\n&quot;, fig3C_prot_name[1])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
# fig3_panelC1

# p38
fig3_panelC2 &lt;- ggplot() +
  geom_point(data = subset(fig3C_data, target_nospace == fig3C_prot[2]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3C_mean, target_nospace == fig3C_prot[2]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  # facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title =  paste0(&quot;\n&quot;, fig3C_prot_name[2])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  textsize_small
# fig3_panelC2

fig3_panelC &lt;- plot_grid(fig3_panelC1, fig3_panelC2, ncol = 2, rel_widths = c(1, 1.5), align = &quot;h&quot;)
fig3_panelC</code></pre>
</div>
<div id="panel-d-2" class="section level4">
<h4>Panel D</h4>
<p>Effect of inhibitor concentration on phospho-protein levels:
Inhibition in both basal and activated signaling states</p>
<pre class="r"><code># Select data
fig3D_prot &lt;- c(&quot;ATF2_T71&quot;, &quot;S6_S240_S244&quot;)
fig3D_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3D_prot &amp; inhib_conc %in% fig3_conc)
fig3D_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3D_prot &amp; inhib_conc %in% fig3_conc) 
fig3D_prot_name &lt;- unique(fig3D_data$target)

# Figure
# ATF2
fig3_panelD1 &lt;- ggplot() +
  geom_point(data = subset(fig3D_data, target_nospace == fig3D_prot[1]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3D_mean, target_nospace == fig3D_prot[1]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = paste0(&quot;Inhibition in basal and activated states\n&quot;, fig3D_prot_name[1])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
# fig3_panelD1

# S6
fig3_panelD2 &lt;- ggplot() +
  geom_point(data = subset(fig3D_data, target_nospace == fig3D_prot[2]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3D_mean, target_nospace == fig3D_prot[2]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title =  paste0(&quot;\n&quot;, fig3D_prot_name[2])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  textsize_small
# fig3_panelD2

fig3_panelD &lt;- plot_grid(fig3_panelD1, fig3_panelD2, ncol = 2, rel_widths = c(1, 1.5), align = &quot;h&quot;)
fig3_panelD</code></pre>
</div>
<div id="panel-e-2" class="section level4">
<h4>Panel E</h4>
<p>Effect of inhibitor concentration on phospho-protein levels:
Increased levels upon inhibition</p>
<pre class="r"><code># Select data
fig3E_prot &lt;- c(&quot;AKT_T308&quot;, &quot;cMYC_T58&quot;) # &quot;STAT6_Y641&quot;
fig3E_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3E_prot &amp; inhib_conc %in% fig3_conc)
fig3E_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3E_prot &amp; inhib_conc %in% fig3_conc) 
fig3E_prot_name &lt;- unique(fig3E_data$target)

# Figure
# AKT
fig3_panelE1 &lt;- ggplot() +
  geom_point(data = subset(fig3E_data, target_nospace == fig3E_prot[1]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3E_mean, target_nospace == fig3E_prot[1]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = paste0(&quot;Increased levels upon inhibition\n&quot;, fig3E_prot_name[1])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
# fig3_panelE1

# cMYC
fig3_panelE2 &lt;- ggplot() +
  geom_point(data = subset(fig3E_data, target_nospace == fig3E_prot[2]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3E_mean, target_nospace == fig3E_prot[2]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title =  paste0(&quot;\n&quot;, fig3E_prot_name[2])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  textsize_small
# fig3_panelE2

fig3_panelE &lt;- plot_grid(fig3_panelE1, fig3_panelE2, ncol = 2, rel_widths = c(1, 1.5), align = &quot;h&quot;)
fig3_panelE</code></pre>
</div>
<div id="panel-f-1" class="section level4">
<h4>Panel F</h4>
<p>Effect of inhibitor concentration on phospho-protein levels:
Increased or decreased levels upon inhibition depending on the inhibitor
in question</p>
<pre class="r"><code># Select data
fig3F_prot &lt;- c(&quot;IKKa_b_S176_S180&quot;, &quot;SYK_Y525_Y526&quot;) # &quot;PARP_D214&quot;
fig3F_data &lt;- fig3_data_id %&gt;%
  subset(target_nospace %in% fig3F_prot &amp; inhib_conc %in% fig3_conc)
fig3F_mean &lt;- fig3_mean_id %&gt;%
  subset(target_nospace %in% fig3F_prot &amp; inhib_conc %in% fig3_conc) 
fig3F_prot_name &lt;- unique(fig3F_data$target)

# Figure
# IKKa/b
fig3_panelF1 &lt;- ggplot() +
  geom_point(data = subset(fig3F_data, target_nospace == fig3F_prot[1]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3F_mean, target_nospace == fig3F_prot[1]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title = paste0(&quot;Inhibitor-dependent in- or decreased levels\n&quot;, fig3F_prot_name[1])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  # theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.position = &quot;none&quot;) +
  textsize_small
# fig3_panelF1

# S6
fig3_panelF2 &lt;- ggplot() +
  geom_point(data = subset(fig3F_data, target_nospace == fig3F_prot[2]), 
             aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label), 
                 counts_norm, 
                 color = paste(cell_line, &quot;-&quot;, stimulus), 
                 shape = paste(cell_line, &quot;-&quot;, stimulus)
             ), 
             alpha = 1, size = 0.75) +
  geom_line(data = subset(fig3F_mean, target_nospace == fig3F_prot[2]), 
            aes(factor(inhib_conc, levels = fig3_conc, labels = fig3_conc_label),
                counts_norm,
                group = paste(cell_line, &quot;-&quot;, stimulus),
                color = paste(cell_line, &quot;-&quot;, stimulus)
            ),
            size = 0.5) +
  facet_wrap(vars(factor(inhibitor, levels = fig3_inhib))) +
  # facet_grid(vars(factor(cell_line, levels = fig3_cells)), vars(factor(inhibitor, levels = fig3_inhib))) +
  scale_shape_manual(values = c(16, 16, 17, 17), breaks = fig3_cell_stim, name = &quot;Cells - Stimulus&quot;) +
  scale_color_manual(values = colors_cell_stim, 
                     breaks = fig3_cell_stim, 
                     name = &quot;Cells - Stimulus&quot;) +
  labs(x = &quot;Inhibitor conc.&quot;, y = &quot;Normalized counts&quot;, title =  paste0(&quot;\n&quot;, fig3F_prot_name[2])) +
  theme_bw() +
  # theme(legend.position = &quot;none&quot;, panel.grid.major.x = element_blank()) +
  theme(panel.grid.minor.y = element_blank(), panel.grid.major.x = element_blank(), legend.key.size = unit(0.3, &quot;cm&quot;), legend.spacing.y = unit(0.1, &quot;cm&quot;), legend.box.spacing = unit(0.1, &quot;cm&quot;)) + #legend.position = &quot;none&quot;
  textsize_small
# fig3_panelF2

fig3_panelF &lt;- plot_grid(fig3_panelF1, fig3_panelF2, ncol = 2, rel_widths = c(1, 1.5), align = &quot;h&quot;)
fig3_panelF</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.1 (2024-06-14 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_Netherlands.utf8  LC_CTYPE=English_Netherlands.utf8   
[3] LC_MONETARY=English_Netherlands.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Netherlands.utf8    

time zone: Europe/Amsterdam
tzcode source: internal

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] png_0.1-8                   magick_2.8.3               
 [3] ggridges_0.5.6              forecast_8.23.0            
 [5] zoo_1.8-12                  ggcyto_1.32.0              
 [7] flowWorkspace_4.16.0        ncdfFlow_2.50.0            
 [9] BH_1.84.0-0                 flowCore_2.16.0            
[11] ashr_2.2-63                 apeglm_1.26.1              
[13] MASS_7.3-60.2               ggbreak_0.1.2              
[15] geomtextpath_0.1.4          ggh4x_0.2.8                
[17] cowplot_1.1.3               patchwork_1.2.0            
[19] greekLetters_1.0.2          ggpubr_0.6.0               
[21] ggrepel_0.9.5               scales_1.3.0               
[23] ggthemes_5.1.0              RColorBrewer_1.1-3         
[25] scico_1.5.0                 viridis_0.6.5              
[27] viridisLite_0.4.2           gridExtra_2.3              
[29] mmtable2_0.1.3              gt_0.10.1                  
[31] ggvenn_0.1.10               rgl_1.3.1                  
[33] plotly_4.10.4               ggcorrplot_0.1.4.1         
[35] corrplot_0.92               umap_0.2.10.0              
[37] ggplotify_0.1.2             dendsort_0.3.4             
[39] vsn_3.72.0                  ComplexHeatmap_2.20.0      
[41] platetools_0.1.7            kableExtra_1.4.0           
[43] DESeq2_1.44.0               SummarizedExperiment_1.34.0
[45] Biobase_2.64.0              MatrixGenerics_1.16.0      
[47] matrixStats_1.3.0           GenomicRanges_1.56.1       
[49] GenomeInfoDb_1.40.1         IRanges_2.38.0             
[51] S4Vectors_0.42.0            BiocGenerics_0.50.0        
[53] Seurat_5.1.0                SeuratObject_5.0.2         
[55] sp_2.1-4                    rstatix_0.7.2              
[57] factoextra_1.0.7            FactoMineR_2.11            
[59] readxl_1.4.3                Matrix_1.7-0               
[61] lubridate_1.9.3             forcats_1.0.0              
[63] stringr_1.5.1               dplyr_1.1.4                
[65] purrr_1.0.2                 readr_2.1.5                
[67] tidyr_1.3.1                 tibble_3.2.1               
[69] ggplot2_3.5.1               tidyverse_2.0.0            
[71] workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] vroom_1.6.5             nnet_7.3-19             goftest_1.2-3          
  [4] DT_0.33                 TH.data_1.1-2           vctrs_0.6.5            
  [7] spatstat.random_3.2-3   digest_0.6.36           shape_1.4.6.1          
 [10] git2r_0.33.0            mixsqp_0.3-54           deldir_2.0-4           
 [13] parallelly_1.37.1       SQUAREM_2021.1          reshape2_1.4.4         
 [16] httpuv_1.6.15           foreach_1.5.2           withr_3.0.0            
 [19] xfun_0.45               ggfun_0.1.5             survival_3.6-4         
 [22] memoise_2.0.1           hexbin_1.28.3           RProtoBufLib_2.16.0    
 [25] emmeans_1.10.2          systemfonts_1.1.0       ragg_1.3.2             
 [28] GlobalOptions_0.1.2     quantmod_0.4.26         pbapply_1.7-2          
 [31] promises_1.3.0          scatterplot3d_0.3-44    httr_1.4.7             
 [34] globals_0.16.3          fitdistrplus_1.1-11     ps_1.7.6               
 [37] rstudioapi_0.16.0       UCSC.utils_1.0.0        miniUI_0.1.1.1         
 [40] generics_0.1.3          base64enc_0.1-3         processx_3.8.4         
 [43] curl_5.2.1              zlibbioc_1.50.0         polyclip_1.10-6        
 [46] quadprog_1.5-8          GenomeInfoDbData_1.2.12 SparseArray_1.4.8      
 [49] xtable_1.8-4            doParallel_1.0.17       evaluate_0.24.0        
 [52] S4Arrays_1.4.1          preprocessCore_1.66.0   fracdiff_1.5-3         
 [55] hms_1.1.3               irlba_2.3.5.1           colorspace_2.1-0       
 [58] urca_1.3-4              ROCR_1.0-11             reticulate_1.38.0      
 [61] spatstat.data_3.1-2     Rgraphviz_2.48.0        magrittr_2.0.3         
 [64] lmtest_0.9-40           later_1.3.2             lattice_0.22-6         
 [67] spatstat.geom_3.2-9     future.apply_1.11.2     getPass_0.2-4          
 [70] XML_3.99-0.17           scattermore_1.2         xts_0.14.0             
 [73] RcppAnnoy_0.0.22        pillar_1.9.0            nlme_3.1-164           
 [76] iterators_1.0.14        compiler_4.4.1          RSpectra_0.16-1        
 [79] stringi_1.8.4           tensor_1.5              plyr_1.8.9             
 [82] crayon_1.5.3            abind_1.4-5             truncnorm_1.0-9        
 [85] gridGraphics_0.5-1      emdbook_1.3.13          locfit_1.5-9.10        
 [88] bit_4.0.5               sandwich_3.1-0          whisker_0.4.1          
 [91] codetools_0.2-20        multcomp_1.4-25         textshaping_0.4.0      
 [94] TTR_0.24.4              openssl_2.2.0           bslib_0.7.0            
 [97] GetoptLong_1.0.5        mime_0.12               splines_4.4.1          
[100] circlize_0.4.16         Rcpp_1.0.12             fastDummies_1.7.3      
[103] tseries_0.10-56         cellranger_1.1.0        Rttf2pt1_1.3.12        
[106] leaps_3.2               knitr_1.47              utf8_1.2.4             
[109] clue_0.3-65             fs_1.6.4                listenv_0.9.1          
[112] ggsignif_0.6.4          estimability_1.5.1      callr_3.7.6            
[115] statmod_1.5.0           tzdb_0.4.0              svglite_2.1.3          
[118] pkgconfig_2.0.3         tools_4.4.1             cachem_1.1.0           
[121] numDeriv_2016.8-1.1     fastmap_1.2.0           rmarkdown_2.27         
[124] ica_1.0-3               broom_1.0.6             sass_0.4.9             
[127] coda_0.19-4.1           BiocManager_1.30.23     dotCall64_1.1-1        
[130] graph_1.82.0            carData_3.0-5           RANN_2.6.1             
[133] farver_2.1.2            yaml_2.3.8              cli_3.6.3              
[136] leiden_0.4.3.1          lifecycle_1.0.4         askpass_1.2.0          
[139] uwot_0.2.2              mvtnorm_1.2-5           backports_1.5.0        
[142] cytolib_2.16.0          BiocParallel_1.38.0     timechange_0.3.0       
[145] gtable_0.3.5            rjson_0.2.21            progressr_0.14.0       
[148] parallel_4.4.1          limma_3.60.3            jsonlite_1.8.8         
[151] RcppHNSW_0.6.0          bit64_4.0.5             multcompView_0.1-10    
[154] assertthat_0.2.1        Rtsne_0.17              yulab.utils_0.1.4      
[157] spatstat.utils_3.0-5    bdsmatrix_1.3-7         highr_0.11             
[160] jquerylib_0.1.4         timeDate_4032.109       lazyeval_0.2.2         
[163] shiny_1.8.1.1           htmltools_0.5.8.1       affy_1.82.0            
[166] sctransform_0.4.1       glue_1.7.0              spam_2.10-0            
[169] XVector_0.44.0          rprojroot_2.0.4         flashClust_1.01-2      
[172] invgamma_1.1            igraph_2.0.3            extrafontdb_1.0        
[175] R6_2.5.1                labeling_0.4.3          bbmle_1.0.25.1         
[178] cluster_2.1.6           aplot_0.2.3             DelayedArray_0.30.1    
[181] tidyselect_1.2.1        xml2_1.3.6              car_3.1-2              
[184] future_1.33.2           munsell_0.5.1           KernSmooth_2.23-24     
[187] affyio_1.74.0           data.table_1.15.4       htmlwidgets_1.6.4      
[190] rlang_1.1.4             spatstat.sparse_3.1-0   extrafont_0.19         
[193] spatstat.explore_3.2-7  fansi_1.0.6            </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = false;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
