---
title: "DS108 ID-seq analysis: Comparisons (clean)"
author: "mwitmond"
date: "2024-02-01"
output:
  workflowr::wflow_html:
    toc: TRUE
    toc_float: 
      collapsed: FALSE
    toc_depth: 3
    code_folding: hide
editor_options:
  chunk_output_type: console
---

## Set-up

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  message = F, warning = F, echo = T, eval = T
)

# Load required packages
source("code/packages_seq.R", local = knitr::knit_global())
```

```{r fig_settings}
row_order = c("A", "B", "C", "D", "E", "F", "G", "H")
col_order = c("1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12")
panel_labels <- c("a", "b", "c","d", "e", "f", "g", "h", "i", "j", "k", "l", "m")

textsize <- theme(axis.text.x = element_text(colour = "grey", size = 11), #, face = "bold"
                  axis.text.y = element_text(colour = "grey", size = 11),
                  axis.title = element_text(colour = "black", size = 12), 
                  legend.title = element_text(colour = "black", size = 12),
                  # legend.title = element_blank(), 
                  legend.text = element_text(colour = "grey", size = 11), 
                  strip.text.x = element_text(colour = "black", size = 12)
)

# textsize_small <- theme(text = element_text(size = 7, family = "sans", colour = "black"),
#                         plot.title = element_text(size = 8)
# )
textsize_small <- theme(axis.text.x = element_text(colour = "black", size = 9),
                        axis.text.y = element_text(colour = "black", size = 9),
                        axis.title = element_text(colour = "black", size = 10),
                        legend.text = element_text(colour = "black", size = 9),
                        title = element_text(color = "black", size = 10),
                        strip.text.x = element_text(colour = "black", size = 10, face = "bold"),
                        strip.text.y = element_text(colour = "black", size = 10, face = "bold")
)

colors_dark9 <- c("#4daf4a", "#984ea3", "#377eb8", "#ff7f00", "#f781bf", "#ffff33", "#e41a1c", "#a65628", "#999999")
colors_light12 <- c("#8dd3c7","#ffffb3","#bebada","#fb8072","#80b1d3","#fdb462","#b3de69","#fccde5","#d9d9d9","#bc80bd","#ccebc5","#ffed6f")
colors_paired10 <- c("#a6cee3", "#1f78b4", "#b2df8a", "#33a02c", "#fdbf6f", "#ff7f00", "#cab2d6", "#6a3d9a", "#fb9a99", "#e31a1c")

colors_blue9 <- c("#f7fbff", "#deebf7", "#c6dbef", "#9ecae1", "#6baed6", "#4292c6", "#2171b5", "#08519c", "#08306b")
colors_green9 <- c("#f7fcf5", "#e5f5e0", "#c7e9c0", "#a1d99b", "#74c476", "#41ab5d", "#238b45", "#006d2c", "#00441b")
colors_purple9 <- c("#fcfbfd", "#efedf5", "#dadaeb", "#bcbddc", "#9e9ac8", "#807dba", "#6a51a3", "#54278f", "#3f007d")
colors_red9 <- c("#fff5f0", "#fee0d2", "#fcbba1", "#fc9272", "#fb6a4a", "#ef3b2c", "#cb181d", "#a50f15", "#67000d")
colors_orange9 <- c("#fff5eb", "#fee6ce", "#fdd0a2", "#fdae6b", "#fd8d3c", "#f16913", "#d94801", "#a63603", "#7f2704")
colors_grey9 <- c("#ffffff", "#f0f0f0", "#d9d9d9", "#bdbdbd", "#969696", "#737373", "#525252", "#252525", "#000000")

colors_yb9 <- c("#ffffd9", "#edf8b1", "#c7e9b4", "#7fcdbb", "#41b6c4", "#1d91c0", "#225ea8", "#253494", "#081d58")
colors_pr9 <- c("#f7f4f9", "#e7e1ef", "#d4b9da", "#c994c7", "#df65b0", "#e7298a", "#ce1256", "#980043", "#67001f")

colors_proteins <- c("pCD79a (Y182)" = "#892BE1", "pSYK (Y525/Y526)" = "#1E8FFF", "pPLCy2 (Y759)" = "#20B1A9")
colors_inhib <- c("iSYK" = "#41ab5d", "iBTK" = "#4292c6", "iPI3Kd" = "#807dba", "iNFkB" = "#f16913")
colors_stim <- c("PBS" = "#525252", "aIg+H2O2" = "#fdae6b")
colors_cell <- c("HBL1" = "#df65b0", "OCI-Ly8" = "#41b6c4")
```

```{r fig_labels}
plates_DS108 <- c("A", "B", "C", "D")
cells_DS108 <- c("HBL1", "OCI-Ly8")
stim_DS108 <- c("PBS", "aIg+H2O2")
inhib_DS108 <- c("iSYK", "iBTK", "iPI3Kd", "iNFkB")
inhib_conc_DS108 <- c("0 uM (nostain)", "0 uM (PBS)", "0 uM (DMSO)", "0.1 uM", "1 uM", "10 uM", "100 uM")
inhib_conc_label <- c("0 uM\n(nostain)", "0 uM\n(PBS)", "0 uM\n(DMSO)", "0.1 uM", "1 uM", "10 uM", "100 uM")
inhib_full_DS108 <- c("0 uM iSYK (nostain)", "0 uM iSYK (PBS)", "0 uM iSYK (DMSO)", "0.1 uM iSYK", "1 uM iSYK", "10 uM iSYK", "100 uM iSYK", 
                      "0 uM iBTK (nostain)", "0 uM iBTK (PBS)", "0 uM iBTK (DMSO)", "0.1 uM iBTK", "1 uM iBTK", "10 uM iBTK", "100 uM iBTK", 
                      "0 uM iPI3Kd (nostain)", "0 uM iPI3Kd (PBS)", "0 uM iPI3Kd (DMSO)", "0.1 uM iPI3Kd", "1 uM iPI3Kd", "10 uM iPI3Kd", "100 uM iPI3Kd", 
                      "0 uM iNFkB (nostain)", "0 uM iNFkB (PBS)", "0 uM iNFkB (DMSO)", "0.1 uM iNFkB", "1 uM iNFkB", "10 uM iNFkB", "100 uM iNFkB")
```

## Data & QC

#### Raw data

```{r data}
# Load sparse matrix format all data
counts_PlateA <- read_count_output("data/DS108_StimInhibIDseq/counts/PlateA/featurecounts", name = "featurecounts")
counts_PlateB <- read_count_output("data/DS108_StimInhibIDseq/counts/PlateB/featurecounts", name = "featurecounts")
counts_PlateC <- read_count_output("data/DS108_StimInhibIDseq/counts/PlateC/featurecounts", name = "featurecounts")
counts_PlateD <- read_count_output("data/DS108_StimInhibIDseq/counts/PlateD/featurecounts", name = "featurecounts")

meta_wellID_plateA <- read_csv("data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateA_HBL1pbs.csv")
meta_wellID_plateB <- read_csv("data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateB_HBL1stim.csv")
meta_wellID_plateC <- read_csv("data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateC_Ly8pbs.csv")
meta_wellID_plateD <- read_csv("data/DS108_StimInhibIDseq/config/DynSign.108_wellBC_metadata_seq_plateD_Ly8stim.csv")
meta_wellID <- list(meta_wellID_plateA, meta_wellID_plateB, meta_wellID_plateC, meta_wellID_plateD) %>% purrr::reduce(full_join) %>%
  mutate(plate_well = paste0(plate, "_", well))

meta_Abs <- read_excel("data/DS108_StimInhibIDseq/config/BulkIDseq_Ab_info_seq_analysis_202212.xlsx")

# Reshape counts in data table format
# Plate A
counts_plateA <- data.frame(counts_PlateA) %>% 
  mutate(barcode_name = rownames(counts_PlateA)) %>% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %>%
  gather("well_BC_seq", "counts", 2:c(ncol(counts_PlateA)+1)) %>%
  filter(counts >= 1) %>% # Remove undetected proteins counts
  left_join(meta_Abs) %>%
  unique() %>%
  # filter(!is.na(target_nospace)) %>% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateA, plate == "A")) %>% # Add sample metadata
  filter(!is.na(experiment)) %>%
  arrange(plate, well)

# Plate B
counts_plateB <- data.frame(counts_PlateB) %>% 
  mutate(barcode_name = rownames(counts_PlateB)) %>% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %>%
  gather("well_BC_seq", "counts", 2:c(ncol(counts_PlateB)+1)) %>%
  filter(counts >= 1) %>% # Remove undetected proteins counts
  left_join(meta_Abs) %>%
  unique() %>%
  # filter(!is.na(target_nospace)) %>% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateB, plate == "B")) %>% # Add sample metadata
  filter(!is.na(experiment)) %>%
  arrange(plate, well)

# Plate C
counts_plateC <- data.frame(counts_PlateC) %>% 
  mutate(barcode_name = rownames(counts_PlateC)) %>% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %>%
  gather("well_BC_seq", "counts", 2:c(ncol(counts_PlateC)+1)) %>%
  filter(counts >= 1) %>% # Remove undetected proteins counts
  left_join(meta_Abs) %>%
  unique() %>%
  # filter(!is.na(target_nospace)) %>% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateC, plate == "C")) %>% # Add sample metadata
  filter(!is.na(experiment)) %>%
  arrange(plate, well)

# Plate D
counts_plateD <- data.frame(counts_PlateD) %>% 
  mutate(barcode_name = rownames(counts_PlateD)) %>% #barcode_name #Barcodename
  dplyr::select(barcode_name, everything()) %>%
  gather("well_BC_seq", "counts", 2:c(ncol(counts_PlateD)+1)) %>%
  filter(counts >= 1) %>% # Remove undetected proteins counts
  left_join(meta_Abs) %>%
  unique() %>%
  # filter(!is.na(target_nospace)) %>% # Remove barcodes not in Ab metadata (not conjugated/added in panel)
  left_join(subset(meta_wellID_plateD, plate == "D")) %>% # Add sample metadata
  filter(!is.na(experiment)) %>%
  arrange(plate, well)

counts_dtbl <- list(counts_plateA, counts_plateB, counts_plateC, counts_plateD) %>% purrr::reduce(full_join)

counts_dtbl <- counts_dtbl %>%
  mutate(plate_well = paste0(plate, "_", well), 
         stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhibitor = factor(inhibitor, inhib_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108),
         inhib_conc_text = factor(inhib_conc_text, levels = inhib_full_DS108))

# Calculate per-sample properties
data_properties <- counts_dtbl %>%
  unique()%>%
  group_by(plate, well_BC_seq) %>%
  summarize(nCount = sum(counts), 
            nProt = n()) %>%
  left_join(meta_wellID) %>% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108))
```

***Table**: Unfiltered data-table properties loaded into R*

|                           | nCols (#samples)        | nRows (\# barcodes)     |
|------------------|---------------------------|---------------------------|
| Plate A: HBL1 PBS         | `r ncol(counts_PlateA)` | `r nrow(counts_PlateA)` |
| Plate B: HBL1 aIg+H2O2    | `r ncol(counts_PlateA)` | `r nrow(counts_PlateA)` |
| Plate C: OCI-Ly8 PBS      | `r ncol(counts_PlateA)` | `r nrow(counts_PlateA)` |
| Plate D: OCI-Ly8 aIg+H2O2 | `r ncol(counts_PlateA)` | `r nrow(counts_PlateA)` |

The full panel of mapped barcodes and 96 wells are present in the data table. First have a look at the total counts and number of detected protein barcodes per sample: There are a lot more proteins detected than put in the panel, due to barcode contamination and cross-ver between wells. The nostain samples have relatively low counts and proteins detected.

```{r fig_count_prot, fig.width=6, fig.height=4}
options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(data_properties, aes(log10(nCount), nProt, color = factor(inhibitor, levels = inhib_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) +
  scale_color_manual(values = colors_inhib, name = "Inhibitor") +
  theme_bw() +
  labs(title = "Correlation total counts and number of detected barcodes", 
       x = expression("Total counts Log"[10]),
       y = "Detected number of protein barcodes") +
  textsize_small

ggplot(data_properties, aes(log10(nCount), nProt, color = factor(inhib_conc, levels = inhib_conc_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) + 
  scale_color_manual(values = colors_blue9[2:9], name = "Inhibitor conc") +
  theme_bw() +
  labs(title = "Correlation total counts and number of detected barcodes", 
       x = expression("Total counts Log"[10]),
       y = "Detected number of protein barcodes") +
  textsize_small
```

***Figure:** Correlation between total counts and number of detected protein barcodes. We observe many more detected proteins than present in the staining.*

As we have a 96-wells plate, we can also have a look at total counts per plate detected:

```{r fig_plate_totalUMI, fig.width=10, fig.height=6}
options(repr.plot.width = 10, repr.plot.height = 6)

rng <- range(log10(data_properties$nCount))

# for (this_plate in plates_DS108) {
#   plot <- raw_map(data = log10(subset(data_properties, plate == this_plate)$nCount), well = subset(data_properties, plate == this_plate)$well, plate = 96, size = 7) +
#     scale_fill_viridis(expression("Total counts Log"[10]), limits = c(rng[1], rng[2])) +
#     labs(title = paste("Plate", this_plate))
#   theme_bw()
#   
#   print(plot)
# }

plot <- raw_grid(data = log10(data_properties$nCount), well = data_properties$well, plate_id = data_properties$plate, plate = 96, size = 6) +
  scale_fill_viridis(expression("Total counts Log"[10]), limits = c(rng[1], rng[2])) +
  theme_bw()

print(plot)

```

***Figure:** Plate overviews of total counts. Row A are no-stain samples.*


#### Barcode contamination comp

Several barcodes are contaminated with \<2.5% of other barcodes (all barcodes were sequenced individually to determine the contamination). As this can skew the results, especially for lowly present barcodes that are contaminated with highly present barcodes, we will compensate for this.

```{r comp_apply}
comp_matrix <- read_csv("data/DS108_StimInhibIDseq/BC_compensation_matrix.csv")

data_comp <- counts_dtbl %>%
  # filter(well == "A01") %>%
  dplyr::select(plate, well, barcode_name, counts) %>%
  left_join(comp_matrix, by = join_by(barcode_name == BC_to_correct)) #, multiple = "all"

# Get counts of OG BC
data_comp_OG <- data_comp %>%
  # filter(well == "A01") %>%
  filter(barcode_name == OG_BC) %>%
  dplyr::select(plate, well, OG_BC, counts) %>%
  dplyr::rename(counts_OG_BC = counts)

# Calculate false reads
data_false <- left_join(data_comp, data_comp_OG) %>% 
  mutate(false_counts = (counts_OG_BC / percent_OG_BC) * percent_BC_to_correct) %>%
  filter(barcode_name != OG_BC) %>%
  group_by(plate, well, barcode_name, counts) %>%
  summarize(total_false_counts = sum(false_counts, na.rm = TRUE))

counts_dtbl <- counts_dtbl %>%
  # filter(well == "A01") %>%
  left_join(data_false) %>% 
  mutate(total_false_counts = replace_na(total_false_counts, 0), 
         true_counts = as.integer(counts - total_false_counts), 
         percent_true = (true_counts / counts) * 100) %>%
  filter(!is.na(target_nospace)) # Remove barcodes not in Ab metadata (not conjugated/added in panel)

compensation <- counts_dtbl %>%
  dplyr::select(plate, well, barcode_name, counts, total_false_counts, true_counts, percent_true, target_nospace)

# Calculate per-sample properties
data_properties_true <- counts_dtbl %>%
  unique()%>%
  group_by(plate, well_BC_seq) %>%
  summarize(nCount = sum(true_counts), 
            nProt = n()) %>%
  left_join(meta_wellID) %>% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_count_raw_comp, fig.width=6, fig.height=4}
options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(counts_dtbl, aes(log10(counts), log10(true_counts), color = factor(inhibitor, levels = inhib_DS108))) +
  geom_point(alpha = 1, size = 0.8) +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") + 
  facet_grid(cell_line~stimulus) +
  scale_color_manual(values = colors_inhib, name = "Inhibitor") +
  theme_bw() +
  labs(title = "Correlation raw counts and compensated counts",
       x = expression("Raw counts Log"[10]),
       y = expression("Compensated counts Log"[10])) +
  textsize_small

ggplot(data_properties_true, aes(log10(nCount), nProt, color = factor(inhib_conc, levels = inhib_conc_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) + # stimulus, staining_cells, staining_conc, staining_volume
  scale_color_manual(values = colors_blue9[2:9], name = "Inhibitor conc") +
  # scale_color_binned(type = "viridis", name = "Staining conc (ug/mL)") +
  theme_bw() +
  labs(title = "Correlation total counts and number of detected barcodes", 
       x = expression("Total counts Log"[10]),
       y = "Detected number of protein barcodes") +
  textsize_small
```

***Figure:** Correlation between raw counts and compensated counts, and between compensated counts and detected number of barcodes. After compensation, all samples (except 2 nostain samples) have 107 detected proteins (the number of Ab in the panel).*


#### Count normalisation

To remove variation between replicates, the data is normalised per cell line and per target protein with a geometric mean scaling factor: First, the geometric mean of each target is calculated separately for each cell line, and then the median of the protein scaling factors is calculated separately for each sample.

```{r normalisation}
scaling_factors <- counts_dtbl %>%
  filter(staining == "yes") %>%
  # filter() %>% #target_nospace == "GAPDH" # modification == "no"
  ungroup() %>%
  group_by(cell_line, target_nospace) %>%
  mutate(scaling_factor = counts / median(counts),
         scaling_factor_geo = counts / exp(mean(log((counts)))), 
         true_scaling_factor = true_counts / median(true_counts),
         true_scaling_factor_geo = true_counts / exp(mean(log((true_counts))))) %>%
  ungroup() %>%
  group_by(cell_line, descript_rep) %>%
  summarize(scaling_factor = median(scaling_factor), .groups = "keep",
            scaling_factor_geo = median(scaling_factor_geo), 
            true_scaling_factor = median(true_scaling_factor),
            true_scaling_factor_geo = median(true_scaling_factor_geo, na.rm = TRUE))

counts_dtbl <- counts_dtbl %>%
  filter(staining == "yes") %>%
  left_join(scaling_factors) %>%
  ungroup() %>%
  mutate(counts_norm = counts / scaling_factor,
         counts_norm_geo = counts / scaling_factor_geo, 
         true_counts_norm = true_counts / true_scaling_factor, 
         true_counts_norm_geo = true_counts / true_scaling_factor_geo)

counts_dtbl <- filter(counts_dtbl, !is.na(target_nospace))

# Calculate per-sample properties
data_properties_norm <- counts_dtbl %>%
  unique()%>%
  group_by(plate, well_BC_seq) %>%
  summarize(nCount_norm = sum(counts_norm_geo), 
            nProt = n()) %>%
  left_join(meta_wellID) %>% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108))

data_properties_norm_true <- counts_dtbl %>%
  unique()%>%
  group_by(plate, well_BC_seq) %>%
  summarize(nCount_norm = sum(true_counts_norm_geo), 
            nProt = n()) %>%
  left_join(meta_wellID) %>% # Add sample metadata 
  mutate(stimulus = factor(stimulus, levels = stim_DS108), 
         cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_count_norm, fig.width=6, fig.height=4}
options(repr.plot.width = 6, repr.plot.height = 4)

ggplot(counts_dtbl, aes(log10(counts), log10(true_counts_norm_geo), color = factor(inhibitor, levels = inhib_DS108))) +
  geom_point(alpha = 1, size = 0.8) +
  geom_abline(intercept = 0, slope = 1, linetype = "dashed") + 
  facet_grid(cell_line~stimulus) +
  scale_color_manual(values = colors_inhib, name = "Inhibitor") +
  theme_bw() +
  labs(title = "Correlation raw counts and normalised counts",
       x = expression("Raw counts Log"[10]),
       y = expression("Normalised counts Log"[10])) +
  textsize_small

ggplot(data_properties_norm_true, aes(log10(nCount_norm), nProt, color = factor(inhib_conc, levels = inhib_conc_DS108))) +
  geom_point(alpha = 1, size = 1) +
  facet_grid(cell_line~stimulus) + 
  scale_color_manual(values = colors_blue9[2:9], name = "Inhibitor conc") +
  theme_bw() +
  labs(title = "Correlation normalised counts and number of detected barcodes", 
       x = expression("Normalised counts Log"[10]),
       y = "Detected number of protein barcodes") +
  textsize_small
```

***Figure:** Correlation between raw counts and compensated counts.*


#### Data wrangling

Calculate the following:

-   Mean + sd for replicates (true_counts + true_counts_norm_geo)

-   Manual fold changes of all samples compared to the mean of 0 uM (DMSO) + PBS samples

  *% phosphoprotein of total protein (only for proteins with both total + phospho targets)*

```{r mean_rep}
# Calculate mean + sd of true_counts and true_counts_norm_geo
mean_dtbl <- counts_dtbl %>%
  dplyr::group_by(description, target_nospace) %>%
  summarise(
    counts_sd = sd(true_counts, na.rm = T),
    counts = mean(true_counts, na.rm = T), 
    counts_norm_sd = sd(true_counts_norm_geo, na.rm = T),
    counts_norm = mean(true_counts_norm_geo, na.rm = T)
    )

# Add metadata
mean_dtbl <- mean_dtbl %>%
  left_join(dplyr::select(counts_dtbl,
                   c(barcode_name, DS_Ab_nr, target, modification, epitope, marker_cell, marker_pathway, marker_pathway_detail, isotype, DS_barcode_nr, DS_barcode_seq, target_nospace, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, staining, description))) %>%
  distinct()
```

```{r fig_wrangle1, fig.width=10, fig.height=4}
prot <- "BTK_Y551"

# Raw (compensated) counts
plot1 <- ggplot(subset(mean_dtbl, staining == "yes" & cell_line == "HBL1" & inhibitor == "iSYK" & target_nospace == prot)) +
  geom_col(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = counts, 
      group = stimulus,
      fill = stimulus),
      position = "dodge2") + 
  geom_errorbar(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      ymin = counts - counts_sd,
      ymax = counts + counts_sd,
      group = stimulus),
      position = position_dodge2(width = 0.9),
      width = 0.9,
    ) +
  facet_wrap(vars(factor(cell_line, labels = "Raw (compensated)")), scales = "free_y") +
  theme_bw() +
  # scale_y_log10() +
  scale_fill_manual(values = colors_stim) +
  labs(x = "", y = "", title = prot) +
  textsize_small

# Normalised counts
plot2 <- ggplot(subset(mean_dtbl, staining == "yes" & cell_line == "HBL1" & inhibitor == "iSYK" & target_nospace == prot)) +
  geom_col(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = counts_norm, 
      group = stimulus,
      fill = stimulus),
      position = "dodge2") + 
  geom_errorbar(aes(
      x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      ymin = counts_norm - counts_norm_sd,
      ymax = counts_norm + counts_norm_sd,
      group = stimulus),
      position = position_dodge2(width = 0.9),
      width = 0.9,
    ) +
  facet_wrap(vars(factor(cell_line, labels = "Normalised")), scales = "free_y") +
  theme_bw() +
  # scale_y_log10() +
  scale_fill_manual(values = colors_stim) +
  labs(x = "", y = "", title = prot) +
  textsize_small

plot_grid(plot1, plot2, ncol = 2)
```

***Figure:** Mean signal (+ sd) of replicates for BTK (Y551) (HBL1 cells, iSYK inhibitor). Raw counts on the left, normalised counts on the right. Normalisation removes the variation between replicates but not the effect of iSYK concentration.*

```{r fold_change}
# Manually calculate the fold change of true_counts and true_counts_norm_geo
# Use the mean of 0 uM (DMSO) + PBS as baseline
counts_FC0 <- counts_dtbl %>%
  filter(stimulus == "PBS" & inhib_conc == "0 uM (DMSO)") %>%
  group_by(cell_line, target_nospace) %>%
  summarise(counts_0 = mean(true_counts), 
            counts_norm_0 = mean(true_counts_norm_geo))

counts_dtbl <- left_join(counts_dtbl, counts_FC0) %>%
  mutate(man_FC_counts = (true_counts / counts_0), 
         man_log2FC_counts = log2(man_FC_counts), 
         man_FC_norm = (true_counts_norm_geo / counts_norm_0), 
         man_log2FC_norm = log2(man_FC_norm))

# Calculate mean + sd of manual fold changes of replicates
mean_FC <- counts_dtbl %>%
  dplyr::group_by(description, target_nospace) %>%
  summarise(
    FC_counts_sd = sd(man_FC_counts, na.rm = T),
    FC_counts = mean(man_FC_counts, na.rm = T), 
    log2FC_counts_sd = sd(man_log2FC_counts, na.rm = T), 
    log2FC_counts = mean(man_log2FC_counts, na.rm = T), 
    FC_norm_sd = sd(man_FC_norm, na.rm = T),
    FC_norm = mean(man_FC_norm, na.rm = T), 
    log2FC_norm_sd = sd(man_log2FC_norm, na.rm = T), 
    log2FC_norm = mean(man_log2FC_norm, na.rm = T)
)

mean_dtbl <- mean_dtbl %>%
  left_join(mean_FC) %>%
  distinct()
```

```{r fig_wrangle2, fig.width=8, fig.height=5}
prot <- "BTK_Y551"

# Normalised counts
plot <- ggplot(subset(counts_dtbl, staining == "yes" & cell_line == "HBL1" & target_nospace == prot)) +
  geom_point(aes(
    x = factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
      y = man_FC_norm,
    color = stimulus)) +
  facet_wrap(vars(factor(inhibitor, levels = inhib_DS108))) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim) +
  labs(x = "", y = "Fold change over 0 uM DMSO + PBS \n(normalised counts)", title = prot) +
  textsize_small

print(plot)
```

***Figure:** Manually calculated fold change of signal compared to 0 uM inhibitor (DMSO ctrl) + PBS stimulation for BTK (Y551) (HBL1 cells, normalised counts). *

```{r percent_phos, eval=F}

```


Finally, we will save the datasets as .csv file

-   IDseq_data_sample.csv: Dataset of counts per protein per sample 

-   IDseq_data_condition.csv: Dataset of counts per protein per condition (mean of replicates)

```{r save}
# Dataset of counts per protein per sample 
data_well_all <- counts_dtbl %>% 
  arrange(plate_well, target_nospace) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

write.csv(data_well_all, file = "output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_sample.csv", row.names = F)

# Dataset of mean counts per protein per condition (replicates combined)
data_mean_all <- mean_dtbl %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

write.csv(data_mean_all, file = "output/DS108_StimInhibIDseq/IDseq_ann/IDseq_data_condition.csv", row.names = F)
```

```{r clear}
rm(list = ls(pattern = "Plate"))
gc()
```



## Counts analysis

#### Counts per sample

To gain insight into the effect of experimental conditions on total signal, we first have a look at the total (raw) and normalised counts per sample. 

```{r fig_counts_sample, fig.width=10, fig.height=5}
# Compensated counts
fig_sample_true <- ggplot(data_properties_true, aes(inhib_conc, nCount, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10() +
  scale_color_manual(values = c("PBS" = "black", "aIg+H2O2" = "orange"), name = "") +
  labs(x = "Inhibitor conc", y = "Compensated counts", title = "Compensated counts") + 
  textsize_small

print(fig_sample_true)

# Normalised compensated counts
fig_sample_norm <- ggplot(data_properties_norm_true, aes(inhib_conc, nCount_norm, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10() +
  scale_color_manual(values = c("PBS" = "black", "aIg+H2O2" = "orange"), name = "") +
  labs(x = "Inhibitor conc", y = "Normalised compensated counts", title = "Normalised compensated counts") + 
  textsize_small

print(fig_sample_norm)
```

```{r fig_counts_sample2, fig.width=15, fig.height=5}
# Compensated counts
fig_sample_true <- ggplot(subset(data_properties_true, staining == "yes"), aes(inhib_conc, nCount, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(limits = c(5e4, 5e5)) +
  scale_color_manual(values = c("PBS" = "black", "aIg+H2O2" = "orange"), name = "") +
  labs(x = "Inhibitor conc", y = "Compensated counts", title = "Compensated counts") + 
  textsize_small

# Normalised compensated counts
fig_sample_norm <- ggplot(data_properties_norm_true, aes(inhib_conc, nCount_norm, color = stimulus)) +
  geom_jitter(alpha = 1, size = 1, width = 0.1) +
  facet_grid(cell_line~inhibitor) +
  theme_bw() +
  RotatedAxis() +
  scale_y_log10(limits = c(5e4, 5e5)) +
  scale_color_manual(values = c("PBS" = "black", "aIg+H2O2" = "orange"), name = "") +
  labs(x = "Inhibitor conc", y = "Normalised compensated counts", title = "Normalised compensated counts") + 
  textsize_small

plot_grid(fig_sample_true, fig_sample_norm, ncol = 2)
```

***Figure:** Distribution counts per sample with compensation (left) and normalisation (right).*

#### Counts per Ab

We can also look at the counts per Ab for selected samples to get an indication of any differences between proteins and between aIg+H2O2 and PBS stimulation.

```{r fig_Ab_counts, fig.width=15, fig.height=8}
# Compensated counts
fig_Ab_true <- ggplot(unique(subset(counts_dtbl, inhib_conc == "0 uM (DMSO)")), 
       aes(x = reorder(target_nospace, true_counts), 
           y = true_counts, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = "black", size = 6)) +
  scale_y_log10(
    breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") +
  labs(x = "Protein", y = "Compensated counts", title = "Compensated counts") +
  theme_bw() +
  facet_grid2(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

print(fig_Ab_true)
```

***Figure:** Distribution of compensated counts per Ab-barcode for the 0 uM (DMSO) samples (non-normalised).*

```{r fig_Ab_counts_norm, fig.width=15, fig.height=8}
# Compensated counts
fig_Ab_norm <- ggplot(unique(subset(counts_dtbl, inhib_conc == "0 uM (DMSO)")), 
       aes(x = reorder(target_nospace, true_counts_norm_geo), 
           y = true_counts_norm_geo, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = "black", size = 6)) +
  scale_y_log10(
    breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") +
  labs(x = "Protein", y = "Normalised compensated counts", title = "Normalised compensated counts") +
  theme_bw() +
  facet_grid2(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

print(fig_Ab_norm)
```

***Figure:** Distribution of normalised compensated counts per Ab-barcode for the 0 uM (DMSO) samples. Here we see some protein targets that clearly show a difference between the aIg+H2O2 and PBS samples.*

```{r fig_Ab_counts_HBL1, fig.width=16, fig.height=10}
# Compensated
fig_Ab_true_HBL1 <- ggplot(unique(subset(counts_dtbl, cell_line == "HBL1" & inhib_conc_text == "0 uM iSYK (DMSO)")), 
       aes(x = reorder(target_nospace, true_counts), 
           y = true_counts, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = "black", size = 6)) +
  scale_y_log10(
    breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") +
  labs(x = "Protein", y = "Compensated counts", title = "Compensated counts") +
  theme_bw() +
  facet_grid(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

# Normalised
fig_Ab_norm_HBL1 <- ggplot(unique(subset(counts_dtbl, cell_line == "HBL1" & inhib_conc_text == "0 uM iSYK (DMSO)")), 
       aes(x = reorder(target_nospace, true_counts_norm_geo), 
           y = true_counts_norm_geo, 
           color = as.factor(stimulus), 
           shape = as.factor(replicate))) +
  geom_point(alpha = 0.7) +
  scale_x_reordered() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1, vjust = 0.5, colour = "black", size = 6)) +
  scale_y_log10(
    breaks = trans_breaks("log10", function(x) 10^x),
    labels = trans_format("log10", math_format(10^.x))) +
  annotation_logticks(sides = "l") +
  labs(x = "Protein", y = "Normalised compensated counts", title = "Normalised compensated counts") +
  theme_bw() +
  facet_grid(cell_line~inhib_conc_text) +
  RotatedAxis() +
  scale_color_colorblind() +
  textsize_small

plot_grid(fig_Ab_true_HBL1, fig_Ab_norm_HBL1, ncol = 1)
```

***Figure:** Distribution of counts per Ab-barcode for the HBL1 0 uM iSYK (DMSO) samples. The normalisation removes the variation between replicates, but preserves the variation between proteins and highlights the differences between aIg+H2O2 and PBS stimulated samples.*



## Basal cell/signaling state

First, we look at the basal signaling levels in both HBL1 and OCI-Ly8 cells (without any inhibitor/DMSO and without any stimulus). We can investigate differences in basal signaling between the cell lines.

### DEseq2 {.tabset}

<https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html>

#### Cell

DESeq2 analysis parameters:

-   0 uM inhibitor (PBS ctrls), PBS stimulus, both cell lines samples (24 samples)

-   Compensated counts (>= 1)

-   Model design: cell_line (ref: HBL1)

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_basal}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts <- counts_dtbl  %>%
  filter(stimulus == "PBS" & inhib_conc == "0 uM (PBS)" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(meta_wellID) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         cell_inhib = paste(cell_line, inhibitor)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign <- ~ cell_line # cell_inhib

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_line <- relevel(dds$cell_line, ref = "HBL1")
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$cell_inhib <- relevel(dds$cell_inhib, ref = "HBL1 iSYK")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_basal}
# Create results for all DESeq_comparisons
results_cell <- results(dds, name = "cell_line_OCI.Ly8_vs_HBL1", alpha = 0.05)

# results_iBTK <- results(dds, name = "inhibitor_iBTK_vs_iSYK", alpha = 0.05)
# results_iNFkB <- results(dds, name = "inhibitor_iNFkB_vs_iSYK", alpha = 0.05)
# results_iPI3Kd <- results(dds, name = "inhibitor_iPI3Kd_vs_iSYK", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_cell) 
                     # results_iBTK, results_iNFkB, results_iPI3Kd)

# Results summary
summary(results_cell, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_basal, fig.width=3, fig.height=4}
# par(mfrow = c(1,3), mar = c(4,4,2,1))
par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_cell, alpha = signcutoff, main = "OCI-Ly8 vs HBL1")
# plotMA(results_iBTK, alpha = signcutoff, main = "iBTK vs iSYK")
# plotMA(results_iPI3Kd, alpha = signcutoff, main = "iPI3Kd vs iSYK")
```

##### Data transformation {.tabset}

```{r transformation_basal}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_basal, fig.width=4, fig.height=3}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_conc"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(cell_line, levels = cells_DS108), factor(cell_line, levels = cells_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhib_conc) +
    scale_color_manual(values = colors_cell, name = "Cell line") +
    scale_fill_manual(values = colors_cell, name = "Cell line") +
    scale_shape_manual(values = c(16, 17), name = "Cell line") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 shows a clear separation on cell line, and PC2 contributes little. The different types of data transformation also give similar results.*

###### Variance

```{r fig_sd_basal, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform (log2(n+1))") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising transform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log transform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_basal, fig.width=5, fig.height=5}
options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$cell_inhib,
                                      list_datatransform[[data]]$replicate,
                                      sep = " - ")
  # rownames(sampleDistMatrix) <- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```

##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_basal}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_cell")
                        # "volc_iBTK", "volc_iNFkB", "volc_iPI3Kd")

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_basal, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_cell)
                          # volc_iBTK, volc_iNFkB, volc_iPI3Kd)
titles_volcano <- c("OCI-Ly8 vs HBL1 (basal signaling w/o inhibitor)")
                    # "volc_iBTK", "volc_iNFkB", "volc_iPI3Kd")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plot of DESeq2 model comparison: OCI-Ly8 vs HBL1. Various (phospho)proteins are significantly lower in the OCI-Ly8 cells (red), but there are also some (phospho)proteins with higher presence in OCI-Ly8 (blue).*

##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_basal}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_cell")
                        # "heat_iBTK", "heat_iNFkB", "heat_iPI3Kd")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("cell_line", "inhibitor")]) %>%
  arrange(cell_line, factor(inhibitor, levels = inhib_DS108))
```

```{r fig_heatmap_basal, fig.width=10, fig.height=8}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# OCI-Ly8 vs HBL1
heat_cell$marker_pathway[is.na(heat_cell$marker_pathway)] <- "Other"

ann_colors_man <- list(cell_line = colors_cell,
                       inhibitor = colors_inhib,
                       modification = c(phospho = "#f768a1", no = "grey", cleaved = "#fff7bc"),
                       marker_pathway = setNames(colors_light12[1:length(unique(heat_cell$marker_pathway))], unique(heat_cell$marker_pathway))
)

# Get the rld transformed data for the heatmap and scale per row
data_cell <- t(scale(t(assay(rld)[unique(c(heat_cell$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_cell,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$cell_line,
  row_split = subset(heat_cell)$modification,
  cellwidth = 10,
  cellheight = 10,
  fontsize = 8,
  border_color = NA,
  # main = "scaled per row, ComplexHeatmap"
)
```

***Figure:** Heatmap of all significant results in the OCI-Ly8 vs HBL1 comparison (no threshold for fold change).*

*The apoptosis markers cleaved PARP and active Caspase 3 are higher in the OCI-Ly8 cells, which is consistent with the FACS data, where we saw more debris and dead cells in the OCI-Ly8 samples. Regarding the phospho-proteins, several upstream BCR signaling proteins and MAPK-JNK signaling proteins show lower presence in OCI-Ly8, while some JAK-STAT signaling proteins show higher presence. Regarding the non-phospho-proteins, serveral surface markers are increased in OCI-Ly8, including IgE, while IgM, CD79a and several intracellular proteins are decreased in OCI-Ly8.*


### Protein comparisons

Finally, we visualised the normalised true counts of all significant proteins in the OCI-Ly8 vs HBL1 comparison (41 in total).

```{r sign_data_basal}
# Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins
sign_prot_cell <- sort(unique(c(heat_cell$target_nospace)))
# print(c("Significant proteins (OCI-ly8 vs HBL1 basal signaling):", sign_prot_cell))

# Datasets generated during analysis
# heat_cell: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_cell: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
clean_data_well <- data.frame(data_cell) %>% 
  mutate(target_nospace = rownames(data_cell)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value", 2:c(ncol(data_cell)+1)) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_cell) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

prot_labels <- sort(setNames(unique(pull(clean_data_well, target)), unique(pull(clean_data_well, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
clean_data_sign <- heat_cell %>%
  filter(target_nospace %in% sign_prot_cell) %>%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean = baseMean, 
                log2FC = log2FoldChange, 
                log2FC_se = lfcSE)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_cell & stimulus == "PBS" & inhib_conc == "0 uM (PBS)") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)
```

```{r fig_counts_basal_all, fig.width=15, fig.height=27}
plot <- ggplot(clean_data_well, 
               aes(factor(cell_line, levels = cells_DS108),
                   counts_norm, 
                   color = cell_line, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target_nospace, labels = prot_labels)), ncol = 5) +
  theme_bw() +
  scale_y_log10() +
  scale_color_manual(values = colors_cell, name = "Cell line") + 
  labs(x = "", y = "Normalized counts (log10 scale)") +
  textsize_small

print(plot)
```

***Figure:** Normalised counts for all significantly changing proteins. For some proteins the effect size is quite small.*

```{r fig_counts_basal, fig.width=2.5, fig.height=2.5}
options(repr.plot.width = 3.5, repr.plot.height = 2.5)

# Re-redesigned fig
filt_clean_data_mean <- clean_data_mean %>%
  dplyr::group_by(cell_line, target_nospace) %>%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_cell) {
  plot <- ggplot() +
    geom_jitter(data = subset(clean_data_well, target_nospace == protein), 
                aes(factor(cell_line, levels = cells_DS108), counts_norm, color = cell_line), 
                alpha = 1, size = 1.5, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
                  aes(factor(cell_line, levels = cells_DS108),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = cell_line),
                  width = 0.5, size = 1, color = "black") +
    # geom_col(data = subset(filt_clean_data_mean, target_nospace == protein), 
    #          aes(factor(cell_line, levels = cells_DS108), mean_counts_norm, group = cell_line, fill = cell_line), 
    #          position = "dodge2", alpha = 0.5) +
    # geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
    #               aes(factor(cell_line, levels = cells_DS108),
    #                   ymin = mean_counts_norm - mean_counts_norm_sd,
    #                   ymax = mean_counts_norm + mean_counts_norm_sd,
    #                   group = cell_line),
    #               position = position_dodge2(width = 0.9), width = 0.5) +
    facet_wrap(vars(factor(target_nospace))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_cell, name = "Cell line") + 
    labs(x = "", y = "Normalized counts") +
    theme_bw() +
    theme(legend.position = "none", panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}
```


### Save data

Save all information and datasets as .csv file

-   sign_proteins.txt: List of significantly changing proteins

-   sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis

-   sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   sign_mean.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins

```{r save_basal}
# List of significantly changing proteins: sign_prot_cell
write(sign_prot_cell, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_proteins.txt")

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_DESeq2data.csv", row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_sample.csv", row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal/sign_condition.csv", row.names = F)
```

```{r clear_basal}
# Remove unnecessary files to clear up memory
rm(list = ls(pattern = "results_"))
rm(list = ls(pattern = "volc_"))
rm(list = ls(pattern = "heat_"))
rm(list = ls(pattern = "list_"))
rm(list = ls(pattern = "names_"))
rm(list = ls(pattern = "titles_"))
rm(list = ls(pattern = "clean_"))
rm(list = ls(pattern = "sign_"))
gc()
```



## Inhibitor effects on basal signaling

Now, we look at the effect of the inhibitors and their different concentrations on basal signaling levels in both HBL1 and OCI-Ly8 cells (without any stimulus). We can investigate differences between inhibitors, between inhibitor concentrations, and between cell lines.

### DEseq2 {.tabset}

<https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html>

#### HBL1 conc

DESeq2 analysis parameters:

-   4 inhibitors with all conc, PBS stimulus, HBL1 cells only (72 samples)

-   Compensated counts (>= 1)

-   Model design: inhib_text_new (ref: DMSO ctrl)

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_bas_HBL1_conc}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(stimulus == "PBS" & cell_line == "HBL1" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ inhib_text_new

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhib_new <- relevel(dds$inhib_new, ref = "DMSO_ctrl")
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$inhib_conc <- relevel(dds$inhib_conc, ref = "0 uM (DMSO)")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_bas_HBL1_conc}
# Create results for all DESeq_comparisons
results_PBS_iH <- results(dds, name = "inhib_text_new_PBS.ctrl_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK0.1_H <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_H <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_H <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_H <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_H <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_H <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_H <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_H <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_H <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_H <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_H <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_H <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_H <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_H <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_H <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_H <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_PBS_iH, 
                     results_iSYK0.1_H, results_iSYK1_H, results_iSYK10_H, results_iSYK100_H, 
                     results_iBTK0.1_H, results_iBTK1_H, results_iBTK10_H, results_iBTK100_H, 
                     results_iPI3Kd0.1_H, results_iPI3Kd1_H, results_iPI3Kd10_H, results_iPI3Kd100_H, 
                     results_iNFkB0.1_H, results_iNFkB1_H, results_iNFkB10_H, results_iNFkB100_H)

# Results summary
summary(results_iSYK100_H, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_bas_HBL1_conc, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_PBS_iH, alpha = signcutoff, main = "PBS vs DMSO")
plotMA(results_iSYK100_H, alpha = signcutoff, main = "100 uM iSYK vs DMSO")
plotMA(results_iNFkB100_H, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
```


##### Data transformation {.tabset}

```{r transformation_bas_HBL1_conc}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_bas_HBL1_conc, fig.width=8, fig.height=3}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 8, repr.plot.height = 3)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is mainly the difference between the 100 uM iNFkB samples and all other samples, and PC2 is mainly the difference between 1 repl of 100 uM iSYK and all other samples.*

###### Variance

```{r fig_sd_bas_HBL1_conc, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_bas_HBL1_conc, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$inhib_conc_text,
                                      list_datatransform[[data]]$replicate,
                                      sep = " - ")
  # rownames(sampleDistMatrix) <- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_bas_HBL1_conc}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_PBS_H",
                        "volc_iSYK0.1_H", "volc_iSYK1_H", "volc_iSYK10_H", "volc_iSYK100_H",
                        "volc_iBTK0.1_H", "volc_iBTK1_H", "volc_iBTK10_H", "volc_iBTK100_H",
                        "volc_iPI3Kd0.1_H", "volc_iPI3Kd1_H", "volc_iPI3Kd10_H", "volc_iPI3Kd100_H",
                        "volc_iNFkB0.1_H", "volc_iNFkB1_H", "volc_iNFkB10_H", "volc_iNFkB100_H")

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_bas_HBL1_conc, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_PBS_H, 
                          volc_iSYK0.1_H, volc_iSYK1_H, volc_iSYK10_H, volc_iSYK100_H, 
                          volc_iBTK0.1_H, volc_iBTK1_H, volc_iBTK10_H, volc_iBTK100_H, 
                          volc_iPI3Kd0.1_H, volc_iPI3Kd1_H, volc_iPI3Kd10_H, volc_iPI3Kd100_H, 
                          volc_iNFkB0.1_H, volc_iNFkB1_H, volc_iNFkB10_H, volc_iNFkB100_H)
# titles_volcano <- c("OCI-Ly8 vs HBL1 (... signaling w/o inhibitor)")
titles_volcano <- c("PBS ctrl vs DMSO ctrl (HBL1 w/o stim)",
                    "0.1 uM iSYK vs DMSO ctrl (HBL1 w/o stim)", "1 uM iSYK vs DMSO ctrl (HBL1 w/o stim)", "10 uM iSYK vs DMSO ctrl (HBL1 w/o stim)", "100 uM iSYK vs DMSO ctrl (HBL1 w/o stim)",
                    "0.1 uM iBTK vs DMSO ctrl (HBL1 w/o stim)", "1 uM iBTK vs DMSO ctrl (HBL1 w/o stim)", "10 uM iBTK vs DMSO ctrl (HBL1 w/o stim)", "100 uM iBTK vs DMSO ctrl (HBL1 w/o stim)",
                    "0.1 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)", "1 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)", "10 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)", "100 uM iPI3Kd vs DMSO ctrl (HBL1 w/o stim)",
                    "0.1 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)", "1 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)", "10 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)", "100 uM iNFkB vs DMSO ctrl (HBL1 w/o stim)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plot of DESeq2 model comparisons: each inhibitor conc vs DMSO ctrl. iSYK shows 1 protein oging down at 1 uM, more at 10 uM and even more at 100 uM. iBTK starts showing significantly changing proteins at 10 uM. iPI3Kd shows 3 proteins significantly increasing at 0.1 uM, and only 1 protein significantly decreasing at 100 uM. iNFkB shows several proteins significantly changing at 10 uM, and even more proteins changing at 100 uM.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_bas_HBL1_conc}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_PBS_H",
                        "heat_iSYK0.1_H", "heat_iSYK1_H", "heat_iSYK10_H", "heat_iSYK100_H",
                        "heat_iBTK0.1_H", "heat_iBTK1_H", "heat_iBTK10_H", "heat_iBTK100_H",
                        "heat_iPI3Kd0.1_H", "heat_iPI3Kd1_H", "heat_iPI3Kd10_H", "heat_iPI3Kd100_H",
                        "heat_iNFkB0.1_H", "heat_iNFkB1_H", "heat_iNFkB10_H", "heat_iNFkB100_H")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_100uM_H <- list(heat_iSYK100_H, heat_iBTK100_H, heat_iPI3Kd100_H, heat_iNFkB100_H) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("inhib_new", "inhib_conc")]) %>%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_bas_HBL1_conc_100, fig.width=14, fig.height=8}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# Get the rld transformed data for the heatmap and scale per row
data_100uM_H <- t(scale(t(assay(rld)[unique(c(heat_100uM_H$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_100uM_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)"
)
```

```{r fig_heatmap_bas_HBL1_conc, fig.width=14, fig.height=8}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM iNFkB vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iNFkB_H <- t(scale(t(assay(rld)[unique(c(heat_iNFkB100_H$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_iNFkB_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iNFkB100_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins in 100 uM iNFkB (HBL1 cells w/o stim)"
)

# 100 uM iSYK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iSYK_H <- t(scale(t(assay(rld)[unique(c(heat_iSYK100_H$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iSYK_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iSYK100_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins in 100 uM iSYK (HBL1 cells w/o stim)"
)

# 100 uM iBTK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iBTK_H <- t(scale(t(assay(rld)[unique(c(heat_iBTK100_H$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iBTK_H,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iBTK100_H)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins in 100 uM iBTK (HBL1 cells w/o stim)"
)
```
***Figure:** Heatmaps of the significant results in the 100 uM inhibitor vs DMSO ctrl comparison (no threshold for fold change).*


#### OCI-Ly8 conc

DESeq2 analysis parameters:

-   4 inhibitors with all conc, PBS stimulus, OCI-Ly8 cells only (72 samples)

-   Compensated counts (>= 1)

-   Model design: inhib_text_new (ref: DMSO ctrl)

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_bas_Ly8_conc}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(stimulus == "PBS" & cell_line == "OCI-Ly8" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ inhib_text_new

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhib_new <- relevel(dds$inhib_new, ref = "DMSO_ctrl")
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$inhib_conc <- relevel(dds$inhib_conc, ref = "0 uM (DMSO)")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_bas_Ly8_conc}
# Create results for all DESeq_comparisons
results_PBS_iO <- results(dds, name = "inhib_text_new_PBS.ctrl_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK0.1_O <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_O <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_O <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_O <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_O <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_O <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_O <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_O <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_O <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_O <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_O <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_O <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_O <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_O <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_O <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_O <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_PBS_iO, 
                     results_iSYK0.1_O, results_iSYK1_O, results_iSYK10_O, results_iSYK100_O, 
                     results_iBTK0.1_O, results_iBTK1_O, results_iBTK10_O, results_iBTK100_O, 
                     results_iPI3Kd0.1_O, results_iPI3Kd1_O, results_iPI3Kd10_O, results_iPI3Kd100_O, 
                     results_iNFkB0.1_O, results_iNFkB1_O, results_iNFkB10_O, results_iNFkB100_O)

# Results summary
summary(results_iSYK100_O, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_bas_Ly8_conc, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_PBS_iO, alpha = signcutoff, main = "PBS vs DMSO")
plotMA(results_iSYK100_O, alpha = signcutoff, main = "100 uM iSYK vs DMSO")
plotMA(results_iNFkB100_O, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
```


##### Data transformation {.tabset}

```{r transformation_bas_Ly8_conc}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_bas_Ly8_conc, fig.width=6, fig.height=4}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 6, repr.plot.height = 4)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is mainly the difference between the 100 uM iNFkB samples and all other samples, and PC2 is mainly the difference between iSYK and iBTK inhibitor concentrations.*

###### Variance

```{r fig_sd_bas_Ly8_conc, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_bas_Ly8_conc, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$inhib_conc_text,
                                      list_datatransform[[data]]$replicate,
                                      sep = " - ")
  # rownames(sampleDistMatrix) <- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_bas_Ly8_conc}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_PBS_O",
                        "volc_iSYK0.1_O", "volc_iSYK1_O", "volc_iSYK10_O", "volc_iSYK100_O",
                        "volc_iBTK0.1_O", "volc_iBTK1_O", "volc_iBTK10_O", "volc_iBTK100_O",
                        "volc_iPI3Kd0.1_O", "volc_iPI3Kd1_O", "volc_iPI3Kd10_O", "volc_iPI3Kd100_O",
                        "volc_iNFkB0.1_O", "volc_iNFkB1_O", "volc_iNFkB10_O", "volc_iNFkB100_O")

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_bas_Ly8_conc, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_PBS_O, 
                          volc_iSYK0.1_O, volc_iSYK1_O, volc_iSYK10_O, volc_iSYK100_O, 
                          volc_iBTK0.1_O, volc_iBTK1_O, volc_iBTK10_O, volc_iBTK100_O, 
                          volc_iPI3Kd0.1_O, volc_iPI3Kd1_O, volc_iPI3Kd10_O, volc_iPI3Kd100_O, 
                          volc_iNFkB0.1_O, volc_iNFkB1_O, volc_iNFkB10_O, volc_iNFkB100_O)
# titles_volcano <- c("OCI-Ly8 vs HBL1 (... signaling w/o inhibitor)")
titles_volcano <- c("PBS ctrl vs DMSO ctrl (OCI-Ly8 w/o stim)",
                    "0.1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)", "1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)", "10 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)", "100 uM iSYK vs DMSO ctrl (OCI-Ly8 w/o stim)",
                    "0.1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)", "1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)", "10 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)", "100 uM iBTK vs DMSO ctrl (OCI-Ly8 w/o stim)",
                    "0.1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)", "1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)", "10 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)", "100 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/o stim)",
                    "0.1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)", "1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)", "10 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)", "100 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/o stim)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plot of DESeq2 model comparisons: each inhibitor conc vs DMSO ctrl. iSYK shows 1 protein going down at 0.1 uM, and more at increasingly higher conc. iBTK starts showing significantly changing proteins at 1 uM. iPI3Kd shows a few proteins significantly changing at all conc. iNFkB shows several proteins significantly changing at 10 uM, and even more proteins changing at 100 uM.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_bas_Ly8_conc}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_PBS_O",
                        "heat_iSYK0.1_O", "heat_iSYK1_O", "heat_iSYK10_O", "heat_iSYK100_O",
                        "heat_iBTK0.1_O", "heat_iBTK1_O", "heat_iBTK10_O", "heat_iBTK100_O",
                        "heat_iPI3Kd0.1_O", "heat_iPI3Kd1_O", "heat_iPI3Kd10_O", "heat_iPI3Kd100_O",
                        "heat_iNFkB0.1_O", "heat_iNFkB1_O", "heat_iNFkB10_O", "heat_iNFkB100_O")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_100uM_O <- list(heat_iSYK100_O, heat_iBTK100_O, heat_iPI3Kd100_O, heat_iNFkB100_O) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("inhib_new", "inhib_conc")]) %>%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_bas_Ly8_conc_100, fig.width=14, fig.height=8}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# Get the rld transformed data for the heatmap and scale per row
data_100uM_O <- t(scale(t(assay(rld)[unique(c(heat_100uM_O$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_100uM_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)"
)
```

```{r fig_heatmap_bas_Ly8_conc, fig.width=14, fig.height=8}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM iNFkB vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iNFkB_O <- t(scale(t(assay(rld)[unique(c(heat_iNFkB100_O$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_iNFkB_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iNFkB100_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins in 100 uM iNFkB (OCI-Ly8 cells w/o stim)"
)

# 100 uM iSYK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iSYK_O <- t(scale(t(assay(rld)[unique(c(heat_iSYK100_O$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iSYK_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iSYK100_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins in 100 uM iSYK (OCI-Ly8 cells w/o stim)"
)

# 100 uM iBTK vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_iBTK_O <- t(scale(t(assay(rld)[unique(c(heat_iBTK100_O$target_nospace)), rownames(data_heatmap)])))

ComplexHeatmap::pheatmap(
  data_iBTK_O,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_iBTK100_O)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins in 100 uM iBTK (OCI-Ly8 cells w/o stim)"
)
```
***Figure:** Heatmap of all significant results in the 100 uM inhibitor vs DMSO ctrl comparisons (no threshold for fold change).*


#### Cell:conc

DESeq2 analysis parameters:

-   4 inhibitors with all conc, PBS stimulus, both cell lines (144 samples)

-   Compensated counts (>= 1)

-   Model design: cell_line (ref: HBL1) + inhib_text_new (ref: DMSO ctrl) + cell_line:inhib_text_new

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_bas_cell_conc}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(stimulus == "PBS" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ cell_line + inhib_text_new + cell_line:inhib_text_new

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhib_new <- relevel(dds$inhib_new, ref = "DMSO_ctrl")
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
dds$cell_line <- relevel(dds$cell_line, ref = "HBL1")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_bas_cell_conc}
# Create results for all DESeq_comparisons
results_cell_icc <- results(dds, name = "cell_line_OCI.Ly8_vs_HBL1", alpha = 0.05)

results_iSYK0.1_cc <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_cc <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_cc <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_cc <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_cc <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_cc <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_cc <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_cc <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_cc <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_cc <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_cc <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_cc <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_cc <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_cc <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_cc <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_cc <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

results_cell_iSYK_cc <- results(dds, name = "cell_lineOCI.Ly8.inhib_text_new100.uM.iSYK", alpha = 0.05)
results_cell_iBTK_cc <- results(dds, name = "cell_lineOCI.Ly8.inhib_text_new100.uM.iBTK", alpha = 0.05)
results_cell_iPI3Kd_cc <- results(dds, name = "cell_lineOCI.Ly8.inhib_text_new100.uM.iPI3Kd", alpha = 0.05)
results_cell_iNFkB_cc <- results(dds, name = "cell_lineOCI.Ly8.inhib_text_new100.uM.iNFkB", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_cell_icc, 
                     results_iSYK0.1_cc, results_iSYK1_cc, results_iSYK10_cc, results_iSYK100_cc,
                     results_iBTK0.1_cc, results_iBTK1_cc, results_iBTK10_cc, results_iBTK100_cc,
                     results_iPI3Kd0.1_cc, results_iPI3Kd1_cc, results_iPI3Kd10_cc, results_iPI3Kd100_cc,
                     results_iNFkB0.1_cc, results_iNFkB1_cc, results_iNFkB10_cc, results_iNFkB100_cc,
                     results_cell_iSYK_cc, results_cell_iBTK_cc, results_cell_iPI3Kd_cc, results_cell_iNFkB_cc)

# Results summary
summary(results_iNFkB100_cc, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_bas_cell_conc, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_cell_icc, alpha = signcutoff, main = "OCI-Ly8 vs HBL1")
# plotMA(results_iSYK100_O, alpha = signcutoff, main = "100 uM iSYK vs DMSO")
plotMA(results_iNFkB100_O, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
plotMA(results_cell_iNFkB_cc, alpha = signcutoff, main = "Effect of OCI-Ly8 on\n100 uM iNFkB vs DMSO")
```


##### Data transformation {.tabset}

```{r transformation_bas_cell_conc}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_bas_cell_conc, fig.width=6, fig.height=4}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 6, repr.plot.height = 4)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  # plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), fill = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_color_manual(values = colors_blue9[4:9], name = "Inhib conc") +
    scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    # scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    scale_shape_manual(values = c(16, 17), name = "Cell line") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is mainly the difference between the cell lines, and PC2 is mainly the difference between inhibitor concentrations (especially the 100 uM iNFkB and the rest).*

###### Variance

```{r fig_sd_bas_cell_conc, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_bas_cell_conc, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$cell_line, 
                                      list_datatransform[[data]]$inhib_conc_text,
                                      list_datatransform[[data]]$replicate,
                                      sep = " - ")
  # rownames(sampleDistMatrix) <- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_bas_cell_conc}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_cell_icc", 
                        "volc_iSYK0.1_cc", "volc_iSYK1_cc", "volc_iSYK10_cc", "volc_iSYK100_cc",
                        "volc_iBTK0.1_cc", "volc_iBTK1_cc", "volc_iBTK10_cc", "volc_iBTK100_cc",
                        "volc_iPI3Kd0.1_cc", "volc_iPI3Kd1_cc", "volc_iPI3Kd10_cc", "volc_iPI3Kd100_cc",
                        "volc_iNFkB0.1_cc", "volc_iNFkB1_cc", "volc_iNFkB10_cc", "volc_iNFkB100_cc",
                        "volc_cell_iSYK_cc", "volc_cell_iBTK_cc", "volc_cell_iPI3Kd_cc", "volc_cell_iNFkB_cc")

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_bas_cell_conc, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_cell_icc, 
                     volc_iSYK0.1_cc, volc_iSYK1_cc, volc_iSYK10_cc, volc_iSYK100_cc,
                     volc_iBTK0.1_cc, volc_iBTK1_cc, volc_iBTK10_cc, volc_iBTK100_cc,
                     volc_iPI3Kd0.1_cc, volc_iPI3Kd1_cc, volc_iPI3Kd10_cc, volc_iPI3Kd100_cc,
                     volc_iNFkB0.1_cc, volc_iNFkB1_cc, volc_iNFkB10_cc, volc_iNFkB100_cc,
                     volc_cell_iSYK_cc, volc_cell_iBTK_cc, volc_cell_iPI3Kd_cc, volc_cell_iNFkB_cc)
titles_volcano <- c("OCI-Ly8 vs HBL1 (all inhibitor together? w/o stim)", 
                        "0.1 uM iSYK vs DMSO ctrl (both cell lines w/o stim)", "1 uM iSYK vs DMSO ctrl (both cell lines w/o stim)", "10 uM iSYK vs DMSO ctrl (both cell lines w/o stim)", "100 uM iSYK vs DMSO ctrl (both cell lines w/o stim)",
                        "0.1 uM iBTK vs DMSO ctrl (both cell lines w/o stim)", "1 uM iBTK vs DMSO ctrl (both cell lines w/o stim)", "10 uM iBTK vs DMSO ctrl (both cell lines w/o stim)", "100 uM iBTK vs DMSO ctrl (both cell lines w/o stim)",
                        "0.1 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)", "1 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)", "10 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)", "100 uM iPI3Kd vs DMSO ctrl (both cell lines w/o stim)",
                        "0.1 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)", "1 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)", "10 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)", "100 uM iNFkB vs DMSO ctrl (both cell lines w/o stim)",
                        "Effect of OCI-Ly8 on 100 uM iSYK vs DMSO ctrl (w/o stim)", "Effect of OCI-Ly8 on 100 uM iBTK vs DMSO ctrl (w/o stim)", "Effect of OCI-Ly8 on 100 uM iPI3Kd vs DMSO ctrl (w/o stim)", "Effect of OCI-Ly8 on 100 uM iNFkB vs DMSO ctrl (w/o stim)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plot of DESeq2 model comparisons: each inhibitor conc vs DMSO ctrl and the effect of OCI-Ly8 vs HBL1 on inhibitor conc vs DMSo ctrl. iSYK shows 2 protein going down at 1 uM, and more at increasingly higher conc. iBTK starts showing significantly changing proteins at 1 uM. iPI3Kd shows almost no significantly changing proteins at all conc. iNFkB shows several proteins significantly changing at 1 uM, and more at increasingly higher conc. iSYK affects many proteins significantly less in OCI-Ly8 cells, whereas the other inhibitors show only a few proteins that are differently affected in OCI-Ly8 cells.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_bas_cell_conc}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_cell_icc", 
                        "heat_iSYK0.1_cc", "heat_iSYK1_cc", "heat_iSYK10_cc", "heat_iSYK100_cc",
                        "heat_iBTK0.1_cc", "heat_iBTK1_cc", "heat_iBTK10_cc", "heat_iBTK100_cc",
                        "heat_iPI3Kd0.1_cc", "heat_iPI3Kd1_cc", "heat_iPI3Kd10_cc", "heat_iPI3Kd100_cc",
                        "heat_iNFkB0.1_cc", "heat_iNFkB1_cc", "heat_iNFkB10_cc", "heat_iNFkB100_cc",
                        "heat_cell_iSYK_cc", "heat_cell_iBTK_cc", "heat_cell_iPI3Kd_cc", "heat_cell_iNFkB_cc")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_100uM_cc <- list(heat_iSYK100_cc, heat_iBTK100_cc, heat_iPI3Kd100_cc, heat_iNFkB100_cc) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  distinct()

# Combine all heatmap datasets of 100 uM inhibitor (without DESeq2 analysis values)
heat_cell_inhib_cc <- list(heat_cell_iSYK_cc, heat_cell_iBTK_cc, heat_cell_iPI3Kd_cc, heat_cell_iNFkB_cc) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("cell_line", "inhib_new", "inhib_conc")]) %>%
  arrange(cell_line, factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_bas_cell_conc, fig.width=22, fig.height=9}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc)), 
                       cell_line = colors_cell
)

# 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_100uM_cc <- t(scale(t(assay(rld)[unique(c(heat_100uM_cc$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_100uM_cc,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_cc)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significant proteins with 100 uM inhibitor (both cell lines w/o stim)"
)

# Effect of OCI-Ly8 on 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_cell_inhib_cc <- t(scale(t(assay(rld)[unique(c(heat_cell_inhib_cc$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_cell_inhib_cc,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = paste(data_heatmap$inhib_new, data_heatmap$cell_line),
  row_split = subset(heat_cell_inhib_cc)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  main = "Significantly different proteins in OCI-Ly8 with 100 uM inhibitor (w/o stim)"
)
```
***Figure:** Heatmap of all significant results in the 100 uM inhibitor vs DMSO comparison and the effect of OCI-Ly8 on 100 uM inhibitor vs DMSO ctrl comparison (no threshold for fold change).*


### Protein comparisons

Finally, we visualised the normalised true counts of all significant proteins in the OCI-Ly8 vs HBL1 comparison (80 total).

```{r sign_data_bas_inhib_prot}
# Create datasets with all relevant information on significantly changing results

# Datasets generated during analysis
# heat_xxx: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_xxx: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# List of all significantly changing proteins
sign_prot_100uM_H <- sort(unique(c(heat_100uM_H$target_nospace)))
sign_prot_100uM_O <- sort(unique(c(heat_100uM_O$target_nospace)))
sign_prot_100uM <- sort(unique(c(sign_prot_100uM_H, sign_prot_100uM_O)))
sign_prot_100uM_cc <- sort(unique(c(heat_100uM_cc$target_nospace)))
sign_prot_cell_cc <- sort(unique(c(heat_cell_inhib_cc$target_nospace)))

sign_prot_all <- sort(unique(c(sign_prot_100uM, sign_prot_cell_cc)))

# print(c("Significant proteins (OCI-ly8 vs HBL1 basal signaling):", sign_prot_cell))
```

```{r sign_data_bas_inhib_sign}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
# 100 uM inhibitor in HBL1
heat_list_H <- list(heat_iSYK100_H, heat_iBTK100_H, heat_iPI3Kd100_H, heat_iNFkB100_H)
heat_names <- c("sign_data_iSYK", "sign_data_iBTK", "sign_data_iPI3Kd", "sign_data_iNFkB")
heat_suffix <- c("_iSYK", "_iBTK", "_iPI3Kd", "_iNFkB")
for (heat_data in c(1:length(heat_list_H))) {
  # print(heat_data)
  alt_data <- heat_list_H[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_100uM_H) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("base_mean", "log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_100uM_H <- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# 100 uM inhibitor in OCI-Ly8
heat_list_O <- list(heat_iSYK100_O, heat_iBTK100_O, heat_iPI3Kd100_O, heat_iNFkB100_O)
heat_names <- c("sign_data_iSYK", "sign_data_iBTK", "sign_data_iPI3Kd", "sign_data_iNFkB")
heat_suffix <- c("_iSYK", "_iBTK", "_iPI3Kd", "_iNFkB")
for (heat_data in c(1:length(heat_list_O))) {
  # print(heat_data)
  alt_data <- heat_list_O[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_100uM_H) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("base_mean", "log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_100uM_O <- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of cell line on 100 uM inhibitor
heat_list_cc <- list(heat_cell_iSYK_cc, heat_cell_iBTK_cc, heat_cell_iPI3Kd_cc, heat_cell_iNFkB_cc)
heat_names <- c("sign_data_iSYK", "sign_data_iBTK", "sign_data_iPI3Kd", "sign_data_iNFkB")
heat_suffix <- c("_iSYK", "_iBTK", "_iPI3Kd", "_iNFkB")
for (heat_data in c(1:length(heat_list_cc))) {
  # print(heat_data)
  alt_data <- heat_list_cc[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_100uM_H) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("base_mean", "log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_cell_cc <- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)
```

```{r sign_data_bas_inhib_well}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
# 100 uM inhibitor vs DMSO ctrl
alt_data_100uM_H <- data.frame(data_100uM_H) %>% 
  mutate(target_nospace = rownames(data_100uM_H)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_HBL1", 2:c(ncol(data_100uM_H)+1))
alt_data_100uM_O <- data.frame(data_100uM_O) %>% 
  mutate(target_nospace = rownames(data_100uM_O)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_Ly8", 2:c(ncol(data_100uM_O)+1))
clean_data_well_100uM <- full_join(alt_data_100uM_H, alt_data_100uM_O) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_100uM) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_HBL1, heatmap_Ly8) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

# Effect of cell line on 100 uM inhibitor
clean_data_well_cell_cc <- data.frame(data_cell_inhib_cc) %>% 
  mutate(target_nospace = rownames(data_cell_inhib_cc)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value", 2:c(ncol(data_cell_inhib_cc)+1)) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_cell_cc) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

# All signifocant proteins for plotting
clean_data_well <- counts_dtbl %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_all & stimulus == "PBS") %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)
```

```{r sign_data_bas_inhib_cond}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with mean values per condition (replicates combined)
# 100 uM inhibitor vs DMSO ctrl
clean_data_mean_100uM <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_100uM & stimulus == "PBS") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# Effect of cell line on 100 uM inhibitor
clean_data_mean_cell_cc <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_cell_cc & stimulus == "PBS") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# All signifocant proteins for plotting
clean_data_mean <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_all & stimulus == "PBS") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)
```

```{r fig_counts_bas_inhib_all, fig.width=15, fig.height=48}
plot <- ggplot(clean_data_well, 
               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                   counts_norm, 
                   color = cell_line, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_cell, name = "Cell line") + 
  labs(x = "", y = "Normalized counts") +
  textsize_small

print(plot)
```

***Figure:** Normalised counts for all significantly changing proteins. For some proteins the effect size is quite small.*

```{r fig_counts_bas_inhib, fig.width=12, fig.height=5}
options(repr.plot.width = 5, repr.plot.height = 5)

# Re-redesigned fig
filt_clean_data_mean <- clean_data_mean %>%
  dplyr::group_by(cell_line, inhibitor, inhib_conc, target_nospace) %>%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_all) {
  plot <- ggplot() +
    geom_jitter(data = subset(clean_data_well, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
                    counts_norm, 
                    color = inhibitor, 
                    # shape = cell_line
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = inhibitor,
                      # color = inhibitor
                      ),
                  width = 0.5, size = 0.5, color = "black") + #
    # facet_wrap(vars(factor(target_nospace))) +
    # facet_wrap(vars(factor(inhibitor, levels = inhib_DS108)), ncol = 1) +
    facet_grid(vars(factor(cell_line, levels = cells_DS108)), vars(factor(inhibitor, levels = inhib_DS108))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_inhib, name = "Inhibitors") + 
    labs(x = "", y = "Normalized counts", title = protein) +
    theme_bw() +
    theme(legend.position = "none", panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}

# for(protein in sign_prot_all[68]) {
#   plot <- ggplot() +
#     geom_jitter(data = subset(clean_data_well, target_nospace == protein), 
#                 aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
#                     counts_norm, 
#                     color = cell_line, 
#                     shape = cell_line), 
#                 alpha = 1, size = 1.5, width = 0.1) +
#     # geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
#     #               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
#     #                   ymin = mean_counts_norm,
#     #                   ymax = mean_counts_norm,
#     #                   group = cell_line, 
#     #                   color = cell_line),
#     #               width = 0.5, size = 1) + #, color = "black"
#     geom_line(data = subset(filt_clean_data_mean, target_nospace == protein),
#               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
#                   mean_counts_norm, 
#                   group = cell_line, 
#                   color = cell_line)) +
#     # facet_wrap(vars(factor(target_nospace))) +
#     facet_grid(vars(factor(inhibitor, levels = inhib_DS108))) +
#     # facet_wrap(vars(factor(cell_line, levels = cells_DS108)), ncol = 1) +
#     # scale_y_log10() +
#     scale_color_manual(values = colors_cell, name = "Cell line") + 
#     labs(x = "", y = "Normalized counts", title = protein) +
#     theme_bw() +
#     theme(legend.position = "none", panel.grid.major.x = element_blank()) +
#     textsize_small
#   
#   print(plot)
# }
```


### Save data

Save all information and datasets as .csv file

100 uM inhibitor vs DMSO ctrl comparisons (done separately for each cell line):

-   100uM_sign_proteins.txt: List of significantly changing proteins in the 100 uM inhibitor vs DMSO ctrl comparisons (in at least one of the cell lines)

- 100uM_HBL1_sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis (HBL1 samples only)

- 100uM_Ly8_sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis (OCI-Ly8 samples only)

- 100uM_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins (in at least one of the cell lines)

- 100uM_sign_condition.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins (in at least one of the cell lines)


Effect of cell line on 100 uM inhibitor vs DMSO ctrl:

- cell_100uM_sign_proteins.txt: List of significantly different proteins in OCI-Ly8 vs HBL1 when looking at the 100 uM inhibitor vs DMSO ctrl comparisons

-   cell_100uM_sign_DESeq2data.csv: Dataset with significantly different proteins, and log2FC + pvalues from DESeq2 analysis

-   cell_100uM_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   cell_100uM_sign_condition.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins

```{r save_bas_inhib}
# List of significantly changing proteins: sign_prot_cell
write(sign_prot_100uM, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_sign_proteins.txt")
write(sign_prot_cell_cc, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_proteins.txt")

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_100uM_H, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_HBL1_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_100uM_O, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_Ly8_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_cell_cc, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_DESeq2data.csv", row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_100uM, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_sign_sample.csv", row.names = F)
write.csv(clean_data_well_cell_cc, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_sample.csv", row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_100uM, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/100uM_sign_condition.csv", row.names = F)
write.csv(clean_data_mean_cell_cc, file = "output/DS108_StimInhibIDseq/IDseq_sign/basal_inhibitor/cell_100uM_sign_condition.csv", row.names = F)
```

```{r clear_bas_inhib}
# Remove unnecessary files to clear up memory
rm(list = ls(pattern = "results_"))
rm(list = ls(pattern = "volc_"))
rm(list = ls(pattern = "heat_"))
rm(list = ls(pattern = "list_"))
rm(list = ls(pattern = "names_"))
rm(list = ls(pattern = "titles_"))
rm(list = ls(pattern = "clean_"))
rm(list = ls(pattern = "sign_"))
gc()
```



## Activated cell/signaling state

Now, we will look at the activated signaling levels in both HBL1 and OCI-Ly8 cells (without any inhibitor/DMSO, but after 20 min incubation with either PBS or aIg+H2O2). We can investigate the effect of activation and differences between the cell lines.


### DEseq2 {.tabset}

<https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html>

#### Interaction

DESeq2 analysis parameters:

-   0 uM inhibitor (PBS ctrls), PBS and aIg+H2O2 stimulus, both cell lines samples (48 samples)

-   Compensated counts (>= 1)

-   Model design: cell_line (ref: HBL1) + stimulus (ref: PBS) + cell_line:stimulus

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_act_int}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
cts <- counts_dtbl  %>%
  filter(inhib_conc == "0 uM (PBS)" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(meta_wellID) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         stimulus = factor(stimulus, levels = stim_DS108), 
         cell_inhib = paste(cell_line, inhibitor)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign <- ~ cell_line + stimulus + cell_line:stimulus

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_line <- relevel(dds$cell_line, ref = "HBL1")
dds$stimulus <- relevel(dds$stimulus, ref = "PBS")
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$cell_inhib <- relevel(dds$cell_inhib, ref = "HBL1 iSYK")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_act_int}
# Create results for all DESeq_comparisons
results_cell <- results(dds, name = "cell_line_OCI.Ly8_vs_HBL1", alpha = 0.05)
results_stim_i <- results(dds, name = "stimulus_aIg.H2O2_vs_PBS", alpha = 0.05)
results_cell_stim <- results(dds, name = "cell_lineOCI.Ly8.stimulusaIg.H2O2", alpha = 0.05)

list_results <- list(results_cell, results_stim_i, results_cell_stim)

# Results summary
summary(results_cell, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_act_int, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_cell, alpha = signcutoff, main = "OCI-Ly8 vs HBL1")
plotMA(results_stim_i, alpha = signcutoff, main = "aIg+H2O2 vs PBS")
plotMA(results_cell_stim, alpha = signcutoff, main = "effect of OCI-Ly8 cell line\non aIg+H2O2 vs PBS")
```


##### Data transformation {.tabset}

```{r transformation_act_int}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_act_int, fig.width=4, fig.height=3}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_conc"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhib_conc) +
    scale_color_manual(values = colors_stim, name = "Stimulus") +
    scale_fill_manual(values = colors_stim, name = "Stimulus") +
    scale_shape_manual(values = c(16, 17), name = "Cell line") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 shows a clear separation on stimulus, and PC2 separates the cell lines. The different types of data transformation also give similar results.*

###### Variance

```{r fig_sd_act_int, fig.width=7, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Norm (log2(n+1))") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_act_int, fig.width=5, fig.height=7}
options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$cell_line,
                                      list_datatransform[[data]]$stimulus,
                                      # list_datatransform[[data]]$inhibitor,
                                      list_datatransform[[data]]$replicate,
                                      sep = " - ")
  # rownames(sampleDistMatrix) <- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_act_int}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_cell", "volc_stim_i", "volc_cell_stim")

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_act_int, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_cell, volc_stim_i, volc_cell_stim)
titles_volcano <- c("OCI-Ly8 vs HBL1 (basal + activated signaling w/o inhibitor)", 
                    "aIg+H2O2 vs PBS (both cell lines, w/o inhibitor)", 
                    "Effect of OCI-Ly8 (vs HBL1) on aIg+H2O2 vs PBS (w/o inhibitor)")
# titles_volcano <- names_results_volc

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plot of DESeq2 model comparisons.*

*As also seen in the basal signaling comparison, several (phospho)proteins are significantly different in the OCI-Ly8 cells compared to HBL1 cells. As expected, upon aIg+H2O2 stimulation, many (phospho)proteins are increased (both cell line together). Interestingly, many proteins are significantly less activated in OCI-Ly8 cells than in HBL1 cells, but a few proteins are activated more in OCI0-Ly8 cells.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_act_int}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_cell", "heat_stim_i", "heat_cell_stim")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("cell_line", "stimulus")]) %>%
  arrange(cell_line, factor(stimulus, levels = stim_DS108))
```

```{r fig_heatmap_act_int, fig.width=15, fig.height=15}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# aIg+H2O2 vs PBS
heat_stim_i$marker_pathway[is.na(heat_stim_i$marker_pathway)] <- "Other"

ann_colors_man <- list(cell_line = colors_cell,
                       stimulus = colors_stim,
                       # modification = c(phospho = "#f768a1", no = "grey", cleaved = "#fff7bc"),
                       marker_pathway = setNames(colors_light12[1:length(unique(heat_stim_i$marker_pathway))], unique(heat_stim_i$marker_pathway))
)

# Get the rld transformed data for the heatmap and scale per row
data_stim_i <- t(scale(t(assay(rld)[unique(c(heat_stim_i$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_stim_i,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  annotation_row = heat_stim_i[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$cell_line,
  row_split = subset(heat_stim_i)$modification,
  cellwidth = 10,
  cellheight = 10,
  fontsize = 8,
  border_color = NA,
  # main = "scaled per row, ComplexHeatmap"
)
```

***Figure:** Heatmap of all significant results in the aIg+H2O2 vs PBS comparison (no threshold for fold change).*

*Focussing on the phospho-proteins: Many upstream BCR proteins are activated in the HBL1 cells and to a lesser extend in the OCI-Ly8 cells. Some MAPK-JNK signaling proteins are activated more in the OCI-Ly8 cells. In addition, several JAK-STAT, ERK, and cell cylcle phospho-proteins are decreased upon stimulation, in both cell lines.*


#### Separate comp

DESeq2 analysis parameters:

-   0 uM inhibitor (PBS ctrls), PBS and aIg+H2O2 stimulus, both cell lines samples (48 samples)

-   Compensated counts (>= 1)

-   Model design: cell_stim (ref: HBL1 PBS)

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_act_sep}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns
counts_dtbl <- counts_dtbl %>%
  mutate(cell_stim = paste(cell_line, stimulus))

cts <- counts_dtbl  %>%
  filter(inhib_conc == "0 uM (PBS)" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, cell_stim)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         stimulus = factor(stimulus, levels = stim_DS108), 
         cell_inhib = paste(cell_line, inhibitor)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
modeldesign <- ~ cell_stim

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
dds$cell_stim <- relevel(dds$cell_stim, ref = "HBL1 PBS")
# dds$cell_inhib <- relevel(dds$cell_inhib, ref = "HBL1 iSYK")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_act_sep}
# Create results for all DESeq_comparisons
results_HBL1 <- results(dds, name = "cell_stim_HBL1.aIg.H2O2_vs_HBL1.PBS", alpha = 0.05)
results_Ly8 <- results(dds, contrast = c("cell_stim", "OCI-Ly8 aIg+H2O2", "OCI-Ly8 PBS"), alpha = 0.05)

results_PBS <- results(dds, name = "cell_stim_OCI.Ly8.PBS_vs_HBL1.PBS", alpha = 0.05)
results_stim <- results(dds, contrast = c("cell_stim", "OCI-Ly8 aIg+H2O2", "HBL1 aIg+H2O2"), alpha = 0.05)

list_results <- list(results_HBL1, results_Ly8, results_PBS, results_stim)

# Results summary
summary(results_HBL1, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_act_sep, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_HBL1, alpha = signcutoff, main = "HBL1:\naIg+H2O2 vs PBS")
plotMA(results_Ly8, alpha = signcutoff, main = "OCI-Ly8:\naIg+H2O2 vs PBS")
plotMA(results_stim, alpha = signcutoff, main = "aIg+H2O2:\nOCI-Ly8 vs HBL1")
```

##### Data transformation {.tabset}

```{r transformation_act_sep}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_act_sep, fig.width=4, fig.height=3}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 4, repr.plot.height = 3)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_conc"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(cell_line, levels = cells_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhib_conc) +
    scale_color_manual(values = colors_stim, name = "Stimulus") +
    scale_fill_manual(values = colors_stim, name = "Stimulus") +
    scale_shape_manual(values = c(16, 17), name = "Cell line") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 shows a clear separation on stimulus, and PC2 separates the cell lines. The different types of data transformation also give similar results.*

###### Variance

```{r fig_sd_act_sep, fig.width=7, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Norm (log2(n+1))") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_act_sep, fig.width=5, fig.height=7}
options(repr.plot.width = 5, repr.plot.height = 5)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$cell_line,
                                      list_datatransform[[data]]$stimulus,
                                      # list_datatransform[[data]]$inhibitor,
                                      list_datatransform[[data]]$replicate,
                                      sep = " - ")
  # rownames(sampleDistMatrix) <- list_datatransform[[data]]$cell_inhib
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```

##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_act_sep}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_HBL1", "volc_Ly8", "volc_PBS", "volc_stim")

for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_act_sep, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_HBL1, volc_Ly8, volc_PBS, volc_stim)
titles_volcano <- c("HBL1: aIg+H2O2 vs PBS (w/o inhibitor)",
                    "OCI-Ly8: aIg+H2O2 vs PBS (w/o inhibitor)",
                    "OCI-Ly8 vs HBL1 (basal signaling w/o inhibitor)", 
                    "OCI-Ly8 vs HBL1 (activated signaling w/o inhibitor)")
# titles_volcano <- names_results_volc

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plot of DESeq2 model comparisons.* 

*Stimulation in HBL1 cells results in the activation of many phospho-proteins. Stimulation in OCI-Ly8 cells also results in the activation of phospho-proteins, but to a much lesser extend (less proteins and lower fold changes). Comparison of the activated signaling states of OCI-Ly8 and HBL1 reveals many (phospho)proteins with significantly lower levels in OCI-Ly8 cells, but also some with higher levels in OCI-Ly8.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_act_sep}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_HBL1", "heat_Ly8", "heat_PBS", "heat_stim")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("cell_line", "stimulus")]) %>%
  arrange(cell_line, factor(stimulus, levels = stim_DS108))
```

```{r fig_heatmap_act_sep, fig.width=15, fig.height=15}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

# aIg+H2O2 vs PBS in HBL1 cells
heat_HBL1$marker_pathway[is.na(heat_HBL1$marker_pathway)] <- "Other"

ann_colors_man <- list(cell_line = colors_cell,
                       stimulus = colors_stim,
                       # modification = c(phospho = "#f768a1", no = "grey", cleaved = "#fff7bc"),
                       marker_pathway = setNames(colors_light12[1:length(unique(heat_HBL1$marker_pathway))], unique(heat_HBL1$marker_pathway))
)

# Get the rld transformed data for the heatmap and scale per row
data_HBL1 <- t(scale(t(assay(rld)[unique(c(heat_HBL1$target_nospace)), rownames(data_heatmap)])))
data_Ly8 <- t(scale(t(assay(rld)[unique(c(heat_Ly8$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

ComplexHeatmap::pheatmap(
  data_HBL1,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  annotation_row = heat_HBL1[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$cell_line,
  row_split = subset(heat_HBL1)$modification,
  cellwidth = 10,
  cellheight = 10,
  fontsize = 8,
  border_color = NA,
  # main = "scaled per row, ComplexHeatmap"
)
```

***Figure:** Heatmap of all significant results in the aIg+H2O2 vs PBS comparison for HBL1 cells (no threshold for fold change).*

*Focussing on the phospho-proteins: Many upstream BCR proteins are activated in the HBL1 cells and to a lesser extend in the OCI-Ly8 cells. Some MAPK-JNK signaling proteins are activated more in the OCI-Ly8 cells. In addition, several JAK-STAT, ERK, and cell cylcle phospho-proteins are decreased upon stimulation, in both cell lines.*


### Protein comparisons

Finally, we visualised the normalised true counts of all significant proteins in the aIg+H2O2 vs PBS comparison (85 in total).

```{r sign_data_act_int}
# Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins (for the HBL1 cells and the OCI-Ly8 cells)
sign_prot_cell_int <- sort(unique(c(heat_stim_i$target_nospace)))
# sign_prot_cell_sep <- sort(unique(c(heat_HBL1$target_nospace, heat_Ly8$target_nospace)))
# print(c("Significant proteins (aIg+H2O2 vs PBS in both HBL1 and OCI-Ly8):", sign_prot_cell_int))

# Datasets generated during analysis
# heat_stim_i: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_stim_i: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
clean_data_well_int <- data.frame(data_stim_i) %>% 
  mutate(target_nospace = rownames(data_stim_i)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value", 2:c(ncol(data_stim_i)+1)) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_cell_int) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, cell_stim, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

prot_labels_int <- sort(setNames(unique(pull(clean_data_well_int, target)), unique(pull(clean_data_well_int, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
clean_data_sign_int <- heat_stim_i %>%
  filter(target_nospace %in% sign_prot_cell_int) %>%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean = baseMean, 
                log2FC = log2FoldChange, 
                log2FC_se = lfcSE)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean_int <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_cell_int & inhib_conc == "0 uM (PBS)") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)
```

```{r sign_data_act_sep}
# Create a dataset with all relevant information on significantly changing results
# List of all significantly changing proteins (for the HBL1 cells and the OCI-Ly8 cells)
# sign_prot_cell_int <- sort(unique(c(heat_stim_i$target_nospace)))
sign_prot_cell_sep <- sort(unique(c(heat_HBL1$target_nospace, heat_Ly8$target_nospace)))
# print(c("Significant proteins (aIg+H2O2 vs PBS in HBL1 and OCI-Ly8 separately):", sign_prot_cell_sep))

# Datasets generated during analysis
# heat_HBL1: DESeq2 data of sign prot for HBL1 (repl combined), log2FC and pvalues per protein
# heat_Ly8: DESeq2 data of sign prot for OCI-Ly8 (repl combined), log2FC and pvalues per protein
# data_HBL1: matrix of data for heatmap for HBL1, well (col) x prot (row) (rld transformed, scaled per row)
# data_Ly8: matrix of data for heatmap for OCI-Ly8, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
heat_data_HBL1 <- data.frame(data_HBL1) %>% 
  mutate(target_nospace = rownames(data_HBL1)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value_HBL1", 2:c(ncol(data_HBL1)+1))
heat_data_Ly8 <- data.frame(data_Ly8) %>% 
  mutate(target_nospace = rownames(data_Ly8)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value_Ly8", 2:c(ncol(data_Ly8)+1))

clean_data_well_sep <- full_join(heat_data_HBL1, heat_data_Ly8) %>% 
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_cell_sep) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, cell_stim, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value_HBL1, heatmap_value_Ly8) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap_HBL1 = heatmap_value_HBL1, 
                heatmap_Ly8 = heatmap_value_Ly8)

prot_labels_sep <- sort(setNames(unique(pull(clean_data_well_sep, target)), unique(pull(clean_data_well_sep, target_nospace))))

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
sign_data_HBL1 <- heat_HBL1 %>%
  filter(target_nospace %in% sign_prot_cell_sep) %>%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean_HBL1 = baseMean, 
                log2FC_HBL1 = log2FoldChange, 
                log2FC_se_HBL1 = lfcSE, 
                stat_HBL1 = stat, 
                pvalue_HBL1 = pvalue,
                padj_HBL1 = padj, 
                diff_express_HBL1 = diff_express)
sign_data_Ly8 <- heat_Ly8 %>%
  filter(target_nospace %in% sign_prot_cell_sep) %>%
  dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
         baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                base_mean_Ly8 = baseMean, 
                log2FC_Ly8 = log2FoldChange, 
                log2FC_se_Ly8 = lfcSE, 
                stat_Ly8 = stat, 
                pvalue_Ly8 = pvalue,
                padj_Ly8 = padj,
                diff_express_Ly8 = diff_express)
clean_data_sign_sep <- full_join(sign_data_HBL1, sign_data_Ly8)

# Make a tidy dataset with mean values per condition (replicates combined)
clean_data_mean_sep <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_cell_sep & inhib_conc == "0 uM (PBS)") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)
```

```{r fig_counts_act_all, fig.width=15, fig.height=51}
# I visualised all significant proteins from the separate analysis (aIg+H2O2 vs PBS for the HBL1 cells and OCI-Ly8 cells separately), as this were more proteins, and all proteins in the interaction analysis also popped up in the separate analysis

plot <- ggplot(clean_data_well_sep, 
               aes(factor(cell_line, levels = cells_DS108),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  scale_y_log10() +
  scale_color_manual(values = colors_stim, name = "Stimulus") + 
  labs(x = "", y = "Normalized counts (log10 scale)") +
  textsize_small

print(plot)
```

***Figure:** Normalised counts for all significantly changing proteins. For some proteins the effect size is quite small.*

```{r fig_counts_act, fig.width=2.5, fig.height=2.5}
options(repr.plot.width = 3.5, repr.plot.height = 2.5)

# Re-redesigned fig
filt_clean_data_mean <- clean_data_mean_sep %>%
  dplyr::group_by(cell_line, stimulus, target_nospace) %>%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_cell_sep) {
  plot <- ggplot() +
    geom_jitter(data = subset(clean_data_well_sep, target_nospace == protein), 
                aes(factor(cell_line, levels = cells_DS108), counts_norm, color = stimulus), 
                alpha = 1, size = 1.5, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean, target_nospace == protein),
                  aes(factor(cell_line, levels = cells_DS108),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = cell_line, 
                      color = stimulus),
                  width = 0.5, size = 1) +
    facet_wrap(vars(factor(target))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = "Stimulus") + 
    labs(x = "", y = "Normalized counts") +
    theme_bw() +
    theme(legend.position = "none", panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}
```


### Save data

Save all information and datasets as .csv file

Interaction analysis for both cell lines together:

-   int_sign_proteins.txt: List of significantly changing proteins

-   int_sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis

-   int_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   int_sign_mean.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins


Separate analysis for each cell line:

-   sep_sign_proteins.txt: List of significantly changing proteins

-   sep_sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis

-   sep_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   sep_sign_mean.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins

```{r save_act}
# List of significantly changing proteins: sign_prot_cell
write(sign_prot_cell_int, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_proteins.txt")
write(sign_prot_cell_sep, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_proteins.txt")

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_int, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_sep, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_DESeq2data.csv", row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_int, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_sample.csv", row.names = F)
write.csv(clean_data_well_sep, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_sample.csv", row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_int, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/int_sign_condition.csv", row.names = F)
write.csv(clean_data_mean_sep, file = "output/DS108_StimInhibIDseq/IDseq_sign/active/sep_sign_condition.csv", row.names = F)
```

```{r clear_act}
# Remove unnecessary files to clear up memory
rm(list = ls(pattern = "results_"))
rm(list = ls(pattern = "volc_"))
rm(list = ls(pattern = "heat_"))
rm(list = ls(pattern = "list_"))
rm(list = ls(pattern = "names_"))
rm(list = ls(pattern = "titles_"))
rm(list = ls(pattern = "clean_"))
rm(list = ls(pattern = "sign_"))
gc()
```




## Inhibitor effects on activated signaling

Finally, we look at the effect of inhibitors on activated signaling state. We can investigate the effect of different inhibitors and concentrations on activation, in both cell lines separately.


### HBL1: DEseq2 {.tabset}

<https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html>

#### Inhib conc:stim

DESeq2 analysis parameters:

-   All HBL1 samples: 4 inhibitors, 6 conc, 2 stimuli (144 samples)

-   Compensated counts (>= 1)

-   Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref: PBS) + inhib_text_new:stimulus

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_act_HBL1_int}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(cell_line == "HBL1" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$inhib_conc <- relevel(dds$inhib_conc, ref = "0 uM (DMSO)")
dds$stimulus <- relevel(dds$stimulus, ref = "PBS")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_act_HBL1_int}
# Create results for all DESeq_comparisons
results_stim_Hi <- results(dds, name = "stimulus_aIg.H2O2_vs_PBS", alpha = 0.05)

results_PBS_Hi <- results(dds, name = "inhib_text_new_PBS.ctrl_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK0.1_Hi <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_Hi <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_Hi <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_Hi <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_Hi <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_Hi <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_Hi <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_Hi <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_Hi <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_Hi <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_Hi <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_Hi <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_Hi <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_Hi <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_Hi <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_Hi <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK100_stim_Hi <- results(dds, name = "inhib_text_new100.uM.iSYK.stimulusaIg.H2O2", alpha = 0.05)
results_iBTK100_stim_Hi <- results(dds, name = "inhib_text_new100.uM.iBTK.stimulusaIg.H2O2", alpha = 0.05)
results_iPI3Kd100_stim_Hi <- results(dds, name = "inhib_text_new100.uM.iPI3Kd.stimulusaIg.H2O2", alpha = 0.05)
results_iNFkB100_stim_Hi <- results(dds, name = "inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_stim_Hi, results_PBS_Hi, 
                     results_iSYK0.1_Hi, results_iSYK1_Hi, results_iSYK10_Hi, results_iSYK100_Hi, 
                     results_iBTK0.1_Hi, results_iBTK1_Hi, results_iBTK10_Hi, results_iBTK100_Hi, 
                     results_iPI3Kd0.1_Hi, results_iPI3Kd1_Hi, results_iPI3Kd10_Hi, results_iPI3Kd100_Hi, 
                     results_iNFkB0.1_Hi, results_iNFkB1_Hi, results_iNFkB10_Hi, results_iNFkB100_Hi, 
                     results_iSYK100_stim_Hi, results_iBTK100_stim_Hi, results_iPI3Kd100_stim_Hi, results_iNFkB100_stim_Hi)

# Results summary
summary(results_stim_Hi, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_act_HBL1_int, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_stim_Hi, alpha = signcutoff, main = "aIg+H2O2 vs PBS")
plotMA(results_iNFkB100_Hi, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
plotMA(results_iNFkB100_stim_Hi, alpha = signcutoff, main = "Effect of 100 uM iNFkB on\n aIg+H2O2 vs PBS")
```


##### Data transformation {.tabset}

```{r transformation_act_HBL1_int}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_act_HBL1_int, fig.width=10, fig.height=4}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    # scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_stim, name = "Stimulus") +
    scale_fill_manual(values = colors_stim, name = "Stimulus") +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is mainly the difference between stimulus, but also shows the effect of increasing inhibitor concentration (as the aIg+H2O2 samples with high inhibitor conc are closer to the PBS samples), and PC2 is mainly the difference between 100 uM iNFkB and all other samples.*

###### Variance

```{r fig_sd_act_HBL1_int, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_act_HBL1_int, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = " - ")
  rownames(sampleDistMatrix) <- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_act_HBL1_int}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_stim_Hi", "volc_PBS_Hi", 
                        "volc_iSYK0.1_Hi", "volc_iSYK1_Hi", "volc_iSYK10_Hi", "volc_iSYK100_Hi", 
                        "volc_iBTK0.1_Hi", "volc_iBTK1_Hi", "volc_iBTK10_Hi", "volc_iBTK100_Hi", 
                        "volc_iPI3Kd0.1_Hi", "volc_iPI3Kd1_Hi", "volc_iPI3Kd10_Hi", "volc_iPI3Kd100_Hi", 
                        "volc_iNFkB0.1_Hi", "volc_iNFkB1_Hi", "volc_iNFkB10_Hi", "volc_iNFkB100_Hi", 
                        "volc_iSYK100_stim_Hi", "volc_iBTK100_stim_Hi", "volc_iPI3Kd100_stim_Hi", "volc_iNFkB100_stim_Hi")


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_act_HBL1_int, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_stim_Hi, volc_PBS_Hi, 
                          volc_iSYK0.1_Hi, volc_iSYK1_Hi, volc_iSYK10_Hi, volc_iSYK100_Hi, 
                          volc_iBTK0.1_Hi, volc_iBTK1_Hi, volc_iBTK10_Hi, volc_iBTK100_Hi, 
                          volc_iPI3Kd0.1_Hi, volc_iPI3Kd1_Hi, volc_iPI3Kd10_Hi, volc_iPI3Kd100_Hi, 
                          volc_iNFkB0.1_Hi, volc_iNFkB1_Hi, volc_iNFkB10_Hi, volc_iNFkB100_Hi, 
                          volc_iSYK100_stim_Hi, volc_iBTK100_stim_Hi, volc_iPI3Kd100_stim_Hi, volc_iNFkB100_stim_Hi)
titles_volcano <- c("aIg+H2O2 vs PBS (HBL1 w/ all inhibitors?)", "PBS ctrl vs DMSO ctrl (HBL1 w/ both stimuli)", 
                    "0.1 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)", "1 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)", "10 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)", "100 uM iSYK vs DMSO ctrl (HBL1 w/ both stimuli)", 
                    "0.1 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)", "1 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)", "10 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)", "100 uM iBTK vs DMSO ctrl (HBL1 w/ both stimuli)",
                    "0.1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)", "1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)", "10 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)", "100 uM iPI3Kd vs DMSO ctrl (HBL1 w/ both stimuli)",
                    "0.1 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)", "1 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)", "10 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)", "100 uM iNFkB vs DMSO ctrl (HBL1 w/ both stimuli)",
                    "Effect of 100 uM iSYK on aIg+H2O2 vs PBS (HBL1)", "Effect of 100 uM iBTK on aIg+H2O2 vs PBS (HBL1)", "Effect of 100 uM iPI3Kd on aIg+H2O2 vs PBS (HBL1)", "Effect of 100 uM iNFkB on aIg+H2O2 vs PBS (HBL1)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plots of DESeq2 model comparisons: aIg+H2O2 vs PBS, each inhibitor conc vs DMSO ctrl, and the effect of 100 uM inhibitor on aIg+H2O2 vs PBS (HBL1 cells only). Many proteins are activated upon aIg+H2O2 stimulation. Higher conc of inhibitor significantly affects more proteins, except iPI3Kd. And many proteins are significantly less activated in the presence of 100 uM inhibitor (again except iPI3Kd).*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_act_HBL1_int}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_stim_Hi", "heat_PBS_Hi", 
                        "heat_iSYK0.1_Hi", "heat_iSYK1_Hi", "heat_iSYK10_Hi", "heat_iSYK100_Hi", 
                        "heat_iBTK0.1_Hi", "heat_iBTK1_Hi", "heat_iBTK10_Hi", "heat_iBTK100_Hi", 
                        "heat_iPI3Kd0.1_Hi", "heat_iPI3Kd1_Hi", "heat_iPI3Kd10_Hi", "heat_iPI3Kd100_Hi", 
                        "heat_iNFkB0.1_Hi", "heat_iNFkB1_Hi", "heat_iNFkB10_Hi", "heat_iNFkB100_Hi", 
                        "heat_iSYK100_stim_Hi", "heat_iBTK100_stim_Hi", "heat_iPI3Kd100_stim_Hi", "heat_iNFkB100_stim_Hi")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Hi <- list(heat_iSYK100_Hi, heat_iBTK100_Hi, heat_iPI3Kd100_Hi, heat_iNFkB100_Hi) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()

# Combine all heatmap datasets of 100 uM inhibitor effect on aIg+H2O2 vs PBS (without DESeq2 analysis values)
heat_100_stim_Hi <- list(heat_iSYK100_stim_Hi, heat_iBTK100_stim_Hi, heat_iPI3Kd100_stim_Hi, heat_iNFkB100_stim_Hi) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("stimulus", "inhib_new", "inhib_conc")]) %>%
  arrange(factor(stimulus, levels = stim_DS108), factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_act_HBL1_int_100, fig.width=21, fig.height=8}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Hi <- t(scale(t(assay(rld)[unique(c(heat_100uM_Hi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Hi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Hi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = "Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)"
)) +
  labs(title = "100 uM inhibitor vs DMSO ctrl (HBL1 w/ both stimuli) (significant proteins)")

# 100 uM inhibitor effect on aIg+H2O2 vs PBS
# Get the rld transformed data for the heatmap and scale per row
data_100_stim_Hi <- t(scale(t(assay(rld)[unique(c(heat_100_stim_Hi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

as.ggplot(ComplexHeatmap::pheatmap(
  data_100_stim_Hi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100_stim_Hi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = "Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)"
)) +
  labs(title = "Effect of 100 uM inhibitor on aIg+H2O2 vs PBS (HBL1) (significant proteins)")
```

***Figure:** Heatmaps of the significant results in the 100 uM inhibitor vs DMSO ctrl comparisons and the effect of 100 uM inhibitor on aIg+H2O2 vs PBS comparisons (no threshold for fold change).*


#### Conc (aIg only)

DESeq2 analysis parameters:

-   All HBL1 samples: 4 inhibitors, 6 conc, aIg+H2O2 stimulus (74 samples)

-   Compensated counts (>= 1)

-   Model design: inhib_text_new (ref: DMSO ctrl)

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_act_HBL1_stim}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(cell_line == "HBL1" & stimulus == "aIg+H2O2" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ inhib_text_new

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$inhib_conc <- relevel(dds$inhib_conc, ref = "0 uM (DMSO)")
# dds$stimulus <- relevel(dds$stimulus, ref = "PBS")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_act_HBL1_stim}
# Create results for all DESeq_comparisons
results_PBS_Hs <- results(dds, name = "inhib_text_new_PBS.ctrl_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK0.1_Hs <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_Hs <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_Hs <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_Hs <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_Hs <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_Hs <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_Hs <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_Hs <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_Hs <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_Hs <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_Hs <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_Hs <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_Hs <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_Hs <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_Hs <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_Hs <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_PBS_Hs, 
                     results_iSYK0.1_Hs, results_iSYK1_Hs, results_iSYK10_Hs, results_iSYK100_Hs, 
                     results_iBTK0.1_Hs, results_iBTK1_Hs, results_iBTK10_Hs, results_iBTK100_Hs, 
                     results_iPI3Kd0.1_Hs, results_iPI3Kd1_Hs, results_iPI3Kd10_Hs, results_iPI3Kd100_Hs, 
                     results_iNFkB0.1_Hs, results_iNFkB1_Hs, results_iNFkB10_Hs, results_iNFkB100_Hs)

# Results summary
summary(results_iNFkB100_Hs, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_act_HBL1_stim, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_PBS_Hs, alpha = signcutoff, main = "PBS vs DMSO")
plotMA(results_iNFkB100_Hs, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
plotMA(results_iBTK100_Hs, alpha = signcutoff, main = "100 uM iBTK vs DMSO")
```


##### Data transformation {.tabset}

```{r transformation_act_HBL1_stim}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_act_HBL1_stim, fig.width=10, fig.height=4}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    # scale_color_manual(values = colors_stim, name = "Stimulus") +
    # scale_fill_manual(values = colors_stim, name = "Stimulus") +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is mainly the difference between inhibitor conc, and PC2 is mainly the difference between 100 uM iNFkB and all other samples.*

###### Variance

```{r fig_sd_act_HBL1_stim, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_act_HBL1_stim, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = " - ")
  rownames(sampleDistMatrix) <- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_act_HBL1_stim}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_PBS_Hs", 
                        "volc_iSYK0.1_Hs", "volc_iSYK1_Hs", "volc_iSYK10_Hs", "volc_iSYK100_Hs", 
                        "volc_iBTK0.1_Hs", "volc_iBTK1_Hs", "volc_iBTK10_Hs", "volc_iBTK100_Hs", 
                        "volc_iPI3Kd0.1_Hs", "volc_iPI3Kd1_Hs", "volc_iPI3Kd10_Hs", "volc_iPI3Kd100_Hs", 
                        "volc_iNFkB0.1_Hs", "volc_iNFkB1_Hs", "volc_iNFkB10_Hs", "volc_iNFkB100_Hs")


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_act_HBL1_stim, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_PBS_Hs, 
                          volc_iSYK0.1_Hs, volc_iSYK1_Hs, volc_iSYK10_Hs, volc_iSYK100_Hs, 
                          volc_iBTK0.1_Hs, volc_iBTK1_Hs, volc_iBTK10_Hs, volc_iBTK100_Hs, 
                          volc_iPI3Kd0.1_Hs, volc_iPI3Kd1_Hs, volc_iPI3Kd10_Hs, volc_iPI3Kd100_Hs, 
                          volc_iNFkB0.1_Hs, volc_iNFkB1_Hs, volc_iNFkB10_Hs, volc_iNFkB100_Hs)
titles_volcano <- c("PBS ctrl vs DMSO ctrl (HBL1 w/ stim)",
                    "0.1 uM iSYK vs DMSO ctrl (HBL1 w/ stim)", "1 uM iSYK vs DMSO ctrl (HBL1 w/ stim)", "10 uM iSYK vs DMSO ctrl (HBL1 w/ stim)", "100 uM iSYK vs DMSO ctrl (HBL1 w/ stim)",
                    "0.1 uM iBTK vs DMSO ctrl (HBL1 w/ stim)", "1 uM iBTK vs DMSO ctrl (HBL1 w/ stim)", "10 uM iBTK vs DMSO ctrl (HBL1 w/ stim)", "100 uM iBTK vs DMSO ctrl (HBL1 w/ stim)",
                    "0.1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)", "1 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)", "10 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)", "100 uM iPI3Kd vs DMSO ctrl (HBL1 w/ stim)",
                    "0.1 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)", "1 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)", "10 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)", "100 uM iNFkB vs DMSO ctrl (HBL1 w/ stim)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0)) + 
    coord_cartesian(xlim = c(-1.5, 1.5), ylim = c(0, 175))
  
  print(plot)
}
```

***Figure:** Volcano plots of DESeq2 model comparisons: each inhibitor conc vs DMSO ctrl (HBL1 w/ aIg+H2O2 samples only). Already at low conc inhibitor, many proteins are decreased compared to the DMSO ctrl, and in general the effect size increases with the inhibitor conc, except iPI3Kd.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_act_HBL1_stim}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_PBS_Hs", 
                        "heat_iSYK0.1_Hs", "heat_iSYK1_Hs", "heat_iSYK10_Hs", "heat_iSYK100_Hs", 
                        "heat_iBTK0.1_Hs", "heat_iBTK1_Hs", "heat_iBTK10_Hs", "heat_iBTK100_Hs", 
                        "heat_iPI3Kd0.1_Hs", "heat_iPI3Kd1_Hs", "heat_iPI3Kd10_Hs", "heat_iPI3Kd100_Hs", 
                        "heat_iNFkB0.1_Hs", "heat_iNFkB1_Hs", "heat_iNFkB10_Hs", "heat_iNFkB100_Hs")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Hs <- list(heat_iSYK100_Hs, heat_iBTK100_Hs, heat_iPI3Kd100_Hs, heat_iNFkB100_Hs) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("inhib_new", "inhib_conc")]) %>%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_act_HBL1_stim, fig.width=13, fig.height=10}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       # stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl (w/ stim)
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Hs <- t(scale(t(assay(rld)[unique(c(heat_100uM_Hs$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Hs,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Hs)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = "Significant proteins with 100 uM inhibitor (HBL1 cells w/o stim)"
)) +
  labs(title = "100 uM inhibitor vs DMSO ctrl (HBL1 w/ stim) (significant proteins)")
```

***Figure:** Heatmap of the significant results in the 100 uM inhibitor vs DMSO ctrl comparisons (no threshold for fold change).*



### HBL1: Protein comparisons

Finally, we visualised the normalised true counts of all proteins for which the aIg+H2O2 vs PBS comparison was significantly affected by 100 uM inhibitor (31 in total).

```{r sign_data_act_HBL1_prot}
# Create datasets with all relevant information on significantly changing results

# Datasets generated during analysis
# heat_xxx: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_xxx: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# List of all significantly changing proteins
sign_prot_Hi <- sort(unique(c(heat_100_stim_Hi$target_nospace)))

heat_iSYK_Hs <- list(heat_iSYK0.1_Hs, heat_iSYK1_Hs, heat_iSYK10_Hs, heat_iSYK100_Hs) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iSYK_Hs <- sort(unique(c(heat_iSYK_Hs$target_nospace)))

heat_iBTK_Hs <- list(heat_iBTK0.1_Hs, heat_iBTK1_Hs, heat_iBTK10_Hs, heat_iBTK100_Hs) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iBTK_Hs <- sort(unique(c(heat_iBTK_Hs$target_nospace)))

heat_iPI3Kd_Hs <- list(heat_iPI3Kd0.1_Hs, heat_iPI3Kd1_Hs, heat_iPI3Kd10_Hs, heat_iPI3Kd100_Hs) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iPI3Kd_Hs <- sort(unique(c(heat_iPI3Kd_Hs$target_nospace)))

heat_iNFkB_Hs <- list(heat_iNFkB0.1_Hs, heat_iNFkB1_Hs, heat_iNFkB10_Hs, heat_iNFkB100_Hs) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iNFkB_Hs <- sort(unique(c(heat_iNFkB_Hs$target_nospace)))

sign_prot_inhib_Hs <- sort(unique(c(sign_prot_iSYK_Hs, sign_prot_iBTK_Hs, sign_prot_iPI3Kd_Hs, sign_prot_iNFkB_Hs)))
sign_prot_all_H <- sort(unique(c(sign_prot_Hi, sign_prot_inhib_Hs)))

# print(c("Significant proteins (OCI-ly8 vs HBL1 basal signaling):", sign_prot_cell))
```

```{r sign_data_act_HBL1_sign}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
# Effect of 100 uM inhibitor on stimulation
heat_list_Hi <- list(heat_iSYK100_stim_Hi, heat_iBTK100_stim_Hi, heat_iPI3Kd100_stim_Hi, heat_iNFkB100_stim_Hi)
heat_names <- c("sign_data_iSYK", "sign_data_iBTK", "sign_data_iPI3Kd", "sign_data_iNFkB")
heat_suffix <- c("_iSYK", "_iBTK", "_iPI3Kd", "_iNFkB")
for (heat_data in c(1:length(heat_list_Hi))) {
  # print(heat_data)
  alt_data <- heat_list_Hi[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_Hi) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_Hi <- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iSYK conc on activated signaling
heat_list_iSYK_Hs <- list(heat_iSYK0.1_Hs, heat_iSYK1_Hs, heat_iSYK10_Hs, heat_iSYK100_Hs)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iSYK_Hs))) {
  # print(heat_data)
  alt_data <- heat_list_iSYK_Hs[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iSYK_Hs) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iSYK_Hs <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iBTK conc on activated signaling
heat_list_iBTK_Hs <- list(heat_iBTK0.1_Hs, heat_iBTK1_Hs, heat_iBTK10_Hs, heat_iBTK100_Hs)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iBTK_Hs))) {
  # print(heat_data)
  alt_data <- heat_list_iBTK_Hs[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iBTK_Hs) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iBTK_Hs <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iPI3Kd conc on activated signaling
heat_list_iPI3Kd_Hs <- list(heat_iPI3Kd0.1_Hs, heat_iPI3Kd1_Hs, heat_iPI3Kd10_Hs, heat_iPI3Kd100_Hs)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iPI3Kd_Hs))) {
  # print(heat_data)
  alt_data <- heat_list_iPI3Kd_Hs[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iPI3Kd_Hs) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iPI3Kd_Hs <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iNFkB conc on activated signaling
heat_list_iNFkB_Hs <- list(heat_iNFkB0.1_Hs, heat_iNFkB1_Hs, heat_iNFkB10_Hs, heat_iNFkB100_Hs)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iNFkB_Hs))) {
  # print(heat_data)
  alt_data <- heat_list_iNFkB_Hs[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iNFkB_Hs) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iNFkB_Hs <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)
```

```{r sign_data_act_HBL1_well}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
# Effect of 100 uM inhibitor on stimulation
clean_data_well_Hi <- data.frame(data_100_stim_Hi) %>% 
  mutate(target_nospace = rownames(data_100_stim_Hi)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value", 2:c(ncol(data_100_stim_Hi)+1)) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_Hi) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

# Effect of inhibitor conc on activated signaling
data_iSYK_Hs <- t(scale(t(assay(rld)[unique(c(sign_prot_iSYK_Hs)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iSYK")))])))
alt_data_iSYK_Hs <- data.frame(data_iSYK_Hs) %>% 
  mutate(target_nospace = rownames(data_iSYK_Hs)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iSYK", 2:c(ncol(data_iSYK_Hs)+1))
data_iBTK_Hs <- t(scale(t(assay(rld)[unique(c(sign_prot_iBTK_Hs)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iBTK")))])))
alt_data_iBTK_Hs <- data.frame(data_iBTK_Hs) %>% 
  mutate(target_nospace = rownames(data_iBTK_Hs)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iBTK", 2:c(ncol(data_iBTK_Hs)+1))
data_iPI3Kd_Hs <- t(scale(t(assay(rld)[unique(c(sign_prot_iPI3Kd_Hs)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iPI3Kd")))])))
alt_data_iPI3Kd_Hs <- data.frame(data_iPI3Kd_Hs) %>% 
  mutate(target_nospace = rownames(data_iPI3Kd_Hs)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iPI3Kd", 2:c(ncol(data_iPI3Kd_Hs)+1))
data_iNFkB_Hs <- t(scale(t(assay(rld)[unique(c(sign_prot_iNFkB_Hs)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iNFkB")))])))
alt_data_iNFkB_Hs <- data.frame(data_iNFkB_Hs) %>% 
  mutate(target_nospace = rownames(data_iNFkB_Hs)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iNFkB", 2:c(ncol(data_iNFkB_Hs)+1))

clean_data_well_inhib_Hs <- list(alt_data_iSYK_Hs, alt_data_iBTK_Hs, alt_data_iPI3Kd_Hs, alt_data_iNFkB_Hs) %>% 
  purrr::reduce(full_join) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_inhib_Hs) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_iSYK, heatmap_iBTK, heatmap_iPI3Kd, heatmap_iNFkB) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

# All significant proteins for plotting
clean_data_well_H <- counts_dtbl %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_Hi) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)
```

```{r sign_data_act_HBL1_cond}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with mean values per condition (replicates combined)
# Effect of 100 uM inhibitor on stimulation
clean_data_mean_Hi <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_Hi & cell_line == "HBL1") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# Effect of inhibitor conc on activated signaling
clean_data_mean_Hs <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_inhib_Hs & cell_line == "HBL1") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# All significant proteins for plotting
clean_data_mean_H <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_Hi) %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)
```

```{r fig_counts_act_HBL1_all, fig.width=15, fig.height=21}
plot <- ggplot(clean_data_well_Hi, 
               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim, name = "Stimulus") + 
  labs(x = "", y = "Normalized counts") +
  textsize_small

print(plot)
```

***Figure:** Normalised counts for all significantly changing proteins. For some proteins the effect size is quite small.*

```{r fig_counts_act_HBL1, fig.width=12, fig.height=5}
options(repr.plot.width = 5, repr.plot.height = 5)

# Re-redesigned fig
filt_clean_data_mean_H <- clean_data_mean_H %>%
  dplyr::group_by(cell_line, inhibitor, inhib_conc, stimulus, target_nospace) %>%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_Hi) {
  plot <- ggplot() +
    geom_jitter(data = subset(clean_data_well_H, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
                    counts_norm, 
                    color = stimulus, 
                    # shape = cell_line
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean_H, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = stimulus,
                      color = stimulus
                      ),
                  width = 0.5, size = 0.75) + #, color = "black"
    # facet_wrap(vars(factor(target_nospace))) +
    # facet_wrap(vars(factor(inhibitor, levels = inhib_DS108)), ncol = 1) +
    facet_grid(vars(factor(cell_line, levels = cells_DS108)), vars(factor(inhibitor, levels = inhib_DS108))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = "Stimulus") + 
    labs(x = "", y = "Normalized counts", title = protein) +
    theme_bw() +
    theme(legend.position = "none", panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}
```


### HBL1: Save data

Save all information and datasets as .csv file

Effect of 100 uM inhibitor on aIg+H2O2 vs PBS comparison (HBL1 cells only):

-   HBL1_100uM_stim_sign_proteins.txt: List of significantly affected proteins by 100 uM inhibitor in the aIg+H2O2 vs PBS comparison (by at least one inhibitor)

-   HBL1_100uM_stim_sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis

-   HBL1_100uM_stim_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   HBL1_100uM_stim_sign_condition.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins


Effect of inhibitor conc vs DMSO ctrl (for each inhibitor, HBL1 w/ stim only):
For each inhibitor (iSYK, iBTK, iPI3Kd, iNFkB):

-   HBL1_inhib_sign_proteins.txt: List of significantly different proteins in any of the inhibitor conc vs DMSO ctrl comparisons

-   HBL1_inhib_sign_DESeq2data.csv: Dataset with significantly different proteins, and log2FC + pvalues from DESeq2 analysis

-   HBL1_inhib_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   HBL1_inhib_sign_condition.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins

```{r save_act_HBL1}
# List of significantly changing proteins: sign_prot_cell
write(sign_prot_Hi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_proteins.txt")
write(sign_prot_iSYK_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iSYK_sign_proteins.txt")
write(sign_prot_iBTK_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iBTK_sign_proteins.txt")
write(sign_prot_iPI3Kd_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iPI3Kd_sign_proteins.txt")
write(sign_prot_iNFkB_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iNFkB_sign_proteins.txt")

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_Hi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iSYK_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iSYK_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iBTK_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iBTK_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iPI3Kd_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iPI3Kd_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iNFkB_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_iNFkB_sign_DESeq2data.csv", row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_Hi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_sample.csv", row.names = F)
write.csv(clean_data_well_inhib_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_inhib_sign_sample.csv", row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_Hi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_100uM_stim_sign_condition.csv", row.names = F)
write.csv(clean_data_mean_Hs, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/HBL1_inhib_sign_condition.csv", row.names = F)
```

```{r clear_act_HBL1, eval=F}
# Remove unnecessary files to clear up memory
rm(list = ls(pattern = "results_"))
rm(list = ls(pattern = "volc_"))
rm(list = ls(pattern = "heat_"))
rm(list = ls(pattern = "list_"))
rm(list = ls(pattern = "names_"))
rm(list = ls(pattern = "titles_"))
rm(list = ls(pattern = "clean_"))
rm(list = ls(pattern = "sign_"))
gc()
```


### OCI-Ly8: DEseq2 {.tabset}

<https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html>

#### Inhib conc:stim

DESeq2 analysis parameters:

-   All OCI-Ly8 samples: 4 inhibitors, 6 conc, 2 stimuli (144 samples)

-   Compensated counts (>= 1)

-   Model design: inhib_text_new (ref: DMSO ctrl) + stimulus (ref: PBS) + inhib_text_new:stimulus

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_act_Ly8_int}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(cell_line == "OCI-Ly8" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ inhib_text_new + stimulus + inhib_text_new:stimulus

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$inhib_conc <- relevel(dds$inhib_conc, ref = "0 uM (DMSO)")
dds$stimulus <- relevel(dds$stimulus, ref = "PBS")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_act_Ly8_int}
# Create results for all DESeq_comparisons
results_stim_Oi <- results(dds, name = "stimulus_aIg.H2O2_vs_PBS", alpha = 0.05)

results_PBS_Oi <- results(dds, name = "inhib_text_new_PBS.ctrl_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK0.1_Oi <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_Oi <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_Oi <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_Oi <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_Oi <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_Oi <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_Oi <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_Oi <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_Oi <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_Oi <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_Oi <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_Oi <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_Oi <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_Oi <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_Oi <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_Oi <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK100_stim_Oi <- results(dds, name = "inhib_text_new100.uM.iSYK.stimulusaIg.H2O2", alpha = 0.05)
results_iBTK100_stim_Oi <- results(dds, name = "inhib_text_new100.uM.iBTK.stimulusaIg.H2O2", alpha = 0.05)
results_iPI3Kd100_stim_Oi <- results(dds, name = "inhib_text_new100.uM.iPI3Kd.stimulusaIg.H2O2", alpha = 0.05)
results_iNFkB100_stim_Oi <- results(dds, name = "inhib_text_new100.uM.iNFkB.stimulusaIg.H2O2", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_stim_Oi, results_PBS_Oi, 
                     results_iSYK0.1_Oi, results_iSYK1_Oi, results_iSYK10_Oi, results_iSYK100_Oi, 
                     results_iBTK0.1_Oi, results_iBTK1_Oi, results_iBTK10_Oi, results_iBTK100_Oi, 
                     results_iPI3Kd0.1_Oi, results_iPI3Kd1_Oi, results_iPI3Kd10_Oi, results_iPI3Kd100_Oi, 
                     results_iNFkB0.1_Oi, results_iNFkB1_Oi, results_iNFkB10_Oi, results_iNFkB100_Oi, 
                     results_iSYK100_stim_Oi, results_iBTK100_stim_Oi, results_iPI3Kd100_stim_Oi, results_iNFkB100_stim_Oi)

# Results summary
summary(results_stim_Oi, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_act_Ly8_int, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_stim_Oi, alpha = signcutoff, main = "aIg+H2O2 vs PBS")
plotMA(results_iNFkB100_Oi, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
plotMA(results_iNFkB100_stim_Oi, alpha = signcutoff, main = "Effect of 100 uM iNFkB on\n aIg+H2O2 vs PBS")
```


##### Data transformation {.tabset}

```{r transformation_act_Ly8_int}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_act_Ly8_int, fig.width=10, fig.height=4}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(stimulus, levels = stim_DS108), fill = factor(stimulus, levels = stim_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    # scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_stim, name = "Stimulus") +
    scale_fill_manual(values = colors_stim, name = "Stimulus") +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is the difference between stimuli and the effect of increasing inhibitor concentration (as the aIg+H2O2 samples with high inhibitor conc are closer to the PBS samples), and PC2 is mainly the difference between 100 uM iNFkB and all other samples.*

###### Variance

```{r fig_sd_act_Ly8_int, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_act_Ly8_int, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = " - ")
  rownames(sampleDistMatrix) <- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_act_Ly8_int}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_stim_Oi", "volc_PBS_Oi", 
                        "volc_iSYK0.1_Oi", "volc_iSYK1_Oi", "volc_iSYK10_Oi", "volc_iSYK100_Oi", 
                        "volc_iBTK0.1_Oi", "volc_iBTK1_Oi", "volc_iBTK10_Oi", "volc_iBTK100_Oi", 
                        "volc_iPI3Kd0.1_Oi", "volc_iPI3Kd1_Oi", "volc_iPI3Kd10_Oi", "volc_iPI3Kd100_Oi", 
                        "volc_iNFkB0.1_Oi", "volc_iNFkB1_Oi", "volc_iNFkB10_Oi", "volc_iNFkB100_Oi", 
                        "volc_iSYK100_stim_Oi", "volc_iBTK100_stim_Oi", "volc_iPI3Kd100_stim_Oi", "volc_iNFkB100_stim_Oi")


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_act_Ly8_int, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_stim_Oi, volc_PBS_Oi, 
                          volc_iSYK0.1_Oi, volc_iSYK1_Oi, volc_iSYK10_Oi, volc_iSYK100_Oi, 
                          volc_iBTK0.1_Oi, volc_iBTK1_Oi, volc_iBTK10_Oi, volc_iBTK100_Oi, 
                          volc_iPI3Kd0.1_Oi, volc_iPI3Kd1_Oi, volc_iPI3Kd10_Oi, volc_iPI3Kd100_Oi, 
                          volc_iNFkB0.1_Oi, volc_iNFkB1_Oi, volc_iNFkB10_Oi, volc_iNFkB100_Oi, 
                          volc_iSYK100_stim_Oi, volc_iBTK100_stim_Oi, volc_iPI3Kd100_stim_Oi, volc_iNFkB100_stim_Oi)
titles_volcano <- c("aIg+H2O2 vs PBS (OCI-Ly8 w/ all inhibitors?)", "PBS ctrl vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", 
                    "0.1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "10 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "100 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", 
                    "0.1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "10 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "100 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ both stimuli)",
                    "0.1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "10 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "100 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ both stimuli)",
                    "0.1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "10 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)", "100 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ both stimuli)",
                    "Effect of 100 uM iSYK on aIg+H2O2 vs PBS (OCI-Ly8)", "Effect of 100 uM iBTK on aIg+H2O2 vs PBS (OCI-Ly8)", "Effect of 100 uM iPI3Kd on aIg+H2O2 vs PBS (OCI-Ly8)", "Effect of 100 uM iNFkB on aIg+H2O2 vs PBS (OCI-Ly8)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0))
  print(plot)
}
```

***Figure:** Volcano plots of DESeq2 model comparisons: aIg+H2O2 vs PBS, each inhibitor conc vs DMSO ctrl, and the effect of 100 uM inhibitor on aIg+H2O2 vs PBS (OCI-Ly8 cells only). Many proteins are activated upon aIg+H2O2 stimulation. Higher conc of inhibitor significantly affects more proteins, except iPI3Kd. And many proteins are significantly less activated in the presence of 100 uM inhibitor (again except iPI3Kd).*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_act_Ly8_int}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_stim_Oi", "heat_PBS_Oi", 
                        "heat_iSYK0.1_Oi", "heat_iSYK1_Oi", "heat_iSYK10_Oi", "heat_iSYK100_Oi", 
                        "heat_iBTK0.1_Oi", "heat_iBTK1_Oi", "heat_iBTK10_Oi", "heat_iBTK100_Oi", 
                        "heat_iPI3Kd0.1_Oi", "heat_iPI3Kd1_Oi", "heat_iPI3Kd10_Oi", "heat_iPI3Kd100_Oi", 
                        "heat_iNFkB0.1_Oi", "heat_iNFkB1_Oi", "heat_iNFkB10_Oi", "heat_iNFkB100_Oi", 
                        "heat_iSYK100_stim_Oi", "heat_iBTK100_stim_Oi", "heat_iPI3Kd100_stim_Oi", "heat_iNFkB100_stim_Oi")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Oi <- list(heat_iSYK100_Oi, heat_iBTK100_Oi, heat_iPI3Kd100_Oi, heat_iNFkB100_Oi) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()

# Combine all heatmap datasets of 100 uM inhibitor effect on aIg+H2O2 vs PBS (without DESeq2 analysis values)
heat_100_stim_Oi <- list(heat_iSYK100_stim_Oi, heat_iBTK100_stim_Oi, heat_iPI3Kd100_stim_Oi, heat_iNFkB100_stim_Oi) %>% 
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("stimulus", "inhib_new", "inhib_conc")]) %>%
  arrange(factor(stimulus, levels = stim_DS108), factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_act_Ly8_int_100, fig.width=21, fig.height=9}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Oi <- t(scale(t(assay(rld)[unique(c(heat_100uM_Oi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Oi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Oi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = "Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)"
)) +
  labs(title = "100 uM inhibitor vs DMSO ctrl (OCI-Ly8 w/ both stimuli) (significant proteins)")

# 100 uM inhibitor effect on aIg+H2O2 vs PBS
# Get the rld transformed data for the heatmap and scale per row
data_100_stim_Oi <- t(scale(t(assay(rld)[unique(c(heat_100_stim_Oi$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

as.ggplot(ComplexHeatmap::pheatmap(
  data_100_stim_Oi,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100_stim_Oi)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = "Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)"
)) +
  labs(title = "Effect of 100 uM inhibitor on aIg+H2O2 vs PBS (OCI-Ly8) (significant proteins)")
```

***Figure:** Heatmaps of the significant results in the 100 uM inhibitor vs DMSO ctrl comparisons and the effect of 100 uM inhibitor on aIg+H2O2 vs PBS comparisons (no threshold for fold change).*


#### Conc (aIg only)

DESeq2 analysis parameters:

-   All OCI-Ly8 samples: 4 inhibitors, 6 conc, aIg+H2O2 stimulus (74 samples)

-   Compensated counts (>= 1)

-   Model design: inhib_text_new (ref: DMSO ctrl)

    *Parameter options: cell_line + stimulus + inhibitor + inhib_conc + inhib_conc_text + description*

```{r DESeq_act_Ly8_stim}
# Prepare data to load into DESeq dataset
# cts: dataframe with proteins as row names, wells as column names, and count data as cell values
# coldata: dataframe with wells as row names and all metadata as columns

# Add a column with DMSO ctrl and PBS ctrl as inhibitors (instead of only under conc)
counts_dtbl <- counts_dtbl %>%
  mutate(inhib_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                               inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                               .default = inhibitor), 
         inhib_text_new = case_when(inhib_conc == "0 uM (PBS)" ~ "PBS ctrl", 
                                    inhib_conc == "0 uM (DMSO)" ~ "DMSO ctrl", 
                                    .default = inhib_conc_text))
inhib_new_DS108 <- c("DMSO ctrl", "PBS ctrl", "iSYK", "iBTK", "iPI3Kd", "iNFkB")

cts <- counts_dtbl  %>%
  filter(cell_line == "OCI-Ly8" & stimulus == "aIg+H2O2" & true_counts >= 1) %>%
  dplyr::select(target_nospace, plate_well, true_counts) %>%
  filter(!is.na(true_counts)) %>%
  spread(plate_well, true_counts) %>%
  replace(is.na(.), 0)

rownames(cts) <- cts$target_nospace
cts <- as.matrix(as.data.frame(cts[2:ncol(cts)]))

coldata <- data.frame(plate_well = colnames(cts)) %>%
  left_join(distinct(dplyr::select(counts_dtbl, c(plate, row, column, well, experiment, sample, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep, plate_well, inhib_new, inhib_text_new)))) %>%
  mutate(cell_line = factor(cell_line, levels = cells_DS108), 
         inhib_new = factor(inhib_new, levels = inhib_new_DS108), 
         inhib_conc = factor(inhib_conc, levels = inhib_conc_DS108), 
         inhib_conc_uM = as.character(inhib_conc_uM), 
         stimulus = factor(stimulus, levels = stim_DS108)) #If necessary, make factors of some metadata columns

rownames(coldata) <- coldata$plate_well

# Create DESeq object
# First define how the model is designed. Place most important parameter last
# modeldesign <- ~ inhibitor + inhib_conc + inhib_conc:inhibitor # works
modeldesign <- ~ inhib_text_new

# Then create the DESeq dataset
dds <- DESeqDataSetFromMatrix(countData = cts,
                              colData = coldata,
                              design = modeldesign)
# Define the reference samples for design parameter
# dds$inhibitor <- relevel(dds$inhibitor, ref = "iSYK")
# dds$inhib_conc <- relevel(dds$inhib_conc, ref = "0 uM (DMSO)")
# dds$stimulus <- relevel(dds$stimulus, ref = "PBS")
dds$inhib_text_new <- relevel(dds$inhib_text_new, ref = "DMSO ctrl")

# Run DESeq2: 
  # This function performs a default analysis through the steps:
  # Estimation of size factors: estimateSizeFactors
  # Estimation of dispersion: estimateDispersions
  # Negative Binomial GLM fitting and Wald statistics: nbinomWaldTest

# dds <- estimateSizeFactors(dds, controlGenes = ctrl_genes) # OPTIONAL: normalise data with specified control genes
dds <- DESeq(dds, test = "Wald", fitType = "local")
DESeq_comparisons <- resultsNames(dds) # lists the coefficients
DESeq_comparisons
```

**Log2FC results:** Significance threshold: padj < 0.05

```{r log2FC_act_Ly8_stim}
# Create results for all DESeq_comparisons
results_PBS_Os <- results(dds, name = "inhib_text_new_PBS.ctrl_vs_DMSO.ctrl", alpha = 0.05)

results_iSYK0.1_Os <- results(dds, name = "inhib_text_new_0.1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK1_Os <- results(dds, name = "inhib_text_new_1.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK10_Os <- results(dds, name = "inhib_text_new_10.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)
results_iSYK100_Os <- results(dds, name = "inhib_text_new_100.uM.iSYK_vs_DMSO.ctrl", alpha = 0.05)

results_iBTK0.1_Os <- results(dds, name = "inhib_text_new_0.1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK1_Os <- results(dds, name = "inhib_text_new_1.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK10_Os <- results(dds, name = "inhib_text_new_10.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)
results_iBTK100_Os <- results(dds, name = "inhib_text_new_100.uM.iBTK_vs_DMSO.ctrl", alpha = 0.05)

results_iPI3Kd0.1_Os <- results(dds, name = "inhib_text_new_0.1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd1_Os <- results(dds, name = "inhib_text_new_1.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd10_Os <- results(dds, name = "inhib_text_new_10.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)
results_iPI3Kd100_Os <- results(dds, name = "inhib_text_new_100.uM.iPI3Kd_vs_DMSO.ctrl", alpha = 0.05)

results_iNFkB0.1_Os <- results(dds, name = "inhib_text_new_0.1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB1_Os <- results(dds, name = "inhib_text_new_1.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB10_Os <- results(dds, name = "inhib_text_new_10.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)
results_iNFkB100_Os <- results(dds, name = "inhib_text_new_100.uM.iNFkB_vs_DMSO.ctrl", alpha = 0.05)

# results_CD53_aIg <- results(dds, contrast = c("description_cell_stim", "CD53_KO anti-Ig", "WT_ctrl anti-Ig"), alpha = 0.05)

list_results <- list(results_PBS_Os, 
                     results_iSYK0.1_Os, results_iSYK1_Os, results_iSYK10_Os, results_iSYK100_Os, 
                     results_iBTK0.1_Os, results_iBTK1_Os, results_iBTK10_Os, results_iBTK100_Os, 
                     results_iPI3Kd0.1_Os, results_iPI3Kd1_Os, results_iPI3Kd10_Os, results_iPI3Kd100_Os, 
                     results_iNFkB0.1_Os, results_iNFkB1_Os, results_iNFkB10_Os, results_iNFkB100_Os)

# Results summary
summary(results_iNFkB100_Os, padj = 0.05)
# sum(results_aIg$padj < 0.05, na.rm = TRUE)
```

```{r fig_log2FC_act_Ly8_stim, fig.width=7, fig.height=4}
par(mfrow = c(1,3), mar = c(4,4,2,1))
# par(mfrow = c(1,1))
signcutoff <- 0.05
plotMA(results_PBS_Os, alpha = signcutoff, main = "PBS vs DMSO")
plotMA(results_iNFkB100_Os, alpha = signcutoff, main = "100 uM iNFkB vs DMSO")
plotMA(results_iBTK100_Os, alpha = signcutoff, main = "100 uM iBTK vs DMSO")
```


##### Data transformation {.tabset}

```{r transformation_act_Ly8_stim}
# Normtransform (log2(n+1))
ntd <- normTransform(dds)

# Variance stabilising transform
vsd <- vst(dds, blind = FALSE, nsub = nrow(dds))

# Regularized log transform
rld <- rlog(dds, blind = FALSE)
```

###### PCA

```{r fig_PCA_act_Ly8_stim, fig.width=10, fig.height=4}
list_datatransform <- list(ntd, vsd, rld)
titles_datatransform <- c("ntd", "vsd", "rld")

for (data in 1:3){
  options(repr.plot.width = 10, repr.plot.height = 4)
  pcaData <- plotPCA(unlist(list_datatransform[[data]]), intgroup = c("stimulus", "cell_line", "inhibitor", "inhib_new", "inhib_conc", "inhib_text_new"), returnData = TRUE)
  percentVar <- round(100 * attr(pcaData, "percentVar"))
  
  plot <- ggplot(pcaData, aes(PC1, PC2, color = factor(inhib_conc, levels = inhib_conc_DS108), shape = factor(inhib_conc, levels = inhib_conc_DS108))) +
    geom_point(size = 2) +
    xlab(paste0("PC1: ", percentVar[1], "% variance")) +
    ylab(paste0("PC2: ", percentVar[2], "% variance")) + 
    coord_fixed() +
    facet_wrap(~inhibitor) +
    # scale_fill_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    scale_color_manual(values = colors_blue9[c(2, 3, 5, 6, 8, 9)], name = "Inhib conc") +
    # scale_color_manual(values = colors_stim, name = "Stimulus") +
    # scale_fill_manual(values = colors_stim, name = "Stimulus") +
    scale_shape_manual(values = c(16, 17, 10, 6, 7, 8), name = "Inhib conc") +
    ggtitle(titles_datatransform[data]) +
    theme_bw() +
    textsize_small
  
  print(plot)
}
```

***Figure:** PCA of all samples included in the DESeq2 analysis. PC1 is mainly the difference between inhibitor conc, and PC2 is mainly the difference between 100 uM iNFkB and all other samples.*

###### Variance

```{r fig_sd_act_Ly8_stim, fig.width=6, fig.height=3}
options(repr.plot.width = 6, repr.plot.height = 3)

msd_ntd <- meanSdPlot(assay(ntd))
plot_ntd <- msd_ntd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Normtransform") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_vsd <- meanSdPlot(assay(vsd))
plot_vsd <- msd_vsd$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Variance stabilising") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

msd_rld <- meanSdPlot(assay(rld))
plot_rld <- msd_rld$gg + 
  labs(x = "Ranked mean", y = "Standard deviation", title = "Regularized log") + 
  theme_bw() + theme(legend.position = "none") + textsize_small

plot_grid(plot_ntd, plot_vsd, plot_rld, ncol = 3)
```

###### Sample-to-sample distances

```{r fig_distances_act_Ly8_stim, fig.width=10, fig.height=15}
options(repr.plot.width = 10, repr.plot.height = 15)

for (data in 1:3){
  sampleDists <- dist(t(assay(unlist(list_datatransform[[data]]))))
  sampleDistMatrix <- as.matrix(sampleDists)
  # rownames(sampleDistMatrix) <- paste(list_datatransform[[data]]$inhib_conc_text,
  #                                     list_datatransform[[data]]$replicate,
  #                                     sep = " - ")
  rownames(sampleDistMatrix) <- list_datatransform[[data]]$description
  colnames(sampleDistMatrix) <- NULL
  colors <- colorRampPalette(rev(brewer.pal(9, "Blues")))(255)
  
  plot <- pheatmap(sampleDistMatrix,
                   clustering_distance_rows = sampleDists,
                   clustering_distance_cols = sampleDists,
                   col = colors, 
                   main = titles_datatransform[data]) 
  
  print(plot)
}
```


##### Volcano

Highlight the significantly changing proteins (no cutoff for fold change):

-   padj < 0.05

```{r prep_volcano_act_Ly8_stim}
# Function to prepare DESeq2 results for volcano plots
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
prep_forvulcano <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5)){ 
  dataset$proteins <- rownames(dataset)
  dataset$target_nospace <- rownames(dataset)
  
  # Add a column for differential expression
  dataset$diff_express <- "NO"
  dataset$diff_express[dataset$log2FoldChange > foldchange_filter & dataset$padj < padj_filter] <- "UP"
  dataset$diff_express[dataset$log2FoldChange < -foldchange_filter & dataset$padj < padj_filter] <- "DOWN"
  
  # Add a column for labeling
  dataset$delabel <- NA
  dataset$delabel[dataset$diff_express != "NO"] <- dataset$proteins[dataset$diff_express != "NO"]
  
  # Add Ab metadata
  dataset <- left_join(as.data.frame(dataset), meta_Abs)
  
  return(dataset)
}

# Apply function to each dataset
names_results_volc <- c("volc_PBS_Os", 
                        "volc_iSYK0.1_Os", "volc_iSYK1_Os", "volc_iSYK10_Os", "volc_iSYK100_Os", 
                        "volc_iBTK0.1_Os", "volc_iBTK1_Os", "volc_iBTK10_Os", "volc_iBTK100_Os", 
                        "volc_iPI3Kd0.1_Os", "volc_iPI3Kd1_Os", "volc_iPI3Kd10_Os", "volc_iPI3Kd100_Os", 
                        "volc_iNFkB0.1_Os", "volc_iNFkB1_Os", "volc_iNFkB10_Os", "volc_iNFkB100_Os")


for(dataset in c(1:length(list_results))){
  assign(names_results_volc[dataset], prep_forvulcano(dataset = list_results[[dataset]], foldchange_filter = 0))
}
```

```{r fig_volcano_act_Ly8_stim, fig.width=8, fig.height=6.5}
# Function to create volcano plots
# Standard log2FC threshold = log2(1.5); standard padj threshold = 0.05
vulcanoplot_results <- function(data = data, 
                                title_plot = "", 
                                xintercepts = c(-log2(1.5), log2(1.5)), 
                                yintercept = -log10(0.05), 
                                line_colors = c("red"), 
                                point_colors = c(UP = "navy", NO = "black", DOWN = "firebrick")){
  
  # Create a plot (ggplot2)
  ggplot(data = as.data.frame(data), aes(x = log2FoldChange, y = -log10(padj), col = diff_express, label = delabel)) +
    geom_point() + 
    theme_minimal() +
    geom_text_repel(size = 2.5, 
                    segment.size = 0.2,
                    hjust = 0,
                    direction = "y",
                    nudge_y = -log10(0.005),
                    nudge_x = ifelse(data$log2FoldChange < 0, -0.4 - data$log2FoldChange, 0.4 - data$log2FoldChange)
                    ) +
    scale_color_manual(values = point_colors) +
    geom_vline(xintercept = xintercepts, col = line_colors) +
    geom_hline(yintercept = yintercept, col = line_colors) +
    ggtitle(title_plot) +
    theme(axis.text.x = element_text(colour = 'black', size = 9),
          axis.text.y = element_text(colour = 'black', size = 9),
          text = element_text(size = 9),
          axis.text = element_text(size = 9),
          plot.title = element_text(size = 11))
}

# Apply function to each dataset
list_results_volc <- list(volc_PBS_Os, 
                          volc_iSYK0.1_Os, volc_iSYK1_Os, volc_iSYK10_Os, volc_iSYK100_Os, 
                          volc_iBTK0.1_Os, volc_iBTK1_Os, volc_iBTK10_Os, volc_iBTK100_Os, 
                          volc_iPI3Kd0.1_Os, volc_iPI3Kd1_Os, volc_iPI3Kd10_Os, volc_iPI3Kd100_Os, 
                          volc_iNFkB0.1_Os, volc_iNFkB1_Os, volc_iNFkB10_Os, volc_iNFkB100_Os)
titles_volcano <- c("PBS ctrl vs DMSO ctrl (OCI-Ly8 w/ stim)",
                    "0.1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)", "1 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)", "10 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)", "100 uM iSYK vs DMSO ctrl (OCI-Ly8 w/ stim)",
                    "0.1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)", "1 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)", "10 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)", "100 uM iBTK vs DMSO ctrl (OCI-Ly8 w/ stim)",
                    "0.1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)", "1 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)", "10 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)", "100 uM iPI3Kd vs DMSO ctrl (OCI-Ly8 w/ stim)",
                    "0.1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)", "1 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)", "10 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)", "100 uM iNFkB vs DMSO ctrl (OCI-Ly8 w/ stim)")

for(dataset in c(1:length(list_results_volc))){
  plot <- vulcanoplot_results(data = list_results_volc[[dataset]], 
                              title_plot = titles_volcano[dataset], 
                              xintercepts = c(0)) + 
    coord_cartesian(xlim = c(-1.25, 1.25), ylim = c(0, 125))
  
  print(plot)
}
```

***Figure:** Volcano plots of DESeq2 model comparisons: each inhibitor conc vs DMSO ctrl (OCI-Ly8 w/ aIg+H2O2 samples only). Already at low conc inhibitor, several proteins are decreased compared to the DMSO ctrl, and in general the number op proteins, effect size and significance increase with the inhibitor conc, except iPI3Kd.*


##### Heatmap

We continue with the regularised log transformed results (see Data transform section). We create heatmaps of significant changes:

-   padj < 0.05

```{r prep_heatmap_act_Ly8_stim}
# Function to filter the data
# Standard log2FC filter = log2(1.5); standard padj filter = 0.05
filter_heatmap <- function(dataset = dataset, padj_filter = 0.05, foldchange_filter = log2(1.5), select_top = TRUE, select_n = 10) {
  data <- as.data.frame(dataset) %>%
    filter(padj <= padj_filter & abs(log2FoldChange) >= foldchange_filter) %>%
    arrange(-log2FoldChange) 
  rownames(data) <- data$target_nospace
  
  # Select top 10 proteins with most increased and decreased expression
  if (select_top == TRUE) {
    data_top <- data %>%
      slice_max(order_by = abs(log2FoldChange), n = select_n)
    return(data_top)
  } else {
    return(data)
  }
}

# Apply function to each dataset
names_results_heat <- c("heat_PBS_Os", 
                        "heat_iSYK0.1_Os", "heat_iSYK1_Os", "heat_iSYK10_Os", "heat_iSYK100_Os", 
                        "heat_iBTK0.1_Os", "heat_iBTK1_Os", "heat_iBTK10_Os", "heat_iBTK100_Os", 
                        "heat_iPI3Kd0.1_Os", "heat_iPI3Kd1_Os", "heat_iPI3Kd10_Os", "heat_iPI3Kd100_Os", 
                        "heat_iNFkB0.1_Os", "heat_iNFkB1_Os", "heat_iNFkB10_Os", "heat_iNFkB100_Os")

for(dataset in c(1:length(list_results_volc))){
  assign(names_results_heat[[dataset]], filter_heatmap(dataset = list_results_volc[[dataset]], select_top = FALSE, foldchange_filter = 0))
  dataset
}

# heat_results_all <- filter_heatmap(dataset = volc_cell, padj_filter = 1, foldchange_filter = 0, select_top = FALSE)

# Combine all heatmap datasets of 100 uM inhibitor vs DMSO ctrl (without DESeq2 analysis values)
heat_100uM_Os <- list(heat_iSYK100_Os, heat_iBTK100_Os, heat_iPI3Kd100_Os, heat_iNFkB100_Os) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()

# Subset the full dataframe to get metadata for heatmap plotting
data_heatmap <- as.data.frame(colData(dds)[, c("inhib_new", "inhib_conc")]) %>%
  arrange(factor(inhib_new, levels = inhib_new_DS108), factor(inhib_conc, levels = inhib_conc_DS108))
```

```{r fig_heatmap_act_Ly8_stim, fig.width=13, fig.height=10}
# , fig.width=10, fig.height=10
# options(repr.plot.width = 10, repr.plot.height = 10)

ann_colors_man <- list(inhib_new = c("PBS ctrl" = "black", "DMSO ctrl" = "grey", colors_inhib),
                       # stimulus = colors_stim, 
                       inhib_conc = setNames(colors_blue9[2:(length(unique(data_heatmap$inhib_conc))+1)], unique(data_heatmap$inhib_conc))
)

# 100 uM inhibitor vs DMSO ctrl (w/ stim)
# Get the rld transformed data for the heatmap and scale per row
data_100uM_Os <- t(scale(t(assay(rld)[unique(c(heat_100uM_Os$target_nospace)), rownames(data_heatmap)])))
# sort_hclust <- function(...) as.hclust(dendsort(as.dendrogram(...)))
# data_cluster_rows <- sort_hclust(hclust(dist(data_cell)))
# plot(data_cluster_rows, main = "Unsorted Dendrogram", xlab = "", sub = "")

as.ggplot(ComplexHeatmap::pheatmap(
  data_100uM_Os,
  cluster_rows = TRUE,
  show_rownames = TRUE,
  show_colnames = FALSE,
  cluster_cols = FALSE,
  # annotation_row = heat_cell[, c("modification", "marker_pathway")],
  # annotation_row = heat_cell[, "marker_pathway", drop = FALSE],
  annotation_col = data_heatmap,
  annotation_colors = ann_colors_man,
  # scale = "row",
  colorRampPalette(c("navy", "white", "firebrick3"))(50),
  column_split = data_heatmap$inhib_new,
  row_split = subset(heat_100uM_Os)$modification,
  cellwidth = 8,
  cellheight = 8,
  fontsize = 8,
  border_color = NA,
  # main = "Significant proteins with 100 uM inhibitor (OCI-Ly8 cells w/o stim)"
)) +
  labs(title = "100 uM inhibitor vs DMSO ctrl (OCI-Ly8 w/ stim) (significant proteins)")
```

***Figure:** Heatmap of the significant results in the 100 uM inhibitor vs DMSO ctrl comparisons (no threshold for fold change).*



### OCI-Ly8: Protein comparisons

Finally, we visualised the normalised true counts of all proteins for which the aIg+H2O2 vs PBS comparison was significantly affected by 100 uM inhibitor (27 in total).

```{r sign_data_act_Ly8_prot}
# Create datasets with all relevant information on significantly changing results

# Datasets generated during analysis
# heat_xxx: DESeq2 data of sign prot (repl combined), log2FC and pvalues per protein
# data_xxx: matrix of data for heatmap, well (col) x prot (row) (rld transformed, scaled per row)
# counts_dtbl: all input info (counts, compensation, normalisation, manual fold changes), tidy format, organised per well + prot
# mean_dtbl: same as counts_dtbl but with mean values of repl

# List of all significantly changing proteins
sign_prot_Oi <- sort(unique(c(heat_100_stim_Oi$target_nospace)))

heat_iSYK_Os <- list(heat_iSYK0.1_Os, heat_iSYK1_Os, heat_iSYK10_Os, heat_iSYK100_Os) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iSYK_Os <- sort(unique(c(heat_iSYK_Os$target_nospace)))

heat_iBTK_Os <- list(heat_iBTK0.1_Os, heat_iBTK1_Os, heat_iBTK10_Os, heat_iBTK100_Os) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iBTK_Os <- sort(unique(c(heat_iBTK_Os$target_nospace)))

heat_iPI3Kd_Os <- list(heat_iPI3Kd0.1_Os, heat_iPI3Kd1_Os, heat_iPI3Kd10_Os, heat_iPI3Kd100_Os) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iPI3Kd_Os <- sort(unique(c(heat_iPI3Kd_Os$target_nospace)))

heat_iNFkB_Os <- list(heat_iNFkB0.1_Os, heat_iNFkB1_Os, heat_iNFkB10_Os, heat_iNFkB100_Os) %>%
  purrr::reduce(full_join) %>%
  dplyr::select(-c(baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express)) %>%
  filter(target_nospace != "BIM") %>%
  distinct()
sign_prot_iNFkB_Os <- sort(unique(c(heat_iNFkB_Os$target_nospace)))

sign_prot_inhib_Os <- sort(unique(c(sign_prot_iSYK_Os, sign_prot_iBTK_Os, sign_prot_iPI3Kd_Os, sign_prot_iNFkB_Os)))
sign_prot_all_O <- sort(unique(c(sign_prot_Oi, sign_prot_inhib_Os)))

# print(c("Significant proteins (OCI-ly8 vs OCI-Ly8 basal signaling):", sign_prot_cell))
```

```{r sign_data_act_Ly8_sign}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis
# Effect of 100 uM inhibitor on stimulation
heat_list_Oi <- list(heat_iSYK100_stim_Oi, heat_iBTK100_stim_Oi, heat_iPI3Kd100_stim_Oi, heat_iNFkB100_stim_Oi)
heat_names <- c("sign_data_iSYK", "sign_data_iBTK", "sign_data_iPI3Kd", "sign_data_iNFkB")
heat_suffix <- c("_iSYK", "_iBTK", "_iPI3Kd", "_iNFkB")
for (heat_data in c(1:length(heat_list_Oi))) {
  # print(heat_data)
  alt_data <- heat_list_Oi[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_Oi) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_Oi <- list(sign_data_iSYK, sign_data_iBTK, sign_data_iPI3Kd, sign_data_iNFkB) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iSYK conc on activated signaling
heat_list_iSYK_Os <- list(heat_iSYK0.1_Os, heat_iSYK1_Os, heat_iSYK10_Os, heat_iSYK100_Os)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iSYK_Os))) {
  # print(heat_data)
  alt_data <- heat_list_iSYK_Os[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iSYK_Os) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iSYK_Os <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iBTK conc on activated signaling
heat_list_iBTK_Os <- list(heat_iBTK0.1_Os, heat_iBTK1_Os, heat_iBTK10_Os, heat_iBTK100_Os)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iBTK_Os))) {
  # print(heat_data)
  alt_data <- heat_list_iBTK_Os[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iBTK_Os) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iBTK_Os <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iPI3Kd conc on activated signaling
heat_list_iPI3Kd_Os <- list(heat_iPI3Kd0.1_Os, heat_iPI3Kd1_Os, heat_iPI3Kd10_Os, heat_iPI3Kd100_Os)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iPI3Kd_Os))) {
  # print(heat_data)
  alt_data <- heat_list_iPI3Kd_Os[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iPI3Kd_Os) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iPI3Kd_Os <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)

# Effect of iNFkB conc on activated signaling
heat_list_iNFkB_Os <- list(heat_iNFkB0.1_Os, heat_iNFkB1_Os, heat_iNFkB10_Os, heat_iNFkB100_Os)
heat_names <- c("sign_data_0.1", "sign_data_1", "sign_data_10", "sign_data_100")
heat_suffix <- c("_0.1", "_1", "_10", "_100")
for (heat_data in c(1:length(heat_list_iNFkB_Os))) {
  # print(heat_data)
  alt_data <- heat_list_iNFkB_Os[[heat_data]] %>%
    filter(target_nospace %in% sign_prot_iNFkB_Os) %>%
    dplyr::select(target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, isotype, barcode_name, DS_Ab_nr, 
           baseMean, log2FoldChange, lfcSE, stat, pvalue, padj, diff_express) %>% # reorder cols to more logical order
    dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                  marker_path_detail = marker_pathway_detail, 
                  Ab_BC_nr = barcode_name, 
                  base_mean = baseMean, 
                  log2FC = log2FoldChange,
                  log2FC_se = lfcSE) %>%
  dplyr::rename_with(~paste0(., heat_suffix[heat_data]), c("log2FC", "log2FC_se", "stat", "pvalue", "padj", "diff_express"))
  assign(heat_names[heat_data], alt_data)
  heat_data
}
clean_data_sign_iNFkB_Os <- list(sign_data_0.1, sign_data_1, sign_data_10, sign_data_100) %>%
  purrr::reduce(full_join) %>%
  arrange(target_nospace)
```

```{r sign_data_act_Ly8_well}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset of the heatmap data matrix + combine with counts_dtbl
# Effect of 100 uM inhibitor on stimulation
clean_data_well_Oi <- data.frame(data_100_stim_Oi) %>% 
  mutate(target_nospace = rownames(data_100_stim_Oi)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_value", 2:c(ncol(data_100_stim_Oi)+1)) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_Oi) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_value) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo, 
                heatmap = heatmap_value)

# Effect of inhibitor conc on activated signaling
data_iSYK_Os <- t(scale(t(assay(rld)[unique(c(sign_prot_iSYK_Os)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iSYK")))])))
alt_data_iSYK_Os <- data.frame(data_iSYK_Os) %>% 
  mutate(target_nospace = rownames(data_iSYK_Os)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iSYK", 2:c(ncol(data_iSYK_Os)+1))
data_iBTK_Os <- t(scale(t(assay(rld)[unique(c(sign_prot_iBTK_Os)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iBTK")))])))
alt_data_iBTK_Os <- data.frame(data_iBTK_Os) %>% 
  mutate(target_nospace = rownames(data_iBTK_Os)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iBTK", 2:c(ncol(data_iBTK_Os)+1))
data_iPI3Kd_Os <- t(scale(t(assay(rld)[unique(c(sign_prot_iPI3Kd_Os)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iPI3Kd")))])))
alt_data_iPI3Kd_Os <- data.frame(data_iPI3Kd_Os) %>% 
  mutate(target_nospace = rownames(data_iPI3Kd_Os)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iPI3Kd", 2:c(ncol(data_iPI3Kd_Os)+1))
data_iNFkB_Os <- t(scale(t(assay(rld)[unique(c(sign_prot_iNFkB_Os)), rownames(filter(data_heatmap, inhib_new %in% c("PBS ctrl", "DMSO ctrl", "iNFkB")))])))
alt_data_iNFkB_Os <- data.frame(data_iNFkB_Os) %>% 
  mutate(target_nospace = rownames(data_iNFkB_Os)) %>%
  dplyr::select(target_nospace, everything()) %>%
  gather("plate_well", "heatmap_iNFkB", 2:c(ncol(data_iNFkB_Os)+1))

clean_data_well_inhib_Os <- list(alt_data_iSYK_Os, alt_data_iBTK_Os, alt_data_iPI3Kd_Os, alt_data_iNFkB_Os) %>% 
  purrr::reduce(full_join) %>%
  left_join(counts_dtbl) %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_inhib_Os) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm, heatmap_iSYK, heatmap_iBTK, heatmap_iPI3Kd, heatmap_iNFkB) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)

# All significant proteins for plotting
clean_data_well_O <- counts_dtbl %>%
  arrange(plate_well, target_nospace) %>%
  filter(target_nospace %in% sign_prot_Oi) %>%
  dplyr::select(plate_well, plate, well, experiment, cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, replicate, description, descript_rep,
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         true_counts, true_scaling_factor_geo, true_counts_norm_geo, counts_norm_0, man_FC_norm, man_log2FC_norm) %>% # reorder columns for a more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                counts = true_counts, 
                scaling_factor = true_scaling_factor_geo, 
                counts_norm = true_counts_norm_geo)
```

```{r sign_data_act_Ly8_cond}
# Create datasets with all relevant information on significantly changing results

# Make a tidy dataset with mean values per condition (replicates combined)
# Effect of 100 uM inhibitor on stimulation
clean_data_mean_Oi <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_Oi & cell_line == "OCI-Ly8") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# Effect of inhibitor conc on activated signaling
clean_data_mean_Os <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_inhib_Os & cell_line == "OCI-Ly8") %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)

# All significant proteins for plotting
clean_data_mean_O <- mean_dtbl %>%
  filter(target_nospace %in% sign_prot_Oi) %>%
  dplyr::select(cell_line, stimulus, inhibitor, inhib_conc_uM, inhib_conc, inhib_conc_text, description, 
         target_nospace, target, modification, marker_cell, marker_pathway, marker_pathway_detail, barcode_name, DS_Ab_nr, 
         counts, counts_sd, counts_norm, counts_norm_sd, FC_norm, FC_norm_sd, log2FC_norm, log2FC_norm_sd) %>% # reorder cols to more logical order
  dplyr::rename(marker_path = marker_pathway, # rename some columns for clarity
                marker_path_detail = marker_pathway_detail, 
                Ab_BC_nr = barcode_name, 
                man_FC_norm = FC_norm, 
                man_FC_norm_sd = FC_norm_sd, 
                man_log2FC_norm = log2FC_norm, 
                man_log2FC_norm_sd = log2FC_norm_sd)
```

```{r fig_counts_act_Ly8_all, fig.width=15, fig.height=18}
plot <- ggplot(clean_data_well_Oi, 
               aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                   counts_norm, 
                   color = stimulus, 
                   shape = inhibitor)
) +
  geom_jitter(alpha = 1, size = 1.5, width = 0.1) +
  facet_wrap(vars(factor(target)), ncol = 5) +
  theme_bw() +
  # scale_y_log10() +
  scale_color_manual(values = colors_stim, name = "Stimulus") + 
  labs(x = "", y = "Normalized counts") +
  textsize_small

print(plot)
```

***Figure:** Normalised counts for all significantly changing proteins. For some proteins the effect size is quite small.*

```{r fig_counts_act_Ly8, fig.width=12, fig.height=5}
options(repr.plot.width = 5, repr.plot.height = 5)

# Re-redesigned fig
filt_clean_data_mean_O <- clean_data_mean_O %>%
  dplyr::group_by(cell_line, inhibitor, inhib_conc, stimulus, target_nospace) %>%
  summarise(mean_counts_norm_sd = sd(counts_norm_sd, na.rm = T), 
            mean_counts_norm = mean(counts_norm, na.rm = T))
  
for(protein in sign_prot_Oi) {
  plot <- ggplot() +
    geom_jitter(data = subset(clean_data_well_O, target_nospace == protein), 
                aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label), 
                    counts_norm, 
                    color = stimulus, 
                    # shape = cell_line
                    ), 
                alpha = 1, size = 1.25, width = 0.1) +
    geom_errorbar(data = subset(filt_clean_data_mean_O, target_nospace == protein),
                  aes(factor(inhib_conc, levels = inhib_conc_DS108, labels = inhib_conc_label),
                      ymin = mean_counts_norm,
                      ymax = mean_counts_norm,
                      group = stimulus,
                      color = stimulus
                      ),
                  width = 0.5, size = 0.75) + #, color = "black"
    # facet_wrap(vars(factor(target_nospace))) +
    # facet_wrap(vars(factor(inhibitor, levels = inhib_DS108)), ncol = 1) +
    facet_grid(vars(factor(cell_line, levels = cells_DS108)), vars(factor(inhibitor, levels = inhib_DS108))) +
    # scale_y_log10() +
    scale_color_manual(values = colors_stim, name = "Stimulus") + 
    labs(x = "", y = "Normalized counts", title = protein) +
    theme_bw() +
    theme(legend.position = "none", panel.grid.major.x = element_blank()) +
    textsize_small
  
  print(plot)
}
```


### OCI-Ly8: Save data

Save all information and datasets as .csv file

Effect of 100 uM inhibitor on aIg+H2O2 vs PBS comparison (OCI-Ly8 cells only):

-   Ly8_100uM_stim_sign_proteins.txt: List of significantly affected proteins by 100 uM inhibitor in the aIg+H2O2 vs PBS comparison (by at least one inhibitor)

-   Ly8_100uM_stim_sign_DESeq2data.csv: Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis

-   Ly8_100uM_stim_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   Ly8_100uM_stim_sign_condition.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins


Effect of inhibitor conc vs DMSO ctrl (for each inhibitor, OCI-Ly8 w/ stim only):
For each inhibitor (iSYK, iBTK, iPI3Kd, iNFkB):

-   Ly8_inhib_sign_proteins.txt: List of significantly different proteins in any of the inhibitor conc vs DMSO ctrl comparisons

-   Ly8_inhib_sign_DESeq2data.csv: Dataset with significantly different proteins, and log2FC + pvalues from DESeq2 analysis

-   Ly8_inhib_sign_sample.csv: Dataset of counts per sample per protein + heatmap data for significantly changing proteins

-   Ly8_inhib_sign_condition.csv: Dataset of counts per condition (mean of repl) per protein for significantly changing proteins

```{r save_act_Ly8}
# List of significantly changing proteins: sign_prot_cell
write(sign_prot_Oi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_proteins.txt")
write(sign_prot_iSYK_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iSYK_sign_proteins.txt")
write(sign_prot_iBTK_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iBTK_sign_proteins.txt")
write(sign_prot_iPI3Kd_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iPI3Kd_sign_proteins.txt")
write(sign_prot_iNFkB_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iNFkB_sign_proteins.txt")

# Dataset with significantly changing proteins, and log2FC + pvalues from DESeq2 analysis: clean_data_sign
write.csv(clean_data_sign_Oi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iSYK_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iSYK_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iBTK_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iBTK_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iPI3Kd_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iPI3Kd_sign_DESeq2data.csv", row.names = F)
write.csv(clean_data_sign_iNFkB_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_iNFkB_sign_DESeq2data.csv", row.names = F)

# Dataset of counts per protein per sample + heatmap data for significantly changing proteins: clean_data_well
write.csv(clean_data_well_Oi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_sample.csv", row.names = F)
write.csv(clean_data_well_inhib_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_inhib_sign_sample.csv", row.names = F)

# Dataset of counts per protein per condition (mean of repl) for significantly changing proteins: clean_data_mean
write.csv(clean_data_mean_Oi, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_100uM_stim_sign_condition.csv", row.names = F)
write.csv(clean_data_mean_Os, file = "output/DS108_StimInhibIDseq/IDseq_sign/active_inhibitor/Ly8_inhib_sign_condition.csv", row.names = F)
```

```{r clear_act_Ly8}
# Remove unnecessary files to clear up memory
rm(list = ls(pattern = "results_"))
rm(list = ls(pattern = "volc_"))
rm(list = ls(pattern = "heat_"))
rm(list = ls(pattern = "list_"))
rm(list = ls(pattern = "names_"))
rm(list = ls(pattern = "titles_"))
rm(list = ls(pattern = "clean_"))
rm(list = ls(pattern = "sign_"))
gc()
```


